{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import defaultdict\n",
    "import os, sys\n",
    "\n",
    "mx_base = \"/mnt/extproj/projekte/textmining/mxplore/\"\n",
    "model_base = mx_base + \"spacy_models/\"\n",
    "mirexplore_base = mx_base + \"/python/textmine/\"\n",
    "scai_base = mirexplore_base + \"/scai_eval/\"\n",
    "\n",
    "if not mirexplore_base in sys.path:\n",
    "    sys.path.insert(0, mirexplore_base)\n",
    "    \n",
    "\n",
    "import scispacy\n",
    "import spacy\n",
    "from spacy import displacy\n",
    "\n",
    "#nlp = spacy.load('/mnt/d/spacy/models/en_core_sci_lg-0.2.4/en_core_sci_lg/en_core_sci_lg-0.2.4/')\n",
    "\n",
    "from MirGeneRelCheck import SentenceRelationChecker, SentenceRelationClassifier\n",
    "from mxutils.tmutils import normalize_gene_names\n",
    "from collections import Counter\n",
    "from lxml import etree\n",
    "from sklearn.metrics import multilabel_confusion_matrix, classification_report\n",
    "from itertools import chain, combinations\n",
    "\n",
    "\n",
    "normGeneSymbols= normalize_gene_names(path=scai_base + \"/hgnc_no_withdrawn.syn\")\n",
    "relationsFile = mirexplore_base + \"allrels.csv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "nlp_weblg = spacy.load(model_base+\"/en_core_web_lg-2.2.0/en_core_web_lg/en_core_web_lg-2.2.0/\")\n",
    "nlp = spacy.load(model_base+\"/en_core_sci_lg-0.2.4/en_core_sci_lg/en_core_sci_lg-0.2.4/\")\n",
    "nlp_ent = spacy.load(model_base+\"/en_ner_bionlp13cg_md-0.2.4/en_ner_bionlp13cg_md/en_ner_bionlp13cg_md-0.2.4\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "import MirGeneRelCheck"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BANCR {'start': 25, 'end': 30, 'label': 'GENE_OR_GENE_PRODUCT'}\n",
      "miR-204-3p {'start': 35, 'end': 45, 'label': 'GENE_OR_GENE_PRODUCT'}\n",
      "extrathyroidal {'start': 102, 'end': 116, 'label': 'MULTI-TISSUE_STRUCTURE'}\n",
      "PTC {'start': 152, 'end': 155, 'label': 'CANCER'}\n"
     ]
    }
   ],
   "source": [
    "doc = nlp_ent(\"Therefore, it seems that BANCR and miR-204-3p might act in a cumulative way on the development of the extrathyroidal invasion of the BRAFV600E-negative PTC.\")\n",
    "\n",
    "ents = doc.to_json()[\"ents\"]\n",
    "\n",
    "for ent in ents:\n",
    "    print(doc.text[ent[\"start\"]:ent[\"end\"]], ent)\n",
    "\n",
    "#doc.to_json()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def analyse(testCase, dist=80):\n",
    "    relCheck = MirGeneRelCheck.MirGeneRelCheck()\n",
    "\n",
    "    doc = nlp(testCase[0])\n",
    "    \n",
    "    #print(\"conjuncts\", doc.conjuncts)\n",
    "    \n",
    "    lWord = doc[testCase[1]]\n",
    "    rWord = doc[testCase[2]]\n",
    "\n",
    "    for t in doc:\n",
    "        print(t.idx, t.i, t, t.pos_, t.dep_, t.head, t.head.idx, t.conjuncts)\n",
    "\n",
    "    print(lWord, rWord)\n",
    "\n",
    "    if not \"mir\" in str(lWord).lower() and not \"let\" in str(lWord).lower():\n",
    "        tmp = lWord\n",
    "        lWord = rWord\n",
    "        rWord = tmp\n",
    "        \n",
    "    print(lWord, rWord)\n",
    "\n",
    "    relClassifier = SentenceRelationClassifier(relationsFile)    \n",
    "    checkResults = relCheck.checkRelation(doc, lWord, rWord, verbose=True, relClassifier=relClassifier.classify)\n",
    "    print(checkResults)\n",
    "    \n",
    "    displacy.render(doc, style=\"dep\", options={\"compact\": True, \"distance\": dist})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 0 Among ADP prep promoted 24 ()\n",
      "6 1 them PRON pobj Among 0 ()\n",
      "10 2 , PUNCT punct promoted 24 ()\n",
      "12 3 circ_002178 NOUN nsubj promoted 24 ()\n",
      "24 4 promoted VERB ROOT promoted 24 ()\n",
      "33 5 PD-L1 NOUN compound expression 39 ()\n",
      "39 6 expression NOUN dobj promoted 24 ()\n",
      "50 7 via ADP mark sponging 54 ()\n",
      "54 8 sponging ADJ advcl promoted 24 ()\n",
      "63 9 miR-34a NOUN dobj sponging 54 ()\n",
      "70 10 . PUNCT punct promoted 24 ()\n",
      "PD-L1 miR-34a\n",
      "miR-34a PD-L1\n",
      "Conjunctions ( conj )\n",
      "Conjunctions ( nmod )\n",
      "SDP sent: Among them, circ_002178 promoted PD-L1 expression via sponging miR-34a.\n",
      "SDP     : [(PD-L1, 'NOUN', 'compound', False), (expression, 'NOUN', 'dobj', False), (promoted, 'VERB', 'ROOT', False), (sponging, 'ADJ', 'advcl', False), (miR-34a, 'NOUN', 'dobj', False)]\n",
      "SDP DOBJ RULE\n",
      "0 [Among, them, ,, circ_002178, promoted, PD-L1, expression, via, sponging, miR-34a] 10 0 Among them, circ_002178 promoted PD-L1 expression via sponging miR-34a.\n",
      "0 [Among, them, ,, circ_002178, promoted, PD-L1, expression, via, sponging, miR-34a] 10 0 Among them, circ_002178 promoted PD-L1 expression via sponging miR-34a.\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "Compartment Text Among them , circ_002178 promoted PD-L1 expression via sponging miR-34a\n",
      "[Among, them, ,, circ_002178, promoted, PD-L1, expression, via, sponging, miR-34a]\n",
      "TextBetween: Among them , circ_002178 promoted PD-L1 expression via sponging miR-34a\n",
      "infer direction from token set [Among, them, ,, circ_002178, promoted, PD-L1, expression, via, sponging, miR-34a]\n",
      "stemTypes2Count: Counter({'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c620>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'POS': 1, 'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe3b0d0>, {'gene': Counter({'POS': 1, 'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(promoted, 'POS'), (expression, 'NEU')]\n",
      "counts between 3 Counter() POS\n",
      "Between Tokens []\n",
      "counts between final False\n",
      "['circ_002178', 'promoted', 'PD-L1', 'expression', 'via', 'sponging', 'miR-34a']\n",
      "Counter()\n",
      "[]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Setting relfile /mnt/extproj/projekte/textmining/miRExplore/python/allrels.csv\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<svg xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" xml:lang=\"en\" id=\"6d3f105e85ae4a938ae7a12a1032b491-0\" class=\"displacy\" width=\"770\" height=\"297.0\" direction=\"ltr\" style=\"max-width: none; height: 297.0px; color: #000000; background: #ffffff; font-family: Arial; direction: ltr\">\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"50\">Among</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"50\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"130\">them,</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"130\">PRON</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"210\">circ_002178</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"210\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"290\">promoted</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"290\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"370\">PD-L1</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"370\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"450\">expression</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"450\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"530\">via</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"530\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"610\">sponging</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"610\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"690\">miR-34a.</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"690\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6d3f105e85ae4a938ae7a12a1032b491-0-0\" stroke-width=\"2px\" d=\"M62,162.0 62,122.0 287.0,122.0 287.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6d3f105e85ae4a938ae7a12a1032b491-0-0\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">prep</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M62,164.0 L58,156.0 66,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6d3f105e85ae4a938ae7a12a1032b491-0-1\" stroke-width=\"2px\" d=\"M62,162.0 62,148.66666666666666 121.0,148.66666666666666 121.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6d3f105e85ae4a938ae7a12a1032b491-0-1\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">pobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M121.0,164.0 L125.0,156.0 117.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6d3f105e85ae4a938ae7a12a1032b491-0-2\" stroke-width=\"2px\" d=\"M222,162.0 222,148.66666666666666 281.0,148.66666666666666 281.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6d3f105e85ae4a938ae7a12a1032b491-0-2\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M222,164.0 L218,156.0 226,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6d3f105e85ae4a938ae7a12a1032b491-0-3\" stroke-width=\"2px\" d=\"M382,162.0 382,148.66666666666666 441.0,148.66666666666666 441.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6d3f105e85ae4a938ae7a12a1032b491-0-3\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M382,164.0 L378,156.0 386,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6d3f105e85ae4a938ae7a12a1032b491-0-4\" stroke-width=\"2px\" d=\"M302,162.0 302,135.33333333333334 444.0,135.33333333333334 444.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6d3f105e85ae4a938ae7a12a1032b491-0-4\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M444.0,164.0 L448.0,156.0 440.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6d3f105e85ae4a938ae7a12a1032b491-0-5\" stroke-width=\"2px\" d=\"M542,162.0 542,148.66666666666666 601.0,148.66666666666666 601.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6d3f105e85ae4a938ae7a12a1032b491-0-5\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">mark</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M542,164.0 L538,156.0 546,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6d3f105e85ae4a938ae7a12a1032b491-0-6\" stroke-width=\"2px\" d=\"M302,162.0 302,108.66666666666666 610.0,108.66666666666666 610.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6d3f105e85ae4a938ae7a12a1032b491-0-6\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">advcl</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M610.0,164.0 L614.0,156.0 606.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6d3f105e85ae4a938ae7a12a1032b491-0-7\" stroke-width=\"2px\" d=\"M622,162.0 622,148.66666666666666 681.0,148.66666666666666 681.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6d3f105e85ae4a938ae7a12a1032b491-0-7\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M681.0,164.0 L685.0,156.0 677.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "</svg>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%aimport MirGeneRelCheck\n",
    "\n",
    "testcase = (u\"Among them, circ_002178 promoted PD-L1 expression via sponging miR-34a.\", 5, 9)\n",
    "\n",
    "analyse(testcase)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 0 Functional ADJ amod association 11 ()\n",
      "11 1 association NOUN nsubjpass established 72 ()\n",
      "23 2 between ADP case miR 31 ()\n",
      "31 3 miR NOUN nmod association 11 ()\n",
      "35 4 124a NOUN dep miR 31 (CDK6,)\n",
      "40 5 and CCONJ cc 124a 35 ()\n",
      "44 6 CDK6 NOUN conj 124a 35 (124a,)\n",
      "49 7 in ADP case medulloblastoma 52 ()\n",
      "52 8 medulloblastoma NOUN nmod association 11 ()\n",
      "68 9 was AUX auxpass established 72 ()\n",
      "72 10 established VERB ROOT established 72 ()\n",
      "84 11 using VERB xcomp established 72 ()\n",
      "90 12 luciferase NOUN compound assays 101 ()\n",
      "101 13 assays NOUN dobj using 84 ()\n",
      "107 14 . PUNCT punct established 72 ()\n",
      "124a CDK6\n",
      "CDK6 124a\n",
      "Conjuncts [CDK6, 124a]\n",
      "neighbour CDK6 in\n",
      "neighbour CDK6 in\n",
      "Conjunction before fc 124a\n",
      "Conjuncts [124a, CDK6]\n",
      "neighbour 124a and\n",
      "neighbour 124a and\n",
      "Conjunction before fc CDK6\n",
      "Conjunctions ( conj )\n",
      "124a [124a, CDK6]\n",
      "achildren {CDK6}\n",
      "achildren False\n",
      "Conjuncts [association, miR]\n",
      "neighbour association between\n",
      "neighbour association between\n",
      "Conjunction before fc miR\n",
      "fc add children for miR\n",
      "fc add children for miR\n",
      "Conjuncts [association, medulloblastoma]\n",
      "neighbour association between\n",
      "neighbour association between\n",
      "Conjunction before fc medulloblastoma\n",
      "fc add children for medulloblastoma\n",
      "Conjunctions ( nmod )\n",
      "miR [association, between, miR, 124a, in, medulloblastoma]\n",
      "SDP sent: Functional association between miR 124a and CDK6 in medulloblastoma was established using luciferase assays.\n",
      "SDP     : [(124a, 'NOUN', 'dep', False), (established, 'VERB', 'ROOT', False), (association, 'NOUN', 'nsubjpass', False), (miR, 'NOUN', 'nmod', False), (124a, 'NOUN', 'dep', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[CDK6]\n",
      "deepcheck connection rule noun found using luciferase [using, luciferase, assays]\n",
      "0 [Functional, association, between, miR, 124a, and, CDK6, in, medulloblastoma, was, established] 11 0 Functional association between miR 124a and CDK6 in medulloblastoma was established using luciferase assays.\n",
      "1 [using, luciferase, assays] 3 84 using luciferase assays.\n",
      "DeepCheck Conj Subtree\n",
      "[between, miR, 124a, and, CDK6]\n",
      "DeepCheck Conj Subtree\n",
      "[CDK6]\n",
      "DeepCheck Conj Subtree\n",
      "[in, medulloblastoma]\n",
      "deepcheck connection rule noun found using luciferase [using, luciferase, assays]\n",
      "0 [Functional, association, between, miR, 124a, and, CDK6, in, medulloblastoma, was, established] 11 0 Functional association between miR 124a and CDK6 in medulloblastoma was established using luciferase assays.\n",
      "1 [using, luciferase, assays] 3 84 using luciferase assays.\n",
      "passive? True\n",
      "Counter()\n",
      "Compartment Text Functional association between miR 124a and CDK6 in medulloblastoma was established\n",
      "[Functional, association, between, miR, 124a, and, CDK6, in, medulloblastoma, was, established]\n",
      "TextBetween: Functional association between miR 124a and CDK6 in medulloblastoma was established\n",
      "infer direction from token set [Functional, association, between, miR, 124a, and, CDK6, in, medulloblastoma, was, established]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8ce18>, {})\n",
      "counts between 1 stemTypes2Count Counter()\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe4ad90>, {})\n",
      "counts between 1 stemTypes2Count []\n",
      "(False, {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': True, 'reg_detect': 'return', 'reg_detect_major': 'return'}})\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Setting relfile /mnt/extproj/projekte/textmining/miRExplore/python/allrels.csv\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<svg xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" xml:lang=\"en\" id=\"f3ce1f68cb244eabbea9dda5370292fb-0\" class=\"displacy\" width=\"1170\" height=\"297.0\" direction=\"ltr\" style=\"max-width: none; height: 297.0px; color: #000000; background: #ffffff; font-family: Arial; direction: ltr\">\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"50\">Functional</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"50\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"130\">association</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"130\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"210\">between</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"210\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"290\">miR</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"290\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"370\">124a</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"370\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"450\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"450\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"530\">CDK6</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"530\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"610\">in</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"610\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"690\">medulloblastoma</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"690\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"770\">was</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"770\">AUX</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"850\">established</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"850\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"930\">using</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"930\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1010\">luciferase</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1010\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"207.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1090\">assays.</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1090\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-f3ce1f68cb244eabbea9dda5370292fb-0-0\" stroke-width=\"2px\" d=\"M62,162.0 62,148.66666666666666 121.0,148.66666666666666 121.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-f3ce1f68cb244eabbea9dda5370292fb-0-0\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M62,164.0 L58,156.0 66,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-f3ce1f68cb244eabbea9dda5370292fb-0-1\" stroke-width=\"2px\" d=\"M142,162.0 142,108.66666666666666 850.0,108.66666666666666 850.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-f3ce1f68cb244eabbea9dda5370292fb-0-1\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubjpass</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M142,164.0 L138,156.0 146,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-f3ce1f68cb244eabbea9dda5370292fb-0-2\" stroke-width=\"2px\" d=\"M222,162.0 222,148.66666666666666 281.0,148.66666666666666 281.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-f3ce1f68cb244eabbea9dda5370292fb-0-2\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M222,164.0 L218,156.0 226,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-f3ce1f68cb244eabbea9dda5370292fb-0-3\" stroke-width=\"2px\" d=\"M142,162.0 142,135.33333333333334 284.0,135.33333333333334 284.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-f3ce1f68cb244eabbea9dda5370292fb-0-3\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M284.0,164.0 L288.0,156.0 280.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-f3ce1f68cb244eabbea9dda5370292fb-0-4\" stroke-width=\"2px\" d=\"M302,162.0 302,148.66666666666666 361.0,148.66666666666666 361.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-f3ce1f68cb244eabbea9dda5370292fb-0-4\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dep</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M361.0,164.0 L365.0,156.0 357.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-f3ce1f68cb244eabbea9dda5370292fb-0-5\" stroke-width=\"2px\" d=\"M382,162.0 382,148.66666666666666 441.0,148.66666666666666 441.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-f3ce1f68cb244eabbea9dda5370292fb-0-5\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M441.0,164.0 L445.0,156.0 437.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-f3ce1f68cb244eabbea9dda5370292fb-0-6\" stroke-width=\"2px\" d=\"M382,162.0 382,135.33333333333334 524.0,135.33333333333334 524.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-f3ce1f68cb244eabbea9dda5370292fb-0-6\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M524.0,164.0 L528.0,156.0 520.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-f3ce1f68cb244eabbea9dda5370292fb-0-7\" stroke-width=\"2px\" d=\"M622,162.0 622,148.66666666666666 681.0,148.66666666666666 681.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-f3ce1f68cb244eabbea9dda5370292fb-0-7\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M622,164.0 L618,156.0 626,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-f3ce1f68cb244eabbea9dda5370292fb-0-8\" stroke-width=\"2px\" d=\"M142,162.0 142,122.0 687.0,122.0 687.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-f3ce1f68cb244eabbea9dda5370292fb-0-8\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M687.0,164.0 L691.0,156.0 683.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-f3ce1f68cb244eabbea9dda5370292fb-0-9\" stroke-width=\"2px\" d=\"M782,162.0 782,148.66666666666666 841.0,148.66666666666666 841.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-f3ce1f68cb244eabbea9dda5370292fb-0-9\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">auxpass</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M782,164.0 L778,156.0 786,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-f3ce1f68cb244eabbea9dda5370292fb-0-10\" stroke-width=\"2px\" d=\"M862,162.0 862,148.66666666666666 921.0,148.66666666666666 921.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-f3ce1f68cb244eabbea9dda5370292fb-0-10\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">xcomp</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M921.0,164.0 L925.0,156.0 917.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-f3ce1f68cb244eabbea9dda5370292fb-0-11\" stroke-width=\"2px\" d=\"M1022,162.0 1022,148.66666666666666 1081.0,148.66666666666666 1081.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-f3ce1f68cb244eabbea9dda5370292fb-0-11\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1022,164.0 L1018,156.0 1026,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-f3ce1f68cb244eabbea9dda5370292fb-0-12\" stroke-width=\"2px\" d=\"M942,162.0 942,135.33333333333334 1084.0,135.33333333333334 1084.0,162.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-f3ce1f68cb244eabbea9dda5370292fb-0-12\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1084.0,164.0 L1088.0,156.0 1080.0,156.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "</svg>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%aimport MirGeneRelCheck\n",
    "\n",
    "testcase = (u\"Functional association between miR 124a and CDK6 in medulloblastoma was established using luciferase assays.\", 4, 6)\n",
    "\n",
    "analyse(testcase)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 0 Overexpression NOUN nsubj reduces 50 ()\n",
      "15 1 of ADP case mir-7 18 ()\n",
      "18 2 mir-7 NOUN nmod Overexpression 0 (mir-153,)\n",
      "24 3 and CCONJ cc mir-7 18 ()\n",
      "28 4 mir-153 NOUN conj mir-7 18 (mir-7,)\n",
      "36 5 significantly ADV advmod reduces 50 ()\n",
      "50 6 reduces VERB ROOT reduces 50 ()\n",
      "58 7 endogenous ADJ amod levels 85 ()\n",
      "69 8 alpha-synuclein ADJ amod levels 85 ()\n",
      "85 9 levels NOUN dobj reduces 50 ()\n",
      "91 10 , PUNCT punct reduces 50 ()\n",
      "93 11 whereas SCONJ dep reduces 50 ()\n",
      "101 12 inhibition NOUN nsubj enhances 133 ()\n",
      "112 13 of ADP case mir-7 115 ()\n",
      "115 14 mir-7 NOUN nmod inhibition 101 (mir-153,)\n",
      "121 15 and CCONJ cc mir-7 115 ()\n",
      "125 16 mir-153 NOUN conj mir-7 115 (mir-7,)\n",
      "133 17 enhances VERB parataxis reduces 50 ()\n",
      "142 18 translation NOUN dobj enhances 133 ()\n",
      "154 19 of ADP case construct 170 ()\n",
      "157 20 a DET det construct 170 ()\n",
      "159 21 luciferase NOUN compound construct 170 ()\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Setting relfile /mnt/extproj/projekte/textmining/miRExplore/python/allrels.csv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "170 22 construct NOUN nmod translation 142 ()\n",
      "180 23 bearing VERB acl construct 170 ()\n",
      "188 24 the DET det alpha-s 192 ()\n",
      "192 25 alpha-s NOUN dobj bearing 180 ()\n",
      "reduces ,\n",
      ", reduces\n",
      "Conjuncts [mir-153, mir-7]\n",
      "neighbour mir-153 significantly\n",
      "neighbour mir-153 significantly\n",
      "Conjunction before fc mir-7\n",
      "fc add children for mir-7\n",
      "Conjuncts [mir-7, mir-153]\n",
      "neighbour mir-7 and\n",
      "neighbour mir-7 and\n",
      "Conjunction before fc mir-153\n",
      "Conjuncts [mir-153, mir-7]\n",
      "neighbour mir-153 enhances\n",
      "neighbour mir-153 enhances\n",
      "Conjunction before fc mir-7\n",
      "fc add children for mir-7\n",
      "Conjuncts [mir-7, mir-153]\n",
      "neighbour mir-7 and\n",
      "neighbour mir-7 and\n",
      "Conjunction before fc mir-153\n",
      "Conunct merge check: mir-7 mir-7\n",
      "mir-153 significantly reduces endogenous alpha-synuclein levels, whereas inhibition\n",
      "Conjunctions ( conj )\n",
      "mir-7 [of, mir-7, mir-153]\n",
      "mir-7 [of, mir-7, mir-153]\n",
      "Conjuncts [Overexpression, mir-7]\n",
      "neighbour Overexpression of\n",
      "neighbour Overexpression of\n",
      "Conjunction before fc mir-7\n",
      "fc add children for mir-7\n",
      "Conjuncts [inhibition, mir-7]\n",
      "neighbour inhibition of\n",
      "neighbour inhibition of\n",
      "Conjunction before fc mir-7\n",
      "fc add children for mir-7\n",
      "Conjuncts [translation, construct]\n",
      "neighbour translation of\n",
      "neighbour translation of\n",
      "Conjunction before fc construct\n",
      "fc add children for construct\n",
      "fc add children for construct\n",
      "fc add children for construct\n",
      "Conunct merge check: mir-7 mir-7\n",
      "mir-7 and mir-153 significantly reduces endogenous alpha-synuclein levels, whereas\n",
      "Conunct merge check: mir-7 construct\n",
      "mir-7 and mir-153 enhances\n",
      "Double Conjunction Found\n",
      "Conjunctions ( nmod )\n",
      "mir-7 [Overexpression, of, mir-7]\n",
      "mir-7 [inhibition, of, mir-7]\n",
      "construct [translation, of, luciferase, construct, bearing, alpha-s]\n",
      "(construct, 'conj') [inhibition, of, mir-7, translation, of, luciferase, construct, bearing, alpha-s]\n",
      "SDP sent: Overexpression of mir-7 and mir-153 significantly reduces endogenous alpha-synuclein levels, whereas inhibition of mir-7 and mir-153 enhances translation of a luciferase construct bearing the alpha-s\n",
      "SDP     : [(reduces, 'VERB', 'ROOT', False), (reduces, 'VERB', 'ROOT', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[mir-153]\n",
      "DeepCheck Conj Subtree\n",
      "[mir-153]\n",
      "deepcheck connection rule noun found enhances inhibition [inhibition, of, mir-7, and, mir-153, enhances, translation, of, a, luciferase, construct, bearing, the, alpha-s]\n",
      "0 [Overexpression, of, mir-7, and, mir-153, significantly, reduces, endogenous, alpha-synuclein, levels] 10 0 Overexpression of mir-7 and mir-153 significantly reduces endogenous alpha-synuclein levels, whereas inhibition of mir-7 and mir-153 enhances translation of a luciferase construct bearing the alpha-s\n",
      "1 [inhibition, of, mir-7, and, mir-153, enhances, translation, of, a, luciferase, construct, bearing, the, alpha-s] 14 101 inhibition of mir-7 and mir-153 enhances translation of a luciferase construct bearing the alpha-s\n",
      "DeepCheck Conj Subtree\n",
      "[of, mir-7, and, mir-153]\n",
      "DeepCheck Conj Subtree\n",
      "[mir-153]\n",
      "DeepCheck Conj Subtree\n",
      "[of, mir-7, and, mir-153]\n",
      "DeepCheck Conj Subtree\n",
      "[mir-153]\n",
      "deepcheck connection rule noun found enhances inhibition [inhibition, of, mir-7, and, mir-153, enhances, translation, of, a, luciferase, construct, bearing, the, alpha-s]\n",
      "DeepCheck Conj Subtree\n",
      "[of, a, luciferase, construct, bearing, the, alpha-s]\n",
      "0 [Overexpression, of, mir-7, and, mir-153, significantly, reduces, endogenous, alpha-synuclein, levels] 10 0 Overexpression of mir-7 and mir-153 significantly reduces endogenous alpha-synuclein levels, whereas inhibition of mir-7 and mir-153 enhances translation of a luciferase construct bearing the alpha-s\n",
      "1 [inhibition, of, mir-7, and, mir-153, enhances, translation, of, a, luciferase, construct, bearing, the, alpha-s] 14 101 inhibition of mir-7 and mir-153 enhances translation of a luciferase construct bearing the alpha-s\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Overexpression of mir-7 and mir-153 significantly reduces endogenous alpha-synuclein levels, whereas inhibition of mir-7 and mir-153 enhances translation of a luciferase construct bearing the alpha-s\n",
      "TextBetween: mir-153 significantly reduces endogenous alpha-synuclein levels , whereas inhibition\n",
      "infer direction from token set [mir-153, significantly, reduces, endogenous, alpha-synuclein, levels, ,, whereas, inhibition]\n",
      "stemTypes2Count: Counter({'NEG': 2})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fc178c8>, {'gene': Counter({'NEG': 1}), 'mir': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEG': 2, 'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fc17e18>, {'gene': Counter({'NEG': 1}), 'mir': Counter({'NEU': 1, 'NEG': 1})})\n",
      "counts between 1 stemTypes2Count [(reduces, 'NEG'), (levels, 'NEU'), (inhibition, 'NEG')]\n",
      "Between Tokens [(levels, 'NEU')]\n",
      "counts between final False\n",
      "['mir-153', 'significantly', 'reduces', 'endogenous', 'alpha-synuclein', 'levels', ',', 'whereas', 'inhibition']\n",
      "Counter({'NEG': 2, 'NEU': 1})\n",
      "[(reduces, 'NEG'), (levels, 'NEU'), (inhibition, 'NEG')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<svg xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" xml:lang=\"en\" id=\"08d3c6def6934d97a84f0c80f267a30b-0\" class=\"displacy\" width=\"2050\" height=\"417.0\" direction=\"ltr\" style=\"max-width: none; height: 417.0px; color: #000000; background: #ffffff; font-family: Arial; direction: ltr\">\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"50\">Overexpression</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"50\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"130\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"130\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"210\">mir-7</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"210\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"290\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"290\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"370\">mir-153</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"370\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"450\">significantly</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"450\">ADV</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"530\">reduces</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"530\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"610\">endogenous</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"610\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"690\">alpha-synuclein</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"690\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"770\">levels,</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"770\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"850\">whereas</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"850\">SCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"930\">inhibition</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"930\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1010\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1010\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1090\">mir-7</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1090\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1170\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1170\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1250\">mir-153</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1250\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1330\">enhances</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1330\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1410\">translation</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1410\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1490\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1490\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1570\">a</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1570\">DET</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1650\">luciferase</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1650\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1730\">construct</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1730\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1810\">bearing</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1810\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1890\">the</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1890\">DET</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"327.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1970\">alpha-s</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1970\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-0\" stroke-width=\"2px\" d=\"M62,282.0 62,202.0 527.0,202.0 527.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-0\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M62,284.0 L58,276.0 66,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-1\" stroke-width=\"2px\" d=\"M142,282.0 142,268.6666666666667 192.0,268.6666666666667 192.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-1\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M142,284.0 L138,276.0 146,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-2\" stroke-width=\"2px\" d=\"M62,282.0 62,255.33333333333334 195.0,255.33333333333334 195.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-2\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M195.0,284.0 L199.0,276.0 191.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-3\" stroke-width=\"2px\" d=\"M222,282.0 222,268.6666666666667 272.0,268.6666666666667 272.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-3\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M272.0,284.0 L276.0,276.0 268.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-4\" stroke-width=\"2px\" d=\"M222,282.0 222,255.33333333333334 355.0,255.33333333333334 355.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-4\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M355.0,284.0 L359.0,276.0 351.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-5\" stroke-width=\"2px\" d=\"M462,282.0 462,268.6666666666667 512.0,268.6666666666667 512.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-5\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">advmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M462,284.0 L458,276.0 466,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-6\" stroke-width=\"2px\" d=\"M622,282.0 622,255.33333333333334 755.0,255.33333333333334 755.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-6\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M622,284.0 L618,276.0 626,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-7\" stroke-width=\"2px\" d=\"M702,282.0 702,268.6666666666667 752.0,268.6666666666667 752.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-7\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M702,284.0 L698,276.0 706,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-8\" stroke-width=\"2px\" d=\"M542,282.0 542,242.0 758.0,242.0 758.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-8\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M758.0,284.0 L762.0,276.0 754.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-9\" stroke-width=\"2px\" d=\"M542,282.0 542,228.66666666666666 841.0,228.66666666666666 841.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-9\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dep</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M841.0,284.0 L845.0,276.0 837.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-10\" stroke-width=\"2px\" d=\"M942,282.0 942,215.33333333333331 1324.0,215.33333333333331 1324.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-10\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M942,284.0 L938,276.0 946,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-11\" stroke-width=\"2px\" d=\"M1022,282.0 1022,268.6666666666667 1072.0,268.6666666666667 1072.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-11\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1022,284.0 L1018,276.0 1026,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-12\" stroke-width=\"2px\" d=\"M942,282.0 942,255.33333333333334 1075.0,255.33333333333334 1075.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-12\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1075.0,284.0 L1079.0,276.0 1071.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-13\" stroke-width=\"2px\" d=\"M1102,282.0 1102,268.6666666666667 1152.0,268.6666666666667 1152.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-13\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1152.0,284.0 L1156.0,276.0 1148.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-14\" stroke-width=\"2px\" d=\"M1102,282.0 1102,255.33333333333334 1235.0,255.33333333333334 1235.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-14\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1235.0,284.0 L1239.0,276.0 1231.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-15\" stroke-width=\"2px\" d=\"M542,282.0 542,188.66666666666669 1330.0,188.66666666666669 1330.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-15\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">parataxis</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1330.0,284.0 L1334.0,276.0 1326.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-16\" stroke-width=\"2px\" d=\"M1342,282.0 1342,268.6666666666667 1392.0,268.6666666666667 1392.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-16\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1392.0,284.0 L1396.0,276.0 1388.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-17\" stroke-width=\"2px\" d=\"M1502,282.0 1502,242.0 1718.0,242.0 1718.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-17\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1502,284.0 L1498,276.0 1506,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-18\" stroke-width=\"2px\" d=\"M1582,282.0 1582,255.33333333333334 1715.0,255.33333333333334 1715.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-18\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">det</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1582,284.0 L1578,276.0 1586,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-19\" stroke-width=\"2px\" d=\"M1662,282.0 1662,268.6666666666667 1712.0,268.6666666666667 1712.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-19\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1662,284.0 L1658,276.0 1666,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-20\" stroke-width=\"2px\" d=\"M1422,282.0 1422,228.66666666666666 1721.0,228.66666666666666 1721.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-20\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1721.0,284.0 L1725.0,276.0 1717.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-21\" stroke-width=\"2px\" d=\"M1742,282.0 1742,268.6666666666667 1792.0,268.6666666666667 1792.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-21\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">acl</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1792.0,284.0 L1796.0,276.0 1788.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-22\" stroke-width=\"2px\" d=\"M1902,282.0 1902,268.6666666666667 1952.0,268.6666666666667 1952.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-22\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">det</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1902,284.0 L1898,276.0 1906,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-08d3c6def6934d97a84f0c80f267a30b-0-23\" stroke-width=\"2px\" d=\"M1822,282.0 1822,255.33333333333334 1955.0,255.33333333333334 1955.0,282.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-08d3c6def6934d97a84f0c80f267a30b-0-23\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1955.0,284.0 L1959.0,276.0 1951.0,276.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "</svg>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%aimport MirGeneRelCheck\n",
    "\n",
    "testcase = (u\"Overexpression of mir-7 and mir-153 significantly reduces endogenous alpha-synuclein levels, whereas inhibition of mir-7 and mir-153 enhances translation of a luciferase construct bearing the alpha-s\", 6, 10)\n",
    "\n",
    "analyse(testcase)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 0 The DET det relevance 13 ()\n",
      "4 1 clinical ADJ amod relevance 13 ()\n",
      "13 2 relevance NOUN nsubjpass evaluated 43 ()\n",
      "23 3 of ADP case findings 30 ()\n",
      "26 4 our PRON nmod:poss findings 30 ()\n",
      "30 5 findings NOUN nmod relevance 13 ()\n",
      "39 6 was AUX auxpass evaluated 43 ()\n",
      "43 7 evaluated VERB ROOT evaluated 43 ()\n",
      "53 8 in ADP case samples 86 ()\n",
      "56 9 HIV-encephalitis NOUN compound samples 86 ()\n",
      "73 "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Setting relfile /mnt/extproj/projekte/textmining/miRExplore/python/allrels.csv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 ( PUNCT punct HIVE 74 ()\n",
      "74 11 HIVE NOUN appos HIV-encephalitis 56 ()\n",
      "78 12 ) PUNCT punct HIVE 74 ()\n",
      "80 13 brain NOUN compound samples 86 ()\n",
      "86 14 samples NOUN nmod evaluated 43 ()\n",
      "94 15 in ADP case which 97 ()\n",
      "97 16 which PRON nmod observed 167 ()\n",
      "103 17 decreased VERB amod levels 113 ()\n",
      "113 18 levels NOUN nsubjpass observed 167 (levels,)\n",
      "120 19 of ADP case MCP-2 123 ()\n",
      "123 20 MCP-2 NOUN nmod levels 113 ()\n",
      "129 21 and CCONJ cc levels 113 ()\n",
      "133 22 increased VERB amod levels 143 ()\n",
      "143 23 levels NOUN conj levels 113 (levels,)\n",
      "150 24 of ADP case mir-146a 153 ()\n",
      "153 25 mir-146a NOUN nmod levels 143 ()\n",
      "162 26 were AUX auxpass observed 167 ()\n",
      "167 27 observed VERB acl:relcl samples 86 ()\n",
      "175 28 , PUNCT punct evaluated 43 ()\n",
      "177 29 suggesting VERB xcomp evaluated 43 ()\n",
      "188 30 a DET det role 190 ()\n",
      "190 31 role NOUN dobj suggesting 177 ()\n",
      "195 32 for ADP case mir-146a 199 ()\n",
      "199 33 mir-146a NOUN nmod role 190 ()\n",
      "208 34 in ADP case maintenance 215 ()\n",
      "211 35 the DET det maintenance 215 ()\n",
      "215 36 maintenance NOUN nmod role 190 ()\n",
      "227 37 of ADP case inflammation 251 ()\n",
      "230 38 HIV-mediated ADJ amod inflammation 251 ()\n",
      "243 39 chronic ADJ amod inflammation 251 ()\n",
      "251 40 inflammation NOUN nmod maintenance 215 ()\n",
      "264 41 of ADP case brain 271 ()\n",
      "267 42 the DET det brain 271 ()\n",
      "271 43 brain NOUN nmod inflammation 251 ()\n",
      "276 44 . PUNCT punct evaluated 43 ()\n",
      "MCP-2 mir-146a\n",
      "mir-146a MCP-2\n",
      "Conjuncts [levels, levels]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conjuncts [levels, levels]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conunct merge check: HIVE levels\n",
      ") brain samples in which\n",
      "Conjunctions ( conj )\n",
      "HIVE [(, )]\n",
      "levels [decreased, levels, of, MCP-2, increased, levels, of, mir-146a]\n",
      "achildren {mir-146a}\n",
      "achildren False\n",
      "Conjuncts [relevance, findings]\n",
      "neighbour relevance of\n",
      "neighbour relevance of\n",
      "Conjunction before fc findings\n",
      "fc add children for findings\n",
      "Conjuncts [evaluated, samples]\n",
      "neighbour evaluated in\n",
      "neighbour evaluated in\n",
      "Conjunction before fc samples\n",
      "fc add children for samples\n",
      "fc add children for samples\n",
      "fc add children for samples\n",
      "Conjuncts [observed, which]\n",
      "neighbour observed ,\n",
      "neighbour observed suggesting\n",
      "Conjunction before fc which\n",
      "fc add children for which\n",
      "Conjuncts [levels, MCP-2]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc MCP-2\n",
      "fc add children for MCP-2\n",
      "Conjuncts [levels, mir-146a]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc mir-146a\n",
      "fc add children for mir-146a\n",
      "Conjuncts [role, mir-146a]\n",
      "neighbour role for\n",
      "neighbour role for\n",
      "Conjunction before fc mir-146a\n",
      "fc add children for mir-146a\n",
      "Conjuncts [role, maintenance]\n",
      "neighbour role for\n",
      "neighbour role for\n",
      "Conjunction before fc maintenance\n",
      "fc add children for maintenance\n",
      "fc add children for maintenance\n",
      "Conjuncts [maintenance, inflammation]\n",
      "neighbour maintenance of\n",
      "neighbour maintenance of\n",
      "Conjunction before fc inflammation\n",
      "fc add children for inflammation\n",
      "fc add children for inflammation\n",
      "fc add children for inflammation\n",
      "fc add children for inflammation\n",
      "Conjuncts [inflammation, brain]\n",
      "neighbour inflammation of\n",
      "neighbour inflammation of\n",
      "Conjunction before fc brain\n",
      "fc add children for brain\n",
      "Conunct merge check: findings samples\n",
      "findings was\n",
      "Conunct merge check: samples HIVE\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: HIVE which\n",
      ") brain samples\n",
      "Conunct merge check: which MCP-2\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: MCP-2 mir-146a\n",
      "MCP-2 and increased\n",
      "Double Conjunction Found\n",
      "Conunct merge check: mir-146a mir-146a\n",
      "mir-146a were observed, suggesting a\n",
      "Conjunctions ( nmod )\n",
      "findings [relevance, of, findings]\n",
      "HIVE [(, )]\n",
      "samples [evaluated, in, HIV-encephalitis, HIVE, brain, samples]\n",
      "which [in, which, observed]\n",
      "MCP-2 [levels, of, MCP-2]\n",
      "mir-146a [levels, of, mir-146a]\n",
      "mir-146a [role, for, mir-146a, in, maintenance, of, HIV-mediated, chronic, inflammation, of, brain]\n",
      "(mir-146a, 'conj') [levels, of, MCP-2, levels, of, mir-146a]\n",
      "achildren {mir-146a}\n",
      "achildren False\n",
      "SDP sent: The clinical relevance of our findings was evaluated in HIV-encephalitis (HIVE) brain samples in which decreased levels of MCP-2 and increased levels of mir-146a were observed, suggesting a role for mir-146a in the maintenance of HIV-mediated chronic inflammation of the brain.\n",
      "SDP     : [(MCP-2, 'NOUN', 'nmod', False), (levels, 'NOUN', 'nsubjpass', True), (levels, 'NOUN', 'conj', False), (mir-146a, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[increased, levels, of, mir-146a]\n",
      "deepcheck conj verb\n",
      "deepcheck conj verb\n",
      "0 [The, clinical, relevance, of, our, findings, was, evaluated, in, HIV-encephalitis, (, HIVE, ), brain, samples, in, which, decreased, levels, of, MCP-2, and, increased, levels, of, mir-146a, were, observed] 28 0 The clinical relevance of our findings was evaluated in HIV-encephalitis (HIVE) brain samples in which decreased levels of MCP-2 and increased levels of mir-146a were observed, suggesting a role for mir-146a in the maintenance of HIV-mediated chronic inflammation of the brain.\n",
      "1 [suggesting, a, role, for, mir-146a, in, the, maintenance, of, HIV-mediated, chronic, inflammation, of, the, brain] 15 177 suggesting a role for mir-146a in the maintenance of HIV-mediated chronic inflammation of the brain.\n",
      "DeepCheck Conj Subtree\n",
      "[of, our, findings]\n",
      "DeepCheck Conj Subtree\n",
      "[in, HIV-encephalitis, (, HIVE, ), brain, samples, in, which, decreased, levels, of, MCP-2, and, increased, levels, of, mir-146a, were, observed]\n",
      "DeepCheck Conj Subtree\n",
      "[in, which]\n",
      "DeepCheck Conj Subtree\n",
      "[of, MCP-2]\n",
      "DeepCheck Conj Subtree\n",
      "[increased, levels, of, mir-146a]\n",
      "deepcheck conj verb\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[of, mir-146a]\n",
      "DeepCheck Conj Subtree\n",
      "[for, mir-146a]\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, maintenance, of, HIV-mediated, chronic, inflammation, of, the, brain]\n",
      "DeepCheck Conj Subtree\n",
      "[of, HIV-mediated, chronic, inflammation, of, the, brain]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, brain]\n",
      "0 [The, clinical, relevance, of, our, findings, was, evaluated, in, HIV-encephalitis, (, HIVE, ), brain, samples, in, which, decreased, levels, of, MCP-2, and, increased, levels, of, mir-146a, were, observed] 28 0 The clinical relevance of our findings was evaluated in HIV-encephalitis (HIVE) brain samples in which decreased levels of MCP-2 and increased levels of mir-146a were observed, suggesting a role for mir-146a in the maintenance of HIV-mediated chronic inflammation of the brain.\n",
      "1 [suggesting, a, role, for, mir-146a, in, the, maintenance, of, HIV-mediated, chronic, inflammation, of, the, brain] 15 177 suggesting a role for mir-146a in the maintenance of HIV-mediated chronic inflammation of the brain.\n",
      "passive? True\n",
      "Counter()\n",
      "Compartment Text The clinical relevance of our findings was evaluated in HIV-encephalitis ( HIVE ) brain samples in which decreased levels of MCP-2 and increased levels of mir-146a were observed\n",
      "[The, clinical, relevance, of, our, findings, was, evaluated, in, HIV-encephalitis, (, HIVE, ), brain, samples, in, which, decreased, levels, of, MCP-2, and, increased, levels, of, mir-146a, were, observed]\n",
      "TextBetween: The clinical relevance of our findings was evaluated in HIV-encephalitis ( HIVE ) brain samples in which decreased levels of MCP-2 and increased levels of mir-146a were observed\n",
      "infer direction from token set [The, clinical, relevance, of, our, findings, was, evaluated, in, HIV-encephalitis, (, HIVE, ), brain, samples, in, which, decreased, levels, of, MCP-2, and, increased, levels, of, mir-146a, were, observed]\n",
      "stemTypes2Count: Counter({'NEG': 1, 'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8cd08>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'POS': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8cae8>, {'gene': Counter({'NEU': 1, 'POS': 1}), 'mir': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(levels, 'NEU'), (increased, 'POS'), (levels, 'NEU')]\n",
      "Between Tokens [(increased, 'POS'), (levels, 'NEU')]\n",
      "counts between final True\n",
      "['levels', 'of', 'MCP-2', 'and', 'increased', 'levels', 'of', 'mir-146a', 'were', 'observed']\n",
      "Counter({'NEU': 2, 'POS': 1})\n",
      "[(levels, 'NEU'), (increased, 'POS'), (levels, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': False, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<svg xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" xml:lang=\"en\" id=\"6264ada9beaa4685a5265a19bcbb7d9c-0\" class=\"displacy\" width=\"3330\" height=\"497.0\" direction=\"ltr\" style=\"max-width: none; height: 497.0px; color: #000000; background: #ffffff; font-family: Arial; direction: ltr\">\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"50\">The</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"50\">DET</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"130\">clinical</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"130\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"210\">relevance</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"210\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"290\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"290\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"370\">our</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"370\">PRON</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"450\">findings</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"450\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"530\">was</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"530\">AUX</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"610\">evaluated</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"610\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"690\">in</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"690\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"770\">HIV-encephalitis (</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"770\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"850\">HIVE)</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"850\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"930\">brain</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"930\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1010\">samples</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1010\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1090\">in</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1090\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1170\">which</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1170\">PRON</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1250\">decreased</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1250\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1330\">levels</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1330\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1410\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1410\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1490\">MCP-2</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1490\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1570\">and</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1570\">CCONJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1650\">increased</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1650\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1730\">levels</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1730\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1810\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1810\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1890\">mir-146a</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1890\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"1970\">were</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"1970\">AUX</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2050\">observed,</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2050\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2130\">suggesting</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2130\">VERB</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2210\">a</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2210\">DET</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2290\">role</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2290\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2370\">for</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2370\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2450\">mir-146a</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2450\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2530\">in</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2530\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2610\">the</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2610\">DET</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2690\">maintenance</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2690\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2770\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2770\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2850\">HIV-mediated</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2850\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"2930\">chronic</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"2930\">ADJ</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3010\">inflammation</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3010\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3090\">of</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3090\">ADP</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3170\">the</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3170\">DET</tspan>\n",
       "</text>\n",
       "\n",
       "<text class=\"displacy-token\" fill=\"currentColor\" text-anchor=\"middle\" y=\"407.0\">\n",
       "    <tspan class=\"displacy-word\" fill=\"currentColor\" x=\"3250\">brain.</tspan>\n",
       "    <tspan class=\"displacy-tag\" dy=\"2em\" fill=\"currentColor\" x=\"3250\">NOUN</tspan>\n",
       "</text>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-0\" stroke-width=\"2px\" d=\"M62,362.0 62,335.3333333333333 189.0,335.3333333333333 189.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-0\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">det</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M62,364.0 L58,356.0 66,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-1\" stroke-width=\"2px\" d=\"M142,362.0 142,348.6666666666667 186.0,348.6666666666667 186.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-1\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M142,364.0 L138,356.0 146,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-2\" stroke-width=\"2px\" d=\"M222,362.0 222,295.3333333333333 598.0,295.3333333333333 598.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-2\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubjpass</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M222,364.0 L218,356.0 226,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-3\" stroke-width=\"2px\" d=\"M302,362.0 302,335.3333333333333 429.0,335.3333333333333 429.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-3\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M302,364.0 L298,356.0 306,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-4\" stroke-width=\"2px\" d=\"M382,362.0 382,348.6666666666667 426.0,348.6666666666667 426.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-4\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod:poss</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M382,364.0 L378,356.0 386,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-5\" stroke-width=\"2px\" d=\"M222,362.0 222,322.0 432.0,322.0 432.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-5\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M432.0,364.0 L436.0,356.0 428.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-6\" stroke-width=\"2px\" d=\"M542,362.0 542,348.6666666666667 586.0,348.6666666666667 586.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-6\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">auxpass</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M542,364.0 L538,356.0 546,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-7\" stroke-width=\"2px\" d=\"M702,362.0 702,308.6666666666667 995.0,308.6666666666667 995.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-7\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M702,364.0 L698,356.0 706,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-8\" stroke-width=\"2px\" d=\"M782,362.0 782,322.0 992.0,322.0 992.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-8\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M782,364.0 L778,356.0 786,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-9\" stroke-width=\"2px\" d=\"M782,362.0 782,348.6666666666667 826.0,348.6666666666667 826.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-9\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">appos</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M826.0,364.0 L830.0,356.0 822.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-10\" stroke-width=\"2px\" d=\"M942,362.0 942,348.6666666666667 986.0,348.6666666666667 986.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-10\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">compound</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M942,364.0 L938,356.0 946,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-11\" stroke-width=\"2px\" d=\"M622,362.0 622,295.3333333333333 998.0,295.3333333333333 998.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-11\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M998.0,364.0 L1002.0,356.0 994.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-12\" stroke-width=\"2px\" d=\"M1102,362.0 1102,348.6666666666667 1146.0,348.6666666666667 1146.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-12\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1102,364.0 L1098,356.0 1106,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-13\" stroke-width=\"2px\" d=\"M1182,362.0 1182,268.6666666666667 2044.0,268.6666666666667 2044.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-13\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1182,364.0 L1178,356.0 1186,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-14\" stroke-width=\"2px\" d=\"M1262,362.0 1262,348.6666666666667 1306.0,348.6666666666667 1306.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-14\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1262,364.0 L1258,356.0 1266,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-15\" stroke-width=\"2px\" d=\"M1342,362.0 1342,282.0 2041.0,282.0 2041.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-15\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nsubjpass</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1342,364.0 L1338,356.0 1346,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-16\" stroke-width=\"2px\" d=\"M1422,362.0 1422,348.6666666666667 1466.0,348.6666666666667 1466.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-16\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1422,364.0 L1418,356.0 1426,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-17\" stroke-width=\"2px\" d=\"M1342,362.0 1342,335.3333333333333 1469.0,335.3333333333333 1469.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-17\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1469.0,364.0 L1473.0,356.0 1465.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-18\" stroke-width=\"2px\" d=\"M1342,362.0 1342,322.0 1552.0,322.0 1552.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-18\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">cc</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1552.0,364.0 L1556.0,356.0 1548.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-19\" stroke-width=\"2px\" d=\"M1662,362.0 1662,348.6666666666667 1706.0,348.6666666666667 1706.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-19\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1662,364.0 L1658,356.0 1666,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-20\" stroke-width=\"2px\" d=\"M1342,362.0 1342,295.3333333333333 1718.0,295.3333333333333 1718.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-20\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">conj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1718.0,364.0 L1722.0,356.0 1714.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-21\" stroke-width=\"2px\" d=\"M1822,362.0 1822,348.6666666666667 1866.0,348.6666666666667 1866.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-21\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1822,364.0 L1818,356.0 1826,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-22\" stroke-width=\"2px\" d=\"M1742,362.0 1742,335.3333333333333 1869.0,335.3333333333333 1869.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-22\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1869.0,364.0 L1873.0,356.0 1865.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-23\" stroke-width=\"2px\" d=\"M1982,362.0 1982,348.6666666666667 2026.0,348.6666666666667 2026.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-23\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">auxpass</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M1982,364.0 L1978,356.0 1986,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-24\" stroke-width=\"2px\" d=\"M1022,362.0 1022,255.33333333333331 2047.0,255.33333333333331 2047.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-24\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">acl:relcl</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2047.0,364.0 L2051.0,356.0 2043.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-25\" stroke-width=\"2px\" d=\"M622,362.0 622,242.0 2130.0,242.0 2130.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-25\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">xcomp</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2130.0,364.0 L2134.0,356.0 2126.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-26\" stroke-width=\"2px\" d=\"M2222,362.0 2222,348.6666666666667 2266.0,348.6666666666667 2266.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-26\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">det</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2222,364.0 L2218,356.0 2226,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-27\" stroke-width=\"2px\" d=\"M2142,362.0 2142,335.3333333333333 2269.0,335.3333333333333 2269.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-27\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">dobj</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2269.0,364.0 L2273.0,356.0 2265.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-28\" stroke-width=\"2px\" d=\"M2382,362.0 2382,348.6666666666667 2426.0,348.6666666666667 2426.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-28\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2382,364.0 L2378,356.0 2386,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-29\" stroke-width=\"2px\" d=\"M2302,362.0 2302,335.3333333333333 2429.0,335.3333333333333 2429.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-29\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2429.0,364.0 L2433.0,356.0 2425.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-30\" stroke-width=\"2px\" d=\"M2542,362.0 2542,335.3333333333333 2669.0,335.3333333333333 2669.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-30\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2542,364.0 L2538,356.0 2546,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-31\" stroke-width=\"2px\" d=\"M2622,362.0 2622,348.6666666666667 2666.0,348.6666666666667 2666.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-31\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">det</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2622,364.0 L2618,356.0 2626,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-32\" stroke-width=\"2px\" d=\"M2302,362.0 2302,295.3333333333333 2678.0,295.3333333333333 2678.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-32\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2678.0,364.0 L2682.0,356.0 2674.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-33\" stroke-width=\"2px\" d=\"M2782,362.0 2782,322.0 2992.0,322.0 2992.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-33\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2782,364.0 L2778,356.0 2786,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-34\" stroke-width=\"2px\" d=\"M2862,362.0 2862,335.3333333333333 2989.0,335.3333333333333 2989.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-34\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2862,364.0 L2858,356.0 2866,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-35\" stroke-width=\"2px\" d=\"M2942,362.0 2942,348.6666666666667 2986.0,348.6666666666667 2986.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-35\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">amod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2942,364.0 L2938,356.0 2946,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-36\" stroke-width=\"2px\" d=\"M2702,362.0 2702,308.6666666666667 2995.0,308.6666666666667 2995.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-36\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M2995.0,364.0 L2999.0,356.0 2991.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-37\" stroke-width=\"2px\" d=\"M3102,362.0 3102,335.3333333333333 3229.0,335.3333333333333 3229.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-37\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">case</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3102,364.0 L3098,356.0 3106,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-38\" stroke-width=\"2px\" d=\"M3182,362.0 3182,348.6666666666667 3226.0,348.6666666666667 3226.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-38\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">det</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3182,364.0 L3178,356.0 3186,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "\n",
       "<g class=\"displacy-arrow\">\n",
       "    <path class=\"displacy-arc\" id=\"arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-39\" stroke-width=\"2px\" d=\"M3022,362.0 3022,322.0 3232.0,322.0 3232.0,362.0\" fill=\"none\" stroke=\"currentColor\"/>\n",
       "    <text dy=\"1.25em\" style=\"font-size: 0.8em; letter-spacing: 1px\">\n",
       "        <textPath xlink:href=\"#arrow-6264ada9beaa4685a5265a19bcbb7d9c-0-39\" class=\"displacy-label\" startOffset=\"50%\" side=\"left\" fill=\"currentColor\" text-anchor=\"middle\">nmod</textPath>\n",
       "    </text>\n",
       "    <path class=\"displacy-arrowhead\" d=\"M3232.0,364.0 L3236.0,356.0 3228.0,356.0\" fill=\"currentColor\"/>\n",
       "</g>\n",
       "</svg>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "analyse((u\"The clinical relevance of our findings was evaluated in HIV-encephalitis (HIVE) brain samples in which decreased levels of MCP-2 and increased levels of mir-146a were observed, suggesting a role for mir-146a in the maintenance of HIV-mediated chronic inflammation of the brain.\", 20, 25)\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "%aimport MirGeneRelCheck\n",
    "\n",
    "testCase = (u\"We show here that miR-29a, -29b-1, and -9 can regulate BACE1 expression in vitro.\", 4, 12)\n",
    "testCase = (u\"It has been predicted that miR-146a may target Tata Binding Protein (TBP).\",5, 12)\n",
    "testCase = (u\"Our results showed that the expression of miR-106b was inversely correlated with TRII protein levels and miR-106b can directly inhibit the TRII translation in vitro.\", 7, 21)\n",
    "\n",
    "relCheck = MirGeneRelCheck.MirGeneRelCheck()\n",
    "\n",
    "doc = nlp(testCase[0])\n",
    "\n",
    "mirword = doc[testCase[1]]\n",
    "geneword = doc[testCase[2]]\n",
    "\n",
    "verbose=False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "conjResult, conjs = relCheck.checkCommonConj(doc, mirword, geneword, verbose)\n",
    "sdpPass, passive, negated = relCheck.checkSDP(doc, mirword, geneword, verbose)\n",
    "compPass = relCheck.checkCompartments(doc, mirword, geneword, verbose)\n",
    "sigPathway = relCheck.checkSurContext(doc, mirword, geneword, verbose)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[(miR-106b, 'NOUN', 'nmod', False),\n",
       " (expression, 'NOUN', 'nsubjpass', False),\n",
       " (correlated, 'VERB', 'ccomp', True),\n",
       " (inhibit, 'VERB', 'conj', False),\n",
       " (translation, 'NOUN', 'dobj', False),\n",
       " (TRII, 'NOUN', 'compound', False)]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sdpRes = relCheck._get_sdp_path(doc, mirword.i, geneword.i)\n",
    "sdpRes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "%aimport MirGeneRelCheck\n",
    "\n",
    "testCase = (u\"We show here that miR-29a, -29b-1, and -9 can regulate BACE1 expression in vitro.\", 4, 12)\n",
    "testCase = (u\"It has been predicted that miR-146a may target Tata Binding Protein (TBP).\",5, 12)\n",
    "testCase = (u\"Our results showed that the expression of miR-106b was inversely correlated with TRII protein levels and miR-106b can directly inhibit the TRII translation in vitro.\", 7, 21)\n",
    "testCase = (u\"The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 6, 16)\n",
    "testCase = (u\"These findings identify miR-29b as a novel posttranscriptional regulator of PGRN expression, raising the possibility that miR-29b or other miRNAs might be targeted therapeutically to increase hPGRN levels in some FTD patients.\", 3,10)\n",
    "relCheck = MirGeneRelCheck.MirGeneRelCheck()\n",
    "\n",
    "doc = nlp(testCase[0])\n",
    "\n",
    "mirword = doc[testCase[1]]\n",
    "geneword = doc[testCase[2]]\n",
    "\n",
    "verbose=False\n",
    "\n",
    "from collections import Counter\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading stems\n"
     ]
    }
   ],
   "source": [
    "def loadStems():\n",
    "    \n",
    "    print(\"Loading stems\")\n",
    "    allRels = []\n",
    "    stem2class = {}\n",
    "\n",
    "    renameClass = {\n",
    "        'INT': 'NEU',\n",
    "        \"BOOST\": \"POS\",\n",
    "        \"COMBINE\": \"NEU\",\n",
    "        \"CHANGE\": \"NEU\",\n",
    "    }\n",
    "\n",
    "    with open(mirexplore_base+\"/allrels.csv\") as fin:\n",
    "\n",
    "        for line in fin:\n",
    "            aline = line.strip().split(\",\", 1)\n",
    "            aregSyns = aline[1].split(\":\")\n",
    "            regSyns = []\n",
    "            if len(aregSyns) > 1:\n",
    "                regSyns = aregSyns[1].split(\"|\")\n",
    "            \n",
    "            aline[1] = aline[1].split(\":\")[0]           \n",
    "\n",
    "            regClass = aline[0]\n",
    "            regStem = aline[1]\n",
    "\n",
    "            if regClass in renameClass:\n",
    "                regClass = renameClass[regClass]\n",
    "                \n",
    "            stem2class[regStem] = regClass\n",
    "                \n",
    "            for regSyn in regSyns:\n",
    "                if not regSyn in stem2class:\n",
    "                    stem2class[regSyn] = regClass\n",
    "\n",
    "\n",
    "            allRels.append((regClass, regStem))\n",
    "\n",
    "    allStems = [x for x in stem2class]\n",
    "    \n",
    "    return allStems, allRels, stem2class\n",
    "\n",
    "allStems, allRels, stem2class = loadStems()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "scaiBase = mirexplore_base + \"/scai_corpus/\"\n",
    "\n",
    "base = \"TRAIN\"\n",
    "\n",
    "if base == \"TRAIN\":\n",
    "    scaiFile = \"miRNA_train_fixed.xml\"\n",
    "    relsFile = \"scai_train_rels.tsv\"\n",
    "elif base == \"TEST\":\n",
    "    scaiFile = \"miRNA_test_fixed.xml\"\n",
    "    relsFile = \"scai_test_rels.tsv\"\n",
    "    \n",
    "normGeneSymbols= normalize_gene_names(path=scai_base + \"/hgnc_no_withdrawn.syn\")\n",
    "\n",
    "relChecker = SentenceRelationChecker(nlp)\n",
    "\n",
    "correctIdentified = 0\n",
    "incorrectIdentified = 0\n",
    "totalChecks = 0\n",
    "incorrectClass = Counter()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "%aimport MirGeneRelCheck\n",
    "\n",
    "def loadRelations(relsFile):\n",
    "    reldir2new = {}\n",
    "    reldir2new[\"TARGET\"] = \"NEU\"\n",
    "    reldir2new[\"INVCORR\"] = \"NEU\"\n",
    "    reldir2new[\"REGULATE\"] = \"NEU\"\n",
    "    relClasses = {}\n",
    "    with open(os.path.join(scai_base, relsFile)) as fin:\n",
    "        \n",
    "        print(\"Loading relations\", relsFile)\n",
    "        \n",
    "        for lidx, line in enumerate(fin):\n",
    "\n",
    "            if lidx == 0:\n",
    "                continue\n",
    "\n",
    "            line = line.strip().split(\"\\t\")\n",
    "\n",
    "            interactionID = line[0].strip()\n",
    "            relType = line[1].strip()\n",
    "            relRDir = line[2].strip()\n",
    "            relIndirect = line[3].strip()\n",
    "            relPassive = line[4].strip()\n",
    "\n",
    "            if relRDir in reldir2new:\n",
    "                relRDir = reldir2new[relRDir]\n",
    "\n",
    "            relClasses[interactionID] = (relType, relRDir, relIndirect, relPassive)\n",
    "            \n",
    "        print(\"Interactions\", len(relClasses))\n",
    "        print(\"Non NA Interactions\", len([x for x in relClasses if not relClasses[x][0] == \"NA\"]))\n",
    "            \n",
    "    return relClasses\n",
    "\n",
    "def all_subsets(ss):\n",
    "    \n",
    "    return chain(*map(lambda x: combinations(ss, x), range(0, len(ss)+1)))\n",
    "\n",
    "def runCheck(numelems, base, relation_ids=None, original=False):\n",
    "    relChecker = SentenceRelationChecker(nlp, nlp_ent=nlp_ent)\n",
    "    relClassifier = SentenceRelationClassifier(relationsFile)\n",
    "    \n",
    "    if base == \"TRAIN\":\n",
    "        scaiFile = \"miRNA_train_fixed.xml\"\n",
    "        relsFile = \"scai_train_rels.tsv\"\n",
    "    elif base == \"TEST\":\n",
    "        scaiFile = \"miRNA_test_fixed.xml\"\n",
    "        relsFile = \"scai_test_rels.tsv\"\n",
    "        \n",
    "    if original:\n",
    "        if base == \"TRAIN\":\n",
    "            scaiFile = \"miRNA-Train-Corpus.xml\"\n",
    "        elif base == \"TEST\":\n",
    "            scaiFile = \"miRNA-Test-Corpus.xml\"\n",
    "    \n",
    "    relClasses = loadRelations(relsFile)\n",
    "    \n",
    "    check2results = {}\n",
    "    checkSubsets = [x for x in all_subsets(relClassifier.major_checks)]\n",
    "    checkSubsets = [checkSubsets[-1]]\n",
    "    \n",
    "    print(\"Will test {} subsets.\".format(len(checkSubsets)))\n",
    "    \n",
    "    for subset_checks in checkSubsets:\n",
    "    \n",
    "        print(\"Testing checks\", subset_checks)\n",
    "        \n",
    "        #relClassifier.active_checks = subset_checks\n",
    "\n",
    "        #allStems, allRels, stem2class = loadStems()\n",
    "\n",
    "        with open(os.path.join(scaiBase, scaiFile), 'r') as fin:\n",
    "            tree = etree.parse(fin)\n",
    "            root = tree.getroot()\n",
    "            scaiPairs = []\n",
    "            totalChecks = 0\n",
    "            correctIdentified = 0\n",
    "            incorrectIdentified = 0\n",
    "\n",
    "            incorrectClassified = 0\n",
    "            correctClassified = 0\n",
    "            totalClassified = 0\n",
    "            correctInteractClassified = 0\n",
    "            totalValidClassified = 0\n",
    "            totalClassifiable = 0\n",
    "            totalMirGeneDownClassified = 0\n",
    "\n",
    "            errorByDetect = Counter()\n",
    "            classifiedByDetect = Counter()\n",
    "            elemCaseCounter = Counter()\n",
    "            incorrectClass = Counter()\n",
    "            relationFalseCounter = Counter()\n",
    "\n",
    "            classifyTrue = []\n",
    "            classifyPred = []\n",
    "\n",
    "            for elem in root.findall(\".//document\"):\n",
    "\n",
    "                pmid = elem.attrib['origId']\n",
    "\n",
    "                for sentElem in elem:\n",
    "\n",
    "                    allEntities = sentElem.findall(\".//entity\")\n",
    "                    allPairs = sentElem.findall(\".//pair\")\n",
    "\n",
    "                    sentText = sentElem.attrib[\"text\"]\n",
    "\n",
    "                    entId2elem = {}\n",
    "\n",
    "                    for entity in allEntities:\n",
    "                        entId = entity.attrib['id']\n",
    "                        entText = entity.attrib['text']\n",
    "                        entType = entity.attrib['type']\n",
    "                        entOffset = tuple([int(x) for x in entity.attrib['charOffset'].split(\"-\")])\n",
    "\n",
    "                        if entType in [\"Specific_miRNAs\", \"Genes/Proteins\"]:\n",
    "\n",
    "                            if \"Genes\" in entType:\n",
    "                                if entText in normGeneSymbols:\n",
    "                                    entText = normGeneSymbols[entText]\n",
    "                                elif entText.upper() in normGeneSymbols:\n",
    "                                    gene = normGeneSymbols[entText.upper()]\n",
    "                            else:\n",
    "                                try:\n",
    "                                    entText = miRNA(entText).getStringFromParts([miRNAPART.MATURE, miRNAPART.ID, miRNAPART.PRECURSOR])\n",
    "                                except:\n",
    "                                    pass\n",
    "\n",
    "                            entTuple = (entText, entType, (entOffset[0], entOffset[1]+1))\n",
    "                            entId2elem[entId] = entTuple\n",
    "\n",
    "\n",
    "                            sentEntText = sentText[entTuple[2][0]:entTuple[2][1]]\n",
    "\n",
    "                    for pair in allPairs:\n",
    "\n",
    "                        validInteraction = pair.attrib['interaction'].lower() == \"true\"\n",
    "                        pairE1 = pair.attrib['e1']\n",
    "                        pairE2 = pair.attrib['e2']\n",
    "                        relationID = pair.attrib[\"id\"]\n",
    "                        if not relation_ids is None:\n",
    "                            if not relationID in relation_ids:\n",
    "                                continue\n",
    "\n",
    "                        #if pairInt == 'true':\n",
    "                        if pairE1 in entId2elem and pairE2 in entId2elem:\n",
    "\n",
    "                            totalChecks += 1\n",
    "\n",
    "                            e1 = entId2elem[pairE1]\n",
    "                            e2 = entId2elem[pairE2]\n",
    "\n",
    "                            if not e1[1] in [\"Specific_miRNAs\"]:\n",
    "\n",
    "                                tmp=e1\n",
    "                                e1=e2\n",
    "                                e2=tmp\n",
    "\n",
    "\n",
    "                            relRes = relChecker.check_sentence(sentText\n",
    "                                                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": e1[2]}\n",
    "                                                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": e2[2]}\n",
    "                                                                , fix_special_chars=False\n",
    "                                                                , relClassifier=relClassifier.classify, verbose=False\n",
    "                                                                )\n",
    "\n",
    "\n",
    "                            fullsentence = relRes['full_sentence']\n",
    "                            acceptInteraction = relRes['accept_relation']\n",
    "                            \n",
    "                            \n",
    "\n",
    "                            if not acceptInteraction == validInteraction:\n",
    "\n",
    "                                incorrectIdentified += 1\n",
    "                                \n",
    "                                print()\n",
    "                                print()\n",
    "                                print(relationID)\n",
    "                                print(\"IS:\", acceptInteraction)\n",
    "                                print(\"SB:\", validInteraction)\n",
    "                                \n",
    "                                print(\"Repeat Analysis\")\n",
    "                                relResV=relChecker.check_sentence(sentText\n",
    "                                                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": e1[2]}\n",
    "                                                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": e2[2]}\n",
    "                                                                , fix_special_chars=False\n",
    "                                                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                                                )\n",
    "                                \n",
    "                                print(relResV)\n",
    "                                \n",
    "                                for check in ['conj', 'conj_nmod', 'sdp', 'compartment', 'compartment_nmod', 'context', 'entity']:\n",
    "                                    if relResV[\"check_results\"][check] == False:\n",
    "                                        relationFalseCounter[check] += 1\n",
    "\n",
    "                                \n",
    "                            else:\n",
    "                                correctIdentified += 1\n",
    "\n",
    "                            elemCaseCounter[(validInteraction, acceptInteraction)]+=1\n",
    "\n",
    "                            totalClassified += 1\n",
    "\n",
    "                            if validInteraction:# and acceptInteraction:\n",
    "                                #print(relRes[\"check_results\"])   \n",
    "                                totalValidClassified += 1\n",
    "\n",
    "                                foundClasses = relRes[\"check_results\"][\"classification\"] #{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'regulation_keys': set()}\n",
    "                                foundTuple = (foundClasses[\"interaction_dir\"], foundClasses[\"regulation_dir\"])                                \n",
    "                                \n",
    "                                testTuple = relClasses[relationID]\n",
    "\n",
    "                                classifyTrue.append((testTuple[0], testTuple[1]))\n",
    "                                classifyPred.append((foundTuple[0], foundTuple[1]))\n",
    "\n",
    "                                if testTuple[0] == \"MIR_GENE\" and testTuple[1] == \"DOWN\":\n",
    "                                    totalMirGeneDownClassified += 1\n",
    "\n",
    "                                classifiedByDetect[foundClasses[\"reg_detect\"]] += 1\n",
    "\n",
    "                                if testTuple[0] != foundTuple[0] or testTuple[1] != foundTuple[1]:\n",
    "\n",
    "\n",
    "                                    #print(sentText)\n",
    "                                    #print(e1, e2)\n",
    "                                    #print(foundTuple, foundClasses[\"reg_detect\"], \"passive?\", foundClasses[\"passive\"])\n",
    "                                    #print(testTuple)\n",
    "\n",
    "                                    print()\n",
    "                                    print()\n",
    "                                    print()\n",
    "                                    \n",
    "                                    print(relationID)\n",
    "                                    print(\"IS: \", foundTuple)\n",
    "                                    print(\"SB: \", testTuple)\n",
    "                                    print(relRes)\n",
    "                                    \n",
    "                                    relResV=relChecker.check_sentence(sentText\n",
    "                                                                , {\"entity_type\": \"mirna\", \"entity_type_token\": \"e1\", \"entity_location\": e1[2]}\n",
    "                                                                , {\"entity_type\": \"gene\", \"entity_type_token\": \"e2\", \"entity_location\": e2[2]}\n",
    "                                                                , fix_special_chars=False\n",
    "                                                                , relClassifier=relClassifier.classify, verbose=True\n",
    "                                                                )\n",
    "                                \n",
    "                                    print(relResV)\n",
    "                                    \n",
    "                                    print()\n",
    "                                    print()\n",
    "                                    print()\n",
    "\n",
    "                                    errorByDetect[foundClasses[\"reg_detect\"]] += 1\n",
    "                                    \n",
    "                                    \n",
    "\n",
    "                                    incorrectClassified += 1\n",
    "\n",
    "\n",
    "\n",
    "                                    #numelems -= 1\n",
    "                                    #if numelems == 0:\n",
    "                                    #    return\n",
    "                                else:\n",
    "                                    \n",
    "                                    if foundClasses[\"reg_detect_major\"] == \"return\":\n",
    "                                        print(foundClasses)\n",
    "                                        print(sentText)\n",
    "                                        print(e1, e2)\n",
    "                                        print(foundTuple, foundClasses[\"reg_detect\"], \"passive?\", foundClasses[\"passive\"])\n",
    "                                        print(testTuple)\n",
    "\n",
    "                                        print()\n",
    "                                        print()\n",
    "                                        print()\n",
    "                                    correctClassified += 1\n",
    "\n",
    "                                    if acceptInteraction:\n",
    "                                        correctInteractClassified += 1\n",
    "\n",
    "\n",
    "\n",
    "            def printStats(outfile):\n",
    "                print(\"Total:     \", totalChecks, file=outfile)\n",
    "                print(\"Correct:   \", correctIdentified, correctIdentified/totalChecks, file=outfile)\n",
    "                print(\"Incorrect: \", incorrectIdentified, incorrectIdentified/totalChecks, file=outfile)\n",
    "                print(\"classes\", incorrectClass, file=outfile)\n",
    "                \n",
    "                precision = elemCaseCounter[(True, True)] / (elemCaseCounter[(True, True)]+elemCaseCounter[(True, False)])\n",
    "                recall = elemCaseCounter[(True, True)] / (elemCaseCounter[(True, True)]+elemCaseCounter[(False, True)])\n",
    "\n",
    "                f1 = 2* precision * recall / (precision+recall)\n",
    "\n",
    "                specificity = elemCaseCounter[(False, False)] / (elemCaseCounter[(True, False)] + elemCaseCounter[(False, False)])\n",
    "\n",
    "                print(\"precision\", precision, file=outfile)\n",
    "                print(\"recall\", recall, file=outfile)\n",
    "                print(\"specificity\", specificity, file=outfile)\n",
    "                print(\"f1\", f1, file=outfile)\n",
    "                \n",
    "                print(\"relationFalseCounter\", relationFalseCounter, file=outfile)\n",
    "\n",
    "                print(\"Correct classified:     \", correctClassified, correctClassified/totalClassified, correctClassified/totalValidClassified,file=outfile)\n",
    "                print(\"Incorrect classified:   \", incorrectClassified, incorrectClassified/totalClassified, incorrectClassified/totalValidClassified,file=outfile)\n",
    "                print(\"Random classified: \", totalMirGeneDownClassified, totalMirGeneDownClassified / totalClassified, totalMirGeneDownClassified/totalValidClassified, file=outfile)\n",
    "                print(errorByDetect)\n",
    "\n",
    "                print(\"Correct interaction&classified:     \", correctInteractClassified, correctInteractClassified/totalClassified, correctInteractClassified/totalValidClassified,file=outfile)\n",
    "\n",
    "                print(\"Classified by: \", classifiedByDetect)\n",
    "\n",
    "                lClassifyTrue = [\"_\".join(x) for x in classifyTrue]\n",
    "                lClassifyPred = [\"_\".join(x) for x in classifyPred]\n",
    "\n",
    "                allLabels = sorted(set(lClassifyTrue+lClassifyPred))\n",
    "                matrix = multilabel_confusion_matrix(lClassifyTrue, lClassifyPred, labels=allLabels)\n",
    "                print(matrix)\n",
    "                print(classification_report(lClassifyTrue,lClassifyPred))\n",
    "                \n",
    "                return classification_report(lClassifyTrue, lClassifyPred, output_dict=True)\n",
    "\n",
    "            rep = printStats(sys.stdout)\n",
    "            check2results[subset_checks] = rep\n",
    "        #printStats(sys.stderr)\n",
    "    return check2results\n",
    "        \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Setting relfile /mnt/extproj/projekte/textmining/miRExplore/python/allrels.csv\n",
      "Setting relfile /mnt/extproj/projekte/textmining/miRExplore/python/allrels.csv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading relations scai_train_rels.tsv\n",
      "Interactions 397\n",
      "Non NA Interactions 194\n",
      "Will test 1 subsets.\n",
      "Testing checks ('compartment', 'between', 'counts', 'return')\n",
      "\n",
      "\n",
      "miRNA-corp.d1.s8.p0\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case microRNA-200a\n",
      "add special case E-cadherin\n",
      "microRNA-200a 1 E-cadherin 9\n",
      "(u\"Downregulated microRNA-200a in meningiomas promotes tumor growth by reducing E-cadherin and activating the Wnt/beta-catenin signaling pathway.\", 1, 9)\n",
      "geneword E-cadherin mirword microRNA-200a\n",
      "Conjuncts [activating, reducing]\n",
      "neighbour activating the\n",
      "neighbour activating the\n",
      "Conjunction before fc reducing\n",
      "fc add children for reducing\n",
      "Conjuncts [reducing, activating]\n",
      "neighbour reducing E-cadherin\n",
      "neighbour reducing E-cadherin\n",
      "Conjunction before fc activating\n",
      "fc add children for activating\n",
      "Conjunctions ( conj )\n",
      "reducing [reducing, E-cadherin, activating, Wnt/beta-catenin, signaling, pathway]\n",
      "Pathway False\n",
      "Conjuncts [microRNA-200a, meningiomas]\n",
      "neighbour microRNA-200a in\n",
      "neighbour microRNA-200a in\n",
      "Conjunction before fc meningiomas\n",
      "fc add children for meningiomas\n",
      "Conjunctions ( nmod )\n",
      "meningiomas [microRNA-200a, in, meningiomas]\n",
      "SDP sent: Downregulated microRNA-200a in meningiomas promotes tumor growth by reducing E-cadherin and activating the Wnt/beta-catenin signaling pathway.\n",
      "SDP     : [(microRNA-200a, 'NOUN', 'nsubj', False), (promotes, 'VERB', 'ROOT', False), (reducing, 'VERB', 'advcl', False), (E-cadherin, 'NOUN', 'dobj', False)]\n",
      "deepcheck connection rule [by, reducing, E-cadherin, and, activating, the, Wnt/beta-catenin, signaling, pathway]\n",
      "deepcheck conj verb\n",
      "0 [Downregulated, microRNA-200a, in, meningiomas, promotes, tumor, growth, by, reducing, E-cadherin, and, activating, the, Wnt/beta-catenin, signaling, pathway] 16 0 Downregulated microRNA-200a in meningiomas promotes tumor growth by reducing E-cadherin and activating the Wnt/beta-catenin signaling pathway.\n",
      "DeepCheck Conj Subtree\n",
      "[in, meningiomas]\n",
      "deepcheck connection rule [by, reducing, E-cadherin, and, activating, the, Wnt/beta-catenin, signaling, pathway]\n",
      "deepcheck conj verb\n",
      "0 [Downregulated, microRNA-200a, in, meningiomas, promotes, tumor, growth, by, reducing, E-cadherin, and, activating, the, Wnt/beta-catenin, signaling, pathway] 16 0 Downregulated microRNA-200a in meningiomas promotes tumor growth by reducing E-cadherin and activating the Wnt/beta-catenin signaling pathway.\n",
      "Recognized overlapping entities: {Interval(77, 87, ('GENE_OR_GENE_PRODUCT', 'E-cadherin'))}\n",
      "passive? False\n",
      "Counter({'POS': 1, 'NEG': 1})\n",
      "Compartment Text Downregulated microRNA-200a in meningiomas promotes tumor growth by reducing E-cadherin and activating the Wnt/beta-catenin signaling pathway\n",
      "[Downregulated, microRNA-200a, in, meningiomas, promotes, tumor, growth, by, reducing, E-cadherin, and, activating, the, Wnt/beta-catenin, signaling, pathway]\n",
      "TextBetween: Downregulated microRNA-200a in meningiomas promotes tumor growth by reducing E-cadherin and activating the Wnt/beta-catenin signaling pathway\n",
      "infer direction from token set [Downregulated, microRNA-200a, in, meningiomas, promotes, tumor, growth, by, reducing, E-cadherin, and, activating, the, Wnt/beta-catenin, signaling, pathway]\n",
      "stemTypes2Count: Counter({'NEG': 2, 'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e2730>, {'mir': Counter({'NEG': 1}), 'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e2840>, {'mir': Counter({'NEG': 1}), 'gene': Counter({'NEG': 1, 'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(Downregulated, 'NEG'), (promotes, 'POS'), (reducing, 'NEG'), (activating, 'NEU')]\n",
      "Between Tokens [(promotes, 'POS'), (reducing, 'NEG')]\n",
      "counts between final False\n",
      "['Downregulated', 'microRNA-200a', 'in', 'meningiomas', 'promotes', 'tumor', 'growth', 'by', 'reducing', 'E-cadherin', 'and', 'activating']\n",
      "Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "[(Downregulated, 'NEG'), (promotes, 'POS'), (reducing, 'NEG'), (activating, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Downregulated microRNA-200a in meningiomas promotes tumor growth by reducing E-cadherin and activating the Wnt/beta-catenin signaling pathway.\n",
      "full sentence Downregulated <e1>microRNA-200a</e1> in meningiomas promotes tumor growth by reducing <e2>E-cadherin</e2> and activating the Wnt/beta-catenin signaling pathway.\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Downregulated <e1>microRNA-200a</e1> in meningiomas promotes tumor growth by reducing <e2>E-cadherin</e2> and activating the Wnt/beta-catenin signaling pathway.', 'full_sentence': 'Downregulated <e1>microRNA-200a</e1> in meningiomas promotes tumor growth by reducing <e2>E-cadherin</e2> and activating the Wnt/beta-catenin signaling pathway.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (14, 27), 'entity_text': 'microRNA-200a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (77, 87), 'entity_text': 'E-cadherin'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d5.s2.p2\n",
      "IS:  ('GENE_MIR', 'NEU')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Strikingly, cells with high levels of <e2>PKM2</e2> expressed lower levels of <e1>miR-326</e1>, suggestive of endogenous regulation of PKM2 by miR-326.', 'full_sentence': 'Strikingly, cells with high levels of <e2>PKM2</e2> expressed lower levels of <e1>miR-326</e1>, suggestive of endogenous regulation of PKM2 by miR-326.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (38, 42), 'entity_text': 'PKM2'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (69, 76), 'entity_text': 'miR-326'}}\n",
      "add special case PKM2\n",
      "add special case miR-326\n",
      "PKM2 7 miR-326 12\n",
      "(u\"Strikingly, cells with high levels of PKM2 expressed lower levels of miR-326, suggestive of endogenous regulation of PKM2 by miR-326.\", 7, 12)\n",
      "geneword PKM2 mirword miR-326\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [cells, levels]\n",
      "neighbour cells with\n",
      "neighbour cells with\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conjuncts [levels, PKM2]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc PKM2\n",
      "fc add children for PKM2\n",
      "Conjuncts [levels, miR-326]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc miR-326\n",
      "fc add children for miR-326\n",
      "fc add children for miR-326\n",
      "Conjuncts [suggestive, regulation]\n",
      "neighbour suggestive of\n",
      "neighbour suggestive of\n",
      "Conjunction before fc regulation\n",
      "fc add children for regulation\n",
      "fc add children for regulation\n",
      "fc add children for regulation\n",
      "fc add children for regulation\n",
      "Conjuncts [regulation, PKM2]\n",
      "neighbour regulation of\n",
      "neighbour regulation of\n",
      "Conjunction before fc PKM2\n",
      "fc add children for PKM2\n",
      "Conjuncts [regulation, miR-326]\n",
      "neighbour regulation of\n",
      "neighbour regulation of\n",
      "Conjunction before fc miR-326\n",
      "fc add children for miR-326\n",
      "Conunct merge check: levels miR-326\n",
      "PKM2 expressed lower\n",
      "Conjunctions ( nmod )\n",
      "levels [cells, with, high, levels, of, PKM2]\n",
      "miR-326 [levels, of, miR-326, suggestive, of, endogenous, regulation, of, PKM2, by, miR-326]\n",
      "SDP sent: Strikingly, cells with high levels of PKM2 expressed lower levels of miR-326, suggestive of endogenous regulation of PKM2 by miR-326.\n",
      "SDP     : [(PKM2, 'NOUN', 'nmod', False), (levels, 'NOUN', 'nmod', False), (cells, 'NOUN', 'nsubj', False), (expressed, 'VERB', 'ROOT', False), (levels, 'NOUN', 'dobj', False), (miR-326, 'NOUN', 'nmod', False)]\n",
      "amodCheck is true\n",
      "0 [Strikingly, ,, cells, with, high, levels, of, PKM2, expressed, lower, levels, of, miR-326] 13 0 Strikingly, cells with high levels of PKM2 expressed lower levels of miR-326, suggestive of endogenous regulation of PKM2 by miR-326.\n",
      "1 [suggestive, of, endogenous, regulation, of, PKM2, by, miR-326] 8 78 suggestive of endogenous regulation of PKM2 by miR-326.\n",
      "DeepCheck Conj Subtree\n",
      "[with, high, levels, of, PKM2]\n",
      "DeepCheck Conj Subtree\n",
      "[of, PKM2]\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-326, ,, suggestive, of, endogenous, regulation, of, PKM2, by, miR-326]\n",
      "amodCheck is true\n",
      "DeepCheck Conj Subtree\n",
      "[of, endogenous, regulation, of, PKM2, by, miR-326]\n",
      "DeepCheck Conj Subtree\n",
      "[of, PKM2]\n",
      "DeepCheck Conj Subtree\n",
      "[by, miR-326]\n",
      "0 [Strikingly, ,, cells, with, high, levels, of, PKM2, expressed, lower, levels, of, miR-326] 13 0 Strikingly, cells with high levels of PKM2 expressed lower levels of miR-326, suggestive of endogenous regulation of PKM2 by miR-326.\n",
      "1 [suggestive, of, endogenous, regulation, of, PKM2, by, miR-326] 8 78 suggestive of endogenous regulation of PKM2 by miR-326.\n",
      "Recognized overlapping entities: {Interval(38, 42, ('GENE_OR_GENE_PRODUCT', 'PKM2'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text Strikingly , cells with high levels of PKM2 expressed lower levels of miR-326\n",
      "[Strikingly, ,, cells, with, high, levels, of, PKM2, expressed, lower, levels, of, miR-326]\n",
      "TextBetween: Strikingly , cells with high levels of PKM2 expressed lower levels of miR-326\n",
      "infer direction from token set [Strikingly, ,, cells, with, high, levels, of, PKM2, expressed, lower, levels, of, miR-326]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70950>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 3})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbecba9e158>, {'gene': Counter({'NEU': 2}), 'mir': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(levels, 'NEU'), (expressed, 'NEU'), (levels, 'NEU')]\n",
      "Between Tokens [(expressed, 'NEU'), (levels, 'NEU')]\n",
      "counts between final False\n",
      "['levels', 'of', 'PKM2', 'expressed', 'lower', 'levels', 'of', 'miR-326']\n",
      "Counter({'NEU': 3})\n",
      "[(levels, 'NEU'), (expressed, 'NEU'), (levels, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Strikingly, cells with high levels of PKM2 expressed lower levels of miR-326, suggestive of endogenous regulation of PKM2 by miR-326.\n",
      "full sentence Strikingly, cells with high levels of <e2>PKM2</e2> expressed lower levels of <e1>miR-326</e1>, suggestive of endogenous regulation of PKM2 by miR-326.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Strikingly, cells with high levels of <e2>PKM2</e2> expressed lower levels of <e1>miR-326</e1>, suggestive of endogenous regulation of PKM2 by miR-326.', 'full_sentence': 'Strikingly, cells with high levels of <e2>PKM2</e2> expressed lower levels of <e1>miR-326</e1>, suggestive of endogenous regulation of PKM2 by miR-326.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (38, 42), 'entity_text': 'PKM2'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (69, 76), 'entity_text': 'miR-326'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d5.s10.p0\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miR-326\n",
      "add special case PKM2\n",
      "miR-326 11 PKM2 19\n",
      "(u\"To investigate whether it might be a functionally important target of miR-326, we used RNA interference to knockdown PKM2 expression in glioma cells.\", 11, 19)\n",
      "geneword PKM2 mirword miR-326\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [target, miR-326]\n",
      "neighbour target of\n",
      "neighbour target of\n",
      "Conjunction before fc miR-326\n",
      "fc add children for miR-326\n",
      "Conjuncts [used, expression]\n",
      "neighbour used RNA\n",
      "neighbour used RNA\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conjuncts [expression, cells]\n",
      "neighbour expression in\n",
      "neighbour expression in\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conunct merge check: miR-326 expression\n",
      "miR-326, we\n",
      "Conjunctions ( nmod )\n",
      "miR-326 [target, of, miR-326]\n",
      "expression [used, to, knockdown, PKM2, expression, in, glioma, cells]\n",
      "SDP sent: To investigate whether it might be a functionally important target of miR-326, we used RNA interference to knockdown PKM2 expression in glioma cells.\n",
      "SDP     : [(miR-326, 'NOUN', 'nmod', False), (target, 'NOUN', 'ccomp', False), (investigate, 'VERB', 'advcl', False), (used, 'VERB', 'ROOT', False), (expression, 'NOUN', 'nmod', False), (PKM2, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule [To, investigate, whether, it, might, be, a, functionally, important, target, of, miR-326]\n",
      "0 [To, investigate, whether, it, might, be, a, functionally, important, target, of, miR-326, ,, we, used, RNA, interference, to, knockdown, PKM2, expression, in, glioma, cells] 24 0 To investigate whether it might be a functionally important target of miR-326, we used RNA interference to knockdown PKM2 expression in glioma cells.\n",
      "deepcheck connection rule [To, investigate, whether, it, might, be, a, functionally, important, target, of, miR-326]\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-326]\n",
      "DeepCheck Conj Subtree\n",
      "[to, knockdown, PKM2, expression, in, glioma, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[in, glioma, cells]\n",
      "0 [To, investigate, whether, it, might, be, a, functionally, important, target, of, miR-326, ,, we, used, RNA, interference, to, knockdown, PKM2, expression, in, glioma, cells] 24 0 To investigate whether it might be a functionally important target of miR-326, we used RNA interference to knockdown PKM2 expression in glioma cells.\n",
      "Recognized overlapping entities: {Interval(117, 121, ('GENE_OR_GENE_PRODUCT', 'PKM2'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text To investigate whether it might be a functionally important target of miR-326 , we used RNA interference to knockdown PKM2 expression in glioma cells\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene by', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence To investigate whether it might be a functionally important target of miR-326, we used RNA interference to knockdown PKM2 expression in glioma cells.\n",
      "full sentence To investigate whether it might be a functionally important target of <e1>miR-326</e1>, we used RNA interference to knockdown <e2>PKM2</e2> expression in glioma cells.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene by', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'To investigate whether it might be a functionally important target of <e1>miR-326</e1>, we used RNA interference to knockdown <e2>PKM2</e2> expression in glioma cells.', 'full_sentence': 'To investigate whether it might be a functionally important target of <e1>miR-326</e1>, we used RNA interference to knockdown <e2>PKM2</e2> expression in glioma cells.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (70, 77), 'entity_text': 'miR-326'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (117, 121), 'entity_text': 'PKM2'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d19.s2.p1\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case Gli1\n",
      "add special case miR-17/92\n",
      "Gli1 10 miR-17/92 13\n",
      "(u\"In CGNPs, the Shh effector N-myc, but not Gli1, induced miR-17/92 expression.\", 10, 13)\n",
      "geneword Gli1 mirword miR-17/92\n",
      "Conjuncts [Gli1, N-myc]\n",
      "neighbour Gli1 ,\n",
      "neighbour Gli1 induced\n",
      "Conjunction before fc N-myc\n",
      "fc add children for N-myc\n",
      "fc add children for N-myc\n",
      "Conjuncts [N-myc, Gli1]\n",
      "neighbour N-myc ,\n",
      "neighbour N-myc but\n",
      "Conjunction before fc Gli1\n",
      "Conjunctions ( conj )\n",
      "N-myc [Shh, effector, N-myc, Gli1]\n",
      "Conjuncts [induced, CGNPs]\n",
      "neighbour induced miR-17/92\n",
      "neighbour induced miR-17/92\n",
      "Conjunction before fc CGNPs\n",
      "fc add children for CGNPs\n",
      "Conjunctions ( nmod )\n",
      "CGNPs [In, CGNPs, induced]\n",
      "SDP sent: In CGNPs, the Shh effector N-myc, but not Gli1, induced miR-17/92 expression.\n",
      "SDP     : [(Gli1, 'NOUN', 'conj', False), (N-myc, 'NOUN', 'nsubj', True), (induced, 'VERB', 'ROOT', False), (expression, 'NOUN', 'dobj', False), (miR-17/92, 'NOUN', 'compound', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[Gli1]\n",
      "0 [In, CGNPs, ,, the, Shh, effector, N-myc, ,, but, not, Gli1, ,, induced, miR-17/92, expression] 15 0 In CGNPs, the Shh effector N-myc, but not Gli1, induced miR-17/92 expression.\n",
      "DeepCheck Conj Subtree\n",
      "[In, CGNPs]\n",
      "DeepCheck Conj Subtree\n",
      "[Gli1]\n",
      "0 [In, CGNPs, ,, the, Shh, effector, N-myc, ,, but, not, Gli1, ,, induced, miR-17/92, expression] 15 0 In CGNPs, the Shh effector N-myc, but not Gli1, induced miR-17/92 expression.\n",
      "Recognized overlapping entities: {Interval(42, 46, ('GENE_OR_GENE_PRODUCT', 'Gli1'))}\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "Compartment Text In CGNPs , the Shh effector N-myc , but not Gli1 , induced miR-17/92 expression\n",
      "[In, CGNPs, ,, the, Shh, effector, N-myc, ,, but, not, Gli1, ,, induced, miR-17/92, expression]\n",
      "TextBetween: In CGNPs , the Shh effector N-myc , but not Gli1 , induced miR-17/92 expression\n",
      "infer direction from token set [In, CGNPs, ,, the, Shh, effector, N-myc, ,, but, not, Gli1, ,, induced, miR-17/92, expression]\n",
      "stemTypes2Count: Counter({'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954d08>, {'mir': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'POS': 1, 'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954f28>, {'mir': Counter({'POS': 1, 'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(induced, 'POS'), (expression, 'NEU')]\n",
      "Between Tokens [(induced, 'POS')]\n",
      "counts between final False\n",
      "['but', 'not', 'Gli1', ',', 'induced', 'miR-17/92', 'expression']\n",
      "Counter({'POS': 1, 'NEU': 1})\n",
      "[(induced, 'POS'), (expression, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence In CGNPs, the Shh effector N-myc, but not Gli1, induced miR-17/92 expression.\n",
      "full sentence In CGNPs, the Shh effector N-myc, but not <e2>Gli1</e2>, induced <e1>miR-17/92</e1> expression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'In CGNPs, the Shh effector N-myc, but not <e2>Gli1</e2>, induced <e1>miR-17/92</e1> expression.', 'full_sentence': 'In CGNPs, the Shh effector N-myc, but not <e2>Gli1</e2>, induced <e1>miR-17/92</e1> expression.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (42, 46), 'entity_text': 'Gli1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (56, 65), 'entity_text': 'miR-17/92'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d19.s2.p2\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case Shh\n",
      "add special case miR-17/92\n",
      "Shh 4 miR-17/92 13\n",
      "(u\"In CGNPs, the Shh effector N-myc, but not Gli1, induced miR-17/92 expression.\", 4, 13)\n",
      "geneword Shh mirword miR-17/92\n",
      "Conjuncts [Gli1, N-myc]\n",
      "neighbour Gli1 ,\n",
      "neighbour Gli1 induced\n",
      "Conjunction before fc N-myc\n",
      "fc add children for N-myc\n",
      "fc add children for N-myc\n",
      "Conjuncts [N-myc, Gli1]\n",
      "neighbour N-myc ,\n",
      "neighbour N-myc but\n",
      "Conjunction before fc Gli1\n",
      "Conjunctions ( conj )\n",
      "N-myc [Shh, effector, N-myc, Gli1]\n",
      "Conjuncts [induced, CGNPs]\n",
      "neighbour induced miR-17/92\n",
      "neighbour induced miR-17/92\n",
      "Conjunction before fc CGNPs\n",
      "fc add children for CGNPs\n",
      "Conjunctions ( nmod )\n",
      "CGNPs [In, CGNPs, induced]\n",
      "SDP sent: In CGNPs, the Shh effector N-myc, but not Gli1, induced miR-17/92 expression.\n",
      "SDP     : [(Shh, 'NOUN', 'compound', False), (N-myc, 'NOUN', 'nsubj', False), (induced, 'VERB', 'ROOT', False), (expression, 'NOUN', 'dobj', False), (miR-17/92, 'NOUN', 'compound', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[Gli1]\n",
      "0 [In, CGNPs, ,, the, Shh, effector, N-myc, ,, but, not, Gli1, ,, induced, miR-17/92, expression] 15 0 In CGNPs, the Shh effector N-myc, but not Gli1, induced miR-17/92 expression.\n",
      "DeepCheck Conj Subtree\n",
      "[In, CGNPs]\n",
      "DeepCheck Conj Subtree\n",
      "[Gli1]\n",
      "0 [In, CGNPs, ,, the, Shh, effector, N-myc, ,, but, not, Gli1, ,, induced, miR-17/92, expression] 15 0 In CGNPs, the Shh effector N-myc, but not Gli1, induced miR-17/92 expression.\n",
      "Recognized overlapping entities: {Interval(14, 17, ('GENE_OR_GENE_PRODUCT', 'Shh'))}\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "Compartment Text In CGNPs , the Shh effector N-myc , but not Gli1 , induced miR-17/92 expression\n",
      "[In, CGNPs, ,, the, Shh, effector, N-myc, ,, but, not, Gli1, ,, induced, miR-17/92, expression]\n",
      "TextBetween: In CGNPs , the Shh effector N-myc , but not Gli1 , induced miR-17/92 expression\n",
      "infer direction from token set [In, CGNPs, ,, the, Shh, effector, N-myc, ,, but, not, Gli1, ,, induced, miR-17/92, expression]\n",
      "stemTypes2Count: Counter({'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbecba9e158>, {'mir': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'POS': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e2268>, {'gene': Counter({'NEU': 1}), 'mir': Counter({'POS': 1, 'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(effector, 'NEU'), (induced, 'POS'), (expression, 'NEU')]\n",
      "Between Tokens [(effector, 'NEU'), (induced, 'POS')]\n",
      "counts between final False\n",
      "[',', 'the', 'Shh', 'effector', 'N-myc', ',', 'but', 'not', 'Gli1', ',', 'induced', 'miR-17/92', 'expression']\n",
      "Counter({'NEU': 2, 'POS': 1})\n",
      "[(effector, 'NEU'), (induced, 'POS'), (expression, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence In CGNPs, the Shh effector N-myc, but not Gli1, induced miR-17/92 expression.\n",
      "full sentence In CGNPs, the <e2>Shh</e2> effector N-myc, but not Gli1, induced <e1>miR-17/92</e1> expression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'In CGNPs, the <e2>Shh</e2> effector N-myc, but not Gli1, induced <e1>miR-17/92</e1> expression.', 'full_sentence': 'In CGNPs, the <e2>Shh</e2> effector N-myc, but not Gli1, induced <e1>miR-17/92</e1> expression.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (14, 17), 'entity_text': 'Shh'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (56, 65), 'entity_text': 'miR-17/92'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d33.s6.p7\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miR-9\n",
      "add special case CoREST\n",
      "miR-9 0 CoREST 7\n",
      "(u\"miR-9 targets REST and miR-9* targets CoREST.\", 0, 7)\n",
      "geneword CoREST mirword miR-9\n",
      "Conjuncts [*, REST]\n",
      "neighbour * targets\n",
      "neighbour * targets\n",
      "Conjunction before fc REST\n",
      "Conjuncts [REST, *]\n",
      "neighbour REST and\n",
      "neighbour REST and\n",
      "Conjunction before fc *\n",
      "fc add children for *\n",
      "Conjunctions ( conj )\n",
      "REST [REST, miR-9, *]\n",
      "Conjunctions ( nmod )\n",
      "SDP sent: miR-9 targets REST and miR-9* targets CoREST.\n",
      "SDP     : [(miR-9, 'NOUN', 'nsubj', False), (targets, 'VERB', 'nsubj', False), (targets, 'VERB', 'ROOT', False), (CoREST, 'NOUN', 'dobj', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-9, *]\n",
      "0 [miR-9, targets, REST, and, miR-9, *, targets, CoREST] 8 0 miR-9 targets REST and miR-9* targets CoREST.\n",
      "DeepCheck Conj Subtree\n",
      "[miR-9, *]\n",
      "0 [miR-9, targets, REST, and, miR-9, *, targets, CoREST] 8 0 miR-9 targets REST and miR-9* targets CoREST.\n",
      "Recognized overlapping entities: {Interval(38, 44, ('GENE_OR_GENE_PRODUCT', 'CoREST'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text miR-9 targets REST and miR-9 * targets CoREST\n",
      "[miR-9, targets, REST, and, miR-9, *, targets, CoREST]\n",
      "TextBetween: miR-9 targets REST and miR-9 * targets CoREST\n",
      "infer direction from token set [miR-9, targets, REST, and, miR-9, *, targets, CoREST]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954e18>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954bf8>, {'mir': Counter({'NEU': 1}), 'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(targets, 'NEU'), (targets, 'NEU')]\n",
      "Between Tokens [(targets, 'NEU'), (targets, 'NEU')]\n",
      "counts between final False\n",
      "['miR-9', 'targets', 'REST', 'and', 'miR-9', '*', 'targets', 'CoREST']\n",
      "Counter({'NEU': 2})\n",
      "[(targets, 'NEU'), (targets, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence miR-9 targets REST and miR-9* targets CoREST.\n",
      "full sentence <e1>miR-9</e1> targets REST and miR-9* targets <e2>CoREST</e2>.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e1>miR-9</e1> targets REST and miR-9* targets <e2>CoREST</e2>.', 'full_sentence': 'Interestingly, one of these, the bifunctional brain enriched miR-9/miR-9*, targets two components of the REST complex; <e1>miR-9</e1> targets REST and miR-9* targets <e2>CoREST</e2>.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (0, 5), 'entity_text': 'miR-9'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (38, 44), 'entity_text': 'CoREST'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d47.s5.p0\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case BACE1\n",
      "add special case miR-485-5p\n",
      "BACE1 9 miR-485-5p 17\n",
      "(u\"We report here that BACE1-antisense prevents miRNA-induced repression of BACE1 mRNA by masking the binding site for miR-485-5p.\", 9, 17)\n",
      "geneword BACE1 mirword miR-485-5p\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [repression, mRNA]\n",
      "neighbour repression of\n",
      "neighbour repression of\n",
      "Conjunction before fc mRNA\n",
      "fc add children for mRNA\n",
      "fc add children for mRNA\n",
      "Conjuncts [site, miR-485-5p]\n",
      "neighbour site for\n",
      "neighbour site for\n",
      "Conjunction before fc miR-485-5p\n",
      "fc add children for miR-485-5p\n",
      "Conunct merge check: mRNA miR-485-5p\n",
      "mRNA by masking the binding\n",
      "Conjunctions ( nmod )\n",
      "mRNA [repression, of, BACE1, mRNA]\n",
      "miR-485-5p [site, for, miR-485-5p]\n",
      "SDP sent: We report here that BACE1-antisense prevents miRNA-induced repression of BACE1 mRNA by masking the binding site for miR-485-5p.\n",
      "SDP     : [(BACE1, 'NOUN', 'compound', False), (mRNA, 'NOUN', 'nmod', False), (repression, 'NOUN', 'dobj', False), (prevents, 'VERB', 'ccomp', False), (masking, 'VERB', 'advcl', False), (site, 'NOUN', 'dobj', False), (miR-485-5p, 'NOUN', 'nmod', False)]\n",
      "SDP DOBJ RULE\n",
      "deepcheck connection rule noun found prevents BACE1-antisense [that, BACE1-antisense, prevents, miRNA-induced, repression, of, BACE1, mRNA, by, masking, the, binding, site, for, miR-485-5p]\n",
      "deepcheck connection rule [by, masking, the, binding, site, for, miR-485-5p]\n",
      "0 [We, report, here] 3 0 We report here that BACE1-antisense prevents miRNA-induced repression of BACE1 mRNA by masking the binding site for miR-485-5p.\n",
      "1 [BACE1-antisense, prevents, miRNA-induced, repression, of, BACE1, mRNA, by, masking, the, binding, site, for, miR-485-5p] 14 20 BACE1-antisense prevents miRNA-induced repression of BACE1 mRNA by masking the binding site for miR-485-5p.\n",
      "deepcheck connection rule noun found prevents BACE1-antisense [that, BACE1-antisense, prevents, miRNA-induced, repression, of, BACE1, mRNA, by, masking, the, binding, site, for, miR-485-5p]\n",
      "DeepCheck Conj Subtree\n",
      "[of, BACE1, mRNA]\n",
      "deepcheck connection rule [by, masking, the, binding, site, for, miR-485-5p]\n",
      "DeepCheck Conj Subtree\n",
      "[for, miR-485-5p]\n",
      "0 [We, report, here] 3 0 We report here that BACE1-antisense prevents miRNA-induced repression of BACE1 mRNA by masking the binding site for miR-485-5p.\n",
      "1 [BACE1-antisense, prevents, miRNA-induced, repression, of, BACE1, mRNA, by, masking, the, binding, site, for, miR-485-5p] 14 20 BACE1-antisense prevents miRNA-induced repression of BACE1 mRNA by masking the binding site for miR-485-5p.\n",
      "Recognized overlapping entities: {Interval(73, 78, ('GENE_OR_GENE_PRODUCT', 'BACE1'))}\n",
      "passive? False\n",
      "Counter({'NEG': 2})\n",
      "Compartment Text BACE1-antisense prevents miRNA-induced repression of BACE1 mRNA by masking the binding site for miR-485-5p\n",
      "[BACE1-antisense, prevents, miRNA-induced, repression, of, BACE1, mRNA, by, masking, the, binding, site, for, miR-485-5p]\n",
      "TextBetween: BACE1-antisense prevents miRNA-induced repression of BACE1 mRNA by masking the binding site for miR-485-5p\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between gene mir', 'reg_detect_major': 'between'}})\n",
      " hit sentence We report here that BACE1-antisense prevents miRNA-induced repression of BACE1 mRNA by masking the binding site for miR-485-5p.\n",
      "full sentence We report here that BACE1-antisense prevents miRNA-induced repression of <e2>BACE1</e2> mRNA by masking the binding site for <e1>miR-485-5p</e1>.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between gene mir', 'reg_detect_major': 'between'}}, 'tagged_sent': 'We report here that BACE1-antisense prevents miRNA-induced repression of <e2>BACE1</e2> mRNA by masking the binding site for <e1>miR-485-5p</e1>.', 'full_sentence': 'RESULTS; We report here that BACE1-antisense prevents miRNA-induced repression of <e2>BACE1</e2> mRNA by masking the binding site for <e1>miR-485-5p</e1>.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (73, 78), 'entity_text': 'BACE1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (116, 126), 'entity_text': 'miR-485-5p'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d47.s8.p3\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case BACE1\n",
      "add special case miR-485-5p\n",
      "BACE1 9 miR-485-5p 22\n",
      "(u\"We observed opposing effects of BACE1-antisense and miR-485-5p on BACE1 protein in vitro and showed that Locked Nucleic Acid-antimiR mediated knockdown of miR-485-5p as well as BACE1-antisense over-expression can prevent the miRNA-induced BACE1 suppression.\", 9, 22)\n",
      "geneword BACE1 mirword miR-485-5p\n",
      "Conjuncts [showed, observed]\n",
      "neighbour showed that\n",
      "neighbour showed that\n",
      "Conjunction before fc observed\n",
      "fc add children for observed\n",
      "Conjuncts [miR-485-5p, BACE1-antisense]\n",
      "neighbour miR-485-5p on\n",
      "neighbour miR-485-5p on\n",
      "Conjunction before fc BACE1-antisense\n",
      "fc add children for BACE1-antisense\n",
      "fc add children for BACE1-antisense\n",
      "Conjuncts [BACE1-antisense, miR-485-5p]\n",
      "neighbour BACE1-antisense and\n",
      "neighbour BACE1-antisense and\n",
      "Conjunction before fc miR-485-5p\n",
      "Conjuncts [observed, showed]\n",
      "neighbour observed opposing\n",
      "neighbour observed opposing\n",
      "Conjunction before fc showed\n",
      "Conjunctions ( conj )\n",
      "observed [observed, opposing, effects, of, BACE1-antisense, miR-485-5p, on, BACE1, protein, showed]\n",
      "Conjuncts [effects, BACE1-antisense]\n",
      "neighbour effects of\n",
      "neighbour effects of\n",
      "Conjunction before fc BACE1-antisense\n",
      "fc add children for BACE1-antisense\n",
      "fc add children for BACE1-antisense\n",
      "Conjuncts [BACE1-antisense, protein]\n",
      "neighbour BACE1-antisense and\n",
      "neighbour BACE1-antisense and\n",
      "Conjunction before fc protein\n",
      "fc add children for protein\n",
      "fc add children for protein\n",
      "Conjuncts [knockdown, miR-485-5p]\n",
      "neighbour knockdown of\n",
      "neighbour knockdown of\n",
      "Conjunction before fc miR-485-5p\n",
      "fc add children for miR-485-5p\n",
      "Conunct merge check: BACE1-antisense miR-485-5p\n",
      "protein in vitro and showed that Locked Nucleic Acid-antimiR mediated\n",
      "Conjunctions ( nmod )\n",
      "BACE1-antisense [effects, of, BACE1-antisense, on, BACE1, protein]\n",
      "miR-485-5p [knockdown, of, miR-485-5p]\n",
      "SDP sent: We observed opposing effects of BACE1-antisense and miR-485-5p on BACE1 protein in vitro and showed that Locked Nucleic Acid-antimiR mediated knockdown of miR-485-5p as well as BACE1-antisense over-expression can prevent the miRNA-induced BACE1 suppression.\n",
      "SDP     : [(BACE1, 'NOUN', 'compound', False), (protein, 'NOUN', 'nmod', False), (BACE1-antisense, 'NOUN', 'nmod', False), (effects, 'NOUN', 'dobj', False), (observed, 'VERB', 'ROOT', True), (showed, 'VERB', 'conj', False), (mediated, 'ADJ', 'ccomp', False), (knockdown, 'NOUN', 'dobj', False), (miR-485-5p, 'NOUN', 'nmod', False)]\n",
      "SDP DOBJ RULE\n",
      "DeepCheck Conj Subtree\n",
      "[miR-485-5p]\n",
      "deepcheck conj verb\n",
      "deepcheck connection rule noun found prevent over-expression [BACE1-antisense, over-expression, can, prevent, the, miRNA-induced, BACE1, suppression]\n",
      "0 [We, observed, opposing, effects, of, BACE1-antisense, and, miR-485-5p, on, BACE1, protein, in, vitro, and, showed, that, Locked, Nucleic, Acid-antimiR, mediated, knockdown, of, miR-485-5p, as, well] 25 0 We observed opposing effects of BACE1-antisense and miR-485-5p on BACE1 protein in vitro and showed that Locked Nucleic Acid-antimiR mediated knockdown of miR-485-5p as well as BACE1-antisense over-expression can prevent the miRNA-induced BACE1 suppression.\n",
      "1 [BACE1-antisense, over-expression, can, prevent, the, miRNA-induced, BACE1, suppression] 8 177 BACE1-antisense over-expression can prevent the miRNA-induced BACE1 suppression.\n",
      "DeepCheck Conj Subtree\n",
      "[of, BACE1-antisense, and, miR-485-5p, on, BACE1, protein]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-485-5p]\n",
      "DeepCheck Conj Subtree\n",
      "[on, BACE1, protein]\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-485-5p, as, well, as]\n",
      "deepcheck connection rule noun found prevent over-expression [BACE1-antisense, over-expression, can, prevent, the, miRNA-induced, BACE1, suppression]\n",
      "0 [We, observed, opposing, effects, of, BACE1-antisense, and, miR-485-5p, on, BACE1, protein, in, vitro, and, showed, that, Locked, Nucleic, Acid-antimiR, mediated, knockdown, of, miR-485-5p, as, well] 25 0 We observed opposing effects of BACE1-antisense and miR-485-5p on BACE1 protein in vitro and showed that Locked Nucleic Acid-antimiR mediated knockdown of miR-485-5p as well as BACE1-antisense over-expression can prevent the miRNA-induced BACE1 suppression.\n",
      "1 [BACE1-antisense, over-expression, can, prevent, the, miRNA-induced, BACE1, suppression] 8 177 BACE1-antisense over-expression can prevent the miRNA-induced BACE1 suppression.\n",
      "Recognized overlapping entities: {Interval(66, 71, ('GENE_OR_GENE_PRODUCT', 'BACE1'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text We observed opposing effects of BACE1-antisense and miR-485-5p on BACE1 protein in vitro and showed that Locked Nucleic Acid-antimiR mediated knockdown of miR-485-5p as well\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'compartment gene mir by', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence We observed opposing effects of BACE1-antisense and miR-485-5p on BACE1 protein in vitro and showed that Locked Nucleic Acid-antimiR mediated knockdown of miR-485-5p as well as BACE1-antisense over-expression can prevent the miRNA-induced BACE1 suppression.\n",
      "full sentence We observed opposing effects of BACE1-antisense and miR-485-5p on <e2>BACE1</e2> protein in vitro and showed that Locked Nucleic Acid-antimiR mediated knockdown of <e1>miR-485-5p</e1> as well as BACE1-antisense over-expression can prevent the miRNA-induced BACE1 suppression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'compartment gene mir by', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'We observed opposing effects of BACE1-antisense and miR-485-5p on <e2>BACE1</e2> protein in vitro and showed that Locked Nucleic Acid-antimiR mediated knockdown of <e1>miR-485-5p</e1> as well as BACE1-antisense over-expression can prevent the miRNA-induced BACE1 suppression.', 'full_sentence': 'We observed opposing effects of BACE1-antisense and miR-485-5p on <e2>BACE1</e2> protein in vitro and showed that Locked Nucleic Acid-antimiR mediated knockdown of <e1>miR-485-5p</e1> as well as BACE1-antisense over-expression can prevent the miRNA-induced BACE1 suppression.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (66, 71), 'entity_text': 'BACE1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (155, 165), 'entity_text': 'miR-485-5p'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d48.s5.p0\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Antagomir-mediated prevention of <e1>mir-145</e1> expression led to an increased protein expression of its downstream target <e2>superoxide dismutase-2</e2> in the postischemic brain.', 'full_sentence': 'Antagomir-mediated prevention of <e1>mir-145</e1> expression led to an increased protein expression of its downstream target <e2>superoxide dismutase-2</e2> in the postischemic brain.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (33, 40), 'entity_text': 'mir-145'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (116, 138), 'entity_text': 'superoxide dismutase-2'}}\n",
      "add special case mir-145\n",
      "add special case superoxide dismutase-2\n",
      "mir-145 3 dismutase-2 16\n",
      "(u\"Antagomir-mediated prevention of mir-145 expression led to an increased protein expression of its downstream target superoxide dismutase-2 in the postischemic brain.\", 3, 16)\n",
      "geneword dismutase-2 mirword mir-145\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [prevention, expression]\n",
      "neighbour prevention of\n",
      "neighbour prevention of\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conjuncts [led, expression]\n",
      "neighbour led to\n",
      "neighbour led to\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conjuncts [expression, dismutase-2]\n",
      "neighbour expression of\n",
      "neighbour expression of\n",
      "Conjunction before fc dismutase-2\n",
      "fc add children for dismutase-2\n",
      "fc add children for dismutase-2\n",
      "fc add children for dismutase-2\n",
      "fc add children for dismutase-2\n",
      "Conjuncts [expression, brain]\n",
      "neighbour expression of\n",
      "neighbour expression of\n",
      "Conjunction before fc brain\n",
      "fc add children for brain\n",
      "fc add children for brain\n",
      "Conunct merge check: expression expression\n",
      "expression\n",
      "Conjunctions ( nmod )\n",
      "expression [prevention, of, mir-145, expression]\n",
      "expression [led, to, increased, protein, expression, of, downstream, target, superoxide, dismutase-2, in, postischemic, brain]\n",
      "SDP sent: Antagomir-mediated prevention of mir-145 expression led to an increased protein expression of its downstream target superoxide dismutase-2 in the postischemic brain.\n",
      "SDP     : [(mir-145, 'NOUN', 'compound', False), (expression, 'NOUN', 'nmod', False), (prevention, 'NOUN', 'nsubj', False), (led, 'VERB', 'ROOT', False), (expression, 'NOUN', 'nmod', False), (dismutase-2, 'NOUN', 'nmod', False)]\n",
      "0 [Antagomir-mediated, prevention, of, mir-145, expression, led, to, an, increased, protein, expression, of, its, downstream, target, superoxide, dismutase-2, in, the, postischemic, brain] 21 0 Antagomir-mediated prevention of mir-145 expression led to an increased protein expression of its downstream target superoxide dismutase-2 in the postischemic brain.\n",
      "DeepCheck Conj Subtree\n",
      "[of, mir-145, expression]\n",
      "DeepCheck Conj Subtree\n",
      "[to, an, increased, protein, expression, of, its, downstream, target, superoxide, dismutase-2, in, the, postischemic, brain]\n",
      "DeepCheck Conj Subtree\n",
      "[of, its, downstream, target, superoxide, dismutase-2]\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, postischemic, brain]\n",
      "0 [Antagomir-mediated, prevention, of, mir-145, expression, led, to, an, increased, protein, expression, of, its, downstream, target, superoxide, dismutase-2, in, the, postischemic, brain] 21 0 Antagomir-mediated prevention of mir-145 expression led to an increased protein expression of its downstream target superoxide dismutase-2 in the postischemic brain.\n",
      "Recognized overlapping entities: {Interval(116, 138, ('GENE_OR_GENE_PRODUCT', 'superoxide dismutase-2'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text Antagomir-mediated prevention of mir-145 expression led to an increased protein expression of its downstream target superoxide dismutase-2 in the postischemic brain\n",
      "[Antagomir-mediated, prevention, of, mir-145, expression, led, to, an, increased, protein, expression, of, its, downstream, target, superoxide, dismutase-2, in, the, postischemic, brain]\n",
      "TextBetween: Antagomir-mediated prevention of mir-145 expression led to an increased protein expression of its downstream target superoxide dismutase-2 in the postischemic brain\n",
      "infer direction from token set [Antagomir-mediated, prevention, of, mir-145, expression, led, to, an, increased, protein, expression, of, its, downstream, target, superoxide, dismutase-2, in, the, postischemic, brain]\n",
      "stemTypes2Count: Counter({'NEG': 1, 'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954bf8>, {'mir': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 3, 'NEG': 1, 'POS': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70510>, {'mir': Counter({'NEG': 1, 'NEU': 1}), 'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(prevention, 'NEG'), (expression, 'NEU'), (increased, 'POS'), (expression, 'NEU'), (target, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Antagomir-mediated prevention of mir-145 expression led to an increased protein expression of its downstream target superoxide dismutase-2 in the postischemic brain.\n",
      "full sentence Antagomir-mediated prevention of <e1>mir-145</e1> expression led to an increased protein expression of its downstream target <e2>superoxide dismutase-2</e2> in the postischemic brain.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Antagomir-mediated prevention of <e1>mir-145</e1> expression led to an increased protein expression of its downstream target <e2>superoxide dismutase-2</e2> in the postischemic brain.', 'full_sentence': 'Antagomir-mediated prevention of <e1>mir-145</e1> expression led to an increased protein expression of its downstream target <e2>superoxide dismutase-2</e2> in the postischemic brain.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (33, 40), 'entity_text': 'mir-145'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (116, 138), 'entity_text': 'superoxide dismutase-2'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d51.s3.p0\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case miR-107\n",
      "add special case BACE1\n",
      "miR-107 1 BACE1 3\n",
      "(u\"Adjusted miR-107 and BACE1 mRNA levels tended to correlate negatively (trend with regression P< 0.07).\", 1, 3)\n",
      "geneword BACE1 mirword miR-107\n",
      "Conjuncts [levels, miR-107]\n",
      "neighbour levels tended\n",
      "neighbour levels tended\n",
      "Conjunction before fc miR-107\n",
      "fc add children for miR-107\n",
      "Conjuncts [miR-107, levels]\n",
      "neighbour miR-107 and\n",
      "neighbour miR-107 and\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conjunctions ( conj )\n",
      "miR-107 [Adjusted, miR-107, BACE1, mRNA, levels]\n",
      "achildren {miR-107}\n",
      "achildren False\n",
      "Conjuncts [trend, <]\n",
      "neighbour trend with\n",
      "neighbour trend with\n",
      "Conjunction before fc <\n",
      "fc add children for <\n",
      "fc add children for <\n",
      "fc add children for <\n",
      "fc add children for <\n",
      "Conjunctions ( nmod )\n",
      "< [trend, with, regression, P, <, 0.07]\n",
      "SDP sent: Adjusted miR-107 and BACE1 mRNA levels tended to correlate negatively (trend with regression P< 0.07).\n",
      "SDP     : [(miR-107, 'NOUN', 'nsubj', False), (tended, 'VERB', 'ROOT', False), (miR-107, 'NOUN', 'nsubj', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[BACE1, mRNA, levels]\n",
      "deepcheck connection rule [to, correlate, negatively, (, trend, with, regression, P, <, 0.07, )]\n",
      "0 [Adjusted, miR-107, and, BACE1, mRNA, levels, tended, to, correlate, negatively, (, trend, with, regression, P, <, 0.07] 17 0 Adjusted miR-107 and BACE1 mRNA levels tended to correlate negatively (trend with regression P< 0.07).\n",
      "DeepCheck Conj Subtree\n",
      "[BACE1, mRNA, levels]\n",
      "deepcheck connection rule [to, correlate, negatively, (, trend, with, regression, P, <, 0.07, )]\n",
      "DeepCheck Conj Subtree\n",
      "[with, regression, P, <, 0.07]\n",
      "0 [Adjusted, miR-107, and, BACE1, mRNA, levels, tended, to, correlate, negatively, (, trend, with, regression, P, <, 0.07] 17 0 Adjusted miR-107 and BACE1 mRNA levels tended to correlate negatively (trend with regression P< 0.07).\n",
      "Recognized overlapping entities: {Interval(21, 26, ('GENE_OR_GENE_PRODUCT', 'BACE1'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text Adjusted miR-107 and BACE1 mRNA levels tended to correlate negatively ( trend with regression P < 0.07\n",
      "[Adjusted, miR-107, and, BACE1, mRNA, levels, tended, to, correlate, negatively, (, trend, with, regression, P, <, 0.07]\n",
      "TextBetween: Adjusted miR-107 and BACE1 mRNA levels tended to correlate negatively ( trend with regression P < 0.07\n",
      "infer direction from token set [Adjusted, miR-107, and, BACE1, mRNA, levels, tended, to, correlate, negatively, (, trend, with, regression, P, <, 0.07]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e2400>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e2598>, {'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(levels, 'NEU')]\n",
      "Between Tokens []\n",
      "counts between final False\n",
      "['Adjusted', 'miR-107', 'and', 'BACE1', 'mRNA', 'levels']\n",
      "Counter({'NEU': 1})\n",
      "[(levels, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Adjusted miR-107 and BACE1 mRNA levels tended to correlate negatively (trend with regression P< 0.07).\n",
      "full sentence Adjusted <e1>miR-107</e1> and <e2>BACE1</e2> mRNA levels tended to correlate negatively (trend with regression P< 0.07).\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Adjusted <e1>miR-107</e1> and <e2>BACE1</e2> mRNA levels tended to correlate negatively (trend with regression P< 0.07).', 'full_sentence': 'Adjusted <e1>miR-107</e1> and <e2>BACE1</e2> mRNA levels tended to correlate negatively (trend with regression P< 0.07).', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (9, 16), 'entity_text': 'miR-107'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (21, 26), 'entity_text': 'BACE1'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d56.s5.p0\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miR-126\n",
      "add special case GALC\n",
      "miR-126 2 GALC 12\n",
      "(u\"By incorporating miR-126 target sequences into a GALC-expressing vector, we suppressed GALC expression in HSCs while maintaining robust expression in mature hematopoietic cells.\", 2, 12)\n",
      "geneword GALC mirword miR-126\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [sequences, vector]\n",
      "neighbour sequences into\n",
      "neighbour sequences into\n",
      "Conjunction before fc vector\n",
      "fc add children for vector\n",
      "fc add children for vector\n",
      "Conjuncts [expression, HSCs]\n",
      "neighbour expression in\n",
      "neighbour expression in\n",
      "Conjunction before fc HSCs\n",
      "fc add children for HSCs\n",
      "Conjuncts [expression, cells]\n",
      "neighbour expression in\n",
      "neighbour expression in\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conunct merge check: vector HSCs\n",
      "vector, we suppressed GALC\n",
      "Conunct merge check: HSCs cells\n",
      "HSCs while maintaining robust\n",
      "Conjunctions ( nmod )\n",
      "vector [sequences, into, GALC-expressing, vector]\n",
      "HSCs [expression, in, HSCs]\n",
      "cells [expression, in, mature, hematopoietic, cells]\n",
      "SDP sent: By incorporating miR-126 target sequences into a GALC-expressing vector, we suppressed GALC expression in HSCs while maintaining robust expression in mature hematopoietic cells.\n",
      "SDP     : [(miR-126, 'NOUN', 'amod', False), (sequences, 'NOUN', 'dobj', False), (incorporating, 'VERB', 'advcl', False), (suppressed, 'VERB', 'ROOT', False), (expression, 'NOUN', 'dobj', False), (GALC, 'NOUN', 'compound', False)]\n",
      "SDP DOBJ RULE\n",
      "deepcheck connection rule [By, incorporating, miR-126, target, sequences, into, a, GALC-expressing, vector]\n",
      "0 [By, incorporating, miR-126, target, sequences, into, a, GALC-expressing, vector, ,, we, suppressed, GALC, expression, in, HSCs] 16 0 By incorporating miR-126 target sequences into a GALC-expressing vector, we suppressed GALC expression in HSCs while maintaining robust expression in mature hematopoietic cells.\n",
      "1 [maintaining, robust, expression, in, mature, hematopoietic, cells] 7 117 maintaining robust expression in mature hematopoietic cells.\n",
      "deepcheck connection rule [By, incorporating, miR-126, target, sequences, into, a, GALC-expressing, vector]\n",
      "DeepCheck Conj Subtree\n",
      "[into, a, GALC-expressing, vector]\n",
      "DeepCheck Conj Subtree\n",
      "[in, HSCs]\n",
      "DeepCheck Conj Subtree\n",
      "[in, mature, hematopoietic, cells]\n",
      "0 [By, incorporating, miR-126, target, sequences, into, a, GALC-expressing, vector, ,, we, suppressed, GALC, expression, in, HSCs] 16 0 By incorporating miR-126 target sequences into a GALC-expressing vector, we suppressed GALC expression in HSCs while maintaining robust expression in mature hematopoietic cells.\n",
      "1 [maintaining, robust, expression, in, mature, hematopoietic, cells] 7 117 maintaining robust expression in mature hematopoietic cells.\n",
      "Recognized overlapping entities: {Interval(87, 91, ('SIMPLE_CHEMICAL', 'GALC'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text By incorporating miR-126 target sequences into a GALC-expressing vector , we suppressed GALC expression in HSCs\n",
      "[By, incorporating, miR-126, target, sequences, into, a, GALC-expressing, vector, ,, we, suppressed, GALC, expression, in, HSCs]\n",
      "TextBetween: By incorporating miR-126 target sequences into a GALC-expressing vector , we suppressed GALC expression in HSCs\n",
      "infer direction from token set [By, incorporating, miR-126, target, sequences, into, a, GALC-expressing, vector, ,, we, suppressed, GALC, expression, in, HSCs]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbecba9e158>, {'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbecba9e048>, {'mir': Counter({'NEU': 1}), 'gene': Counter({'NEG': 1, 'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(target, 'NEU'), (suppressed, 'NEG'), (expression, 'NEU')]\n",
      "Between Tokens [(target, 'NEU'), (suppressed, 'NEG')]\n",
      "counts between final False\n",
      "['By', 'incorporating', 'miR-126', 'target', 'sequences', 'into', 'a', 'GALC-expressing', 'vector', ',', 'we', 'suppressed', 'GALC', 'expression', 'in']\n",
      "Counter({'NEU': 2, 'NEG': 1})\n",
      "[(target, 'NEU'), (suppressed, 'NEG'), (expression, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence By incorporating miR-126 target sequences into a GALC-expressing vector, we suppressed GALC expression in HSCs while maintaining robust expression in mature hematopoietic cells.\n",
      "full sentence By incorporating <e1>miR-126</e1> target sequences into a GALC-expressing vector, we suppressed <e2>GALC</e2> expression in HSCs while maintaining robust expression in mature hematopoietic cells.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'By incorporating <e1>miR-126</e1> target sequences into a GALC-expressing vector, we suppressed <e2>GALC</e2> expression in HSCs while maintaining robust expression in mature hematopoietic cells.', 'full_sentence': 'By incorporating <e1>miR-126</e1> target sequences into a GALC-expressing vector, we suppressed <e2>GALC</e2> expression in HSCs while maintaining robust expression in mature hematopoietic cells.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (17, 24), 'entity_text': 'miR-126'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (87, 91), 'entity_text': 'GALC'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d56.s5.p1\n",
      "IS:  ('MIR_GENE', 'NEU')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'By incorporating <e1>miR-126</e1> target sequences into a GALC-expressing vector, we suppressed <e2>GALC</e2> expression in HSCs while maintaining robust expression in mature hematopoietic cells.', 'full_sentence': 'By incorporating <e1>miR-126</e1> target sequences into a GALC-expressing vector, we suppressed <e2>GALC</e2> expression in HSCs while maintaining robust expression in mature hematopoietic cells.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (17, 24), 'entity_text': 'miR-126'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (87, 91), 'entity_text': 'GALC'}}\n",
      "add special case miR-126\n",
      "add special case GALC\n",
      "miR-126 2 GALC 12\n",
      "(u\"By incorporating miR-126 target sequences into a GALC-expressing vector, we suppressed GALC expression in HSCs while maintaining robust expression in mature hematopoietic cells.\", 2, 12)\n",
      "geneword GALC mirword miR-126\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [sequences, vector]\n",
      "neighbour sequences into\n",
      "neighbour sequences into\n",
      "Conjunction before fc vector\n",
      "fc add children for vector\n",
      "fc add children for vector\n",
      "Conjuncts [expression, HSCs]\n",
      "neighbour expression in\n",
      "neighbour expression in\n",
      "Conjunction before fc HSCs\n",
      "fc add children for HSCs\n",
      "Conjuncts [expression, cells]\n",
      "neighbour expression in\n",
      "neighbour expression in\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conunct merge check: vector HSCs\n",
      "vector, we suppressed GALC\n",
      "Conunct merge check: HSCs cells\n",
      "HSCs while maintaining robust\n",
      "Conjunctions ( nmod )\n",
      "vector [sequences, into, GALC-expressing, vector]\n",
      "HSCs [expression, in, HSCs]\n",
      "cells [expression, in, mature, hematopoietic, cells]\n",
      "SDP sent: By incorporating miR-126 target sequences into a GALC-expressing vector, we suppressed GALC expression in HSCs while maintaining robust expression in mature hematopoietic cells.\n",
      "SDP     : [(miR-126, 'NOUN', 'amod', False), (sequences, 'NOUN', 'dobj', False), (incorporating, 'VERB', 'advcl', False), (suppressed, 'VERB', 'ROOT', False), (expression, 'NOUN', 'dobj', False), (GALC, 'NOUN', 'compound', False)]\n",
      "SDP DOBJ RULE\n",
      "deepcheck connection rule [By, incorporating, miR-126, target, sequences, into, a, GALC-expressing, vector]\n",
      "0 [By, incorporating, miR-126, target, sequences, into, a, GALC-expressing, vector, ,, we, suppressed, GALC, expression, in, HSCs] 16 0 By incorporating miR-126 target sequences into a GALC-expressing vector, we suppressed GALC expression in HSCs while maintaining robust expression in mature hematopoietic cells.\n",
      "1 [maintaining, robust, expression, in, mature, hematopoietic, cells] 7 117 maintaining robust expression in mature hematopoietic cells.\n",
      "deepcheck connection rule [By, incorporating, miR-126, target, sequences, into, a, GALC-expressing, vector]\n",
      "DeepCheck Conj Subtree\n",
      "[into, a, GALC-expressing, vector]\n",
      "DeepCheck Conj Subtree\n",
      "[in, HSCs]\n",
      "DeepCheck Conj Subtree\n",
      "[in, mature, hematopoietic, cells]\n",
      "0 [By, incorporating, miR-126, target, sequences, into, a, GALC-expressing, vector, ,, we, suppressed, GALC, expression, in, HSCs] 16 0 By incorporating miR-126 target sequences into a GALC-expressing vector, we suppressed GALC expression in HSCs while maintaining robust expression in mature hematopoietic cells.\n",
      "1 [maintaining, robust, expression, in, mature, hematopoietic, cells] 7 117 maintaining robust expression in mature hematopoietic cells.\n",
      "Recognized overlapping entities: {Interval(87, 91, ('SIMPLE_CHEMICAL', 'GALC'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text By incorporating miR-126 target sequences into a GALC-expressing vector , we suppressed GALC expression in HSCs\n",
      "[By, incorporating, miR-126, target, sequences, into, a, GALC-expressing, vector, ,, we, suppressed, GALC, expression, in, HSCs]\n",
      "TextBetween: By incorporating miR-126 target sequences into a GALC-expressing vector , we suppressed GALC expression in HSCs\n",
      "infer direction from token set [By, incorporating, miR-126, target, sequences, into, a, GALC-expressing, vector, ,, we, suppressed, GALC, expression, in, HSCs]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70730>, {'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954f28>, {'mir': Counter({'NEU': 1}), 'gene': Counter({'NEG': 1, 'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(target, 'NEU'), (suppressed, 'NEG'), (expression, 'NEU')]\n",
      "Between Tokens [(target, 'NEU'), (suppressed, 'NEG')]\n",
      "counts between final False\n",
      "['By', 'incorporating', 'miR-126', 'target', 'sequences', 'into', 'a', 'GALC-expressing', 'vector', ',', 'we', 'suppressed', 'GALC', 'expression', 'in']\n",
      "Counter({'NEU': 2, 'NEG': 1})\n",
      "[(target, 'NEU'), (suppressed, 'NEG'), (expression, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence By incorporating miR-126 target sequences into a GALC-expressing vector, we suppressed GALC expression in HSCs while maintaining robust expression in mature hematopoietic cells.\n",
      "full sentence By incorporating <e1>miR-126</e1> target sequences into a GALC-expressing vector, we suppressed <e2>GALC</e2> expression in HSCs while maintaining robust expression in mature hematopoietic cells.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'By incorporating <e1>miR-126</e1> target sequences into a GALC-expressing vector, we suppressed <e2>GALC</e2> expression in HSCs while maintaining robust expression in mature hematopoietic cells.', 'full_sentence': 'By incorporating <e1>miR-126</e1> target sequences into a GALC-expressing vector, we suppressed <e2>GALC</e2> expression in HSCs while maintaining robust expression in mature hematopoietic cells.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (17, 24), 'entity_text': 'miR-126'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (87, 91), 'entity_text': 'GALC'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d68.s0.p7\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e1>miR-29a</e1> decreased in Alzheimer disease brains targets <e2>neurone navigator 3</e2>.  AIMS.', 'full_sentence': 'Aberrant microRNA expression in the brains of neurodegenerative diseases; <e1>miR-29a</e1> decreased in Alzheimer disease brains targets <e2>neurone navigator 3</e2>.  AIMS.;  MicroRNAs (miRNAs) are small non-coding RNAs that regulate translational repression of target mRNAs. ', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (0, 7), 'entity_text': 'miR-29a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (54, 73), 'entity_text': 'neurone navigator 3'}}\n",
      "add special case miR-29a\n",
      "add special case neurone navigator 3\n",
      "miR-29a 0 3 9\n",
      "(u\"miR-29a decreased in Alzheimer disease brains targets neurone navigator 3.  AIMS.\", 0, 9)\n",
      "geneword 3 mirword miR-29a\n",
      "Conjuncts [targets, decreased]\n",
      "neighbour targets neurone\n",
      "neighbour targets neurone\n",
      "Conjunction before fc decreased\n",
      "fc add children for decreased\n",
      "Conjuncts [decreased, targets]\n",
      "neighbour decreased in\n",
      "neighbour decreased in\n",
      "Conjunction before fc targets\n",
      "fc add children for targets\n",
      "Conjunctions ( conj )\n",
      "decreased [decreased, in, Alzheimer, disease, brains, targets, neurone, navigator, 3]\n",
      "Conjuncts [decreased, brains]\n",
      "neighbour decreased in\n",
      "neighbour decreased in\n",
      "Conjunction before fc brains\n",
      "fc add children for brains\n",
      "fc add children for brains\n",
      "fc add children for brains\n",
      "Conjunctions ( nmod )\n",
      "brains [decreased, in, Alzheimer, disease, brains]\n",
      "SDP sent: miR-29a decreased in Alzheimer disease brains targets neurone navigator 3.  AIMS.\n",
      "SDP     : [(miR-29a, 'NOUN', 'nsubj', False), (decreased, 'VERB', 'ROOT', True), (targets, 'VERB', 'conj', False), (navigator, 'NOUN', 'dobj', False), (3, 'NUM', 'nummod', False)]\n",
      "deepcheck connection rule noun found targets neurone [targets, neurone, navigator, 3]\n",
      "deepcheck conj verb\n",
      "0 [miR-29a, decreased, in, Alzheimer, disease, brains, targets, neurone, navigator, 3, .,  , AIMS] 13 0 miR-29a decreased in Alzheimer disease brains targets neurone navigator 3.  AIMS.\n",
      "DeepCheck Conj Subtree\n",
      "[in, Alzheimer, disease, brains]\n",
      "deepcheck connection rule noun found targets neurone [targets, neurone, navigator, 3]\n",
      "deepcheck conj verb\n",
      "0 [miR-29a, decreased, in, Alzheimer, disease, brains, targets, neurone, navigator, 3, .,  , AIMS] 13 0 miR-29a decreased in Alzheimer disease brains targets neurone navigator 3.  AIMS.\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text miR-29a decreased in Alzheimer disease brains targets neurone navigator 3 . AIMS\n",
      "[miR-29a, decreased, in, Alzheimer, disease, brains, targets, neurone, navigator, 3, .,  , AIMS]\n",
      "TextBetween: miR-29a decreased in Alzheimer disease brains targets neurone navigator 3 . AIMS\n",
      "infer direction from token set [miR-29a, decreased, in, Alzheimer, disease, brains, targets, neurone, navigator, 3, .,  , AIMS]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70730>, {'mir': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEG': 1, 'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70950>, {'mir': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count [(decreased, 'NEG'), (targets, 'NEU')]\n",
      "Between Tokens [(decreased, 'NEG'), (targets, 'NEU')]\n",
      "counts between final False\n",
      "['miR-29a', 'decreased', 'in', 'Alzheimer', 'disease', 'brains', 'targets', 'neurone', 'navigator', '3', '.', ' ']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "[(decreased, 'NEG'), (targets, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence miR-29a decreased in Alzheimer disease brains targets neurone navigator 3.  AIMS.\n",
      "full sentence <e1>miR-29a</e1> decreased in Alzheimer disease brains targets <e2>neurone navigator 3</e2>.  AIMS.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e1>miR-29a</e1> decreased in Alzheimer disease brains targets <e2>neurone navigator 3</e2>.  AIMS.', 'full_sentence': 'Aberrant microRNA expression in the brains of neurodegenerative diseases; <e1>miR-29a</e1> decreased in Alzheimer disease brains targets <e2>neurone navigator 3</e2>.  AIMS.;  MicroRNAs (miRNAs) are small non-coding RNAs that regulate translational repression of target mRNAs. ', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (0, 7), 'entity_text': 'miR-29a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (54, 73), 'entity_text': 'neurone navigator 3'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d68.s3.p0\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case neurone navigator 3\n",
      "add special case miR-29a\n",
      "3 13 miR-29a 29\n",
      "(u\"The database search on TargetScan, PicTar and miRBase Target identified neurone navigator 3 (NAV3), a regulator of axon guidance, as a principal target of miR-29a, and actually NAV3 mRNA levels were elevated in AD brains.\", 13, 29)\n",
      "geneword 3 mirword miR-29a\n",
      "Conjuncts [PicTar, Target, TargetScan]\n",
      "neighbour PicTar and\n",
      "neighbour PicTar and\n",
      "neighbour Target identified\n",
      "neighbour Target identified\n",
      "Conjunction before fc TargetScan\n",
      "fc add children for TargetScan\n",
      "Conjuncts [TargetScan, Target, PicTar]\n",
      "neighbour TargetScan ,\n",
      "neighbour TargetScan PicTar\n",
      "neighbour Target identified\n",
      "neighbour Target identified\n",
      "Conjunction before fc PicTar\n",
      "Conjuncts [TargetScan, PicTar, Target]\n",
      "neighbour TargetScan ,\n",
      "neighbour TargetScan PicTar\n",
      "neighbour PicTar and\n",
      "neighbour PicTar and\n",
      "Conjunction before fc Target\n",
      "fc add children for Target\n",
      "Conjuncts [elevated, identified]\n",
      "neighbour elevated in\n",
      "neighbour elevated in\n",
      "Conjunction before fc identified\n",
      "fc add children for identified\n",
      "fc add children for identified\n",
      "Conjuncts [identified, elevated]\n",
      "neighbour identified neurone\n",
      "neighbour identified neurone\n",
      "Conjunction before fc elevated\n",
      "fc add children for elevated\n",
      "Conunct merge check: TargetScan identified\n",
      "Target\n",
      "Conunct merge check: identified NAV3\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( conj )\n",
      "TargetScan [on, TargetScan, PicTar, miRBase, Target]\n",
      "identified [identified, neurone, navigator, 3, NAV3, a, regulator, of, axon, guidance, as, principal, target, of, miR-29a, elevated, in, AD, brains]\n",
      "NAV3 [(, )]\n",
      "CONJ TARGET OF RULE 3 (NAV3), a regulator of axon guidance, as a principal target of\n",
      "Conjuncts [search, TargetScan]\n",
      "neighbour search on\n",
      "neighbour search on\n",
      "Conjunction before fc TargetScan\n",
      "fc add children for TargetScan\n",
      "Conjuncts [regulator, guidance]\n",
      "neighbour regulator of\n",
      "neighbour regulator of\n",
      "Conjunction before fc guidance\n",
      "fc add children for guidance\n",
      "fc add children for guidance\n",
      "Conjuncts [identified, target]\n",
      "neighbour identified neurone\n",
      "neighbour identified neurone\n",
      "Conjunction before fc target\n",
      "fc add children for target\n",
      "fc add children for target\n",
      "fc add children for target\n",
      "Conjuncts [target, miR-29a]\n",
      "neighbour target of\n",
      "neighbour target of\n",
      "Conjunction before fc miR-29a\n",
      "fc add children for miR-29a\n",
      "Conjuncts [elevated, brains]\n",
      "neighbour elevated in\n",
      "neighbour elevated in\n",
      "Conjunction before fc brains\n",
      "fc add children for brains\n",
      "fc add children for brains\n",
      "Conunct merge check: TargetScan target\n",
      "TargetScan, PicTar and miRBase Target\n",
      "Conunct merge check: target NAV3\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: NAV3 regulator\n",
      "),\n",
      "Conunct merge check: regulator brains\n",
      "guidance, as a principal target of miR-29a, and actually NAV3 mRNA levels were\n",
      "Conjunctions ( nmod )\n",
      "TargetScan [search, on, TargetScan]\n",
      "NAV3 [(, )]\n",
      "regulator [a, regulator, of, axon, guidance]\n",
      "target [identified, as, principal, target, of, miR-29a]\n",
      "brains [elevated, in, AD, brains]\n",
      "SDP sent: The database search on TargetScan, PicTar and miRBase Target identified neurone navigator 3 (NAV3), a regulator of axon guidance, as a principal target of miR-29a, and actually NAV3 mRNA levels were elevated in AD brains.\n",
      "SDP     : [(3, 'NUM', 'nummod', False), (navigator, 'NOUN', 'dobj', False), (identified, 'VERB', 'ROOT', False), (target, 'NOUN', 'nmod', False), (miR-29a, 'NOUN', 'nmod', False)]\n",
      "SDP DOBJ RULE\n",
      "DeepCheck Conj Subtree\n",
      "[PicTar]\n",
      "DeepCheck Conj Subtree\n",
      "[miRBase, Target]\n",
      "DeepCheck Conj Subtree\n",
      "[actually, NAV3, mRNA, levels, were, elevated, in, AD, brains, .]\n",
      "0 [The, database, search, on, TargetScan, ,, PicTar, and, miRBase, Target, identified, neurone, navigator, 3, (, NAV3, ), ,, a, regulator, of, axon, guidance, ,, as, a, principal, target, of, miR-29a] 30 0 The database search on TargetScan, PicTar and miRBase Target identified neurone navigator 3 (NAV3), a regulator of axon guidance, as a principal target of miR-29a, and actually NAV3 mRNA levels were elevated in AD brains.\n",
      "1 [actually, NAV3, mRNA, levels, were, elevated, in, AD, brains] 9 168 actually NAV3 mRNA levels were elevated in AD brains.\n",
      "DeepCheck Conj Subtree\n",
      "[on, TargetScan, ,, PicTar, and, miRBase, Target]\n",
      "DeepCheck Conj Subtree\n",
      "[PicTar]\n",
      "DeepCheck Conj Subtree\n",
      "[miRBase, Target]\n",
      "DeepCheck Conj Subtree\n",
      "[of, axon, guidance]\n",
      "DeepCheck Conj Subtree\n",
      "[as, a, principal, target, of, miR-29a, ,, and]\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-29a]\n",
      "DeepCheck Conj Subtree\n",
      "[actually, NAV3, mRNA, levels, were, elevated, in, AD, brains, .]\n",
      "DeepCheck Conj Subtree\n",
      "[in, AD, brains]\n",
      "0 [The, database, search, on, TargetScan, ,, PicTar, and, miRBase, Target, identified, neurone, navigator, 3, (, NAV3, ), ,, a, regulator, of, axon, guidance, ,, as, a, principal, target, of, miR-29a] 30 0 The database search on TargetScan, PicTar and miRBase Target identified neurone navigator 3 (NAV3), a regulator of axon guidance, as a principal target of miR-29a, and actually NAV3 mRNA levels were elevated in AD brains.\n",
      "1 [actually, NAV3, mRNA, levels, were, elevated, in, AD, brains] 9 168 actually NAV3 mRNA levels were elevated in AD brains.\n",
      "Recognized overlapping entities: {Interval(72, 91, ('CELL', 'neurone navigator 3'))}\n",
      "Gene reject: {'CELL'} due to {'CELL'}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text The database search on TargetScan , PicTar and miRBase Target identified neurone navigator 3 ( NAV3 ) , a regulator of axon guidance , as a principal target of miR-29a\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': False, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'compartment gene mir', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence The database search on TargetScan, PicTar and miRBase Target identified neurone navigator 3 (NAV3), a regulator of axon guidance, as a principal target of miR-29a, and actually NAV3 mRNA levels were elevated in AD brains.\n",
      "full sentence The database search on TargetScan, PicTar and miRBase Target identified <e2>neurone navigator 3</e2> (NAV3), a regulator of axon guidance, as a principal target of <e1>miR-29a</e1>, and actually NAV3 mRNA levels were elevated in AD brains.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': False, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'compartment gene mir', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'The database search on TargetScan, PicTar and miRBase Target identified <e2>neurone navigator 3</e2> (NAV3), a regulator of axon guidance, as a principal target of <e1>miR-29a</e1>, and actually NAV3 mRNA levels were elevated in AD brains.', 'full_sentence': 'The database search on TargetScan, PicTar and miRBase Target identified <e2>neurone navigator 3</e2> (NAV3), a regulator of axon guidance, as a principal target of <e1>miR-29a</e1>, and actually NAV3 mRNA levels were elevated in AD brains.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (72, 91), 'entity_text': 'neurone navigator 3'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (155, 162), 'entity_text': 'miR-29a'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d68.s3.p0\n",
      "IS:  ('GENE_MIR', 'NEU')\n",
      "SB:  ('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': False, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'compartment gene mir', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'The database search on TargetScan, PicTar and miRBase Target identified <e2>neurone navigator 3</e2> (NAV3), a regulator of axon guidance, as a principal target of <e1>miR-29a</e1>, and actually NAV3 mRNA levels were elevated in AD brains.', 'full_sentence': 'The database search on TargetScan, PicTar and miRBase Target identified <e2>neurone navigator 3</e2> (NAV3), a regulator of axon guidance, as a principal target of <e1>miR-29a</e1>, and actually NAV3 mRNA levels were elevated in AD brains.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (72, 91), 'entity_text': 'neurone navigator 3'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (155, 162), 'entity_text': 'miR-29a'}}\n",
      "add special case neurone navigator 3\n",
      "add special case miR-29a\n",
      "3 13 miR-29a 29\n",
      "(u\"The database search on TargetScan, PicTar and miRBase Target identified neurone navigator 3 (NAV3), a regulator of axon guidance, as a principal target of miR-29a, and actually NAV3 mRNA levels were elevated in AD brains.\", 13, 29)\n",
      "geneword 3 mirword miR-29a\n",
      "Conjuncts [PicTar, Target, TargetScan]\n",
      "neighbour PicTar and\n",
      "neighbour PicTar and\n",
      "neighbour Target identified\n",
      "neighbour Target identified\n",
      "Conjunction before fc TargetScan\n",
      "fc add children for TargetScan\n",
      "Conjuncts [TargetScan, Target, PicTar]\n",
      "neighbour TargetScan ,\n",
      "neighbour TargetScan PicTar\n",
      "neighbour Target identified\n",
      "neighbour Target identified\n",
      "Conjunction before fc PicTar\n",
      "Conjuncts [TargetScan, PicTar, Target]\n",
      "neighbour TargetScan ,\n",
      "neighbour TargetScan PicTar\n",
      "neighbour PicTar and\n",
      "neighbour PicTar and\n",
      "Conjunction before fc Target\n",
      "fc add children for Target\n",
      "Conjuncts [elevated, identified]\n",
      "neighbour elevated in\n",
      "neighbour elevated in\n",
      "Conjunction before fc identified\n",
      "fc add children for identified\n",
      "fc add children for identified\n",
      "Conjuncts [identified, elevated]\n",
      "neighbour identified neurone\n",
      "neighbour identified neurone\n",
      "Conjunction before fc elevated\n",
      "fc add children for elevated\n",
      "Conunct merge check: TargetScan identified\n",
      "Target\n",
      "Conunct merge check: identified NAV3\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( conj )\n",
      "TargetScan [on, TargetScan, PicTar, miRBase, Target]\n",
      "identified [identified, neurone, navigator, 3, NAV3, a, regulator, of, axon, guidance, as, principal, target, of, miR-29a, elevated, in, AD, brains]\n",
      "NAV3 [(, )]\n",
      "CONJ TARGET OF RULE 3 (NAV3), a regulator of axon guidance, as a principal target of\n",
      "Conjuncts [search, TargetScan]\n",
      "neighbour search on\n",
      "neighbour search on\n",
      "Conjunction before fc TargetScan\n",
      "fc add children for TargetScan\n",
      "Conjuncts [regulator, guidance]\n",
      "neighbour regulator of\n",
      "neighbour regulator of\n",
      "Conjunction before fc guidance\n",
      "fc add children for guidance\n",
      "fc add children for guidance\n",
      "Conjuncts [identified, target]\n",
      "neighbour identified neurone\n",
      "neighbour identified neurone\n",
      "Conjunction before fc target\n",
      "fc add children for target\n",
      "fc add children for target\n",
      "fc add children for target\n",
      "Conjuncts [target, miR-29a]\n",
      "neighbour target of\n",
      "neighbour target of\n",
      "Conjunction before fc miR-29a\n",
      "fc add children for miR-29a\n",
      "Conjuncts [elevated, brains]\n",
      "neighbour elevated in\n",
      "neighbour elevated in\n",
      "Conjunction before fc brains\n",
      "fc add children for brains\n",
      "fc add children for brains\n",
      "Conunct merge check: TargetScan target\n",
      "TargetScan, PicTar and miRBase Target\n",
      "Conunct merge check: target NAV3\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: NAV3 regulator\n",
      "),\n",
      "Conunct merge check: regulator brains\n",
      "guidance, as a principal target of miR-29a, and actually NAV3 mRNA levels were\n",
      "Conjunctions ( nmod )\n",
      "TargetScan [search, on, TargetScan]\n",
      "NAV3 [(, )]\n",
      "regulator [a, regulator, of, axon, guidance]\n",
      "target [identified, as, principal, target, of, miR-29a]\n",
      "brains [elevated, in, AD, brains]\n",
      "SDP sent: The database search on TargetScan, PicTar and miRBase Target identified neurone navigator 3 (NAV3), a regulator of axon guidance, as a principal target of miR-29a, and actually NAV3 mRNA levels were elevated in AD brains.\n",
      "SDP     : [(3, 'NUM', 'nummod', False), (navigator, 'NOUN', 'dobj', False), (identified, 'VERB', 'ROOT', False), (target, 'NOUN', 'nmod', False), (miR-29a, 'NOUN', 'nmod', False)]\n",
      "SDP DOBJ RULE\n",
      "DeepCheck Conj Subtree\n",
      "[PicTar]\n",
      "DeepCheck Conj Subtree\n",
      "[miRBase, Target]\n",
      "DeepCheck Conj Subtree\n",
      "[actually, NAV3, mRNA, levels, were, elevated, in, AD, brains, .]\n",
      "0 [The, database, search, on, TargetScan, ,, PicTar, and, miRBase, Target, identified, neurone, navigator, 3, (, NAV3, ), ,, a, regulator, of, axon, guidance, ,, as, a, principal, target, of, miR-29a] 30 0 The database search on TargetScan, PicTar and miRBase Target identified neurone navigator 3 (NAV3), a regulator of axon guidance, as a principal target of miR-29a, and actually NAV3 mRNA levels were elevated in AD brains.\n",
      "1 [actually, NAV3, mRNA, levels, were, elevated, in, AD, brains] 9 168 actually NAV3 mRNA levels were elevated in AD brains.\n",
      "DeepCheck Conj Subtree\n",
      "[on, TargetScan, ,, PicTar, and, miRBase, Target]\n",
      "DeepCheck Conj Subtree\n",
      "[PicTar]\n",
      "DeepCheck Conj Subtree\n",
      "[miRBase, Target]\n",
      "DeepCheck Conj Subtree\n",
      "[of, axon, guidance]\n",
      "DeepCheck Conj Subtree\n",
      "[as, a, principal, target, of, miR-29a, ,, and]\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-29a]\n",
      "DeepCheck Conj Subtree\n",
      "[actually, NAV3, mRNA, levels, were, elevated, in, AD, brains, .]\n",
      "DeepCheck Conj Subtree\n",
      "[in, AD, brains]\n",
      "0 [The, database, search, on, TargetScan, ,, PicTar, and, miRBase, Target, identified, neurone, navigator, 3, (, NAV3, ), ,, a, regulator, of, axon, guidance, ,, as, a, principal, target, of, miR-29a] 30 0 The database search on TargetScan, PicTar and miRBase Target identified neurone navigator 3 (NAV3), a regulator of axon guidance, as a principal target of miR-29a, and actually NAV3 mRNA levels were elevated in AD brains.\n",
      "1 [actually, NAV3, mRNA, levels, were, elevated, in, AD, brains] 9 168 actually NAV3 mRNA levels were elevated in AD brains.\n",
      "Recognized overlapping entities: {Interval(72, 91, ('CELL', 'neurone navigator 3'))}\n",
      "Gene reject: {'CELL'} due to {'CELL'}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text The database search on TargetScan , PicTar and miRBase Target identified neurone navigator 3 ( NAV3 ) , a regulator of axon guidance , as a principal target of miR-29a\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': False, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'compartment gene mir', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence The database search on TargetScan, PicTar and miRBase Target identified neurone navigator 3 (NAV3), a regulator of axon guidance, as a principal target of miR-29a, and actually NAV3 mRNA levels were elevated in AD brains.\n",
      "full sentence The database search on TargetScan, PicTar and miRBase Target identified <e2>neurone navigator 3</e2> (NAV3), a regulator of axon guidance, as a principal target of <e1>miR-29a</e1>, and actually NAV3 mRNA levels were elevated in AD brains.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': False, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'compartment gene mir', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'The database search on TargetScan, PicTar and miRBase Target identified <e2>neurone navigator 3</e2> (NAV3), a regulator of axon guidance, as a principal target of <e1>miR-29a</e1>, and actually NAV3 mRNA levels were elevated in AD brains.', 'full_sentence': 'The database search on TargetScan, PicTar and miRBase Target identified <e2>neurone navigator 3</e2> (NAV3), a regulator of axon guidance, as a principal target of <e1>miR-29a</e1>, and actually NAV3 mRNA levels were elevated in AD brains.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (72, 91), 'entity_text': 'neurone navigator 3'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (155, 162), 'entity_text': 'miR-29a'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d68.s3.p1\n",
      "IS:  ('GENE_MIR', 'NEU')\n",
      "SB:  ('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'compartment gene mir', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'The database search on TargetScan, PicTar and miRBase Target identified neurone navigator 3 (<e2>NAV3</e2>), a regulator of axon guidance, as a principal target of <e1>miR-29a</e1>, and actually NAV3 mRNA levels were elevated in AD brains.', 'full_sentence': 'The database search on TargetScan, PicTar and miRBase Target identified neurone navigator 3 (<e2>NAV3</e2>), a regulator of axon guidance, as a principal target of <e1>miR-29a</e1>, and actually NAV3 mRNA levels were elevated in AD brains.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (93, 97), 'entity_text': 'NAV3'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (155, 162), 'entity_text': 'miR-29a'}}\n",
      "add special case NAV3\n",
      "add special case miR-29a\n",
      "NAV3 15 miR-29a 29\n",
      "(u\"The database search on TargetScan, PicTar and miRBase Target identified neurone navigator 3 (NAV3), a regulator of axon guidance, as a principal target of miR-29a, and actually NAV3 mRNA levels were elevated in AD brains.\", 15, 29)\n",
      "geneword NAV3 mirword miR-29a\n",
      "Conjuncts [PicTar, Target, TargetScan]\n",
      "neighbour PicTar and\n",
      "neighbour PicTar and\n",
      "neighbour Target identified\n",
      "neighbour Target identified\n",
      "Conjunction before fc TargetScan\n",
      "fc add children for TargetScan\n",
      "Conjuncts [TargetScan, Target, PicTar]\n",
      "neighbour TargetScan ,\n",
      "neighbour TargetScan PicTar\n",
      "neighbour Target identified\n",
      "neighbour Target identified\n",
      "Conjunction before fc PicTar\n",
      "Conjuncts [TargetScan, PicTar, Target]\n",
      "neighbour TargetScan ,\n",
      "neighbour TargetScan PicTar\n",
      "neighbour PicTar and\n",
      "neighbour PicTar and\n",
      "Conjunction before fc Target\n",
      "fc add children for Target\n",
      "Conjuncts [elevated, identified]\n",
      "neighbour elevated in\n",
      "neighbour elevated in\n",
      "Conjunction before fc identified\n",
      "fc add children for identified\n",
      "fc add children for identified\n",
      "Conjuncts [identified, elevated]\n",
      "neighbour identified neurone\n",
      "neighbour identified neurone\n",
      "Conjunction before fc elevated\n",
      "fc add children for elevated\n",
      "Conunct merge check: TargetScan identified\n",
      "Target\n",
      "Conunct merge check: identified NAV3\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( conj )\n",
      "TargetScan [on, TargetScan, PicTar, miRBase, Target]\n",
      "identified [identified, neurone, navigator, 3, NAV3, a, regulator, of, axon, guidance, as, principal, target, of, miR-29a, elevated, in, AD, brains]\n",
      "NAV3 [(, )]\n",
      "CONJ TARGET OF RULE NAV3), a regulator of axon guidance, as a principal target of\n",
      "Conjuncts [search, TargetScan]\n",
      "neighbour search on\n",
      "neighbour search on\n",
      "Conjunction before fc TargetScan\n",
      "fc add children for TargetScan\n",
      "Conjuncts [regulator, guidance]\n",
      "neighbour regulator of\n",
      "neighbour regulator of\n",
      "Conjunction before fc guidance\n",
      "fc add children for guidance\n",
      "fc add children for guidance\n",
      "Conjuncts [identified, target]\n",
      "neighbour identified neurone\n",
      "neighbour identified neurone\n",
      "Conjunction before fc target\n",
      "fc add children for target\n",
      "fc add children for target\n",
      "fc add children for target\n",
      "Conjuncts [target, miR-29a]\n",
      "neighbour target of\n",
      "neighbour target of\n",
      "Conjunction before fc miR-29a\n",
      "fc add children for miR-29a\n",
      "Conjuncts [elevated, brains]\n",
      "neighbour elevated in\n",
      "neighbour elevated in\n",
      "Conjunction before fc brains\n",
      "fc add children for brains\n",
      "fc add children for brains\n",
      "Conunct merge check: TargetScan target\n",
      "TargetScan, PicTar and miRBase Target\n",
      "Conunct merge check: target NAV3\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: NAV3 regulator\n",
      "),\n",
      "Conunct merge check: regulator brains\n",
      "guidance, as a principal target of miR-29a, and actually NAV3 mRNA levels were\n",
      "Conjunctions ( nmod )\n",
      "TargetScan [search, on, TargetScan]\n",
      "NAV3 [(, )]\n",
      "regulator [a, regulator, of, axon, guidance]\n",
      "target [identified, as, principal, target, of, miR-29a]\n",
      "brains [elevated, in, AD, brains]\n",
      "SDP sent: The database search on TargetScan, PicTar and miRBase Target identified neurone navigator 3 (NAV3), a regulator of axon guidance, as a principal target of miR-29a, and actually NAV3 mRNA levels were elevated in AD brains.\n",
      "SDP     : [(NAV3, 'NOUN', 'appos', False), (navigator, 'NOUN', 'dobj', False), (identified, 'VERB', 'ROOT', False), (target, 'NOUN', 'nmod', False), (miR-29a, 'NOUN', 'nmod', False)]\n",
      "SDP DOBJ RULE\n",
      "DeepCheck Conj Subtree\n",
      "[PicTar]\n",
      "DeepCheck Conj Subtree\n",
      "[miRBase, Target]\n",
      "DeepCheck Conj Subtree\n",
      "[actually, NAV3, mRNA, levels, were, elevated, in, AD, brains, .]\n",
      "0 [The, database, search, on, TargetScan, ,, PicTar, and, miRBase, Target, identified, neurone, navigator, 3, (, NAV3, ), ,, a, regulator, of, axon, guidance, ,, as, a, principal, target, of, miR-29a] 30 0 The database search on TargetScan, PicTar and miRBase Target identified neurone navigator 3 (NAV3), a regulator of axon guidance, as a principal target of miR-29a, and actually NAV3 mRNA levels were elevated in AD brains.\n",
      "1 [actually, NAV3, mRNA, levels, were, elevated, in, AD, brains] 9 168 actually NAV3 mRNA levels were elevated in AD brains.\n",
      "DeepCheck Conj Subtree\n",
      "[on, TargetScan, ,, PicTar, and, miRBase, Target]\n",
      "DeepCheck Conj Subtree\n",
      "[PicTar]\n",
      "DeepCheck Conj Subtree\n",
      "[miRBase, Target]\n",
      "DeepCheck Conj Subtree\n",
      "[of, axon, guidance]\n",
      "DeepCheck Conj Subtree\n",
      "[as, a, principal, target, of, miR-29a, ,, and]\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-29a]\n",
      "DeepCheck Conj Subtree\n",
      "[actually, NAV3, mRNA, levels, were, elevated, in, AD, brains, .]\n",
      "DeepCheck Conj Subtree\n",
      "[in, AD, brains]\n",
      "0 [The, database, search, on, TargetScan, ,, PicTar, and, miRBase, Target, identified, neurone, navigator, 3, (, NAV3, ), ,, a, regulator, of, axon, guidance, ,, as, a, principal, target, of, miR-29a] 30 0 The database search on TargetScan, PicTar and miRBase Target identified neurone navigator 3 (NAV3), a regulator of axon guidance, as a principal target of miR-29a, and actually NAV3 mRNA levels were elevated in AD brains.\n",
      "1 [actually, NAV3, mRNA, levels, were, elevated, in, AD, brains] 9 168 actually NAV3 mRNA levels were elevated in AD brains.\n",
      "Recognized overlapping entities: {Interval(93, 97, ('GENE_OR_GENE_PRODUCT', 'NAV3'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text The database search on TargetScan , PicTar and miRBase Target identified neurone navigator 3 ( NAV3 ) , a regulator of axon guidance , as a principal target of miR-29a\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'compartment gene mir', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence The database search on TargetScan, PicTar and miRBase Target identified neurone navigator 3 (NAV3), a regulator of axon guidance, as a principal target of miR-29a, and actually NAV3 mRNA levels were elevated in AD brains.\n",
      "full sentence The database search on TargetScan, PicTar and miRBase Target identified neurone navigator 3 (<e2>NAV3</e2>), a regulator of axon guidance, as a principal target of <e1>miR-29a</e1>, and actually NAV3 mRNA levels were elevated in AD brains.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'compartment gene mir', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'The database search on TargetScan, PicTar and miRBase Target identified neurone navigator 3 (<e2>NAV3</e2>), a regulator of axon guidance, as a principal target of <e1>miR-29a</e1>, and actually NAV3 mRNA levels were elevated in AD brains.', 'full_sentence': 'The database search on TargetScan, PicTar and miRBase Target identified neurone navigator 3 (<e2>NAV3</e2>), a regulator of axon guidance, as a principal target of <e1>miR-29a</e1>, and actually NAV3 mRNA levels were elevated in AD brains.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (93, 97), 'entity_text': 'NAV3'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (155, 162), 'entity_text': 'miR-29a'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d92.s4.p1\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miR-133b\n",
      "add special case tyrosine hydroxylase\n",
      "miR-133b 0 hydroxylase 15\n",
      "(u\"miR-133b is specifically expressed in midbrain dopaminergic neurons, and regulates the production of tyrosine hydroxylase and the dopamine transporter.\", 0, 15)\n",
      "geneword hydroxylase mirword miR-133b\n",
      "Conjuncts [regulates, expressed]\n",
      "neighbour regulates the\n",
      "neighbour regulates the\n",
      "Conjunction before fc expressed\n",
      "fc add children for expressed\n",
      "Conjuncts [expressed, regulates]\n",
      "neighbour expressed in\n",
      "neighbour expressed in\n",
      "Conjunction before fc regulates\n",
      "fc add children for regulates\n",
      "Conjuncts [transporter, hydroxylase]\n",
      "neighbour transporter .\n",
      "neighbour transporter None\n",
      "Conjunction before fc hydroxylase\n",
      "fc add children for hydroxylase\n",
      "fc add children for hydroxylase\n",
      "Conjuncts [hydroxylase, transporter]\n",
      "neighbour hydroxylase and\n",
      "neighbour hydroxylase and\n",
      "Conjunction before fc transporter\n",
      "fc add children for transporter\n",
      "Conjunctions ( conj )\n",
      "expressed [expressed, in, midbrain, dopaminergic, neurons, regulates, production, of, tyrosine, hydroxylase, dopamine, transporter]\n",
      "Conjuncts [expressed, neurons]\n",
      "neighbour expressed in\n",
      "neighbour expressed in\n",
      "Conjunction before fc neurons\n",
      "fc add children for neurons\n",
      "fc add children for neurons\n",
      "fc add children for neurons\n",
      "Conjuncts [production, hydroxylase]\n",
      "neighbour production of\n",
      "neighbour production of\n",
      "Conjunction before fc hydroxylase\n",
      "fc add children for hydroxylase\n",
      "fc add children for hydroxylase\n",
      "Conunct merge check: neurons hydroxylase\n",
      "neurons, and regulates the\n",
      "Double Conjunction Found\n",
      "Conjunctions ( nmod )\n",
      "neurons [expressed, in, midbrain, dopaminergic, neurons]\n",
      "hydroxylase [production, of, tyrosine, hydroxylase]\n",
      "(hydroxylase, 'conj') [expressed, in, midbrain, dopaminergic, neurons, production, of, tyrosine, hydroxylase]\n",
      "SDP sent: miR-133b is specifically expressed in midbrain dopaminergic neurons, and regulates the production of tyrosine hydroxylase and the dopamine transporter.\n",
      "SDP     : [(miR-133b, 'NOUN', 'nsubjpass', False), (expressed, 'VERB', 'ROOT', True), (regulates, 'VERB', 'conj', False), (production, 'NOUN', 'dobj', False), (hydroxylase, 'NOUN', 'nmod', False)]\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[the, dopamine, transporter]\n",
      "0 [miR-133b, is, specifically, expressed, in, midbrain, dopaminergic, neurons, ,, and, regulates, the, production, of, tyrosine, hydroxylase, and, the, dopamine, transporter] 20 0 miR-133b is specifically expressed in midbrain dopaminergic neurons, and regulates the production of tyrosine hydroxylase and the dopamine transporter.\n",
      "DeepCheck Conj Subtree\n",
      "[in, midbrain, dopaminergic, neurons]\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[of, tyrosine, hydroxylase, and, the, dopamine, transporter]\n",
      "DeepCheck Conj Subtree\n",
      "[the, dopamine, transporter]\n",
      "0 [miR-133b, is, specifically, expressed, in, midbrain, dopaminergic, neurons, ,, and, regulates, the, production, of, tyrosine, hydroxylase, and, the, dopamine, transporter] 20 0 miR-133b is specifically expressed in midbrain dopaminergic neurons, and regulates the production of tyrosine hydroxylase and the dopamine transporter.\n",
      "Recognized overlapping entities: set()\n",
      "passive? True\n",
      "Counter()\n",
      "Compartment Text miR-133b is specifically expressed in midbrain dopaminergic neurons , and regulates the production of tyrosine hydroxylase and the dopamine transporter\n",
      "[miR-133b, is, specifically, expressed, in, midbrain, dopaminergic, neurons, ,, and, regulates, the, production, of, tyrosine, hydroxylase, and, the, dopamine, transporter]\n",
      "TextBetween: miR-133b is specifically expressed in midbrain dopaminergic neurons , and regulates the production of tyrosine hydroxylase and the dopamine transporter\n",
      "infer direction from token set [miR-133b, is, specifically, expressed, in, midbrain, dopaminergic, neurons, ,, and, regulates, the, production, of, tyrosine, hydroxylase, and, the, dopamine, transporter]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c2f0>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 3})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c048>, {})\n",
      "counts between 1 stemTypes2Count [(expressed, 'NEU'), (regulates, 'NEU'), (production, 'NEU')]\n",
      "Between Tokens [(expressed, 'NEU'), (regulates, 'NEU'), (production, 'NEU')]\n",
      "counts between final True\n",
      "['miR-133b', 'is', 'specifically', 'expressed', 'in', 'midbrain', 'dopaminergic', 'neurons', ',', 'and', 'regulates', 'the', 'production', 'of', 'tyrosine', 'hydroxylase', 'and', 'the']\n",
      "Counter({'NEU': 3})\n",
      "[(expressed, 'NEU'), (regulates, 'NEU'), (production, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence miR-133b is specifically expressed in midbrain dopaminergic neurons, and regulates the production of tyrosine hydroxylase and the dopamine transporter.\n",
      "full sentence <e1>miR-133b</e1> is specifically expressed in midbrain dopaminergic neurons, and regulates the production of <e2>tyrosine hydroxylase</e2> and the dopamine transporter.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e1>miR-133b</e1> is specifically expressed in midbrain dopaminergic neurons, and regulates the production of <e2>tyrosine hydroxylase</e2> and the dopamine transporter.', 'full_sentence': 'Recent studies have identified changes of several specific miRNA expression profiles and polymorphisms affecting the interactions between miRNAs and their targets in various brain disorders, including addiction;  miR-16 causes adaptive changes in production of the serotonin transporter; <e1>miR-133b</e1> is specifically expressed in midbrain dopaminergic neurons, and regulates the production of <e2>tyrosine hydroxylase</e2> and the dopamine transporter.;  miR-212 affects production of striatal brain-derived neurotrophic factor and synaptic plasticity upon cocaine. ', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (0, 8), 'entity_text': 'miR-133b'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (101, 121), 'entity_text': 'tyrosine hydroxylase'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d94.s2.p2\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case CDKN2A\n",
      "add special case miRNA-125b\n",
      "CDKN2A 26 miRNA-125b 30\n",
      "(u\"In vitro, anti-miRNA-125b added exogenously to IL-6-stressed NHA cultures attenuated both glial cell proliferation and increased the expression of the cyclin-dependent kinase inhibitor 2A (CDKN2A), a miRNA-125b target and negative regulator of cell growth.\", 26, 30)\n",
      "geneword CDKN2A mirword miRNA-125b\n",
      "Conjuncts [attenuated, increased, added]\n",
      "neighbour attenuated both\n",
      "neighbour attenuated both\n",
      "neighbour increased the\n",
      "neighbour increased the\n",
      "Conjunction before fc added\n",
      "fc add children for added\n",
      "Conjuncts [added, increased, attenuated]\n",
      "neighbour added exogenously\n",
      "neighbour added exogenously\n",
      "neighbour increased the\n",
      "neighbour increased the\n",
      "Conjunction before fc attenuated\n",
      "fc add children for attenuated\n",
      "Conjuncts [added, attenuated, increased]\n",
      "neighbour added exogenously\n",
      "neighbour added exogenously\n",
      "neighbour attenuated both\n",
      "neighbour attenuated both\n",
      "Conjunction before fc increased\n",
      "fc add children for increased\n",
      "Conjuncts [regulator, target]\n",
      "neighbour regulator of\n",
      "neighbour regulator of\n",
      "Conjunction before fc target\n",
      "fc add children for target\n",
      "Conjuncts [target, regulator]\n",
      "neighbour target and\n",
      "neighbour target and\n",
      "Conjunction before fc regulator\n",
      "fc add children for regulator\n",
      "fc add children for regulator\n",
      "Conunct merge check: added CDKN2A\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( conj )\n",
      "added [added, to, IL-6-stressed, NHA, cultures, attenuated, glial, cell, proliferation, increased, expression, of, cyclin-dependent, kinase, inhibitor, 2A, CDKN2A, miRNA-125b, target, negative, regulator, of, cell, growth]\n",
      "CDKN2A [(, )]\n",
      "achildren {target, 2A, miRNA-125b}\n",
      "achildren False\n",
      "Conjuncts [added, cultures]\n",
      "neighbour added exogenously\n",
      "neighbour added exogenously\n",
      "Conjunction before fc cultures\n",
      "fc add children for cultures\n",
      "fc add children for cultures\n",
      "fc add children for cultures\n",
      "Conjuncts [expression, 2A]\n",
      "neighbour expression of\n",
      "neighbour expression of\n",
      "Conjunction before fc 2A\n",
      "fc add children for 2A\n",
      "fc add children for 2A\n",
      "fc add children for 2A\n",
      "fc add children for 2A\n",
      "fc add children for 2A\n",
      "fc add children for 2A\n",
      "Conjuncts [regulator, growth]\n",
      "neighbour regulator of\n",
      "neighbour regulator of\n",
      "Conjunction before fc growth\n",
      "fc add children for growth\n",
      "fc add children for growth\n",
      "Conunct merge check: cultures 2A\n",
      "cultures attenuated both glial cell proliferation and increased the\n",
      "Conunct merge check: 2A CDKN2A\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: CDKN2A target\n",
      "),\n",
      "Conjunctions ( nmod )\n",
      "cultures [added, to, IL-6-stressed, NHA, cultures]\n",
      "2A [expression, of, cyclin-dependent, kinase, inhibitor, 2A, CDKN2A, miRNA-125b, target]\n",
      "CDKN2A [(, )]\n",
      "target [a, regulator, of, cell, growth]\n",
      "achildren {target, 2A, miRNA-125b}\n",
      "achildren False\n",
      "SDP sent: In vitro, anti-miRNA-125b added exogenously to IL-6-stressed NHA cultures attenuated both glial cell proliferation and increased the expression of the cyclin-dependent kinase inhibitor 2A (CDKN2A), a miRNA-125b target and negative regulator of cell growth.\n",
      "SDP     : [(CDKN2A, 'NOUN', 'appos', False), (2A, 'NOUN', 'nmod', False), (target, 'NOUN', 'appos', False), (miRNA-125b, 'NOUN', 'amod', False)]\n",
      "deepcheck conj verb\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[negative, regulator, of, cell, growth]\n",
      "0 [In, vitro, ,, anti-miRNA-125b, added, exogenously, to, IL-6-stressed, NHA, cultures, attenuated, both, glial, cell, proliferation, and, increased, the, expression, of, the, cyclin-dependent, kinase, inhibitor, 2A, (, CDKN2A, ), ,, a, miRNA-125b, target, and, negative, regulator, of, cell, growth] 38 0 In vitro, anti-miRNA-125b added exogenously to IL-6-stressed NHA cultures attenuated both glial cell proliferation and increased the expression of the cyclin-dependent kinase inhibitor 2A (CDKN2A), a miRNA-125b target and negative regulator of cell growth.\n",
      "DeepCheck Conj Subtree\n",
      "[to, IL-6-stressed, NHA, cultures]\n",
      "deepcheck conj verb\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, cyclin-dependent, kinase, inhibitor, 2A, (, CDKN2A, ), ,, a, miRNA-125b, target, and, negative, regulator, of, cell, growth]\n",
      "DeepCheck Conj Subtree\n",
      "[negative, regulator, of, cell, growth]\n",
      "DeepCheck Conj Subtree\n",
      "[of, cell, growth]\n",
      "0 [In, vitro, ,, anti-miRNA-125b, added, exogenously, to, IL-6-stressed, NHA, cultures, attenuated, both, glial, cell, proliferation, and, increased, the, expression, of, the, cyclin-dependent, kinase, inhibitor, 2A, (, CDKN2A, ), ,, a, miRNA-125b, target, and, negative, regulator, of, cell, growth] 38 0 In vitro, anti-miRNA-125b added exogenously to IL-6-stressed NHA cultures attenuated both glial cell proliferation and increased the expression of the cyclin-dependent kinase inhibitor 2A (CDKN2A), a miRNA-125b target and negative regulator of cell growth.\n",
      "Recognized overlapping entities: {Interval(189, 195, ('GENE_OR_GENE_PRODUCT', 'CDKN2A'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text In vitro , anti-miRNA-125b added exogenously to IL-6-stressed NHA cultures attenuated both glial cell proliferation and increased the expression of the cyclin-dependent kinase inhibitor 2A ( CDKN2A ) , a miRNA-125b target and negative regulator of cell growth\n",
      "[In, vitro, ,, anti-miRNA-125b, added, exogenously, to, IL-6-stressed, NHA, cultures, attenuated, both, glial, cell, proliferation, and, increased, the, expression, of, the, cyclin-dependent, kinase, inhibitor, 2A, (, CDKN2A, ), ,, a, miRNA-125b, target, and, negative, regulator, of, cell, growth]\n",
      "TextBetween: In vitro , anti-miRNA-125b added exogenously to IL-6-stressed NHA cultures attenuated both glial cell proliferation and increased the expression of the cyclin-dependent kinase inhibitor 2A ( CDKN2A ) , a miRNA-125b target and negative regulator of cell growth\n",
      "infer direction from token set [In, vitro, ,, anti-miRNA-125b, added, exogenously, to, IL-6-stressed, NHA, cultures, attenuated, both, glial, cell, proliferation, and, increased, the, expression, of, the, cyclin-dependent, kinase, inhibitor, 2A, (, CDKN2A, ), ,, a, miRNA-125b, target, and, negative, regulator, of, cell, growth]\n",
      "stemTypes2Count: Counter({'POS': 4, 'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954ae8>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954bf8>, {'mir': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(target, 'NEU')]\n",
      "Between Tokens []\n",
      "counts between final False\n",
      "['2A', '(', 'CDKN2A', ')', ',', 'a', 'miRNA-125b', 'target', 'and']\n",
      "Counter({'NEU': 1})\n",
      "[(target, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': False, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence In vitro, anti-miRNA-125b added exogenously to IL-6-stressed NHA cultures attenuated both glial cell proliferation and increased the expression of the cyclin-dependent kinase inhibitor 2A (CDKN2A), a miRNA-125b target and negative regulator of cell growth.\n",
      "full sentence In vitro, anti-miRNA-125b added exogenously to IL-6-stressed NHA cultures attenuated both glial cell proliferation and increased the expression of the cyclin-dependent kinase inhibitor 2A (<e2>CDKN2A</e2>), a <e1>miRNA-125b</e1> target and negative regulator of cell growth.\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'In vitro, anti-miRNA-125b added exogenously to IL-6-stressed NHA cultures attenuated both glial cell proliferation and increased the expression of the cyclin-dependent kinase inhibitor 2A (<e2>CDKN2A</e2>), a <e1>miRNA-125b</e1> target and negative regulator of cell growth.', 'full_sentence': 'In vitro, anti-miRNA-125b added exogenously to IL-6-stressed NHA cultures attenuated both glial cell proliferation and increased the expression of the cyclin-dependent kinase inhibitor 2A (<e2>CDKN2A</e2>), a <e1>miRNA-125b</e1> target and negative regulator of cell growth.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (189, 195), 'entity_text': 'CDKN2A'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (200, 210), 'entity_text': 'miRNA-125b'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d94.s3.p0\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case miRNA-125b\n",
      "add special case interleukin-6\n",
      "miRNA-125b 10 interleukin-6 14\n",
      "(u\"Here we report that the levels of a human brain-enriched miRNA-125b are up-regulated in interleukin-6 (IL-6)-stressed normal human astrocytes (NHA), a treatment known to induce astrogliosis.\", 10, 14)\n",
      "geneword interleukin-6 mirword miRNA-125b\n",
      "Conunct merge check: NHA treatment\n",
      "),\n",
      "Conjunctions ( conj )\n",
      "NHA [(, )]\n",
      "treatment [a, astrogliosis]\n",
      "Conjuncts [levels, miRNA-125b]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc miRNA-125b\n",
      "fc add children for miRNA-125b\n",
      "fc add children for miRNA-125b\n",
      "fc add children for miRNA-125b\n",
      "Conjuncts [up-regulated, astrocytes]\n",
      "neighbour up-regulated in\n",
      "neighbour up-regulated in\n",
      "Conjunction before fc astrocytes\n",
      "fc add children for astrocytes\n",
      "fc add children for astrocytes\n",
      "fc add children for astrocytes\n",
      "fc add children for astrocytes\n",
      "fc add children for astrocytes\n",
      "fc add children for astrocytes\n",
      "fc add children for astrocytes\n",
      "Conunct merge check: miRNA-125b astrocytes\n",
      "miRNA-125b are\n",
      "Conunct merge check: astrocytes NHA\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: NHA treatment\n",
      "),\n",
      "Conjunctions ( nmod )\n",
      "miRNA-125b [levels, of, human, brain-enriched, miRNA-125b]\n",
      "astrocytes [up-regulated, in, interleukin-6, (IL-6)-stressed, normal, human, astrocytes, NHA, treatment, known]\n",
      "NHA [(, )]\n",
      "treatment [a, astrogliosis]\n",
      "SDP sent: Here we report that the levels of a human brain-enriched miRNA-125b are up-regulated in interleukin-6 (IL-6)-stressed normal human astrocytes (NHA), a treatment known to induce astrogliosis.\n",
      "SDP     : [(miRNA-125b, 'NOUN', 'nmod', False), (levels, 'NOUN', 'nsubjpass', False), (up-regulated, 'VERB', 'ccomp', False), (astrocytes, 'NOUN', 'nmod', False), (interleukin-6, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule [to, induce, astrogliosis]\n",
      "0 [Here, we, report] 3 0 Here we report that the levels of a human brain-enriched miRNA-125b are up-regulated in interleukin-6 (IL-6)-stressed normal human astrocytes (NHA), a treatment known to induce astrogliosis.\n",
      "1 [the, levels, of, a, human, brain-enriched, miRNA-125b, are, up-regulated, in, interleukin-6, (IL-6)-stressed, normal, human, astrocytes, (, NHA, ), ,, a, treatment, known, to, induce, astrogliosis] 25 20 the levels of a human brain-enriched miRNA-125b are up-regulated in interleukin-6 (IL-6)-stressed normal human astrocytes (NHA), a treatment known to induce astrogliosis.\n",
      "DeepCheck Conj Subtree\n",
      "[of, a, human, brain-enriched, miRNA-125b]\n",
      "DeepCheck Conj Subtree\n",
      "[in, interleukin-6, (IL-6)-stressed, normal, human, astrocytes, (, NHA, ), ,, a, treatment, known, to, induce, astrogliosis]\n",
      "deepcheck connection rule [to, induce, astrogliosis]\n",
      "0 [Here, we, report] 3 0 Here we report that the levels of a human brain-enriched miRNA-125b are up-regulated in interleukin-6 (IL-6)-stressed normal human astrocytes (NHA), a treatment known to induce astrogliosis.\n",
      "1 [the, levels, of, a, human, brain-enriched, miRNA-125b, are, up-regulated, in, interleukin-6, (IL-6)-stressed, normal, human, astrocytes, (, NHA, ), ,, a, treatment, known, to, induce, astrogliosis] 25 20 the levels of a human brain-enriched miRNA-125b are up-regulated in interleukin-6 (IL-6)-stressed normal human astrocytes (NHA), a treatment known to induce astrogliosis.\n",
      "Context Fail pre <regex.Match object; span=(2, 18), match=' up-regulated in', fuzzy_counts=(0, 1, 0)> regex.Regex('(up-regulated in){e<=1}', flags=regex.V0)\n",
      "Recognized overlapping entities: {Interval(88, 101, ('GENE_OR_GENE_PRODUCT', 'interleukin-6'))}\n",
      "passive? True\n",
      "Counter({'POS': 1})\n",
      "Compartment Text the levels of a human brain-enriched miRNA-125b are up-regulated in interleukin-6 (IL-6)-stressed normal human astrocytes ( NHA ) , a treatment known to induce astrogliosis\n",
      "[the, levels, of, a, human, brain-enriched, miRNA-125b, are, up-regulated, in, interleukin-6, (IL-6)-stressed, normal, human, astrocytes, (, NHA, ), ,, a, treatment, known, to, induce, astrogliosis]\n",
      "TextBetween: the levels of a human brain-enriched miRNA-125b are up-regulated in interleukin-6 (IL-6)-stressed normal human astrocytes ( NHA ) , a treatment known to induce astrogliosis\n",
      "infer direction from token set [the, levels, of, a, human, brain-enriched, miRNA-125b, are, up-regulated, in, interleukin-6, (IL-6)-stressed, normal, human, astrocytes, (, NHA, ), ,, a, treatment, known, to, induce, astrogliosis]\n",
      "stemTypes2Count: Counter({'POS': 2})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e27b8>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'POS': 2})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e20d0>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count [(up-regulated, 'POS'), ((IL-6)-stressed, 'POS')]\n",
      "Between Tokens [(up-regulated, 'POS')]\n",
      "counts between final True\n",
      "['human', 'brain-enriched', 'miRNA-125b', 'are', 'up-regulated', 'in', 'interleukin-6', '(IL-6)-stressed', 'normal']\n",
      "Counter({'POS': 2})\n",
      "[(up-regulated, 'POS'), ((IL-6)-stressed, 'POS')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': True, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': True, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Here we report that the levels of a human brain-enriched miRNA-125b are up-regulated in interleukin-6 (IL-6)-stressed normal human astrocytes (NHA), a treatment known to induce astrogliosis.\n",
      "full sentence Here we report that the levels of a human brain-enriched <e1>miRNA-125b</e1> are up-regulated in <e2>interleukin-6</e2> (IL-6)-stressed normal human astrocytes (NHA), a treatment known to induce astrogliosis.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': True, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Here we report that the levels of a human brain-enriched <e1>miRNA-125b</e1> are up-regulated in <e2>interleukin-6</e2> (IL-6)-stressed normal human astrocytes (NHA), a treatment known to induce astrogliosis.', 'full_sentence': 'Here we report that the levels of a human brain-enriched <e1>miRNA-125b</e1> are up-regulated in <e2>interleukin-6</e2> (IL-6)-stressed normal human astrocytes (NHA), a treatment known to induce astrogliosis.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (57, 67), 'entity_text': 'miRNA-125b'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (88, 101), 'entity_text': 'interleukin-6'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d94.s5.p6\n",
      "IS:  ('MIR_GENE', 'UP')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene pos corr', 'reg_detect_major': 'compartment'}}, 'tagged_sent': \"A strong positive correlation between <e1>miRNA-125b</e1> abundance and the glial cell markers glial fibrillary acidic protein (GFAP) and vimentin, and <e2>CDKN2A</e2> down-regulation was noted in advanced Alzheimer's disease (AD) and in Down's syndrome (DS) brain, chronic neurological disorders associated with astrogliosis.\", 'full_sentence': \"A strong positive correlation between <e1>miRNA-125b</e1> abundance and the glial cell markers glial fibrillary acidic protein (GFAP) and vimentin, and <e2>CDKN2A</e2> down-regulation was noted in advanced Alzheimer's disease (AD) and in Down's syndrome (DS) brain, chronic neurological disorders associated with astrogliosis.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (38, 48), 'entity_text': 'miRNA-125b'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (143, 149), 'entity_text': 'CDKN2A'}}\n",
      "add special case miRNA-125b\n",
      "add special case CDKN2A\n",
      "miRNA-125b 5 CDKN2A 23\n",
      "(u\"A strong positive correlation between miRNA-125b abundance and the glial cell markers glial fibrillary acidic protein (GFAP) and vimentin, and CDKN2A down-regulation was noted in advanced Alzheimer's disease (AD) and in Down's syndrome (DS) brain, chronic neurological disorders associated with astrogliosis.\", 5, 23)\n",
      "geneword CDKN2A mirword miRNA-125b\n",
      "Conjuncts [protein, down-regulation, vimentin, abundance]\n",
      "neighbour protein (\n",
      "neighbour protein GFAP\n",
      "neighbour down-regulation was\n",
      "neighbour down-regulation was\n",
      "neighbour vimentin ,\n",
      "neighbour vimentin and\n",
      "Conjunction before fc abundance\n",
      "fc add children for abundance\n",
      "fc add children for abundance\n",
      "Conjuncts [abundance, down-regulation, vimentin, protein]\n",
      "neighbour abundance and\n",
      "neighbour abundance and\n",
      "neighbour down-regulation was\n",
      "neighbour down-regulation was\n",
      "neighbour vimentin ,\n",
      "neighbour vimentin and\n",
      "Conjunction before fc protein\n",
      "fc add children for protein\n",
      "fc add children for protein\n",
      "fc add children for protein\n",
      "fc add children for protein\n",
      "fc add children for protein\n",
      "Conjuncts [abundance, protein, down-regulation, vimentin]\n",
      "neighbour abundance and\n",
      "neighbour abundance and\n",
      "neighbour protein (\n",
      "neighbour protein GFAP\n",
      "neighbour down-regulation was\n",
      "neighbour down-regulation was\n",
      "Conjunction before fc vimentin\n",
      "Conjuncts [abundance, protein, vimentin, down-regulation]\n",
      "neighbour abundance and\n",
      "neighbour abundance and\n",
      "neighbour protein (\n",
      "neighbour protein GFAP\n",
      "neighbour vimentin ,\n",
      "neighbour vimentin and\n",
      "Conjunction before fc down-regulation\n",
      "fc add children for down-regulation\n",
      "Conjuncts [brain, disorders, disease]\n",
      "neighbour brain ,\n",
      "neighbour brain chronic\n",
      "neighbour disorders associated\n",
      "neighbour disorders associated\n",
      "Conjunction before fc disease\n",
      "fc add children for disease\n",
      "fc add children for disease\n",
      "fc add children for disease\n",
      "fc add children for disease\n",
      "Conjuncts [disease, disorders, brain]\n",
      "neighbour disease (\n",
      "neighbour disease AD\n",
      "neighbour disorders associated\n",
      "neighbour disorders associated\n",
      "Conjunction before fc brain\n",
      "fc add children for brain\n",
      "Conjuncts [disease, brain, disorders]\n",
      "neighbour disease (\n",
      "neighbour disease AD\n",
      "neighbour brain ,\n",
      "neighbour brain chronic\n",
      "Conjunction before fc disorders\n",
      "fc add children for disorders\n",
      "fc add children for disorders\n",
      "fc add children for disorders\n",
      "Conunct merge check: abundance GFAP\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: GFAP disease\n",
      ") and vimentin, and CDKN2A down-regulation was noted\n",
      "Conunct merge check: disease AD\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: AD DS\n",
      ") and in Down's syndrome\n",
      "Double Conjunction Found\n",
      "Conjunctions ( conj )\n",
      "abundance [between, miRNA-125b, abundance, glial, cell, markers, glial, fibrillary, acidic, protein, GFAP, vimentin, CDKN2A, down-regulation]\n",
      "GFAP [(, )]\n",
      "disease [in, advanced, 's, disease, AD, in, 's, syndrome, DS, brain, chronic, neurological, disorders, associated, with, astrogliosis]\n",
      "AD [(, )]\n",
      "DS [(, )]\n",
      "(DS, 'conj') [(, ), (, )]\n",
      "Conjunction case rule [between, abundance, glial, cell, markers, glial, fibrillary, acidic, protein, GFAP, vimentin, down-regulation]\n",
      "Conjuncts [correlation, abundance]\n",
      "neighbour correlation between\n",
      "neighbour correlation between\n",
      "Conjunction before fc abundance\n",
      "fc add children for abundance\n",
      "fc add children for abundance\n",
      "Conjuncts [noted, disease]\n",
      "neighbour noted in\n",
      "neighbour noted in\n",
      "Conjunction before fc disease\n",
      "fc add children for disease\n",
      "fc add children for disease\n",
      "fc add children for disease\n",
      "fc add children for disease\n",
      "Conjuncts [associated, astrogliosis]\n",
      "neighbour associated with\n",
      "neighbour associated with\n",
      "Conjunction before fc astrogliosis\n",
      "fc add children for astrogliosis\n",
      "Conunct merge check: abundance GFAP\n",
      "abundance and the glial cell markers glial fibrillary acidic protein\n",
      "Conunct merge check: GFAP disease\n",
      ") and vimentin, and CDKN2A down-regulation was\n",
      "Conunct merge check: disease AD\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: AD DS\n",
      ") and in Down's syndrome\n",
      "Double Conjunction Found\n",
      "Conunct merge check: DS astrogliosis\n",
      ") brain, chronic neurological disorders\n",
      "Conjunctions ( nmod )\n",
      "abundance [correlation, between, miRNA-125b, abundance]\n",
      "GFAP [(, )]\n",
      "disease [noted, in, advanced, 's, disease, AD]\n",
      "AD [(, )]\n",
      "DS [(, )]\n",
      "astrogliosis [associated, with, astrogliosis]\n",
      "(DS, 'conj') [(, ), (, )]\n",
      "SDP sent: A strong positive correlation between miRNA-125b abundance and the glial cell markers glial fibrillary acidic protein (GFAP) and vimentin, and CDKN2A down-regulation was noted in advanced Alzheimer's disease (AD) and in Down's syndrome (DS) brain, chronic neurological disorders associated with astrogliosis.\n",
      "SDP     : [(miRNA-125b, 'NOUN', 'amod', False), (abundance, 'NOUN', 'nmod', True), (down-regulation, 'NOUN', 'conj', False), (CDKN2A, 'NOUN', 'compound', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[the, glial, cell, markers, glial, fibrillary, acidic, protein, (, GFAP, ), and, vimentin]\n",
      "DeepCheck Conj Subtree\n",
      "[vimentin]\n",
      "DeepCheck Conj Subtree\n",
      "[CDKN2A, down-regulation]\n",
      "DeepCheck Conj Subtree\n",
      "[in, Down, 's, syndrome, (, DS, ), brain, ,]\n",
      "DeepCheck Conj Subtree\n",
      "[chronic, neurological, disorders, associated, with, astrogliosis]\n",
      "0 [A, strong, positive, correlation, between, miRNA-125b, abundance, and, the, glial, cell, markers, glial, fibrillary, acidic, protein, (, GFAP, ), and, vimentin, ,, and, CDKN2A, down-regulation, was, noted, in, advanced, Alzheimer, 's, disease, (, AD, ), and, in, Down, 's, syndrome, (, DS, ), brain, ,, chronic, neurological, disorders, associated, with, astrogliosis] 51 0 A strong positive correlation between miRNA-125b abundance and the glial cell markers glial fibrillary acidic protein (GFAP) and vimentin, and CDKN2A down-regulation was noted in advanced Alzheimer's disease (AD) and in Down's syndrome (DS) brain, chronic neurological disorders associated with astrogliosis.\n",
      "DeepCheck Conj Subtree\n",
      "[between, miRNA-125b, abundance, and, the, glial, cell, markers, glial, fibrillary, acidic, protein, (, GFAP, ), and, vimentin, ,, and, CDKN2A, down-regulation]\n",
      "DeepCheck Conj Subtree\n",
      "[the, glial, cell, markers, glial, fibrillary, acidic, protein, (, GFAP, ), and, vimentin]\n",
      "DeepCheck Conj Subtree\n",
      "[vimentin]\n",
      "DeepCheck Conj Subtree\n",
      "[CDKN2A, down-regulation]\n",
      "DeepCheck Conj Subtree\n",
      "[in, advanced, Alzheimer, 's, disease, (, AD, ), and, in, Down, 's, syndrome, (, DS, ), brain, ,, chronic, neurological, disorders, associated, with, astrogliosis]\n",
      "DeepCheck Conj Subtree\n",
      "[in, Down, 's, syndrome, (, DS, ), brain, ,]\n",
      "DeepCheck Conj Subtree\n",
      "[chronic, neurological, disorders, associated, with, astrogliosis]\n",
      "DeepCheck Conj Subtree\n",
      "[with, astrogliosis]\n",
      "0 [A, strong, positive, correlation, between, miRNA-125b, abundance, and, the, glial, cell, markers, glial, fibrillary, acidic, protein, (, GFAP, ), and, vimentin, ,, and, CDKN2A, down-regulation, was, noted, in, advanced, Alzheimer, 's, disease, (, AD, ), and, in, Down, 's, syndrome, (, DS, ), brain, ,, chronic, neurological, disorders, associated, with, astrogliosis] 51 0 A strong positive correlation between miRNA-125b abundance and the glial cell markers glial fibrillary acidic protein (GFAP) and vimentin, and CDKN2A down-regulation was noted in advanced Alzheimer's disease (AD) and in Down's syndrome (DS) brain, chronic neurological disorders associated with astrogliosis.\n",
      "Recognized overlapping entities: {Interval(143, 149, ('GENE_OR_GENE_PRODUCT', 'CDKN2A'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text A strong positive correlation between miRNA-125b abundance and the glial cell markers glial fibrillary acidic protein ( GFAP ) and vimentin , and CDKN2A down-regulation was noted in advanced Alzheimer 's disease ( AD ) and in Down 's syndrome ( DS ) brain , chronic neurological disorders associated with astrogliosis\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene pos corr', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence A strong positive correlation between miRNA-125b abundance and the glial cell markers glial fibrillary acidic protein (GFAP) and vimentin, and CDKN2A down-regulation was noted in advanced Alzheimer's disease (AD) and in Down's syndrome (DS) brain, chronic neurological disorders associated with astrogliosis.\n",
      "full sentence A strong positive correlation between <e1>miRNA-125b</e1> abundance and the glial cell markers glial fibrillary acidic protein (GFAP) and vimentin, and <e2>CDKN2A</e2> down-regulation was noted in advanced Alzheimer's disease (AD) and in Down's syndrome (DS) brain, chronic neurological disorders associated with astrogliosis.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene pos corr', 'reg_detect_major': 'compartment'}}, 'tagged_sent': \"A strong positive correlation between <e1>miRNA-125b</e1> abundance and the glial cell markers glial fibrillary acidic protein (GFAP) and vimentin, and <e2>CDKN2A</e2> down-regulation was noted in advanced Alzheimer's disease (AD) and in Down's syndrome (DS) brain, chronic neurological disorders associated with astrogliosis.\", 'full_sentence': \"A strong positive correlation between <e1>miRNA-125b</e1> abundance and the glial cell markers glial fibrillary acidic protein (GFAP) and vimentin, and <e2>CDKN2A</e2> down-regulation was noted in advanced Alzheimer's disease (AD) and in Down's syndrome (DS) brain, chronic neurological disorders associated with astrogliosis.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (38, 48), 'entity_text': 'miRNA-125b'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (143, 149), 'entity_text': 'CDKN2A'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d98.s0.p4\n",
      "IS:  ('GENE_MIR', 'UP')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"<e2>BACE1</e2> mRNA levels tended to increase as <e1>miR-107</e1> levels decreased in the progression of AD. Cell culture reporter assays performed with a subset of the predicted miR-107 binding sites indicate the presence of at least one physiological miR-107 miRNA recognition sequence in the 3'-UTR of BACE1 mRNA.\", 'full_sentence': \"<e2>BACE1</e2> mRNA levels tended to increase as <e1>miR-107</e1> levels decreased in the progression of AD. Cell culture reporter assays performed with a subset of the predicted miR-107 binding sites indicate the presence of at least one physiological miR-107 miRNA recognition sequence in the 3'-UTR of BACE1 mRNA.\", 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (0, 5), 'entity_text': 'BACE1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (40, 47), 'entity_text': 'miR-107'}}\n",
      "add special case BACE1\n",
      "add special case miR-107\n",
      "BACE1 0 miR-107 7\n",
      "(u\"BACE1 mRNA levels tended to increase as miR-107 levels decreased in the progression of AD. Cell culture reporter assays performed with a subset of the predicted miR-107 binding sites indicate the presence of at least one physiological miR-107 miRNA recognition sequence in the 3'-UTR of BACE1 mRNA.\", 0, 7)\n",
      "geneword BACE1 mirword miR-107\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [increase, levels]\n",
      "neighbour increase as\n",
      "neighbour increase as\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conjuncts [decreased, progression]\n",
      "neighbour decreased in\n",
      "neighbour decreased in\n",
      "Conjunction before fc progression\n",
      "fc add children for progression\n",
      "fc add children for progression\n",
      "Conjuncts [progression, AD]\n",
      "neighbour progression of\n",
      "neighbour progression of\n",
      "Conjunction before fc AD\n",
      "fc add children for AD\n",
      "Conjuncts [performed, subset]\n",
      "neighbour performed with\n",
      "neighbour performed with\n",
      "Conjunction before fc subset\n",
      "fc add children for subset\n",
      "fc add children for subset\n",
      "Conjuncts [subset, sites]\n",
      "neighbour subset of\n",
      "neighbour subset of\n",
      "Conjunction before fc sites\n",
      "fc add children for sites\n",
      "fc add children for sites\n",
      "fc add children for sites\n",
      "fc add children for sites\n",
      "Conjuncts [presence, sequence]\n",
      "neighbour presence of\n",
      "neighbour presence of\n",
      "Conjunction before fc sequence\n",
      "fc add children for sequence\n",
      "fc add children for sequence\n",
      "fc add children for sequence\n",
      "fc add children for sequence\n",
      "fc add children for sequence\n",
      "fc add children for sequence\n",
      "Conjuncts [presence, 3'-UTR]\n",
      "neighbour presence of\n",
      "neighbour presence of\n",
      "Conjunction before fc 3'-UTR\n",
      "fc add children for 3'-UTR\n",
      "fc add children for 3'-UTR\n",
      "Conjuncts [3'-UTR, mRNA]\n",
      "neighbour 3'-UTR of\n",
      "neighbour 3'-UTR of\n",
      "Conjunction before fc mRNA\n",
      "fc add children for mRNA\n",
      "fc add children for mRNA\n",
      "Conunct merge check: levels progression\n",
      "levels\n",
      "Conunct merge check: progression subset\n",
      "AD. Cell culture reporter assays\n",
      "Conunct merge check: subset sequence\n",
      "sites indicate the\n",
      "Conjunctions ( nmod )\n",
      "levels [increase, as, miR-107, levels]\n",
      "progression [decreased, in, progression, of, AD]\n",
      "subset [performed, with, subset, of, predicted, miR-107, binding, sites]\n",
      "sequence [presence, of, one, physiological, miR-107, miRNA, recognition, sequence, in, 3'-UTR, of, BACE1, mRNA]\n",
      "SDP sent: BACE1 mRNA levels tended to increase as miR-107 levels decreased in the progression of AD. Cell culture reporter assays performed with a subset of the predicted miR-107 binding sites indicate the presence of at least one physiological miR-107 miRNA recognition sequence in the 3'-UTR of BACE1 mRNA.\n",
      "SDP     : [(BACE1, 'NOUN', 'compound', False), (levels, 'NOUN', 'nsubj', False), (tended, 'VERB', 'ROOT', False), (increase, 'VERB', 'xcomp', False), (levels, 'NOUN', 'nmod', False), (miR-107, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule [to, increase, as, miR-107, levels]\n",
      "0 [BACE1, mRNA, levels, tended, to, increase, as, miR-107, levels] 9 0 BACE1 mRNA levels tended to increase as miR-107 levels decreased in the progression of AD. Cell culture reporter assays performed with a subset of the predicted miR-107 binding sites indicate the presence of at least one physiological miR-107 miRNA recognition sequence in the 3'-UTR of BACE1 mRNA.\n",
      "1 [decreased, in, the, progression, of, AD] 6 55 decreased in the progression of AD. Cell culture reporter assays performed with a subset of the predicted miR-107 binding sites indicate the presence of at least one physiological miR-107 miRNA recognition sequence in the 3'-UTR of BACE1 mRNA.\n",
      "2 [Cell, culture, reporter, assays, performed, with, a, subset, of, the, predicted, miR-107, binding, sites, indicate, the, presence, of, at, least, one, physiological, miR-107, miRNA, recognition, sequence, in, the, 3'-UTR, of, BACE1, mRNA] 32 91 Cell culture reporter assays performed with a subset of the predicted miR-107 binding sites indicate the presence of at least one physiological miR-107 miRNA recognition sequence in the 3'-UTR of BACE1 mRNA.\n",
      "deepcheck connection rule [to, increase, as, miR-107, levels]\n",
      "DeepCheck Conj Subtree\n",
      "[as, miR-107, levels]\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, progression, of, AD]\n",
      "DeepCheck Conj Subtree\n",
      "[of, AD]\n",
      "DeepCheck Conj Subtree\n",
      "[with, a, subset, of, the, predicted, miR-107, binding, sites]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, predicted, miR-107, binding, sites]\n",
      "DeepCheck Conj Subtree\n",
      "[of, at, least, one, physiological, miR-107, miRNA, recognition, sequence]\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, 3'-UTR, of, BACE1, mRNA]\n",
      "DeepCheck Conj Subtree\n",
      "[of, BACE1, mRNA]\n",
      "0 [BACE1, mRNA, levels, tended, to, increase, as, miR-107, levels] 9 0 BACE1 mRNA levels tended to increase as miR-107 levels decreased in the progression of AD. Cell culture reporter assays performed with a subset of the predicted miR-107 binding sites indicate the presence of at least one physiological miR-107 miRNA recognition sequence in the 3'-UTR of BACE1 mRNA.\n",
      "1 [decreased, in, the, progression, of, AD] 6 55 decreased in the progression of AD. Cell culture reporter assays performed with a subset of the predicted miR-107 binding sites indicate the presence of at least one physiological miR-107 miRNA recognition sequence in the 3'-UTR of BACE1 mRNA.\n",
      "2 [Cell, culture, reporter, assays, performed, with, a, subset, of, the, predicted, miR-107, binding, sites, indicate, the, presence, of, at, least, one, physiological, miR-107, miRNA, recognition, sequence, in, the, 3'-UTR, of, BACE1, mRNA] 32 91 Cell culture reporter assays performed with a subset of the predicted miR-107 binding sites indicate the presence of at least one physiological miR-107 miRNA recognition sequence in the 3'-UTR of BACE1 mRNA.\n",
      "Recognized overlapping entities: {Interval(0, 5, ('GENE_OR_GENE_PRODUCT', 'BACE1'))}\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "Compartment Text BACE1 mRNA levels tended to increase as miR-107 levels\n",
      "[BACE1, mRNA, levels, tended, to, increase, as, miR-107, levels]\n",
      "TextBetween: BACE1 mRNA levels tended to increase as miR-107 levels\n",
      "infer direction from token set [BACE1, mRNA, levels, tended, to, increase, as, miR-107, levels]\n",
      "stemTypes2Count: Counter({'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c598>, {'mir': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'POS': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c9d8>, {'gene': Counter({'NEU': 1}), 'mir': Counter({'POS': 1, 'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(levels, 'NEU'), (increase, 'POS'), (levels, 'NEU')]\n",
      "Between Tokens [(levels, 'NEU'), (increase, 'POS')]\n",
      "counts between final False\n",
      "['BACE1', 'mRNA', 'levels', 'tended', 'to', 'increase', 'as', 'miR-107', 'levels']\n",
      "Counter({'NEU': 2, 'POS': 1})\n",
      "[(levels, 'NEU'), (increase, 'POS'), (levels, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence BACE1 mRNA levels tended to increase as miR-107 levels decreased in the progression of AD. Cell culture reporter assays performed with a subset of the predicted miR-107 binding sites indicate the presence of at least one physiological miR-107 miRNA recognition sequence in the 3'-UTR of BACE1 mRNA.\n",
      "full sentence <e2>BACE1</e2> mRNA levels tended to increase as <e1>miR-107</e1> levels decreased in the progression of AD. Cell culture reporter assays performed with a subset of the predicted miR-107 binding sites indicate the presence of at least one physiological miR-107 miRNA recognition sequence in the 3'-UTR of BACE1 mRNA.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"<e2>BACE1</e2> mRNA levels tended to increase as <e1>miR-107</e1> levels decreased in the progression of AD. Cell culture reporter assays performed with a subset of the predicted miR-107 binding sites indicate the presence of at least one physiological miR-107 miRNA recognition sequence in the 3'-UTR of BACE1 mRNA.\", 'full_sentence': \"<e2>BACE1</e2> mRNA levels tended to increase as <e1>miR-107</e1> levels decreased in the progression of AD. Cell culture reporter assays performed with a subset of the predicted miR-107 binding sites indicate the presence of at least one physiological miR-107 miRNA recognition sequence in the 3'-UTR of BACE1 mRNA.\", 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (0, 5), 'entity_text': 'BACE1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (40, 47), 'entity_text': 'miR-107'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d98.s10.p1\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case miR-107\n",
      "add special case beta-site amyloid precursor protein-cleaving enzyme 1\n",
      "miR-107 4 1 24\n",
      "(u\"The expression of microRNA miR-107 decreases early in Alzheimer's disease and may accelerate disease progression through regulation of beta-site amyloid precursor protein-cleaving enzyme 1.\", 4, 24)\n",
      "geneword 1 mirword miR-107\n",
      "Conjuncts [accelerate, decreases]\n",
      "neighbour accelerate disease\n",
      "neighbour accelerate disease\n",
      "Conjunction before fc decreases\n",
      "Conjuncts [decreases, accelerate]\n",
      "neighbour decreases early\n",
      "neighbour decreases early\n",
      "Conjunction before fc accelerate\n",
      "fc add children for accelerate\n",
      "fc add children for accelerate\n",
      "Conjunctions ( conj )\n",
      "decreases [decreases, accelerate, disease, progression, through, regulation, of, beta-site, amyloid, protein-cleaving, enzyme, 1]\n",
      "Conjuncts [expression, miR-107]\n",
      "neighbour expression of\n",
      "neighbour expression of\n",
      "Conjunction before fc miR-107\n",
      "fc add children for miR-107\n",
      "fc add children for miR-107\n",
      "Conjuncts [early, disease]\n",
      "neighbour early in\n",
      "neighbour early in\n",
      "Conjunction before fc disease\n",
      "fc add children for disease\n",
      "fc add children for disease\n",
      "Conjuncts [accelerate, regulation]\n",
      "neighbour accelerate disease\n",
      "neighbour accelerate disease\n",
      "Conjunction before fc regulation\n",
      "fc add children for regulation\n",
      "fc add children for regulation\n",
      "Conjuncts [regulation, enzyme]\n",
      "neighbour regulation of\n",
      "neighbour regulation of\n",
      "Conjunction before fc enzyme\n",
      "fc add children for enzyme\n",
      "fc add children for enzyme\n",
      "fc add children for enzyme\n",
      "fc add children for enzyme\n",
      "fc add children for enzyme\n",
      "Conunct merge check: miR-107 disease\n",
      "miR-107 decreases\n",
      "Conunct merge check: disease regulation\n",
      "disease and may\n",
      "Double Conjunction Found\n",
      "Conjunctions ( nmod )\n",
      "miR-107 [expression, of, microRNA, miR-107]\n",
      "disease [early, in, 's, disease]\n",
      "regulation [accelerate, through, regulation, of, beta-site, amyloid, protein-cleaving, enzyme, 1]\n",
      "(regulation, 'conj') [early, in, 's, disease, accelerate, through, regulation, of, beta-site, amyloid, protein-cleaving, enzyme, 1]\n",
      "SDP sent: The expression of microRNA miR-107 decreases early in Alzheimer's disease and may accelerate disease progression through regulation of beta-site amyloid precursor protein-cleaving enzyme 1.\n",
      "SDP     : [(miR-107, 'NOUN', 'nmod', False), (expression, 'NOUN', 'nsubj', False), (decreases, 'VERB', 'ROOT', True), (accelerate, 'VERB', 'conj', False), (regulation, 'NOUN', 'nmod', False), (enzyme, 'NOUN', 'nmod', False), (1, 'NUM', 'nummod', False)]\n",
      "0 [The, expression, of, microRNA, miR-107, decreases, early, in, Alzheimer, 's, disease] 11 0 The expression of microRNA miR-107 decreases early in Alzheimer's disease and may accelerate disease progression through regulation of beta-site amyloid precursor protein-cleaving enzyme 1.\n",
      "1 [may, accelerate, disease, progression, through, regulation, of, beta-site, amyloid, precursor, protein-cleaving, enzyme, 1] 13 78 may accelerate disease progression through regulation of beta-site amyloid precursor protein-cleaving enzyme 1.\n",
      "DeepCheck Conj Subtree\n",
      "[of, microRNA, miR-107]\n",
      "DeepCheck Conj Subtree\n",
      "[in, Alzheimer, 's, disease]\n",
      "DeepCheck Conj Subtree\n",
      "[through, regulation, of, beta-site, amyloid, precursor, protein-cleaving, enzyme, 1]\n",
      "DeepCheck Conj Subtree\n",
      "[of, beta-site, amyloid, precursor, protein-cleaving, enzyme, 1]\n",
      "0 [The, expression, of, microRNA, miR-107, decreases, early, in, Alzheimer, 's, disease] 11 0 The expression of microRNA miR-107 decreases early in Alzheimer's disease and may accelerate disease progression through regulation of beta-site amyloid precursor protein-cleaving enzyme 1.\n",
      "1 [may, accelerate, disease, progression, through, regulation, of, beta-site, amyloid, precursor, protein-cleaving, enzyme, 1] 13 78 may accelerate disease progression through regulation of beta-site amyloid precursor protein-cleaving enzyme 1.\n",
      "Recognized overlapping entities: {Interval(163, 188, ('GENE_OR_GENE_PRODUCT', 'protein-cleaving enzyme 1'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "The expression of microRNA miR-107 decreases early in Alzheimer's disease and may accelerate disease progression through regulation of beta-site amyloid precursor protein-cleaving enzyme 1.\n",
      "TextBetween: of microRNA miR-107 decreases early in Alzheimer 's disease and may accelerate disease progression through regulation of beta-site amyloid precursor protein-cleaving enzyme 1 .\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between mir gene', 'reg_detect_major': 'between'}})\n",
      " hit sentence The expression of microRNA miR-107 decreases early in Alzheimer's disease and may accelerate disease progression through regulation of beta-site amyloid precursor protein-cleaving enzyme 1.\n",
      "full sentence The expression of microRNA <e1>miR-107</e1> decreases early in Alzheimer's disease and may accelerate disease progression through regulation of <e2>beta-site amyloid precursor protein-cleaving enzyme 1</e2>.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between mir gene', 'reg_detect_major': 'between'}}, 'tagged_sent': \"The expression of microRNA <e1>miR-107</e1> decreases early in Alzheimer's disease and may accelerate disease progression through regulation of <e2>beta-site amyloid precursor protein-cleaving enzyme 1</e2>.\", 'full_sentence': \"The expression of microRNA <e1>miR-107</e1> decreases early in Alzheimer's disease and may accelerate disease progression through regulation of <e2>beta-site amyloid precursor protein-cleaving enzyme 1</e2>.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (27, 34), 'entity_text': 'miR-107'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (135, 188), 'entity_text': 'beta-site amyloid precursor protein-cleaving enzyme 1'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d107.s0.p0\n",
      "IS:  ('MIR_GENE', 'UP')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"Loss of microRNA cluster <e1>miR-29a/b-1</e1> in sporadic Alzheimer's disease correlates with increased <e2>BACE1</e2>/beta-secretase expression.\", 'full_sentence': \"Loss of microRNA cluster <e1>miR-29a/b-1</e1> in sporadic Alzheimer's disease correlates with increased <e2>BACE1</e2>/beta-secretase expression.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (25, 36), 'entity_text': 'miR-29a/b-1'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (95, 100), 'entity_text': 'BACE1'}}\n",
      "add special case miR-29a/b-1\n",
      "add special case BACE1\n",
      "miR-29a/b-1 4 BACE1/beta-secretase 13\n",
      "(u\"Loss of microRNA cluster miR-29a/b-1 in sporadic Alzheimer's disease correlates with increased BACE1/beta-secretase expression.\", 4, 13)\n",
      "geneword BACE1/beta-secretase mirword miR-29a/b-1\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [Loss, miR-29a/b-1]\n",
      "neighbour Loss of\n",
      "neighbour Loss of\n",
      "Conjunction before fc miR-29a/b-1\n",
      "fc add children for miR-29a/b-1\n",
      "fc add children for miR-29a/b-1\n",
      "fc add children for miR-29a/b-1\n",
      "Conjuncts [Loss, disease]\n",
      "neighbour Loss of\n",
      "neighbour Loss of\n",
      "Conjunction before fc disease\n",
      "fc add children for disease\n",
      "fc add children for disease\n",
      "fc add children for disease\n",
      "Conjuncts [correlates, expression]\n",
      "neighbour correlates with\n",
      "neighbour correlates with\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conunct merge check: miR-29a/b-1 expression\n",
      "disease\n",
      "Conjunctions ( nmod )\n",
      "miR-29a/b-1 [Loss, of, microRNA, cluster, miR-29a/b-1, in, sporadic, 's, disease]\n",
      "expression [correlates, with, increased, BACE1/beta-secretase, expression]\n",
      "SDP sent: Loss of microRNA cluster miR-29a/b-1 in sporadic Alzheimer's disease correlates with increased BACE1/beta-secretase expression.\n",
      "SDP     : [(miR-29a/b-1, 'NOUN', 'nmod', False), (Loss, 'NOUN', 'nsubj', False), (correlates, 'VERB', 'ROOT', False), (expression, 'NOUN', 'nmod', False), (BACE1/beta-secretase, 'NUM', 'compound', False)]\n",
      "0 [Loss, of, microRNA, cluster, miR-29a/b-1, in, sporadic, Alzheimer, 's, disease, correlates, with, increased, BACE1/beta-secretase, expression] 15 0 Loss of microRNA cluster miR-29a/b-1 in sporadic Alzheimer's disease correlates with increased BACE1/beta-secretase expression.\n",
      "DeepCheck Conj Subtree\n",
      "[of, microRNA, cluster, miR-29a/b-1]\n",
      "DeepCheck Conj Subtree\n",
      "[in, sporadic, Alzheimer, 's, disease]\n",
      "DeepCheck Conj Subtree\n",
      "[with, increased, BACE1/beta-secretase, expression]\n",
      "0 [Loss, of, microRNA, cluster, miR-29a/b-1, in, sporadic, Alzheimer, 's, disease, correlates, with, increased, BACE1/beta-secretase, expression] 15 0 Loss of microRNA cluster miR-29a/b-1 in sporadic Alzheimer's disease correlates with increased BACE1/beta-secretase expression.\n",
      "Recognized overlapping entities: {Interval(95, 115, ('GENE_OR_GENE_PRODUCT', 'BACE1/beta-secretase'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text Loss of microRNA cluster miR-29a/b-1 in sporadic Alzheimer 's disease correlates with increased BACE1/beta-secretase expression\n",
      "[Loss, of, microRNA, cluster, miR-29a/b-1, in, sporadic, Alzheimer, 's, disease, correlates, with, increased, BACE1/beta-secretase, expression]\n",
      "TextBetween: Loss of microRNA cluster miR-29a/b-1 in sporadic Alzheimer 's disease correlates with increased BACE1/beta-secretase expression\n",
      "infer direction from token set [Loss, of, microRNA, cluster, miR-29a/b-1, in, sporadic, Alzheimer, 's, disease, correlates, with, increased, BACE1/beta-secretase, expression]\n",
      "stemTypes2Count: Counter({'NEG': 1, 'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62f28>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 3, 'POS': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62bf8>, {'mir': Counter({'NEU': 1}), 'gene': Counter({'POS': 1, 'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(cluster, 'NEU'), (correlates, 'NEU'), (increased, 'POS'), (expression, 'NEU')]\n",
      "Between Tokens [(correlates, 'NEU'), (increased, 'POS')]\n",
      "counts between final False\n",
      "['microRNA', 'cluster', 'miR-29a/b-1', 'in', 'sporadic', 'Alzheimer', \"'s\", 'disease', 'correlates', 'with', 'increased', 'BACE1/beta-secretase', 'expression']\n",
      "Counter({'NEU': 3, 'POS': 1})\n",
      "[(cluster, 'NEU'), (correlates, 'NEU'), (increased, 'POS'), (expression, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Loss of microRNA cluster miR-29a/b-1 in sporadic Alzheimer's disease correlates with increased BACE1/beta-secretase expression.\n",
      "full sentence Loss of microRNA cluster <e1>miR-29a/b-1</e1> in sporadic Alzheimer's disease correlates with increased <e2>BACE1</e2>/beta-secretase expression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"Loss of microRNA cluster <e1>miR-29a/b-1</e1> in sporadic Alzheimer's disease correlates with increased <e2>BACE1</e2>/beta-secretase expression.\", 'full_sentence': \"Loss of microRNA cluster <e1>miR-29a/b-1</e1> in sporadic Alzheimer's disease correlates with increased <e2>BACE1</e2>/beta-secretase expression.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (25, 36), 'entity_text': 'miR-29a/b-1'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (95, 100), 'entity_text': 'BACE1'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d107.s0.p2\n",
      "IS:  ('MIR_GENE', 'UP')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"Loss of microRNA cluster <e1>miR-29a/b-1</e1> in sporadic Alzheimer's disease correlates with increased BACE1/<e2>beta-secretase</e2> expression.\", 'full_sentence': \"Loss of microRNA cluster <e1>miR-29a/b-1</e1> in sporadic Alzheimer's disease correlates with increased BACE1/<e2>beta-secretase</e2> expression.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (25, 36), 'entity_text': 'miR-29a/b-1'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (101, 115), 'entity_text': 'beta-secretase'}}\n",
      "add special case miR-29a/b-1\n",
      "add special case beta-secretase\n",
      "miR-29a/b-1 4 BACE1/beta-secretase 13\n",
      "(u\"Loss of microRNA cluster miR-29a/b-1 in sporadic Alzheimer's disease correlates with increased BACE1/beta-secretase expression.\", 4, 13)\n",
      "geneword BACE1/beta-secretase mirword miR-29a/b-1\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [Loss, miR-29a/b-1]\n",
      "neighbour Loss of\n",
      "neighbour Loss of\n",
      "Conjunction before fc miR-29a/b-1\n",
      "fc add children for miR-29a/b-1\n",
      "fc add children for miR-29a/b-1\n",
      "fc add children for miR-29a/b-1\n",
      "Conjuncts [Loss, disease]\n",
      "neighbour Loss of\n",
      "neighbour Loss of\n",
      "Conjunction before fc disease\n",
      "fc add children for disease\n",
      "fc add children for disease\n",
      "fc add children for disease\n",
      "Conjuncts [correlates, expression]\n",
      "neighbour correlates with\n",
      "neighbour correlates with\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conunct merge check: miR-29a/b-1 expression\n",
      "disease\n",
      "Conjunctions ( nmod )\n",
      "miR-29a/b-1 [Loss, of, microRNA, cluster, miR-29a/b-1, in, sporadic, 's, disease]\n",
      "expression [correlates, with, increased, BACE1/beta-secretase, expression]\n",
      "SDP sent: Loss of microRNA cluster miR-29a/b-1 in sporadic Alzheimer's disease correlates with increased BACE1/beta-secretase expression.\n",
      "SDP     : [(miR-29a/b-1, 'NOUN', 'nmod', False), (Loss, 'NOUN', 'nsubj', False), (correlates, 'VERB', 'ROOT', False), (expression, 'NOUN', 'nmod', False), (BACE1/beta-secretase, 'NUM', 'compound', False)]\n",
      "0 [Loss, of, microRNA, cluster, miR-29a/b-1, in, sporadic, Alzheimer, 's, disease, correlates, with, increased, BACE1/beta-secretase, expression] 15 0 Loss of microRNA cluster miR-29a/b-1 in sporadic Alzheimer's disease correlates with increased BACE1/beta-secretase expression.\n",
      "DeepCheck Conj Subtree\n",
      "[of, microRNA, cluster, miR-29a/b-1]\n",
      "DeepCheck Conj Subtree\n",
      "[in, sporadic, Alzheimer, 's, disease]\n",
      "DeepCheck Conj Subtree\n",
      "[with, increased, BACE1/beta-secretase, expression]\n",
      "0 [Loss, of, microRNA, cluster, miR-29a/b-1, in, sporadic, Alzheimer, 's, disease, correlates, with, increased, BACE1/beta-secretase, expression] 15 0 Loss of microRNA cluster miR-29a/b-1 in sporadic Alzheimer's disease correlates with increased BACE1/beta-secretase expression.\n",
      "Recognized overlapping entities: {Interval(95, 115, ('GENE_OR_GENE_PRODUCT', 'BACE1/beta-secretase'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text Loss of microRNA cluster miR-29a/b-1 in sporadic Alzheimer 's disease correlates with increased BACE1/beta-secretase expression\n",
      "[Loss, of, microRNA, cluster, miR-29a/b-1, in, sporadic, Alzheimer, 's, disease, correlates, with, increased, BACE1/beta-secretase, expression]\n",
      "TextBetween: Loss of microRNA cluster miR-29a/b-1 in sporadic Alzheimer 's disease correlates with increased BACE1/beta-secretase expression\n",
      "infer direction from token set [Loss, of, microRNA, cluster, miR-29a/b-1, in, sporadic, Alzheimer, 's, disease, correlates, with, increased, BACE1/beta-secretase, expression]\n",
      "stemTypes2Count: Counter({'NEG': 1, 'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62b70>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 3, 'POS': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62c80>, {'mir': Counter({'NEU': 1}), 'gene': Counter({'POS': 1, 'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(cluster, 'NEU'), (correlates, 'NEU'), (increased, 'POS'), (expression, 'NEU')]\n",
      "Between Tokens [(correlates, 'NEU'), (increased, 'POS')]\n",
      "counts between final False\n",
      "['microRNA', 'cluster', 'miR-29a/b-1', 'in', 'sporadic', 'Alzheimer', \"'s\", 'disease', 'correlates', 'with', 'increased', 'BACE1/beta-secretase', 'expression']\n",
      "Counter({'NEU': 3, 'POS': 1})\n",
      "[(cluster, 'NEU'), (correlates, 'NEU'), (increased, 'POS'), (expression, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Loss of microRNA cluster miR-29a/b-1 in sporadic Alzheimer's disease correlates with increased BACE1/beta-secretase expression.\n",
      "full sentence Loss of microRNA cluster <e1>miR-29a/b-1</e1> in sporadic Alzheimer's disease correlates with increased BACE1/<e2>beta-secretase</e2> expression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"Loss of microRNA cluster <e1>miR-29a/b-1</e1> in sporadic Alzheimer's disease correlates with increased BACE1/<e2>beta-secretase</e2> expression.\", 'full_sentence': \"Loss of microRNA cluster <e1>miR-29a/b-1</e1> in sporadic Alzheimer's disease correlates with increased BACE1/<e2>beta-secretase</e2> expression.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (25, 36), 'entity_text': 'miR-29a/b-1'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (101, 115), 'entity_text': 'beta-secretase'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d108.s0.p1\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case GRN\n",
      "add special case miR-659\n",
      "GRN 6 miR-659 34\n",
      "(u\"Here we expand the role of GRN in FTLD-U and demonstrate that a common genetic variant (rs5848), located in the 3'-untranslated region (UTR) of GRN in a binding-site for miR-659, is a major susceptibility factor for FTLD-U.\", 6, 34)\n",
      "geneword GRN mirword miR-659\n",
      "Conjuncts [demonstrate, expand]\n",
      "neighbour demonstrate that\n",
      "neighbour demonstrate that\n",
      "Conjunction before fc expand\n",
      "fc add children for expand\n",
      "Conjuncts [expand, demonstrate]\n",
      "neighbour expand the\n",
      "neighbour expand the\n",
      "Conjunction before fc demonstrate\n",
      "Conunct merge check: expand rs5848\n",
      "demonstrate that a common genetic variant\n",
      "Conunct merge check: rs5848 UTR\n",
      "), located in the 3'-untranslated region\n",
      "Conjunctions ( conj )\n",
      "expand [expand, role, of, GRN, in, FTLD-U, demonstrate]\n",
      "rs5848 [(, )]\n",
      "UTR [(, )]\n",
      "Conjuncts [role, GRN]\n",
      "neighbour role of\n",
      "neighbour role of\n",
      "Conjunction before fc GRN\n",
      "fc add children for GRN\n",
      "Conjuncts [role, FTLD-U]\n",
      "neighbour role of\n",
      "neighbour role of\n",
      "Conjunction before fc FTLD-U\n",
      "fc add children for FTLD-U\n",
      "Conjuncts [located, region]\n",
      "neighbour located in\n",
      "neighbour located in\n",
      "Conjunction before fc region\n",
      "fc add children for region\n",
      "fc add children for region\n",
      "fc add children for region\n",
      "fc add children for region\n",
      "Conjuncts [region, GRN]\n",
      "neighbour region (\n",
      "neighbour region UTR\n",
      "Conjunction before fc GRN\n",
      "fc add children for GRN\n",
      "Conjuncts [located, binding-site]\n",
      "neighbour located in\n",
      "neighbour located in\n",
      "Conjunction before fc binding-site\n",
      "fc add children for binding-site\n",
      "fc add children for binding-site\n",
      "Conjuncts [binding-site, miR-659]\n",
      "neighbour binding-site for\n",
      "neighbour binding-site for\n",
      "Conjunction before fc miR-659\n",
      "fc add children for miR-659\n",
      "Conjuncts [factor, FTLD-U.]\n",
      "neighbour factor for\n",
      "neighbour factor for\n",
      "Conjunction before fc FTLD-U.\n",
      "fc add children for FTLD-U.\n",
      "Conunct merge check: GRN rs5848\n",
      "FTLD-U and demonstrate that a common genetic variant\n",
      "Conunct merge check: rs5848 region\n",
      "),\n",
      "Conunct merge check: region UTR\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: UTR FTLD-U.\n",
      ") of GRN in a binding-site for miR-659, is a major susceptibility\n",
      "Conjunctions ( nmod )\n",
      "GRN [role, of, GRN, in, FTLD-U]\n",
      "rs5848 [(, )]\n",
      "region [located, in, 3'-untranslated, region, UTR, of, GRN, in, binding-site, for, miR-659]\n",
      "UTR [(, )]\n",
      "FTLD-U. [factor, for, FTLD-U.]\n",
      "SDP sent: Here we expand the role of GRN in FTLD-U and demonstrate that a common genetic variant (rs5848), located in the 3'-untranslated region (UTR) of GRN in a binding-site for miR-659, is a major susceptibility factor for FTLD-U.\n",
      "SDP     : [(GRN, 'NOUN', 'nmod', False), (role, 'NOUN', 'dobj', False), (expand, 'VERB', 'ROOT', True), (demonstrate, 'VERB', 'conj', False), (factor, 'NOUN', 'ccomp', False), (variant, 'NOUN', 'nsubj', False), (located, 'ADJ', 'amod', False), (binding-site, 'NOUN', 'nmod', False), (miR-659, 'NOUN', 'nmod', False)]\n",
      "SDP DOBJ RULE\n",
      "deepcheck conj verb\n",
      "0 [Here, we, expand, the, role, of, GRN, in, FTLD-U, and, demonstrate, that, a, common, genetic, variant, (, rs5848, ), ,, located, in, the, 3'-untranslated, region, (, UTR, ), of, GRN, in, a, binding-site, for, miR-659, ,, is, a, major, susceptibility, factor, for, FTLD-U.] 43 0 Here we expand the role of GRN in FTLD-U and demonstrate that a common genetic variant (rs5848), located in the 3'-untranslated region (UTR) of GRN in a binding-site for miR-659, is a major susceptibility factor for FTLD-U.\n",
      "DeepCheck Conj Subtree\n",
      "[of, GRN]\n",
      "DeepCheck Conj Subtree\n",
      "[in, FTLD-U]\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, 3'-untranslated, region, (, UTR, ), of, GRN]\n",
      "DeepCheck Conj Subtree\n",
      "[of, GRN]\n",
      "DeepCheck Conj Subtree\n",
      "[in, a, binding-site, for, miR-659]\n",
      "DeepCheck Conj Subtree\n",
      "[for, miR-659]\n",
      "DeepCheck Conj Subtree\n",
      "[for, FTLD-U.]\n",
      "0 [Here, we, expand, the, role, of, GRN, in, FTLD-U, and, demonstrate, that, a, common, genetic, variant, (, rs5848, ), ,, located, in, the, 3'-untranslated, region, (, UTR, ), of, GRN, in, a, binding-site, for, miR-659, ,, is, a, major, susceptibility, factor, for, FTLD-U.] 43 0 Here we expand the role of GRN in FTLD-U and demonstrate that a common genetic variant (rs5848), located in the 3'-untranslated region (UTR) of GRN in a binding-site for miR-659, is a major susceptibility factor for FTLD-U.\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text Here we expand the role of GRN in FTLD-U and demonstrate that a common genetic variant ( rs5848 ) , located in the 3'-untranslated region ( UTR ) of GRN in a binding-site for miR-659 , is a major susceptibility factor for FTLD-U.\n",
      "[Here, we, expand, the, role, of, GRN, in, FTLD-U, and, demonstrate, that, a, common, genetic, variant, (, rs5848, ), ,, located, in, the, 3'-untranslated, region, (, UTR, ), of, GRN, in, a, binding-site, for, miR-659, ,, is, a, major, susceptibility, factor, for, FTLD-U.]\n",
      "TextBetween: Here we expand the role of GRN in FTLD-U and demonstrate that a common genetic variant ( rs5848 ) , located in the 3'-untranslated region ( UTR ) of GRN in a binding-site for miR-659 , is a major susceptibility factor for FTLD-U.\n",
      "infer direction from token set [Here, we, expand, the, role, of, GRN, in, FTLD-U, and, demonstrate, that, a, common, genetic, variant, (, rs5848, ), ,, located, in, the, 3'-untranslated, region, (, UTR, ), of, GRN, in, a, binding-site, for, miR-659, ,, is, a, major, susceptibility, factor, for, FTLD-U.]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8cd08>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70510>, {'mir': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(binding-site, 'NEU')]\n",
      "Between Tokens [(binding-site, 'NEU')]\n",
      "counts between final False\n",
      "['role', 'of', 'GRN', 'in', 'FTLD-U', 'and', 'demonstrate', 'that', 'a', 'common', 'genetic', 'variant', '(', 'rs5848', ')', ',', 'located', 'in', 'the', \"3'-untranslated\", 'region', '(', 'UTR', ')', 'of', 'GRN', 'in', 'a', 'binding-site', 'for', 'miR-659', ',', 'is']\n",
      "Counter({'NEU': 1})\n",
      "[(binding-site, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Here we expand the role of GRN in FTLD-U and demonstrate that a common genetic variant (rs5848), located in the 3'-untranslated region (UTR) of GRN in a binding-site for miR-659, is a major susceptibility factor for FTLD-U.\n",
      "full sentence Here we expand the role of <e2>GRN</e2> in FTLD-U and demonstrate that a common genetic variant (rs5848), located in the 3'-untranslated region (UTR) of GRN in a binding-site for <e1>miR-659</e1>, is a major susceptibility factor for FTLD-U.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"Here we expand the role of <e2>GRN</e2> in FTLD-U and demonstrate that a common genetic variant (rs5848), located in the 3'-untranslated region (UTR) of GRN in a binding-site for <e1>miR-659</e1>, is a major susceptibility factor for FTLD-U.\", 'full_sentence': \"Here we expand the role of <e2>GRN</e2> in FTLD-U and demonstrate that a common genetic variant (rs5848), located in the 3'-untranslated region (UTR) of GRN in a binding-site for <e1>miR-659</e1>, is a major susceptibility factor for FTLD-U.\", 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (27, 30), 'entity_text': 'GRN'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (170, 177), 'entity_text': 'miR-659'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d108.s0.p3\n",
      "IS:  ('GENE_MIR', 'NEU')\n",
      "SB:  ('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"Here we expand the role of GRN in FTLD-U and demonstrate that a common genetic variant (rs5848), located in the 3'-untranslated region (UTR) of <e2>GRN</e2> in a binding-site for <e1>miR-659</e1>, is a major susceptibility factor for FTLD-U.\", 'full_sentence': \"Here we expand the role of GRN in FTLD-U and demonstrate that a common genetic variant (rs5848), located in the 3'-untranslated region (UTR) of <e2>GRN</e2> in a binding-site for <e1>miR-659</e1>, is a major susceptibility factor for FTLD-U.\", 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (144, 147), 'entity_text': 'GRN'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (170, 177), 'entity_text': 'miR-659'}}\n",
      "add special case GRN\n",
      "add special case miR-659\n",
      "GRN 29 miR-659 34\n",
      "(u\"Here we expand the role of GRN in FTLD-U and demonstrate that a common genetic variant (rs5848), located in the 3'-untranslated region (UTR) of GRN in a binding-site for miR-659, is a major susceptibility factor for FTLD-U.\", 29, 34)\n",
      "geneword GRN mirword miR-659\n",
      "Conjuncts [demonstrate, expand]\n",
      "neighbour demonstrate that\n",
      "neighbour demonstrate that\n",
      "Conjunction before fc expand\n",
      "fc add children for expand\n",
      "Conjuncts [expand, demonstrate]\n",
      "neighbour expand the\n",
      "neighbour expand the\n",
      "Conjunction before fc demonstrate\n",
      "Conunct merge check: expand rs5848\n",
      "demonstrate that a common genetic variant\n",
      "Conunct merge check: rs5848 UTR\n",
      "), located in the 3'-untranslated region\n",
      "Conjunctions ( conj )\n",
      "expand [expand, role, of, GRN, in, FTLD-U, demonstrate]\n",
      "rs5848 [(, )]\n",
      "UTR [(, )]\n",
      "Conjuncts [role, GRN]\n",
      "neighbour role of\n",
      "neighbour role of\n",
      "Conjunction before fc GRN\n",
      "fc add children for GRN\n",
      "Conjuncts [role, FTLD-U]\n",
      "neighbour role of\n",
      "neighbour role of\n",
      "Conjunction before fc FTLD-U\n",
      "fc add children for FTLD-U\n",
      "Conjuncts [located, region]\n",
      "neighbour located in\n",
      "neighbour located in\n",
      "Conjunction before fc region\n",
      "fc add children for region\n",
      "fc add children for region\n",
      "fc add children for region\n",
      "fc add children for region\n",
      "Conjuncts [region, GRN]\n",
      "neighbour region (\n",
      "neighbour region UTR\n",
      "Conjunction before fc GRN\n",
      "fc add children for GRN\n",
      "Conjuncts [located, binding-site]\n",
      "neighbour located in\n",
      "neighbour located in\n",
      "Conjunction before fc binding-site\n",
      "fc add children for binding-site\n",
      "fc add children for binding-site\n",
      "Conjuncts [binding-site, miR-659]\n",
      "neighbour binding-site for\n",
      "neighbour binding-site for\n",
      "Conjunction before fc miR-659\n",
      "fc add children for miR-659\n",
      "Conjuncts [factor, FTLD-U.]\n",
      "neighbour factor for\n",
      "neighbour factor for\n",
      "Conjunction before fc FTLD-U.\n",
      "fc add children for FTLD-U.\n",
      "Conunct merge check: GRN rs5848\n",
      "FTLD-U and demonstrate that a common genetic variant\n",
      "Conunct merge check: rs5848 region\n",
      "),\n",
      "Conunct merge check: region UTR\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: UTR FTLD-U.\n",
      ") of GRN in a binding-site for miR-659, is a major susceptibility\n",
      "Conjunctions ( nmod )\n",
      "GRN [role, of, GRN, in, FTLD-U]\n",
      "rs5848 [(, )]\n",
      "region [located, in, 3'-untranslated, region, UTR, of, GRN, in, binding-site, for, miR-659]\n",
      "UTR [(, )]\n",
      "FTLD-U. [factor, for, FTLD-U.]\n",
      "achildren {miR-659}\n",
      "achildren False\n",
      "SDP sent: Here we expand the role of GRN in FTLD-U and demonstrate that a common genetic variant (rs5848), located in the 3'-untranslated region (UTR) of GRN in a binding-site for miR-659, is a major susceptibility factor for FTLD-U.\n",
      "SDP     : [(GRN, 'NOUN', 'nmod', False), (region, 'NOUN', 'nmod', False), (located, 'ADJ', 'amod', False), (binding-site, 'NOUN', 'nmod', False), (miR-659, 'NOUN', 'nmod', False)]\n",
      "deepcheck conj verb\n",
      "0 [Here, we, expand, the, role, of, GRN, in, FTLD-U, and, demonstrate, that, a, common, genetic, variant, (, rs5848, ), ,, located, in, the, 3'-untranslated, region, (, UTR, ), of, GRN, in, a, binding-site, for, miR-659, ,, is, a, major, susceptibility, factor, for, FTLD-U.] 43 0 Here we expand the role of GRN in FTLD-U and demonstrate that a common genetic variant (rs5848), located in the 3'-untranslated region (UTR) of GRN in a binding-site for miR-659, is a major susceptibility factor for FTLD-U.\n",
      "DeepCheck Conj Subtree\n",
      "[of, GRN]\n",
      "DeepCheck Conj Subtree\n",
      "[in, FTLD-U]\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, 3'-untranslated, region, (, UTR, ), of, GRN]\n",
      "DeepCheck Conj Subtree\n",
      "[of, GRN]\n",
      "DeepCheck Conj Subtree\n",
      "[in, a, binding-site, for, miR-659]\n",
      "DeepCheck Conj Subtree\n",
      "[for, miR-659]\n",
      "DeepCheck Conj Subtree\n",
      "[for, FTLD-U.]\n",
      "0 [Here, we, expand, the, role, of, GRN, in, FTLD-U, and, demonstrate, that, a, common, genetic, variant, (, rs5848, ), ,, located, in, the, 3'-untranslated, region, (, UTR, ), of, GRN, in, a, binding-site, for, miR-659, ,, is, a, major, susceptibility, factor, for, FTLD-U.] 43 0 Here we expand the role of GRN in FTLD-U and demonstrate that a common genetic variant (rs5848), located in the 3'-untranslated region (UTR) of GRN in a binding-site for miR-659, is a major susceptibility factor for FTLD-U.\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text Here we expand the role of GRN in FTLD-U and demonstrate that a common genetic variant ( rs5848 ) , located in the 3'-untranslated region ( UTR ) of GRN in a binding-site for miR-659 , is a major susceptibility factor for FTLD-U.\n",
      "[Here, we, expand, the, role, of, GRN, in, FTLD-U, and, demonstrate, that, a, common, genetic, variant, (, rs5848, ), ,, located, in, the, 3'-untranslated, region, (, UTR, ), of, GRN, in, a, binding-site, for, miR-659, ,, is, a, major, susceptibility, factor, for, FTLD-U.]\n",
      "TextBetween: Here we expand the role of GRN in FTLD-U and demonstrate that a common genetic variant ( rs5848 ) , located in the 3'-untranslated region ( UTR ) of GRN in a binding-site for miR-659 , is a major susceptibility factor for FTLD-U.\n",
      "infer direction from token set [Here, we, expand, the, role, of, GRN, in, FTLD-U, and, demonstrate, that, a, common, genetic, variant, (, rs5848, ), ,, located, in, the, 3'-untranslated, region, (, UTR, ), of, GRN, in, a, binding-site, for, miR-659, ,, is, a, major, susceptibility, factor, for, FTLD-U.]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c268>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8cbf8>, {'mir': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(binding-site, 'NEU')]\n",
      "Between Tokens [(binding-site, 'NEU')]\n",
      "counts between final False\n",
      "[')', 'of', 'GRN', 'in', 'a', 'binding-site', 'for', 'miR-659', ',', 'is']\n",
      "Counter({'NEU': 1})\n",
      "[(binding-site, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Here we expand the role of GRN in FTLD-U and demonstrate that a common genetic variant (rs5848), located in the 3'-untranslated region (UTR) of GRN in a binding-site for miR-659, is a major susceptibility factor for FTLD-U.\n",
      "full sentence Here we expand the role of GRN in FTLD-U and demonstrate that a common genetic variant (rs5848), located in the 3'-untranslated region (UTR) of <e2>GRN</e2> in a binding-site for <e1>miR-659</e1>, is a major susceptibility factor for FTLD-U.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"Here we expand the role of GRN in FTLD-U and demonstrate that a common genetic variant (rs5848), located in the 3'-untranslated region (UTR) of <e2>GRN</e2> in a binding-site for <e1>miR-659</e1>, is a major susceptibility factor for FTLD-U.\", 'full_sentence': \"Here we expand the role of GRN in FTLD-U and demonstrate that a common genetic variant (rs5848), located in the 3'-untranslated region (UTR) of <e2>GRN</e2> in a binding-site for <e1>miR-659</e1>, is a major susceptibility factor for FTLD-U.\", 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (144, 147), 'entity_text': 'GRN'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (170, 177), 'entity_text': 'miR-659'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d108.s5.p1\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case GRN\n",
      "add special case miR-659\n",
      "GRN 7 miR-659 13\n",
      "(u\"We further demonstrate that miR-659 can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\", 7, 13)\n",
      "geneword GRN mirword miR-659\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [regulate, binding]\n",
      "neighbour regulate GRN\n",
      "neighbour regulate GRN\n",
      "Conjunction before fc binding\n",
      "fc add children for binding\n",
      "fc add children for binding\n",
      "fc add children for binding\n",
      "Conjuncts [binding, T-allele]\n",
      "neighbour binding more\n",
      "neighbour binding more\n",
      "Conjunction before fc T-allele\n",
      "fc add children for T-allele\n",
      "fc add children for T-allele\n",
      "fc add children for T-allele\n",
      "fc add children for T-allele\n",
      "fc add children for T-allele\n",
      "Conjuncts [T-allele, rs5848]\n",
      "neighbour T-allele of\n",
      "neighbour T-allele of\n",
      "Conjunction before fc rs5848\n",
      "fc add children for rs5848\n",
      "Conjuncts [resulting, inhibition]\n",
      "neighbour resulting in\n",
      "neighbour resulting in\n",
      "Conjunction before fc inhibition\n",
      "fc add children for inhibition\n",
      "fc add children for inhibition\n",
      "fc add children for inhibition\n",
      "fc add children for inhibition\n",
      "Conjuncts [inhibition, GRN]\n",
      "neighbour inhibition of\n",
      "neighbour inhibition of\n",
      "Conjunction before fc GRN\n",
      "fc add children for GRN\n",
      "Conjunctions ( nmod )\n",
      "binding [regulate, with, miR-659, binding, to, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN]\n",
      "SDP sent: We further demonstrate that miR-659 can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\n",
      "SDP     : [(GRN, 'NOUN', 'compound', False), (expression, 'NOUN', 'dobj', False), (regulate, 'VERB', 'ccomp', False), (binding, 'NOUN', 'nmod', False), (miR-659, 'NOUN', 'compound', False)]\n",
      "SDP DOBJ RULE\n",
      "deepcheck connection rule noun found regulate miR-659 [that, miR-659, can, regulate, GRN, expression, in, vitro, ,, with, miR-659, binding, more, efficiently, to, the, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN]\n",
      "0 [We, further, demonstrate] 3 0 We further demonstrate that miR-659 can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\n",
      "1 [miR-659, can, regulate, GRN, expression, in, vitro, ,, with, miR-659, binding, more, efficiently, to, the, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN] 27 28 miR-659 can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\n",
      "deepcheck connection rule noun found regulate miR-659 [that, miR-659, can, regulate, GRN, expression, in, vitro, ,, with, miR-659, binding, more, efficiently, to, the, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN]\n",
      "DeepCheck Conj Subtree\n",
      "[with, miR-659, binding, more, efficiently, to, the, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN]\n",
      "DeepCheck Conj Subtree\n",
      "[to, the, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN]\n",
      "DeepCheck Conj Subtree\n",
      "[of, rs5848]\n",
      "DeepCheck Conj Subtree\n",
      "[in, augmented, translational, inhibition, of, GRN]\n",
      "DeepCheck Conj Subtree\n",
      "[of, GRN]\n",
      "0 [We, further, demonstrate] 3 0 We further demonstrate that miR-659 can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\n",
      "1 [miR-659, can, regulate, GRN, expression, in, vitro, ,, with, miR-659, binding, more, efficiently, to, the, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN] 27 28 miR-659 can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text miR-659 can regulate GRN expression in vitro , with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN\n",
      "[miR-659, can, regulate, GRN, expression, in, vitro, ,, with, miR-659, binding, more, efficiently, to, the, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN]\n",
      "TextBetween: miR-659 can regulate GRN expression in vitro , with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN\n",
      "infer direction from token set [miR-659, can, regulate, GRN, expression, in, vitro, ,, with, miR-659, binding, more, efficiently, to, the, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN]\n",
      "stemTypes2Count: Counter({'POS': 1, 'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fc0bbbd3158>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 3})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c2f0>, {'gene': Counter({'NEU': 2}), 'mir': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(regulate, 'NEU'), (expression, 'NEU'), (binding, 'NEU')]\n",
      "counts between 3 Counter() NEU\n",
      "Between Tokens []\n",
      "counts between final False\n",
      "['can', 'regulate', 'GRN', 'expression', 'in', 'vitro', ',', 'with', 'miR-659', 'binding', 'more']\n",
      "Counter()\n",
      "[]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence We further demonstrate that miR-659 can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\n",
      "full sentence We further demonstrate that miR-659 can regulate <e2>GRN</e2> expression in vitro, with <e1>miR-659</e1> binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'We further demonstrate that miR-659 can regulate <e2>GRN</e2> expression in vitro, with <e1>miR-659</e1> binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.', 'full_sentence': 'We further demonstrate that miR-659 can regulate <e2>GRN</e2> expression in vitro, with <e1>miR-659</e1> binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (49, 52), 'entity_text': 'GRN'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (79, 86), 'entity_text': 'miR-659'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d108.s5.p2\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miR-659\n",
      "add special case GRN\n",
      "miR-659 4 GRN 30\n",
      "(u\"We further demonstrate that miR-659 can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\", 4, 30)\n",
      "geneword GRN mirword miR-659\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [regulate, binding]\n",
      "neighbour regulate GRN\n",
      "neighbour regulate GRN\n",
      "Conjunction before fc binding\n",
      "fc add children for binding\n",
      "fc add children for binding\n",
      "fc add children for binding\n",
      "Conjuncts [binding, T-allele]\n",
      "neighbour binding more\n",
      "neighbour binding more\n",
      "Conjunction before fc T-allele\n",
      "fc add children for T-allele\n",
      "fc add children for T-allele\n",
      "fc add children for T-allele\n",
      "fc add children for T-allele\n",
      "fc add children for T-allele\n",
      "Conjuncts [T-allele, rs5848]\n",
      "neighbour T-allele of\n",
      "neighbour T-allele of\n",
      "Conjunction before fc rs5848\n",
      "fc add children for rs5848\n",
      "Conjuncts [resulting, inhibition]\n",
      "neighbour resulting in\n",
      "neighbour resulting in\n",
      "Conjunction before fc inhibition\n",
      "fc add children for inhibition\n",
      "fc add children for inhibition\n",
      "fc add children for inhibition\n",
      "fc add children for inhibition\n",
      "Conjuncts [inhibition, GRN]\n",
      "neighbour inhibition of\n",
      "neighbour inhibition of\n",
      "Conjunction before fc GRN\n",
      "fc add children for GRN\n",
      "Conjunctions ( nmod )\n",
      "binding [regulate, with, miR-659, binding, to, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN]\n",
      "SDP sent: We further demonstrate that miR-659 can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\n",
      "SDP     : [(miR-659, 'NOUN', 'nsubj', False), (regulate, 'VERB', 'ccomp', False), (binding, 'NOUN', 'nmod', False), (T-allele, 'NOUN', 'nmod', False), (resulting, 'VERB', 'acl', False), (inhibition, 'NOUN', 'nmod', False), (GRN, 'NOUN', 'nmod', False)]\n",
      "deepcheck connection rule noun found regulate miR-659 [that, miR-659, can, regulate, GRN, expression, in, vitro, ,, with, miR-659, binding, more, efficiently, to, the, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN]\n",
      "0 [We, further, demonstrate] 3 0 We further demonstrate that miR-659 can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\n",
      "1 [miR-659, can, regulate, GRN, expression, in, vitro, ,, with, miR-659, binding, more, efficiently, to, the, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN] 27 28 miR-659 can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\n",
      "deepcheck connection rule noun found regulate miR-659 [that, miR-659, can, regulate, GRN, expression, in, vitro, ,, with, miR-659, binding, more, efficiently, to, the, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN]\n",
      "DeepCheck Conj Subtree\n",
      "[with, miR-659, binding, more, efficiently, to, the, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN]\n",
      "DeepCheck Conj Subtree\n",
      "[to, the, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN]\n",
      "DeepCheck Conj Subtree\n",
      "[of, rs5848]\n",
      "DeepCheck Conj Subtree\n",
      "[in, augmented, translational, inhibition, of, GRN]\n",
      "DeepCheck Conj Subtree\n",
      "[of, GRN]\n",
      "0 [We, further, demonstrate] 3 0 We further demonstrate that miR-659 can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\n",
      "1 [miR-659, can, regulate, GRN, expression, in, vitro, ,, with, miR-659, binding, more, efficiently, to, the, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN] 27 28 miR-659 can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text miR-659 can regulate GRN expression in vitro , with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN\n",
      "[miR-659, can, regulate, GRN, expression, in, vitro, ,, with, miR-659, binding, more, efficiently, to, the, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN]\n",
      "TextBetween: miR-659 can regulate GRN expression in vitro , with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between mir gene reg corr', 'reg_detect_major': 'between'}})\n",
      " hit sentence We further demonstrate that miR-659 can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\n",
      "full sentence We further demonstrate that <e1>miR-659</e1> can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of <e2>GRN</e2>.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between mir gene reg corr', 'reg_detect_major': 'between'}}, 'tagged_sent': 'We further demonstrate that <e1>miR-659</e1> can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of <e2>GRN</e2>.', 'full_sentence': 'We further demonstrate that <e1>miR-659</e1> can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of <e2>GRN</e2>.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (28, 35), 'entity_text': 'miR-659'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (199, 202), 'entity_text': 'GRN'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d108.s5.p3\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between mir gene reg corr', 'reg_detect_major': 'between'}}, 'tagged_sent': 'We further demonstrate that <e1>miR-659</e1> can regulate <e2>GRN</e2> expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.', 'full_sentence': 'We further demonstrate that <e1>miR-659</e1> can regulate <e2>GRN</e2> expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (28, 35), 'entity_text': 'miR-659'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (49, 52), 'entity_text': 'GRN'}}\n",
      "add special case miR-659\n",
      "add special case GRN\n",
      "miR-659 4 GRN 7\n",
      "(u\"We further demonstrate that miR-659 can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\", 4, 7)\n",
      "geneword GRN mirword miR-659\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [regulate, binding]\n",
      "neighbour regulate GRN\n",
      "neighbour regulate GRN\n",
      "Conjunction before fc binding\n",
      "fc add children for binding\n",
      "fc add children for binding\n",
      "fc add children for binding\n",
      "Conjuncts [binding, T-allele]\n",
      "neighbour binding more\n",
      "neighbour binding more\n",
      "Conjunction before fc T-allele\n",
      "fc add children for T-allele\n",
      "fc add children for T-allele\n",
      "fc add children for T-allele\n",
      "fc add children for T-allele\n",
      "fc add children for T-allele\n",
      "Conjuncts [T-allele, rs5848]\n",
      "neighbour T-allele of\n",
      "neighbour T-allele of\n",
      "Conjunction before fc rs5848\n",
      "fc add children for rs5848\n",
      "Conjuncts [resulting, inhibition]\n",
      "neighbour resulting in\n",
      "neighbour resulting in\n",
      "Conjunction before fc inhibition\n",
      "fc add children for inhibition\n",
      "fc add children for inhibition\n",
      "fc add children for inhibition\n",
      "fc add children for inhibition\n",
      "Conjuncts [inhibition, GRN]\n",
      "neighbour inhibition of\n",
      "neighbour inhibition of\n",
      "Conjunction before fc GRN\n",
      "fc add children for GRN\n",
      "Conjunctions ( nmod )\n",
      "binding [regulate, with, miR-659, binding, to, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN]\n",
      "SDP sent: We further demonstrate that miR-659 can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\n",
      "SDP     : [(miR-659, 'NOUN', 'nsubj', False), (regulate, 'VERB', 'ccomp', False), (expression, 'NOUN', 'dobj', False), (GRN, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule noun found regulate miR-659 [that, miR-659, can, regulate, GRN, expression, in, vitro, ,, with, miR-659, binding, more, efficiently, to, the, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN]\n",
      "0 [We, further, demonstrate] 3 0 We further demonstrate that miR-659 can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\n",
      "1 [miR-659, can, regulate, GRN, expression, in, vitro, ,, with, miR-659, binding, more, efficiently, to, the, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN] 27 28 miR-659 can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\n",
      "deepcheck connection rule noun found regulate miR-659 [that, miR-659, can, regulate, GRN, expression, in, vitro, ,, with, miR-659, binding, more, efficiently, to, the, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN]\n",
      "DeepCheck Conj Subtree\n",
      "[with, miR-659, binding, more, efficiently, to, the, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN]\n",
      "DeepCheck Conj Subtree\n",
      "[to, the, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN]\n",
      "DeepCheck Conj Subtree\n",
      "[of, rs5848]\n",
      "DeepCheck Conj Subtree\n",
      "[in, augmented, translational, inhibition, of, GRN]\n",
      "DeepCheck Conj Subtree\n",
      "[of, GRN]\n",
      "0 [We, further, demonstrate] 3 0 We further demonstrate that miR-659 can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\n",
      "1 [miR-659, can, regulate, GRN, expression, in, vitro, ,, with, miR-659, binding, more, efficiently, to, the, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN] 27 28 miR-659 can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text miR-659 can regulate GRN expression in vitro , with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN\n",
      "[miR-659, can, regulate, GRN, expression, in, vitro, ,, with, miR-659, binding, more, efficiently, to, the, high, risk, T-allele, of, rs5848, resulting, in, augmented, translational, inhibition, of, GRN]\n",
      "TextBetween: miR-659 can regulate GRN expression in vitro , with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between mir gene reg corr', 'reg_detect_major': 'between'}})\n",
      " hit sentence We further demonstrate that miR-659 can regulate GRN expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\n",
      "full sentence We further demonstrate that <e1>miR-659</e1> can regulate <e2>GRN</e2> expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between mir gene reg corr', 'reg_detect_major': 'between'}}, 'tagged_sent': 'We further demonstrate that <e1>miR-659</e1> can regulate <e2>GRN</e2> expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.', 'full_sentence': 'We further demonstrate that <e1>miR-659</e1> can regulate <e2>GRN</e2> expression in vitro, with miR-659 binding more efficiently to the high risk T-allele of rs5848 resulting in augmented translational inhibition of GRN.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (28, 35), 'entity_text': 'miR-659'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (49, 52), 'entity_text': 'GRN'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d114.s4.p0\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miRNA-146a\n",
      "add special case CFH\n",
      "miRNA-146a-promoter-luciferase 2 CFH 21\n",
      "(u\"An NF-kappaB-containing miRNA-146a-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of miRNA-146a after aluminum-sulfate treatment that corresponded to decreased CFH gene expression.\", 2, 21)\n",
      "geneword CFH mirword miRNA-146a-promoter-luciferase\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [transfected, cells]\n",
      "neighbour transfected into\n",
      "neighbour transfected into\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conjuncts [up-regulation, miRNA-146a]\n",
      "neighbour up-regulation of\n",
      "neighbour up-regulation of\n",
      "Conjunction before fc miRNA-146a\n",
      "fc add children for miRNA-146a\n",
      "Conjuncts [up-regulation, treatment]\n",
      "neighbour up-regulation of\n",
      "neighbour up-regulation of\n",
      "Conjunction before fc treatment\n",
      "fc add children for treatment\n",
      "fc add children for treatment\n",
      "Conjuncts [corresponded, expression]\n",
      "neighbour corresponded to\n",
      "neighbour corresponded to\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conunct merge check: cells miRNA-146a\n",
      "cells showed significant\n",
      "Conunct merge check: miRNA-146a expression\n",
      "treatment that\n",
      "Conjunctions ( nmod )\n",
      "cells [transfected, into, HN, cells]\n",
      "miRNA-146a [up-regulation, of, miRNA-146a, after, aluminum-sulfate, treatment]\n",
      "expression [corresponded, to, decreased, CFH, gene, expression]\n",
      "SDP sent: An NF-kappaB-containing miRNA-146a-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of miRNA-146a after aluminum-sulfate treatment that corresponded to decreased CFH gene expression.\n",
      "SDP     : [(miRNA-146a-promoter-luciferase, 'NOUN', 'compound', False), (construct, 'NOUN', 'nsubj', False), (showed, 'VERB', 'ROOT', False), (up-regulation, 'NOUN', 'dobj', False), (treatment, 'NOUN', 'nmod', False), (corresponded, 'VERB', 'acl:relcl', False), (expression, 'NOUN', 'nmod', False), (CFH, 'NOUN', 'compound', False)]\n",
      "0 [An, NF-kappaB-containing, miRNA-146a-promoter-luciferase, reporter, construct, transfected, into, HN, cells, showed, significant, up-regulation, of, miRNA-146a, after, aluminum-sulfate, treatment, that, corresponded, to, decreased, CFH, gene, expression] 24 0 An NF-kappaB-containing miRNA-146a-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of miRNA-146a after aluminum-sulfate treatment that corresponded to decreased CFH gene expression.\n",
      "DeepCheck Conj Subtree\n",
      "[into, HN, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[of, miRNA-146a]\n",
      "DeepCheck Conj Subtree\n",
      "[after, aluminum-sulfate, treatment, that, corresponded, to, decreased, CFH, gene, expression]\n",
      "DeepCheck Conj Subtree\n",
      "[to, decreased, CFH, gene, expression]\n",
      "0 [An, NF-kappaB-containing, miRNA-146a-promoter-luciferase, reporter, construct, transfected, into, HN, cells, showed, significant, up-regulation, of, miRNA-146a, after, aluminum-sulfate, treatment, that, corresponded, to, decreased, CFH, gene, expression] 24 0 An NF-kappaB-containing miRNA-146a-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of miRNA-146a after aluminum-sulfate treatment that corresponded to decreased CFH gene expression.\n",
      "Recognized overlapping entities: {Interval(211, 214, ('GENE_OR_GENE_PRODUCT', 'CFH'))}\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "Compartment Text An NF-kappaB-containing miRNA-146a-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of miRNA-146a after aluminum-sulfate treatment that corresponded to decreased CFH gene expression\n",
      "[An, NF-kappaB-containing, miRNA-146a-promoter-luciferase, reporter, construct, transfected, into, HN, cells, showed, significant, up-regulation, of, miRNA-146a, after, aluminum-sulfate, treatment, that, corresponded, to, decreased, CFH, gene, expression]\n",
      "TextBetween: An NF-kappaB-containing miRNA-146a-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of miRNA-146a after aluminum-sulfate treatment that corresponded to decreased CFH gene expression\n",
      "infer direction from token set [An, NF-kappaB-containing, miRNA-146a-promoter-luciferase, reporter, construct, transfected, into, HN, cells, showed, significant, up-regulation, of, miRNA-146a, after, aluminum-sulfate, treatment, that, corresponded, to, decreased, CFH, gene, expression]\n",
      "stemTypes2Count: Counter({'POS': 1, 'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8cf28>, {'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'POS': 1, 'NEG': 1, 'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8cd90>, {'gene': Counter({'NEG': 1, 'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(up-regulation, 'POS'), (decreased, 'NEG'), (expression, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}})\n",
      " hit sentence An NF-kappaB-containing miRNA-146a-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of miRNA-146a after aluminum-sulfate treatment that corresponded to decreased CFH gene expression.\n",
      "full sentence An NF-kappaB-containing <e1>miRNA-146a</e1>-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of miRNA-146a after aluminum-sulfate treatment that corresponded to decreased <e2>CFH</e2> gene expression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'An NF-kappaB-containing <e1>miRNA-146a</e1>-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of miRNA-146a after aluminum-sulfate treatment that corresponded to decreased <e2>CFH</e2> gene expression.', 'full_sentence': 'An NF-kappaB-containing <e1>miRNA-146a</e1>-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of miRNA-146a after aluminum-sulfate treatment that corresponded to decreased <e2>CFH</e2> gene expression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (24, 34), 'entity_text': 'miRNA-146a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (211, 214), 'entity_text': 'CFH'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d114.s4.p1\n",
      "IS:  ('GENE_MIR', 'DOWN')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'An NF-kappaB-containing miRNA-146a-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of <e1>miRNA-146a</e1> after aluminum-sulfate treatment that corresponded to decreased <e2>CFH</e2> gene expression.', 'full_sentence': 'An NF-kappaB-containing miRNA-146a-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of <e1>miRNA-146a</e1> after aluminum-sulfate treatment that corresponded to decreased <e2>CFH</e2> gene expression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (136, 146), 'entity_text': 'miRNA-146a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (211, 214), 'entity_text': 'CFH'}}\n",
      "add special case miRNA-146a\n",
      "add special case CFH\n",
      "miRNA-146a 13 CFH 21\n",
      "(u\"An NF-kappaB-containing miRNA-146a-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of miRNA-146a after aluminum-sulfate treatment that corresponded to decreased CFH gene expression.\", 13, 21)\n",
      "geneword CFH mirword miRNA-146a\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [transfected, cells]\n",
      "neighbour transfected into\n",
      "neighbour transfected into\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conjuncts [up-regulation, miRNA-146a]\n",
      "neighbour up-regulation of\n",
      "neighbour up-regulation of\n",
      "Conjunction before fc miRNA-146a\n",
      "fc add children for miRNA-146a\n",
      "Conjuncts [up-regulation, treatment]\n",
      "neighbour up-regulation of\n",
      "neighbour up-regulation of\n",
      "Conjunction before fc treatment\n",
      "fc add children for treatment\n",
      "fc add children for treatment\n",
      "Conjuncts [corresponded, expression]\n",
      "neighbour corresponded to\n",
      "neighbour corresponded to\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conunct merge check: cells miRNA-146a\n",
      "cells showed significant\n",
      "Conunct merge check: miRNA-146a expression\n",
      "treatment that\n",
      "Conjunctions ( nmod )\n",
      "cells [transfected, into, HN, cells]\n",
      "miRNA-146a [up-regulation, of, miRNA-146a, after, aluminum-sulfate, treatment]\n",
      "expression [corresponded, to, decreased, CFH, gene, expression]\n",
      "SDP sent: An NF-kappaB-containing miRNA-146a-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of miRNA-146a after aluminum-sulfate treatment that corresponded to decreased CFH gene expression.\n",
      "SDP     : [(miRNA-146a, 'NOUN', 'nmod', False), (up-regulation, 'NOUN', 'dobj', False), (treatment, 'NOUN', 'nmod', False), (corresponded, 'VERB', 'acl:relcl', False), (expression, 'NOUN', 'nmod', False), (CFH, 'NOUN', 'compound', False)]\n",
      "SDP DOBJ RULE\n",
      "0 [An, NF-kappaB-containing, miRNA-146a-promoter-luciferase, reporter, construct, transfected, into, HN, cells, showed, significant, up-regulation, of, miRNA-146a, after, aluminum-sulfate, treatment, that, corresponded, to, decreased, CFH, gene, expression] 24 0 An NF-kappaB-containing miRNA-146a-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of miRNA-146a after aluminum-sulfate treatment that corresponded to decreased CFH gene expression.\n",
      "DeepCheck Conj Subtree\n",
      "[into, HN, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[of, miRNA-146a]\n",
      "DeepCheck Conj Subtree\n",
      "[after, aluminum-sulfate, treatment, that, corresponded, to, decreased, CFH, gene, expression]\n",
      "DeepCheck Conj Subtree\n",
      "[to, decreased, CFH, gene, expression]\n",
      "0 [An, NF-kappaB-containing, miRNA-146a-promoter-luciferase, reporter, construct, transfected, into, HN, cells, showed, significant, up-regulation, of, miRNA-146a, after, aluminum-sulfate, treatment, that, corresponded, to, decreased, CFH, gene, expression] 24 0 An NF-kappaB-containing miRNA-146a-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of miRNA-146a after aluminum-sulfate treatment that corresponded to decreased CFH gene expression.\n",
      "Recognized overlapping entities: {Interval(211, 214, ('GENE_OR_GENE_PRODUCT', 'CFH'))}\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "Compartment Text An NF-kappaB-containing miRNA-146a-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of miRNA-146a after aluminum-sulfate treatment that corresponded to decreased CFH gene expression\n",
      "[An, NF-kappaB-containing, miRNA-146a-promoter-luciferase, reporter, construct, transfected, into, HN, cells, showed, significant, up-regulation, of, miRNA-146a, after, aluminum-sulfate, treatment, that, corresponded, to, decreased, CFH, gene, expression]\n",
      "TextBetween: An NF-kappaB-containing miRNA-146a-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of miRNA-146a after aluminum-sulfate treatment that corresponded to decreased CFH gene expression\n",
      "infer direction from token set [An, NF-kappaB-containing, miRNA-146a-promoter-luciferase, reporter, construct, transfected, into, HN, cells, showed, significant, up-regulation, of, miRNA-146a, after, aluminum-sulfate, treatment, that, corresponded, to, decreased, CFH, gene, expression]\n",
      "stemTypes2Count: Counter({'POS': 1, 'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c840>, {'mir': Counter({'POS': 1}), 'gene': Counter({'NEG': 1})})\n",
      "defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c840>, {'mir': Counter({'POS': 1}), 'gene': Counter({'NEG': 1})})\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}})\n",
      " hit sentence An NF-kappaB-containing miRNA-146a-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of miRNA-146a after aluminum-sulfate treatment that corresponded to decreased CFH gene expression.\n",
      "full sentence An NF-kappaB-containing miRNA-146a-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of <e1>miRNA-146a</e1> after aluminum-sulfate treatment that corresponded to decreased <e2>CFH</e2> gene expression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'An NF-kappaB-containing miRNA-146a-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of <e1>miRNA-146a</e1> after aluminum-sulfate treatment that corresponded to decreased <e2>CFH</e2> gene expression.', 'full_sentence': 'An NF-kappaB-containing miRNA-146a-promoter-luciferase reporter construct transfected into HN cells showed significant up-regulation of <e1>miRNA-146a</e1> after aluminum-sulfate treatment that corresponded to decreased <e2>CFH</e2> gene expression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (136, 146), 'entity_text': 'miRNA-146a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (211, 214), 'entity_text': 'CFH'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d114.s5.p1\n",
      "IS:  ('MIR_GENE', 'NEU')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'These data suggest that (1) as in AD brain, NF-kappaB-sensitive, <e1>miRNA-146a</e1>-mediated, modulation of <e2>CFH</e2> gene expression may contribute to inflammatory responses in aluminum-stressed HN cells, and (2) underscores the potential of nanomolar aluminum to drive genotoxic mechanisms characteristic of neurodegenerative disease processes.', 'full_sentence': 'These data suggest that (1) as in AD brain, NF-kappaB-sensitive, <e1>miRNA-146a</e1>-mediated, modulation of <e2>CFH</e2> gene expression may contribute to inflammatory responses in aluminum-stressed HN cells, and (2) underscores the potential of nanomolar aluminum to drive genotoxic mechanisms characteristic of neurodegenerative disease processes.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (65, 75), 'entity_text': 'miRNA-146a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (100, 103), 'entity_text': 'CFH'}}\n",
      "add special case miRNA-146a\n",
      "add special case CFH\n",
      "miRNA-146a-mediated 14 CFH 18\n",
      "(u\"These data suggest that (1) as in AD brain, NF-kappaB-sensitive, miRNA-146a-mediated, modulation of CFH gene expression may contribute to inflammatory responses in aluminum-stressed HN cells, and (2) underscores the potential of nanomolar aluminum to drive genotoxic mechanisms characteristic of neurodegenerative disease processes.\", 14, 18)\n",
      "geneword CFH mirword miRNA-146a-mediated\n",
      "Conjuncts [miRNA-146a-mediated, NF-kappaB-sensitive]\n",
      "neighbour miRNA-146a-mediated ,\n",
      "neighbour miRNA-146a-mediated modulation\n",
      "Conjunction before fc NF-kappaB-sensitive\n",
      "Conjuncts [NF-kappaB-sensitive, miRNA-146a-mediated]\n",
      "neighbour NF-kappaB-sensitive ,\n",
      "neighbour NF-kappaB-sensitive miRNA-146a-mediated\n",
      "Conjunction before fc miRNA-146a-mediated\n",
      "Conjuncts [underscores, contribute]\n",
      "neighbour underscores the\n",
      "neighbour underscores the\n",
      "Conjunction before fc contribute\n",
      "fc add children for contribute\n",
      "fc add children for contribute\n",
      "fc add children for contribute\n",
      "Conjuncts [contribute, underscores]\n",
      "neighbour contribute to\n",
      "neighbour contribute to\n",
      "Conjunction before fc underscores\n",
      "fc add children for underscores\n",
      "fc add children for underscores\n",
      "Conunct merge check: contribute NF-kappaB-sensitive\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( conj )\n",
      "NF-kappaB-sensitive [NF-kappaB-sensitive, miRNA-146a-mediated]\n",
      "contribute [1, as, in, AD, brain, contribute, to, inflammatory, responses, in, aluminum-stressed, HN, cells, 2, underscores, potential, of, nanomolar, aluminum, drive, genotoxic, mechanisms, characteristic, of, neurodegenerative, disease, processes]\n",
      "Conjuncts [contribute, brain]\n",
      "neighbour contribute to\n",
      "neighbour contribute to\n",
      "Conjunction before fc brain\n",
      "fc add children for brain\n",
      "fc add children for brain\n",
      "fc add children for brain\n",
      "Conjuncts [modulation, expression]\n",
      "neighbour modulation of\n",
      "neighbour modulation of\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conjuncts [contribute, responses]\n",
      "neighbour contribute to\n",
      "neighbour contribute to\n",
      "Conjunction before fc responses\n",
      "fc add children for responses\n",
      "fc add children for responses\n",
      "fc add children for responses\n",
      "Conjuncts [responses, cells]\n",
      "neighbour responses in\n",
      "neighbour responses in\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conjuncts [potential, aluminum]\n",
      "neighbour potential of\n",
      "neighbour potential of\n",
      "Conjunction before fc aluminum\n",
      "fc add children for aluminum\n",
      "fc add children for aluminum\n",
      "Conjuncts [characteristic, processes]\n",
      "neighbour characteristic of\n",
      "neighbour characteristic of\n",
      "Conjunction before fc processes\n",
      "fc add children for processes\n",
      "fc add children for processes\n",
      "fc add children for processes\n",
      "Conunct merge check: brain expression\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: expression aluminum\n",
      "expression may contribute to inflammatory responses in aluminum-stressed HN cells, and (2) underscores the\n",
      "Conunct merge check: aluminum processes\n",
      "aluminum to drive genotoxic mechanisms\n",
      "Conjunctions ( nmod )\n",
      "brain [as, in, AD, brain, contribute, to, inflammatory, responses, in, aluminum-stressed, HN, cells]\n",
      "expression [modulation, of, CFH, gene, expression]\n",
      "aluminum [potential, of, nanomolar, aluminum]\n",
      "processes [characteristic, of, neurodegenerative, disease, processes]\n",
      "SDP sent: These data suggest that (1) as in AD brain, NF-kappaB-sensitive, miRNA-146a-mediated, modulation of CFH gene expression may contribute to inflammatory responses in aluminum-stressed HN cells, and (2) underscores the potential of nanomolar aluminum to drive genotoxic mechanisms characteristic of neurodegenerative disease processes.\n",
      "SDP     : [(miRNA-146a-mediated, 'ADJ', 'conj', False), (NF-kappaB-sensitive, 'ADJ', 'amod', True), (modulation, 'NOUN', 'nsubj', False), (expression, 'NOUN', 'nmod', False), (CFH, 'NOUN', 'compound', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[miRNA-146a-mediated]\n",
      "0 [These, data, suggest] 3 0 These data suggest that (1) as in AD brain, NF-kappaB-sensitive, miRNA-146a-mediated, modulation of CFH gene expression may contribute to inflammatory responses in aluminum-stressed HN cells, and (2) underscores the potential of nanomolar aluminum to drive genotoxic mechanisms characteristic of neurodegenerative disease processes.\n",
      "1 [1, ), as, in, AD, brain, ,, NF-kappaB-sensitive, ,, miRNA-146a-mediated, ,, modulation, of, CFH, gene, expression, may, contribute, to, inflammatory, responses, in, aluminum-stressed, HN, cells] 25 25 1) as in AD brain, NF-kappaB-sensitive, miRNA-146a-mediated, modulation of CFH gene expression may contribute to inflammatory responses in aluminum-stressed HN cells, and (2) underscores the potential of nanomolar aluminum to drive genotoxic mechanisms characteristic of neurodegenerative disease processes.\n",
      "2 [2, ), underscores, the, potential, of, nanomolar, aluminum, to, drive, genotoxic, mechanisms, characteristic, of, neurodegenerative, disease, processes] 17 197 2) underscores the potential of nanomolar aluminum to drive genotoxic mechanisms characteristic of neurodegenerative disease processes.\n",
      "DeepCheck Conj Subtree\n",
      "[as, in, AD, brain]\n",
      "DeepCheck Conj Subtree\n",
      "[miRNA-146a-mediated]\n",
      "DeepCheck Conj Subtree\n",
      "[of, CFH, gene, expression]\n",
      "DeepCheck Conj Subtree\n",
      "[to, inflammatory, responses, in, aluminum-stressed, HN, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[in, aluminum-stressed, HN, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[of, nanomolar, aluminum]\n",
      "DeepCheck Conj Subtree\n",
      "[of, neurodegenerative, disease, processes]\n",
      "0 [These, data, suggest] 3 0 These data suggest that (1) as in AD brain, NF-kappaB-sensitive, miRNA-146a-mediated, modulation of CFH gene expression may contribute to inflammatory responses in aluminum-stressed HN cells, and (2) underscores the potential of nanomolar aluminum to drive genotoxic mechanisms characteristic of neurodegenerative disease processes.\n",
      "1 [1, ), as, in, AD, brain, ,, NF-kappaB-sensitive, ,, miRNA-146a-mediated, ,, modulation, of, CFH, gene, expression, may, contribute, to, inflammatory, responses, in, aluminum-stressed, HN, cells] 25 25 1) as in AD brain, NF-kappaB-sensitive, miRNA-146a-mediated, modulation of CFH gene expression may contribute to inflammatory responses in aluminum-stressed HN cells, and (2) underscores the potential of nanomolar aluminum to drive genotoxic mechanisms characteristic of neurodegenerative disease processes.\n",
      "2 [2, ), underscores, the, potential, of, nanomolar, aluminum, to, drive, genotoxic, mechanisms, characteristic, of, neurodegenerative, disease, processes] 17 197 2) underscores the potential of nanomolar aluminum to drive genotoxic mechanisms characteristic of neurodegenerative disease processes.\n",
      "Recognized overlapping entities: {Interval(100, 103, ('GENE_OR_GENE_PRODUCT', 'CFH'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text 1 ) as in AD brain , NF-kappaB-sensitive , miRNA-146a-mediated , modulation of CFH gene expression may contribute to inflammatory responses in aluminum-stressed HN cells\n",
      "[1, ), as, in, AD, brain, ,, NF-kappaB-sensitive, ,, miRNA-146a-mediated, ,, modulation, of, CFH, gene, expression, may, contribute, to, inflammatory, responses, in, aluminum-stressed, HN, cells]\n",
      "TextBetween: 1 ) as in AD brain , NF-kappaB-sensitive , miRNA-146a-mediated , modulation of CFH gene expression may contribute to inflammatory responses in aluminum-stressed HN cells\n",
      "infer direction from token set [1, ), as, in, AD, brain, ,, NF-kappaB-sensitive, ,, miRNA-146a-mediated, ,, modulation, of, CFH, gene, expression, may, contribute, to, inflammatory, responses, in, aluminum-stressed, HN, cells]\n",
      "stemTypes2Count: Counter({'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e28c8>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e2620>, {'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(modulation, 'NEU'), (expression, 'NEU')]\n",
      "Between Tokens [(modulation, 'NEU')]\n",
      "counts between final False\n",
      "['NF-kappaB-sensitive', ',', 'miRNA-146a-mediated', ',', 'modulation', 'of', 'CFH', 'gene', 'expression']\n",
      "Counter({'NEU': 2})\n",
      "[(modulation, 'NEU'), (expression, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence These data suggest that (1) as in AD brain, NF-kappaB-sensitive, miRNA-146a-mediated, modulation of CFH gene expression may contribute to inflammatory responses in aluminum-stressed HN cells, and (2) underscores the potential of nanomolar aluminum to drive genotoxic mechanisms characteristic of neurodegenerative disease processes.\n",
      "full sentence These data suggest that (1) as in AD brain, NF-kappaB-sensitive, <e1>miRNA-146a</e1>-mediated, modulation of <e2>CFH</e2> gene expression may contribute to inflammatory responses in aluminum-stressed HN cells, and (2) underscores the potential of nanomolar aluminum to drive genotoxic mechanisms characteristic of neurodegenerative disease processes.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'These data suggest that (1) as in AD brain, NF-kappaB-sensitive, <e1>miRNA-146a</e1>-mediated, modulation of <e2>CFH</e2> gene expression may contribute to inflammatory responses in aluminum-stressed HN cells, and (2) underscores the potential of nanomolar aluminum to drive genotoxic mechanisms characteristic of neurodegenerative disease processes.', 'full_sentence': 'These data suggest that (1) as in AD brain, NF-kappaB-sensitive, <e1>miRNA-146a</e1>-mediated, modulation of <e2>CFH</e2> gene expression may contribute to inflammatory responses in aluminum-stressed HN cells, and (2) underscores the potential of nanomolar aluminum to drive genotoxic mechanisms characteristic of neurodegenerative disease processes.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (65, 75), 'entity_text': 'miRNA-146a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (100, 103), 'entity_text': 'CFH'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d116.s1.p0\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case miR-630\n",
      "add special case p27\n",
      "Pre-miR-630 0 p27(Kip1 22\n",
      "(u\"Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\", 0, 22)\n",
      "geneword p27(Kip1 mirword Pre-miR-630\n",
      "Conjuncts [resulting, correlating]\n",
      "neighbour resulting in\n",
      "neighbour resulting in\n",
      "Conjunction before fc correlating\n",
      "fc add children for correlating\n",
      "Conjuncts [rates, levels]\n",
      "neighbour rates and\n",
      "neighbour rates and\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conjuncts [levels, rates]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc rates\n",
      "fc add children for rates\n",
      "fc add children for rates\n",
      "fc add children for rates\n",
      "Conjuncts [correlating, resulting]\n",
      "neighbour correlating with\n",
      "neighbour correlating with\n",
      "Conjunction before fc resulting\n",
      "fc add children for resulting\n",
      "Conjunctions ( conj )\n",
      "correlating [correlating, with, increased, levels, of, cell, cycle, inhibitor, p27(Kip1, with, reduced, proliferation, rates, resulting, in, diminished, sensitivity, of, A549, cells, to, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "Conjuncts [cells, phase]\n",
      "neighbour cells in\n",
      "neighbour cells in\n",
      "Conjunction before fc phase\n",
      "fc add children for phase\n",
      "fc add children for phase\n",
      "fc add children for phase\n",
      "Conjuncts [phase, cycle]\n",
      "neighbour phase of\n",
      "neighbour phase of\n",
      "Conjunction before fc cycle\n",
      "fc add children for cycle\n",
      "fc add children for cycle\n",
      "Conjuncts [correlating, levels]\n",
      "neighbour correlating with\n",
      "neighbour correlating with\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conjuncts [levels, p27(Kip1]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc p27(Kip1\n",
      "fc add children for p27(Kip1\n",
      "fc add children for p27(Kip1\n",
      "fc add children for p27(Kip1\n",
      "fc add children for p27(Kip1\n",
      "Conjuncts [resulting, sensitivity]\n",
      "neighbour resulting in\n",
      "neighbour resulting in\n",
      "Conjunction before fc sensitivity\n",
      "fc add children for sensitivity\n",
      "fc add children for sensitivity\n",
      "fc add children for sensitivity\n",
      "fc add children for sensitivity\n",
      "Conjuncts [sensitivity, cells]\n",
      "neighbour sensitivity of\n",
      "neighbour sensitivity of\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conjuncts [sensitivity, arrest]\n",
      "neighbour sensitivity of\n",
      "neighbour sensitivity of\n",
      "Conjunction before fc arrest\n",
      "fc add children for arrest\n",
      "fc add children for arrest\n",
      "fc add children for arrest\n",
      "fc add children for arrest\n",
      "fc add children for arrest\n",
      "fc add children for arrest\n",
      "Conjuncts [mediated, CDDP]\n",
      "neighbour mediated by\n",
      "neighbour mediated by\n",
      "Conjunction before fc CDDP\n",
      "fc add children for CDDP\n",
      "Conunct merge check: phase levels\n",
      "cycle,\n",
      "Conunct merge check: levels sensitivity\n",
      "p27(Kip1) as well as with reduced proliferation rates and\n",
      "Conjunctions ( nmod )\n",
      "phase [cells, in, G0-G1, phase, of, cell, cycle]\n",
      "levels [correlating, with, increased, levels, of, cell, cycle, inhibitor, p27(Kip1]\n",
      "sensitivity [resulting, in, diminished, sensitivity, of, A549, cells, to, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "SDP sent: Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\n",
      "SDP     : [(Pre-miR-630, 'NOUN', 'amod', False), (cells, 'NOUN', 'ROOT', False), (correlating, 'VERB', 'acl', False), (levels, 'NOUN', 'nmod', False), (p27(Kip1, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[with, reduced, proliferation, rates]\n",
      "deepcheck conj verb\n",
      "0 [Pre-miR-630, arrested, A549, cells, in, the, G0-G1, phase, of, the, cell, cycle, ,, correlating, with, increased, levels, of, the, cell, cycle, inhibitor, p27(Kip1, ), as, well, as, with, reduced, proliferation, rates, and, resulting, in, greatly, diminished, sensitivity, of, A549, cells, to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP] 50 0 Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, G0-G1, phase, of, the, cell, cycle]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, cell, cycle]\n",
      "DeepCheck Conj Subtree\n",
      "[with, increased, levels, of, the, cell, cycle, inhibitor, p27(Kip1, ), as, well, as, with, reduced, proliferation, rates]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, cell, cycle, inhibitor, p27(Kip1, )]\n",
      "DeepCheck Conj Subtree\n",
      "[with, reduced, proliferation, rates]\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[in, greatly, diminished, sensitivity, of, A549, cells, to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "DeepCheck Conj Subtree\n",
      "[of, A549, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "DeepCheck Conj Subtree\n",
      "[by, CDDP]\n",
      "0 [Pre-miR-630, arrested, A549, cells, in, the, G0-G1, phase, of, the, cell, cycle, ,, correlating, with, increased, levels, of, the, cell, cycle, inhibitor, p27(Kip1, ), as, well, as, with, reduced, proliferation, rates, and, resulting, in, greatly, diminished, sensitivity, of, A549, cells, to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP] 50 0 Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\n",
      "Context Fail <regex.Match object; span=(25, 30), match='cells'> regex.Regex('(cells){e<=0}', flags=regex.V0)\n",
      "Recognized overlapping entities: {Interval(132, 140, ('GENE_OR_GENE_PRODUCT', 'p27(Kip1'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle , correlating with increased levels of the cell cycle inhibitor p27(Kip1 ) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP\n",
      "[Pre-miR-630, arrested, A549, cells, in, the, G0-G1, phase, of, the, cell, cycle, ,, correlating, with, increased, levels, of, the, cell, cycle, inhibitor, p27(Kip1, ), as, well, as, with, reduced, proliferation, rates, and, resulting, in, greatly, diminished, sensitivity, of, A549, cells, to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "TextBetween: Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle , correlating with increased levels of the cell cycle inhibitor p27(Kip1 ) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP\n",
      "infer direction from token set [Pre-miR-630, arrested, A549, cells, in, the, G0-G1, phase, of, the, cell, cycle, ,, correlating, with, increased, levels, of, the, cell, cycle, inhibitor, p27(Kip1, ), as, well, as, with, reduced, proliferation, rates, and, resulting, in, greatly, diminished, sensitivity, of, A549, cells, to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "stemTypes2Count: Counter({'NEG': 2, 'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c488>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'POS': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8cbf8>, {})\n",
      "counts between 1 stemTypes2Count [(correlating, 'NEU'), (increased, 'POS'), (levels, 'NEU')]\n",
      "Between Tokens [(correlating, 'NEU'), (increased, 'POS'), (levels, 'NEU')]\n",
      "counts between final False\n",
      "['Pre-miR-630', 'arrested', 'A549', 'cells', 'in', 'the', 'G0-G1', 'phase', 'of', 'the', 'cell', 'cycle', ',', 'correlating', 'with', 'increased', 'levels', 'of', 'the', 'cell', 'cycle', 'inhibitor', 'p27(Kip1', ')', 'as']\n",
      "Counter({'NEU': 2, 'POS': 1})\n",
      "[(correlating, 'NEU'), (increased, 'POS'), (levels, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\n",
      "full sentence Pre-<e1>miR-630</e1> arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor <e2>p27</e2>(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Pre-<e1>miR-630</e1> arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor <e2>p27</e2>(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.', 'full_sentence': 'Pre-<e1>miR-630</e1> arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor <e2>p27</e2>(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (4, 11), 'entity_text': 'miR-630'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (132, 135), 'entity_text': 'p27'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d116.s1.p0\n",
      "IS:  ('MIR_GENE', 'UP')\n",
      "SB:  ('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Pre-<e1>miR-630</e1> arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor <e2>p27</e2>(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.', 'full_sentence': 'Pre-<e1>miR-630</e1> arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor <e2>p27</e2>(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (4, 11), 'entity_text': 'miR-630'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (132, 135), 'entity_text': 'p27'}}\n",
      "add special case miR-630\n",
      "add special case p27\n",
      "Pre-miR-630 0 p27(Kip1 22\n",
      "(u\"Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\", 0, 22)\n",
      "geneword p27(Kip1 mirword Pre-miR-630\n",
      "Conjuncts [resulting, correlating]\n",
      "neighbour resulting in\n",
      "neighbour resulting in\n",
      "Conjunction before fc correlating\n",
      "fc add children for correlating\n",
      "Conjuncts [rates, levels]\n",
      "neighbour rates and\n",
      "neighbour rates and\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conjuncts [levels, rates]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc rates\n",
      "fc add children for rates\n",
      "fc add children for rates\n",
      "fc add children for rates\n",
      "Conjuncts [correlating, resulting]\n",
      "neighbour correlating with\n",
      "neighbour correlating with\n",
      "Conjunction before fc resulting\n",
      "fc add children for resulting\n",
      "Conjunctions ( conj )\n",
      "correlating [correlating, with, increased, levels, of, cell, cycle, inhibitor, p27(Kip1, with, reduced, proliferation, rates, resulting, in, diminished, sensitivity, of, A549, cells, to, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "Conjuncts [cells, phase]\n",
      "neighbour cells in\n",
      "neighbour cells in\n",
      "Conjunction before fc phase\n",
      "fc add children for phase\n",
      "fc add children for phase\n",
      "fc add children for phase\n",
      "Conjuncts [phase, cycle]\n",
      "neighbour phase of\n",
      "neighbour phase of\n",
      "Conjunction before fc cycle\n",
      "fc add children for cycle\n",
      "fc add children for cycle\n",
      "Conjuncts [correlating, levels]\n",
      "neighbour correlating with\n",
      "neighbour correlating with\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conjuncts [levels, p27(Kip1]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc p27(Kip1\n",
      "fc add children for p27(Kip1\n",
      "fc add children for p27(Kip1\n",
      "fc add children for p27(Kip1\n",
      "fc add children for p27(Kip1\n",
      "Conjuncts [resulting, sensitivity]\n",
      "neighbour resulting in\n",
      "neighbour resulting in\n",
      "Conjunction before fc sensitivity\n",
      "fc add children for sensitivity\n",
      "fc add children for sensitivity\n",
      "fc add children for sensitivity\n",
      "fc add children for sensitivity\n",
      "Conjuncts [sensitivity, cells]\n",
      "neighbour sensitivity of\n",
      "neighbour sensitivity of\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conjuncts [sensitivity, arrest]\n",
      "neighbour sensitivity of\n",
      "neighbour sensitivity of\n",
      "Conjunction before fc arrest\n",
      "fc add children for arrest\n",
      "fc add children for arrest\n",
      "fc add children for arrest\n",
      "fc add children for arrest\n",
      "fc add children for arrest\n",
      "fc add children for arrest\n",
      "Conjuncts [mediated, CDDP]\n",
      "neighbour mediated by\n",
      "neighbour mediated by\n",
      "Conjunction before fc CDDP\n",
      "fc add children for CDDP\n",
      "Conunct merge check: phase levels\n",
      "cycle,\n",
      "Conunct merge check: levels sensitivity\n",
      "p27(Kip1) as well as with reduced proliferation rates and\n",
      "Conjunctions ( nmod )\n",
      "phase [cells, in, G0-G1, phase, of, cell, cycle]\n",
      "levels [correlating, with, increased, levels, of, cell, cycle, inhibitor, p27(Kip1]\n",
      "sensitivity [resulting, in, diminished, sensitivity, of, A549, cells, to, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "SDP sent: Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\n",
      "SDP     : [(Pre-miR-630, 'NOUN', 'amod', False), (cells, 'NOUN', 'ROOT', False), (correlating, 'VERB', 'acl', False), (levels, 'NOUN', 'nmod', False), (p27(Kip1, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[with, reduced, proliferation, rates]\n",
      "deepcheck conj verb\n",
      "0 [Pre-miR-630, arrested, A549, cells, in, the, G0-G1, phase, of, the, cell, cycle, ,, correlating, with, increased, levels, of, the, cell, cycle, inhibitor, p27(Kip1, ), as, well, as, with, reduced, proliferation, rates, and, resulting, in, greatly, diminished, sensitivity, of, A549, cells, to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP] 50 0 Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, G0-G1, phase, of, the, cell, cycle]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, cell, cycle]\n",
      "DeepCheck Conj Subtree\n",
      "[with, increased, levels, of, the, cell, cycle, inhibitor, p27(Kip1, ), as, well, as, with, reduced, proliferation, rates]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, cell, cycle, inhibitor, p27(Kip1, )]\n",
      "DeepCheck Conj Subtree\n",
      "[with, reduced, proliferation, rates]\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[in, greatly, diminished, sensitivity, of, A549, cells, to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "DeepCheck Conj Subtree\n",
      "[of, A549, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "DeepCheck Conj Subtree\n",
      "[by, CDDP]\n",
      "0 [Pre-miR-630, arrested, A549, cells, in, the, G0-G1, phase, of, the, cell, cycle, ,, correlating, with, increased, levels, of, the, cell, cycle, inhibitor, p27(Kip1, ), as, well, as, with, reduced, proliferation, rates, and, resulting, in, greatly, diminished, sensitivity, of, A549, cells, to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP] 50 0 Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\n",
      "Context Fail <regex.Match object; span=(25, 30), match='cells'> regex.Regex('(cells){e<=0}', flags=regex.V0)\n",
      "Recognized overlapping entities: {Interval(132, 140, ('GENE_OR_GENE_PRODUCT', 'p27(Kip1'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle , correlating with increased levels of the cell cycle inhibitor p27(Kip1 ) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP\n",
      "[Pre-miR-630, arrested, A549, cells, in, the, G0-G1, phase, of, the, cell, cycle, ,, correlating, with, increased, levels, of, the, cell, cycle, inhibitor, p27(Kip1, ), as, well, as, with, reduced, proliferation, rates, and, resulting, in, greatly, diminished, sensitivity, of, A549, cells, to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "TextBetween: Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle , correlating with increased levels of the cell cycle inhibitor p27(Kip1 ) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP\n",
      "infer direction from token set [Pre-miR-630, arrested, A549, cells, in, the, G0-G1, phase, of, the, cell, cycle, ,, correlating, with, increased, levels, of, the, cell, cycle, inhibitor, p27(Kip1, ), as, well, as, with, reduced, proliferation, rates, and, resulting, in, greatly, diminished, sensitivity, of, A549, cells, to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "stemTypes2Count: Counter({'NEG': 2, 'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70510>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'POS': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e28c8>, {})\n",
      "counts between 1 stemTypes2Count [(correlating, 'NEU'), (increased, 'POS'), (levels, 'NEU')]\n",
      "Between Tokens [(correlating, 'NEU'), (increased, 'POS'), (levels, 'NEU')]\n",
      "counts between final False\n",
      "['Pre-miR-630', 'arrested', 'A549', 'cells', 'in', 'the', 'G0-G1', 'phase', 'of', 'the', 'cell', 'cycle', ',', 'correlating', 'with', 'increased', 'levels', 'of', 'the', 'cell', 'cycle', 'inhibitor', 'p27(Kip1', ')', 'as']\n",
      "Counter({'NEU': 2, 'POS': 1})\n",
      "[(correlating, 'NEU'), (increased, 'POS'), (levels, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\n",
      "full sentence Pre-<e1>miR-630</e1> arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor <e2>p27</e2>(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Pre-<e1>miR-630</e1> arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor <e2>p27</e2>(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.', 'full_sentence': 'Pre-<e1>miR-630</e1> arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor <e2>p27</e2>(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (4, 11), 'entity_text': 'miR-630'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (132, 135), 'entity_text': 'p27'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d116.s1.p1\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case miR-630\n",
      "add special case Kip1\n",
      "Pre-miR-630 0 p27(Kip1 22\n",
      "(u\"Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\", 0, 22)\n",
      "geneword p27(Kip1 mirword Pre-miR-630\n",
      "Conjuncts [resulting, correlating]\n",
      "neighbour resulting in\n",
      "neighbour resulting in\n",
      "Conjunction before fc correlating\n",
      "fc add children for correlating\n",
      "Conjuncts [rates, levels]\n",
      "neighbour rates and\n",
      "neighbour rates and\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conjuncts [levels, rates]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc rates\n",
      "fc add children for rates\n",
      "fc add children for rates\n",
      "fc add children for rates\n",
      "Conjuncts [correlating, resulting]\n",
      "neighbour correlating with\n",
      "neighbour correlating with\n",
      "Conjunction before fc resulting\n",
      "fc add children for resulting\n",
      "Conjunctions ( conj )\n",
      "correlating [correlating, with, increased, levels, of, cell, cycle, inhibitor, p27(Kip1, with, reduced, proliferation, rates, resulting, in, diminished, sensitivity, of, A549, cells, to, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "Conjuncts [cells, phase]\n",
      "neighbour cells in\n",
      "neighbour cells in\n",
      "Conjunction before fc phase\n",
      "fc add children for phase\n",
      "fc add children for phase\n",
      "fc add children for phase\n",
      "Conjuncts [phase, cycle]\n",
      "neighbour phase of\n",
      "neighbour phase of\n",
      "Conjunction before fc cycle\n",
      "fc add children for cycle\n",
      "fc add children for cycle\n",
      "Conjuncts [correlating, levels]\n",
      "neighbour correlating with\n",
      "neighbour correlating with\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conjuncts [levels, p27(Kip1]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc p27(Kip1\n",
      "fc add children for p27(Kip1\n",
      "fc add children for p27(Kip1\n",
      "fc add children for p27(Kip1\n",
      "fc add children for p27(Kip1\n",
      "Conjuncts [resulting, sensitivity]\n",
      "neighbour resulting in\n",
      "neighbour resulting in\n",
      "Conjunction before fc sensitivity\n",
      "fc add children for sensitivity\n",
      "fc add children for sensitivity\n",
      "fc add children for sensitivity\n",
      "fc add children for sensitivity\n",
      "Conjuncts [sensitivity, cells]\n",
      "neighbour sensitivity of\n",
      "neighbour sensitivity of\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conjuncts [sensitivity, arrest]\n",
      "neighbour sensitivity of\n",
      "neighbour sensitivity of\n",
      "Conjunction before fc arrest\n",
      "fc add children for arrest\n",
      "fc add children for arrest\n",
      "fc add children for arrest\n",
      "fc add children for arrest\n",
      "fc add children for arrest\n",
      "fc add children for arrest\n",
      "Conjuncts [mediated, CDDP]\n",
      "neighbour mediated by\n",
      "neighbour mediated by\n",
      "Conjunction before fc CDDP\n",
      "fc add children for CDDP\n",
      "Conunct merge check: phase levels\n",
      "cycle,\n",
      "Conunct merge check: levels sensitivity\n",
      "p27(Kip1) as well as with reduced proliferation rates and\n",
      "Conjunctions ( nmod )\n",
      "phase [cells, in, G0-G1, phase, of, cell, cycle]\n",
      "levels [correlating, with, increased, levels, of, cell, cycle, inhibitor, p27(Kip1]\n",
      "sensitivity [resulting, in, diminished, sensitivity, of, A549, cells, to, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "SDP sent: Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\n",
      "SDP     : [(Pre-miR-630, 'NOUN', 'amod', False), (cells, 'NOUN', 'ROOT', False), (correlating, 'VERB', 'acl', False), (levels, 'NOUN', 'nmod', False), (p27(Kip1, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[with, reduced, proliferation, rates]\n",
      "deepcheck conj verb\n",
      "0 [Pre-miR-630, arrested, A549, cells, in, the, G0-G1, phase, of, the, cell, cycle, ,, correlating, with, increased, levels, of, the, cell, cycle, inhibitor, p27(Kip1, ), as, well, as, with, reduced, proliferation, rates, and, resulting, in, greatly, diminished, sensitivity, of, A549, cells, to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP] 50 0 Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, G0-G1, phase, of, the, cell, cycle]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, cell, cycle]\n",
      "DeepCheck Conj Subtree\n",
      "[with, increased, levels, of, the, cell, cycle, inhibitor, p27(Kip1, ), as, well, as, with, reduced, proliferation, rates]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, cell, cycle, inhibitor, p27(Kip1, )]\n",
      "DeepCheck Conj Subtree\n",
      "[with, reduced, proliferation, rates]\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[in, greatly, diminished, sensitivity, of, A549, cells, to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "DeepCheck Conj Subtree\n",
      "[of, A549, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "DeepCheck Conj Subtree\n",
      "[by, CDDP]\n",
      "0 [Pre-miR-630, arrested, A549, cells, in, the, G0-G1, phase, of, the, cell, cycle, ,, correlating, with, increased, levels, of, the, cell, cycle, inhibitor, p27(Kip1, ), as, well, as, with, reduced, proliferation, rates, and, resulting, in, greatly, diminished, sensitivity, of, A549, cells, to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP] 50 0 Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\n",
      "Context Fail <regex.Match object; span=(25, 30), match='cells'> regex.Regex('(cells){e<=0}', flags=regex.V0)\n",
      "Recognized overlapping entities: {Interval(132, 140, ('GENE_OR_GENE_PRODUCT', 'p27(Kip1'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle , correlating with increased levels of the cell cycle inhibitor p27(Kip1 ) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP\n",
      "[Pre-miR-630, arrested, A549, cells, in, the, G0-G1, phase, of, the, cell, cycle, ,, correlating, with, increased, levels, of, the, cell, cycle, inhibitor, p27(Kip1, ), as, well, as, with, reduced, proliferation, rates, and, resulting, in, greatly, diminished, sensitivity, of, A549, cells, to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "TextBetween: Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle , correlating with increased levels of the cell cycle inhibitor p27(Kip1 ) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP\n",
      "infer direction from token set [Pre-miR-630, arrested, A549, cells, in, the, G0-G1, phase, of, the, cell, cycle, ,, correlating, with, increased, levels, of, the, cell, cycle, inhibitor, p27(Kip1, ), as, well, as, with, reduced, proliferation, rates, and, resulting, in, greatly, diminished, sensitivity, of, A549, cells, to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "stemTypes2Count: Counter({'NEG': 2, 'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954e18>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'POS': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70510>, {})\n",
      "counts between 1 stemTypes2Count [(correlating, 'NEU'), (increased, 'POS'), (levels, 'NEU')]\n",
      "Between Tokens [(correlating, 'NEU'), (increased, 'POS'), (levels, 'NEU')]\n",
      "counts between final False\n",
      "['Pre-miR-630', 'arrested', 'A549', 'cells', 'in', 'the', 'G0-G1', 'phase', 'of', 'the', 'cell', 'cycle', ',', 'correlating', 'with', 'increased', 'levels', 'of', 'the', 'cell', 'cycle', 'inhibitor', 'p27(Kip1', ')', 'as']\n",
      "Counter({'NEU': 2, 'POS': 1})\n",
      "[(correlating, 'NEU'), (increased, 'POS'), (levels, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\n",
      "full sentence Pre-<e1>miR-630</e1> arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(<e2>Kip1</e2>) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Pre-<e1>miR-630</e1> arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(<e2>Kip1</e2>) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.', 'full_sentence': 'Pre-<e1>miR-630</e1> arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(<e2>Kip1</e2>) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (4, 11), 'entity_text': 'miR-630'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (136, 140), 'entity_text': 'Kip1'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d116.s1.p1\n",
      "IS:  ('MIR_GENE', 'UP')\n",
      "SB:  ('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Pre-<e1>miR-630</e1> arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(<e2>Kip1</e2>) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.', 'full_sentence': 'Pre-<e1>miR-630</e1> arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(<e2>Kip1</e2>) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (4, 11), 'entity_text': 'miR-630'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (136, 140), 'entity_text': 'Kip1'}}\n",
      "add special case miR-630\n",
      "add special case Kip1\n",
      "Pre-miR-630 0 p27(Kip1 22\n",
      "(u\"Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\", 0, 22)\n",
      "geneword p27(Kip1 mirword Pre-miR-630\n",
      "Conjuncts [resulting, correlating]\n",
      "neighbour resulting in\n",
      "neighbour resulting in\n",
      "Conjunction before fc correlating\n",
      "fc add children for correlating\n",
      "Conjuncts [rates, levels]\n",
      "neighbour rates and\n",
      "neighbour rates and\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conjuncts [levels, rates]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc rates\n",
      "fc add children for rates\n",
      "fc add children for rates\n",
      "fc add children for rates\n",
      "Conjuncts [correlating, resulting]\n",
      "neighbour correlating with\n",
      "neighbour correlating with\n",
      "Conjunction before fc resulting\n",
      "fc add children for resulting\n",
      "Conjunctions ( conj )\n",
      "correlating [correlating, with, increased, levels, of, cell, cycle, inhibitor, p27(Kip1, with, reduced, proliferation, rates, resulting, in, diminished, sensitivity, of, A549, cells, to, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "Conjuncts [cells, phase]\n",
      "neighbour cells in\n",
      "neighbour cells in\n",
      "Conjunction before fc phase\n",
      "fc add children for phase\n",
      "fc add children for phase\n",
      "fc add children for phase\n",
      "Conjuncts [phase, cycle]\n",
      "neighbour phase of\n",
      "neighbour phase of\n",
      "Conjunction before fc cycle\n",
      "fc add children for cycle\n",
      "fc add children for cycle\n",
      "Conjuncts [correlating, levels]\n",
      "neighbour correlating with\n",
      "neighbour correlating with\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conjuncts [levels, p27(Kip1]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc p27(Kip1\n",
      "fc add children for p27(Kip1\n",
      "fc add children for p27(Kip1\n",
      "fc add children for p27(Kip1\n",
      "fc add children for p27(Kip1\n",
      "Conjuncts [resulting, sensitivity]\n",
      "neighbour resulting in\n",
      "neighbour resulting in\n",
      "Conjunction before fc sensitivity\n",
      "fc add children for sensitivity\n",
      "fc add children for sensitivity\n",
      "fc add children for sensitivity\n",
      "fc add children for sensitivity\n",
      "Conjuncts [sensitivity, cells]\n",
      "neighbour sensitivity of\n",
      "neighbour sensitivity of\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conjuncts [sensitivity, arrest]\n",
      "neighbour sensitivity of\n",
      "neighbour sensitivity of\n",
      "Conjunction before fc arrest\n",
      "fc add children for arrest\n",
      "fc add children for arrest\n",
      "fc add children for arrest\n",
      "fc add children for arrest\n",
      "fc add children for arrest\n",
      "fc add children for arrest\n",
      "Conjuncts [mediated, CDDP]\n",
      "neighbour mediated by\n",
      "neighbour mediated by\n",
      "Conjunction before fc CDDP\n",
      "fc add children for CDDP\n",
      "Conunct merge check: phase levels\n",
      "cycle,\n",
      "Conunct merge check: levels sensitivity\n",
      "p27(Kip1) as well as with reduced proliferation rates and\n",
      "Conjunctions ( nmod )\n",
      "phase [cells, in, G0-G1, phase, of, cell, cycle]\n",
      "levels [correlating, with, increased, levels, of, cell, cycle, inhibitor, p27(Kip1]\n",
      "sensitivity [resulting, in, diminished, sensitivity, of, A549, cells, to, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "SDP sent: Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\n",
      "SDP     : [(Pre-miR-630, 'NOUN', 'amod', False), (cells, 'NOUN', 'ROOT', False), (correlating, 'VERB', 'acl', False), (levels, 'NOUN', 'nmod', False), (p27(Kip1, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[with, reduced, proliferation, rates]\n",
      "deepcheck conj verb\n",
      "0 [Pre-miR-630, arrested, A549, cells, in, the, G0-G1, phase, of, the, cell, cycle, ,, correlating, with, increased, levels, of, the, cell, cycle, inhibitor, p27(Kip1, ), as, well, as, with, reduced, proliferation, rates, and, resulting, in, greatly, diminished, sensitivity, of, A549, cells, to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP] 50 0 Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, G0-G1, phase, of, the, cell, cycle]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, cell, cycle]\n",
      "DeepCheck Conj Subtree\n",
      "[with, increased, levels, of, the, cell, cycle, inhibitor, p27(Kip1, ), as, well, as, with, reduced, proliferation, rates]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, cell, cycle, inhibitor, p27(Kip1, )]\n",
      "DeepCheck Conj Subtree\n",
      "[with, reduced, proliferation, rates]\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[in, greatly, diminished, sensitivity, of, A549, cells, to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "DeepCheck Conj Subtree\n",
      "[of, A549, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "DeepCheck Conj Subtree\n",
      "[by, CDDP]\n",
      "0 [Pre-miR-630, arrested, A549, cells, in, the, G0-G1, phase, of, the, cell, cycle, ,, correlating, with, increased, levels, of, the, cell, cycle, inhibitor, p27(Kip1, ), as, well, as, with, reduced, proliferation, rates, and, resulting, in, greatly, diminished, sensitivity, of, A549, cells, to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP] 50 0 Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\n",
      "Context Fail <regex.Match object; span=(25, 30), match='cells'> regex.Regex('(cells){e<=0}', flags=regex.V0)\n",
      "Recognized overlapping entities: {Interval(132, 140, ('GENE_OR_GENE_PRODUCT', 'p27(Kip1'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle , correlating with increased levels of the cell cycle inhibitor p27(Kip1 ) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP\n",
      "[Pre-miR-630, arrested, A549, cells, in, the, G0-G1, phase, of, the, cell, cycle, ,, correlating, with, increased, levels, of, the, cell, cycle, inhibitor, p27(Kip1, ), as, well, as, with, reduced, proliferation, rates, and, resulting, in, greatly, diminished, sensitivity, of, A549, cells, to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "TextBetween: Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle , correlating with increased levels of the cell cycle inhibitor p27(Kip1 ) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP\n",
      "infer direction from token set [Pre-miR-630, arrested, A549, cells, in, the, G0-G1, phase, of, the, cell, cycle, ,, correlating, with, increased, levels, of, the, cell, cycle, inhibitor, p27(Kip1, ), as, well, as, with, reduced, proliferation, rates, and, resulting, in, greatly, diminished, sensitivity, of, A549, cells, to, the, late, S-G2-M, cell, cycle, arrest, mediated, by, CDDP]\n",
      "stemTypes2Count: Counter({'NEG': 2, 'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954bf8>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'POS': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e2048>, {})\n",
      "counts between 1 stemTypes2Count [(correlating, 'NEU'), (increased, 'POS'), (levels, 'NEU')]\n",
      "Between Tokens [(correlating, 'NEU'), (increased, 'POS'), (levels, 'NEU')]\n",
      "counts between final False\n",
      "['Pre-miR-630', 'arrested', 'A549', 'cells', 'in', 'the', 'G0-G1', 'phase', 'of', 'the', 'cell', 'cycle', ',', 'correlating', 'with', 'increased', 'levels', 'of', 'the', 'cell', 'cycle', 'inhibitor', 'p27(Kip1', ')', 'as']\n",
      "Counter({'NEU': 2, 'POS': 1})\n",
      "[(correlating, 'NEU'), (increased, 'POS'), (levels, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Pre-miR-630 arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(Kip1) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\n",
      "full sentence Pre-<e1>miR-630</e1> arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(<e2>Kip1</e2>) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Pre-<e1>miR-630</e1> arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(<e2>Kip1</e2>) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.', 'full_sentence': 'Pre-<e1>miR-630</e1> arrested A549 cells in the G0-G1 phase of the cell cycle, correlating with increased levels of the cell cycle inhibitor p27(<e2>Kip1</e2>) as well as with reduced proliferation rates and resulting in greatly diminished sensitivity of A549 cells to the late S-G2-M cell cycle arrest mediated by CDDP.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (4, 11), 'entity_text': 'miR-630'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (136, 140), 'entity_text': 'Kip1'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d122.s6.p0\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case high-mobility group A2\n",
      "add special case let-7\n",
      "A2 6 let-7 12\n",
      "(u\"Recent studies report that high-mobility group A2 is negatively regulated by the let-7 microRNAs (miRNAs) family in vitro.\", 6, 12)\n",
      "geneword A2 mirword let-7\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [regulated, family]\n",
      "neighbour regulated by\n",
      "neighbour regulated by\n",
      "Conjunction before fc family\n",
      "fc add children for family\n",
      "fc add children for family\n",
      "fc add children for family\n",
      "fc add children for family\n",
      "Conjunctions ( nmod )\n",
      "family [regulated, by, let-7, microRNAs, miRNAs, family]\n",
      "SDP sent: Recent studies report that high-mobility group A2 is negatively regulated by the let-7 microRNAs (miRNAs) family in vitro.\n",
      "SDP     : [(A2, 'NOUN', 'nsubjpass', False), (regulated, 'VERB', 'ccomp', False), (family, 'NOUN', 'nmod', False), (let-7, 'NOUN', 'compound', False)]\n",
      "0 [Recent, studies, report] 3 0 Recent studies report that high-mobility group A2 is negatively regulated by the let-7 microRNAs (miRNAs) family in vitro.\n",
      "1 [high-mobility, group, A2, is, negatively, regulated, by, the, let-7, microRNAs, (, miRNAs, ), family, in, vitro] 16 27 high-mobility group A2 is negatively regulated by the let-7 microRNAs (miRNAs) family in vitro.\n",
      "DeepCheck Conj Subtree\n",
      "[by, the, let-7, microRNAs, (, miRNAs, ), family]\n",
      "0 [Recent, studies, report] 3 0 Recent studies report that high-mobility group A2 is negatively regulated by the let-7 microRNAs (miRNAs) family in vitro.\n",
      "1 [high-mobility, group, A2, is, negatively, regulated, by, the, let-7, microRNAs, (, miRNAs, ), family, in, vitro] 16 27 high-mobility group A2 is negatively regulated by the let-7 microRNAs (miRNAs) family in vitro.\n",
      "Recognized overlapping entities: {Interval(27, 49, ('GENE_OR_GENE_PRODUCT', 'high-mobility group A2'))}\n",
      "passive? True\n",
      "Counter()\n",
      "Compartment Text high-mobility group A2 is negatively regulated by the let-7 microRNAs ( miRNAs ) family in vitro\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'compartment gene mir by', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence Recent studies report that high-mobility group A2 is negatively regulated by the let-7 microRNAs (miRNAs) family in vitro.\n",
      "full sentence Recent studies report that <e2>high-mobility group A2</e2> is negatively regulated by the <e1>let-7</e1> microRNAs (miRNAs) family in vitro.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'compartment gene mir by', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'Recent studies report that <e2>high-mobility group A2</e2> is negatively regulated by the <e1>let-7</e1> microRNAs (miRNAs) family in vitro.', 'full_sentence': 'Recent studies report that <e2>high-mobility group A2</e2> is negatively regulated by the <e1>let-7</e1> microRNAs (miRNAs) family in vitro.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (27, 49), 'entity_text': 'high-mobility group A2'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (81, 86), 'entity_text': 'let-7'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "These findings support a causal link between let-7 and high-mobility group A2 whereby loss of let-7 expression induces high-mobility group A2 upregulation that represents an important mechanism in pituitary tumorigenesis and progression.   \n",
      "('let-7', 'Specific_miRNAs', (45, 50)) ('high-mobility group A2', 'Genes/Proteins', (55, 77))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d134.s0.p0\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miR-106b\n",
      "add special case TR II\n",
      "miR-106b 8 II 17\n",
      "(u\"Our in vitro results suggested that A42 oligomer-induced miR-106b leads to impairment in TGF- signaling through TR II, concomitant with retinoic acid-induced neurodegeneration in SH-SY5Y cells.\", 8, 17)\n",
      "geneword II mirword miR-106b\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [leads, impairment]\n",
      "neighbour leads to\n",
      "neighbour leads to\n",
      "Conjunction before fc impairment\n",
      "fc add children for impairment\n",
      "fc add children for impairment\n",
      "Conjuncts [impairment, signaling]\n",
      "neighbour impairment in\n",
      "neighbour impairment in\n",
      "Conjunction before fc signaling\n",
      "fc add children for signaling\n",
      "fc add children for signaling\n",
      "fc add children for signaling\n",
      "Conjuncts [signaling, TR]\n",
      "neighbour signaling through\n",
      "neighbour signaling through\n",
      "Conjunction before fc TR\n",
      "fc add children for TR\n",
      "fc add children for TR\n",
      "Conjuncts [concomitant, neurodegeneration]\n",
      "neighbour concomitant with\n",
      "neighbour concomitant with\n",
      "Conjunction before fc neurodegeneration\n",
      "fc add children for neurodegeneration\n",
      "fc add children for neurodegeneration\n",
      "fc add children for neurodegeneration\n",
      "Conjuncts [neurodegeneration, cells]\n",
      "neighbour neurodegeneration in\n",
      "neighbour neurodegeneration in\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conunct merge check: impairment neurodegeneration\n",
      "II,\n",
      "Conjunctions ( nmod )\n",
      "impairment [leads, to, impairment, in, TGF-, signaling, through, TR, II]\n",
      "neurodegeneration [concomitant, with, retinoic, acid-induced, neurodegeneration, in, SH-SY5Y, cells]\n",
      "SDP sent: Our in vitro results suggested that A42 oligomer-induced miR-106b leads to impairment in TGF- signaling through TR II, concomitant with retinoic acid-induced neurodegeneration in SH-SY5Y cells.\n",
      "SDP     : [(miR-106b, 'NOUN', 'nsubj', False), (leads, 'VERB', 'ccomp', False), (impairment, 'NOUN', 'nmod', False), (signaling, 'NOUN', 'nmod', False), (TR, 'NOUN', 'nmod', False), (II, 'NUM', 'nummod', False)]\n",
      "deepcheck connection rule noun found leads A42 [that, A42, oligomer-induced, miR-106b, leads, to, impairment, in, TGF-, signaling, through, TR, II, ,, concomitant, with, retinoic, acid-induced, neurodegeneration, in, SH-SY5Y, cells]\n",
      "0 [Our, in, vitro, results, suggested] 5 0 Our in vitro results suggested that A42 oligomer-induced miR-106b leads to impairment in TGF- signaling through TR II, concomitant with retinoic acid-induced neurodegeneration in SH-SY5Y cells.\n",
      "1 [A42, oligomer-induced, miR-106b, leads, to, impairment, in, TGF-, signaling, through, TR, II, ,, concomitant, with, retinoic, acid-induced, neurodegeneration, in, SH-SY5Y, cells] 21 36 A42 oligomer-induced miR-106b leads to impairment in TGF- signaling through TR II, concomitant with retinoic acid-induced neurodegeneration in SH-SY5Y cells.\n",
      "deepcheck connection rule noun found leads A42 [that, A42, oligomer-induced, miR-106b, leads, to, impairment, in, TGF-, signaling, through, TR, II, ,, concomitant, with, retinoic, acid-induced, neurodegeneration, in, SH-SY5Y, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[to, impairment, in, TGF-, signaling, through, TR, II]\n",
      "DeepCheck Conj Subtree\n",
      "[in, TGF-, signaling, through, TR, II]\n",
      "DeepCheck Conj Subtree\n",
      "[through, TR, II]\n",
      "DeepCheck Conj Subtree\n",
      "[with, retinoic, acid-induced, neurodegeneration, in, SH-SY5Y, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[in, SH-SY5Y, cells]\n",
      "0 [Our, in, vitro, results, suggested] 5 0 Our in vitro results suggested that A42 oligomer-induced miR-106b leads to impairment in TGF- signaling through TR II, concomitant with retinoic acid-induced neurodegeneration in SH-SY5Y cells.\n",
      "1 [A42, oligomer-induced, miR-106b, leads, to, impairment, in, TGF-, signaling, through, TR, II, ,, concomitant, with, retinoic, acid-induced, neurodegeneration, in, SH-SY5Y, cells] 21 36 A42 oligomer-induced miR-106b leads to impairment in TGF- signaling through TR II, concomitant with retinoic acid-induced neurodegeneration in SH-SY5Y cells.\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text A42 oligomer-induced miR-106b leads to impairment in TGF- signaling through TR II , concomitant with retinoic acid-induced neurodegeneration in SH-SY5Y cells\n",
      "[A42, oligomer-induced, miR-106b, leads, to, impairment, in, TGF-, signaling, through, TR, II, ,, concomitant, with, retinoic, acid-induced, neurodegeneration, in, SH-SY5Y, cells]\n",
      "TextBetween: A42 oligomer-induced miR-106b leads to impairment in TGF- signaling through TR II , concomitant with retinoic acid-induced neurodegeneration in SH-SY5Y cells\n",
      "infer direction from token set [A42, oligomer-induced, miR-106b, leads, to, impairment, in, TGF-, signaling, through, TR, II, ,, concomitant, with, retinoic, acid-induced, neurodegeneration, in, SH-SY5Y, cells]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e2620>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e21e0>, {})\n",
      "counts between 1 stemTypes2Count [(signaling, 'NEU')]\n",
      "Between Tokens [(signaling, 'NEU')]\n",
      "counts between final False\n",
      "['A42', 'oligomer-induced', 'miR-106b', 'leads', 'to', 'impairment', 'in', 'TGF-', 'signaling', 'through', 'TR', 'II', ',', 'concomitant']\n",
      "Counter({'NEU': 1})\n",
      "[(signaling, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Our in vitro results suggested that A42 oligomer-induced miR-106b leads to impairment in TGF- signaling through TR II, concomitant with retinoic acid-induced neurodegeneration in SH-SY5Y cells.\n",
      "full sentence Our in vitro results suggested that A42 oligomer-induced <e1>miR-106b</e1> leads to impairment in TGF- signaling through <e2>TR II</e2>, concomitant with retinoic acid-induced neurodegeneration in SH-SY5Y cells.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Our in vitro results suggested that A42 oligomer-induced <e1>miR-106b</e1> leads to impairment in TGF- signaling through <e2>TR II</e2>, concomitant with retinoic acid-induced neurodegeneration in SH-SY5Y cells.', 'full_sentence': 'Our in vitro results suggested that A42 oligomer-induced <e1>miR-106b</e1> leads to impairment in TGF- signaling through <e2>TR II</e2>, concomitant with retinoic acid-induced neurodegeneration in SH-SY5Y cells.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (58, 66), 'entity_text': 'miR-106b'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (114, 120), 'entity_text': 'TR II'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Sequence analysis revealed two putative binding sites for miR-106b in the 3' UTR of the TR II mRNA. \n",
      "('miR-106b', 'Specific_miRNAs', (58, 66)) ('TR II', 'Genes/Proteins', (88, 94))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d134.s9.p3\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case miR-106b\n",
      "add special case TR II\n",
      "miR-106b 17 II 23\n",
      "(u\"Our results showed that the expression of miR-106b was inversely correlated with TR II protein levels and miR-106b can directly inhibit the TR II translation in vitro.\", 17, 23)\n",
      "geneword II mirword miR-106b\n",
      "Conjuncts [inhibit, correlated]\n",
      "neighbour inhibit the\n",
      "neighbour inhibit the\n",
      "Conjunction before fc correlated\n",
      "fc add children for correlated\n",
      "Conjuncts [miR-106b, levels]\n",
      "neighbour miR-106b can\n",
      "neighbour miR-106b can\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conjuncts [levels, miR-106b]\n",
      "neighbour levels and\n",
      "neighbour levels and\n",
      "Conjunction before fc miR-106b\n",
      "Conjuncts [correlated, inhibit]\n",
      "neighbour correlated with\n",
      "neighbour correlated with\n",
      "Conjunction before fc inhibit\n",
      "fc add children for inhibit\n",
      "Conjunctions ( conj )\n",
      "correlated [correlated, with, TR, II, protein, levels, miR-106b, inhibit, TR, II, translation]\n",
      "achildren {miR-106b}\n",
      "achildren False\n",
      "Conjuncts [expression, miR-106b]\n",
      "neighbour expression of\n",
      "neighbour expression of\n",
      "Conjunction before fc miR-106b\n",
      "fc add children for miR-106b\n",
      "Conjuncts [correlated, levels]\n",
      "neighbour correlated with\n",
      "neighbour correlated with\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conunct merge check: miR-106b levels\n",
      "miR-106b was inversely\n",
      "Conjunctions ( nmod )\n",
      "miR-106b [expression, of, miR-106b]\n",
      "levels [correlated, with, TR, II, protein, levels]\n",
      "SDP sent: Our results showed that the expression of miR-106b was inversely correlated with TR II protein levels and miR-106b can directly inhibit the TR II translation in vitro.\n",
      "SDP     : [(miR-106b, 'NOUN', 'conj', False), (levels, 'NOUN', 'nmod', True), (correlated, 'VERB', 'ccomp', True), (inhibit, 'VERB', 'conj', False), (translation, 'NOUN', 'dobj', False), (II, 'NUM', 'nummod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-106b]\n",
      "0 [Our, results, showed] 3 0 Our results showed that the expression of miR-106b was inversely correlated with TR II protein levels and miR-106b can directly inhibit the TR II translation in vitro.\n",
      "1 [the, expression, of, miR-106b, was, inversely, correlated, with, TR, II, protein, levels, and, miR-106b] 14 24 the expression of miR-106b was inversely correlated with TR II protein levels and miR-106b can directly inhibit the TR II translation in vitro.\n",
      "2 [can, directly, inhibit, the, TR, II, translation, in, vitro] 9 116 can directly inhibit the TR II translation in vitro.\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-106b]\n",
      "DeepCheck Conj Subtree\n",
      "[with, TR, II, protein, levels, and, miR-106b]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-106b]\n",
      "0 [Our, results, showed] 3 0 Our results showed that the expression of miR-106b was inversely correlated with TR II protein levels and miR-106b can directly inhibit the TR II translation in vitro.\n",
      "1 [the, expression, of, miR-106b, was, inversely, correlated, with, TR, II, protein, levels, and, miR-106b] 14 24 the expression of miR-106b was inversely correlated with TR II protein levels and miR-106b can directly inhibit the TR II translation in vitro.\n",
      "2 [can, directly, inhibit, the, TR, II, translation, in, vitro] 9 116 can directly inhibit the TR II translation in vitro.\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Our results showed that the expression of miR-106b was inversely correlated with TR II protein levels and miR-106b can directly inhibit the TR II translation in vitro.\n",
      "TextBetween: levels and miR-106b can directly inhibit the TR II translation in\n",
      "infer direction from token set [levels, and, miR-106b, can, directly, inhibit, the, TR, II, translation, in]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954bf8>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e2510>, {'mir': Counter({'NEU': 1}), 'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(levels, 'NEU'), (inhibit, 'NEG'), (translation, 'NEU')]\n",
      "Between Tokens [(inhibit, 'NEG')]\n",
      "counts between final False\n",
      "['levels', 'and', 'miR-106b', 'can', 'directly', 'inhibit', 'the', 'TR', 'II', 'translation', 'in']\n",
      "Counter({'NEU': 2, 'NEG': 1})\n",
      "[(levels, 'NEU'), (inhibit, 'NEG'), (translation, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Our results showed that the expression of miR-106b was inversely correlated with TR II protein levels and miR-106b can directly inhibit the TR II translation in vitro.\n",
      "full sentence Our results showed that the expression of miR-106b was inversely correlated with TR II protein levels and <e1>miR-106b</e1> can directly inhibit the <e2>TR II</e2> translation in vitro.\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Our results showed that the expression of miR-106b was inversely correlated with TR II protein levels and <e1>miR-106b</e1> can directly inhibit the <e2>TR II</e2> translation in vitro.', 'full_sentence': 'Our results showed that the expression of miR-106b was inversely correlated with TR II protein levels and <e1>miR-106b</e1> can directly inhibit the <e2>TR II</e2> translation in vitro.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (107, 115), 'entity_text': 'miR-106b'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (141, 147), 'entity_text': 'TR II'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d134.s11.p3\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case Smad6/7\n",
      "add special case miR-106b\n",
      "Smad6/7 18 miR-106b 23\n",
      "(u\"Western blot analysis revealed unchanged total Smad2/3 protein levels, but reduced phospho-Smad2/3 (p-Smad2/3) and increased Smad6/7 protein levels in the miR-106b stably transfected cell line.\", 18, 23)\n",
      "geneword Smad6/7 mirword miR-106b\n",
      "Conjuncts [levels, revealed]\n",
      "neighbour levels in\n",
      "neighbour levels in\n",
      "Conjunction before fc revealed\n",
      "fc add children for revealed\n",
      "Conjuncts [revealed, levels]\n",
      "neighbour revealed unchanged\n",
      "neighbour revealed unchanged\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conunct merge check: revealed p-Smad2/3\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( conj )\n",
      "revealed [revealed, unchanged, total, Smad2/3, protein, levels, reduced, phospho-Smad2/3, p-Smad2/3, increased, Smad6/7, protein, levels, in, miR-106b, transfected, cell, line]\n",
      "p-Smad2/3 [(, )]\n",
      "achildren {miR-106b}\n",
      "achildren False\n",
      "Conjuncts [levels, miR-106b]\n",
      "neighbour levels in\n",
      "neighbour levels in\n",
      "Conjunction before fc miR-106b\n",
      "fc add children for miR-106b\n",
      "fc add children for miR-106b\n",
      "Conunct merge check: p-Smad2/3 miR-106b\n",
      ") and increased Smad6/7 protein\n",
      "Conjunctions ( nmod )\n",
      "p-Smad2/3 [(, )]\n",
      "miR-106b [levels, in, miR-106b, transfected, cell, line]\n",
      "SDP sent: Western blot analysis revealed unchanged total Smad2/3 protein levels, but reduced phospho-Smad2/3 (p-Smad2/3) and increased Smad6/7 protein levels in the miR-106b stably transfected cell line.\n",
      "SDP     : [(Smad6/7, 'NOUN', 'compound', False), (levels, 'NOUN', 'conj', False), (miR-106b, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[reduced, phospho-Smad2/3, (, p-Smad2/3, ), and, increased, Smad6/7, protein, levels, in, the, miR-106b, stably, transfected, cell, line]\n",
      "deepcheck conj verb\n",
      "deepcheck conj verb\n",
      "0 [Western, blot, analysis, revealed, unchanged, total, Smad2/3, protein, levels, ,, but, reduced, phospho-Smad2/3, (, p-Smad2/3, ), and, increased, Smad6/7, protein, levels, in, the, miR-106b, stably, transfected, cell, line] 28 0 Western blot analysis revealed unchanged total Smad2/3 protein levels, but reduced phospho-Smad2/3 (p-Smad2/3) and increased Smad6/7 protein levels in the miR-106b stably transfected cell line.\n",
      "DeepCheck Conj Subtree\n",
      "[reduced, phospho-Smad2/3, (, p-Smad2/3, ), and, increased, Smad6/7, protein, levels, in, the, miR-106b, stably, transfected, cell, line]\n",
      "deepcheck conj verb\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, miR-106b, stably, transfected, cell, line]\n",
      "0 [Western, blot, analysis, revealed, unchanged, total, Smad2/3, protein, levels, ,, but, reduced, phospho-Smad2/3, (, p-Smad2/3, ), and, increased, Smad6/7, protein, levels, in, the, miR-106b, stably, transfected, cell, line] 28 0 Western blot analysis revealed unchanged total Smad2/3 protein levels, but reduced phospho-Smad2/3 (p-Smad2/3) and increased Smad6/7 protein levels in the miR-106b stably transfected cell line.\n",
      "Context Fail for Noun Rule\n",
      "Recognized overlapping entities: {Interval(125, 132, ('GENE_OR_GENE_PRODUCT', 'Smad6/7'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text Western blot analysis revealed unchanged total Smad2/3 protein levels , but reduced phospho-Smad2/3 ( p-Smad2/3 ) and increased Smad6/7 protein levels in the miR-106b stably transfected cell line\n",
      "[Western, blot, analysis, revealed, unchanged, total, Smad2/3, protein, levels, ,, but, reduced, phospho-Smad2/3, (, p-Smad2/3, ), and, increased, Smad6/7, protein, levels, in, the, miR-106b, stably, transfected, cell, line]\n",
      "TextBetween: Western blot analysis revealed unchanged total Smad2/3 protein levels , but reduced phospho-Smad2/3 ( p-Smad2/3 ) and increased Smad6/7 protein levels in the miR-106b stably transfected cell line\n",
      "infer direction from token set [Western, blot, analysis, revealed, unchanged, total, Smad2/3, protein, levels, ,, but, reduced, phospho-Smad2/3, (, p-Smad2/3, ), and, increased, Smad6/7, protein, levels, in, the, miR-106b, stably, transfected, cell, line]\n",
      "stemTypes2Count: Counter({'NEG': 1, 'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fc0b806d158>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'POS': 1, 'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954b70>, {'gene': Counter({'POS': 1, 'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(increased, 'POS'), (levels, 'NEU')]\n",
      "Between Tokens [(levels, 'NEU')]\n",
      "counts between final False\n",
      "['and', 'increased', 'Smad6/7', 'protein', 'levels', 'in', 'the', 'miR-106b', 'stably', 'transfected']\n",
      "Counter({'POS': 1, 'NEU': 1})\n",
      "[(increased, 'POS'), (levels, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Western blot analysis revealed unchanged total Smad2/3 protein levels, but reduced phospho-Smad2/3 (p-Smad2/3) and increased Smad6/7 protein levels in the miR-106b stably transfected cell line.\n",
      "full sentence Western blot analysis revealed unchanged total Smad2/3 protein levels, but reduced phospho-Smad2/3 (p-Smad2/3) and increased <e2>Smad6/7</e2> protein levels in the <e1>miR-106b</e1> stably transfected cell line.\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Western blot analysis revealed unchanged total Smad2/3 protein levels, but reduced phospho-Smad2/3 (p-Smad2/3) and increased <e2>Smad6/7</e2> protein levels in the <e1>miR-106b</e1> stably transfected cell line.', 'full_sentence': 'Western blot analysis revealed unchanged total Smad2/3 protein levels, but reduced phospho-Smad2/3 (p-Smad2/3) and increased <e2>Smad6/7</e2> protein levels in the <e1>miR-106b</e1> stably transfected cell line.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (125, 132), 'entity_text': 'Smad6/7'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (155, 163), 'entity_text': 'miR-106b'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d134.s11.p3\n",
      "IS:  ('GENE_MIR', 'UP')\n",
      "SB:  ('MIR_GENE', 'UP', 'FALSE', 'FALSE')\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Western blot analysis revealed unchanged total Smad2/3 protein levels, but reduced phospho-Smad2/3 (p-Smad2/3) and increased <e2>Smad6/7</e2> protein levels in the <e1>miR-106b</e1> stably transfected cell line.', 'full_sentence': 'Western blot analysis revealed unchanged total Smad2/3 protein levels, but reduced phospho-Smad2/3 (p-Smad2/3) and increased <e2>Smad6/7</e2> protein levels in the <e1>miR-106b</e1> stably transfected cell line.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (125, 132), 'entity_text': 'Smad6/7'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (155, 163), 'entity_text': 'miR-106b'}}\n",
      "add special case Smad6/7\n",
      "add special case miR-106b\n",
      "Smad6/7 18 miR-106b 23\n",
      "(u\"Western blot analysis revealed unchanged total Smad2/3 protein levels, but reduced phospho-Smad2/3 (p-Smad2/3) and increased Smad6/7 protein levels in the miR-106b stably transfected cell line.\", 18, 23)\n",
      "geneword Smad6/7 mirword miR-106b\n",
      "Conjuncts [levels, revealed]\n",
      "neighbour levels in\n",
      "neighbour levels in\n",
      "Conjunction before fc revealed\n",
      "fc add children for revealed\n",
      "Conjuncts [revealed, levels]\n",
      "neighbour revealed unchanged\n",
      "neighbour revealed unchanged\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conunct merge check: revealed p-Smad2/3\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( conj )\n",
      "revealed [revealed, unchanged, total, Smad2/3, protein, levels, reduced, phospho-Smad2/3, p-Smad2/3, increased, Smad6/7, protein, levels, in, miR-106b, transfected, cell, line]\n",
      "p-Smad2/3 [(, )]\n",
      "achildren {miR-106b}\n",
      "achildren False\n",
      "Conjuncts [levels, miR-106b]\n",
      "neighbour levels in\n",
      "neighbour levels in\n",
      "Conjunction before fc miR-106b\n",
      "fc add children for miR-106b\n",
      "fc add children for miR-106b\n",
      "Conunct merge check: p-Smad2/3 miR-106b\n",
      ") and increased Smad6/7 protein\n",
      "Conjunctions ( nmod )\n",
      "p-Smad2/3 [(, )]\n",
      "miR-106b [levels, in, miR-106b, transfected, cell, line]\n",
      "SDP sent: Western blot analysis revealed unchanged total Smad2/3 protein levels, but reduced phospho-Smad2/3 (p-Smad2/3) and increased Smad6/7 protein levels in the miR-106b stably transfected cell line.\n",
      "SDP     : [(Smad6/7, 'NOUN', 'compound', False), (levels, 'NOUN', 'conj', False), (miR-106b, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[reduced, phospho-Smad2/3, (, p-Smad2/3, ), and, increased, Smad6/7, protein, levels, in, the, miR-106b, stably, transfected, cell, line]\n",
      "deepcheck conj verb\n",
      "deepcheck conj verb\n",
      "0 [Western, blot, analysis, revealed, unchanged, total, Smad2/3, protein, levels, ,, but, reduced, phospho-Smad2/3, (, p-Smad2/3, ), and, increased, Smad6/7, protein, levels, in, the, miR-106b, stably, transfected, cell, line] 28 0 Western blot analysis revealed unchanged total Smad2/3 protein levels, but reduced phospho-Smad2/3 (p-Smad2/3) and increased Smad6/7 protein levels in the miR-106b stably transfected cell line.\n",
      "DeepCheck Conj Subtree\n",
      "[reduced, phospho-Smad2/3, (, p-Smad2/3, ), and, increased, Smad6/7, protein, levels, in, the, miR-106b, stably, transfected, cell, line]\n",
      "deepcheck conj verb\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, miR-106b, stably, transfected, cell, line]\n",
      "0 [Western, blot, analysis, revealed, unchanged, total, Smad2/3, protein, levels, ,, but, reduced, phospho-Smad2/3, (, p-Smad2/3, ), and, increased, Smad6/7, protein, levels, in, the, miR-106b, stably, transfected, cell, line] 28 0 Western blot analysis revealed unchanged total Smad2/3 protein levels, but reduced phospho-Smad2/3 (p-Smad2/3) and increased Smad6/7 protein levels in the miR-106b stably transfected cell line.\n",
      "Context Fail for Noun Rule\n",
      "Recognized overlapping entities: {Interval(125, 132, ('GENE_OR_GENE_PRODUCT', 'Smad6/7'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text Western blot analysis revealed unchanged total Smad2/3 protein levels , but reduced phospho-Smad2/3 ( p-Smad2/3 ) and increased Smad6/7 protein levels in the miR-106b stably transfected cell line\n",
      "[Western, blot, analysis, revealed, unchanged, total, Smad2/3, protein, levels, ,, but, reduced, phospho-Smad2/3, (, p-Smad2/3, ), and, increased, Smad6/7, protein, levels, in, the, miR-106b, stably, transfected, cell, line]\n",
      "TextBetween: Western blot analysis revealed unchanged total Smad2/3 protein levels , but reduced phospho-Smad2/3 ( p-Smad2/3 ) and increased Smad6/7 protein levels in the miR-106b stably transfected cell line\n",
      "infer direction from token set [Western, blot, analysis, revealed, unchanged, total, Smad2/3, protein, levels, ,, but, reduced, phospho-Smad2/3, (, p-Smad2/3, ), and, increased, Smad6/7, protein, levels, in, the, miR-106b, stably, transfected, cell, line]\n",
      "stemTypes2Count: Counter({'NEG': 1, 'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e28c8>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'POS': 1, 'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e27b8>, {'gene': Counter({'POS': 1, 'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(increased, 'POS'), (levels, 'NEU')]\n",
      "Between Tokens [(levels, 'NEU')]\n",
      "counts between final False\n",
      "['and', 'increased', 'Smad6/7', 'protein', 'levels', 'in', 'the', 'miR-106b', 'stably', 'transfected']\n",
      "Counter({'POS': 1, 'NEU': 1})\n",
      "[(increased, 'POS'), (levels, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Western blot analysis revealed unchanged total Smad2/3 protein levels, but reduced phospho-Smad2/3 (p-Smad2/3) and increased Smad6/7 protein levels in the miR-106b stably transfected cell line.\n",
      "full sentence Western blot analysis revealed unchanged total Smad2/3 protein levels, but reduced phospho-Smad2/3 (p-Smad2/3) and increased <e2>Smad6/7</e2> protein levels in the <e1>miR-106b</e1> stably transfected cell line.\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Western blot analysis revealed unchanged total Smad2/3 protein levels, but reduced phospho-Smad2/3 (p-Smad2/3) and increased <e2>Smad6/7</e2> protein levels in the <e1>miR-106b</e1> stably transfected cell line.', 'full_sentence': 'Western blot analysis revealed unchanged total Smad2/3 protein levels, but reduced phospho-Smad2/3 (p-Smad2/3) and increased <e2>Smad6/7</e2> protein levels in the <e1>miR-106b</e1> stably transfected cell line.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (125, 132), 'entity_text': 'Smad6/7'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (155, 163), 'entity_text': 'miR-106b'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d142.s4.p0\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miR-298\n",
      "add special case APP\n",
      "miR-298 30 (APP)-converting 44\n",
      "(u\"Using transiently transfected murine neuronal N2a cells in culture, in parallel to a mouse model of AD, we were able to demonstrate a role for two miRNAs (miR-298 and miR-328) in the regulation of beta-amyloid (Abeta) precursor protein (APP)-converting enzyme (BACE) messenger RNA (mRNA) translation, thereby providing key insights into the molecular basis underlying BACE deregulation in AD.\", 30, 44)\n",
      "geneword (APP)-converting mirword miR-298\n",
      "Conjuncts [miR-328, miR-298]\n",
      "neighbour miR-328 )\n",
      "neighbour miR-328 in\n",
      "Conjunction before fc miR-298\n",
      "Conjuncts [miR-298, miR-328]\n",
      "neighbour miR-298 and\n",
      "neighbour miR-298 and\n",
      "Conjunction before fc miR-328\n",
      "Conunct merge check: miR-298 mRNA\n",
      "miR-328) in the regulation of beta-amyloid (Abeta) precursor protein (APP)-converting enzyme (BACE) messenger RNA\n",
      "Conjunctions ( conj )\n",
      "miR-298 [miR-298, miR-328]\n",
      "mRNA [(, )]\n",
      "Conjuncts [Using, culture]\n",
      "neighbour Using transiently\n",
      "neighbour Using transiently\n",
      "Conjunction before fc culture\n",
      "fc add children for culture\n",
      "Conjuncts [Using, parallel]\n",
      "neighbour Using transiently\n",
      "neighbour Using transiently\n",
      "Conjunction before fc parallel\n",
      "fc add children for parallel\n",
      "fc add children for parallel\n",
      "Conjuncts [parallel, model]\n",
      "neighbour parallel to\n",
      "neighbour parallel to\n",
      "Conjunction before fc model\n",
      "fc add children for model\n",
      "fc add children for model\n",
      "fc add children for model\n",
      "Conjuncts [model, AD]\n",
      "neighbour model of\n",
      "neighbour model of\n",
      "Conjunction before fc AD\n",
      "fc add children for AD\n",
      "Conjuncts [role, miRNAs]\n",
      "neighbour role for\n",
      "neighbour role for\n",
      "Conjunction before fc miRNAs\n",
      "fc add children for miRNAs\n",
      "fc add children for miRNAs\n",
      "fc add children for miRNAs\n",
      "Conjuncts [role, regulation]\n",
      "neighbour role for\n",
      "neighbour role for\n",
      "Conjunction before fc regulation\n",
      "fc add children for regulation\n",
      "fc add children for regulation\n",
      "Conjuncts [regulation, translation]\n",
      "neighbour regulation of\n",
      "neighbour regulation of\n",
      "Conjunction before fc translation\n",
      "fc add children for translation\n",
      "fc add children for translation\n",
      "fc add children for translation\n",
      "fc add children for translation\n",
      "fc add children for translation\n",
      "fc add children for translation\n",
      "fc add children for translation\n",
      "fc add children for translation\n",
      "fc add children for translation\n",
      "fc add children for translation\n",
      "Conjuncts [providing, basis]\n",
      "neighbour providing key\n",
      "neighbour providing key\n",
      "Conjunction before fc basis\n",
      "fc add children for basis\n",
      "fc add children for basis\n",
      "fc add children for basis\n",
      "Conjuncts [deregulation, AD]\n",
      "neighbour deregulation in\n",
      "neighbour deregulation in\n",
      "Conjunction before fc AD\n",
      "fc add children for AD\n",
      "Conunct merge check: culture miRNAs\n",
      "AD, we were able to demonstrate a\n",
      "Conunct merge check: miRNAs mRNA\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: mRNA basis\n",
      ") translation, thereby\n",
      "Conjunctions ( nmod )\n",
      "culture [Using, in, culture, in, parallel, to, mouse, model, of, AD]\n",
      "miRNAs [role, for, two, miRNAs, miR-298, in, regulation, of, beta-amyloid, Abeta, precursor, protein, (APP)-converting, enzyme, BACE, messenger, RNA, mRNA, translation]\n",
      "mRNA [(, )]\n",
      "basis [providing, into, molecular, basis, underlying, BACE, deregulation, in, AD]\n",
      "achildren {miR-298}\n",
      "achildren False\n",
      "SDP sent: Using transiently transfected murine neuronal N2a cells in culture, in parallel to a mouse model of AD, we were able to demonstrate a role for two miRNAs (miR-298 and miR-328) in the regulation of beta-amyloid (Abeta) precursor protein (APP)-converting enzyme (BACE) messenger RNA (mRNA) translation, thereby providing key insights into the molecular basis underlying BACE deregulation in AD.\n",
      "SDP     : [(miR-298, 'NOUN', 'dep', False), (miRNAs, 'NOUN', 'nmod', False), (role, 'NOUN', 'dobj', False), (regulation, 'NOUN', 'nmod', False), (translation, 'NOUN', 'nmod', False), ((APP)-converting, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule [to, demonstrate, a, role, for, two, miRNAs, (, miR-298, and, miR-328, ), in, the, regulation, of, beta-amyloid, (, Abeta, ), precursor, protein, (APP)-converting, enzyme, (, BACE, ), messenger, RNA, (, mRNA, ), translation]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-328]\n",
      "0 [we, were, able, to, demonstrate, a, role, for, two, miRNAs, (, miR-298, and, miR-328, ), in, the, regulation, of, beta-amyloid, (, Abeta, ), precursor, protein, (APP)-converting, enzyme, (, BACE, ), messenger, RNA, (, mRNA, ), translation] 36 104 we were able to demonstrate a role for two miRNAs (miR-298 and miR-328) in the regulation of beta-amyloid (Abeta) precursor protein (APP)-converting enzyme (BACE) messenger RNA (mRNA) translation, thereby providing key insights into the molecular basis underlying BACE deregulation in AD.\n",
      "1 [Using, transiently, transfected, murine, neuronal, N2a, cells, in, culture, ,, in, parallel, to, a, mouse, model, of, AD] 18 0 Using transiently transfected murine neuronal N2a cells in culture, in parallel to a mouse model of AD, we were able to demonstrate a role for two miRNAs (miR-298 and miR-328) in the regulation of beta-amyloid (Abeta) precursor protein (APP)-converting enzyme (BACE) messenger RNA (mRNA) translation, thereby providing key insights into the molecular basis underlying BACE deregulation in AD.\n",
      "2 [thereby, providing, key, insights, into, the, molecular, basis, underlying, BACE, deregulation, in, AD] 13 301 thereby providing key insights into the molecular basis underlying BACE deregulation in AD.\n",
      "DeepCheck Conj Subtree\n",
      "[in, culture]\n",
      "DeepCheck Conj Subtree\n",
      "[in, parallel, to, a, mouse, model, of, AD]\n",
      "DeepCheck Conj Subtree\n",
      "[to, a, mouse, model, of, AD]\n",
      "DeepCheck Conj Subtree\n",
      "[of, AD]\n",
      "deepcheck connection rule [to, demonstrate, a, role, for, two, miRNAs, (, miR-298, and, miR-328, ), in, the, regulation, of, beta-amyloid, (, Abeta, ), precursor, protein, (APP)-converting, enzyme, (, BACE, ), messenger, RNA, (, mRNA, ), translation]\n",
      "DeepCheck Conj Subtree\n",
      "[for, two, miRNAs, (, miR-298, and, miR-328, )]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-328]\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, regulation, of, beta-amyloid, (, Abeta, ), precursor, protein, (APP)-converting, enzyme, (, BACE, ), messenger, RNA, (, mRNA, ), translation]\n",
      "DeepCheck Conj Subtree\n",
      "[of, beta-amyloid, (, Abeta, ), precursor, protein, (APP)-converting, enzyme, (, BACE, ), messenger, RNA, (, mRNA, ), translation]\n",
      "DeepCheck Conj Subtree\n",
      "[into, the, molecular, basis, underlying, BACE, deregulation, in, AD]\n",
      "DeepCheck Conj Subtree\n",
      "[in, AD]\n",
      "0 [we, were, able, to, demonstrate, a, role, for, two, miRNAs, (, miR-298, and, miR-328, ), in, the, regulation, of, beta-amyloid, (, Abeta, ), precursor, protein, (APP)-converting, enzyme, (, BACE, ), messenger, RNA, (, mRNA, ), translation] 36 104 we were able to demonstrate a role for two miRNAs (miR-298 and miR-328) in the regulation of beta-amyloid (Abeta) precursor protein (APP)-converting enzyme (BACE) messenger RNA (mRNA) translation, thereby providing key insights into the molecular basis underlying BACE deregulation in AD.\n",
      "1 [Using, transiently, transfected, murine, neuronal, N2a, cells, in, culture, ,, in, parallel, to, a, mouse, model, of, AD] 18 0 Using transiently transfected murine neuronal N2a cells in culture, in parallel to a mouse model of AD, we were able to demonstrate a role for two miRNAs (miR-298 and miR-328) in the regulation of beta-amyloid (Abeta) precursor protein (APP)-converting enzyme (BACE) messenger RNA (mRNA) translation, thereby providing key insights into the molecular basis underlying BACE deregulation in AD.\n",
      "2 [thereby, providing, key, insights, into, the, molecular, basis, underlying, BACE, deregulation, in, AD] 13 301 thereby providing key insights into the molecular basis underlying BACE deregulation in AD.\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text we were able to demonstrate a role for two miRNAs ( miR-298 and miR-328 ) in the regulation of beta-amyloid ( Abeta ) precursor protein (APP)-converting enzyme ( BACE ) messenger RNA ( mRNA ) translation\n",
      "(False, {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence Using transiently transfected murine neuronal N2a cells in culture, in parallel to a mouse model of AD, we were able to demonstrate a role for two miRNAs (miR-298 and miR-328) in the regulation of beta-amyloid (Abeta) precursor protein (APP)-converting enzyme (BACE) messenger RNA (mRNA) translation, thereby providing key insights into the molecular basis underlying BACE deregulation in AD.\n",
      "full sentence Using transiently transfected murine neuronal N2a cells in culture, in parallel to a mouse model of AD, we were able to demonstrate a role for two miRNAs (<e1>miR-298</e1> and miR-328) in the regulation of beta-amyloid (Abeta) precursor protein (<e2>APP</e2>)-converting enzyme (BACE) messenger RNA (mRNA) translation, thereby providing key insights into the molecular basis underlying BACE deregulation in AD.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'Using transiently transfected murine neuronal N2a cells in culture, in parallel to a mouse model of AD, we were able to demonstrate a role for two miRNAs (<e1>miR-298</e1> and miR-328) in the regulation of beta-amyloid (Abeta) precursor protein (<e2>APP</e2>)-converting enzyme (BACE) messenger RNA (mRNA) translation, thereby providing key insights into the molecular basis underlying BACE deregulation in AD.', 'full_sentence': 'Using transiently transfected murine neuronal N2a cells in culture, in parallel to a mouse model of AD, we were able to demonstrate a role for two miRNAs (<e1>miR-298</e1> and miR-328) in the regulation of beta-amyloid (Abeta) precursor protein (<e2>APP</e2>)-converting enzyme (BACE) messenger RNA (mRNA) translation, thereby providing key insights into the molecular basis underlying BACE deregulation in AD.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (155, 162), 'entity_text': 'miR-298'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (237, 240), 'entity_text': 'APP'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d142.s4.p6\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miR-328\n",
      "add special case APP\n",
      "miR-328 32 (APP)-converting 44\n",
      "(u\"Using transiently transfected murine neuronal N2a cells in culture, in parallel to a mouse model of AD, we were able to demonstrate a role for two miRNAs (miR-298 and miR-328) in the regulation of beta-amyloid (Abeta) precursor protein (APP)-converting enzyme (BACE) messenger RNA (mRNA) translation, thereby providing key insights into the molecular basis underlying BACE deregulation in AD.\", 32, 44)\n",
      "geneword (APP)-converting mirword miR-328\n",
      "Conjuncts [miR-328, miR-298]\n",
      "neighbour miR-328 )\n",
      "neighbour miR-328 in\n",
      "Conjunction before fc miR-298\n",
      "Conjuncts [miR-298, miR-328]\n",
      "neighbour miR-298 and\n",
      "neighbour miR-298 and\n",
      "Conjunction before fc miR-328\n",
      "Conunct merge check: miR-298 mRNA\n",
      "miR-328) in the regulation of beta-amyloid (Abeta) precursor protein (APP)-converting enzyme (BACE) messenger RNA\n",
      "Conjunctions ( conj )\n",
      "miR-298 [miR-298, miR-328]\n",
      "mRNA [(, )]\n",
      "Conjuncts [Using, culture]\n",
      "neighbour Using transiently\n",
      "neighbour Using transiently\n",
      "Conjunction before fc culture\n",
      "fc add children for culture\n",
      "Conjuncts [Using, parallel]\n",
      "neighbour Using transiently\n",
      "neighbour Using transiently\n",
      "Conjunction before fc parallel\n",
      "fc add children for parallel\n",
      "fc add children for parallel\n",
      "Conjuncts [parallel, model]\n",
      "neighbour parallel to\n",
      "neighbour parallel to\n",
      "Conjunction before fc model\n",
      "fc add children for model\n",
      "fc add children for model\n",
      "fc add children for model\n",
      "Conjuncts [model, AD]\n",
      "neighbour model of\n",
      "neighbour model of\n",
      "Conjunction before fc AD\n",
      "fc add children for AD\n",
      "Conjuncts [role, miRNAs]\n",
      "neighbour role for\n",
      "neighbour role for\n",
      "Conjunction before fc miRNAs\n",
      "fc add children for miRNAs\n",
      "fc add children for miRNAs\n",
      "fc add children for miRNAs\n",
      "Conjuncts [role, regulation]\n",
      "neighbour role for\n",
      "neighbour role for\n",
      "Conjunction before fc regulation\n",
      "fc add children for regulation\n",
      "fc add children for regulation\n",
      "Conjuncts [regulation, translation]\n",
      "neighbour regulation of\n",
      "neighbour regulation of\n",
      "Conjunction before fc translation\n",
      "fc add children for translation\n",
      "fc add children for translation\n",
      "fc add children for translation\n",
      "fc add children for translation\n",
      "fc add children for translation\n",
      "fc add children for translation\n",
      "fc add children for translation\n",
      "fc add children for translation\n",
      "fc add children for translation\n",
      "fc add children for translation\n",
      "Conjuncts [providing, basis]\n",
      "neighbour providing key\n",
      "neighbour providing key\n",
      "Conjunction before fc basis\n",
      "fc add children for basis\n",
      "fc add children for basis\n",
      "fc add children for basis\n",
      "Conjuncts [deregulation, AD]\n",
      "neighbour deregulation in\n",
      "neighbour deregulation in\n",
      "Conjunction before fc AD\n",
      "fc add children for AD\n",
      "Conunct merge check: culture miRNAs\n",
      "AD, we were able to demonstrate a\n",
      "Conunct merge check: miRNAs mRNA\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: mRNA basis\n",
      ") translation, thereby\n",
      "Conjunctions ( nmod )\n",
      "culture [Using, in, culture, in, parallel, to, mouse, model, of, AD]\n",
      "miRNAs [role, for, two, miRNAs, miR-298, in, regulation, of, beta-amyloid, Abeta, precursor, protein, (APP)-converting, enzyme, BACE, messenger, RNA, mRNA, translation]\n",
      "mRNA [(, )]\n",
      "basis [providing, into, molecular, basis, underlying, BACE, deregulation, in, AD]\n",
      "SDP sent: Using transiently transfected murine neuronal N2a cells in culture, in parallel to a mouse model of AD, we were able to demonstrate a role for two miRNAs (miR-298 and miR-328) in the regulation of beta-amyloid (Abeta) precursor protein (APP)-converting enzyme (BACE) messenger RNA (mRNA) translation, thereby providing key insights into the molecular basis underlying BACE deregulation in AD.\n",
      "SDP     : [(miR-328, 'NOUN', 'conj', False), (miR-298, 'NOUN', 'dep', True), (miRNAs, 'NOUN', 'nmod', False), (role, 'NOUN', 'dobj', False), (regulation, 'NOUN', 'nmod', False), (translation, 'NOUN', 'nmod', False), ((APP)-converting, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule [to, demonstrate, a, role, for, two, miRNAs, (, miR-298, and, miR-328, ), in, the, regulation, of, beta-amyloid, (, Abeta, ), precursor, protein, (APP)-converting, enzyme, (, BACE, ), messenger, RNA, (, mRNA, ), translation]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-328]\n",
      "0 [we, were, able, to, demonstrate, a, role, for, two, miRNAs, (, miR-298, and, miR-328, ), in, the, regulation, of, beta-amyloid, (, Abeta, ), precursor, protein, (APP)-converting, enzyme, (, BACE, ), messenger, RNA, (, mRNA, ), translation] 36 104 we were able to demonstrate a role for two miRNAs (miR-298 and miR-328) in the regulation of beta-amyloid (Abeta) precursor protein (APP)-converting enzyme (BACE) messenger RNA (mRNA) translation, thereby providing key insights into the molecular basis underlying BACE deregulation in AD.\n",
      "1 [Using, transiently, transfected, murine, neuronal, N2a, cells, in, culture, ,, in, parallel, to, a, mouse, model, of, AD] 18 0 Using transiently transfected murine neuronal N2a cells in culture, in parallel to a mouse model of AD, we were able to demonstrate a role for two miRNAs (miR-298 and miR-328) in the regulation of beta-amyloid (Abeta) precursor protein (APP)-converting enzyme (BACE) messenger RNA (mRNA) translation, thereby providing key insights into the molecular basis underlying BACE deregulation in AD.\n",
      "2 [thereby, providing, key, insights, into, the, molecular, basis, underlying, BACE, deregulation, in, AD] 13 301 thereby providing key insights into the molecular basis underlying BACE deregulation in AD.\n",
      "DeepCheck Conj Subtree\n",
      "[in, culture]\n",
      "DeepCheck Conj Subtree\n",
      "[in, parallel, to, a, mouse, model, of, AD]\n",
      "DeepCheck Conj Subtree\n",
      "[to, a, mouse, model, of, AD]\n",
      "DeepCheck Conj Subtree\n",
      "[of, AD]\n",
      "deepcheck connection rule [to, demonstrate, a, role, for, two, miRNAs, (, miR-298, and, miR-328, ), in, the, regulation, of, beta-amyloid, (, Abeta, ), precursor, protein, (APP)-converting, enzyme, (, BACE, ), messenger, RNA, (, mRNA, ), translation]\n",
      "DeepCheck Conj Subtree\n",
      "[for, two, miRNAs, (, miR-298, and, miR-328, )]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-328]\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, regulation, of, beta-amyloid, (, Abeta, ), precursor, protein, (APP)-converting, enzyme, (, BACE, ), messenger, RNA, (, mRNA, ), translation]\n",
      "DeepCheck Conj Subtree\n",
      "[of, beta-amyloid, (, Abeta, ), precursor, protein, (APP)-converting, enzyme, (, BACE, ), messenger, RNA, (, mRNA, ), translation]\n",
      "DeepCheck Conj Subtree\n",
      "[into, the, molecular, basis, underlying, BACE, deregulation, in, AD]\n",
      "DeepCheck Conj Subtree\n",
      "[in, AD]\n",
      "0 [we, were, able, to, demonstrate, a, role, for, two, miRNAs, (, miR-298, and, miR-328, ), in, the, regulation, of, beta-amyloid, (, Abeta, ), precursor, protein, (APP)-converting, enzyme, (, BACE, ), messenger, RNA, (, mRNA, ), translation] 36 104 we were able to demonstrate a role for two miRNAs (miR-298 and miR-328) in the regulation of beta-amyloid (Abeta) precursor protein (APP)-converting enzyme (BACE) messenger RNA (mRNA) translation, thereby providing key insights into the molecular basis underlying BACE deregulation in AD.\n",
      "1 [Using, transiently, transfected, murine, neuronal, N2a, cells, in, culture, ,, in, parallel, to, a, mouse, model, of, AD] 18 0 Using transiently transfected murine neuronal N2a cells in culture, in parallel to a mouse model of AD, we were able to demonstrate a role for two miRNAs (miR-298 and miR-328) in the regulation of beta-amyloid (Abeta) precursor protein (APP)-converting enzyme (BACE) messenger RNA (mRNA) translation, thereby providing key insights into the molecular basis underlying BACE deregulation in AD.\n",
      "2 [thereby, providing, key, insights, into, the, molecular, basis, underlying, BACE, deregulation, in, AD] 13 301 thereby providing key insights into the molecular basis underlying BACE deregulation in AD.\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text we were able to demonstrate a role for two miRNAs ( miR-298 and miR-328 ) in the regulation of beta-amyloid ( Abeta ) precursor protein (APP)-converting enzyme ( BACE ) messenger RNA ( mRNA ) translation\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence Using transiently transfected murine neuronal N2a cells in culture, in parallel to a mouse model of AD, we were able to demonstrate a role for two miRNAs (miR-298 and miR-328) in the regulation of beta-amyloid (Abeta) precursor protein (APP)-converting enzyme (BACE) messenger RNA (mRNA) translation, thereby providing key insights into the molecular basis underlying BACE deregulation in AD.\n",
      "full sentence Using transiently transfected murine neuronal N2a cells in culture, in parallel to a mouse model of AD, we were able to demonstrate a role for two miRNAs (miR-298 and <e1>miR-328</e1>) in the regulation of beta-amyloid (Abeta) precursor protein (<e2>APP</e2>)-converting enzyme (BACE) messenger RNA (mRNA) translation, thereby providing key insights into the molecular basis underlying BACE deregulation in AD.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'Using transiently transfected murine neuronal N2a cells in culture, in parallel to a mouse model of AD, we were able to demonstrate a role for two miRNAs (miR-298 and <e1>miR-328</e1>) in the regulation of beta-amyloid (Abeta) precursor protein (<e2>APP</e2>)-converting enzyme (BACE) messenger RNA (mRNA) translation, thereby providing key insights into the molecular basis underlying BACE deregulation in AD.', 'full_sentence': 'Using transiently transfected murine neuronal N2a cells in culture, in parallel to a mouse model of AD, we were able to demonstrate a role for two miRNAs (miR-298 and <e1>miR-328</e1>) in the regulation of beta-amyloid (Abeta) precursor protein (<e2>APP</e2>)-converting enzyme (BACE) messenger RNA (mRNA) translation, thereby providing key insights into the molecular basis underlying BACE deregulation in AD.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (167, 174), 'entity_text': 'miR-328'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (237, 240), 'entity_text': 'APP'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d144.s2.p0\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miRNA-146a\n",
      "add special case interleukin-1beta\n",
      "miRNA-146a 2 interleukin-1beta 15\n",
      "(u\"Up-regulation of miRNA-146a coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.\", 2, 15)\n",
      "geneword interleukin-1beta mirword miRNA-146a\n",
      "Conjuncts [interleukin-1beta, Abeta42, cells, brain]\n",
      "neighbour interleukin-1beta ,\n",
      "neighbour interleukin-1beta Abeta42\n",
      "neighbour Abeta42 ,\n",
      "neighbour Abeta42 and/or\n",
      "neighbour cells in\n",
      "neighbour cells in\n",
      "Conjunction before fc brain\n",
      "fc add children for brain\n",
      "fc add children for brain\n",
      "Conjuncts [brain, Abeta42, cells, interleukin-1beta]\n",
      "neighbour brain and\n",
      "neighbour brain and\n",
      "neighbour Abeta42 ,\n",
      "neighbour Abeta42 and/or\n",
      "neighbour cells in\n",
      "neighbour cells in\n",
      "Conjunction before fc interleukin-1beta\n",
      "fc add children for interleukin-1beta\n",
      "fc add children for interleukin-1beta\n",
      "Conjuncts [brain, interleukin-1beta, cells, Abeta42]\n",
      "neighbour brain and\n",
      "neighbour brain and\n",
      "neighbour interleukin-1beta ,\n",
      "neighbour interleukin-1beta Abeta42\n",
      "neighbour cells in\n",
      "neighbour cells in\n",
      "Conjunction before fc Abeta42\n",
      "Conjuncts [brain, interleukin-1beta, Abeta42, cells]\n",
      "neighbour brain and\n",
      "neighbour brain and\n",
      "neighbour interleukin-1beta ,\n",
      "neighbour interleukin-1beta Abeta42\n",
      "neighbour Abeta42 ,\n",
      "neighbour Abeta42 and/or\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conjunctions ( conj )\n",
      "brain [in, AD, brain, in, interleukin-1beta, Abeta42, stressed, human, neural, HN, cells, in, primary, culture]\n",
      "Conjuncts [Up-regulation, miRNA-146a]\n",
      "neighbour Up-regulation of\n",
      "neighbour Up-regulation of\n",
      "Conjunction before fc miRNA-146a\n",
      "fc add children for miRNA-146a\n",
      "fc add children for miRNA-146a\n",
      "Conjuncts [coupled, down-regulation]\n",
      "neighbour coupled to\n",
      "neighbour coupled to\n",
      "Conjunction before fc down-regulation\n",
      "fc add children for down-regulation\n",
      "fc add children for down-regulation\n",
      "Conjuncts [down-regulation, CFH]\n",
      "neighbour down-regulation of\n",
      "neighbour down-regulation of\n",
      "Conjunction before fc CFH\n",
      "fc add children for CFH\n",
      "Conjuncts [observed, brain]\n",
      "neighbour observed in\n",
      "neighbour observed in\n",
      "Conjunction before fc brain\n",
      "fc add children for brain\n",
      "fc add children for brain\n",
      "Conjuncts [interleukin-1beta, culture]\n",
      "neighbour interleukin-1beta ,\n",
      "neighbour interleukin-1beta Abeta42\n",
      "Conjunction before fc culture\n",
      "fc add children for culture\n",
      "fc add children for culture\n",
      "Conunct merge check: miRNA-146a brain\n",
      "CFH was\n",
      "Conunct merge check: brain culture\n",
      "brain and in\n",
      "Double Conjunction Found\n",
      "Conjunctions ( nmod )\n",
      "miRNA-146a [Up-regulation, of, miRNA-146a, coupled, to, down-regulation, of, CFH]\n",
      "brain [observed, in, AD, brain]\n",
      "culture [interleukin-1beta, in, primary, culture]\n",
      "(culture, 'conj') [observed, in, AD, brain, interleukin-1beta, in, primary, culture]\n",
      "SDP sent: Up-regulation of miRNA-146a coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.\n",
      "SDP     : [(miRNA-146a, 'NOUN', 'nmod', False), (Up-regulation, 'NOUN', 'nsubjpass', False), (observed, 'VERB', 'ROOT', False), (brain, 'NOUN', 'nmod', True), (interleukin-1beta, 'NOUN', 'conj', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[in, interleukin-1beta, ,, Abeta42, ,, and/or, oxidatively, stressed, human, neural, (, HN, ), cells, in, primary, culture]\n",
      "DeepCheck Conj Subtree\n",
      "[Abeta42]\n",
      "DeepCheck Conj Subtree\n",
      "[oxidatively, stressed, human, neural, (, HN, ), cells]\n",
      "0 [Up-regulation, of, miRNA-146a, coupled, to, down-regulation, of, CFH, was, observed, in, AD, brain, and, in, interleukin-1beta, ,, Abeta42, ,, and/or, oxidatively, stressed, human, neural, (, HN, ), cells, in, primary, culture] 31 0 Up-regulation of miRNA-146a coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.\n",
      "DeepCheck Conj Subtree\n",
      "[of, miRNA-146a, coupled, to, down-regulation, of, CFH]\n",
      "DeepCheck Conj Subtree\n",
      "[to, down-regulation, of, CFH]\n",
      "DeepCheck Conj Subtree\n",
      "[of, CFH]\n",
      "DeepCheck Conj Subtree\n",
      "[in, AD, brain, and, in, interleukin-1beta, ,, Abeta42, ,, and/or, oxidatively, stressed, human, neural, (, HN, ), cells, in, primary, culture]\n",
      "DeepCheck Conj Subtree\n",
      "[in, interleukin-1beta, ,, Abeta42, ,, and/or, oxidatively, stressed, human, neural, (, HN, ), cells, in, primary, culture]\n",
      "DeepCheck Conj Subtree\n",
      "[Abeta42]\n",
      "DeepCheck Conj Subtree\n",
      "[oxidatively, stressed, human, neural, (, HN, ), cells]\n",
      "DeepCheck Conj Subtree\n",
      "[in, primary, culture]\n",
      "0 [Up-regulation, of, miRNA-146a, coupled, to, down-regulation, of, CFH, was, observed, in, AD, brain, and, in, interleukin-1beta, ,, Abeta42, ,, and/or, oxidatively, stressed, human, neural, (, HN, ), cells, in, primary, culture] 31 0 Up-regulation of miRNA-146a coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.\n",
      "Recognized overlapping entities: {Interval(94, 111, ('GENE_OR_GENE_PRODUCT', 'interleukin-1beta'))}\n",
      "passive? True\n",
      "Counter({'POS': 1})\n",
      "Compartment Text Up-regulation of miRNA-146a coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta , Abeta42 , and/or oxidatively stressed human neural ( HN ) cells in primary culture\n",
      "[Up-regulation, of, miRNA-146a, coupled, to, down-regulation, of, CFH, was, observed, in, AD, brain, and, in, interleukin-1beta, ,, Abeta42, ,, and/or, oxidatively, stressed, human, neural, (, HN, ), cells, in, primary, culture]\n",
      "TextBetween: Up-regulation of miRNA-146a coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta , Abeta42 , and/or oxidatively stressed human neural ( HN ) cells in primary culture\n",
      "infer direction from token set [Up-regulation, of, miRNA-146a, coupled, to, down-regulation, of, CFH, was, observed, in, AD, brain, and, in, interleukin-1beta, ,, Abeta42, ,, and/or, oxidatively, stressed, human, neural, (, HN, ), cells, in, primary, culture]\n",
      "stemTypes2Count: Counter({'POS': 1, 'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e21e0>, {'mir': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'POS': 1, 'NEU': 1, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e28c8>, {'mir': Counter({'POS': 1, 'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(Up-regulation, 'POS'), (coupled, 'NEU'), (down-regulation, 'NEG')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Up-regulation of miRNA-146a coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.\n",
      "full sentence Up-regulation of <e1>miRNA-146a</e1> coupled to down-regulation of CFH was observed in AD brain and in <e2>interleukin-1beta</e2>, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Up-regulation of <e1>miRNA-146a</e1> coupled to down-regulation of CFH was observed in AD brain and in <e2>interleukin-1beta</e2>, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.', 'full_sentence': 'Up-regulation of <e1>miRNA-146a</e1> coupled to down-regulation of CFH was observed in AD brain and in <e2>interleukin-1beta</e2>, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (17, 27), 'entity_text': 'miRNA-146a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (94, 111), 'entity_text': 'interleukin-1beta'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d144.s2.p1\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miRNA-146a\n",
      "add special case Abeta42\n",
      "miRNA-146a 2 Abeta42 17\n",
      "(u\"Up-regulation of miRNA-146a coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.\", 2, 17)\n",
      "geneword Abeta42 mirword miRNA-146a\n",
      "Conjuncts [interleukin-1beta, Abeta42, cells, brain]\n",
      "neighbour interleukin-1beta ,\n",
      "neighbour interleukin-1beta Abeta42\n",
      "neighbour Abeta42 ,\n",
      "neighbour Abeta42 and/or\n",
      "neighbour cells in\n",
      "neighbour cells in\n",
      "Conjunction before fc brain\n",
      "fc add children for brain\n",
      "fc add children for brain\n",
      "Conjuncts [brain, Abeta42, cells, interleukin-1beta]\n",
      "neighbour brain and\n",
      "neighbour brain and\n",
      "neighbour Abeta42 ,\n",
      "neighbour Abeta42 and/or\n",
      "neighbour cells in\n",
      "neighbour cells in\n",
      "Conjunction before fc interleukin-1beta\n",
      "fc add children for interleukin-1beta\n",
      "fc add children for interleukin-1beta\n",
      "Conjuncts [brain, interleukin-1beta, cells, Abeta42]\n",
      "neighbour brain and\n",
      "neighbour brain and\n",
      "neighbour interleukin-1beta ,\n",
      "neighbour interleukin-1beta Abeta42\n",
      "neighbour cells in\n",
      "neighbour cells in\n",
      "Conjunction before fc Abeta42\n",
      "Conjuncts [brain, interleukin-1beta, Abeta42, cells]\n",
      "neighbour brain and\n",
      "neighbour brain and\n",
      "neighbour interleukin-1beta ,\n",
      "neighbour interleukin-1beta Abeta42\n",
      "neighbour Abeta42 ,\n",
      "neighbour Abeta42 and/or\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conjunctions ( conj )\n",
      "brain [in, AD, brain, in, interleukin-1beta, Abeta42, stressed, human, neural, HN, cells, in, primary, culture]\n",
      "Conjuncts [Up-regulation, miRNA-146a]\n",
      "neighbour Up-regulation of\n",
      "neighbour Up-regulation of\n",
      "Conjunction before fc miRNA-146a\n",
      "fc add children for miRNA-146a\n",
      "fc add children for miRNA-146a\n",
      "Conjuncts [coupled, down-regulation]\n",
      "neighbour coupled to\n",
      "neighbour coupled to\n",
      "Conjunction before fc down-regulation\n",
      "fc add children for down-regulation\n",
      "fc add children for down-regulation\n",
      "Conjuncts [down-regulation, CFH]\n",
      "neighbour down-regulation of\n",
      "neighbour down-regulation of\n",
      "Conjunction before fc CFH\n",
      "fc add children for CFH\n",
      "Conjuncts [observed, brain]\n",
      "neighbour observed in\n",
      "neighbour observed in\n",
      "Conjunction before fc brain\n",
      "fc add children for brain\n",
      "fc add children for brain\n",
      "Conjuncts [interleukin-1beta, culture]\n",
      "neighbour interleukin-1beta ,\n",
      "neighbour interleukin-1beta Abeta42\n",
      "Conjunction before fc culture\n",
      "fc add children for culture\n",
      "fc add children for culture\n",
      "Conunct merge check: miRNA-146a brain\n",
      "CFH was\n",
      "Conunct merge check: brain culture\n",
      "brain and in\n",
      "Double Conjunction Found\n",
      "Conjunctions ( nmod )\n",
      "miRNA-146a [Up-regulation, of, miRNA-146a, coupled, to, down-regulation, of, CFH]\n",
      "brain [observed, in, AD, brain]\n",
      "culture [interleukin-1beta, in, primary, culture]\n",
      "(culture, 'conj') [observed, in, AD, brain, interleukin-1beta, in, primary, culture]\n",
      "SDP sent: Up-regulation of miRNA-146a coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.\n",
      "SDP     : [(miRNA-146a, 'NOUN', 'nmod', False), (Up-regulation, 'NOUN', 'nsubjpass', False), (observed, 'VERB', 'ROOT', False), (brain, 'NOUN', 'nmod', True), (interleukin-1beta, 'NOUN', 'conj', True), (Abeta42, 'NOUN', 'conj', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[in, interleukin-1beta, ,, Abeta42, ,, and/or, oxidatively, stressed, human, neural, (, HN, ), cells, in, primary, culture]\n",
      "DeepCheck Conj Subtree\n",
      "[Abeta42]\n",
      "DeepCheck Conj Subtree\n",
      "[oxidatively, stressed, human, neural, (, HN, ), cells]\n",
      "0 [Up-regulation, of, miRNA-146a, coupled, to, down-regulation, of, CFH, was, observed, in, AD, brain, and, in, interleukin-1beta, ,, Abeta42, ,, and/or, oxidatively, stressed, human, neural, (, HN, ), cells, in, primary, culture] 31 0 Up-regulation of miRNA-146a coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.\n",
      "DeepCheck Conj Subtree\n",
      "[of, miRNA-146a, coupled, to, down-regulation, of, CFH]\n",
      "DeepCheck Conj Subtree\n",
      "[to, down-regulation, of, CFH]\n",
      "DeepCheck Conj Subtree\n",
      "[of, CFH]\n",
      "DeepCheck Conj Subtree\n",
      "[in, AD, brain, and, in, interleukin-1beta, ,, Abeta42, ,, and/or, oxidatively, stressed, human, neural, (, HN, ), cells, in, primary, culture]\n",
      "DeepCheck Conj Subtree\n",
      "[in, interleukin-1beta, ,, Abeta42, ,, and/or, oxidatively, stressed, human, neural, (, HN, ), cells, in, primary, culture]\n",
      "DeepCheck Conj Subtree\n",
      "[Abeta42]\n",
      "DeepCheck Conj Subtree\n",
      "[oxidatively, stressed, human, neural, (, HN, ), cells]\n",
      "DeepCheck Conj Subtree\n",
      "[in, primary, culture]\n",
      "0 [Up-regulation, of, miRNA-146a, coupled, to, down-regulation, of, CFH, was, observed, in, AD, brain, and, in, interleukin-1beta, ,, Abeta42, ,, and/or, oxidatively, stressed, human, neural, (, HN, ), cells, in, primary, culture] 31 0 Up-regulation of miRNA-146a coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.\n",
      "Recognized overlapping entities: set()\n",
      "passive? True\n",
      "Counter({'POS': 1})\n",
      "Compartment Text Up-regulation of miRNA-146a coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta , Abeta42 , and/or oxidatively stressed human neural ( HN ) cells in primary culture\n",
      "[Up-regulation, of, miRNA-146a, coupled, to, down-regulation, of, CFH, was, observed, in, AD, brain, and, in, interleukin-1beta, ,, Abeta42, ,, and/or, oxidatively, stressed, human, neural, (, HN, ), cells, in, primary, culture]\n",
      "TextBetween: Up-regulation of miRNA-146a coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta , Abeta42 , and/or oxidatively stressed human neural ( HN ) cells in primary culture\n",
      "infer direction from token set [Up-regulation, of, miRNA-146a, coupled, to, down-regulation, of, CFH, was, observed, in, AD, brain, and, in, interleukin-1beta, ,, Abeta42, ,, and/or, oxidatively, stressed, human, neural, (, HN, ), cells, in, primary, culture]\n",
      "stemTypes2Count: Counter({'POS': 1, 'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e2488>, {'mir': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'POS': 1, 'NEU': 1, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e2268>, {'mir': Counter({'POS': 1, 'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(Up-regulation, 'POS'), (coupled, 'NEU'), (down-regulation, 'NEG')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Up-regulation of miRNA-146a coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.\n",
      "full sentence Up-regulation of <e1>miRNA-146a</e1> coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta, <e2>Abeta42</e2>, and/or oxidatively stressed human neural (HN) cells in primary culture.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Up-regulation of <e1>miRNA-146a</e1> coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta, <e2>Abeta42</e2>, and/or oxidatively stressed human neural (HN) cells in primary culture.', 'full_sentence': 'Up-regulation of <e1>miRNA-146a</e1> coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta, <e2>Abeta42</e2>, and/or oxidatively stressed human neural (HN) cells in primary culture.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (17, 27), 'entity_text': 'miRNA-146a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (113, 120), 'entity_text': 'Abeta42'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d144.s2.p2\n",
      "IS:  ('GENE_MIR', 'UP')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': True, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Up-regulation of <e1>miRNA-146a</e1> coupled to down-regulation of <e2>CFH</e2> was observed in AD brain and in interleukin-1beta, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.', 'full_sentence': 'Up-regulation of <e1>miRNA-146a</e1> coupled to down-regulation of <e2>CFH</e2> was observed in AD brain and in interleukin-1beta, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (17, 27), 'entity_text': 'miRNA-146a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (58, 61), 'entity_text': 'CFH'}}\n",
      "add special case miRNA-146a\n",
      "add special case CFH\n",
      "miRNA-146a 2 CFH 7\n",
      "(u\"Up-regulation of miRNA-146a coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.\", 2, 7)\n",
      "geneword CFH mirword miRNA-146a\n",
      "Conjuncts [interleukin-1beta, Abeta42, cells, brain]\n",
      "neighbour interleukin-1beta ,\n",
      "neighbour interleukin-1beta Abeta42\n",
      "neighbour Abeta42 ,\n",
      "neighbour Abeta42 and/or\n",
      "neighbour cells in\n",
      "neighbour cells in\n",
      "Conjunction before fc brain\n",
      "fc add children for brain\n",
      "fc add children for brain\n",
      "Conjuncts [brain, Abeta42, cells, interleukin-1beta]\n",
      "neighbour brain and\n",
      "neighbour brain and\n",
      "neighbour Abeta42 ,\n",
      "neighbour Abeta42 and/or\n",
      "neighbour cells in\n",
      "neighbour cells in\n",
      "Conjunction before fc interleukin-1beta\n",
      "fc add children for interleukin-1beta\n",
      "fc add children for interleukin-1beta\n",
      "Conjuncts [brain, interleukin-1beta, cells, Abeta42]\n",
      "neighbour brain and\n",
      "neighbour brain and\n",
      "neighbour interleukin-1beta ,\n",
      "neighbour interleukin-1beta Abeta42\n",
      "neighbour cells in\n",
      "neighbour cells in\n",
      "Conjunction before fc Abeta42\n",
      "Conjuncts [brain, interleukin-1beta, Abeta42, cells]\n",
      "neighbour brain and\n",
      "neighbour brain and\n",
      "neighbour interleukin-1beta ,\n",
      "neighbour interleukin-1beta Abeta42\n",
      "neighbour Abeta42 ,\n",
      "neighbour Abeta42 and/or\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conjunctions ( conj )\n",
      "brain [in, AD, brain, in, interleukin-1beta, Abeta42, stressed, human, neural, HN, cells, in, primary, culture]\n",
      "Conjuncts [Up-regulation, miRNA-146a]\n",
      "neighbour Up-regulation of\n",
      "neighbour Up-regulation of\n",
      "Conjunction before fc miRNA-146a\n",
      "fc add children for miRNA-146a\n",
      "fc add children for miRNA-146a\n",
      "Conjuncts [coupled, down-regulation]\n",
      "neighbour coupled to\n",
      "neighbour coupled to\n",
      "Conjunction before fc down-regulation\n",
      "fc add children for down-regulation\n",
      "fc add children for down-regulation\n",
      "Conjuncts [down-regulation, CFH]\n",
      "neighbour down-regulation of\n",
      "neighbour down-regulation of\n",
      "Conjunction before fc CFH\n",
      "fc add children for CFH\n",
      "Conjuncts [observed, brain]\n",
      "neighbour observed in\n",
      "neighbour observed in\n",
      "Conjunction before fc brain\n",
      "fc add children for brain\n",
      "fc add children for brain\n",
      "Conjuncts [interleukin-1beta, culture]\n",
      "neighbour interleukin-1beta ,\n",
      "neighbour interleukin-1beta Abeta42\n",
      "Conjunction before fc culture\n",
      "fc add children for culture\n",
      "fc add children for culture\n",
      "Conunct merge check: miRNA-146a brain\n",
      "CFH was\n",
      "Conunct merge check: brain culture\n",
      "brain and in\n",
      "Double Conjunction Found\n",
      "Conjunctions ( nmod )\n",
      "miRNA-146a [Up-regulation, of, miRNA-146a, coupled, to, down-regulation, of, CFH]\n",
      "brain [observed, in, AD, brain]\n",
      "culture [interleukin-1beta, in, primary, culture]\n",
      "(culture, 'conj') [observed, in, AD, brain, interleukin-1beta, in, primary, culture]\n",
      "achildren {miRNA-146a}\n",
      "achildren False\n",
      "SDP sent: Up-regulation of miRNA-146a coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.\n",
      "SDP     : [(miRNA-146a, 'NOUN', 'nmod', False), (observed, 'VERB', 'ROOT', False), (Up-regulation, 'NOUN', 'nsubjpass', False), (miRNA-146a, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[in, interleukin-1beta, ,, Abeta42, ,, and/or, oxidatively, stressed, human, neural, (, HN, ), cells, in, primary, culture]\n",
      "DeepCheck Conj Subtree\n",
      "[Abeta42]\n",
      "DeepCheck Conj Subtree\n",
      "[oxidatively, stressed, human, neural, (, HN, ), cells]\n",
      "0 [Up-regulation, of, miRNA-146a, coupled, to, down-regulation, of, CFH, was, observed, in, AD, brain, and, in, interleukin-1beta, ,, Abeta42, ,, and/or, oxidatively, stressed, human, neural, (, HN, ), cells, in, primary, culture] 31 0 Up-regulation of miRNA-146a coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.\n",
      "DeepCheck Conj Subtree\n",
      "[of, miRNA-146a, coupled, to, down-regulation, of, CFH]\n",
      "DeepCheck Conj Subtree\n",
      "[to, down-regulation, of, CFH]\n",
      "DeepCheck Conj Subtree\n",
      "[of, CFH]\n",
      "DeepCheck Conj Subtree\n",
      "[in, AD, brain, and, in, interleukin-1beta, ,, Abeta42, ,, and/or, oxidatively, stressed, human, neural, (, HN, ), cells, in, primary, culture]\n",
      "DeepCheck Conj Subtree\n",
      "[in, interleukin-1beta, ,, Abeta42, ,, and/or, oxidatively, stressed, human, neural, (, HN, ), cells, in, primary, culture]\n",
      "DeepCheck Conj Subtree\n",
      "[Abeta42]\n",
      "DeepCheck Conj Subtree\n",
      "[oxidatively, stressed, human, neural, (, HN, ), cells]\n",
      "DeepCheck Conj Subtree\n",
      "[in, primary, culture]\n",
      "0 [Up-regulation, of, miRNA-146a, coupled, to, down-regulation, of, CFH, was, observed, in, AD, brain, and, in, interleukin-1beta, ,, Abeta42, ,, and/or, oxidatively, stressed, human, neural, (, HN, ), cells, in, primary, culture] 31 0 Up-regulation of miRNA-146a coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.\n",
      "Recognized overlapping entities: {Interval(58, 61, ('GENE_OR_GENE_PRODUCT', 'CFH'))}\n",
      "passive? True\n",
      "Counter({'POS': 1})\n",
      "Compartment Text Up-regulation of miRNA-146a coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta , Abeta42 , and/or oxidatively stressed human neural ( HN ) cells in primary culture\n",
      "[Up-regulation, of, miRNA-146a, coupled, to, down-regulation, of, CFH, was, observed, in, AD, brain, and, in, interleukin-1beta, ,, Abeta42, ,, and/or, oxidatively, stressed, human, neural, (, HN, ), cells, in, primary, culture]\n",
      "TextBetween: Up-regulation of miRNA-146a coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta , Abeta42 , and/or oxidatively stressed human neural ( HN ) cells in primary culture\n",
      "infer direction from token set [Up-regulation, of, miRNA-146a, coupled, to, down-regulation, of, CFH, was, observed, in, AD, brain, and, in, interleukin-1beta, ,, Abeta42, ,, and/or, oxidatively, stressed, human, neural, (, HN, ), cells, in, primary, culture]\n",
      "stemTypes2Count: Counter({'POS': 1, 'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954f28>, {'mir': Counter({'POS': 1}), 'gene': Counter({'NEG': 1})})\n",
      "defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954f28>, {'mir': Counter({'POS': 1}), 'gene': Counter({'NEG': 1})})\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': True, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': True, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Up-regulation of miRNA-146a coupled to down-regulation of CFH was observed in AD brain and in interleukin-1beta, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.\n",
      "full sentence Up-regulation of <e1>miRNA-146a</e1> coupled to down-regulation of <e2>CFH</e2> was observed in AD brain and in interleukin-1beta, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'GENE_MIR', 'passive': True, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Up-regulation of <e1>miRNA-146a</e1> coupled to down-regulation of <e2>CFH</e2> was observed in AD brain and in interleukin-1beta, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.', 'full_sentence': 'Up-regulation of <e1>miRNA-146a</e1> coupled to down-regulation of <e2>CFH</e2> was observed in AD brain and in interleukin-1beta, Abeta42, and/or oxidatively stressed human neural (HN) cells in primary culture.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (17, 27), 'entity_text': 'miRNA-146a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (58, 61), 'entity_text': 'CFH'}}\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Here we provide evidence in AD brains of a specific up-regulation of an NF-kappaB-sensitive miRNA-146a highly complementary to the 3'-untranslated region of complement factor H (CFH), an important repressor of the inflammatory response of the brain. \n",
      "('miRNA-146a', 'Specific_miRNAs', (92, 102)) ('complement factor H ', 'Genes/Proteins', (157, 177))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Here we provide evidence in AD brains of a specific up-regulation of an NF-kappaB-sensitive miRNA-146a highly complementary to the 3'-untranslated region of complement factor H (CFH), an important repressor of the inflammatory response of the brain. \n",
      "('miRNA-146a', 'Specific_miRNAs', (92, 102)) ('CFH', 'Genes/Proteins', (178, 181))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d144.s7.p0\n",
      "IS:  ('MIR_GENE', 'NEU')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'These data indicate that NF-kappaB-sensitive <e1>miRNA-146a</e1>-mediated modulation of <e2>CFH</e2> gene expression may in part regulate an inflammatory response in AD brain and in stressed HN cell models of AD and illustrate the potential for anti-miRNAs as an effective therapeutic strategy against pathogenic inflammatory signaling.', 'full_sentence': 'These data indicate that NF-kappaB-sensitive <e1>miRNA-146a</e1>-mediated modulation of <e2>CFH</e2> gene expression may in part regulate an inflammatory response in AD brain and in stressed HN cell models of AD and illustrate the potential for anti-miRNAs as an effective therapeutic strategy against pathogenic inflammatory signaling.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (45, 55), 'entity_text': 'miRNA-146a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (79, 82), 'entity_text': 'CFH'}}\n",
      "add special case miRNA-146a\n",
      "add special case CFH\n",
      "miRNA-146a-mediated 5 CFH 8\n",
      "(u\"These data indicate that NF-kappaB-sensitive miRNA-146a-mediated modulation of CFH gene expression may in part regulate an inflammatory response in AD brain and in stressed HN cell models of AD and illustrate the potential for anti-miRNAs as an effective therapeutic strategy against pathogenic inflammatory signaling.\", 5, 8)\n",
      "geneword CFH mirword miRNA-146a-mediated\n",
      "Conjuncts [illustrate, regulate]\n",
      "neighbour illustrate the\n",
      "neighbour illustrate the\n",
      "Conjunction before fc regulate\n",
      "fc add children for regulate\n",
      "fc add children for regulate\n",
      "Conjuncts [models, brain]\n",
      "neighbour models of\n",
      "neighbour models of\n",
      "Conjunction before fc brain\n",
      "fc add children for brain\n",
      "fc add children for brain\n",
      "Conjuncts [brain, models]\n",
      "neighbour brain and\n",
      "neighbour brain and\n",
      "Conjunction before fc models\n",
      "fc add children for models\n",
      "fc add children for models\n",
      "fc add children for models\n",
      "fc add children for models\n",
      "fc add children for models\n",
      "Conjuncts [regulate, illustrate]\n",
      "neighbour regulate an\n",
      "neighbour regulate an\n",
      "Conjunction before fc illustrate\n",
      "fc add children for illustrate\n",
      "Conjunctions ( conj )\n",
      "regulate [in, part, regulate, inflammatory, response, in, AD, brain, in, stressed, HN, cell, models, of, AD, illustrate, potential, for, anti-miRNAs, as, effective, therapeutic, strategy, against, pathogenic, inflammatory, signaling]\n",
      "Conjuncts [modulation, expression]\n",
      "neighbour modulation of\n",
      "neighbour modulation of\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conjuncts [regulate, part]\n",
      "neighbour regulate an\n",
      "neighbour regulate an\n",
      "Conjunction before fc part\n",
      "fc add children for part\n",
      "Conjuncts [response, brain]\n",
      "neighbour response in\n",
      "neighbour response in\n",
      "Conjunction before fc brain\n",
      "fc add children for brain\n",
      "fc add children for brain\n",
      "Conjuncts [models, AD]\n",
      "neighbour models of\n",
      "neighbour models of\n",
      "Conjunction before fc AD\n",
      "fc add children for AD\n",
      "Conjuncts [potential, anti-miRNAs]\n",
      "neighbour potential for\n",
      "neighbour potential for\n",
      "Conjunction before fc anti-miRNAs\n",
      "fc add children for anti-miRNAs\n",
      "Conjuncts [potential, strategy]\n",
      "neighbour potential for\n",
      "neighbour potential for\n",
      "Conjunction before fc strategy\n",
      "fc add children for strategy\n",
      "fc add children for strategy\n",
      "fc add children for strategy\n",
      "Conjuncts [strategy, signaling]\n",
      "neighbour strategy against\n",
      "neighbour strategy against\n",
      "Conjunction before fc signaling\n",
      "fc add children for signaling\n",
      "fc add children for signaling\n",
      "fc add children for signaling\n",
      "Conunct merge check: expression part\n",
      "expression may\n",
      "Conunct merge check: part brain\n",
      "regulate an inflammatory\n",
      "Conunct merge check: brain AD\n",
      "brain and in stressed HN cell\n",
      "Double Conjunction Found\n",
      "Conunct merge check: AD anti-miRNAs\n",
      "AD and illustrate the\n",
      "Double Conjunction Found\n",
      "Conjunctions ( nmod )\n",
      "expression [modulation, of, CFH, gene, expression]\n",
      "part [in, part, regulate]\n",
      "brain [response, in, AD, brain]\n",
      "AD [models, of, AD]\n",
      "anti-miRNAs [potential, for, anti-miRNAs, as, effective, therapeutic, strategy, against, pathogenic, inflammatory, signaling]\n",
      "(AD, 'conj') [response, in, AD, brain, models, of, AD]\n",
      "(anti-miRNAs, 'conj') [response, in, AD, brain, models, of, AD, potential, for, anti-miRNAs, as, effective, therapeutic, strategy, against, pathogenic, inflammatory, signaling]\n",
      "SDP sent: These data indicate that NF-kappaB-sensitive miRNA-146a-mediated modulation of CFH gene expression may in part regulate an inflammatory response in AD brain and in stressed HN cell models of AD and illustrate the potential for anti-miRNAs as an effective therapeutic strategy against pathogenic inflammatory signaling.\n",
      "SDP     : [(miRNA-146a-mediated, 'ADJ', 'amod', False), (modulation, 'NOUN', 'nsubj', False), (expression, 'NOUN', 'nmod', False), (CFH, 'NOUN', 'compound', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[in, stressed, HN, cell, models, of, AD]\n",
      "deepcheck conj verb\n",
      "0 [These, data, indicate] 3 0 These data indicate that NF-kappaB-sensitive miRNA-146a-mediated modulation of CFH gene expression may in part regulate an inflammatory response in AD brain and in stressed HN cell models of AD and illustrate the potential for anti-miRNAs as an effective therapeutic strategy against pathogenic inflammatory signaling.\n",
      "1 [NF-kappaB-sensitive, miRNA-146a-mediated, modulation, of, CFH, gene, expression, may, in, part, regulate, an, inflammatory, response, in, AD, brain, and, in, stressed, HN, cell, models, of, AD, and, illustrate, the, potential, for, anti-miRNAs, as, an, effective, therapeutic, strategy, against, pathogenic, inflammatory, signaling] 40 25 NF-kappaB-sensitive miRNA-146a-mediated modulation of CFH gene expression may in part regulate an inflammatory response in AD brain and in stressed HN cell models of AD and illustrate the potential for anti-miRNAs as an effective therapeutic strategy against pathogenic inflammatory signaling.\n",
      "DeepCheck Conj Subtree\n",
      "[of, CFH, gene, expression]\n",
      "DeepCheck Conj Subtree\n",
      "[in, part]\n",
      "DeepCheck Conj Subtree\n",
      "[in, AD, brain, and, in, stressed, HN, cell, models, of, AD]\n",
      "DeepCheck Conj Subtree\n",
      "[in, stressed, HN, cell, models, of, AD]\n",
      "DeepCheck Conj Subtree\n",
      "[of, AD]\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[for, anti-miRNAs]\n",
      "DeepCheck Conj Subtree\n",
      "[as, an, effective, therapeutic, strategy, against, pathogenic, inflammatory, signaling]\n",
      "DeepCheck Conj Subtree\n",
      "[against, pathogenic, inflammatory, signaling]\n",
      "0 [These, data, indicate] 3 0 These data indicate that NF-kappaB-sensitive miRNA-146a-mediated modulation of CFH gene expression may in part regulate an inflammatory response in AD brain and in stressed HN cell models of AD and illustrate the potential for anti-miRNAs as an effective therapeutic strategy against pathogenic inflammatory signaling.\n",
      "1 [NF-kappaB-sensitive, miRNA-146a-mediated, modulation, of, CFH, gene, expression, may, in, part, regulate, an, inflammatory, response, in, AD, brain, and, in, stressed, HN, cell, models, of, AD, and, illustrate, the, potential, for, anti-miRNAs, as, an, effective, therapeutic, strategy, against, pathogenic, inflammatory, signaling] 40 25 NF-kappaB-sensitive miRNA-146a-mediated modulation of CFH gene expression may in part regulate an inflammatory response in AD brain and in stressed HN cell models of AD and illustrate the potential for anti-miRNAs as an effective therapeutic strategy against pathogenic inflammatory signaling.\n",
      "Recognized overlapping entities: {Interval(79, 82, ('GENE_OR_GENE_PRODUCT', 'CFH'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text NF-kappaB-sensitive miRNA-146a-mediated modulation of CFH gene expression may in part regulate an inflammatory response in AD brain and in stressed HN cell models of AD and illustrate the potential for anti-miRNAs as an effective therapeutic strategy against pathogenic inflammatory signaling\n",
      "[NF-kappaB-sensitive, miRNA-146a-mediated, modulation, of, CFH, gene, expression, may, in, part, regulate, an, inflammatory, response, in, AD, brain, and, in, stressed, HN, cell, models, of, AD, and, illustrate, the, potential, for, anti-miRNAs, as, an, effective, therapeutic, strategy, against, pathogenic, inflammatory, signaling]\n",
      "TextBetween: NF-kappaB-sensitive miRNA-146a-mediated modulation of CFH gene expression may in part regulate an inflammatory response in AD brain and in stressed HN cell models of AD and illustrate the potential for anti-miRNAs as an effective therapeutic strategy against pathogenic inflammatory signaling\n",
      "infer direction from token set [NF-kappaB-sensitive, miRNA-146a-mediated, modulation, of, CFH, gene, expression, may, in, part, regulate, an, inflammatory, response, in, AD, brain, and, in, stressed, HN, cell, models, of, AD, and, illustrate, the, potential, for, anti-miRNAs, as, an, effective, therapeutic, strategy, against, pathogenic, inflammatory, signaling]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbecba9e158>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62ae8>, {'mir': Counter({'NEU': 1}), 'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(modulation, 'NEU'), (expression, 'NEU')]\n",
      "Between Tokens [(modulation, 'NEU')]\n",
      "counts between final False\n",
      "['NF-kappaB-sensitive', 'miRNA-146a-mediated', 'modulation', 'of', 'CFH', 'gene', 'expression']\n",
      "Counter({'NEU': 2})\n",
      "[(modulation, 'NEU'), (expression, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence These data indicate that NF-kappaB-sensitive miRNA-146a-mediated modulation of CFH gene expression may in part regulate an inflammatory response in AD brain and in stressed HN cell models of AD and illustrate the potential for anti-miRNAs as an effective therapeutic strategy against pathogenic inflammatory signaling.\n",
      "full sentence These data indicate that NF-kappaB-sensitive <e1>miRNA-146a</e1>-mediated modulation of <e2>CFH</e2> gene expression may in part regulate an inflammatory response in AD brain and in stressed HN cell models of AD and illustrate the potential for anti-miRNAs as an effective therapeutic strategy against pathogenic inflammatory signaling.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'These data indicate that NF-kappaB-sensitive <e1>miRNA-146a</e1>-mediated modulation of <e2>CFH</e2> gene expression may in part regulate an inflammatory response in AD brain and in stressed HN cell models of AD and illustrate the potential for anti-miRNAs as an effective therapeutic strategy against pathogenic inflammatory signaling.', 'full_sentence': 'These data indicate that NF-kappaB-sensitive <e1>miRNA-146a</e1>-mediated modulation of <e2>CFH</e2> gene expression may in part regulate an inflammatory response in AD brain and in stressed HN cell models of AD and illustrate the potential for anti-miRNAs as an effective therapeutic strategy against pathogenic inflammatory signaling.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (45, 55), 'entity_text': 'miRNA-146a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (79, 82), 'entity_text': 'CFH'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d157.s0.p2\n",
      "IS:  ('MIR_GENE', 'NEU')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment gene mir', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'In a screening of MB cell lines, the miRNA miR-199b-5p was seen to be a regulator of the Notch pathway through its targeting of the transcription factor HES1. Down-regulation of <e2>HES1</e2> expression by <e1>miR-199b-5p</e1> negatively regulates the proliferation rate and anchorage-independent growth of MB cells.', 'full_sentence': 'METHODOLOGY/PRINCIPAL FINDINGS; In a screening of MB cell lines, the miRNA miR-199b-5p was seen to be a regulator of the Notch pathway through its targeting of the transcription factor HES1. Down-regulation of <e2>HES1</e2> expression by <e1>miR-199b-5p</e1> negatively regulates the proliferation rate and anchorage-independent growth of MB cells.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (178, 182), 'entity_text': 'HES1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (197, 208), 'entity_text': 'miR-199b-5p'}}\n",
      "add special case HES1\n",
      "add special case miR-199b-5p\n",
      "HES1 32 miR-199b-5p 35\n",
      "(u\"In a screening of MB cell lines, the miRNA miR-199b-5p was seen to be a regulator of the Notch pathway through its targeting of the transcription factor HES1. Down-regulation of HES1 expression by miR-199b-5p negatively regulates the proliferation rate and anchorage-independent growth of MB cells.\", 32, 35)\n",
      "geneword HES1 mirword miR-199b-5p\n",
      "Conjuncts [growth, rate]\n",
      "neighbour growth of\n",
      "neighbour growth of\n",
      "Conjunction before fc rate\n",
      "fc add children for rate\n",
      "fc add children for rate\n",
      "Conjuncts [rate, growth]\n",
      "neighbour rate and\n",
      "neighbour rate and\n",
      "Conjunction before fc growth\n",
      "fc add children for growth\n",
      "Conjunctions ( conj )\n",
      "rate [proliferation, rate, anchorage-independent, growth, of, MB, cells]\n",
      "Conjuncts [seen, screening]\n",
      "neighbour seen to\n",
      "neighbour seen to\n",
      "Conjunction before fc screening\n",
      "fc add children for screening\n",
      "fc add children for screening\n",
      "Conjuncts [screening, lines]\n",
      "neighbour screening of\n",
      "neighbour screening of\n",
      "Conjunction before fc lines\n",
      "fc add children for lines\n",
      "fc add children for lines\n",
      "fc add children for lines\n",
      "Conjuncts [regulator, pathway]\n",
      "neighbour regulator of\n",
      "neighbour regulator of\n",
      "Conjunction before fc pathway\n",
      "fc add children for pathway\n",
      "fc add children for pathway\n",
      "Conjuncts [regulator, targeting]\n",
      "neighbour regulator of\n",
      "neighbour regulator of\n",
      "Conjunction before fc targeting\n",
      "fc add children for targeting\n",
      "fc add children for targeting\n",
      "Conjuncts [targeting, HES1]\n",
      "neighbour targeting of\n",
      "neighbour targeting of\n",
      "Conjunction before fc HES1\n",
      "fc add children for HES1\n",
      "fc add children for HES1\n",
      "fc add children for HES1\n",
      "Conjuncts [Down-regulation, expression]\n",
      "neighbour Down-regulation of\n",
      "neighbour Down-regulation of\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conjuncts [Down-regulation, miR-199b-5p]\n",
      "neighbour Down-regulation of\n",
      "neighbour Down-regulation of\n",
      "Conjunction before fc miR-199b-5p\n",
      "fc add children for miR-199b-5p\n",
      "Conjuncts [rate, cells]\n",
      "neighbour rate and\n",
      "neighbour rate and\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conunct merge check: screening pathway\n",
      "seen to be a\n",
      "Conunct merge check: pathway expression\n",
      "HES1.\n",
      "Conunct merge check: expression cells\n",
      "miR-199b-5p negatively regulates the proliferation\n",
      "Conjunctions ( nmod )\n",
      "screening [In, screening, of, MB, cell, lines, seen]\n",
      "pathway [regulator, of, Notch, pathway, through, targeting, of, transcription, factor, HES1]\n",
      "expression [Down-regulation, of, HES1, expression, by, miR-199b-5p]\n",
      "cells [rate, of, MB, cells]\n",
      "achildren {miR-199b-5p}\n",
      "achildren False\n",
      "SDP sent: In a screening of MB cell lines, the miRNA miR-199b-5p was seen to be a regulator of the Notch pathway through its targeting of the transcription factor HES1. Down-regulation of HES1 expression by miR-199b-5p negatively regulates the proliferation rate and anchorage-independent growth of MB cells.\n",
      "SDP     : [(HES1, 'NOUN', 'compound', False), (expression, 'NOUN', 'nmod', False), (Down-regulation, 'NOUN', 'nsubj', False), (miR-199b-5p, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[anchorage-independent, growth]\n",
      "0 [In, a, screening, of, MB, cell, lines, ,, the, miRNA, miR-199b-5p, was, seen, to, be, a, regulator, of, the, Notch, pathway, through, its, targeting, of, the, transcription, factor, HES1, ., Down-regulation, of, HES1, expression, by, miR-199b-5p, negatively, regulates, the, proliferation, rate, and, anchorage-independent, growth, of, MB, cells] 47 0 In a screening of MB cell lines, the miRNA miR-199b-5p was seen to be a regulator of the Notch pathway through its targeting of the transcription factor HES1. Down-regulation of HES1 expression by miR-199b-5p negatively regulates the proliferation rate and anchorage-independent growth of MB cells.\n",
      "DeepCheck Conj Subtree\n",
      "[In, a, screening, of, MB, cell, lines]\n",
      "DeepCheck Conj Subtree\n",
      "[of, MB, cell, lines]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, Notch, pathway]\n",
      "DeepCheck Conj Subtree\n",
      "[through, its, targeting, of, the, transcription, factor, HES1]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, transcription, factor, HES1]\n",
      "DeepCheck Conj Subtree\n",
      "[of, HES1, expression]\n",
      "DeepCheck Conj Subtree\n",
      "[by, miR-199b-5p]\n",
      "DeepCheck Conj Subtree\n",
      "[anchorage-independent, growth]\n",
      "DeepCheck Conj Subtree\n",
      "[of, MB, cells]\n",
      "0 [In, a, screening, of, MB, cell, lines, ,, the, miRNA, miR-199b-5p, was, seen, to, be, a, regulator, of, the, Notch, pathway, through, its, targeting, of, the, transcription, factor, HES1, ., Down-regulation, of, HES1, expression, by, miR-199b-5p, negatively, regulates, the, proliferation, rate, and, anchorage-independent, growth, of, MB, cells] 47 0 In a screening of MB cell lines, the miRNA miR-199b-5p was seen to be a regulator of the Notch pathway through its targeting of the transcription factor HES1. Down-regulation of HES1 expression by miR-199b-5p negatively regulates the proliferation rate and anchorage-independent growth of MB cells.\n",
      "Recognized overlapping entities: {Interval(178, 182, ('GENE_OR_GENE_PRODUCT', 'HES1'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text In a screening of MB cell lines , the miRNA miR-199b-5p was seen to be a regulator of the Notch pathway through its targeting of the transcription factor HES1 . Down-regulation of HES1 expression by miR-199b-5p negatively regulates the proliferation rate and anchorage-independent growth of MB cells\n",
      "(False, {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment gene mir', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence In a screening of MB cell lines, the miRNA miR-199b-5p was seen to be a regulator of the Notch pathway through its targeting of the transcription factor HES1. Down-regulation of HES1 expression by miR-199b-5p negatively regulates the proliferation rate and anchorage-independent growth of MB cells.\n",
      "full sentence In a screening of MB cell lines, the miRNA miR-199b-5p was seen to be a regulator of the Notch pathway through its targeting of the transcription factor HES1. Down-regulation of <e2>HES1</e2> expression by <e1>miR-199b-5p</e1> negatively regulates the proliferation rate and anchorage-independent growth of MB cells.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment gene mir', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'In a screening of MB cell lines, the miRNA miR-199b-5p was seen to be a regulator of the Notch pathway through its targeting of the transcription factor HES1. Down-regulation of <e2>HES1</e2> expression by <e1>miR-199b-5p</e1> negatively regulates the proliferation rate and anchorage-independent growth of MB cells.', 'full_sentence': 'METHODOLOGY/PRINCIPAL FINDINGS; In a screening of MB cell lines, the miRNA miR-199b-5p was seen to be a regulator of the Notch pathway through its targeting of the transcription factor HES1. Down-regulation of <e2>HES1</e2> expression by <e1>miR-199b-5p</e1> negatively regulates the proliferation rate and anchorage-independent growth of MB cells.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (178, 182), 'entity_text': 'HES1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (197, 208), 'entity_text': 'miR-199b-5p'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d157.s9.p0\n",
      "IS:  ('MIR_GENE', 'NEU')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene', 'reg_detect_major': 'compartment'}}, 'tagged_sent': '<e1>MicroRNA-199b-5p</e1> impairs cancer stem cells through negative regulation of <e2>HES1</e2> in medulloblastoma.', 'full_sentence': '<e1>MicroRNA-199b-5p</e1> impairs cancer stem cells through negative regulation of <e2>HES1</e2> in medulloblastoma.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (0, 16), 'entity_text': 'MicroRNA-199b-5p'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (74, 78), 'entity_text': 'HES1'}}\n",
      "add special case MicroRNA-199b-5p\n",
      "add special case HES1\n",
      "MicroRNA-199b-5p 0 HES1 9\n",
      "(u\"MicroRNA-199b-5p impairs cancer stem cells through negative regulation of HES1 in medulloblastoma.\", 0, 9)\n",
      "geneword HES1 mirword MicroRNA-199b-5p\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [impairs, regulation]\n",
      "neighbour impairs cancer\n",
      "neighbour impairs cancer\n",
      "Conjunction before fc regulation\n",
      "fc add children for regulation\n",
      "fc add children for regulation\n",
      "fc add children for regulation\n",
      "fc add children for regulation\n",
      "Conjuncts [regulation, HES1]\n",
      "neighbour regulation of\n",
      "neighbour regulation of\n",
      "Conjunction before fc HES1\n",
      "fc add children for HES1\n",
      "Conjuncts [regulation, medulloblastoma]\n",
      "neighbour regulation of\n",
      "neighbour regulation of\n",
      "Conjunction before fc medulloblastoma\n",
      "fc add children for medulloblastoma\n",
      "Conjunctions ( nmod )\n",
      "regulation [impairs, through, negative, regulation, of, HES1, in, medulloblastoma]\n",
      "SDP sent: MicroRNA-199b-5p impairs cancer stem cells through negative regulation of HES1 in medulloblastoma.\n",
      "SDP     : [(MicroRNA-199b-5p, 'NOUN', 'nsubj', False), (impairs, 'VERB', 'ROOT', False), (regulation, 'NOUN', 'nmod', False), (HES1, 'NOUN', 'nmod', False)]\n",
      "0 [MicroRNA-199b-5p, impairs, cancer, stem, cells, through, negative, regulation, of, HES1, in, medulloblastoma] 12 0 MicroRNA-199b-5p impairs cancer stem cells through negative regulation of HES1 in medulloblastoma.\n",
      "DeepCheck Conj Subtree\n",
      "[through, negative, regulation, of, HES1, in, medulloblastoma]\n",
      "DeepCheck Conj Subtree\n",
      "[of, HES1]\n",
      "DeepCheck Conj Subtree\n",
      "[in, medulloblastoma]\n",
      "0 [MicroRNA-199b-5p, impairs, cancer, stem, cells, through, negative, regulation, of, HES1, in, medulloblastoma] 12 0 MicroRNA-199b-5p impairs cancer stem cells through negative regulation of HES1 in medulloblastoma.\n",
      "Recognized overlapping entities: {Interval(74, 78, ('GENE_OR_GENE_PRODUCT', 'HES1'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text MicroRNA-199b-5p impairs cancer stem cells through negative regulation of HES1 in medulloblastoma\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence MicroRNA-199b-5p impairs cancer stem cells through negative regulation of HES1 in medulloblastoma.\n",
      "full sentence <e1>MicroRNA-199b-5p</e1> impairs cancer stem cells through negative regulation of <e2>HES1</e2> in medulloblastoma.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene', 'reg_detect_major': 'compartment'}}, 'tagged_sent': '<e1>MicroRNA-199b-5p</e1> impairs cancer stem cells through negative regulation of <e2>HES1</e2> in medulloblastoma.', 'full_sentence': '<e1>MicroRNA-199b-5p</e1> impairs cancer stem cells through negative regulation of <e2>HES1</e2> in medulloblastoma.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (0, 16), 'entity_text': 'MicroRNA-199b-5p'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (74, 78), 'entity_text': 'HES1'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d178.s4.p2\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('GENE_MIR', 'DOWN', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'The levels of primary <e1>miR-21</e1> gene transcripts, precursor miR-21, and mature miR-21 decreased 6 hours after the addition of <e2>IFN-beta</e2>, indicating that the reduction in miR-21 levels was due to transcriptional suppression.', 'full_sentence': 'The levels of primary <e1>miR-21</e1> gene transcripts, precursor miR-21, and mature miR-21 decreased 6 hours after the addition of <e2>IFN-beta</e2>, indicating that the reduction in miR-21 levels was due to transcriptional suppression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (22, 28), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (123, 131), 'entity_text': 'IFN-beta'}}\n",
      "add special case miR-21\n",
      "add special case IFN-beta\n",
      "miR-21 4 IFN-beta 21\n",
      "(u\"The levels of primary miR-21 gene transcripts, precursor miR-21, and mature miR-21 decreased 6 hours after the addition of IFN-beta, indicating that the reduction in miR-21 levels was due to transcriptional suppression.\", 4, 21)\n",
      "geneword IFN-beta mirword miR-21\n",
      "Conjuncts [miR-21, miR-21, transcripts]\n",
      "neighbour miR-21 ,\n",
      "neighbour miR-21 and\n",
      "neighbour miR-21 decreased\n",
      "neighbour miR-21 decreased\n",
      "Conjunction before fc transcripts\n",
      "fc add children for transcripts\n",
      "fc add children for transcripts\n",
      "fc add children for transcripts\n",
      "fc add children for transcripts\n",
      "Conjuncts [transcripts, miR-21, miR-21]\n",
      "neighbour transcripts ,\n",
      "neighbour transcripts precursor\n",
      "neighbour miR-21 decreased\n",
      "neighbour miR-21 decreased\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conjuncts [transcripts, miR-21, miR-21]\n",
      "neighbour transcripts ,\n",
      "neighbour transcripts precursor\n",
      "neighbour miR-21 ,\n",
      "neighbour miR-21 and\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conjunctions ( conj )\n",
      "transcripts [of, primary, miR-21, gene, transcripts, precursor, miR-21, mature, miR-21]\n",
      "Conjuncts [levels, transcripts]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc transcripts\n",
      "fc add children for transcripts\n",
      "fc add children for transcripts\n",
      "fc add children for transcripts\n",
      "fc add children for transcripts\n",
      "Conjuncts [decreased, addition]\n",
      "neighbour decreased 6\n",
      "neighbour decreased 6\n",
      "Conjunction before fc addition\n",
      "fc add children for addition\n",
      "fc add children for addition\n",
      "Conjuncts [addition, IFN-beta]\n",
      "neighbour addition of\n",
      "neighbour addition of\n",
      "Conjunction before fc IFN-beta\n",
      "fc add children for IFN-beta\n",
      "Conjuncts [reduction, levels]\n",
      "neighbour reduction in\n",
      "neighbour reduction in\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conunct merge check: transcripts addition\n",
      "transcripts, precursor miR-21, and mature miR-21\n",
      "Conunct merge check: addition levels\n",
      "IFN-beta, indicating that the\n",
      "Conjunctions ( nmod )\n",
      "transcripts [levels, of, primary, miR-21, gene, transcripts]\n",
      "addition [decreased, after, addition, of, IFN-beta]\n",
      "levels [reduction, in, miR-21, levels]\n",
      "SDP sent: The levels of primary miR-21 gene transcripts, precursor miR-21, and mature miR-21 decreased 6 hours after the addition of IFN-beta, indicating that the reduction in miR-21 levels was due to transcriptional suppression.\n",
      "SDP     : [(miR-21, 'NOUN', 'compound', False), (transcripts, 'NOUN', 'nmod', False), (levels, 'NOUN', 'nsubj', False), (decreased, 'VERB', 'ROOT', False), (addition, 'NOUN', 'nmod', False), (IFN-beta, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[precursor, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[mature, miR-21]\n",
      "0 [The, levels, of, primary, miR-21, gene, transcripts, ,, precursor, miR-21, ,, and, mature, miR-21, decreased, 6, hours, after, the, addition, of, IFN-beta] 22 0 The levels of primary miR-21 gene transcripts, precursor miR-21, and mature miR-21 decreased 6 hours after the addition of IFN-beta, indicating that the reduction in miR-21 levels was due to transcriptional suppression.\n",
      "1 [indicating, that, the, reduction, in, miR-21, levels, was, due, to, transcriptional, suppression] 12 133 indicating that the reduction in miR-21 levels was due to transcriptional suppression.\n",
      "DeepCheck Conj Subtree\n",
      "[of, primary, miR-21, gene, transcripts, ,, precursor, miR-21, ,, and, mature, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[precursor, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[mature, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[after, the, addition, of, IFN-beta]\n",
      "DeepCheck Conj Subtree\n",
      "[of, IFN-beta]\n",
      "DeepCheck Conj Subtree\n",
      "[in, miR-21, levels]\n",
      "0 [The, levels, of, primary, miR-21, gene, transcripts, ,, precursor, miR-21, ,, and, mature, miR-21, decreased, 6, hours, after, the, addition, of, IFN-beta] 22 0 The levels of primary miR-21 gene transcripts, precursor miR-21, and mature miR-21 decreased 6 hours after the addition of IFN-beta, indicating that the reduction in miR-21 levels was due to transcriptional suppression.\n",
      "1 [indicating, that, the, reduction, in, miR-21, levels, was, due, to, transcriptional, suppression] 12 133 indicating that the reduction in miR-21 levels was due to transcriptional suppression.\n",
      "Recognized overlapping entities: {Interval(123, 131, ('GENE_OR_GENE_PRODUCT', 'IFN-beta'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1, 'POS': 1})\n",
      "Compartment Text The levels of primary miR-21 gene transcripts , precursor miR-21 , and mature miR-21 decreased 6 hours after the addition of IFN-beta\n",
      "[The, levels, of, primary, miR-21, gene, transcripts, ,, precursor, miR-21, ,, and, mature, miR-21, decreased, 6, hours, after, the, addition, of, IFN-beta]\n",
      "TextBetween: The levels of primary miR-21 gene transcripts , precursor miR-21 , and mature miR-21 decreased 6 hours after the addition of IFN-beta\n",
      "infer direction from token set [The, levels, of, primary, miR-21, gene, transcripts, ,, precursor, miR-21, ,, and, mature, miR-21, decreased, 6, hours, after, the, addition, of, IFN-beta]\n",
      "stemTypes2Count: Counter({'NEG': 1, 'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbecba9e2f0>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 1, 'NEG': 1, 'POS': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62bf8>, {'mir': Counter({'NEU': 1}), 'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count [(transcripts, 'NEU'), (decreased, 'NEG'), (addition, 'POS')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}})\n",
      " hit sentence The levels of primary miR-21 gene transcripts, precursor miR-21, and mature miR-21 decreased 6 hours after the addition of IFN-beta, indicating that the reduction in miR-21 levels was due to transcriptional suppression.\n",
      "full sentence The levels of primary <e1>miR-21</e1> gene transcripts, precursor miR-21, and mature miR-21 decreased 6 hours after the addition of <e2>IFN-beta</e2>, indicating that the reduction in miR-21 levels was due to transcriptional suppression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'The levels of primary <e1>miR-21</e1> gene transcripts, precursor miR-21, and mature miR-21 decreased 6 hours after the addition of <e2>IFN-beta</e2>, indicating that the reduction in miR-21 levels was due to transcriptional suppression.', 'full_sentence': 'The levels of primary <e1>miR-21</e1> gene transcripts, precursor miR-21, and mature miR-21 decreased 6 hours after the addition of <e2>IFN-beta</e2>, indicating that the reduction in miR-21 levels was due to transcriptional suppression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (22, 28), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (123, 131), 'entity_text': 'IFN-beta'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d178.s4.p3\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('GENE_MIR', 'DOWN', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'The levels of primary miR-21 gene transcripts, precursor <e1>miR-21</e1>, and mature miR-21 decreased 6 hours after the addition of <e2>IFN-beta</e2>, indicating that the reduction in miR-21 levels was due to transcriptional suppression.', 'full_sentence': 'The levels of primary miR-21 gene transcripts, precursor <e1>miR-21</e1>, and mature miR-21 decreased 6 hours after the addition of <e2>IFN-beta</e2>, indicating that the reduction in miR-21 levels was due to transcriptional suppression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (57, 63), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (123, 131), 'entity_text': 'IFN-beta'}}\n",
      "add special case miR-21\n",
      "add special case IFN-beta\n",
      "miR-21 9 IFN-beta 21\n",
      "(u\"The levels of primary miR-21 gene transcripts, precursor miR-21, and mature miR-21 decreased 6 hours after the addition of IFN-beta, indicating that the reduction in miR-21 levels was due to transcriptional suppression.\", 9, 21)\n",
      "geneword IFN-beta mirword miR-21\n",
      "Conjuncts [miR-21, miR-21, transcripts]\n",
      "neighbour miR-21 ,\n",
      "neighbour miR-21 and\n",
      "neighbour miR-21 decreased\n",
      "neighbour miR-21 decreased\n",
      "Conjunction before fc transcripts\n",
      "fc add children for transcripts\n",
      "fc add children for transcripts\n",
      "fc add children for transcripts\n",
      "fc add children for transcripts\n",
      "Conjuncts [transcripts, miR-21, miR-21]\n",
      "neighbour transcripts ,\n",
      "neighbour transcripts precursor\n",
      "neighbour miR-21 decreased\n",
      "neighbour miR-21 decreased\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conjuncts [transcripts, miR-21, miR-21]\n",
      "neighbour transcripts ,\n",
      "neighbour transcripts precursor\n",
      "neighbour miR-21 ,\n",
      "neighbour miR-21 and\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conjunctions ( conj )\n",
      "transcripts [of, primary, miR-21, gene, transcripts, precursor, miR-21, mature, miR-21]\n",
      "Conjuncts [levels, transcripts]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc transcripts\n",
      "fc add children for transcripts\n",
      "fc add children for transcripts\n",
      "fc add children for transcripts\n",
      "fc add children for transcripts\n",
      "Conjuncts [decreased, addition]\n",
      "neighbour decreased 6\n",
      "neighbour decreased 6\n",
      "Conjunction before fc addition\n",
      "fc add children for addition\n",
      "fc add children for addition\n",
      "Conjuncts [addition, IFN-beta]\n",
      "neighbour addition of\n",
      "neighbour addition of\n",
      "Conjunction before fc IFN-beta\n",
      "fc add children for IFN-beta\n",
      "Conjuncts [reduction, levels]\n",
      "neighbour reduction in\n",
      "neighbour reduction in\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conunct merge check: transcripts addition\n",
      "transcripts, precursor miR-21, and mature miR-21\n",
      "Conunct merge check: addition levels\n",
      "IFN-beta, indicating that the\n",
      "Conjunctions ( nmod )\n",
      "transcripts [levels, of, primary, miR-21, gene, transcripts]\n",
      "addition [decreased, after, addition, of, IFN-beta]\n",
      "levels [reduction, in, miR-21, levels]\n",
      "SDP sent: The levels of primary miR-21 gene transcripts, precursor miR-21, and mature miR-21 decreased 6 hours after the addition of IFN-beta, indicating that the reduction in miR-21 levels was due to transcriptional suppression.\n",
      "SDP     : [(miR-21, 'NOUN', 'conj', False), (transcripts, 'NOUN', 'nmod', True), (levels, 'NOUN', 'nsubj', False), (decreased, 'VERB', 'ROOT', False), (addition, 'NOUN', 'nmod', False), (IFN-beta, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[precursor, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[mature, miR-21]\n",
      "0 [The, levels, of, primary, miR-21, gene, transcripts, ,, precursor, miR-21, ,, and, mature, miR-21, decreased, 6, hours, after, the, addition, of, IFN-beta] 22 0 The levels of primary miR-21 gene transcripts, precursor miR-21, and mature miR-21 decreased 6 hours after the addition of IFN-beta, indicating that the reduction in miR-21 levels was due to transcriptional suppression.\n",
      "1 [indicating, that, the, reduction, in, miR-21, levels, was, due, to, transcriptional, suppression] 12 133 indicating that the reduction in miR-21 levels was due to transcriptional suppression.\n",
      "DeepCheck Conj Subtree\n",
      "[of, primary, miR-21, gene, transcripts, ,, precursor, miR-21, ,, and, mature, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[precursor, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[mature, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[after, the, addition, of, IFN-beta]\n",
      "DeepCheck Conj Subtree\n",
      "[of, IFN-beta]\n",
      "DeepCheck Conj Subtree\n",
      "[in, miR-21, levels]\n",
      "0 [The, levels, of, primary, miR-21, gene, transcripts, ,, precursor, miR-21, ,, and, mature, miR-21, decreased, 6, hours, after, the, addition, of, IFN-beta] 22 0 The levels of primary miR-21 gene transcripts, precursor miR-21, and mature miR-21 decreased 6 hours after the addition of IFN-beta, indicating that the reduction in miR-21 levels was due to transcriptional suppression.\n",
      "1 [indicating, that, the, reduction, in, miR-21, levels, was, due, to, transcriptional, suppression] 12 133 indicating that the reduction in miR-21 levels was due to transcriptional suppression.\n",
      "Recognized overlapping entities: {Interval(123, 131, ('GENE_OR_GENE_PRODUCT', 'IFN-beta'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1, 'POS': 1})\n",
      "Compartment Text The levels of primary miR-21 gene transcripts , precursor miR-21 , and mature miR-21 decreased 6 hours after the addition of IFN-beta\n",
      "[The, levels, of, primary, miR-21, gene, transcripts, ,, precursor, miR-21, ,, and, mature, miR-21, decreased, 6, hours, after, the, addition, of, IFN-beta]\n",
      "TextBetween: The levels of primary miR-21 gene transcripts , precursor miR-21 , and mature miR-21 decreased 6 hours after the addition of IFN-beta\n",
      "infer direction from token set [The, levels, of, primary, miR-21, gene, transcripts, ,, precursor, miR-21, ,, and, mature, miR-21, decreased, 6, hours, after, the, addition, of, IFN-beta]\n",
      "stemTypes2Count: Counter({'NEG': 1, 'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8cf28>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEG': 1, 'POS': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c598>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count [(decreased, 'NEG'), (addition, 'POS')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}})\n",
      " hit sentence The levels of primary miR-21 gene transcripts, precursor miR-21, and mature miR-21 decreased 6 hours after the addition of IFN-beta, indicating that the reduction in miR-21 levels was due to transcriptional suppression.\n",
      "full sentence The levels of primary miR-21 gene transcripts, precursor <e1>miR-21</e1>, and mature miR-21 decreased 6 hours after the addition of <e2>IFN-beta</e2>, indicating that the reduction in miR-21 levels was due to transcriptional suppression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'The levels of primary miR-21 gene transcripts, precursor <e1>miR-21</e1>, and mature miR-21 decreased 6 hours after the addition of <e2>IFN-beta</e2>, indicating that the reduction in miR-21 levels was due to transcriptional suppression.', 'full_sentence': 'The levels of primary miR-21 gene transcripts, precursor <e1>miR-21</e1>, and mature miR-21 decreased 6 hours after the addition of <e2>IFN-beta</e2>, indicating that the reduction in miR-21 levels was due to transcriptional suppression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (57, 63), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (123, 131), 'entity_text': 'IFN-beta'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d178.s5.p4\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('GENE_MIR', 'DOWN', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'the addition of signal transducers and activators of transcription 3 (STAT3)-expressing vectors induced the <e2>IFN-beta</e2>-mediated suppression of <e1>miR-21</e1>, whereas STAT3-inhibiting agents inhibited the miR-21 suppression.', 'full_sentence': 'We did reporter assays to elucidate the IFN-beta-mediated suppression of miR-21; the addition of signal transducers and activators of transcription 3 (STAT3)-expressing vectors induced the <e2>IFN-beta</e2>-mediated suppression of <e1>miR-21</e1>, whereas STAT3-inhibiting agents inhibited the miR-21 suppression.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (108, 116), 'entity_text': 'IFN-beta'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (141, 147), 'entity_text': 'miR-21'}}\n",
      "add special case IFN-beta\n",
      "add special case miR-21\n",
      "IFN-beta-mediated 14 miR-21 17\n",
      "(u\"the addition of signal transducers and activators of transcription 3 (STAT3)-expressing vectors induced the IFN-beta-mediated suppression of miR-21, whereas STAT3-inhibiting agents inhibited the miR-21 suppression.\", 14, 17)\n",
      "geneword IFN-beta-mediated mirword miR-21\n",
      "Conjuncts [transcription, transducers]\n",
      "neighbour transcription 3\n",
      "neighbour transcription 3\n",
      "Conjunction before fc transducers\n",
      "fc add children for transducers\n",
      "fc add children for transducers\n",
      "Conjuncts [transducers, transcription]\n",
      "neighbour transducers and\n",
      "neighbour transducers and\n",
      "Conjunction before fc transcription\n",
      "fc add children for transcription\n",
      "fc add children for transcription\n",
      "fc add children for transcription\n",
      "Conjunctions ( conj )\n",
      "transducers [of, signal, transducers, activators, of, transcription, 3]\n",
      "Conjuncts [addition, transducers]\n",
      "neighbour addition of\n",
      "neighbour addition of\n",
      "Conjunction before fc transducers\n",
      "fc add children for transducers\n",
      "fc add children for transducers\n",
      "Conjuncts [suppression, miR-21]\n",
      "neighbour suppression of\n",
      "neighbour suppression of\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conunct merge check: transducers miR-21\n",
      "transducers and activators of transcription 3 (STAT3)-expressing vectors induced the IFN-beta-mediated\n",
      "Conjunctions ( nmod )\n",
      "transducers [addition, of, signal, transducers]\n",
      "miR-21 [suppression, of, miR-21]\n",
      "SDP sent: the addition of signal transducers and activators of transcription 3 (STAT3)-expressing vectors induced the IFN-beta-mediated suppression of miR-21, whereas STAT3-inhibiting agents inhibited the miR-21 suppression.\n",
      "SDP     : [(IFN-beta-mediated, 'ADJ', 'amod', False), (suppression, 'NOUN', 'dobj', False), (miR-21, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[activators, of, transcription, 3]\n",
      "deepcheck connection rule noun found inhibited agents [STAT3-inhibiting, agents, inhibited, the, miR-21, suppression]\n",
      "0 [the, addition, of, signal, transducers, and, activators, of, transcription, 3, (STAT3)-expressing, vectors, induced, the, IFN-beta-mediated, suppression, of, miR-21] 18 0 the addition of signal transducers and activators of transcription 3 (STAT3)-expressing vectors induced the IFN-beta-mediated suppression of miR-21, whereas STAT3-inhibiting agents inhibited the miR-21 suppression.\n",
      "1 [STAT3-inhibiting, agents, inhibited, the, miR-21, suppression] 6 157 STAT3-inhibiting agents inhibited the miR-21 suppression.\n",
      "DeepCheck Conj Subtree\n",
      "[of, signal, transducers, and, activators, of, transcription, 3]\n",
      "DeepCheck Conj Subtree\n",
      "[activators, of, transcription, 3]\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-21]\n",
      "deepcheck connection rule noun found inhibited agents [STAT3-inhibiting, agents, inhibited, the, miR-21, suppression]\n",
      "0 [the, addition, of, signal, transducers, and, activators, of, transcription, 3, (STAT3)-expressing, vectors, induced, the, IFN-beta-mediated, suppression, of, miR-21] 18 0 the addition of signal transducers and activators of transcription 3 (STAT3)-expressing vectors induced the IFN-beta-mediated suppression of miR-21, whereas STAT3-inhibiting agents inhibited the miR-21 suppression.\n",
      "1 [STAT3-inhibiting, agents, inhibited, the, miR-21, suppression] 6 157 STAT3-inhibiting agents inhibited the miR-21 suppression.\n",
      "Context Fail for Noun Rule\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text the addition of signal transducers and activators of transcription 3 (STAT3)-expressing vectors induced the IFN-beta-mediated suppression of miR-21\n",
      "[the, addition, of, signal, transducers, and, activators, of, transcription, 3, (STAT3)-expressing, vectors, induced, the, IFN-beta-mediated, suppression, of, miR-21]\n",
      "TextBetween: the addition of signal transducers and activators of transcription 3 (STAT3)-expressing vectors induced the IFN-beta-mediated suppression of miR-21\n",
      "infer direction from token set [the, addition, of, signal, transducers, and, activators, of, transcription, 3, (STAT3)-expressing, vectors, induced, the, IFN-beta-mediated, suppression, of, miR-21]\n",
      "stemTypes2Count: Counter({'POS': 2, 'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbecba9e1e0>, {'gene': Counter({'POS': 1, 'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'POS': 1, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbecba9e048>, {'gene': Counter({'POS': 1, 'NEG': 1})})\n",
      "counts between 1 stemTypes2Count [(induced, 'POS'), (suppression, 'NEG')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}})\n",
      " hit sentence the addition of signal transducers and activators of transcription 3 (STAT3)-expressing vectors induced the IFN-beta-mediated suppression of miR-21, whereas STAT3-inhibiting agents inhibited the miR-21 suppression.\n",
      "full sentence the addition of signal transducers and activators of transcription 3 (STAT3)-expressing vectors induced the <e2>IFN-beta</e2>-mediated suppression of <e1>miR-21</e1>, whereas STAT3-inhibiting agents inhibited the miR-21 suppression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'the addition of signal transducers and activators of transcription 3 (STAT3)-expressing vectors induced the <e2>IFN-beta</e2>-mediated suppression of <e1>miR-21</e1>, whereas STAT3-inhibiting agents inhibited the miR-21 suppression.', 'full_sentence': 'We did reporter assays to elucidate the IFN-beta-mediated suppression of miR-21; the addition of signal transducers and activators of transcription 3 (STAT3)-expressing vectors induced the <e2>IFN-beta</e2>-mediated suppression of <e1>miR-21</e1>, whereas STAT3-inhibiting agents inhibited the miR-21 suppression.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (108, 116), 'entity_text': 'IFN-beta'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (141, 147), 'entity_text': 'miR-21'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d178.s7.p0\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case IFN-beta\n",
      "add special case miR-21\n",
      "IFN-beta 0 miR-21 3\n",
      "(u\"IFN-beta treatment reduced miR-21 expression in glioma cells markedly, and IFN-beta administration suppressed the growth of glioma-initiating cell-derived intracranial tumors.\", 0, 3)\n",
      "geneword IFN-beta mirword miR-21\n",
      "Conjuncts [suppressed, reduced]\n",
      "neighbour suppressed the\n",
      "neighbour suppressed the\n",
      "Conjunction before fc reduced\n",
      "fc add children for reduced\n",
      "Conjuncts [reduced, suppressed]\n",
      "neighbour reduced miR-21\n",
      "neighbour reduced miR-21\n",
      "Conjunction before fc suppressed\n",
      "fc add children for suppressed\n",
      "Conjunctions ( conj )\n",
      "reduced [reduced, miR-21, expression, in, glioma, cells, suppressed, growth, of, glioma-initiating, cell-derived, intracranial, tumors]\n",
      "Conjuncts [expression, cells]\n",
      "neighbour expression in\n",
      "neighbour expression in\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conjuncts [growth, tumors]\n",
      "neighbour growth of\n",
      "neighbour growth of\n",
      "Conjunction before fc tumors\n",
      "fc add children for tumors\n",
      "fc add children for tumors\n",
      "fc add children for tumors\n",
      "fc add children for tumors\n",
      "Conunct merge check: cells tumors\n",
      "cells markedly, and IFN-beta administration suppressed the\n",
      "Conjunctions ( nmod )\n",
      "cells [expression, in, glioma, cells]\n",
      "tumors [growth, of, glioma-initiating, cell-derived, intracranial, tumors]\n",
      "SDP sent: IFN-beta treatment reduced miR-21 expression in glioma cells markedly, and IFN-beta administration suppressed the growth of glioma-initiating cell-derived intracranial tumors.\n",
      "SDP     : [(IFN-beta, 'NOUN', 'compound', False), (treatment, 'NOUN', 'nsubj', False), (reduced, 'VERB', 'ROOT', False), (expression, 'NOUN', 'dobj', False), (miR-21, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule noun found suppressed IFN-beta [IFN-beta, administration, suppressed, the, growth, of, glioma-initiating, cell-derived, intracranial, tumors]\n",
      "0 [IFN-beta, treatment, reduced, miR-21, expression, in, glioma, cells, markedly] 9 0 IFN-beta treatment reduced miR-21 expression in glioma cells markedly, and IFN-beta administration suppressed the growth of glioma-initiating cell-derived intracranial tumors.\n",
      "1 [IFN-beta, administration, suppressed, the, growth, of, glioma-initiating, cell-derived, intracranial, tumors] 10 75 IFN-beta administration suppressed the growth of glioma-initiating cell-derived intracranial tumors.\n",
      "DeepCheck Conj Subtree\n",
      "[in, glioma, cells]\n",
      "deepcheck connection rule noun found suppressed IFN-beta [IFN-beta, administration, suppressed, the, growth, of, glioma-initiating, cell-derived, intracranial, tumors]\n",
      "DeepCheck Conj Subtree\n",
      "[of, glioma-initiating, cell-derived, intracranial, tumors]\n",
      "0 [IFN-beta, treatment, reduced, miR-21, expression, in, glioma, cells, markedly] 9 0 IFN-beta treatment reduced miR-21 expression in glioma cells markedly, and IFN-beta administration suppressed the growth of glioma-initiating cell-derived intracranial tumors.\n",
      "1 [IFN-beta, administration, suppressed, the, growth, of, glioma-initiating, cell-derived, intracranial, tumors] 10 75 IFN-beta administration suppressed the growth of glioma-initiating cell-derived intracranial tumors.\n",
      "Context Fail <regex.Match object; span=(7, 17), match=' treatment', fuzzy_counts=(0, 1, 0)> regex.Regex('(treatment){e<=1}', flags=regex.V0)\n",
      "Recognized overlapping entities: {Interval(0, 8, ('GENE_OR_GENE_PRODUCT', 'IFN-beta'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text IFN-beta treatment reduced miR-21 expression in glioma cells markedly\n",
      "[IFN-beta, treatment, reduced, miR-21, expression, in, glioma, cells, markedly]\n",
      "TextBetween: IFN-beta treatment reduced miR-21 expression in glioma cells markedly\n",
      "infer direction from token set [IFN-beta, treatment, reduced, miR-21, expression, in, glioma, cells, markedly]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c8c8>, {'mir': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEG': 1, 'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fc0bbbd3158>, {'mir': Counter({'NEG': 1, 'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(reduced, 'NEG'), (expression, 'NEU')]\n",
      "Between Tokens [(reduced, 'NEG')]\n",
      "counts between final False\n",
      "['IFN-beta', 'treatment', 'reduced', 'miR-21', 'expression', 'in']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "[(reduced, 'NEG'), (expression, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence IFN-beta treatment reduced miR-21 expression in glioma cells markedly, and IFN-beta administration suppressed the growth of glioma-initiating cell-derived intracranial tumors.\n",
      "full sentence <e2>IFN-beta</e2> treatment reduced <e1>miR-21</e1> expression in glioma cells markedly, and IFN-beta administration suppressed the growth of glioma-initiating cell-derived intracranial tumors.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e2>IFN-beta</e2> treatment reduced <e1>miR-21</e1> expression in glioma cells markedly, and IFN-beta administration suppressed the growth of glioma-initiating cell-derived intracranial tumors.', 'full_sentence': '<e2>IFN-beta</e2> treatment reduced <e1>miR-21</e1> expression in glioma cells markedly, and IFN-beta administration suppressed the growth of glioma-initiating cell-derived intracranial tumors.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (0, 8), 'entity_text': 'IFN-beta'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (27, 33), 'entity_text': 'miR-21'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d178.s8.p0\n",
      "IS:  ('GENE_MIR', 'NEU')\n",
      "SB:  ('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'compartment mir gene by', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'Thus, the results of our study show that the downregulation of <e1>miR-21</e1> contributes to the antitumor effects of <e2>IFN-beta</e2> and that miR-21 expression is negatively regulated by STAT3 activation.', 'full_sentence': 'Thus, the results of our study show that the downregulation of <e1>miR-21</e1> contributes to the antitumor effects of <e2>IFN-beta</e2> and that miR-21 expression is negatively regulated by STAT3 activation.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (63, 69), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (110, 118), 'entity_text': 'IFN-beta'}}\n",
      "add special case miR-21\n",
      "add special case IFN-beta\n",
      "miR-21 12 IFN-beta 19\n",
      "(u\"Thus, the results of our study show that the downregulation of miR-21 contributes to the antitumor effects of IFN-beta and that miR-21 expression is negatively regulated by STAT3 activation.\", 12, 19)\n",
      "geneword IFN-beta mirword miR-21\n",
      "Conjuncts [regulated, contributes]\n",
      "neighbour regulated by\n",
      "neighbour regulated by\n",
      "Conjunction before fc contributes\n",
      "fc add children for contributes\n",
      "Conjuncts [contributes, regulated]\n",
      "neighbour contributes to\n",
      "neighbour contributes to\n",
      "Conjunction before fc regulated\n",
      "fc add children for regulated\n",
      "Conjunctions ( conj )\n",
      "contributes [contributes, to, antitumor, effects, of, IFN-beta, regulated, by, STAT3, activation]\n",
      "Conjuncts [results, study]\n",
      "neighbour results of\n",
      "neighbour results of\n",
      "Conjunction before fc study\n",
      "fc add children for study\n",
      "Conjuncts [downregulation, miR-21]\n",
      "neighbour downregulation of\n",
      "neighbour downregulation of\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conjuncts [contributes, effects]\n",
      "neighbour contributes to\n",
      "neighbour contributes to\n",
      "Conjunction before fc effects\n",
      "fc add children for effects\n",
      "fc add children for effects\n",
      "fc add children for effects\n",
      "Conjuncts [effects, IFN-beta]\n",
      "neighbour effects of\n",
      "neighbour effects of\n",
      "Conjunction before fc IFN-beta\n",
      "fc add children for IFN-beta\n",
      "Conjuncts [regulated, activation]\n",
      "neighbour regulated by\n",
      "neighbour regulated by\n",
      "Conjunction before fc activation\n",
      "fc add children for activation\n",
      "fc add children for activation\n",
      "Conunct merge check: study miR-21\n",
      "study show that the\n",
      "Conunct merge check: miR-21 effects\n",
      "miR-21\n",
      "Conunct merge check: effects activation\n",
      "IFN-beta and that miR-21 expression is negatively\n",
      "Conjunctions ( nmod )\n",
      "study [results, of, study]\n",
      "miR-21 [downregulation, of, miR-21]\n",
      "effects [contributes, to, antitumor, effects, of, IFN-beta]\n",
      "activation [regulated, by, STAT3, activation]\n",
      "SDP sent: Thus, the results of our study show that the downregulation of miR-21 contributes to the antitumor effects of IFN-beta and that miR-21 expression is negatively regulated by STAT3 activation.\n",
      "SDP     : [(miR-21, 'NOUN', 'nmod', False), (downregulation, 'NOUN', 'nsubj', False), (contributes, 'VERB', 'ccomp', False), (effects, 'NOUN', 'nmod', False), (IFN-beta, 'NOUN', 'nmod', False)]\n",
      "deepcheck connection rule noun found regulated miR-21 [that, miR-21, expression, is, negatively, regulated, by, STAT3, activation]\n",
      "0 [Thus, ,, the, results, of, our, study, show] 8 0 Thus, the results of our study show that the downregulation of miR-21 contributes to the antitumor effects of IFN-beta and that miR-21 expression is negatively regulated by STAT3 activation.\n",
      "1 [the, downregulation, of, miR-21, contributes, to, the, antitumor, effects, of, IFN-beta] 11 41 the downregulation of miR-21 contributes to the antitumor effects of IFN-beta and that miR-21 expression is negatively regulated by STAT3 activation.\n",
      "2 [miR-21, expression, is, negatively, regulated, by, STAT3, activation] 8 128 miR-21 expression is negatively regulated by STAT3 activation.\n",
      "DeepCheck Conj Subtree\n",
      "[of, our, study]\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[to, the, antitumor, effects, of, IFN-beta]\n",
      "DeepCheck Conj Subtree\n",
      "[of, IFN-beta]\n",
      "deepcheck connection rule noun found regulated miR-21 [that, miR-21, expression, is, negatively, regulated, by, STAT3, activation]\n",
      "DeepCheck Conj Subtree\n",
      "[by, STAT3, activation]\n",
      "0 [Thus, ,, the, results, of, our, study, show] 8 0 Thus, the results of our study show that the downregulation of miR-21 contributes to the antitumor effects of IFN-beta and that miR-21 expression is negatively regulated by STAT3 activation.\n",
      "1 [the, downregulation, of, miR-21, contributes, to, the, antitumor, effects, of, IFN-beta] 11 41 the downregulation of miR-21 contributes to the antitumor effects of IFN-beta and that miR-21 expression is negatively regulated by STAT3 activation.\n",
      "2 [miR-21, expression, is, negatively, regulated, by, STAT3, activation] 8 128 miR-21 expression is negatively regulated by STAT3 activation.\n",
      "Recognized overlapping entities: {Interval(110, 118, ('GENE_OR_GENE_PRODUCT', 'IFN-beta'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text the downregulation of miR-21 contributes to the antitumor effects of IFN-beta\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'compartment mir gene by', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence Thus, the results of our study show that the downregulation of miR-21 contributes to the antitumor effects of IFN-beta and that miR-21 expression is negatively regulated by STAT3 activation.\n",
      "full sentence Thus, the results of our study show that the downregulation of <e1>miR-21</e1> contributes to the antitumor effects of <e2>IFN-beta</e2> and that miR-21 expression is negatively regulated by STAT3 activation.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'compartment mir gene by', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'Thus, the results of our study show that the downregulation of <e1>miR-21</e1> contributes to the antitumor effects of <e2>IFN-beta</e2> and that miR-21 expression is negatively regulated by STAT3 activation.', 'full_sentence': 'Thus, the results of our study show that the downregulation of <e1>miR-21</e1> contributes to the antitumor effects of <e2>IFN-beta</e2> and that miR-21 expression is negatively regulated by STAT3 activation.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (63, 69), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (110, 118), 'entity_text': 'IFN-beta'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d178.s8.p3\n",
      "IS:  ('GENE_MIR', 'NEU')\n",
      "SB:  ('GENE_MIR', 'DOWN', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': True, 'reg_detect': 'compartment mir gene by', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'Thus, the results of our study show that the downregulation of miR-21 contributes to the antitumor effects of IFN-beta and that <e1>miR-21</e1> expression is negatively regulated by <e2>STAT3</e2> activation.', 'full_sentence': 'Thus, the results of our study show that the downregulation of miR-21 contributes to the antitumor effects of IFN-beta and that <e1>miR-21</e1> expression is negatively regulated by <e2>STAT3</e2> activation.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (128, 134), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (173, 178), 'entity_text': 'STAT3'}}\n",
      "add special case miR-21\n",
      "add special case STAT3\n",
      "miR-21 22 STAT3 28\n",
      "(u\"Thus, the results of our study show that the downregulation of miR-21 contributes to the antitumor effects of IFN-beta and that miR-21 expression is negatively regulated by STAT3 activation.\", 22, 28)\n",
      "geneword STAT3 mirword miR-21\n",
      "Conjuncts [regulated, contributes]\n",
      "neighbour regulated by\n",
      "neighbour regulated by\n",
      "Conjunction before fc contributes\n",
      "fc add children for contributes\n",
      "Conjuncts [contributes, regulated]\n",
      "neighbour contributes to\n",
      "neighbour contributes to\n",
      "Conjunction before fc regulated\n",
      "fc add children for regulated\n",
      "Conjunctions ( conj )\n",
      "contributes [contributes, to, antitumor, effects, of, IFN-beta, regulated, by, STAT3, activation]\n",
      "Conjuncts [results, study]\n",
      "neighbour results of\n",
      "neighbour results of\n",
      "Conjunction before fc study\n",
      "fc add children for study\n",
      "Conjuncts [downregulation, miR-21]\n",
      "neighbour downregulation of\n",
      "neighbour downregulation of\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conjuncts [contributes, effects]\n",
      "neighbour contributes to\n",
      "neighbour contributes to\n",
      "Conjunction before fc effects\n",
      "fc add children for effects\n",
      "fc add children for effects\n",
      "fc add children for effects\n",
      "Conjuncts [effects, IFN-beta]\n",
      "neighbour effects of\n",
      "neighbour effects of\n",
      "Conjunction before fc IFN-beta\n",
      "fc add children for IFN-beta\n",
      "Conjuncts [regulated, activation]\n",
      "neighbour regulated by\n",
      "neighbour regulated by\n",
      "Conjunction before fc activation\n",
      "fc add children for activation\n",
      "fc add children for activation\n",
      "Conunct merge check: study miR-21\n",
      "study show that the\n",
      "Conunct merge check: miR-21 effects\n",
      "miR-21\n",
      "Conunct merge check: effects activation\n",
      "IFN-beta and that miR-21 expression is negatively\n",
      "Conjunctions ( nmod )\n",
      "study [results, of, study]\n",
      "miR-21 [downregulation, of, miR-21]\n",
      "effects [contributes, to, antitumor, effects, of, IFN-beta]\n",
      "activation [regulated, by, STAT3, activation]\n",
      "SDP sent: Thus, the results of our study show that the downregulation of miR-21 contributes to the antitumor effects of IFN-beta and that miR-21 expression is negatively regulated by STAT3 activation.\n",
      "SDP     : [(miR-21, 'NOUN', 'compound', False), (expression, 'NOUN', 'nsubjpass', False), (regulated, 'VERB', 'conj', False), (activation, 'NOUN', 'nmod', False), (STAT3, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule noun found regulated miR-21 [that, miR-21, expression, is, negatively, regulated, by, STAT3, activation]\n",
      "0 [Thus, ,, the, results, of, our, study, show] 8 0 Thus, the results of our study show that the downregulation of miR-21 contributes to the antitumor effects of IFN-beta and that miR-21 expression is negatively regulated by STAT3 activation.\n",
      "1 [the, downregulation, of, miR-21, contributes, to, the, antitumor, effects, of, IFN-beta] 11 41 the downregulation of miR-21 contributes to the antitumor effects of IFN-beta and that miR-21 expression is negatively regulated by STAT3 activation.\n",
      "2 [miR-21, expression, is, negatively, regulated, by, STAT3, activation] 8 128 miR-21 expression is negatively regulated by STAT3 activation.\n",
      "DeepCheck Conj Subtree\n",
      "[of, our, study]\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[to, the, antitumor, effects, of, IFN-beta]\n",
      "DeepCheck Conj Subtree\n",
      "[of, IFN-beta]\n",
      "deepcheck connection rule noun found regulated miR-21 [that, miR-21, expression, is, negatively, regulated, by, STAT3, activation]\n",
      "DeepCheck Conj Subtree\n",
      "[by, STAT3, activation]\n",
      "0 [Thus, ,, the, results, of, our, study, show] 8 0 Thus, the results of our study show that the downregulation of miR-21 contributes to the antitumor effects of IFN-beta and that miR-21 expression is negatively regulated by STAT3 activation.\n",
      "1 [the, downregulation, of, miR-21, contributes, to, the, antitumor, effects, of, IFN-beta] 11 41 the downregulation of miR-21 contributes to the antitumor effects of IFN-beta and that miR-21 expression is negatively regulated by STAT3 activation.\n",
      "2 [miR-21, expression, is, negatively, regulated, by, STAT3, activation] 8 128 miR-21 expression is negatively regulated by STAT3 activation.\n",
      "Recognized overlapping entities: {Interval(173, 178, ('GENE_OR_GENE_PRODUCT', 'STAT3'))}\n",
      "passive? True\n",
      "Counter()\n",
      "Compartment Text miR-21 expression is negatively regulated by STAT3 activation\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': True, 'reg_detect': 'compartment mir gene by', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence Thus, the results of our study show that the downregulation of miR-21 contributes to the antitumor effects of IFN-beta and that miR-21 expression is negatively regulated by STAT3 activation.\n",
      "full sentence Thus, the results of our study show that the downregulation of miR-21 contributes to the antitumor effects of IFN-beta and that <e1>miR-21</e1> expression is negatively regulated by <e2>STAT3</e2> activation.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': True, 'reg_detect': 'compartment mir gene by', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'Thus, the results of our study show that the downregulation of miR-21 contributes to the antitumor effects of IFN-beta and that <e1>miR-21</e1> expression is negatively regulated by <e2>STAT3</e2> activation.', 'full_sentence': 'Thus, the results of our study show that the downregulation of miR-21 contributes to the antitumor effects of IFN-beta and that <e1>miR-21</e1> expression is negatively regulated by <e2>STAT3</e2> activation.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (128, 134), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (173, 178), 'entity_text': 'STAT3'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d181.s9.p5\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('MIR_GENE', 'UP', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'These data suggest a significantly independent regulation of IRAK-1 and IRAK-2 in AD and in IL-1+A42 peptide-stressed HAG cells and that an inducible, NF-B-sensitive, <e1>miRNA-146a</e1>-mediated down-regulation of IRAK-1 coupled to an NF-B-induced up-regulation of <e2>IRAK-2</e2> expression drives an extensively sustained inflammatory response.', 'full_sentence': 'These data suggest a significantly independent regulation of IRAK-1 and IRAK-2 in AD and in IL-1+A42 peptide-stressed HAG cells and that an inducible, NF-B-sensitive, <e1>miRNA-146a</e1>-mediated down-regulation of IRAK-1 coupled to an NF-B-induced up-regulation of <e2>IRAK-2</e2> expression drives an extensively sustained inflammatory response.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (170, 180), 'entity_text': 'miRNA-146a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (261, 267), 'entity_text': 'IRAK-2'}}\n",
      "add special case miRNA-146a\n",
      "add special case IRAK-2\n",
      "miRNA-146a-mediated 26 IRAK-2 36\n",
      "(u\"These data suggest a significantly independent regulation of IRAK-1 and IRAK-2 in AD and in IL-1+A42 peptide-stressed HAG cells and that an inducible, NF-B-sensitive, miRNA-146a-mediated down-regulation of IRAK-1 coupled to an NF-B-induced up-regulation of IRAK-2 expression drives an extensively sustained inflammatory response.\", 26, 36)\n",
      "geneword IRAK-2 mirword miRNA-146a-mediated\n",
      "Conjuncts [IRAK-2, IRAK-1]\n",
      "neighbour IRAK-2 in\n",
      "neighbour IRAK-2 in\n",
      "Conjunction before fc IRAK-1\n",
      "fc add children for IRAK-1\n",
      "Conjuncts [IRAK-1, IRAK-2]\n",
      "neighbour IRAK-1 and\n",
      "neighbour IRAK-1 and\n",
      "Conjunction before fc IRAK-2\n",
      "Conjuncts [cells, AD]\n",
      "neighbour cells and\n",
      "neighbour cells and\n",
      "Conjunction before fc AD\n",
      "fc add children for AD\n",
      "Conjuncts [AD, cells]\n",
      "neighbour AD and\n",
      "neighbour AD and\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conunct merge check: IRAK-1 AD\n",
      "IRAK-2\n",
      "Conjunctions ( conj )\n",
      "IRAK-1 [of, IRAK-1, IRAK-2]\n",
      "AD [in, AD, in, IL-1+A42, peptide-stressed, HAG, cells]\n",
      "Conjuncts [regulation, IRAK-1]\n",
      "neighbour regulation of\n",
      "neighbour regulation of\n",
      "Conjunction before fc IRAK-1\n",
      "fc add children for IRAK-1\n",
      "Conjuncts [regulation, AD]\n",
      "neighbour regulation of\n",
      "neighbour regulation of\n",
      "Conjunction before fc AD\n",
      "fc add children for AD\n",
      "Conjuncts [down-regulation, IRAK-1]\n",
      "neighbour down-regulation of\n",
      "neighbour down-regulation of\n",
      "Conjunction before fc IRAK-1\n",
      "fc add children for IRAK-1\n",
      "Conjuncts [coupled, up-regulation]\n",
      "neighbour coupled to\n",
      "neighbour coupled to\n",
      "Conjunction before fc up-regulation\n",
      "fc add children for up-regulation\n",
      "fc add children for up-regulation\n",
      "fc add children for up-regulation\n",
      "Conjuncts [up-regulation, expression]\n",
      "neighbour up-regulation of\n",
      "neighbour up-regulation of\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conunct merge check: IRAK-1 IRAK-1\n",
      "AD and in IL-1+A42 peptide-stressed HAG cells and that an inducible, NF-B-sensitive, miRNA-146a-mediated\n",
      "Conunct merge check: IRAK-1 up-regulation\n",
      "IRAK-1\n",
      "Conjunctions ( nmod )\n",
      "IRAK-1 [regulation, of, IRAK-1, in, AD]\n",
      "IRAK-1 [down-regulation, of, IRAK-1]\n",
      "up-regulation [coupled, to, NF-B-induced, up-regulation, of, IRAK-2, expression]\n",
      "SDP sent: These data suggest a significantly independent regulation of IRAK-1 and IRAK-2 in AD and in IL-1+A42 peptide-stressed HAG cells and that an inducible, NF-B-sensitive, miRNA-146a-mediated down-regulation of IRAK-1 coupled to an NF-B-induced up-regulation of IRAK-2 expression drives an extensively sustained inflammatory response.\n",
      "SDP     : [(miRNA-146a-mediated, 'ADJ', 'dep', False), (inducible, 'ADJ', 'amod', False), (down-regulation, 'NOUN', 'nsubj', False), (coupled, 'VERB', 'acl', False), (up-regulation, 'NOUN', 'nmod', False), (expression, 'NOUN', 'nmod', False), (IRAK-2, 'NOUN', 'compound', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[IRAK-2]\n",
      "DeepCheck Conj Subtree\n",
      "[in, IL-1+A42, peptide-stressed, HAG, cells]\n",
      "0 [These, data, suggest, a, significantly, independent, regulation, of, IRAK-1, and, IRAK-2, in, AD, and, in, IL-1+A42, peptide-stressed, HAG, cells] 19 0 These data suggest a significantly independent regulation of IRAK-1 and IRAK-2 in AD and in IL-1+A42 peptide-stressed HAG cells and that an inducible, NF-B-sensitive, miRNA-146a-mediated down-regulation of IRAK-1 coupled to an NF-B-induced up-regulation of IRAK-2 expression drives an extensively sustained inflammatory response.\n",
      "1 [an, inducible, ,, NF-B-sensitive, ,, miRNA-146a-mediated, down-regulation, of, IRAK-1, coupled, to, an, NF-B-induced, up-regulation, of, IRAK-2, expression, drives, an, extensively, sustained, inflammatory, response] 23 139 an inducible, NF-B-sensitive, miRNA-146a-mediated down-regulation of IRAK-1 coupled to an NF-B-induced up-regulation of IRAK-2 expression drives an extensively sustained inflammatory response.\n",
      "DeepCheck Conj Subtree\n",
      "[of, IRAK-1, and, IRAK-2]\n",
      "DeepCheck Conj Subtree\n",
      "[IRAK-2]\n",
      "DeepCheck Conj Subtree\n",
      "[in, AD, and, in, IL-1+A42, peptide-stressed, HAG, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[in, IL-1+A42, peptide-stressed, HAG, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[of, IRAK-1]\n",
      "DeepCheck Conj Subtree\n",
      "[to, an, NF-B-induced, up-regulation, of, IRAK-2, expression]\n",
      "DeepCheck Conj Subtree\n",
      "[of, IRAK-2, expression]\n",
      "0 [These, data, suggest, a, significantly, independent, regulation, of, IRAK-1, and, IRAK-2, in, AD, and, in, IL-1+A42, peptide-stressed, HAG, cells] 19 0 These data suggest a significantly independent regulation of IRAK-1 and IRAK-2 in AD and in IL-1+A42 peptide-stressed HAG cells and that an inducible, NF-B-sensitive, miRNA-146a-mediated down-regulation of IRAK-1 coupled to an NF-B-induced up-regulation of IRAK-2 expression drives an extensively sustained inflammatory response.\n",
      "1 [an, inducible, ,, NF-B-sensitive, ,, miRNA-146a-mediated, down-regulation, of, IRAK-1, coupled, to, an, NF-B-induced, up-regulation, of, IRAK-2, expression, drives, an, extensively, sustained, inflammatory, response] 23 139 an inducible, NF-B-sensitive, miRNA-146a-mediated down-regulation of IRAK-1 coupled to an NF-B-induced up-regulation of IRAK-2 expression drives an extensively sustained inflammatory response.\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter({'POS': 2, 'NEG': 1})\n",
      "Compartment Text an inducible , NF-B-sensitive , miRNA-146a-mediated down-regulation of IRAK-1 coupled to an NF-B-induced up-regulation of IRAK-2 expression drives an extensively sustained inflammatory response\n",
      "[an, inducible, ,, NF-B-sensitive, ,, miRNA-146a-mediated, down-regulation, of, IRAK-1, coupled, to, an, NF-B-induced, up-regulation, of, IRAK-2, expression, drives, an, extensively, sustained, inflammatory, response]\n",
      "TextBetween: an inducible , NF-B-sensitive , miRNA-146a-mediated down-regulation of IRAK-1 coupled to an NF-B-induced up-regulation of IRAK-2 expression drives an extensively sustained inflammatory response\n",
      "infer direction from token set [an, inducible, ,, NF-B-sensitive, ,, miRNA-146a-mediated, down-regulation, of, IRAK-1, coupled, to, an, NF-B-induced, up-regulation, of, IRAK-2, expression, drives, an, extensively, sustained, inflammatory, response]\n",
      "stemTypes2Count: Counter({'POS': 2, 'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70730>, {'mir': Counter({'NEG': 1}), 'gene': Counter({'POS': 1})})\n",
      "defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70730>, {'mir': Counter({'NEG': 1}), 'gene': Counter({'POS': 1})})\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}})\n",
      " hit sentence These data suggest a significantly independent regulation of IRAK-1 and IRAK-2 in AD and in IL-1+A42 peptide-stressed HAG cells and that an inducible, NF-B-sensitive, miRNA-146a-mediated down-regulation of IRAK-1 coupled to an NF-B-induced up-regulation of IRAK-2 expression drives an extensively sustained inflammatory response.\n",
      "full sentence These data suggest a significantly independent regulation of IRAK-1 and IRAK-2 in AD and in IL-1+A42 peptide-stressed HAG cells and that an inducible, NF-B-sensitive, <e1>miRNA-146a</e1>-mediated down-regulation of IRAK-1 coupled to an NF-B-induced up-regulation of <e2>IRAK-2</e2> expression drives an extensively sustained inflammatory response.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'These data suggest a significantly independent regulation of IRAK-1 and IRAK-2 in AD and in IL-1+A42 peptide-stressed HAG cells and that an inducible, NF-B-sensitive, <e1>miRNA-146a</e1>-mediated down-regulation of IRAK-1 coupled to an NF-B-induced up-regulation of <e2>IRAK-2</e2> expression drives an extensively sustained inflammatory response.', 'full_sentence': 'These data suggest a significantly independent regulation of IRAK-1 and IRAK-2 in AD and in IL-1+A42 peptide-stressed HAG cells and that an inducible, NF-B-sensitive, <e1>miRNA-146a</e1>-mediated down-regulation of IRAK-1 coupled to an NF-B-induced up-regulation of <e2>IRAK-2</e2> expression drives an extensively sustained inflammatory response.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (170, 180), 'entity_text': 'miRNA-146a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (261, 267), 'entity_text': 'IRAK-2'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d181.s11.p0\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case miRNA-146a\n",
      "add special case IRAK\n",
      "miRNA-146a 6 IRAK 20\n",
      "(u\"The interactive signaling of NF-B and miRNA-146a further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain IRAK expression.\", 6, 20)\n",
      "geneword IRAK mirword miRNA-146a\n",
      "Conjuncts [miRNA-146a, NF-B]\n",
      "neighbour miRNA-146a further\n",
      "neighbour miRNA-146a further\n",
      "Conjunction before fc NF-B\n",
      "fc add children for NF-B\n",
      "Conjuncts [NF-B, miRNA-146a]\n",
      "neighbour NF-B and\n",
      "neighbour NF-B and\n",
      "Conjunction before fc miRNA-146a\n",
      "Conjuncts [miRNAs, factors]\n",
      "neighbour miRNAs that\n",
      "neighbour miRNAs that\n",
      "Conjunction before fc factors\n",
      "fc add children for factors\n",
      "fc add children for factors\n",
      "fc add children for factors\n",
      "Conjuncts [factors, miRNAs]\n",
      "neighbour factors and\n",
      "neighbour factors and\n",
      "Conjunction before fc miRNAs\n",
      "fc add children for miRNAs\n",
      "Conunct merge check: NF-B factors\n",
      "miRNA-146a further illustrate interplay\n",
      "Conjunctions ( conj )\n",
      "NF-B [of, NF-B, miRNA-146a]\n",
      "factors [between, inducible, transcription, factors, pro-inflammatory, miRNAs]\n",
      "Conjuncts [signaling, NF-B]\n",
      "neighbour signaling of\n",
      "neighbour signaling of\n",
      "Conjunction before fc NF-B\n",
      "fc add children for NF-B\n",
      "Conjuncts [interplay, factors]\n",
      "neighbour interplay between\n",
      "neighbour interplay between\n",
      "Conjunction before fc factors\n",
      "fc add children for factors\n",
      "fc add children for factors\n",
      "fc add children for factors\n",
      "Conunct merge check: NF-B factors\n",
      "NF-B and miRNA-146a further illustrate\n",
      "Conjunctions ( nmod )\n",
      "NF-B [signaling, of, NF-B]\n",
      "factors [interplay, between, inducible, transcription, factors]\n",
      "SDP sent: The interactive signaling of NF-B and miRNA-146a further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain IRAK expression.\n",
      "SDP     : [(miRNA-146a, 'NOUN', 'conj', False), (NF-B, 'NOUN', 'nmod', True), (signaling, 'NOUN', 'nsubj', False), (illustrate, 'VERB', 'ROOT', False), (interplay, 'NOUN', 'dobj', False), (factors, 'NOUN', 'nmod', False), (regulate, 'VERB', 'acl:relcl', False), (expression, 'NOUN', 'dobj', False), (IRAK, 'NOUN', 'compound', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[miRNA-146a]\n",
      "DeepCheck Conj Subtree\n",
      "[pro-inflammatory, miRNAs]\n",
      "0 [The, interactive, signaling, of, NF-B, and, miRNA-146a, further, illustrate, interplay, between, inducible, transcription, factors, and, pro-inflammatory, miRNAs, that, regulate, brain, IRAK, expression] 22 0 The interactive signaling of NF-B and miRNA-146a further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain IRAK expression.\n",
      "DeepCheck Conj Subtree\n",
      "[of, NF-B, and, miRNA-146a]\n",
      "DeepCheck Conj Subtree\n",
      "[miRNA-146a]\n",
      "DeepCheck Conj Subtree\n",
      "[between, inducible, transcription, factors, and, pro-inflammatory, miRNAs, that, regulate, brain, IRAK, expression]\n",
      "DeepCheck Conj Subtree\n",
      "[pro-inflammatory, miRNAs]\n",
      "0 [The, interactive, signaling, of, NF-B, and, miRNA-146a, further, illustrate, interplay, between, inducible, transcription, factors, and, pro-inflammatory, miRNAs, that, regulate, brain, IRAK, expression] 22 0 The interactive signaling of NF-B and miRNA-146a further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain IRAK expression.\n",
      "Recognized overlapping entities: {Interval(161, 171, ('CELL', 'brain IRAK'))}\n",
      "Gene reject: {'CELL'} due to {'CELL'}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text The interactive signaling of NF-B and miRNA-146a further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain IRAK expression\n",
      "[The, interactive, signaling, of, NF-B, and, miRNA-146a, further, illustrate, interplay, between, inducible, transcription, factors, and, pro-inflammatory, miRNAs, that, regulate, brain, IRAK, expression]\n",
      "TextBetween: The interactive signaling of NF-B and miRNA-146a further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain IRAK expression\n",
      "infer direction from token set [The, interactive, signaling, of, NF-B, and, miRNA-146a, further, illustrate, interplay, between, inducible, transcription, factors, and, pro-inflammatory, miRNAs, that, regulate, brain, IRAK, expression]\n",
      "stemTypes2Count: Counter({'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70488>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 3, 'POS': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fc0b806d048>, {'gene': Counter({'NEU': 2})})\n",
      "counts between 1 stemTypes2Count [(inducible, 'POS'), (transcription, 'NEU'), (regulate, 'NEU'), (expression, 'NEU')]\n",
      "Between Tokens [(inducible, 'POS'), (transcription, 'NEU'), (regulate, 'NEU')]\n",
      "counts between final False\n",
      "['NF-B', 'and', 'miRNA-146a', 'further', 'illustrate', 'interplay', 'between', 'inducible', 'transcription', 'factors', 'and', 'pro-inflammatory', 'miRNAs', 'that', 'regulate', 'brain', 'IRAK', 'expression']\n",
      "Counter({'NEU': 3, 'POS': 1})\n",
      "[(inducible, 'POS'), (transcription, 'NEU'), (regulate, 'NEU'), (expression, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': False, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence The interactive signaling of NF-B and miRNA-146a further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain IRAK expression.\n",
      "full sentence The interactive signaling of NF-B and <e1>miRNA-146a</e1> further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain <e2>IRAK</e2> expression.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': False, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'The interactive signaling of NF-B and <e1>miRNA-146a</e1> further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain <e2>IRAK</e2> expression.', 'full_sentence': 'The interactive signaling of NF-B and <e1>miRNA-146a</e1> further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain <e2>IRAK</e2> expression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (39, 49), 'entity_text': 'miRNA-146a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (167, 171), 'entity_text': 'IRAK'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d181.s11.p0\n",
      "IS:  ('MIR_GENE', 'UP')\n",
      "SB:  ('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': False, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'The interactive signaling of NF-B and <e1>miRNA-146a</e1> further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain <e2>IRAK</e2> expression.', 'full_sentence': 'The interactive signaling of NF-B and <e1>miRNA-146a</e1> further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain <e2>IRAK</e2> expression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (39, 49), 'entity_text': 'miRNA-146a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (167, 171), 'entity_text': 'IRAK'}}\n",
      "add special case miRNA-146a\n",
      "add special case IRAK\n",
      "miRNA-146a 6 IRAK 20\n",
      "(u\"The interactive signaling of NF-B and miRNA-146a further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain IRAK expression.\", 6, 20)\n",
      "geneword IRAK mirword miRNA-146a\n",
      "Conjuncts [miRNA-146a, NF-B]\n",
      "neighbour miRNA-146a further\n",
      "neighbour miRNA-146a further\n",
      "Conjunction before fc NF-B\n",
      "fc add children for NF-B\n",
      "Conjuncts [NF-B, miRNA-146a]\n",
      "neighbour NF-B and\n",
      "neighbour NF-B and\n",
      "Conjunction before fc miRNA-146a\n",
      "Conjuncts [miRNAs, factors]\n",
      "neighbour miRNAs that\n",
      "neighbour miRNAs that\n",
      "Conjunction before fc factors\n",
      "fc add children for factors\n",
      "fc add children for factors\n",
      "fc add children for factors\n",
      "Conjuncts [factors, miRNAs]\n",
      "neighbour factors and\n",
      "neighbour factors and\n",
      "Conjunction before fc miRNAs\n",
      "fc add children for miRNAs\n",
      "Conunct merge check: NF-B factors\n",
      "miRNA-146a further illustrate interplay\n",
      "Conjunctions ( conj )\n",
      "NF-B [of, NF-B, miRNA-146a]\n",
      "factors [between, inducible, transcription, factors, pro-inflammatory, miRNAs]\n",
      "Conjuncts [signaling, NF-B]\n",
      "neighbour signaling of\n",
      "neighbour signaling of\n",
      "Conjunction before fc NF-B\n",
      "fc add children for NF-B\n",
      "Conjuncts [interplay, factors]\n",
      "neighbour interplay between\n",
      "neighbour interplay between\n",
      "Conjunction before fc factors\n",
      "fc add children for factors\n",
      "fc add children for factors\n",
      "fc add children for factors\n",
      "Conunct merge check: NF-B factors\n",
      "NF-B and miRNA-146a further illustrate\n",
      "Conjunctions ( nmod )\n",
      "NF-B [signaling, of, NF-B]\n",
      "factors [interplay, between, inducible, transcription, factors]\n",
      "SDP sent: The interactive signaling of NF-B and miRNA-146a further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain IRAK expression.\n",
      "SDP     : [(miRNA-146a, 'NOUN', 'conj', False), (NF-B, 'NOUN', 'nmod', True), (signaling, 'NOUN', 'nsubj', False), (illustrate, 'VERB', 'ROOT', False), (interplay, 'NOUN', 'dobj', False), (factors, 'NOUN', 'nmod', False), (regulate, 'VERB', 'acl:relcl', False), (expression, 'NOUN', 'dobj', False), (IRAK, 'NOUN', 'compound', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[miRNA-146a]\n",
      "DeepCheck Conj Subtree\n",
      "[pro-inflammatory, miRNAs]\n",
      "0 [The, interactive, signaling, of, NF-B, and, miRNA-146a, further, illustrate, interplay, between, inducible, transcription, factors, and, pro-inflammatory, miRNAs, that, regulate, brain, IRAK, expression] 22 0 The interactive signaling of NF-B and miRNA-146a further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain IRAK expression.\n",
      "DeepCheck Conj Subtree\n",
      "[of, NF-B, and, miRNA-146a]\n",
      "DeepCheck Conj Subtree\n",
      "[miRNA-146a]\n",
      "DeepCheck Conj Subtree\n",
      "[between, inducible, transcription, factors, and, pro-inflammatory, miRNAs, that, regulate, brain, IRAK, expression]\n",
      "DeepCheck Conj Subtree\n",
      "[pro-inflammatory, miRNAs]\n",
      "0 [The, interactive, signaling, of, NF-B, and, miRNA-146a, further, illustrate, interplay, between, inducible, transcription, factors, and, pro-inflammatory, miRNAs, that, regulate, brain, IRAK, expression] 22 0 The interactive signaling of NF-B and miRNA-146a further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain IRAK expression.\n",
      "Recognized overlapping entities: {Interval(161, 171, ('CELL', 'brain IRAK'))}\n",
      "Gene reject: {'CELL'} due to {'CELL'}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text The interactive signaling of NF-B and miRNA-146a further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain IRAK expression\n",
      "[The, interactive, signaling, of, NF-B, and, miRNA-146a, further, illustrate, interplay, between, inducible, transcription, factors, and, pro-inflammatory, miRNAs, that, regulate, brain, IRAK, expression]\n",
      "TextBetween: The interactive signaling of NF-B and miRNA-146a further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain IRAK expression\n",
      "infer direction from token set [The, interactive, signaling, of, NF-B, and, miRNA-146a, further, illustrate, interplay, between, inducible, transcription, factors, and, pro-inflammatory, miRNAs, that, regulate, brain, IRAK, expression]\n",
      "stemTypes2Count: Counter({'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70730>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 3, 'POS': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8cbf8>, {'gene': Counter({'NEU': 2})})\n",
      "counts between 1 stemTypes2Count [(inducible, 'POS'), (transcription, 'NEU'), (regulate, 'NEU'), (expression, 'NEU')]\n",
      "Between Tokens [(inducible, 'POS'), (transcription, 'NEU'), (regulate, 'NEU')]\n",
      "counts between final False\n",
      "['NF-B', 'and', 'miRNA-146a', 'further', 'illustrate', 'interplay', 'between', 'inducible', 'transcription', 'factors', 'and', 'pro-inflammatory', 'miRNAs', 'that', 'regulate', 'brain', 'IRAK', 'expression']\n",
      "Counter({'NEU': 3, 'POS': 1})\n",
      "[(inducible, 'POS'), (transcription, 'NEU'), (regulate, 'NEU'), (expression, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': False, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence The interactive signaling of NF-B and miRNA-146a further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain IRAK expression.\n",
      "full sentence The interactive signaling of NF-B and <e1>miRNA-146a</e1> further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain <e2>IRAK</e2> expression.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': False, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'The interactive signaling of NF-B and <e1>miRNA-146a</e1> further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain <e2>IRAK</e2> expression.', 'full_sentence': 'The interactive signaling of NF-B and <e1>miRNA-146a</e1> further illustrate interplay between inducible transcription factors and pro-inflammatory miRNAs that regulate brain <e2>IRAK</e2> expression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (39, 49), 'entity_text': 'miRNA-146a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (167, 171), 'entity_text': 'IRAK'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d182.s5.p0\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'compartment mir gene neg corr', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'We show that the expression of <e1>miR-34a</e1> is inversely correlated with the protein level of <e2>bcl2</e2> in APPswe/PSDeltaE9 mice and age-matched controls, and miR-34a expression directly inhibits bcl2 translation in SH-SY5Y cells.', 'full_sentence': 'We show that the expression of <e1>miR-34a</e1> is inversely correlated with the protein level of <e2>bcl2</e2> in APPswe/PSDeltaE9 mice and age-matched controls, and miR-34a expression directly inhibits bcl2 translation in SH-SY5Y cells.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (31, 38), 'entity_text': 'miR-34a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (89, 93), 'entity_text': 'bcl2'}}\n",
      "add special case miR-34a\n",
      "add special case bcl2\n",
      "miR-34a 6 bcl2 15\n",
      "(u\"We show that the expression of miR-34a is inversely correlated with the protein level of bcl2 in APPswe/PSDeltaE9 mice and age-matched controls, and miR-34a expression directly inhibits bcl2 translation in SH-SY5Y cells.\", 6, 15)\n",
      "geneword bcl2 mirword miR-34a\n",
      "Conjuncts [inhibits, correlated]\n",
      "neighbour inhibits bcl2\n",
      "neighbour inhibits bcl2\n",
      "Conjunction before fc correlated\n",
      "fc add children for correlated\n",
      "Conjuncts [controls, mice]\n",
      "neighbour controls ,\n",
      "neighbour controls and\n",
      "Conjunction before fc mice\n",
      "fc add children for mice\n",
      "fc add children for mice\n",
      "Conjuncts [mice, controls]\n",
      "neighbour mice and\n",
      "neighbour mice and\n",
      "Conjunction before fc controls\n",
      "fc add children for controls\n",
      "Conjuncts [correlated, inhibits]\n",
      "neighbour correlated with\n",
      "neighbour correlated with\n",
      "Conjunction before fc inhibits\n",
      "fc add children for inhibits\n",
      "Conjunctions ( conj )\n",
      "correlated [correlated, with, protein, level, of, bcl2, in, APPswe/PSDeltaE9, mice, age-matched, controls, inhibits, bcl2, translation, in, SH-SY5Y, cells]\n",
      "Conjuncts [expression, miR-34a]\n",
      "neighbour expression of\n",
      "neighbour expression of\n",
      "Conjunction before fc miR-34a\n",
      "fc add children for miR-34a\n",
      "Conjuncts [correlated, level]\n",
      "neighbour correlated with\n",
      "neighbour correlated with\n",
      "Conjunction before fc level\n",
      "fc add children for level\n",
      "fc add children for level\n",
      "fc add children for level\n",
      "fc add children for level\n",
      "Conjuncts [level, bcl2]\n",
      "neighbour level of\n",
      "neighbour level of\n",
      "Conjunction before fc bcl2\n",
      "fc add children for bcl2\n",
      "Conjuncts [level, mice]\n",
      "neighbour level of\n",
      "neighbour level of\n",
      "Conjunction before fc mice\n",
      "fc add children for mice\n",
      "fc add children for mice\n",
      "Conjuncts [translation, cells]\n",
      "neighbour translation in\n",
      "neighbour translation in\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conunct merge check: miR-34a level\n",
      "miR-34a is inversely\n",
      "Conunct merge check: level cells\n",
      "mice and age-matched controls, and miR-34a expression directly inhibits bcl2\n",
      "Conjunctions ( nmod )\n",
      "miR-34a [expression, of, miR-34a]\n",
      "level [correlated, with, protein, level, of, bcl2, in, APPswe/PSDeltaE9, mice]\n",
      "cells [translation, in, SH-SY5Y, cells]\n",
      "SDP sent: We show that the expression of miR-34a is inversely correlated with the protein level of bcl2 in APPswe/PSDeltaE9 mice and age-matched controls, and miR-34a expression directly inhibits bcl2 translation in SH-SY5Y cells.\n",
      "SDP     : [(miR-34a, 'NOUN', 'nmod', False), (expression, 'NOUN', 'nsubjpass', False), (correlated, 'VERB', 'ccomp', False), (level, 'NOUN', 'nmod', False), (bcl2, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[age-matched, controls]\n",
      "deepcheck connection rule noun found inhibits miR-34a [miR-34a, expression, directly, inhibits, bcl2, translation, in, SH-SY5Y, cells]\n",
      "0 [We, show] 2 0 We show that the expression of miR-34a is inversely correlated with the protein level of bcl2 in APPswe/PSDeltaE9 mice and age-matched controls, and miR-34a expression directly inhibits bcl2 translation in SH-SY5Y cells.\n",
      "1 [the, expression, of, miR-34a, is, inversely, correlated, with, the, protein, level, of, bcl2, in, APPswe/PSDeltaE9, mice, and, age-matched, controls] 19 13 the expression of miR-34a is inversely correlated with the protein level of bcl2 in APPswe/PSDeltaE9 mice and age-matched controls, and miR-34a expression directly inhibits bcl2 translation in SH-SY5Y cells.\n",
      "2 [miR-34a, expression, directly, inhibits, bcl2, translation, in, SH-SY5Y, cells] 9 149 miR-34a expression directly inhibits bcl2 translation in SH-SY5Y cells.\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-34a]\n",
      "DeepCheck Conj Subtree\n",
      "[with, the, protein, level, of, bcl2, in, APPswe/PSDeltaE9, mice, and, age-matched, controls]\n",
      "DeepCheck Conj Subtree\n",
      "[of, bcl2]\n",
      "DeepCheck Conj Subtree\n",
      "[in, APPswe/PSDeltaE9, mice, and, age-matched, controls]\n",
      "DeepCheck Conj Subtree\n",
      "[age-matched, controls]\n",
      "deepcheck connection rule noun found inhibits miR-34a [miR-34a, expression, directly, inhibits, bcl2, translation, in, SH-SY5Y, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[in, SH-SY5Y, cells]\n",
      "0 [We, show] 2 0 We show that the expression of miR-34a is inversely correlated with the protein level of bcl2 in APPswe/PSDeltaE9 mice and age-matched controls, and miR-34a expression directly inhibits bcl2 translation in SH-SY5Y cells.\n",
      "1 [the, expression, of, miR-34a, is, inversely, correlated, with, the, protein, level, of, bcl2, in, APPswe/PSDeltaE9, mice, and, age-matched, controls] 19 13 the expression of miR-34a is inversely correlated with the protein level of bcl2 in APPswe/PSDeltaE9 mice and age-matched controls, and miR-34a expression directly inhibits bcl2 translation in SH-SY5Y cells.\n",
      "2 [miR-34a, expression, directly, inhibits, bcl2, translation, in, SH-SY5Y, cells] 9 149 miR-34a expression directly inhibits bcl2 translation in SH-SY5Y cells.\n",
      "Recognized overlapping entities: {Interval(89, 93, ('GENE_OR_GENE_PRODUCT', 'bcl2'))}\n",
      "passive? True\n",
      "Counter()\n",
      "Compartment Text the expression of miR-34a is inversely correlated with the protein level of bcl2 in APPswe/PSDeltaE9 mice and age-matched controls\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'compartment mir gene neg corr', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence We show that the expression of miR-34a is inversely correlated with the protein level of bcl2 in APPswe/PSDeltaE9 mice and age-matched controls, and miR-34a expression directly inhibits bcl2 translation in SH-SY5Y cells.\n",
      "full sentence We show that the expression of <e1>miR-34a</e1> is inversely correlated with the protein level of <e2>bcl2</e2> in APPswe/PSDeltaE9 mice and age-matched controls, and miR-34a expression directly inhibits bcl2 translation in SH-SY5Y cells.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'compartment mir gene neg corr', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'We show that the expression of <e1>miR-34a</e1> is inversely correlated with the protein level of <e2>bcl2</e2> in APPswe/PSDeltaE9 mice and age-matched controls, and miR-34a expression directly inhibits bcl2 translation in SH-SY5Y cells.', 'full_sentence': 'We show that the expression of <e1>miR-34a</e1> is inversely correlated with the protein level of <e2>bcl2</e2> in APPswe/PSDeltaE9 mice and age-matched controls, and miR-34a expression directly inhibits bcl2 translation in SH-SY5Y cells.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (31, 38), 'entity_text': 'miR-34a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (89, 93), 'entity_text': 'bcl2'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d182.s8.p2\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miR-34a\n",
      "add special case bcl2\n",
      "miR-34a 11 bcl2 38\n",
      "(u\"These findings suggested that bcl2 is an important functional target for miR-34a, and the abnormal expression of miR-34a may contribute to the pathogenesis of Alzheimer's disease, at least in part by affecting the expression of bcl2.\", 11, 38)\n",
      "geneword bcl2 mirword miR-34a\n",
      "Conjuncts [contribute, target]\n",
      "neighbour contribute to\n",
      "neighbour contribute to\n",
      "Conjunction before fc target\n",
      "fc add children for target\n",
      "fc add children for target\n",
      "fc add children for target\n",
      "fc add children for target\n",
      "Conjuncts [target, contribute]\n",
      "neighbour target for\n",
      "neighbour target for\n",
      "Conjunction before fc contribute\n",
      "fc add children for contribute\n",
      "Conjunctions ( conj )\n",
      "target [important, functional, target, for, miR-34a, contribute, to, pathogenesis, of, 's, disease, in, part]\n",
      "Conjuncts [target, miR-34a]\n",
      "neighbour target for\n",
      "neighbour target for\n",
      "Conjunction before fc miR-34a\n",
      "fc add children for miR-34a\n",
      "Conjuncts [expression, miR-34a]\n",
      "neighbour expression of\n",
      "neighbour expression of\n",
      "Conjunction before fc miR-34a\n",
      "fc add children for miR-34a\n",
      "Conjuncts [contribute, pathogenesis]\n",
      "neighbour contribute to\n",
      "neighbour contribute to\n",
      "Conjunction before fc pathogenesis\n",
      "fc add children for pathogenesis\n",
      "fc add children for pathogenesis\n",
      "Conjuncts [pathogenesis, disease]\n",
      "neighbour pathogenesis of\n",
      "neighbour pathogenesis of\n",
      "Conjunction before fc disease\n",
      "fc add children for disease\n",
      "fc add children for disease\n",
      "Conjuncts [target, part]\n",
      "neighbour target for\n",
      "neighbour target for\n",
      "Conjunction before fc part\n",
      "fc add children for part\n",
      "Conjuncts [expression, bcl2]\n",
      "neighbour expression of\n",
      "neighbour expression of\n",
      "Conjunction before fc bcl2\n",
      "fc add children for bcl2\n",
      "Conunct merge check: miR-34a miR-34a\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: miR-34a pathogenesis\n",
      "miR-34a may\n",
      "Conunct merge check: pathogenesis bcl2\n",
      "disease, at least in part by affecting the\n",
      "Conjunctions ( nmod )\n",
      "miR-34a [target, for, miR-34a, in, part]\n",
      "miR-34a [expression, of, miR-34a]\n",
      "pathogenesis [contribute, to, pathogenesis, of, 's, disease]\n",
      "bcl2 [expression, of, bcl2]\n",
      "SDP sent: These findings suggested that bcl2 is an important functional target for miR-34a, and the abnormal expression of miR-34a may contribute to the pathogenesis of Alzheimer's disease, at least in part by affecting the expression of bcl2.\n",
      "SDP     : [(miR-34a, 'NOUN', 'nmod', False), (target, 'NOUN', 'ccomp', False), (affecting, 'VERB', 'advcl', False), (expression, 'NOUN', 'dobj', False), (bcl2, 'NOUN', 'nmod', False)]\n",
      "deepcheck connection rule [by, affecting, the, expression, of, bcl2]\n",
      "0 [These, findings, suggested, that, bcl2, is, an, important, functional, target, for, miR-34a, ,, and, ,, at, least, in, part, by, affecting, the, expression, of, bcl2] 25 0 These findings suggested that bcl2 is an important functional target for miR-34a, and the abnormal expression of miR-34a may contribute to the pathogenesis of Alzheimer's disease, at least in part by affecting the expression of bcl2.\n",
      "1 [the, abnormal, expression, of, miR-34a, may, contribute, to, the, pathogenesis, of, Alzheimer, 's, disease] 14 86 the abnormal expression of miR-34a may contribute to the pathogenesis of Alzheimer's disease, at least in part by affecting the expression of bcl2.\n",
      "DeepCheck Conj Subtree\n",
      "[for, miR-34a]\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-34a]\n",
      "DeepCheck Conj Subtree\n",
      "[to, the, pathogenesis, of, Alzheimer, 's, disease]\n",
      "DeepCheck Conj Subtree\n",
      "[of, Alzheimer, 's, disease]\n",
      "DeepCheck Conj Subtree\n",
      "[at, least, in, part]\n",
      "deepcheck connection rule [by, affecting, the, expression, of, bcl2]\n",
      "DeepCheck Conj Subtree\n",
      "[of, bcl2]\n",
      "0 [These, findings, suggested, that, bcl2, is, an, important, functional, target, for, miR-34a, ,, and, ,, at, least, in, part, by, affecting, the, expression, of, bcl2] 25 0 These findings suggested that bcl2 is an important functional target for miR-34a, and the abnormal expression of miR-34a may contribute to the pathogenesis of Alzheimer's disease, at least in part by affecting the expression of bcl2.\n",
      "1 [the, abnormal, expression, of, miR-34a, may, contribute, to, the, pathogenesis, of, Alzheimer, 's, disease] 14 86 the abnormal expression of miR-34a may contribute to the pathogenesis of Alzheimer's disease, at least in part by affecting the expression of bcl2.\n",
      "Recognized overlapping entities: {Interval(228, 232, ('GENE_OR_GENE_PRODUCT', 'bcl2'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text These findings suggested that bcl2 is an important functional target for miR-34a , and , at least in part by affecting the expression of bcl2\n",
      "[These, findings, suggested, that, bcl2, is, an, important, functional, target, for, miR-34a, ,, and, ,, at, least, in, part, by, affecting, the, expression, of, bcl2]\n",
      "TextBetween: These findings suggested that bcl2 is an important functional target for miR-34a , and , at least in part by affecting the expression of bcl2\n",
      "infer direction from token set [These, findings, suggested, that, bcl2, is, an, important, functional, target, for, miR-34a, ,, and, ,, at, least, in, part, by, affecting, the, expression, of, bcl2]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954d90>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 3})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fc0bbbd3158>, {'mir': Counter({'NEU': 1}), 'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(target, 'NEU'), (affecting, 'NEU'), (expression, 'NEU')]\n",
      "Between Tokens [(affecting, 'NEU'), (expression, 'NEU')]\n",
      "counts between final False\n",
      "['target', 'for', 'miR-34a', ',', 'and', ',', 'at', 'least', 'in', 'part', 'by', 'affecting', 'the', 'expression', 'of', 'bcl2']\n",
      "Counter({'NEU': 3})\n",
      "[(target, 'NEU'), (affecting, 'NEU'), (expression, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence These findings suggested that bcl2 is an important functional target for miR-34a, and the abnormal expression of miR-34a may contribute to the pathogenesis of Alzheimer's disease, at least in part by affecting the expression of bcl2.\n",
      "full sentence These findings suggested that bcl2 is an important functional target for <e1>miR-34a</e1>, and the abnormal expression of miR-34a may contribute to the pathogenesis of Alzheimer's disease, at least in part by affecting the expression of <e2>bcl2</e2>.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"These findings suggested that bcl2 is an important functional target for <e1>miR-34a</e1>, and the abnormal expression of miR-34a may contribute to the pathogenesis of Alzheimer's disease, at least in part by affecting the expression of <e2>bcl2</e2>.\", 'full_sentence': \"These findings suggested that bcl2 is an important functional target for <e1>miR-34a</e1>, and the abnormal expression of miR-34a may contribute to the pathogenesis of Alzheimer's disease, at least in part by affecting the expression of <e2>bcl2</e2>.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (73, 80), 'entity_text': 'miR-34a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (228, 232), 'entity_text': 'bcl2'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d182.s8.p5\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case miR-34a\n",
      "add special case bcl2\n",
      "miR-34a 18 bcl2 38\n",
      "(u\"These findings suggested that bcl2 is an important functional target for miR-34a, and the abnormal expression of miR-34a may contribute to the pathogenesis of Alzheimer's disease, at least in part by affecting the expression of bcl2.\", 18, 38)\n",
      "geneword bcl2 mirword miR-34a\n",
      "Conjuncts [contribute, target]\n",
      "neighbour contribute to\n",
      "neighbour contribute to\n",
      "Conjunction before fc target\n",
      "fc add children for target\n",
      "fc add children for target\n",
      "fc add children for target\n",
      "fc add children for target\n",
      "Conjuncts [target, contribute]\n",
      "neighbour target for\n",
      "neighbour target for\n",
      "Conjunction before fc contribute\n",
      "fc add children for contribute\n",
      "Conjunctions ( conj )\n",
      "target [important, functional, target, for, miR-34a, contribute, to, pathogenesis, of, 's, disease, in, part]\n",
      "Conjuncts [target, miR-34a]\n",
      "neighbour target for\n",
      "neighbour target for\n",
      "Conjunction before fc miR-34a\n",
      "fc add children for miR-34a\n",
      "Conjuncts [expression, miR-34a]\n",
      "neighbour expression of\n",
      "neighbour expression of\n",
      "Conjunction before fc miR-34a\n",
      "fc add children for miR-34a\n",
      "Conjuncts [contribute, pathogenesis]\n",
      "neighbour contribute to\n",
      "neighbour contribute to\n",
      "Conjunction before fc pathogenesis\n",
      "fc add children for pathogenesis\n",
      "fc add children for pathogenesis\n",
      "Conjuncts [pathogenesis, disease]\n",
      "neighbour pathogenesis of\n",
      "neighbour pathogenesis of\n",
      "Conjunction before fc disease\n",
      "fc add children for disease\n",
      "fc add children for disease\n",
      "Conjuncts [target, part]\n",
      "neighbour target for\n",
      "neighbour target for\n",
      "Conjunction before fc part\n",
      "fc add children for part\n",
      "Conjuncts [expression, bcl2]\n",
      "neighbour expression of\n",
      "neighbour expression of\n",
      "Conjunction before fc bcl2\n",
      "fc add children for bcl2\n",
      "Conunct merge check: miR-34a miR-34a\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: miR-34a pathogenesis\n",
      "miR-34a may\n",
      "Conunct merge check: pathogenesis bcl2\n",
      "disease, at least in part by affecting the\n",
      "Conjunctions ( nmod )\n",
      "miR-34a [target, for, miR-34a, in, part]\n",
      "miR-34a [expression, of, miR-34a]\n",
      "pathogenesis [contribute, to, pathogenesis, of, 's, disease]\n",
      "bcl2 [expression, of, bcl2]\n",
      "SDP sent: These findings suggested that bcl2 is an important functional target for miR-34a, and the abnormal expression of miR-34a may contribute to the pathogenesis of Alzheimer's disease, at least in part by affecting the expression of bcl2.\n",
      "SDP     : [(miR-34a, 'NOUN', 'nmod', False), (expression, 'NOUN', 'nsubj', False), (contribute, 'VERB', 'conj', False), (target, 'NOUN', 'ccomp', True), (affecting, 'VERB', 'advcl', False), (expression, 'NOUN', 'dobj', False), (bcl2, 'NOUN', 'nmod', False)]\n",
      "deepcheck connection rule [by, affecting, the, expression, of, bcl2]\n",
      "0 [These, findings, suggested, that, bcl2, is, an, important, functional, target, for, miR-34a, ,, and, ,, at, least, in, part, by, affecting, the, expression, of, bcl2] 25 0 These findings suggested that bcl2 is an important functional target for miR-34a, and the abnormal expression of miR-34a may contribute to the pathogenesis of Alzheimer's disease, at least in part by affecting the expression of bcl2.\n",
      "1 [the, abnormal, expression, of, miR-34a, may, contribute, to, the, pathogenesis, of, Alzheimer, 's, disease] 14 86 the abnormal expression of miR-34a may contribute to the pathogenesis of Alzheimer's disease, at least in part by affecting the expression of bcl2.\n",
      "DeepCheck Conj Subtree\n",
      "[for, miR-34a]\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-34a]\n",
      "DeepCheck Conj Subtree\n",
      "[to, the, pathogenesis, of, Alzheimer, 's, disease]\n",
      "DeepCheck Conj Subtree\n",
      "[of, Alzheimer, 's, disease]\n",
      "DeepCheck Conj Subtree\n",
      "[at, least, in, part]\n",
      "deepcheck connection rule [by, affecting, the, expression, of, bcl2]\n",
      "DeepCheck Conj Subtree\n",
      "[of, bcl2]\n",
      "0 [These, findings, suggested, that, bcl2, is, an, important, functional, target, for, miR-34a, ,, and, ,, at, least, in, part, by, affecting, the, expression, of, bcl2] 25 0 These findings suggested that bcl2 is an important functional target for miR-34a, and the abnormal expression of miR-34a may contribute to the pathogenesis of Alzheimer's disease, at least in part by affecting the expression of bcl2.\n",
      "1 [the, abnormal, expression, of, miR-34a, may, contribute, to, the, pathogenesis, of, Alzheimer, 's, disease] 14 86 the abnormal expression of miR-34a may contribute to the pathogenesis of Alzheimer's disease, at least in part by affecting the expression of bcl2.\n",
      "Recognized overlapping entities: {Interval(228, 232, ('GENE_OR_GENE_PRODUCT', 'bcl2'))}\n",
      "passive? False\n",
      "Counter()\n",
      "These findings suggested that bcl2 is an important functional target for miR-34a, and the abnormal expression of miR-34a may contribute to the pathogenesis of Alzheimer's disease, at least in part by affecting the expression of bcl2.\n",
      "TextBetween: expression of miR-34a may contribute to the pathogenesis of Alzheimer 's disease , at least in part by affecting the expression of bcl2 .\n",
      "infer direction from token set [expression, of, miR-34a, may, contribute, to, the, pathogenesis, of, Alzheimer, 's, disease, ,, at, least, in, part, by, affecting, the, expression, of, bcl2, .]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70950>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 4})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fc0b806d158>, {'mir': Counter({'NEU': 2}), 'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(expression, 'NEU'), (contribute, 'NEU'), (affecting, 'NEU'), (expression, 'NEU')]\n",
      "Between Tokens [(contribute, 'NEU'), (affecting, 'NEU'), (expression, 'NEU')]\n",
      "counts between final False\n",
      "['expression', 'of', 'miR-34a', 'may', 'contribute', 'to', 'the', 'pathogenesis', 'of', 'Alzheimer', \"'s\", 'disease', ',', 'at', 'least', 'in', 'part', 'by', 'affecting', 'the', 'expression', 'of', 'bcl2', '.']\n",
      "Counter({'NEU': 4})\n",
      "[(expression, 'NEU'), (contribute, 'NEU'), (affecting, 'NEU'), (expression, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence These findings suggested that bcl2 is an important functional target for miR-34a, and the abnormal expression of miR-34a may contribute to the pathogenesis of Alzheimer's disease, at least in part by affecting the expression of bcl2.\n",
      "full sentence These findings suggested that bcl2 is an important functional target for miR-34a, and the abnormal expression of <e1>miR-34a</e1> may contribute to the pathogenesis of Alzheimer's disease, at least in part by affecting the expression of <e2>bcl2</e2>.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"These findings suggested that bcl2 is an important functional target for miR-34a, and the abnormal expression of <e1>miR-34a</e1> may contribute to the pathogenesis of Alzheimer's disease, at least in part by affecting the expression of <e2>bcl2</e2>.\", 'full_sentence': \"These findings suggested that bcl2 is an important functional target for miR-34a, and the abnormal expression of <e1>miR-34a</e1> may contribute to the pathogenesis of Alzheimer's disease, at least in part by affecting the expression of <e2>bcl2</e2>.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (113, 120), 'entity_text': 'miR-34a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (228, 232), 'entity_text': 'bcl2'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d187.s6.p0\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case let-7\n",
      "add special case LRRK2\n",
      "let-7 3 LRRK2 42\n",
      "(u\"Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2.\", 3, 42)\n",
      "geneword LRRK2 mirword let-7\n",
      "Conjuncts [miR-184, let-7]\n",
      "neighbour miR-184 *\n",
      "neighbour miR-184 action\n",
      "Conjunction before fc let-7\n",
      "fc add children for let-7\n",
      "Conjuncts [let-7, miR-184]\n",
      "neighbour let-7 and\n",
      "neighbour let-7 and\n",
      "Conjunction before fc miR-184\n",
      "Conjuncts [replacement, protector]\n",
      "neighbour replacement of\n",
      "neighbour replacement of\n",
      "Conjunction before fc protector\n",
      "fc add children for protector\n",
      "fc add children for protector\n",
      "fc add children for protector\n",
      "Conjuncts [protector, replacement]\n",
      "neighbour protector ,\n",
      "neighbour protector or\n",
      "Conjunction before fc replacement\n",
      "fc add children for replacement\n",
      "fc add children for replacement\n",
      "Conunct merge check: blockage protector\n",
      "miR-184* action, transgenic expression\n",
      "Conjunctions ( conj )\n",
      "blockage [antagomir-mediated, of, let-7, miR-184]\n",
      "protector [of, dp, target, protector, replacement, of, endogenous, dp, with, dp, transgene, non-responsive, to, let-7]\n",
      "Conjuncts [deletion, let-7]\n",
      "neighbour deletion of\n",
      "neighbour deletion of\n",
      "Conjunction before fc let-7\n",
      "fc add children for let-7\n",
      "fc add children for let-7\n",
      "Conjuncts [blockage, let-7]\n",
      "neighbour blockage of\n",
      "neighbour blockage of\n",
      "Conjunction before fc let-7\n",
      "fc add children for let-7\n",
      "Conjuncts [expression, protector]\n",
      "neighbour expression of\n",
      "neighbour expression of\n",
      "Conjunction before fc protector\n",
      "fc add children for protector\n",
      "fc add children for protector\n",
      "fc add children for protector\n",
      "Conjuncts [replacement, dp]\n",
      "neighbour replacement of\n",
      "neighbour replacement of\n",
      "Conjunction before fc dp\n",
      "fc add children for dp\n",
      "fc add children for dp\n",
      "Conjuncts [replacement, transgene]\n",
      "neighbour replacement of\n",
      "neighbour replacement of\n",
      "Conjunction before fc transgene\n",
      "fc add children for transgene\n",
      "fc add children for transgene\n",
      "fc add children for transgene\n",
      "Conjuncts [non-responsive, let-7]\n",
      "neighbour non-responsive to\n",
      "neighbour non-responsive to\n",
      "Conjunction before fc let-7\n",
      "fc add children for let-7\n",
      "Conjuncts [expression, each]\n",
      "neighbour expression of\n",
      "neighbour expression of\n",
      "Conjunction before fc each\n",
      "Conjuncts [similar, those]\n",
      "neighbour similar to\n",
      "neighbour similar to\n",
      "Conjunction before fc those\n",
      "fc add children for those\n",
      "fc add children for those\n",
      "Conjuncts [those, LRRK2]\n",
      "neighbour those of\n",
      "neighbour those of\n",
      "Conjunction before fc LRRK2\n",
      "fc add children for LRRK2\n",
      "fc add children for LRRK2\n",
      "Conunct merge check: let-7 protector\n",
      "miR-184* action, transgenic\n",
      "Conunct merge check: protector dp\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: dp those\n",
      "let-7 each had toxic effects\n",
      "Conjunctions ( nmod )\n",
      "let-7 [deletion, of, let-7, antagomir-mediated, blockage, of, let-7, miR-184]\n",
      "protector [expression, of, dp, target, protector, each]\n",
      "dp [replacement, of, endogenous, dp, with, dp, transgene, non-responsive, to, let-7]\n",
      "those [similar, to, those, of, pathogenic, LRRK2]\n",
      "SDP sent: Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2.\n",
      "SDP     : [(let-7, 'NOUN', 'nmod', False), (deletion, 'NOUN', 'nsubj', False), (had, 'AUX', 'ROOT', False), (effects, 'NOUN', 'dobj', False), (similar, 'ADJ', 'amod', False), (those, 'DET', 'nmod', False), (LRRK2, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-184]\n",
      "DeepCheck Conj Subtree\n",
      "[replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7]\n",
      "0 [Genetic, deletion, of, let-7, ,, antagomir-mediated, blockage, of, let-7, and, miR-184, *, action, ,, transgenic, expression, of, dp, target, protector, ,, or, replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7, each, had, toxic, effects, similar, to, those, of, pathogenic, LRRK2] 43 0 Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2.\n",
      "DeepCheck Conj Subtree\n",
      "[of, let-7, ,, antagomir-mediated, blockage, of, let-7, and, miR-184]\n",
      "DeepCheck Conj Subtree\n",
      "[of, let-7, and, miR-184]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-184]\n",
      "DeepCheck Conj Subtree\n",
      "[of, dp, target, protector, ,, or, replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7]\n",
      "DeepCheck Conj Subtree\n",
      "[replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7]\n",
      "DeepCheck Conj Subtree\n",
      "[of, endogenous, dp]\n",
      "DeepCheck Conj Subtree\n",
      "[with, a, dp, transgene, non-responsive, to, let-7]\n",
      "DeepCheck Conj Subtree\n",
      "[to, let-7]\n",
      "DeepCheck Conj Subtree\n",
      "[each]\n",
      "DeepCheck Conj Subtree\n",
      "[to, those, of, pathogenic, LRRK2]\n",
      "DeepCheck Conj Subtree\n",
      "[of, pathogenic, LRRK2]\n",
      "0 [Genetic, deletion, of, let-7, ,, antagomir-mediated, blockage, of, let-7, and, miR-184, *, action, ,, transgenic, expression, of, dp, target, protector, ,, or, replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7, each, had, toxic, effects, similar, to, those, of, pathogenic, LRRK2] 43 0 Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2.\n",
      "Recognized overlapping entities: {Interval(261, 266, ('GENE_OR_GENE_PRODUCT', 'LRRK2'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text Genetic deletion of let-7 , antagomir-mediated blockage of let-7 and miR-184 * action , transgenic expression of dp target protector , or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2\n",
      "[Genetic, deletion, of, let-7, ,, antagomir-mediated, blockage, of, let-7, and, miR-184, *, action, ,, transgenic, expression, of, dp, target, protector, ,, or, replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7, each, had, toxic, effects, similar, to, those, of, pathogenic, LRRK2]\n",
      "TextBetween: Genetic deletion of let-7 , antagomir-mediated blockage of let-7 and miR-184 * action , transgenic expression of dp target protector , or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2\n",
      "infer direction from token set [Genetic, deletion, of, let-7, ,, antagomir-mediated, blockage, of, let-7, and, miR-184, *, action, ,, transgenic, expression, of, dp, target, protector, ,, or, replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7, each, had, toxic, effects, similar, to, those, of, pathogenic, LRRK2]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954ea0>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 5, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbecba9e1e0>, {})\n",
      "counts between 1 stemTypes2Count [(blockage, 'NEG'), (action, 'NEU'), (expression, 'NEU'), (target, 'NEU'), (replacement, 'NEU'), (effects, 'NEU')]\n",
      "Between Tokens [(blockage, 'NEG'), (action, 'NEU'), (expression, 'NEU'), (target, 'NEU'), (replacement, 'NEU'), (effects, 'NEU')]\n",
      "counts between final False\n",
      "['deletion', 'of', 'let-7', ',', 'antagomir-mediated', 'blockage', 'of', 'let-7', 'and', 'miR-184', '*', 'action', ',', 'transgenic', 'expression', 'of', 'dp', 'target', 'protector', ',', 'or', 'replacement', 'of', 'endogenous', 'dp', 'with', 'a', 'dp', 'transgene', 'non-responsive', 'to', 'let-7', 'each', 'had', 'toxic', 'effects', 'similar', 'to', 'those', 'of', 'pathogenic', 'LRRK2']\n",
      "Counter({'NEU': 5, 'NEG': 1})\n",
      "[(blockage, 'NEG'), (action, 'NEU'), (expression, 'NEU'), (target, 'NEU'), (replacement, 'NEU'), (effects, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2.\n",
      "full sentence Genetic deletion of <e1>let-7</e1>, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic <e2>LRRK2</e2>.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Genetic deletion of <e1>let-7</e1>, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic <e2>LRRK2</e2>.', 'full_sentence': 'Genetic deletion of <e1>let-7</e1>, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic <e2>LRRK2</e2>.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (20, 25), 'entity_text': 'let-7'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (261, 266), 'entity_text': 'LRRK2'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d187.s6.p1\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case let-7\n",
      "add special case LRRK2\n",
      "let-7 8 LRRK2 42\n",
      "(u\"Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2.\", 8, 42)\n",
      "geneword LRRK2 mirword let-7\n",
      "Conjuncts [miR-184, let-7]\n",
      "neighbour miR-184 *\n",
      "neighbour miR-184 action\n",
      "Conjunction before fc let-7\n",
      "fc add children for let-7\n",
      "Conjuncts [let-7, miR-184]\n",
      "neighbour let-7 and\n",
      "neighbour let-7 and\n",
      "Conjunction before fc miR-184\n",
      "Conjuncts [replacement, protector]\n",
      "neighbour replacement of\n",
      "neighbour replacement of\n",
      "Conjunction before fc protector\n",
      "fc add children for protector\n",
      "fc add children for protector\n",
      "fc add children for protector\n",
      "Conjuncts [protector, replacement]\n",
      "neighbour protector ,\n",
      "neighbour protector or\n",
      "Conjunction before fc replacement\n",
      "fc add children for replacement\n",
      "fc add children for replacement\n",
      "Conunct merge check: blockage protector\n",
      "miR-184* action, transgenic expression\n",
      "Conjunctions ( conj )\n",
      "blockage [antagomir-mediated, of, let-7, miR-184]\n",
      "protector [of, dp, target, protector, replacement, of, endogenous, dp, with, dp, transgene, non-responsive, to, let-7]\n",
      "Conjuncts [deletion, let-7]\n",
      "neighbour deletion of\n",
      "neighbour deletion of\n",
      "Conjunction before fc let-7\n",
      "fc add children for let-7\n",
      "fc add children for let-7\n",
      "Conjuncts [blockage, let-7]\n",
      "neighbour blockage of\n",
      "neighbour blockage of\n",
      "Conjunction before fc let-7\n",
      "fc add children for let-7\n",
      "Conjuncts [expression, protector]\n",
      "neighbour expression of\n",
      "neighbour expression of\n",
      "Conjunction before fc protector\n",
      "fc add children for protector\n",
      "fc add children for protector\n",
      "fc add children for protector\n",
      "Conjuncts [replacement, dp]\n",
      "neighbour replacement of\n",
      "neighbour replacement of\n",
      "Conjunction before fc dp\n",
      "fc add children for dp\n",
      "fc add children for dp\n",
      "Conjuncts [replacement, transgene]\n",
      "neighbour replacement of\n",
      "neighbour replacement of\n",
      "Conjunction before fc transgene\n",
      "fc add children for transgene\n",
      "fc add children for transgene\n",
      "fc add children for transgene\n",
      "Conjuncts [non-responsive, let-7]\n",
      "neighbour non-responsive to\n",
      "neighbour non-responsive to\n",
      "Conjunction before fc let-7\n",
      "fc add children for let-7\n",
      "Conjuncts [expression, each]\n",
      "neighbour expression of\n",
      "neighbour expression of\n",
      "Conjunction before fc each\n",
      "Conjuncts [similar, those]\n",
      "neighbour similar to\n",
      "neighbour similar to\n",
      "Conjunction before fc those\n",
      "fc add children for those\n",
      "fc add children for those\n",
      "Conjuncts [those, LRRK2]\n",
      "neighbour those of\n",
      "neighbour those of\n",
      "Conjunction before fc LRRK2\n",
      "fc add children for LRRK2\n",
      "fc add children for LRRK2\n",
      "Conunct merge check: let-7 protector\n",
      "miR-184* action, transgenic\n",
      "Conunct merge check: protector dp\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: dp those\n",
      "let-7 each had toxic effects\n",
      "Conjunctions ( nmod )\n",
      "let-7 [deletion, of, let-7, antagomir-mediated, blockage, of, let-7, miR-184]\n",
      "protector [expression, of, dp, target, protector, each]\n",
      "dp [replacement, of, endogenous, dp, with, dp, transgene, non-responsive, to, let-7]\n",
      "those [similar, to, those, of, pathogenic, LRRK2]\n",
      "SDP sent: Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2.\n",
      "SDP     : [(let-7, 'NOUN', 'nmod', False), (blockage, 'NOUN', 'appos', False), (let-7, 'NOUN', 'nmod', False), (deletion, 'NOUN', 'nsubj', False), (had, 'AUX', 'ROOT', False), (effects, 'NOUN', 'dobj', False), (similar, 'ADJ', 'amod', False), (those, 'DET', 'nmod', False), (LRRK2, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-184]\n",
      "DeepCheck Conj Subtree\n",
      "[replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7]\n",
      "0 [Genetic, deletion, of, let-7, ,, antagomir-mediated, blockage, of, let-7, and, miR-184, *, action, ,, transgenic, expression, of, dp, target, protector, ,, or, replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7, each, had, toxic, effects, similar, to, those, of, pathogenic, LRRK2] 43 0 Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2.\n",
      "DeepCheck Conj Subtree\n",
      "[of, let-7, ,, antagomir-mediated, blockage, of, let-7, and, miR-184]\n",
      "DeepCheck Conj Subtree\n",
      "[of, let-7, and, miR-184]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-184]\n",
      "DeepCheck Conj Subtree\n",
      "[of, dp, target, protector, ,, or, replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7]\n",
      "DeepCheck Conj Subtree\n",
      "[replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7]\n",
      "DeepCheck Conj Subtree\n",
      "[of, endogenous, dp]\n",
      "DeepCheck Conj Subtree\n",
      "[with, a, dp, transgene, non-responsive, to, let-7]\n",
      "DeepCheck Conj Subtree\n",
      "[to, let-7]\n",
      "DeepCheck Conj Subtree\n",
      "[each]\n",
      "DeepCheck Conj Subtree\n",
      "[to, those, of, pathogenic, LRRK2]\n",
      "DeepCheck Conj Subtree\n",
      "[of, pathogenic, LRRK2]\n",
      "0 [Genetic, deletion, of, let-7, ,, antagomir-mediated, blockage, of, let-7, and, miR-184, *, action, ,, transgenic, expression, of, dp, target, protector, ,, or, replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7, each, had, toxic, effects, similar, to, those, of, pathogenic, LRRK2] 43 0 Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2.\n",
      "Recognized overlapping entities: {Interval(261, 266, ('GENE_OR_GENE_PRODUCT', 'LRRK2'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text Genetic deletion of let-7 , antagomir-mediated blockage of let-7 and miR-184 * action , transgenic expression of dp target protector , or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2\n",
      "[Genetic, deletion, of, let-7, ,, antagomir-mediated, blockage, of, let-7, and, miR-184, *, action, ,, transgenic, expression, of, dp, target, protector, ,, or, replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7, each, had, toxic, effects, similar, to, those, of, pathogenic, LRRK2]\n",
      "TextBetween: Genetic deletion of let-7 , antagomir-mediated blockage of let-7 and miR-184 * action , transgenic expression of dp target protector , or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2\n",
      "infer direction from token set [Genetic, deletion, of, let-7, ,, antagomir-mediated, blockage, of, let-7, and, miR-184, *, action, ,, transgenic, expression, of, dp, target, protector, ,, or, replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7, each, had, toxic, effects, similar, to, those, of, pathogenic, LRRK2]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954d08>, {'mir': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 5, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70510>, {'mir': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count [(blockage, 'NEG'), (action, 'NEU'), (expression, 'NEU'), (target, 'NEU'), (replacement, 'NEU'), (effects, 'NEU')]\n",
      "Between Tokens [(action, 'NEU'), (expression, 'NEU'), (target, 'NEU'), (replacement, 'NEU'), (effects, 'NEU')]\n",
      "counts between final False\n",
      "['blockage', 'of', 'let-7', 'and', 'miR-184', '*', 'action', ',', 'transgenic', 'expression', 'of', 'dp', 'target', 'protector', ',', 'or', 'replacement', 'of', 'endogenous', 'dp', 'with', 'a', 'dp', 'transgene', 'non-responsive', 'to', 'let-7', 'each', 'had', 'toxic', 'effects', 'similar', 'to', 'those', 'of', 'pathogenic', 'LRRK2']\n",
      "Counter({'NEU': 5, 'NEG': 1})\n",
      "[(blockage, 'NEG'), (action, 'NEU'), (expression, 'NEU'), (target, 'NEU'), (replacement, 'NEU'), (effects, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2.\n",
      "full sentence Genetic deletion of let-7, antagomir-mediated blockage of <e1>let-7</e1> and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic <e2>LRRK2</e2>.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Genetic deletion of let-7, antagomir-mediated blockage of <e1>let-7</e1> and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic <e2>LRRK2</e2>.', 'full_sentence': 'Genetic deletion of let-7, antagomir-mediated blockage of <e1>let-7</e1> and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic <e2>LRRK2</e2>.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (58, 63), 'entity_text': 'let-7'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (261, 266), 'entity_text': 'LRRK2'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d187.s6.p2\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miR-184\n",
      "add special case LRRK2\n",
      "miR-184 10 LRRK2 42\n",
      "(u\"Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2.\", 10, 42)\n",
      "geneword LRRK2 mirword miR-184\n",
      "Conjuncts [miR-184, let-7]\n",
      "neighbour miR-184 *\n",
      "neighbour miR-184 action\n",
      "Conjunction before fc let-7\n",
      "fc add children for let-7\n",
      "Conjuncts [let-7, miR-184]\n",
      "neighbour let-7 and\n",
      "neighbour let-7 and\n",
      "Conjunction before fc miR-184\n",
      "Conjuncts [replacement, protector]\n",
      "neighbour replacement of\n",
      "neighbour replacement of\n",
      "Conjunction before fc protector\n",
      "fc add children for protector\n",
      "fc add children for protector\n",
      "fc add children for protector\n",
      "Conjuncts [protector, replacement]\n",
      "neighbour protector ,\n",
      "neighbour protector or\n",
      "Conjunction before fc replacement\n",
      "fc add children for replacement\n",
      "fc add children for replacement\n",
      "Conunct merge check: blockage protector\n",
      "miR-184* action, transgenic expression\n",
      "Conjunctions ( conj )\n",
      "blockage [antagomir-mediated, of, let-7, miR-184]\n",
      "protector [of, dp, target, protector, replacement, of, endogenous, dp, with, dp, transgene, non-responsive, to, let-7]\n",
      "Conjuncts [deletion, let-7]\n",
      "neighbour deletion of\n",
      "neighbour deletion of\n",
      "Conjunction before fc let-7\n",
      "fc add children for let-7\n",
      "fc add children for let-7\n",
      "Conjuncts [blockage, let-7]\n",
      "neighbour blockage of\n",
      "neighbour blockage of\n",
      "Conjunction before fc let-7\n",
      "fc add children for let-7\n",
      "Conjuncts [expression, protector]\n",
      "neighbour expression of\n",
      "neighbour expression of\n",
      "Conjunction before fc protector\n",
      "fc add children for protector\n",
      "fc add children for protector\n",
      "fc add children for protector\n",
      "Conjuncts [replacement, dp]\n",
      "neighbour replacement of\n",
      "neighbour replacement of\n",
      "Conjunction before fc dp\n",
      "fc add children for dp\n",
      "fc add children for dp\n",
      "Conjuncts [replacement, transgene]\n",
      "neighbour replacement of\n",
      "neighbour replacement of\n",
      "Conjunction before fc transgene\n",
      "fc add children for transgene\n",
      "fc add children for transgene\n",
      "fc add children for transgene\n",
      "Conjuncts [non-responsive, let-7]\n",
      "neighbour non-responsive to\n",
      "neighbour non-responsive to\n",
      "Conjunction before fc let-7\n",
      "fc add children for let-7\n",
      "Conjuncts [expression, each]\n",
      "neighbour expression of\n",
      "neighbour expression of\n",
      "Conjunction before fc each\n",
      "Conjuncts [similar, those]\n",
      "neighbour similar to\n",
      "neighbour similar to\n",
      "Conjunction before fc those\n",
      "fc add children for those\n",
      "fc add children for those\n",
      "Conjuncts [those, LRRK2]\n",
      "neighbour those of\n",
      "neighbour those of\n",
      "Conjunction before fc LRRK2\n",
      "fc add children for LRRK2\n",
      "fc add children for LRRK2\n",
      "Conunct merge check: let-7 protector\n",
      "miR-184* action, transgenic\n",
      "Conunct merge check: protector dp\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: dp those\n",
      "let-7 each had toxic effects\n",
      "Conjunctions ( nmod )\n",
      "let-7 [deletion, of, let-7, antagomir-mediated, blockage, of, let-7, miR-184]\n",
      "protector [expression, of, dp, target, protector, each]\n",
      "dp [replacement, of, endogenous, dp, with, dp, transgene, non-responsive, to, let-7]\n",
      "those [similar, to, those, of, pathogenic, LRRK2]\n",
      "SDP sent: Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2.\n",
      "SDP     : [(miR-184, 'NOUN', 'conj', False), (let-7, 'NOUN', 'nmod', True), (blockage, 'NOUN', 'appos', False), (let-7, 'NOUN', 'nmod', False), (deletion, 'NOUN', 'nsubj', False), (had, 'AUX', 'ROOT', False), (effects, 'NOUN', 'dobj', False), (similar, 'ADJ', 'amod', False), (those, 'DET', 'nmod', False), (LRRK2, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-184]\n",
      "DeepCheck Conj Subtree\n",
      "[replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7]\n",
      "0 [Genetic, deletion, of, let-7, ,, antagomir-mediated, blockage, of, let-7, and, miR-184, *, action, ,, transgenic, expression, of, dp, target, protector, ,, or, replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7, each, had, toxic, effects, similar, to, those, of, pathogenic, LRRK2] 43 0 Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2.\n",
      "DeepCheck Conj Subtree\n",
      "[of, let-7, ,, antagomir-mediated, blockage, of, let-7, and, miR-184]\n",
      "DeepCheck Conj Subtree\n",
      "[of, let-7, and, miR-184]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-184]\n",
      "DeepCheck Conj Subtree\n",
      "[of, dp, target, protector, ,, or, replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7]\n",
      "DeepCheck Conj Subtree\n",
      "[replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7]\n",
      "DeepCheck Conj Subtree\n",
      "[of, endogenous, dp]\n",
      "DeepCheck Conj Subtree\n",
      "[with, a, dp, transgene, non-responsive, to, let-7]\n",
      "DeepCheck Conj Subtree\n",
      "[to, let-7]\n",
      "DeepCheck Conj Subtree\n",
      "[each]\n",
      "DeepCheck Conj Subtree\n",
      "[to, those, of, pathogenic, LRRK2]\n",
      "DeepCheck Conj Subtree\n",
      "[of, pathogenic, LRRK2]\n",
      "0 [Genetic, deletion, of, let-7, ,, antagomir-mediated, blockage, of, let-7, and, miR-184, *, action, ,, transgenic, expression, of, dp, target, protector, ,, or, replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7, each, had, toxic, effects, similar, to, those, of, pathogenic, LRRK2] 43 0 Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2.\n",
      "Recognized overlapping entities: {Interval(261, 266, ('GENE_OR_GENE_PRODUCT', 'LRRK2'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text Genetic deletion of let-7 , antagomir-mediated blockage of let-7 and miR-184 * action , transgenic expression of dp target protector , or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2\n",
      "[Genetic, deletion, of, let-7, ,, antagomir-mediated, blockage, of, let-7, and, miR-184, *, action, ,, transgenic, expression, of, dp, target, protector, ,, or, replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7, each, had, toxic, effects, similar, to, those, of, pathogenic, LRRK2]\n",
      "TextBetween: Genetic deletion of let-7 , antagomir-mediated blockage of let-7 and miR-184 * action , transgenic expression of dp target protector , or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2\n",
      "infer direction from token set [Genetic, deletion, of, let-7, ,, antagomir-mediated, blockage, of, let-7, and, miR-184, *, action, ,, transgenic, expression, of, dp, target, protector, ,, or, replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7, each, had, toxic, effects, similar, to, those, of, pathogenic, LRRK2]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e2048>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 5})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e2620>, {'mir': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(action, 'NEU'), (expression, 'NEU'), (target, 'NEU'), (replacement, 'NEU'), (effects, 'NEU')]\n",
      "Between Tokens [(action, 'NEU'), (expression, 'NEU'), (target, 'NEU'), (replacement, 'NEU'), (effects, 'NEU')]\n",
      "counts between final False\n",
      "['let-7', 'and', 'miR-184', '*', 'action', ',', 'transgenic', 'expression', 'of', 'dp', 'target', 'protector', ',', 'or', 'replacement', 'of', 'endogenous', 'dp', 'with', 'a', 'dp', 'transgene', 'non-responsive', 'to', 'let-7', 'each', 'had', 'toxic', 'effects', 'similar', 'to', 'those', 'of', 'pathogenic', 'LRRK2']\n",
      "Counter({'NEU': 5})\n",
      "[(action, 'NEU'), (expression, 'NEU'), (target, 'NEU'), (replacement, 'NEU'), (effects, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2.\n",
      "full sentence Genetic deletion of let-7, antagomir-mediated blockage of let-7 and <e1>miR-184</e1>* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic <e2>LRRK2</e2>.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Genetic deletion of let-7, antagomir-mediated blockage of let-7 and <e1>miR-184</e1>* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic <e2>LRRK2</e2>.', 'full_sentence': 'Genetic deletion of let-7, antagomir-mediated blockage of let-7 and <e1>miR-184</e1>* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic <e2>LRRK2</e2>.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (68, 75), 'entity_text': 'miR-184'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (261, 266), 'entity_text': 'LRRK2'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d187.s6.p3\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case let-7\n",
      "add special case LRRK2\n",
      "let-7 32 LRRK2 42\n",
      "(u\"Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2.\", 32, 42)\n",
      "geneword LRRK2 mirword let-7\n",
      "Conjuncts [miR-184, let-7]\n",
      "neighbour miR-184 *\n",
      "neighbour miR-184 action\n",
      "Conjunction before fc let-7\n",
      "fc add children for let-7\n",
      "Conjuncts [let-7, miR-184]\n",
      "neighbour let-7 and\n",
      "neighbour let-7 and\n",
      "Conjunction before fc miR-184\n",
      "Conjuncts [replacement, protector]\n",
      "neighbour replacement of\n",
      "neighbour replacement of\n",
      "Conjunction before fc protector\n",
      "fc add children for protector\n",
      "fc add children for protector\n",
      "fc add children for protector\n",
      "Conjuncts [protector, replacement]\n",
      "neighbour protector ,\n",
      "neighbour protector or\n",
      "Conjunction before fc replacement\n",
      "fc add children for replacement\n",
      "fc add children for replacement\n",
      "Conunct merge check: blockage protector\n",
      "miR-184* action, transgenic expression\n",
      "Conjunctions ( conj )\n",
      "blockage [antagomir-mediated, of, let-7, miR-184]\n",
      "protector [of, dp, target, protector, replacement, of, endogenous, dp, with, dp, transgene, non-responsive, to, let-7]\n",
      "Conjuncts [deletion, let-7]\n",
      "neighbour deletion of\n",
      "neighbour deletion of\n",
      "Conjunction before fc let-7\n",
      "fc add children for let-7\n",
      "fc add children for let-7\n",
      "Conjuncts [blockage, let-7]\n",
      "neighbour blockage of\n",
      "neighbour blockage of\n",
      "Conjunction before fc let-7\n",
      "fc add children for let-7\n",
      "Conjuncts [expression, protector]\n",
      "neighbour expression of\n",
      "neighbour expression of\n",
      "Conjunction before fc protector\n",
      "fc add children for protector\n",
      "fc add children for protector\n",
      "fc add children for protector\n",
      "Conjuncts [replacement, dp]\n",
      "neighbour replacement of\n",
      "neighbour replacement of\n",
      "Conjunction before fc dp\n",
      "fc add children for dp\n",
      "fc add children for dp\n",
      "Conjuncts [replacement, transgene]\n",
      "neighbour replacement of\n",
      "neighbour replacement of\n",
      "Conjunction before fc transgene\n",
      "fc add children for transgene\n",
      "fc add children for transgene\n",
      "fc add children for transgene\n",
      "Conjuncts [non-responsive, let-7]\n",
      "neighbour non-responsive to\n",
      "neighbour non-responsive to\n",
      "Conjunction before fc let-7\n",
      "fc add children for let-7\n",
      "Conjuncts [expression, each]\n",
      "neighbour expression of\n",
      "neighbour expression of\n",
      "Conjunction before fc each\n",
      "Conjuncts [similar, those]\n",
      "neighbour similar to\n",
      "neighbour similar to\n",
      "Conjunction before fc those\n",
      "fc add children for those\n",
      "fc add children for those\n",
      "Conjuncts [those, LRRK2]\n",
      "neighbour those of\n",
      "neighbour those of\n",
      "Conjunction before fc LRRK2\n",
      "fc add children for LRRK2\n",
      "fc add children for LRRK2\n",
      "Conunct merge check: let-7 protector\n",
      "miR-184* action, transgenic\n",
      "Conunct merge check: protector dp\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: dp those\n",
      "let-7 each had toxic effects\n",
      "Conjunctions ( nmod )\n",
      "let-7 [deletion, of, let-7, antagomir-mediated, blockage, of, let-7, miR-184]\n",
      "protector [expression, of, dp, target, protector, each]\n",
      "dp [replacement, of, endogenous, dp, with, dp, transgene, non-responsive, to, let-7]\n",
      "those [similar, to, those, of, pathogenic, LRRK2]\n",
      "SDP sent: Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2.\n",
      "SDP     : [(let-7, 'NOUN', 'nmod', False), (non-responsive, 'ADJ', 'amod', False), (transgene, 'NOUN', 'nmod', False), (replacement, 'NOUN', 'conj', False), (protector, 'NOUN', 'nmod', True), (expression, 'NOUN', 'nsubj', False), (had, 'AUX', 'ROOT', False), (effects, 'NOUN', 'dobj', False), (similar, 'ADJ', 'amod', False), (those, 'DET', 'nmod', False), (LRRK2, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-184]\n",
      "DeepCheck Conj Subtree\n",
      "[replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7]\n",
      "0 [Genetic, deletion, of, let-7, ,, antagomir-mediated, blockage, of, let-7, and, miR-184, *, action, ,, transgenic, expression, of, dp, target, protector, ,, or, replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7, each, had, toxic, effects, similar, to, those, of, pathogenic, LRRK2] 43 0 Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2.\n",
      "DeepCheck Conj Subtree\n",
      "[of, let-7, ,, antagomir-mediated, blockage, of, let-7, and, miR-184]\n",
      "DeepCheck Conj Subtree\n",
      "[of, let-7, and, miR-184]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-184]\n",
      "DeepCheck Conj Subtree\n",
      "[of, dp, target, protector, ,, or, replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7]\n",
      "DeepCheck Conj Subtree\n",
      "[replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7]\n",
      "DeepCheck Conj Subtree\n",
      "[of, endogenous, dp]\n",
      "DeepCheck Conj Subtree\n",
      "[with, a, dp, transgene, non-responsive, to, let-7]\n",
      "DeepCheck Conj Subtree\n",
      "[to, let-7]\n",
      "DeepCheck Conj Subtree\n",
      "[each]\n",
      "DeepCheck Conj Subtree\n",
      "[to, those, of, pathogenic, LRRK2]\n",
      "DeepCheck Conj Subtree\n",
      "[of, pathogenic, LRRK2]\n",
      "0 [Genetic, deletion, of, let-7, ,, antagomir-mediated, blockage, of, let-7, and, miR-184, *, action, ,, transgenic, expression, of, dp, target, protector, ,, or, replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7, each, had, toxic, effects, similar, to, those, of, pathogenic, LRRK2] 43 0 Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2.\n",
      "Recognized overlapping entities: {Interval(261, 266, ('GENE_OR_GENE_PRODUCT', 'LRRK2'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text Genetic deletion of let-7 , antagomir-mediated blockage of let-7 and miR-184 * action , transgenic expression of dp target protector , or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2\n",
      "[Genetic, deletion, of, let-7, ,, antagomir-mediated, blockage, of, let-7, and, miR-184, *, action, ,, transgenic, expression, of, dp, target, protector, ,, or, replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7, each, had, toxic, effects, similar, to, those, of, pathogenic, LRRK2]\n",
      "TextBetween: Genetic deletion of let-7 , antagomir-mediated blockage of let-7 and miR-184 * action , transgenic expression of dp target protector , or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2\n",
      "infer direction from token set [Genetic, deletion, of, let-7, ,, antagomir-mediated, blockage, of, let-7, and, miR-184, *, action, ,, transgenic, expression, of, dp, target, protector, ,, or, replacement, of, endogenous, dp, with, a, dp, transgene, non-responsive, to, let-7, each, had, toxic, effects, similar, to, those, of, pathogenic, LRRK2]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbecba9e158>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62ae8>, {})\n",
      "counts between 1 stemTypes2Count [(effects, 'NEU')]\n",
      "Between Tokens [(effects, 'NEU')]\n",
      "counts between final False\n",
      "['non-responsive', 'to', 'let-7', 'each', 'had', 'toxic', 'effects', 'similar', 'to', 'those', 'of', 'pathogenic', 'LRRK2']\n",
      "Counter({'NEU': 1})\n",
      "[(effects, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to let-7 each had toxic effects similar to those of pathogenic LRRK2.\n",
      "full sentence Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to <e1>let-7</e1> each had toxic effects similar to those of pathogenic <e2>LRRK2</e2>.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to <e1>let-7</e1> each had toxic effects similar to those of pathogenic <e2>LRRK2</e2>.', 'full_sentence': 'Genetic deletion of let-7, antagomir-mediated blockage of let-7 and miR-184* action, transgenic expression of dp target protector, or replacement of endogenous dp with a dp transgene non-responsive to <e1>let-7</e1> each had toxic effects similar to those of pathogenic <e2>LRRK2</e2>.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (201, 206), 'entity_text': 'let-7'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (261, 266), 'entity_text': 'LRRK2'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d193.s6.p0\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case microRNA-338\n",
      "add special case eiF4E3\n",
      "microRNA-338 3 eiF4E3 9\n",
      "(u\"[Interaction of microRNA-338 and its potential targeting protein eiF4E3].\", 3, 9)\n",
      "geneword eiF4E3 mirword microRNA-338\n",
      "Conjuncts [protein, microRNA-338]\n",
      "neighbour protein eiF4E3\n",
      "neighbour protein eiF4E3\n",
      "Conjunction before fc microRNA-338\n",
      "fc add children for microRNA-338\n",
      "Conjuncts [microRNA-338, protein]\n",
      "neighbour microRNA-338 and\n",
      "neighbour microRNA-338 and\n",
      "Conjunction before fc protein\n",
      "fc add children for protein\n",
      "fc add children for protein\n",
      "fc add children for protein\n",
      "Conjunctions ( conj )\n",
      "microRNA-338 [of, microRNA-338, potential, targeting, protein, eiF4E3]\n",
      "achildren {microRNA-338}\n",
      "achildren False\n",
      "Conjuncts [Interaction, microRNA-338]\n",
      "neighbour Interaction of\n",
      "neighbour Interaction of\n",
      "Conjunction before fc microRNA-338\n",
      "fc add children for microRNA-338\n",
      "Conjunctions ( nmod )\n",
      "microRNA-338 [Interaction, of, microRNA-338]\n",
      "SDP sent: [Interaction of microRNA-338 and its potential targeting protein eiF4E3].\n",
      "SDP     : [(microRNA-338, 'NOUN', 'nmod', False), (Interaction, 'NOUN', 'ROOT', False), (microRNA-338, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[its, potential, targeting, protein, eiF4E3]\n",
      "0 [Interaction, of, microRNA-338, and, its, potential, targeting, protein, eiF4E3] 9 1 Interaction of microRNA-338 and its potential targeting protein eiF4E3].\n",
      "DeepCheck Conj Subtree\n",
      "[of, microRNA-338, and, its, potential, targeting, protein, eiF4E3]\n",
      "DeepCheck Conj Subtree\n",
      "[its, potential, targeting, protein, eiF4E3]\n",
      "0 [Interaction, of, microRNA-338, and, its, potential, targeting, protein, eiF4E3] 9 1 Interaction of microRNA-338 and its potential targeting protein eiF4E3].\n",
      "Recognized overlapping entities: {Interval(65, 71, ('GENE_OR_GENE_PRODUCT', 'eiF4E3'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text Interaction of microRNA-338 and its potential targeting protein eiF4E3\n",
      "[Interaction, of, microRNA-338, and, its, potential, targeting, protein, eiF4E3]\n",
      "TextBetween: Interaction of microRNA-338 and its potential targeting protein eiF4E3\n",
      "infer direction from token set [Interaction, of, microRNA-338, and, its, potential, targeting, protein, eiF4E3]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62a60>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fc0bbbd3158>, {'mir': Counter({'NEU': 1}), 'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(Interaction, 'NEU'), (targeting, 'NEU')]\n",
      "Between Tokens [(targeting, 'NEU')]\n",
      "counts between final False\n",
      "['Interaction', 'of', 'microRNA-338', 'and', 'its', 'potential', 'targeting', 'protein', 'eiF4E3']\n",
      "Counter({'NEU': 2})\n",
      "[(Interaction, 'NEU'), (targeting, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence [Interaction of microRNA-338 and its potential targeting protein eiF4E3].\n",
      "full sentence [Interaction of <e1>microRNA-338</e1> and its potential targeting protein <e2>eiF4E3</e2>].\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': '[Interaction of <e1>microRNA-338</e1> and its potential targeting protein <e2>eiF4E3</e2>].', 'full_sentence': '[Interaction of <e1>microRNA-338</e1> and its potential targeting protein <e2>eiF4E3</e2>].', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (16, 28), 'entity_text': 'microRNA-338'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (65, 71), 'entity_text': 'eiF4E3'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d195.s6.p2\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miR-26a\n",
      "add special case PTEN\n",
      "miR-26a-mediated 5 PTEN 26\n",
      "(u\"Finally, we demonstrate that miR-26a-mediated PTEN repression in a murine glioma model both enhances de novo tumor formation and precludes loss of heterozygosity and the PTEN locus.\", 5, 26)\n",
      "geneword PTEN mirword miR-26a-mediated\n",
      "Conjuncts [precludes, enhances]\n",
      "neighbour precludes loss\n",
      "neighbour precludes loss\n",
      "Conjunction before fc enhances\n",
      "fc add children for enhances\n",
      "Conjuncts [enhances, precludes]\n",
      "neighbour enhances de\n",
      "neighbour enhances de\n",
      "Conjunction before fc precludes\n",
      "fc add children for precludes\n",
      "Conjuncts [locus, heterozygosity]\n",
      "neighbour locus .\n",
      "neighbour locus None\n",
      "Conjunction before fc heterozygosity\n",
      "fc add children for heterozygosity\n",
      "Conjuncts [heterozygosity, locus]\n",
      "neighbour heterozygosity and\n",
      "neighbour heterozygosity and\n",
      "Conjunction before fc locus\n",
      "fc add children for locus\n",
      "Conjunctions ( conj )\n",
      "enhances [enhances, de, novo, tumor, formation, precludes, loss, of, heterozygosity, PTEN, locus]\n",
      "Conjuncts [repression, model]\n",
      "neighbour repression in\n",
      "neighbour repression in\n",
      "Conjunction before fc model\n",
      "fc add children for model\n",
      "fc add children for model\n",
      "fc add children for model\n",
      "Conjuncts [loss, heterozygosity]\n",
      "neighbour loss of\n",
      "neighbour loss of\n",
      "Conjunction before fc heterozygosity\n",
      "fc add children for heterozygosity\n",
      "Conunct merge check: model heterozygosity\n",
      "model both enhances de novo tumor formation and precludes\n",
      "Conjunctions ( nmod )\n",
      "model [repression, in, murine, glioma, model]\n",
      "heterozygosity [loss, of, heterozygosity]\n",
      "SDP sent: Finally, we demonstrate that miR-26a-mediated PTEN repression in a murine glioma model both enhances de novo tumor formation and precludes loss of heterozygosity and the PTEN locus.\n",
      "SDP     : [(miR-26a-mediated, 'ADJ', 'amod', False), (repression, 'NOUN', 'nsubj', False), (enhances, 'VERB', 'ccomp', True), (precludes, 'VERB', 'conj', False), (loss, 'NOUN', 'dobj', False), (heterozygosity, 'NOUN', 'nmod', True), (locus, 'NOUN', 'conj', False), (PTEN, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule noun found precludes loss [precludes, loss, of, heterozygosity, and, the, PTEN, locus]\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[the, PTEN, locus]\n",
      "0 [Finally, ,, we, demonstrate] 4 0 Finally, we demonstrate that miR-26a-mediated PTEN repression in a murine glioma model both enhances de novo tumor formation and precludes loss of heterozygosity and the PTEN locus.\n",
      "1 [miR-26a-mediated, PTEN, repression, in, a, murine, glioma, model, both, enhances, de, novo, tumor, formation, and, precludes, loss, of, heterozygosity, and, the, PTEN, locus] 23 29 miR-26a-mediated PTEN repression in a murine glioma model both enhances de novo tumor formation and precludes loss of heterozygosity and the PTEN locus.\n",
      "DeepCheck Conj Subtree\n",
      "[in, a, murine, glioma, model]\n",
      "deepcheck connection rule noun found precludes loss [precludes, loss, of, heterozygosity, and, the, PTEN, locus]\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[of, heterozygosity, and, the, PTEN, locus]\n",
      "DeepCheck Conj Subtree\n",
      "[the, PTEN, locus]\n",
      "0 [Finally, ,, we, demonstrate] 4 0 Finally, we demonstrate that miR-26a-mediated PTEN repression in a murine glioma model both enhances de novo tumor formation and precludes loss of heterozygosity and the PTEN locus.\n",
      "1 [miR-26a-mediated, PTEN, repression, in, a, murine, glioma, model, both, enhances, de, novo, tumor, formation, and, precludes, loss, of, heterozygosity, and, the, PTEN, locus] 23 29 miR-26a-mediated PTEN repression in a murine glioma model both enhances de novo tumor formation and precludes loss of heterozygosity and the PTEN locus.\n",
      "Recognized overlapping entities: {Interval(170, 174, ('GENE_OR_GENE_PRODUCT', 'PTEN'))}\n",
      "passive? False\n",
      "Counter({'NEG': 2, 'POS': 1})\n",
      "Compartment Text miR-26a-mediated PTEN repression in a murine glioma model both enhances de novo tumor formation and precludes loss of heterozygosity and the PTEN locus\n",
      "[miR-26a-mediated, PTEN, repression, in, a, murine, glioma, model, both, enhances, de, novo, tumor, formation, and, precludes, loss, of, heterozygosity, and, the, PTEN, locus]\n",
      "TextBetween: miR-26a-mediated PTEN repression in a murine glioma model both enhances de novo tumor formation and precludes loss of heterozygosity and the PTEN locus\n",
      "infer direction from token set [miR-26a-mediated, PTEN, repression, in, a, murine, glioma, model, both, enhances, de, novo, tumor, formation, and, precludes, loss, of, heterozygosity, and, the, PTEN, locus]\n",
      "stemTypes2Count: Counter({'NEG': 2, 'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62bf8>, {'mir': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62d08>, {'mir': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count [(repression, 'NEG'), (enhances, 'POS'), (formation, 'NEU'), (loss, 'NEG')]\n",
      "Between Tokens [(repression, 'NEG'), (enhances, 'POS'), (formation, 'NEU'), (loss, 'NEG')]\n",
      "counts between final False\n",
      "['miR-26a-mediated', 'PTEN', 'repression', 'in', 'a', 'murine', 'glioma', 'model', 'both', 'enhances', 'de', 'novo', 'tumor', 'formation', 'and', 'precludes', 'loss', 'of', 'heterozygosity', 'and', 'the', 'PTEN', 'locus']\n",
      "Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "[(repression, 'NEG'), (enhances, 'POS'), (formation, 'NEU'), (loss, 'NEG')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Finally, we demonstrate that miR-26a-mediated PTEN repression in a murine glioma model both enhances de novo tumor formation and precludes loss of heterozygosity and the PTEN locus.\n",
      "full sentence Finally, we demonstrate that <e1>miR-26a</e1>-mediated PTEN repression in a murine glioma model both enhances de novo tumor formation and precludes loss of heterozygosity and the <e2>PTEN</e2> locus.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Finally, we demonstrate that <e1>miR-26a</e1>-mediated PTEN repression in a murine glioma model both enhances de novo tumor formation and precludes loss of heterozygosity and the <e2>PTEN</e2> locus.', 'full_sentence': 'Finally, we demonstrate that <e1>miR-26a</e1>-mediated PTEN repression in a murine glioma model both enhances de novo tumor formation and precludes loss of heterozygosity and the <e2>PTEN</e2> locus.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (29, 36), 'entity_text': 'miR-26a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (170, 174), 'entity_text': 'PTEN'}}\n",
      "Total:      397\n",
      "Correct:    359 0.9042821158690176\n",
      "Incorrect:  38 0.09571788413098237\n",
      "classes Counter()\n",
      "precision 0.9278350515463918\n",
      "recall 0.8823529411764706\n",
      "specificity 0.927461139896373\n",
      "f1 0.9045226130653266\n",
      "relationFalseCounter Counter({'conj': 6, 'context': 4, 'compartment': 3, 'compartment_nmod': 3, 'entity': 2, 'conj_nmod': 2})\n",
      "Correct classified:      165 0.4156171284634761 0.8505154639175257\n",
      "Incorrect classified:    29 0.07304785894206549 0.14948453608247422\n",
      "Random classified:  53 0.13350125944584382 0.27319587628865977\n",
      "Counter({'counts between': 13, 'counts between equal': 4, 'compartment gene mir': 3, 'counts opp': 3, 'compartment mir gene by': 2, 'compartment mir gene pos corr': 1, 'between mir gene reg corr': 1, 'compartment mir gene': 1, 'compartment mir gene neg corr': 1})\n",
      "Correct interaction&classified:      156 0.3929471032745592 0.8041237113402062\n",
      "Classified by:  Counter({'counts between': 101, 'counts between equal': 16, 'between mir gene': 16, 'between gene mir': 11, 'compartment mir gene': 10, 'counts opp': 9, 'compartment gene mir': 7, 'compartment gene mir by': 6, 'compartment mir gene pos corr': 4, 'return': 4, 'compartment mir gene neg corr': 3, 'between mir gene reg corr': 2, 'between gene mir sites': 2, 'compartment mir gene by': 2, 'counts between alternating': 1})\n",
      "[[[186   1]\n",
      "  [  4   3]]\n",
      "\n",
      " [[187   6]\n",
      "  [  0   1]]\n",
      "\n",
      " [[187   3]\n",
      "  [  0   4]]\n",
      "\n",
      " [[133   8]\n",
      "  [ 12  41]]\n",
      "\n",
      " [[ 70   5]\n",
      "  [ 11 108]]\n",
      "\n",
      " [[178   6]\n",
      "  [  2   8]]]\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "GENE_MIR_DOWN       0.75      0.43      0.55         7\n",
      " GENE_MIR_NEU       0.14      1.00      0.25         1\n",
      "  GENE_MIR_UP       0.57      1.00      0.73         4\n",
      "MIR_GENE_DOWN       0.84      0.77      0.80        53\n",
      " MIR_GENE_NEU       0.96      0.91      0.93       119\n",
      "  MIR_GENE_UP       0.57      0.80      0.67        10\n",
      "\n",
      "     accuracy                           0.85       194\n",
      "    macro avg       0.64      0.82      0.65       194\n",
      " weighted avg       0.88      0.85      0.86       194\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%aimport MirGeneRelCheck\n",
    "\n",
    "relClassifier = SentenceRelationClassifier(relationsFile)\n",
    "\n",
    "train_results = runCheck(-1, \"TRAIN\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading relations scai_test_rels.tsv\n",
      "Interactions 232\n",
      "Non NA Interactions 141\n",
      "Will test 1 subsets.\n",
      "Testing checks ('compartment', 'between', 'counts', 'return')\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Setting relfile /mnt/extproj/projekte/textmining/miRExplore/python/allrels.csv\n",
      "Setting relfile /mnt/extproj/projekte/textmining/miRExplore/python/allrels.csv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines. \n",
      "('miRNA-128', 'Specific_miRNAs', (22, 31)) ('APOBEC3C', 'Genes/Proteins', (99, 103))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines. \n",
      "('miRNA-128', 'Specific_miRNAs', (22, 31)) ('ANGPTL6', 'Genes/Proteins', (105, 112))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines. \n",
      "('miRNA-128', 'Specific_miRNAs', (22, 31)) ('Bmi-1', 'Genes/Proteins', (115, 120))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d6.s1.p0\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case miR-21\n",
      "add special case caspase 9\n",
      "miR-21 4 9 7\n",
      "(u\"Moreover, reduction of miR-21 activated caspase 9 and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.\", 4, 7)\n",
      "geneword 9 mirword miR-21\n",
      "Conjuncts [3, caspase]\n",
      "neighbour 3 ,\n",
      "neighbour 3 which\n",
      "Conjunction before fc caspase\n",
      "fc add children for caspase\n",
      "fc add children for caspase\n",
      "fc add children for caspase\n",
      "fc add children for caspase\n",
      "Conjuncts [caspase, 3]\n",
      "neighbour caspase 9\n",
      "neighbour caspase 9\n",
      "Conjunction before fc 3\n",
      "Conunct merge check: reduction caspase\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( conj )\n",
      "reduction [reduction, .]\n",
      "caspase [of, miR-21, activated, caspase, 9, 3]\n",
      "achildren {miR-21}\n",
      "achildren False\n",
      "Conjuncts [reduction, caspase]\n",
      "neighbour reduction of\n",
      "neighbour reduction of\n",
      "Conjunction before fc caspase\n",
      "fc add children for caspase\n",
      "fc add children for caspase\n",
      "fc add children for caspase\n",
      "fc add children for caspase\n",
      "Conjuncts [genes, TIMP3]\n",
      "neighbour genes ,\n",
      "neighbour genes such\n",
      "Conjunction before fc TIMP3\n",
      "fc add children for TIMP3\n",
      "Conunct merge check: reduction TIMP3\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( nmod )\n",
      "reduction [reduction, of, miR-21, activated, caspase, 9, .]\n",
      "TIMP3 [genes, such, TIMP3]\n",
      "achildren {miR-21}\n",
      "achildren False\n",
      "SDP sent: Moreover, reduction of miR-21 activated caspase 9 and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.\n",
      "SDP     : [(miR-21, 'NOUN', 'compound', False), (caspase, 'NOUN', 'nmod', False), (9, 'NUM', 'nummod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[3]\n",
      "deepcheck connection rule [by, modulating, multiple, potential, target, genes, ,, such, as, TIMP3]\n",
      "0 [Moreover, ,, reduction, of, miR-21, activated, caspase, 9, and, 3, ,, which, may, be, mediated, by, modulating, multiple, potential, target, genes, ,, such, as, TIMP3] 25 0 Moreover, reduction of miR-21 activated caspase 9 and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-21, activated, caspase, 9, and, 3, ,, which, may, be, mediated, by, modulating, multiple, potential, target, genes, ,, such, as, TIMP3]\n",
      "DeepCheck Conj Subtree\n",
      "[3]\n",
      "deepcheck connection rule [by, modulating, multiple, potential, target, genes, ,, such, as, TIMP3]\n",
      "DeepCheck Conj Subtree\n",
      "[such, as, TIMP3]\n",
      "0 [Moreover, ,, reduction, of, miR-21, activated, caspase, 9, and, 3, ,, which, may, be, mediated, by, modulating, multiple, potential, target, genes, ,, such, as, TIMP3] 25 0 Moreover, reduction of miR-21 activated caspase 9 and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.\n",
      "Context Fail for Noun Rule\n",
      "Recognized overlapping entities: {Interval(40, 49, ('GENE_OR_GENE_PRODUCT', 'caspase 9'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text Moreover , reduction of miR-21 activated caspase 9 and 3 , which may be mediated by modulating multiple potential target genes , such as TIMP3\n",
      "[Moreover, ,, reduction, of, miR-21, activated, caspase, 9, and, 3, ,, which, may, be, mediated, by, modulating, multiple, potential, target, genes, ,, such, as, TIMP3]\n",
      "TextBetween: Moreover , reduction of miR-21 activated caspase 9 and 3 , which may be mediated by modulating multiple potential target genes , such as TIMP3\n",
      "infer direction from token set [Moreover, ,, reduction, of, miR-21, activated, caspase, 9, and, 3, ,, which, may, be, mediated, by, modulating, multiple, potential, target, genes, ,, such, as, TIMP3]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e27b8>, {'mir': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEG': 1, 'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e21e0>, {'mir': Counter({'NEG': 1, 'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(reduction, 'NEG'), (activated, 'NEU')]\n",
      "Between Tokens [(activated, 'NEU')]\n",
      "counts between final False\n",
      "['reduction', 'of', 'miR-21', 'activated', 'caspase', '9', 'and', '3']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "[(reduction, 'NEG'), (activated, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': False, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Moreover, reduction of miR-21 activated caspase 9 and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.\n",
      "full sentence Moreover, reduction of <e1>miR-21</e1> activated <e2>caspase 9</e2> and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Moreover, reduction of <e1>miR-21</e1> activated <e2>caspase 9</e2> and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.', 'full_sentence': 'Moreover, reduction of <e1>miR-21</e1> activated <e2>caspase 9</e2> and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (23, 29), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (40, 49), 'entity_text': 'caspase 9'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d9.s1.p0\n",
      "IS:  ('MIR_GENE', 'UP')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased <e2>FGF20</e2> expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 'full_sentence': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased <e2>FGF20</e2> expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (33, 40), 'entity_text': 'miR-433'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (81, 86), 'entity_text': 'FGF20'}}\n",
      "add special case miR-433\n",
      "add special case FGF20\n",
      "miR-433 6 FGF20 16\n",
      "(u\"The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 6, 16)\n",
      "geneword FGF20 mirword miR-433\n",
      "Conjuncts [upregulation, expression]\n",
      "neighbour upregulation of\n",
      "neighbour upregulation of\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conjuncts [expression, upregulation]\n",
      "neighbour expression and\n",
      "neighbour expression and\n",
      "Conjunction before fc upregulation\n",
      "fc add children for upregulation\n",
      "Conjunctions ( conj )\n",
      "expression [in, increased, FGF20, expression, upregulation, of, alpha-synuclein]\n",
      "Conjuncts [reduction, affinity]\n",
      "neighbour reduction of\n",
      "neighbour reduction of\n",
      "Conjunction before fc affinity\n",
      "fc add children for affinity\n",
      "fc add children for affinity\n",
      "fc add children for affinity\n",
      "Conjuncts [affinity, miR-433]\n",
      "neighbour affinity of\n",
      "neighbour affinity of\n",
      "Conjunction before fc miR-433\n",
      "fc add children for miR-433\n",
      "Conjuncts [affinity, UTR]\n",
      "neighbour affinity of\n",
      "neighbour affinity of\n",
      "Conjunction before fc UTR\n",
      "fc add children for UTR\n",
      "fc add children for UTR\n",
      "Conjuncts [result, expression]\n",
      "neighbour result in\n",
      "neighbour result in\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conjuncts [upregulation, alpha-synuclein]\n",
      "neighbour upregulation of\n",
      "neighbour upregulation of\n",
      "Conjunction before fc alpha-synuclein\n",
      "fc add children for alpha-synuclein\n",
      "Conjuncts [promote, turn]\n",
      "neighbour promote dopaminergic\n",
      "neighbour promote dopaminergic\n",
      "Conjunction before fc turn\n",
      "fc add children for turn\n",
      "Conunct merge check: affinity expression\n",
      "UTR would\n",
      "Conunct merge check: expression alpha-synuclein\n",
      "expression and\n",
      "Double Conjunction Found\n",
      "Conunct merge check: alpha-synuclein turn\n",
      "alpha-synuclein, which could\n",
      "Conjunctions ( nmod )\n",
      "affinity [reduction, of, affinity, of, miR-433, to, 3, UTR]\n",
      "expression [result, in, increased, FGF20, expression]\n",
      "alpha-synuclein [upregulation, of, alpha-synuclein]\n",
      "turn [in, turn, promote]\n",
      "(alpha-synuclein, 'conj') [result, in, increased, FGF20, expression, upregulation, of, alpha-synuclein]\n",
      "SDP sent: The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "SDP     : [(miR-433, 'NOUN', 'nmod', False), (affinity, 'NOUN', 'nmod', False), (reduction, 'NOUN', 'nsubj', False), (result, 'VERB', 'ROOT', False), (expression, 'NOUN', 'nmod', False), (FGF20, 'NOUN', 'compound', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "0 [The, reduction, of, the, affinity, of, miR-433, to, the, 3, ', UTR, would, result, in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration] 31 0 The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, affinity, of, miR-433, to, the, 3, ', UTR]\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-433]\n",
      "DeepCheck Conj Subtree\n",
      "[to, the, 3, ', UTR]\n",
      "DeepCheck Conj Subtree\n",
      "[in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "DeepCheck Conj Subtree\n",
      "[upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "DeepCheck Conj Subtree\n",
      "[of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "DeepCheck Conj Subtree\n",
      "[in, turn]\n",
      "0 [The, reduction, of, the, affinity, of, miR-433, to, the, 3, ', UTR, would, result, in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration] 31 0 The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "Recognized overlapping entities: {Interval(81, 86, ('GENE_OR_GENE_PRODUCT', 'FGF20'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text The reduction of the affinity of miR-433 to the 3 ' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein , which could in turn promote dopaminergic neurons degeneration\n",
      "[The, reduction, of, the, affinity, of, miR-433, to, the, 3, ', UTR, would, result, in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "TextBetween: The reduction of the affinity of miR-433 to the 3 ' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein , which could in turn promote dopaminergic neurons degeneration\n",
      "infer direction from token set [The, reduction, of, the, affinity, of, miR-433, to, the, 3, ', UTR, would, result, in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "stemTypes2Count: Counter({'POS': 3, 'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62b70>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'POS': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62ea0>, {'gene': Counter({'POS': 1, 'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(result, 'NEU'), (increased, 'POS'), (expression, 'NEU')]\n",
      "Between Tokens [(result, 'NEU'), (increased, 'POS')]\n",
      "counts between final False\n",
      "['affinity', 'of', 'miR-433', 'to', 'the', '3', \"'\", 'UTR', 'would', 'result', 'in', 'increased', 'FGF20', 'expression', 'and']\n",
      "Counter({'NEU': 2, 'POS': 1})\n",
      "[(result, 'NEU'), (increased, 'POS'), (expression, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "full sentence The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased <e2>FGF20</e2> expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased <e2>FGF20</e2> expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 'full_sentence': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased <e2>FGF20</e2> expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (33, 40), 'entity_text': 'miR-433'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (81, 86), 'entity_text': 'FGF20'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d9.s1.p1\n",
      "IS:  ('MIR_GENE', 'UP')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased FGF20 expression and upregulation of <e2>alpha-synuclein</e2>, which could in turn promote dopaminergic neurons degeneration.\", 'full_sentence': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased FGF20 expression and upregulation of <e2>alpha-synuclein</e2>, which could in turn promote dopaminergic neurons degeneration.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (33, 40), 'entity_text': 'miR-433'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (118, 133), 'entity_text': 'alpha-synuclein'}}\n",
      "add special case miR-433\n",
      "add special case alpha-synuclein\n",
      "miR-433 6 alpha-synuclein 21\n",
      "(u\"The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 6, 21)\n",
      "geneword alpha-synuclein mirword miR-433\n",
      "Conjuncts [upregulation, expression]\n",
      "neighbour upregulation of\n",
      "neighbour upregulation of\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conjuncts [expression, upregulation]\n",
      "neighbour expression and\n",
      "neighbour expression and\n",
      "Conjunction before fc upregulation\n",
      "fc add children for upregulation\n",
      "Conjunctions ( conj )\n",
      "expression [in, increased, FGF20, expression, upregulation, of, alpha-synuclein]\n",
      "Conjuncts [reduction, affinity]\n",
      "neighbour reduction of\n",
      "neighbour reduction of\n",
      "Conjunction before fc affinity\n",
      "fc add children for affinity\n",
      "fc add children for affinity\n",
      "fc add children for affinity\n",
      "Conjuncts [affinity, miR-433]\n",
      "neighbour affinity of\n",
      "neighbour affinity of\n",
      "Conjunction before fc miR-433\n",
      "fc add children for miR-433\n",
      "Conjuncts [affinity, UTR]\n",
      "neighbour affinity of\n",
      "neighbour affinity of\n",
      "Conjunction before fc UTR\n",
      "fc add children for UTR\n",
      "fc add children for UTR\n",
      "Conjuncts [result, expression]\n",
      "neighbour result in\n",
      "neighbour result in\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conjuncts [upregulation, alpha-synuclein]\n",
      "neighbour upregulation of\n",
      "neighbour upregulation of\n",
      "Conjunction before fc alpha-synuclein\n",
      "fc add children for alpha-synuclein\n",
      "Conjuncts [promote, turn]\n",
      "neighbour promote dopaminergic\n",
      "neighbour promote dopaminergic\n",
      "Conjunction before fc turn\n",
      "fc add children for turn\n",
      "Conunct merge check: affinity expression\n",
      "UTR would\n",
      "Conunct merge check: expression alpha-synuclein\n",
      "expression and\n",
      "Double Conjunction Found\n",
      "Conunct merge check: alpha-synuclein turn\n",
      "alpha-synuclein, which could\n",
      "Conjunctions ( nmod )\n",
      "affinity [reduction, of, affinity, of, miR-433, to, 3, UTR]\n",
      "expression [result, in, increased, FGF20, expression]\n",
      "alpha-synuclein [upregulation, of, alpha-synuclein]\n",
      "turn [in, turn, promote]\n",
      "(alpha-synuclein, 'conj') [result, in, increased, FGF20, expression, upregulation, of, alpha-synuclein]\n",
      "SDP sent: The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "SDP     : [(miR-433, 'NOUN', 'nmod', False), (affinity, 'NOUN', 'nmod', False), (reduction, 'NOUN', 'nsubj', False), (result, 'VERB', 'ROOT', False), (expression, 'NOUN', 'nmod', True), (upregulation, 'NOUN', 'conj', False), (alpha-synuclein, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "0 [The, reduction, of, the, affinity, of, miR-433, to, the, 3, ', UTR, would, result, in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration] 31 0 The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, affinity, of, miR-433, to, the, 3, ', UTR]\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-433]\n",
      "DeepCheck Conj Subtree\n",
      "[to, the, 3, ', UTR]\n",
      "DeepCheck Conj Subtree\n",
      "[in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "DeepCheck Conj Subtree\n",
      "[upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "DeepCheck Conj Subtree\n",
      "[of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "DeepCheck Conj Subtree\n",
      "[in, turn]\n",
      "0 [The, reduction, of, the, affinity, of, miR-433, to, the, 3, ', UTR, would, result, in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration] 31 0 The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "Recognized overlapping entities: {Interval(118, 133, ('GENE_OR_GENE_PRODUCT', 'alpha-synuclein'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1, 'POS': 1})\n",
      "Compartment Text The reduction of the affinity of miR-433 to the 3 ' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein , which could in turn promote dopaminergic neurons degeneration\n",
      "[The, reduction, of, the, affinity, of, miR-433, to, the, 3, ', UTR, would, result, in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "TextBetween: The reduction of the affinity of miR-433 to the 3 ' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein , which could in turn promote dopaminergic neurons degeneration\n",
      "infer direction from token set [The, reduction, of, the, affinity, of, miR-433, to, the, 3, ', UTR, would, result, in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "stemTypes2Count: Counter({'POS': 3, 'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbecba9e1e0>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'POS': 2})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbecba9e2f0>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count [(result, 'NEU'), (increased, 'POS'), (expression, 'NEU'), (upregulation, 'POS')]\n",
      "Between Tokens [(result, 'NEU'), (increased, 'POS'), (expression, 'NEU'), (upregulation, 'POS')]\n",
      "counts between final False\n",
      "['affinity', 'of', 'miR-433', 'to', 'the', '3', \"'\", 'UTR', 'would', 'result', 'in', 'increased', 'FGF20', 'expression', 'and', 'upregulation', 'of', 'alpha-synuclein', ',', 'which']\n",
      "Counter({'NEU': 2, 'POS': 2})\n",
      "[(result, 'NEU'), (increased, 'POS'), (expression, 'NEU'), (upregulation, 'POS')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "full sentence The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased FGF20 expression and upregulation of <e2>alpha-synuclein</e2>, which could in turn promote dopaminergic neurons degeneration.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased FGF20 expression and upregulation of <e2>alpha-synuclein</e2>, which could in turn promote dopaminergic neurons degeneration.\", 'full_sentence': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased FGF20 expression and upregulation of <e2>alpha-synuclein</e2>, which could in turn promote dopaminergic neurons degeneration.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (33, 40), 'entity_text': 'miR-433'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (118, 133), 'entity_text': 'alpha-synuclein'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d18.s1.p0\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case microRNA-30\n",
      "add special case Oas1\n",
      "microRNA-30 12 Oas1 18\n",
      "(u\"Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\", 12, 18)\n",
      "geneword Oas1 mirword microRNA-30\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [Modification, construct]\n",
      "neighbour Modification of\n",
      "neighbour Modification of\n",
      "Conjunction before fc construct\n",
      "fc add children for construct\n",
      "fc add children for construct\n",
      "Conjuncts [features, precursor]\n",
      "neighbour features of\n",
      "neighbour features of\n",
      "Conjunction before fc precursor\n",
      "fc add children for precursor\n",
      "fc add children for precursor\n",
      "fc add children for precursor\n",
      "fc add children for precursor\n",
      "Conjuncts [induction, cultures]\n",
      "neighbour induction in\n",
      "neighbour induction in\n",
      "Conjunction before fc cultures\n",
      "fc add children for cultures\n",
      "fc add children for cultures\n",
      "fc add children for cultures\n",
      "Conjuncts [modification, strand]\n",
      "neighbour modification of\n",
      "neighbour modification of\n",
      "Conjunction before fc strand\n",
      "fc add children for strand\n",
      "fc add children for strand\n",
      "Conunct merge check: construct precursor\n",
      "construct by implementing\n",
      "Conunct merge check: precursor cultures\n",
      "precursor avoided Oas1\n",
      "Conunct merge check: cultures strand\n",
      "cultures, whereby\n",
      "Conjunctions ( nmod )\n",
      "construct [Modification, of, shRNA, construct]\n",
      "precursor [features, of, occurring, microRNA-30, miR-30, precursor]\n",
      "cultures [induction, in, transduced, primary, cultures]\n",
      "strand [modification, of, passenger, strand]\n",
      "SDP sent: Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "SDP     : [(microRNA-30, 'ADJ', 'compound', False), (precursor, 'NOUN', 'nmod', False), (features, 'NOUN', 'dobj', False), (implementing, 'VERB', 'advcl', False), (Modification, 'NOUN', 'nsubj', False), (avoided, 'VERB', 'ROOT', False), (induction, 'NOUN', 'dobj', False), (Oas1, 'NOUN', 'compound', False)]\n",
      "SDP DOBJ RULE\n",
      "deepcheck connection rule [by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor]\n",
      "deepcheck connection rule noun found seems modification [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression]\n",
      "0 [Modification, of, the, shRNA, construct, by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor, avoided, Oas1, induction, in, transduced, primary, cultures] 24 0 Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "1 [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression] 17 174 whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, shRNA, construct]\n",
      "deepcheck connection rule [by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor]\n",
      "DeepCheck Conj Subtree\n",
      "[in, transduced, primary, cultures]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, passenger, strand]\n",
      "deepcheck connection rule noun found seems modification [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression]\n",
      "0 [Modification, of, the, shRNA, construct, by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor, avoided, Oas1, induction, in, transduced, primary, cultures] 24 0 Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "1 [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression] 17 174 whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "Recognized overlapping entities: {Interval(127, 131, ('GENE_OR_GENE_PRODUCT', 'Oas1'))}\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "Compartment Text Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 ( miR-30 ) precursor avoided Oas1 induction in transduced primary cultures\n",
      "[Modification, of, the, shRNA, construct, by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor, avoided, Oas1, induction, in, transduced, primary, cultures]\n",
      "TextBetween: Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 ( miR-30 ) precursor avoided Oas1 induction in transduced primary cultures\n",
      "infer direction from token set [Modification, of, the, shRNA, construct, by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor, avoided, Oas1, induction, in, transduced, primary, cultures]\n",
      "stemTypes2Count: Counter({'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e27b8>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'POS': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e2048>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count [(induction, 'POS')]\n",
      "Between Tokens []\n",
      "counts between final False\n",
      "['naturally', 'occurring', 'microRNA-30', '(', 'miR-30', ')', 'precursor', 'avoided', 'Oas1', 'induction', 'in']\n",
      "Counter({'POS': 1})\n",
      "[(induction, 'POS')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "full sentence Modification of the shRNA construct by implementing features of the naturally occurring <e1>microRNA-30</e1> (miR-30) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Modification of the shRNA construct by implementing features of the naturally occurring <e1>microRNA-30</e1> (miR-30) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.', 'full_sentence': 'Modification of the shRNA construct by implementing features of the naturally occurring <e1>microRNA-30</e1> (miR-30) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (88, 99), 'entity_text': 'microRNA-30'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (127, 131), 'entity_text': 'Oas1'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d18.s1.p1\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miR-30\n",
      "add special case Oas1\n",
      "miR-30 14 Oas1 18\n",
      "(u\"Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\", 14, 18)\n",
      "geneword Oas1 mirword miR-30\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [Modification, construct]\n",
      "neighbour Modification of\n",
      "neighbour Modification of\n",
      "Conjunction before fc construct\n",
      "fc add children for construct\n",
      "fc add children for construct\n",
      "Conjuncts [features, precursor]\n",
      "neighbour features of\n",
      "neighbour features of\n",
      "Conjunction before fc precursor\n",
      "fc add children for precursor\n",
      "fc add children for precursor\n",
      "fc add children for precursor\n",
      "fc add children for precursor\n",
      "Conjuncts [induction, cultures]\n",
      "neighbour induction in\n",
      "neighbour induction in\n",
      "Conjunction before fc cultures\n",
      "fc add children for cultures\n",
      "fc add children for cultures\n",
      "fc add children for cultures\n",
      "Conjuncts [modification, strand]\n",
      "neighbour modification of\n",
      "neighbour modification of\n",
      "Conjunction before fc strand\n",
      "fc add children for strand\n",
      "fc add children for strand\n",
      "Conunct merge check: construct precursor\n",
      "construct by implementing\n",
      "Conunct merge check: precursor cultures\n",
      "precursor avoided Oas1\n",
      "Conunct merge check: cultures strand\n",
      "cultures, whereby\n",
      "Conjunctions ( nmod )\n",
      "construct [Modification, of, shRNA, construct]\n",
      "precursor [features, of, occurring, microRNA-30, miR-30, precursor]\n",
      "cultures [induction, in, transduced, primary, cultures]\n",
      "strand [modification, of, passenger, strand]\n",
      "SDP sent: Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "SDP     : [(miR-30, 'NOUN', 'compound', False), (precursor, 'NOUN', 'nmod', False), (features, 'NOUN', 'dobj', False), (implementing, 'VERB', 'advcl', False), (Modification, 'NOUN', 'nsubj', False), (avoided, 'VERB', 'ROOT', False), (induction, 'NOUN', 'dobj', False), (Oas1, 'NOUN', 'compound', False)]\n",
      "SDP DOBJ RULE\n",
      "deepcheck connection rule [by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor]\n",
      "deepcheck connection rule noun found seems modification [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression]\n",
      "0 [Modification, of, the, shRNA, construct, by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor, avoided, Oas1, induction, in, transduced, primary, cultures] 24 0 Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "1 [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression] 17 174 whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, shRNA, construct]\n",
      "deepcheck connection rule [by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor]\n",
      "DeepCheck Conj Subtree\n",
      "[in, transduced, primary, cultures]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, passenger, strand]\n",
      "deepcheck connection rule noun found seems modification [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression]\n",
      "0 [Modification, of, the, shRNA, construct, by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor, avoided, Oas1, induction, in, transduced, primary, cultures] 24 0 Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "1 [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression] 17 174 whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "Recognized overlapping entities: {Interval(127, 131, ('GENE_OR_GENE_PRODUCT', 'Oas1'))}\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "Compartment Text Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 ( miR-30 ) precursor avoided Oas1 induction in transduced primary cultures\n",
      "[Modification, of, the, shRNA, construct, by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor, avoided, Oas1, induction, in, transduced, primary, cultures]\n",
      "TextBetween: Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 ( miR-30 ) precursor avoided Oas1 induction in transduced primary cultures\n",
      "infer direction from token set [Modification, of, the, shRNA, construct, by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor, avoided, Oas1, induction, in, transduced, primary, cultures]\n",
      "stemTypes2Count: Counter({'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbecba9e1e0>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'POS': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbecba9e2f0>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count [(induction, 'POS')]\n",
      "Between Tokens []\n",
      "counts between final False\n",
      "['microRNA-30', '(', 'miR-30', ')', 'precursor', 'avoided', 'Oas1', 'induction', 'in']\n",
      "Counter({'POS': 1})\n",
      "[(induction, 'POS')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "full sentence Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (<e1>miR-30</e1>) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (<e1>miR-30</e1>) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.', 'full_sentence': 'Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (<e1>miR-30</e1>) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (101, 107), 'entity_text': 'miR-30'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (127, 131), 'entity_text': 'Oas1'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d40.s5.p3\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case mir-146a\n",
      "add special case CD4\n",
      "mir-146a 6 CD4/CCR5-mediated 24\n",
      "(u\"Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication.\", 6, 24)\n",
      "geneword CD4/CCR5-mediated mirword mir-146a\n",
      "Conjuncts [inhibitor, receptor]\n",
      "neighbour inhibitor of\n",
      "neighbour inhibitor of\n",
      "Conjunction before fc receptor\n",
      "fc add children for receptor\n",
      "fc add children for receptor\n",
      "fc add children for receptor\n",
      "Conjuncts [receptor, inhibitor]\n",
      "neighbour receptor and\n",
      "neighbour receptor and\n",
      "Conjunction before fc inhibitor\n",
      "fc add children for inhibitor\n",
      "fc add children for inhibitor\n",
      "Conjuncts [replication, entry]\n",
      "neighbour replication .\n",
      "neighbour replication None\n",
      "Conjunction before fc entry\n",
      "fc add children for entry\n",
      "fc add children for entry\n",
      "fc add children for entry\n",
      "Conjuncts [entry, replication]\n",
      "neighbour entry and\n",
      "neighbour entry and\n",
      "Conjunction before fc replication\n",
      "Conjunctions ( conj )\n",
      "receptor [for, CCR5, chemokine, receptor, potent, inhibitor, of, CD4/CCR5-mediated, HIV-1, entry, replication]\n",
      "Conjuncts [ligand, targets]\n",
      "neighbour ligand for\n",
      "neighbour ligand for\n",
      "Conjunction before fc targets\n",
      "fc add children for targets\n",
      "fc add children for targets\n",
      "fc add children for targets\n",
      "fc add children for targets\n",
      "Conjuncts [targets, mir-146a]\n",
      "neighbour targets for\n",
      "neighbour targets for\n",
      "Conjunction before fc mir-146a\n",
      "fc add children for mir-146a\n",
      "Conjuncts [ligand, receptor]\n",
      "neighbour ligand for\n",
      "neighbour ligand for\n",
      "Conjunction before fc receptor\n",
      "fc add children for receptor\n",
      "fc add children for receptor\n",
      "fc add children for receptor\n",
      "Conjuncts [inhibitor, entry]\n",
      "neighbour inhibitor of\n",
      "neighbour inhibitor of\n",
      "Conjunction before fc entry\n",
      "fc add children for entry\n",
      "fc add children for entry\n",
      "fc add children for entry\n",
      "Conunct merge check: targets entry\n",
      "receptor and a potent\n",
      "Double Conjunction Found\n",
      "Conjunctions ( nmod )\n",
      "targets [Among, predicted, gene, targets, for, mir-146a, ligand, for, CCR5, chemokine, receptor]\n",
      "entry [inhibitor, of, CD4/CCR5-mediated, HIV-1, entry]\n",
      "(entry, 'conj') [Among, predicted, gene, targets, for, mir-146a, ligand, for, CCR5, chemokine, receptor, inhibitor, of, CD4/CCR5-mediated, HIV-1, entry]\n",
      "achildren {mir-146a}\n",
      "achildren False\n",
      "SDP sent: Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication.\n",
      "SDP     : [(mir-146a, 'NOUN', 'nmod', False), (targets, 'NOUN', 'nmod', False), (ligand, 'NOUN', 'ROOT', False), (receptor, 'NOUN', 'nmod', True), (inhibitor, 'NOUN', 'conj', False), (entry, 'NOUN', 'nmod', False), (CD4/CCR5-mediated, 'ADJ', 'amod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[a, potent, inhibitor, of, CD4/CCR5-mediated, HIV-1, entry, and, replication]\n",
      "DeepCheck Conj Subtree\n",
      "[replication]\n",
      "0 [Among, the, predicted, gene, targets, for, mir-146a, ,, the, chemokine, CCL8/MCP-2, is, a, ligand, for, the, CCR5, chemokine, receptor, and, a, potent, inhibitor, of, CD4/CCR5-mediated, HIV-1, entry, and, replication] 29 0 Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication.\n",
      "DeepCheck Conj Subtree\n",
      "[Among, the, predicted, gene, targets, for, mir-146a]\n",
      "DeepCheck Conj Subtree\n",
      "[for, mir-146a]\n",
      "DeepCheck Conj Subtree\n",
      "[for, the, CCR5, chemokine, receptor, and, a, potent, inhibitor, of, CD4/CCR5-mediated, HIV-1, entry, and, replication]\n",
      "DeepCheck Conj Subtree\n",
      "[a, potent, inhibitor, of, CD4/CCR5-mediated, HIV-1, entry, and, replication]\n",
      "DeepCheck Conj Subtree\n",
      "[of, CD4/CCR5-mediated, HIV-1, entry, and, replication]\n",
      "DeepCheck Conj Subtree\n",
      "[replication]\n",
      "0 [Among, the, predicted, gene, targets, for, mir-146a, ,, the, chemokine, CCL8/MCP-2, is, a, ligand, for, the, CCR5, chemokine, receptor, and, a, potent, inhibitor, of, CD4/CCR5-mediated, HIV-1, entry, and, replication] 29 0 Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication.\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text Among the predicted gene targets for mir-146a , the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication\n",
      "[Among, the, predicted, gene, targets, for, mir-146a, ,, the, chemokine, CCL8/MCP-2, is, a, ligand, for, the, CCR5, chemokine, receptor, and, a, potent, inhibitor, of, CD4/CCR5-mediated, HIV-1, entry, and, replication]\n",
      "TextBetween: Among the predicted gene targets for mir-146a , the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication\n",
      "infer direction from token set [Among, the, predicted, gene, targets, for, mir-146a, ,, the, chemokine, CCL8/MCP-2, is, a, ligand, for, the, CCR5, chemokine, receptor, and, a, potent, inhibitor, of, CD4/CCR5-mediated, HIV-1, entry, and, replication]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c048>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 3})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c8c8>, {'mir': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(targets, 'NEU'), (ligand, 'NEU'), (receptor, 'NEU')]\n",
      "Between Tokens [(ligand, 'NEU'), (receptor, 'NEU')]\n",
      "counts between final False\n",
      "['targets', 'for', 'mir-146a', ',', 'the', 'chemokine', 'CCL8/MCP-2', 'is', 'a', 'ligand', 'for', 'the', 'CCR5', 'chemokine', 'receptor', 'and', 'a', 'potent', 'inhibitor', 'of', 'CD4/CCR5-mediated', 'HIV-1', 'entry']\n",
      "Counter({'NEU': 3})\n",
      "[(targets, 'NEU'), (ligand, 'NEU'), (receptor, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication.\n",
      "full sentence Among the predicted gene targets for <e1>mir-146a</e1>, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of <e2>CD4</e2>/CCR5-mediated HIV-1 entry and replication.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Among the predicted gene targets for <e1>mir-146a</e1>, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of <e2>CD4</e2>/CCR5-mediated HIV-1 entry and replication.', 'full_sentence': 'Among the predicted gene targets for <e1>mir-146a</e1>, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of <e2>CD4</e2>/CCR5-mediated HIV-1 entry and replication.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (37, 45), 'entity_text': 'mir-146a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (142, 145), 'entity_text': 'CD4'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d40.s5.p4\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case mir-146a\n",
      "add special case CCR5\n",
      "mir-146a 6 CD4/CCR5-mediated 24\n",
      "(u\"Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication.\", 6, 24)\n",
      "geneword CD4/CCR5-mediated mirword mir-146a\n",
      "Conjuncts [inhibitor, receptor]\n",
      "neighbour inhibitor of\n",
      "neighbour inhibitor of\n",
      "Conjunction before fc receptor\n",
      "fc add children for receptor\n",
      "fc add children for receptor\n",
      "fc add children for receptor\n",
      "Conjuncts [receptor, inhibitor]\n",
      "neighbour receptor and\n",
      "neighbour receptor and\n",
      "Conjunction before fc inhibitor\n",
      "fc add children for inhibitor\n",
      "fc add children for inhibitor\n",
      "Conjuncts [replication, entry]\n",
      "neighbour replication .\n",
      "neighbour replication None\n",
      "Conjunction before fc entry\n",
      "fc add children for entry\n",
      "fc add children for entry\n",
      "fc add children for entry\n",
      "Conjuncts [entry, replication]\n",
      "neighbour entry and\n",
      "neighbour entry and\n",
      "Conjunction before fc replication\n",
      "Conjunctions ( conj )\n",
      "receptor [for, CCR5, chemokine, receptor, potent, inhibitor, of, CD4/CCR5-mediated, HIV-1, entry, replication]\n",
      "Conjuncts [ligand, targets]\n",
      "neighbour ligand for\n",
      "neighbour ligand for\n",
      "Conjunction before fc targets\n",
      "fc add children for targets\n",
      "fc add children for targets\n",
      "fc add children for targets\n",
      "fc add children for targets\n",
      "Conjuncts [targets, mir-146a]\n",
      "neighbour targets for\n",
      "neighbour targets for\n",
      "Conjunction before fc mir-146a\n",
      "fc add children for mir-146a\n",
      "Conjuncts [ligand, receptor]\n",
      "neighbour ligand for\n",
      "neighbour ligand for\n",
      "Conjunction before fc receptor\n",
      "fc add children for receptor\n",
      "fc add children for receptor\n",
      "fc add children for receptor\n",
      "Conjuncts [inhibitor, entry]\n",
      "neighbour inhibitor of\n",
      "neighbour inhibitor of\n",
      "Conjunction before fc entry\n",
      "fc add children for entry\n",
      "fc add children for entry\n",
      "fc add children for entry\n",
      "Conunct merge check: targets entry\n",
      "receptor and a potent\n",
      "Double Conjunction Found\n",
      "Conjunctions ( nmod )\n",
      "targets [Among, predicted, gene, targets, for, mir-146a, ligand, for, CCR5, chemokine, receptor]\n",
      "entry [inhibitor, of, CD4/CCR5-mediated, HIV-1, entry]\n",
      "(entry, 'conj') [Among, predicted, gene, targets, for, mir-146a, ligand, for, CCR5, chemokine, receptor, inhibitor, of, CD4/CCR5-mediated, HIV-1, entry]\n",
      "achildren {mir-146a}\n",
      "achildren False\n",
      "SDP sent: Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication.\n",
      "SDP     : [(mir-146a, 'NOUN', 'nmod', False), (targets, 'NOUN', 'nmod', False), (ligand, 'NOUN', 'ROOT', False), (receptor, 'NOUN', 'nmod', True), (inhibitor, 'NOUN', 'conj', False), (entry, 'NOUN', 'nmod', False), (CD4/CCR5-mediated, 'ADJ', 'amod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[a, potent, inhibitor, of, CD4/CCR5-mediated, HIV-1, entry, and, replication]\n",
      "DeepCheck Conj Subtree\n",
      "[replication]\n",
      "0 [Among, the, predicted, gene, targets, for, mir-146a, ,, the, chemokine, CCL8/MCP-2, is, a, ligand, for, the, CCR5, chemokine, receptor, and, a, potent, inhibitor, of, CD4/CCR5-mediated, HIV-1, entry, and, replication] 29 0 Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication.\n",
      "DeepCheck Conj Subtree\n",
      "[Among, the, predicted, gene, targets, for, mir-146a]\n",
      "DeepCheck Conj Subtree\n",
      "[for, mir-146a]\n",
      "DeepCheck Conj Subtree\n",
      "[for, the, CCR5, chemokine, receptor, and, a, potent, inhibitor, of, CD4/CCR5-mediated, HIV-1, entry, and, replication]\n",
      "DeepCheck Conj Subtree\n",
      "[a, potent, inhibitor, of, CD4/CCR5-mediated, HIV-1, entry, and, replication]\n",
      "DeepCheck Conj Subtree\n",
      "[of, CD4/CCR5-mediated, HIV-1, entry, and, replication]\n",
      "DeepCheck Conj Subtree\n",
      "[replication]\n",
      "0 [Among, the, predicted, gene, targets, for, mir-146a, ,, the, chemokine, CCL8/MCP-2, is, a, ligand, for, the, CCR5, chemokine, receptor, and, a, potent, inhibitor, of, CD4/CCR5-mediated, HIV-1, entry, and, replication] 29 0 Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication.\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text Among the predicted gene targets for mir-146a , the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication\n",
      "[Among, the, predicted, gene, targets, for, mir-146a, ,, the, chemokine, CCL8/MCP-2, is, a, ligand, for, the, CCR5, chemokine, receptor, and, a, potent, inhibitor, of, CD4/CCR5-mediated, HIV-1, entry, and, replication]\n",
      "TextBetween: Among the predicted gene targets for mir-146a , the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication\n",
      "infer direction from token set [Among, the, predicted, gene, targets, for, mir-146a, ,, the, chemokine, CCL8/MCP-2, is, a, ligand, for, the, CCR5, chemokine, receptor, and, a, potent, inhibitor, of, CD4/CCR5-mediated, HIV-1, entry, and, replication]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbecba9e158>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 3})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62ae8>, {'mir': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(targets, 'NEU'), (ligand, 'NEU'), (receptor, 'NEU')]\n",
      "Between Tokens [(ligand, 'NEU'), (receptor, 'NEU')]\n",
      "counts between final False\n",
      "['targets', 'for', 'mir-146a', ',', 'the', 'chemokine', 'CCL8/MCP-2', 'is', 'a', 'ligand', 'for', 'the', 'CCR5', 'chemokine', 'receptor', 'and', 'a', 'potent', 'inhibitor', 'of', 'CD4/CCR5-mediated', 'HIV-1', 'entry']\n",
      "Counter({'NEU': 3})\n",
      "[(targets, 'NEU'), (ligand, 'NEU'), (receptor, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Among the predicted gene targets for mir-146a, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/CCR5-mediated HIV-1 entry and replication.\n",
      "full sentence Among the predicted gene targets for <e1>mir-146a</e1>, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/<e2>CCR5</e2>-mediated HIV-1 entry and replication.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Among the predicted gene targets for <e1>mir-146a</e1>, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/<e2>CCR5</e2>-mediated HIV-1 entry and replication.', 'full_sentence': 'Among the predicted gene targets for <e1>mir-146a</e1>, the chemokine CCL8/MCP-2 is a ligand for the CCR5 chemokine receptor and a potent inhibitor of CD4/<e2>CCR5</e2>-mediated HIV-1 entry and replication.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (37, 45), 'entity_text': 'mir-146a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (146, 150), 'entity_text': 'CCR5'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d48.s1.p3\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case miR-221/222\n",
      "add special case p27Kip1\n",
      "miR-221/222 2 p27Kip1 25\n",
      "(u\"Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\", 2, 25)\n",
      "geneword p27Kip1 mirword miR-221/222\n",
      "Conjuncts [reduced, p27Kip1]\n",
      "neighbour reduced tumor\n",
      "neighbour reduced tumor\n",
      "Conjunction before fc p27Kip1\n",
      "fc add children for p27Kip1\n",
      "fc add children for p27Kip1\n",
      "Conjuncts [p27Kip1, reduced]\n",
      "neighbour p27Kip1 in\n",
      "neighbour p27Kip1 in\n",
      "Conjunction before fc reduced\n",
      "fc add children for reduced\n",
      "fc add children for reduced\n",
      "Conjunctions ( conj )\n",
      "p27Kip1 [increased, p27Kip1, in, U251, glioma, subcutaneous, mice, reduced, tumor, growth, through, up, regulation, of, p27Kip1]\n",
      "Conjuncts [knocked-down, 2'-OME-oligonucleotides]\n",
      "neighbour knocked-down through\n",
      "neighbour knocked-down through\n",
      "Conjunction before fc 2'-OME-oligonucleotides\n",
      "fc add children for 2'-OME-oligonucleotides\n",
      "fc add children for 2'-OME-oligonucleotides\n",
      "Conjuncts [p27Kip1, mice]\n",
      "neighbour p27Kip1 in\n",
      "neighbour p27Kip1 in\n",
      "Conjunction before fc mice\n",
      "fc add children for mice\n",
      "fc add children for mice\n",
      "fc add children for mice\n",
      "fc add children for mice\n",
      "Conjuncts [reduced, regulation]\n",
      "neighbour reduced tumor\n",
      "neighbour reduced tumor\n",
      "Conjunction before fc regulation\n",
      "fc add children for regulation\n",
      "fc add children for regulation\n",
      "Conjuncts [regulation, p27Kip1]\n",
      "neighbour regulation of\n",
      "neighbour regulation of\n",
      "Conjunction before fc p27Kip1\n",
      "fc add children for p27Kip1\n",
      "Conunct merge check: 2'-OME-oligonucleotides mice\n",
      "2'-OME-oligonucleotides increased\n",
      "Conunct merge check: mice regulation\n",
      "mice and strongly\n",
      "Double Conjunction Found\n",
      "Conjunctions ( nmod )\n",
      "2'-OME-oligonucleotides [knocked-down, through, antisense, 2'-OME-oligonucleotides]\n",
      "mice [p27Kip1, in, U251, glioma, subcutaneous, mice]\n",
      "regulation [reduced, through, up, regulation, of, p27Kip1]\n",
      "(regulation, 'conj') [p27Kip1, in, U251, glioma, subcutaneous, mice, reduced, through, up, regulation, of, p27Kip1]\n",
      "SDP sent: Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "SDP     : [(miR-221/222, 'NOUN', 'amod', False), (knocked-down, 'VERB', 'nsubj', False), (increased, 'VERB', 'amod', False), (p27Kip1, 'NOUN', 'ROOT', True), (reduced, 'VERB', 'conj', False), (regulation, 'NOUN', 'nmod', False), (p27Kip1, 'NOUN', 'nmod', False)]\n",
      "0 [Consistently, ,, miR-221/222, knocked-down, through, antisense, 2'-OME-oligonucleotides, increased, p27Kip1, in, U251, glioma, subcutaneous, mice] 14 0 Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "1 [strongly, reduced, tumor, growth, in, vivo, through, up, regulation, of, p27Kip1] 11 136 strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "DeepCheck Conj Subtree\n",
      "[through, antisense, 2'-OME-oligonucleotides]\n",
      "DeepCheck Conj Subtree\n",
      "[in, U251, glioma, subcutaneous, mice]\n",
      "DeepCheck Conj Subtree\n",
      "[through, up, regulation, of, p27Kip1]\n",
      "DeepCheck Conj Subtree\n",
      "[of, p27Kip1]\n",
      "0 [Consistently, ,, miR-221/222, knocked-down, through, antisense, 2'-OME-oligonucleotides, increased, p27Kip1, in, U251, glioma, subcutaneous, mice] 14 0 Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "1 [strongly, reduced, tumor, growth, in, vivo, through, up, regulation, of, p27Kip1] 11 136 strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "Recognized overlapping entities: {Interval(199, 206, ('GENE_OR_GENE_PRODUCT', 'p27Kip1'))}\n",
      "passive? False\n",
      "Counter({'NEG': 2, 'POS': 1})\n",
      "Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "TextBetween: Consistently , miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1 .\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between mir gene', 'reg_detect_major': 'between'}})\n",
      " hit sentence Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "full sentence Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between mir gene', 'reg_detect_major': 'between'}}, 'tagged_sent': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\", 'full_sentence': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (14, 25), 'entity_text': 'miR-221/222'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (199, 206), 'entity_text': 'p27Kip1'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d48.s1.p3\n",
      "IS:  ('MIR_GENE', 'NEU')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between mir gene', 'reg_detect_major': 'between'}}, 'tagged_sent': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\", 'full_sentence': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (14, 25), 'entity_text': 'miR-221/222'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (199, 206), 'entity_text': 'p27Kip1'}}\n",
      "add special case miR-221/222\n",
      "add special case p27Kip1\n",
      "miR-221/222 2 p27Kip1 25\n",
      "(u\"Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\", 2, 25)\n",
      "geneword p27Kip1 mirword miR-221/222\n",
      "Conjuncts [reduced, p27Kip1]\n",
      "neighbour reduced tumor\n",
      "neighbour reduced tumor\n",
      "Conjunction before fc p27Kip1\n",
      "fc add children for p27Kip1\n",
      "fc add children for p27Kip1\n",
      "Conjuncts [p27Kip1, reduced]\n",
      "neighbour p27Kip1 in\n",
      "neighbour p27Kip1 in\n",
      "Conjunction before fc reduced\n",
      "fc add children for reduced\n",
      "fc add children for reduced\n",
      "Conjunctions ( conj )\n",
      "p27Kip1 [increased, p27Kip1, in, U251, glioma, subcutaneous, mice, reduced, tumor, growth, through, up, regulation, of, p27Kip1]\n",
      "Conjuncts [knocked-down, 2'-OME-oligonucleotides]\n",
      "neighbour knocked-down through\n",
      "neighbour knocked-down through\n",
      "Conjunction before fc 2'-OME-oligonucleotides\n",
      "fc add children for 2'-OME-oligonucleotides\n",
      "fc add children for 2'-OME-oligonucleotides\n",
      "Conjuncts [p27Kip1, mice]\n",
      "neighbour p27Kip1 in\n",
      "neighbour p27Kip1 in\n",
      "Conjunction before fc mice\n",
      "fc add children for mice\n",
      "fc add children for mice\n",
      "fc add children for mice\n",
      "fc add children for mice\n",
      "Conjuncts [reduced, regulation]\n",
      "neighbour reduced tumor\n",
      "neighbour reduced tumor\n",
      "Conjunction before fc regulation\n",
      "fc add children for regulation\n",
      "fc add children for regulation\n",
      "Conjuncts [regulation, p27Kip1]\n",
      "neighbour regulation of\n",
      "neighbour regulation of\n",
      "Conjunction before fc p27Kip1\n",
      "fc add children for p27Kip1\n",
      "Conunct merge check: 2'-OME-oligonucleotides mice\n",
      "2'-OME-oligonucleotides increased\n",
      "Conunct merge check: mice regulation\n",
      "mice and strongly\n",
      "Double Conjunction Found\n",
      "Conjunctions ( nmod )\n",
      "2'-OME-oligonucleotides [knocked-down, through, antisense, 2'-OME-oligonucleotides]\n",
      "mice [p27Kip1, in, U251, glioma, subcutaneous, mice]\n",
      "regulation [reduced, through, up, regulation, of, p27Kip1]\n",
      "(regulation, 'conj') [p27Kip1, in, U251, glioma, subcutaneous, mice, reduced, through, up, regulation, of, p27Kip1]\n",
      "SDP sent: Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "SDP     : [(miR-221/222, 'NOUN', 'amod', False), (knocked-down, 'VERB', 'nsubj', False), (increased, 'VERB', 'amod', False), (p27Kip1, 'NOUN', 'ROOT', True), (reduced, 'VERB', 'conj', False), (regulation, 'NOUN', 'nmod', False), (p27Kip1, 'NOUN', 'nmod', False)]\n",
      "0 [Consistently, ,, miR-221/222, knocked-down, through, antisense, 2'-OME-oligonucleotides, increased, p27Kip1, in, U251, glioma, subcutaneous, mice] 14 0 Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "1 [strongly, reduced, tumor, growth, in, vivo, through, up, regulation, of, p27Kip1] 11 136 strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "DeepCheck Conj Subtree\n",
      "[through, antisense, 2'-OME-oligonucleotides]\n",
      "DeepCheck Conj Subtree\n",
      "[in, U251, glioma, subcutaneous, mice]\n",
      "DeepCheck Conj Subtree\n",
      "[through, up, regulation, of, p27Kip1]\n",
      "DeepCheck Conj Subtree\n",
      "[of, p27Kip1]\n",
      "0 [Consistently, ,, miR-221/222, knocked-down, through, antisense, 2'-OME-oligonucleotides, increased, p27Kip1, in, U251, glioma, subcutaneous, mice] 14 0 Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "1 [strongly, reduced, tumor, growth, in, vivo, through, up, regulation, of, p27Kip1] 11 136 strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "Recognized overlapping entities: {Interval(199, 206, ('GENE_OR_GENE_PRODUCT', 'p27Kip1'))}\n",
      "passive? False\n",
      "Counter({'NEG': 2, 'POS': 1})\n",
      "Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "TextBetween: Consistently , miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1 .\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between mir gene', 'reg_detect_major': 'between'}})\n",
      " hit sentence Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "full sentence Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between mir gene', 'reg_detect_major': 'between'}}, 'tagged_sent': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\", 'full_sentence': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (14, 25), 'entity_text': 'miR-221/222'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (199, 206), 'entity_text': 'p27Kip1'}}\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "The data strongly suggested that a regulatory loop between miR-21 and STAT3 might provide an insight into the mechanism of modulating EGFR/STAT3 signaling. \n",
      "('miR-21', 'Specific_miRNAs', (59, 65)) ('STAT3', 'Genes/Proteins', (70, 75))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We identified an evolutionarily conserved binding site for microRNA-29b (miR-29b) in the 3' untranslated region (3'UTR) of the human PGRN (hPGRN) mRNA. \n",
      "('microRNA-29b', 'Specific_miRNAs', (59, 71)) ('GRN', 'Genes/Proteins', (133, 137))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We identified an evolutionarily conserved binding site for microRNA-29b (miR-29b) in the 3' untranslated region (3'UTR) of the human PGRN (hPGRN) mRNA. \n",
      "('miR-29b', 'Specific_miRNAs', (73, 80)) ('GRN', 'Genes/Proteins', (133, 137))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d54.s9.p2\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case FASLG\n",
      "add special case miR-21\n",
      "FASLG 17 miR-21 36\n",
      "(u\"Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\", 17, 36)\n",
      "geneword FASLG mirword miR-21\n",
      "Conjuncts [mediates, protects]\n",
      "neighbour mediates the\n",
      "neighbour mediates the\n",
      "Conjunction before fc protects\n",
      "fc add children for protects\n",
      "Conjuncts [ligand, FASLG]\n",
      "neighbour ligand whose\n",
      "neighbour ligand whose\n",
      "Conjunction before fc FASLG\n",
      "fc add children for FASLG\n",
      "fc add children for FASLG\n",
      "Conjuncts [FASLG, ligand]\n",
      "neighbour FASLG ,\n",
      "neighbour FASLG a\n",
      "Conjunction before fc ligand\n",
      "fc add children for ligand\n",
      "fc add children for ligand\n",
      "fc add children for ligand\n",
      "Conjuncts [protects, mediates]\n",
      "neighbour protects against\n",
      "neighbour protects against\n",
      "Conjunction before fc mediates\n",
      "fc add children for mediates\n",
      "Conunct merge check: protects FASLG\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( conj )\n",
      "protects [protects, against, ischemic, neuronal, death, mediates, neuroprotective, effect]\n",
      "FASLG [of, FASLG, a, tumor, necrosis, factor-, family, member, important, cell, death-inducing, ligand]\n",
      "Conjuncts [overexpression, miR-21]\n",
      "neighbour overexpression of\n",
      "neighbour overexpression of\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conjuncts [protects, death]\n",
      "neighbour protects against\n",
      "neighbour protects against\n",
      "Conjunction before fc death\n",
      "fc add children for death\n",
      "fc add children for death\n",
      "fc add children for death\n",
      "Conjuncts [downregulation, FASLG]\n",
      "neighbour downregulation of\n",
      "neighbour downregulation of\n",
      "Conjunction before fc FASLG\n",
      "fc add children for FASLG\n",
      "fc add children for FASLG\n",
      "Conjuncts [targeted, miR-21]\n",
      "neighbour targeted by\n",
      "neighbour targeted by\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conunct merge check: miR-21 death\n",
      "miR-21\n",
      "Conunct merge check: death FASLG\n",
      "death, and that\n",
      "Double Conjunction Found\n",
      "Conunct merge check: FASLG miR-21\n",
      "member and an important cell death-inducing ligand whose gene is\n",
      "Conjunctions ( nmod )\n",
      "miR-21 [overexpression, of, miR-21]\n",
      "death [protects, against, ischemic, neuronal, death]\n",
      "FASLG [downregulation, of, FASLG, a, tumor, necrosis, factor-, family, member]\n",
      "miR-21 [targeted, by, miR-21]\n",
      "(FASLG, 'conj') [protects, against, ischemic, neuronal, death, downregulation, of, FASLG, a, tumor, necrosis, factor-, family, member]\n",
      "SDP sent: Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "SDP     : [(FASLG, 'NOUN', 'nmod', False), (indicate, 'VERB', 'ROOT', False), (protects, 'VERB', 'ccomp', True), (mediates, 'VERB', 'conj', False), (downregulation, 'NOUN', 'nsubj', False), (FASLG, 'NOUN', 'nmod', False)]\n",
      "deepcheck connection rule noun found protects overexpression [that, overexpression, of, miR-21, protects, against, ischemic, neuronal, death, ,, and, that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "DeepCheck Conj Subtree\n",
      "[an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "deepcheck connection rule noun found mediates downregulation [that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "0 [Our, data, indicate] 3 0 Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "1 [overexpression, of, miR-21, protects, against, ischemic, neuronal, death] 8 23 overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "2 [downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect] 28 99 downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-21]\n",
      "deepcheck connection rule noun found protects overexpression [that, overexpression, of, miR-21, protects, against, ischemic, neuronal, death, ,, and, that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "DeepCheck Conj Subtree\n",
      "[against, ischemic, neuronal, death]\n",
      "DeepCheck Conj Subtree\n",
      "[of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[by, miR-21]\n",
      "deepcheck connection rule noun found mediates downregulation [that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "0 [Our, data, indicate] 3 0 Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "1 [overexpression, of, miR-21, protects, against, ischemic, neuronal, death] 8 23 overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "2 [downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect] 28 99 downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "Context Fail <regex.Match object; span=(7, 13), match=' tumor', fuzzy_counts=(0, 1, 0)> regex.Regex('(tumor){e<=1}', flags=regex.V0)\n",
      "Recognized overlapping entities: {Interval(117, 122, ('GENE_OR_GENE_PRODUCT', 'FASLG'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text downregulation of FASLG , a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21 , probably mediates the neuroprotective effect\n",
      "[downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "TextBetween: downregulation of FASLG , a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21 , probably mediates the neuroprotective effect\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between gene mir', 'reg_detect_major': 'between'}})\n",
      " hit sentence Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "full sentence Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of <e2>FASLG</e2>, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by <e1>miR-21</e1>, probably mediates the neuroprotective effect.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between gene mir', 'reg_detect_major': 'between'}}, 'tagged_sent': 'Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of <e2>FASLG</e2>, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by <e1>miR-21</e1>, probably mediates the neuroprotective effect.', 'full_sentence': 'Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of <e2>FASLG</e2>, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by <e1>miR-21</e1>, probably mediates the neuroprotective effect.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (117, 122), 'entity_text': 'FASLG'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (234, 240), 'entity_text': 'miR-21'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d56.s2.p9\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case CD47\n",
      "add special case microRNA-155\n",
      "CD47 20 microRNA-155 26\n",
      "(u\"Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites.\", 20, 26)\n",
      "geneword CD47 mirword microRNA-155\n",
      "Conjuncts [microRNA-155, microRNA-326, microRNA-34a]\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-326 )\n",
      "neighbour microRNA-326 targeted\n",
      "Conjunction before fc microRNA-34a\n",
      "Conjuncts [microRNA-34a, microRNA-326, microRNA-155]\n",
      "neighbour microRNA-34a ,\n",
      "neighbour microRNA-34a microRNA-155\n",
      "neighbour microRNA-326 )\n",
      "neighbour microRNA-326 targeted\n",
      "Conjunction before fc microRNA-155\n",
      "Conjuncts [microRNA-34a, microRNA-155, microRNA-326]\n",
      "neighbour microRNA-34a ,\n",
      "neighbour microRNA-34a microRNA-155\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-155 and\n",
      "Conjunction before fc microRNA-326\n",
      "Conjunctions ( conj )\n",
      "microRNA-34a [microRNA-34a, microRNA-155, microRNA-326]\n",
      "Conjuncts [upregulated, lesions]\n",
      "neighbour upregulated in\n",
      "neighbour upregulated in\n",
      "Conjunction before fc lesions\n",
      "fc add children for lesions\n",
      "fc add children for lesions\n",
      "fc add children for lesions\n",
      "fc add children for lesions\n",
      "fc add children for lesions\n",
      "Conjuncts [region, CD47]\n",
      "neighbour region of\n",
      "neighbour region of\n",
      "Conjunction before fc CD47\n",
      "fc add children for CD47\n",
      "Conjuncts [targeted, assays]\n",
      "neighbour targeted the\n",
      "neighbour targeted the\n",
      "Conjunction before fc assays\n",
      "fc add children for assays\n",
      "fc add children for assays\n",
      "Conjuncts [targeted, microRNA-155]\n",
      "neighbour targeted the\n",
      "neighbour targeted the\n",
      "Conjunction before fc microRNA-155\n",
      "fc add children for microRNA-155\n",
      "Conjuncts [targeted, sites]\n",
      "neighbour targeted the\n",
      "neighbour targeted the\n",
      "Conjunction before fc sites\n",
      "fc add children for sites\n",
      "fc add children for sites\n",
      "fc add children for sites\n",
      "Conunct merge check: lesions assays\n",
      "microRNA-34a, microRNA-155 and microRNA-326)\n",
      "Conunct merge check: assays CD47\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( nmod )\n",
      "lesions [upregulated, in, active, multiple, sclerosis, lesions, microRNA-34a]\n",
      "CD47 [region, of, CD47]\n",
      "assays [targeted, in, reporter, assays, with, microRNA-155, at, two, distinct, sites]\n",
      "SDP sent: Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites.\n",
      "SDP     : [(CD47, 'NOUN', 'nmod', False), (region, 'NOUN', 'dobj', False), (targeted, 'VERB', 'ROOT', False), (microRNA-155, 'NOUN', 'nmod', False)]\n",
      "SDP DOBJ RULE\n",
      "DeepCheck Conj Subtree\n",
      "[microRNA-155]\n",
      "DeepCheck Conj Subtree\n",
      "[microRNA-326]\n",
      "0 [Three, microRNA, upregulated, in, active, multiple, sclerosis, lesions, (, microRNA-34a, ,, microRNA-155, and, microRNA-326, ), targeted, the, 3'-untranslated, region, of, CD47, in, reporter, assays, ,, with, microRNA-155, even, at, two, distinct, sites] 32 0 Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites.\n",
      "DeepCheck Conj Subtree\n",
      "[in, active, multiple, sclerosis, lesions, (, microRNA-34a, ,, microRNA-155, and, microRNA-326, )]\n",
      "DeepCheck Conj Subtree\n",
      "[microRNA-155]\n",
      "DeepCheck Conj Subtree\n",
      "[microRNA-326]\n",
      "DeepCheck Conj Subtree\n",
      "[of, CD47]\n",
      "DeepCheck Conj Subtree\n",
      "[in, reporter, assays]\n",
      "DeepCheck Conj Subtree\n",
      "[with, microRNA-155]\n",
      "DeepCheck Conj Subtree\n",
      "[even, at, two, distinct, sites]\n",
      "0 [Three, microRNA, upregulated, in, active, multiple, sclerosis, lesions, (, microRNA-34a, ,, microRNA-155, and, microRNA-326, ), targeted, the, 3'-untranslated, region, of, CD47, in, reporter, assays, ,, with, microRNA-155, even, at, two, distinct, sites] 32 0 Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites.\n",
      "Recognized overlapping entities: {Interval(149, 153, ('GENE_OR_GENE_PRODUCT', 'CD47'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text Three microRNA upregulated in active multiple sclerosis lesions ( microRNA-34a , microRNA-155 and microRNA-326 ) targeted the 3'-untranslated region of CD47 in reporter assays , with microRNA-155 even at two distinct sites\n",
      "[Three, microRNA, upregulated, in, active, multiple, sclerosis, lesions, (, microRNA-34a, ,, microRNA-155, and, microRNA-326, ), targeted, the, 3'-untranslated, region, of, CD47, in, reporter, assays, ,, with, microRNA-155, even, at, two, distinct, sites]\n",
      "TextBetween: Three microRNA upregulated in active multiple sclerosis lesions ( microRNA-34a , microRNA-155 and microRNA-326 ) targeted the 3'-untranslated region of CD47 in reporter assays , with microRNA-155 even at two distinct sites\n",
      "infer direction from token set [Three, microRNA, upregulated, in, active, multiple, sclerosis, lesions, (, microRNA-34a, ,, microRNA-155, and, microRNA-326, ), targeted, the, 3'-untranslated, region, of, CD47, in, reporter, assays, ,, with, microRNA-155, even, at, two, distinct, sites]\n",
      "stemTypes2Count: Counter({'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fc0b812eb70>, {})\n",
      "counts between 1 stemTypes2Count Counter()\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbecba9e158>, {})\n",
      "counts between 1 stemTypes2Count []\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}})\n",
      " hit sentence Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites.\n",
      "full sentence Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of <e2>CD47</e2> in reporter assays, with <e1>microRNA-155</e1> even at two distinct sites.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}}, 'tagged_sent': \"Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of <e2>CD47</e2> in reporter assays, with <e1>microRNA-155</e1> even at two distinct sites.\", 'full_sentence': \"Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of <e2>CD47</e2> in reporter assays, with <e1>microRNA-155</e1> even at two distinct sites.\", 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (149, 153), 'entity_text': 'CD47'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (179, 191), 'entity_text': 'microRNA-155'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d57.s2.p4\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case miR-23a\n",
      "add special case XIAP\n",
      "miR-23a 11 XIAP 18\n",
      "(u\"miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\", 11, 18)\n",
      "geneword XIAP mirword miR-23a\n",
      "Conjuncts [led, bound]\n",
      "neighbour led to\n",
      "neighbour led to\n",
      "Conjunction before fc bound\n",
      "fc add children for bound\n",
      "Conjuncts [miR-23a, UTR]\n",
      "neighbour miR-23a inhibition\n",
      "neighbour miR-23a inhibition\n",
      "Conjunction before fc UTR\n",
      "fc add children for UTR\n",
      "fc add children for UTR\n",
      "Conjuncts [UTR, miR-23a]\n",
      "neighbour UTR of\n",
      "neighbour UTR of\n",
      "Conjunction before fc miR-23a\n",
      "Conjuncts [bound, led]\n",
      "neighbour bound the\n",
      "neighbour bound the\n",
      "Conjunction before fc led\n",
      "fc add children for led\n",
      "Conjunctions ( conj )\n",
      "bound [bound, 3, UTR, of, XIAP, miR-23a, led, to, increase, in, XIAP, mRNA]\n",
      "achildren {miR-23a}\n",
      "achildren False\n",
      "Conjuncts [UTR, XIAP]\n",
      "neighbour UTR of\n",
      "neighbour UTR of\n",
      "Conjunction before fc XIAP\n",
      "fc add children for XIAP\n",
      "Conjuncts [led, increase]\n",
      "neighbour led to\n",
      "neighbour led to\n",
      "Conjunction before fc increase\n",
      "fc add children for increase\n",
      "fc add children for increase\n",
      "Conjuncts [increase, mRNA]\n",
      "neighbour increase in\n",
      "neighbour increase in\n",
      "Conjunction before fc mRNA\n",
      "fc add children for mRNA\n",
      "fc add children for mRNA\n",
      "Conjuncts [target, miR-23a]\n",
      "neighbour target for\n",
      "neighbour target for\n",
      "Conjunction before fc miR-23a\n",
      "fc add children for miR-23a\n",
      "Conunct merge check: XIAP increase\n",
      "XIAP, and miR-23a inhibition\n",
      "Double Conjunction Found\n",
      "Conunct merge check: increase miR-23a\n",
      "mRNA in vitro, demonstrating that XIAP is a previously uncharacterized\n",
      "Conjunctions ( nmod )\n",
      "XIAP [UTR, of, XIAP]\n",
      "increase [led, to, increase, in, XIAP, mRNA]\n",
      "miR-23a [target, for, miR-23a]\n",
      "(increase, 'conj') [UTR, of, XIAP, led, to, increase, in, XIAP, mRNA]\n",
      "SDP sent: miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "SDP     : [(miR-23a, 'NOUN', 'conj', False), (UTR, 'NOUN', 'dobj', True), (bound, 'VERB', 'ROOT', True), (led, 'VERB', 'conj', False), (increase, 'NOUN', 'nmod', False), (mRNA, 'NOUN', 'nmod', False), (XIAP, 'NOUN', 'compound', False)]\n",
      "SDP DOBJ RULE\n",
      "DeepCheck Conj Subtree\n",
      "[miR-23a]\n",
      "deepcheck connection rule noun found led inhibition [inhibition, led, to, an, increase, in, XIAP, mRNA, in, vitro]\n",
      "0 [miR-23a, directly, bound, the, 3, ', UTR, of, XIAP, ,, and, miR-23a] 12 0 miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "1 [inhibition, led, to, an, increase, in, XIAP, mRNA, in, vitro] 10 55 inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "2 [demonstrating, that, XIAP, is, a, previously, uncharacterized, target, for, miR-23a] 10 108 demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "DeepCheck Conj Subtree\n",
      "[of, XIAP]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-23a]\n",
      "deepcheck connection rule noun found led inhibition [inhibition, led, to, an, increase, in, XIAP, mRNA, in, vitro]\n",
      "DeepCheck Conj Subtree\n",
      "[to, an, increase, in, XIAP, mRNA]\n",
      "DeepCheck Conj Subtree\n",
      "[in, XIAP, mRNA]\n",
      "DeepCheck Conj Subtree\n",
      "[for, miR-23a]\n",
      "0 [miR-23a, directly, bound, the, 3, ', UTR, of, XIAP, ,, and, miR-23a] 12 0 miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "1 [inhibition, led, to, an, increase, in, XIAP, mRNA, in, vitro] 10 55 inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "2 [demonstrating, that, XIAP, is, a, previously, uncharacterized, target, for, miR-23a] 10 108 demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "Recognized overlapping entities: {Interval(88, 92, ('GENE_OR_GENE_PRODUCT', 'XIAP'))}\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "TextBetween: , and miR-23a inhibition led to an increase in XIAP mRNA in\n",
      "infer direction from token set [,, and, miR-23a, inhibition, led, to, an, increase, in, XIAP, mRNA, in]\n",
      "stemTypes2Count: Counter({'NEG': 1, 'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8cae8>, {'mir': Counter({'NEG': 1}), 'gene': Counter({'POS': 1})})\n",
      "defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8cae8>, {'mir': Counter({'NEG': 1}), 'gene': Counter({'POS': 1})})\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}})\n",
      " hit sentence miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "full sentence miR-23a directly bound the 3' UTR of XIAP, and <e1>miR-23a</e1> inhibition led to an increase in <e2>XIAP</e2> mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"miR-23a directly bound the 3' UTR of XIAP, and <e1>miR-23a</e1> inhibition led to an increase in <e2>XIAP</e2> mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\", 'full_sentence': \"miR-23a directly bound the 3' UTR of XIAP, and <e1>miR-23a</e1> inhibition led to an increase in <e2>XIAP</e2> mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (47, 54), 'entity_text': 'miR-23a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (88, 92), 'entity_text': 'XIAP'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d66.s2.p0\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'We found that the A allele of rs2507800 suppressed <e2>angiopoietin-1</e2> translation by facilitating <e1>miR-211</e1> binding, but not the T allele.', 'full_sentence': 'We found that the A allele of rs2507800 suppressed <e2>angiopoietin-1</e2> translation by facilitating <e1>miR-211</e1> binding, but not the T allele.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (51, 65), 'entity_text': 'angiopoietin-1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (94, 101), 'entity_text': 'miR-211'}}\n",
      "add special case angiopoietin-1\n",
      "add special case miR-211\n",
      "angiopoietin-1 9 miR-211 13\n",
      "(u\"We found that the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding, but not the T allele.\", 9, 13)\n",
      "geneword angiopoietin-1 mirword miR-211\n",
      "Conjuncts [allele, binding]\n",
      "neighbour allele .\n",
      "neighbour allele None\n",
      "Conjunction before fc binding\n",
      "fc add children for binding\n",
      "Conjuncts [binding, allele]\n",
      "neighbour binding ,\n",
      "neighbour binding but\n",
      "Conjunction before fc allele\n",
      "fc add children for allele\n",
      "Conjunctions ( conj )\n",
      "binding [miR-211, binding, T, allele]\n",
      "Conjuncts [allele, rs2507800]\n",
      "neighbour allele of\n",
      "neighbour allele of\n",
      "Conjunction before fc rs2507800\n",
      "fc add children for rs2507800\n",
      "Conjunctions ( nmod )\n",
      "rs2507800 [allele, of, rs2507800]\n",
      "SDP sent: We found that the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding, but not the T allele.\n",
      "SDP     : [(angiopoietin-1, 'NOUN', 'compound', False), (translation, 'NOUN', 'dobj', False), (suppressed, 'VERB', 'ccomp', False), (facilitating, 'VERB', 'advcl', False), (binding, 'NOUN', 'dobj', False), (miR-211, 'NOUN', 'compound', False)]\n",
      "SDP DOBJ RULE\n",
      "deepcheck connection rule [by, facilitating, miR-211, binding, ,, but, not, the, T, allele]\n",
      "DeepCheck Conj Subtree\n",
      "[the, T, allele]\n",
      "0 [We, found] 2 0 We found that the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding, but not the T allele.\n",
      "1 [the, A, allele, of, rs2507800, suppressed, angiopoietin-1, translation, by, facilitating, miR-211, binding, ,, but, not, the, T, allele] 18 14 the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding, but not the T allele.\n",
      "DeepCheck Conj Subtree\n",
      "[of, rs2507800]\n",
      "deepcheck connection rule [by, facilitating, miR-211, binding, ,, but, not, the, T, allele]\n",
      "DeepCheck Conj Subtree\n",
      "[the, T, allele]\n",
      "0 [We, found] 2 0 We found that the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding, but not the T allele.\n",
      "1 [the, A, allele, of, rs2507800, suppressed, angiopoietin-1, translation, by, facilitating, miR-211, binding, ,, but, not, the, T, allele] 18 14 the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding, but not the T allele.\n",
      "Recognized overlapping entities: {Interval(51, 65, ('GENE_OR_GENE_PRODUCT', 'angiopoietin-1'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding , but not the T allele\n",
      "[the, A, allele, of, rs2507800, suppressed, angiopoietin-1, translation, by, facilitating, miR-211, binding, ,, but, not, the, T, allele]\n",
      "TextBetween: the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding , but not the T allele\n",
      "infer direction from token set [the, A, allele, of, rs2507800, suppressed, angiopoietin-1, translation, by, facilitating, miR-211, binding, ,, but, not, the, T, allele]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbecba9e048>, {'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fc0b806d048>, {'gene': Counter({'NEG': 1, 'NEU': 1}), 'mir': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(suppressed, 'NEG'), (translation, 'NEU'), (binding, 'NEU')]\n",
      "counts between 3 Counter() NEG\n",
      "Between Tokens []\n",
      "counts between final False\n",
      "['rs2507800', 'suppressed', 'angiopoietin-1', 'translation', 'by', 'facilitating', 'miR-211', 'binding', ',']\n",
      "Counter()\n",
      "[]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence We found that the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding, but not the T allele.\n",
      "full sentence We found that the A allele of rs2507800 suppressed <e2>angiopoietin-1</e2> translation by facilitating <e1>miR-211</e1> binding, but not the T allele.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'We found that the A allele of rs2507800 suppressed <e2>angiopoietin-1</e2> translation by facilitating <e1>miR-211</e1> binding, but not the T allele.', 'full_sentence': 'We found that the A allele of rs2507800 suppressed <e2>angiopoietin-1</e2> translation by facilitating <e1>miR-211</e1> binding, but not the T allele.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (51, 65), 'entity_text': 'angiopoietin-1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (94, 101), 'entity_text': 'miR-211'}}\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "To test our hypothesis, a predicted microRNA-binding site was found in the 3'-UTR of angiopoietin-1 using bioinformatics; variant rs2507800 was identified to be located in the miR-211-binding site of angiopoietin-1. \n",
      "('miR-211', 'Specific_miRNAs', (176, 183)) ('angiopoietin-1', 'Genes/Proteins', (200, 214))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d66.s9.p1\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of <e2>angiopoietin-1</e2> might reduce the risk of stroke by interfering with <e1>miR-211</e1> binding.\", 'full_sentence': \"Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of <e2>angiopoietin-1</e2> might reduce the risk of stroke by interfering with <e1>miR-211</e1> binding.\", 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (79, 93), 'entity_text': 'angiopoietin-1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (146, 153), 'entity_text': 'miR-211'}}\n",
      "add special case angiopoietin-1\n",
      "add special case miR-211\n",
      "angiopoietin-1 15 miR-211 25\n",
      "(u\"Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding.\", 15, 25)\n",
      "geneword angiopoietin-1 mirword miR-211\n",
      "Conjunctions ( conj )\n",
      "rs2507800 [(, )]\n",
      "Conjuncts [genotype, 3'-UTR]\n",
      "neighbour genotype (\n",
      "neighbour genotype rs2507800\n",
      "Conjunction before fc 3'-UTR\n",
      "fc add children for 3'-UTR\n",
      "fc add children for 3'-UTR\n",
      "Conjuncts [3'-UTR, angiopoietin-1]\n",
      "neighbour 3'-UTR of\n",
      "neighbour 3'-UTR of\n",
      "Conjunction before fc angiopoietin-1\n",
      "fc add children for angiopoietin-1\n",
      "Conjuncts [risk, stroke]\n",
      "neighbour risk of\n",
      "neighbour risk of\n",
      "Conjunction before fc stroke\n",
      "fc add children for stroke\n",
      "Conjuncts [interfering, binding]\n",
      "neighbour interfering with\n",
      "neighbour interfering with\n",
      "Conjunction before fc binding\n",
      "fc add children for binding\n",
      "fc add children for binding\n",
      "Conunct merge check: 3'-UTR rs2507800\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: rs2507800 stroke\n",
      ") in the 3'-UTR of angiopoietin-1 might reduce the\n",
      "Conunct merge check: stroke binding\n",
      "stroke by\n",
      "Conjunctions ( nmod )\n",
      "rs2507800 [(, )]\n",
      "3'-UTR [genotype, in, 3'-UTR, of, angiopoietin-1]\n",
      "stroke [risk, of, stroke]\n",
      "binding [interfering, with, miR-211, binding]\n",
      "SDP sent: Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding.\n",
      "SDP     : [(angiopoietin-1, 'NOUN', 'nmod', False), (3'-UTR, 'NOUN', 'nmod', False), (genotype, 'NOUN', 'nsubj', False), (reduce, 'VERB', 'ccomp', False), (interfering, 'VERB', 'advcl', False), (binding, 'NOUN', 'nmod', False), (miR-211, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule [by, interfering, with, miR-211, binding]\n",
      "0 [Our, results, provide, evidence] 4 0 Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding.\n",
      "1 [the, TT, genotype, (, rs2507800, ), in, the, 3'-UTR, of, angiopoietin-1, might, reduce, the, risk, of, stroke, by, interfering, with, miR-211, binding] 22 34 the TT genotype (rs2507800) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding.\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, 3'-UTR, of, angiopoietin-1]\n",
      "DeepCheck Conj Subtree\n",
      "[of, angiopoietin-1]\n",
      "DeepCheck Conj Subtree\n",
      "[of, stroke]\n",
      "deepcheck connection rule [by, interfering, with, miR-211, binding]\n",
      "DeepCheck Conj Subtree\n",
      "[with, miR-211, binding]\n",
      "0 [Our, results, provide, evidence] 4 0 Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding.\n",
      "1 [the, TT, genotype, (, rs2507800, ), in, the, 3'-UTR, of, angiopoietin-1, might, reduce, the, risk, of, stroke, by, interfering, with, miR-211, binding] 22 34 the TT genotype (rs2507800) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding.\n",
      "Recognized overlapping entities: {Interval(79, 93, ('GENE_OR_GENE_PRODUCT', 'angiopoietin-1'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text the TT genotype ( rs2507800 ) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding\n",
      "[the, TT, genotype, (, rs2507800, ), in, the, 3'-UTR, of, angiopoietin-1, might, reduce, the, risk, of, stroke, by, interfering, with, miR-211, binding]\n",
      "TextBetween: the TT genotype ( rs2507800 ) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding\n",
      "infer direction from token set [the, TT, genotype, (, rs2507800, ), in, the, 3'-UTR, of, angiopoietin-1, might, reduce, the, risk, of, stroke, by, interfering, with, miR-211, binding]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70ae8>, {'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70510>, {'gene': Counter({'NEG': 1}), 'mir': Counter({'NEU': 2})})\n",
      "counts between 1 stemTypes2Count [(reduce, 'NEG'), (interfering, 'NEU'), (binding, 'NEU')]\n",
      "counts between 3 Counter({'NEG': 1}) NEG\n",
      "Between Tokens [(reduce, 'NEG')]\n",
      "counts between final False\n",
      "[\"3'-UTR\", 'of', 'angiopoietin-1', 'might', 'reduce', 'the', 'risk', 'of', 'stroke', 'by', 'interfering', 'with', 'miR-211', 'binding']\n",
      "Counter({'NEG': 1})\n",
      "[(reduce, 'NEG')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding.\n",
      "full sentence Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of <e2>angiopoietin-1</e2> might reduce the risk of stroke by interfering with <e1>miR-211</e1> binding.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of <e2>angiopoietin-1</e2> might reduce the risk of stroke by interfering with <e1>miR-211</e1> binding.\", 'full_sentence': \"Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of <e2>angiopoietin-1</e2> might reduce the risk of stroke by interfering with <e1>miR-211</e1> binding.\", 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (79, 93), 'entity_text': 'angiopoietin-1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (146, 153), 'entity_text': 'miR-211'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d69.s11.p1\n",
      "IS:  ('MIR_GENE', 'NEU')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Glucose supplementation in cultured cells that leads to increased <e1>miR-107</e1> levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of <e2>GRN</e2> protein.', 'full_sentence': 'Glucose supplementation in cultured cells that leads to increased <e1>miR-107</e1> levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of <e2>GRN</e2> protein.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (66, 73), 'entity_text': 'miR-107'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (193, 196), 'entity_text': 'GRN'}}\n",
      "add special case miR-107\n",
      "add special case GRN\n",
      "miR-107 9 GRN 27\n",
      "(u\"Glucose supplementation in cultured cells that leads to increased miR-107 levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of GRN protein.\", 9, 27)\n",
      "geneword GRN mirword miR-107\n",
      "Conjuncts [secretion, expression]\n",
      "neighbour secretion of\n",
      "neighbour secretion of\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conjuncts [expression, secretion]\n",
      "neighbour expression ,\n",
      "neighbour expression including\n",
      "Conjunction before fc secretion\n",
      "fc add children for secretion\n",
      "fc add children for secretion\n",
      "Conjunctions ( conj )\n",
      "expression [in, decreased, GRN, expression, including, changes, in, cell, compartmentation, decreased, secretion, of, GRN, protein]\n",
      "Conjuncts [supplementation, cells]\n",
      "neighbour supplementation in\n",
      "neighbour supplementation in\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conjuncts [leads, levels]\n",
      "neighbour leads to\n",
      "neighbour leads to\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conjuncts [results, expression]\n",
      "neighbour results in\n",
      "neighbour results in\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conjuncts [expression, changes]\n",
      "neighbour expression ,\n",
      "neighbour expression including\n",
      "Conjunction before fc changes\n",
      "fc add children for changes\n",
      "fc add children for changes\n",
      "Conjuncts [changes, compartmentation]\n",
      "neighbour changes in\n",
      "neighbour changes in\n",
      "Conjunction before fc compartmentation\n",
      "fc add children for compartmentation\n",
      "fc add children for compartmentation\n",
      "Conjuncts [secretion, protein]\n",
      "neighbour secretion of\n",
      "neighbour secretion of\n",
      "Conjunction before fc protein\n",
      "fc add children for protein\n",
      "fc add children for protein\n",
      "Conunct merge check: cells levels\n",
      "cells that\n",
      "Conunct merge check: levels expression\n",
      "levels also\n",
      "Conunct merge check: expression protein\n",
      "compartmentation and decreased\n",
      "Double Conjunction Found\n",
      "Conjunctions ( nmod )\n",
      "cells [supplementation, in, cultured, cells]\n",
      "levels [leads, to, increased, miR-107, levels]\n",
      "expression [results, in, decreased, GRN, expression, including, changes, in, cell, compartmentation]\n",
      "protein [secretion, of, GRN, protein]\n",
      "(protein, 'conj') [results, in, decreased, GRN, expression, including, changes, in, cell, compartmentation, secretion, of, GRN, protein]\n",
      "SDP sent: Glucose supplementation in cultured cells that leads to increased miR-107 levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of GRN protein.\n",
      "SDP     : [(miR-107, 'NOUN', 'compound', False), (levels, 'NOUN', 'nmod', False), (leads, 'VERB', 'acl:relcl', False), (cells, 'NOUN', 'nmod', False), (supplementation, 'NOUN', 'nsubj', False), (results, 'VERB', 'ROOT', False), (expression, 'NOUN', 'nmod', True), (secretion, 'NOUN', 'conj', False), (protein, 'NOUN', 'nmod', False), (GRN, 'NOUN', 'compound', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[decreased, secretion, of, GRN, protein]\n",
      "deepcheck conj verb\n",
      "deepcheck conj verb\n",
      "0 [Glucose, supplementation, in, cultured, cells, that, leads, to, increased, miR-107, levels, also, results, in, decreased, GRN, expression, ,, including, changes, in, cell, compartmentation, and, decreased, secretion, of, GRN, protein] 29 0 Glucose supplementation in cultured cells that leads to increased miR-107 levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of GRN protein.\n",
      "DeepCheck Conj Subtree\n",
      "[in, cultured, cells, that, leads, to, increased, miR-107, levels]\n",
      "DeepCheck Conj Subtree\n",
      "[to, increased, miR-107, levels]\n",
      "DeepCheck Conj Subtree\n",
      "[in, decreased, GRN, expression, ,, including, changes, in, cell, compartmentation, and, decreased, secretion, of, GRN, protein]\n",
      "DeepCheck Conj Subtree\n",
      "[including, changes, in, cell, compartmentation]\n",
      "deepcheck conj verb\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[in, cell, compartmentation]\n",
      "DeepCheck Conj Subtree\n",
      "[decreased, secretion, of, GRN, protein]\n",
      "deepcheck conj verb\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[of, GRN, protein]\n",
      "0 [Glucose, supplementation, in, cultured, cells, that, leads, to, increased, miR-107, levels, also, results, in, decreased, GRN, expression, ,, including, changes, in, cell, compartmentation, and, decreased, secretion, of, GRN, protein] 29 0 Glucose supplementation in cultured cells that leads to increased miR-107 levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of GRN protein.\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "Compartment Text Glucose supplementation in cultured cells that leads to increased miR-107 levels also results in decreased GRN expression , including changes in cell compartmentation and decreased secretion of GRN protein\n",
      "[Glucose, supplementation, in, cultured, cells, that, leads, to, increased, miR-107, levels, also, results, in, decreased, GRN, expression, ,, including, changes, in, cell, compartmentation, and, decreased, secretion, of, GRN, protein]\n",
      "TextBetween: Glucose supplementation in cultured cells that leads to increased miR-107 levels also results in decreased GRN expression , including changes in cell compartmentation and decreased secretion of GRN protein\n",
      "infer direction from token set [Glucose, supplementation, in, cultured, cells, that, leads, to, increased, miR-107, levels, also, results, in, decreased, GRN, expression, ,, including, changes, in, cell, compartmentation, and, decreased, secretion, of, GRN, protein]\n",
      "stemTypes2Count: Counter({'POS': 2, 'NEG': 2})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e26a8>, {'mir': Counter({'POS': 1}), 'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 4, 'POS': 2, 'NEG': 2})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e2620>, {'mir': Counter({'POS': 1, 'NEU': 1}), 'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count [(increased, 'POS'), (levels, 'NEU'), (results, 'NEU'), (decreased, 'NEG'), (expression, 'NEU'), (including, 'NEU'), (decreased, 'NEG'), (secretion, 'POS')]\n",
      "Between Tokens [(levels, 'NEU'), (results, 'NEU'), (decreased, 'NEG'), (expression, 'NEU'), (including, 'NEU'), (decreased, 'NEG'), (secretion, 'POS')]\n",
      "counts between final False\n",
      "['to', 'increased', 'miR-107', 'levels', 'also', 'results', 'in', 'decreased', 'GRN', 'expression', ',', 'including', 'changes', 'in', 'cell', 'compartmentation', 'and', 'decreased', 'secretion', 'of', 'GRN', 'protein']\n",
      "Counter({'NEU': 4, 'POS': 2, 'NEG': 2})\n",
      "[(increased, 'POS'), (levels, 'NEU'), (results, 'NEU'), (decreased, 'NEG'), (expression, 'NEU'), (including, 'NEU'), (decreased, 'NEG'), (secretion, 'POS')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Glucose supplementation in cultured cells that leads to increased miR-107 levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of GRN protein.\n",
      "full sentence Glucose supplementation in cultured cells that leads to increased <e1>miR-107</e1> levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of <e2>GRN</e2> protein.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Glucose supplementation in cultured cells that leads to increased <e1>miR-107</e1> levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of <e2>GRN</e2> protein.', 'full_sentence': 'Glucose supplementation in cultured cells that leads to increased <e1>miR-107</e1> levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of <e2>GRN</e2> protein.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (66, 73), 'entity_text': 'miR-107'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (193, 196), 'entity_text': 'GRN'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d70.s1.p2\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case MiR-21\n",
      "add special case Bax\n",
      "MiR-21 0 Bax/Bcl-2 14\n",
      "(u\"MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\", 0, 14)\n",
      "geneword Bax/Bcl-2 mirword MiR-21\n",
      "Conjuncts [induced, protected]\n",
      "neighbour induced apoptosis\n",
      "neighbour induced apoptosis\n",
      "Conjunction before fc protected\n",
      "fc add children for protected\n",
      "fc add children for protected\n",
      "Conjuncts [protected, induced]\n",
      "neighbour protected human\n",
      "neighbour protected human\n",
      "Conjunction before fc induced\n",
      "fc add children for induced\n",
      "Conjuncts [activity, ratio]\n",
      "neighbour activity .\n",
      "neighbour activity None\n",
      "Conjunction before fc ratio\n",
      "fc add children for ratio\n",
      "Conjuncts [ratio, activity]\n",
      "neighbour ratio and\n",
      "neighbour ratio and\n",
      "Conjunction before fc activity\n",
      "fc add children for activity\n",
      "Conunct merge check: protected ratio\n",
      "apoptosis by decreasing\n",
      "Conjunctions ( conj )\n",
      "protected [protected, human, glioblastoma, U87MG, cells, from, chemotherapeutic, drug, temozolomide, induced, apoptosis]\n",
      "ratio [Bax/Bcl-2, ratio, caspase-3, activity]\n",
      "Conjuncts [protected, temozolomide]\n",
      "neighbour protected human\n",
      "neighbour protected human\n",
      "Conjunction before fc temozolomide\n",
      "fc add children for temozolomide\n",
      "fc add children for temozolomide\n",
      "fc add children for temozolomide\n",
      "Conjunctions ( nmod )\n",
      "temozolomide [protected, from, chemotherapeutic, drug, temozolomide]\n",
      "SDP sent: MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "SDP     : [(MiR-21, 'NOUN', 'nsubj', False), (protected, 'VERB', 'ROOT', False), (decreasing, 'VERB', 'advcl', False), (ratio, 'NOUN', 'dobj', False), (Bax/Bcl-2, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule noun found induced apoptosis [induced, apoptosis]\n",
      "deepcheck conj verb\n",
      "deepcheck connection rule [by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity]\n",
      "DeepCheck Conj Subtree\n",
      "[caspase-3, activity]\n",
      "0 [MiR-21, protected, human, glioblastoma, U87MG, cells, from, chemotherapeutic, drug, temozolomide, induced, apoptosis, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity] 19 0 MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "DeepCheck Conj Subtree\n",
      "[from, chemotherapeutic, drug, temozolomide]\n",
      "deepcheck connection rule noun found induced apoptosis [induced, apoptosis]\n",
      "deepcheck conj verb\n",
      "deepcheck connection rule [by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity]\n",
      "DeepCheck Conj Subtree\n",
      "[caspase-3, activity]\n",
      "0 [MiR-21, protected, human, glioblastoma, U87MG, cells, from, chemotherapeutic, drug, temozolomide, induced, apoptosis, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity] 19 0 MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "Context Fail <regex.Match object; span=(8, 14), match=' ratio'> regex.Regex('( ratio){e<=0}', flags=regex.V0)\n",
      "Recognized overlapping entities: {Interval(120, 129, ('SIMPLE_CHEMICAL', 'Bax/Bcl-2'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity\n",
      "[MiR-21, protected, human, glioblastoma, U87MG, cells, from, chemotherapeutic, drug, temozolomide, induced, apoptosis, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity]\n",
      "TextBetween: MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity\n",
      "infer direction from token set [MiR-21, protected, human, glioblastoma, U87MG, cells, from, chemotherapeutic, drug, temozolomide, induced, apoptosis, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity]\n",
      "stemTypes2Count: Counter({'POS': 1, 'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62c80>, {'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'POS': 1, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e21e0>, {'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count [(induced, 'POS'), (decreasing, 'NEG')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}})\n",
      " hit sentence MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "full sentence <e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing <e2>Bax</e2>/Bcl-2 ratio and caspase-3 activity.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing <e2>Bax</e2>/Bcl-2 ratio and caspase-3 activity.', 'full_sentence': '<e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing <e2>Bax</e2>/Bcl-2 ratio and caspase-3 activity.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (0, 6), 'entity_text': 'MiR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (120, 123), 'entity_text': 'Bax'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d70.s1.p3\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case MiR-21\n",
      "add special case Bcl-2\n",
      "MiR-21 0 Bax/Bcl-2 14\n",
      "(u\"MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\", 0, 14)\n",
      "geneword Bax/Bcl-2 mirword MiR-21\n",
      "Conjuncts [induced, protected]\n",
      "neighbour induced apoptosis\n",
      "neighbour induced apoptosis\n",
      "Conjunction before fc protected\n",
      "fc add children for protected\n",
      "fc add children for protected\n",
      "Conjuncts [protected, induced]\n",
      "neighbour protected human\n",
      "neighbour protected human\n",
      "Conjunction before fc induced\n",
      "fc add children for induced\n",
      "Conjuncts [activity, ratio]\n",
      "neighbour activity .\n",
      "neighbour activity None\n",
      "Conjunction before fc ratio\n",
      "fc add children for ratio\n",
      "Conjuncts [ratio, activity]\n",
      "neighbour ratio and\n",
      "neighbour ratio and\n",
      "Conjunction before fc activity\n",
      "fc add children for activity\n",
      "Conunct merge check: protected ratio\n",
      "apoptosis by decreasing\n",
      "Conjunctions ( conj )\n",
      "protected [protected, human, glioblastoma, U87MG, cells, from, chemotherapeutic, drug, temozolomide, induced, apoptosis]\n",
      "ratio [Bax/Bcl-2, ratio, caspase-3, activity]\n",
      "Conjuncts [protected, temozolomide]\n",
      "neighbour protected human\n",
      "neighbour protected human\n",
      "Conjunction before fc temozolomide\n",
      "fc add children for temozolomide\n",
      "fc add children for temozolomide\n",
      "fc add children for temozolomide\n",
      "Conjunctions ( nmod )\n",
      "temozolomide [protected, from, chemotherapeutic, drug, temozolomide]\n",
      "SDP sent: MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "SDP     : [(MiR-21, 'NOUN', 'nsubj', False), (protected, 'VERB', 'ROOT', False), (decreasing, 'VERB', 'advcl', False), (ratio, 'NOUN', 'dobj', False), (Bax/Bcl-2, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule noun found induced apoptosis [induced, apoptosis]\n",
      "deepcheck conj verb\n",
      "deepcheck connection rule [by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity]\n",
      "DeepCheck Conj Subtree\n",
      "[caspase-3, activity]\n",
      "0 [MiR-21, protected, human, glioblastoma, U87MG, cells, from, chemotherapeutic, drug, temozolomide, induced, apoptosis, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity] 19 0 MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "DeepCheck Conj Subtree\n",
      "[from, chemotherapeutic, drug, temozolomide]\n",
      "deepcheck connection rule noun found induced apoptosis [induced, apoptosis]\n",
      "deepcheck conj verb\n",
      "deepcheck connection rule [by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity]\n",
      "DeepCheck Conj Subtree\n",
      "[caspase-3, activity]\n",
      "0 [MiR-21, protected, human, glioblastoma, U87MG, cells, from, chemotherapeutic, drug, temozolomide, induced, apoptosis, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity] 19 0 MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "Context Fail <regex.Match object; span=(8, 14), match=' ratio'> regex.Regex('( ratio){e<=0}', flags=regex.V0)\n",
      "Recognized overlapping entities: {Interval(120, 129, ('SIMPLE_CHEMICAL', 'Bax/Bcl-2'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity\n",
      "[MiR-21, protected, human, glioblastoma, U87MG, cells, from, chemotherapeutic, drug, temozolomide, induced, apoptosis, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity]\n",
      "TextBetween: MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity\n",
      "infer direction from token set [MiR-21, protected, human, glioblastoma, U87MG, cells, from, chemotherapeutic, drug, temozolomide, induced, apoptosis, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity]\n",
      "stemTypes2Count: Counter({'POS': 1, 'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62c80>, {'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'POS': 1, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62bf8>, {'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count [(induced, 'POS'), (decreasing, 'NEG')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}})\n",
      " hit sentence MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "full sentence <e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/<e2>Bcl-2</e2> ratio and caspase-3 activity.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/<e2>Bcl-2</e2> ratio and caspase-3 activity.', 'full_sentence': '<e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/<e2>Bcl-2</e2> ratio and caspase-3 activity.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (0, 6), 'entity_text': 'MiR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (124, 129), 'entity_text': 'Bcl-2'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d70.s12.p1\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case miR-21\n",
      "add special case Bcl-2\n",
      "miR-21 12 Bcl-2 26\n",
      "(u\"However, such effect was partly prevented by treatment of cells with miR-21 overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.\", 12, 26)\n",
      "geneword Bcl-2 mirword miR-21\n",
      "Conjuncts [upregulate, decrease, prevented]\n",
      "neighbour upregulate the\n",
      "neighbour upregulate the\n",
      "neighbour decrease caspase-3\n",
      "neighbour decrease caspase-3\n",
      "Conjunction before fc prevented\n",
      "fc add children for prevented\n",
      "Conjuncts [prevented, decrease, upregulate]\n",
      "neighbour prevented by\n",
      "neighbour prevented by\n",
      "neighbour decrease caspase-3\n",
      "neighbour decrease caspase-3\n",
      "Conjunction before fc upregulate\n",
      "fc add children for upregulate\n",
      "Conjuncts [prevented, upregulate, decrease]\n",
      "neighbour prevented by\n",
      "neighbour prevented by\n",
      "neighbour upregulate the\n",
      "neighbour upregulate the\n",
      "Conjunction before fc decrease\n",
      "fc add children for decrease\n",
      "Conjunctions ( conj )\n",
      "prevented [prevented, by, treatment, of, cells, with, miR-21, overexpression, before, upregulate, Bcl-2, expression, decrease, caspase-3, activity]\n",
      "achildren {miR-21}\n",
      "achildren False\n",
      "Conjuncts [prevented, treatment]\n",
      "neighbour prevented by\n",
      "neighbour prevented by\n",
      "Conjunction before fc treatment\n",
      "fc add children for treatment\n",
      "fc add children for treatment\n",
      "fc add children for treatment\n",
      "fc add children for treatment\n",
      "Conjuncts [treatment, cells]\n",
      "neighbour treatment of\n",
      "neighbour treatment of\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "Conjuncts [treatment, overexpression]\n",
      "neighbour treatment of\n",
      "neighbour treatment of\n",
      "Conjunction before fc overexpression\n",
      "fc add children for overexpression\n",
      "fc add children for overexpression\n",
      "Conjunctions ( nmod )\n",
      "treatment [prevented, by, treatment, of, cells, with, miR-21, overexpression, before]\n",
      "SDP sent: However, such effect was partly prevented by treatment of cells with miR-21 overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.\n",
      "SDP     : [(miR-21, 'NOUN', 'compound', False), (overexpression, 'NOUN', 'nmod', False), (treatment, 'NOUN', 'nmod', False), (prevented, 'VERB', 'ROOT', True), (upregulate, 'VERB', 'conj', False), (expression, 'NOUN', 'dobj', False), (Bcl-2, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule [to, downregulate, the, Bax, expression]\n",
      "deepcheck conj verb\n",
      "deepcheck connection rule noun found decrease caspase-3 [decrease, caspase-3, activity]\n",
      "deepcheck conj verb\n",
      "0 [However, ,, such, effect, was, partly, prevented, by, treatment, of, cells, with, miR-21, overexpression, before, ,, which, appeared, to, downregulate, the, Bax, expression, ,, upregulate, the, Bcl-2, expression, and, decrease, caspase-3, activity] 32 0 However, such effect was partly prevented by treatment of cells with miR-21 overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.\n",
      "DeepCheck Conj Subtree\n",
      "[by, treatment, of, cells, with, miR-21, overexpression, before, ,, which, appeared, to, downregulate, the, Bax, expression]\n",
      "DeepCheck Conj Subtree\n",
      "[of, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[with, miR-21, overexpression]\n",
      "deepcheck connection rule [to, downregulate, the, Bax, expression]\n",
      "deepcheck conj verb\n",
      "deepcheck connection rule noun found decrease caspase-3 [decrease, caspase-3, activity]\n",
      "deepcheck conj verb\n",
      "0 [However, ,, such, effect, was, partly, prevented, by, treatment, of, cells, with, miR-21, overexpression, before, ,, which, appeared, to, downregulate, the, Bax, expression, ,, upregulate, the, Bcl-2, expression, and, decrease, caspase-3, activity] 32 0 However, such effect was partly prevented by treatment of cells with miR-21 overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.\n",
      "Recognized overlapping entities: {Interval(165, 170, ('GENE_OR_GENE_PRODUCT', 'Bcl-2'))}\n",
      "passive? False\n",
      "Counter({'POS': 2, 'NEG': 1})\n",
      "Compartment Text However , such effect was partly prevented by treatment of cells with miR-21 overexpression before , which appeared to downregulate the Bax expression , upregulate the Bcl-2 expression and decrease caspase-3 activity\n",
      "[However, ,, such, effect, was, partly, prevented, by, treatment, of, cells, with, miR-21, overexpression, before, ,, which, appeared, to, downregulate, the, Bax, expression, ,, upregulate, the, Bcl-2, expression, and, decrease, caspase-3, activity]\n",
      "TextBetween: However , such effect was partly prevented by treatment of cells with miR-21 overexpression before , which appeared to downregulate the Bax expression , upregulate the Bcl-2 expression and decrease caspase-3 activity\n",
      "infer direction from token set [However, ,, such, effect, was, partly, prevented, by, treatment, of, cells, with, miR-21, overexpression, before, ,, which, appeared, to, downregulate, the, Bax, expression, ,, upregulate, the, Bcl-2, expression, and, decrease, caspase-3, activity]\n",
      "stemTypes2Count: Counter({'NEG': 3, 'POS': 2})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70950>, {'mir': Counter({'POS': 1}), 'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'POS': 2, 'NEU': 2, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e2400>, {'mir': Counter({'POS': 1}), 'gene': Counter({'POS': 1, 'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(overexpression, 'POS'), (downregulate, 'NEG'), (expression, 'NEU'), (upregulate, 'POS'), (expression, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between alternating', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between alternating', 'reg_detect_major': 'counts'}})\n",
      " hit sentence However, such effect was partly prevented by treatment of cells with miR-21 overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.\n",
      "full sentence However, such effect was partly prevented by treatment of cells with <e1>miR-21</e1> overexpression before, which appeared to downregulate the Bax expression, upregulate the <e2>Bcl-2</e2> expression and decrease caspase-3 activity.\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between alternating', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'However, such effect was partly prevented by treatment of cells with <e1>miR-21</e1> overexpression before, which appeared to downregulate the Bax expression, upregulate the <e2>Bcl-2</e2> expression and decrease caspase-3 activity.', 'full_sentence': 'However, such effect was partly prevented by treatment of cells with <e1>miR-21</e1> overexpression before, which appeared to downregulate the Bax expression, upregulate the <e2>Bcl-2</e2> expression and decrease caspase-3 activity.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (69, 75), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (165, 170), 'entity_text': 'Bcl-2'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d70.s12.p1\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('MIR_GENE', 'UP', 'TRUE', 'FALSE')\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between alternating', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'However, such effect was partly prevented by treatment of cells with <e1>miR-21</e1> overexpression before, which appeared to downregulate the Bax expression, upregulate the <e2>Bcl-2</e2> expression and decrease caspase-3 activity.', 'full_sentence': 'However, such effect was partly prevented by treatment of cells with <e1>miR-21</e1> overexpression before, which appeared to downregulate the Bax expression, upregulate the <e2>Bcl-2</e2> expression and decrease caspase-3 activity.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (69, 75), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (165, 170), 'entity_text': 'Bcl-2'}}\n",
      "add special case miR-21\n",
      "add special case Bcl-2\n",
      "miR-21 12 Bcl-2 26\n",
      "(u\"However, such effect was partly prevented by treatment of cells with miR-21 overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.\", 12, 26)\n",
      "geneword Bcl-2 mirword miR-21\n",
      "Conjuncts [upregulate, decrease, prevented]\n",
      "neighbour upregulate the\n",
      "neighbour upregulate the\n",
      "neighbour decrease caspase-3\n",
      "neighbour decrease caspase-3\n",
      "Conjunction before fc prevented\n",
      "fc add children for prevented\n",
      "Conjuncts [prevented, decrease, upregulate]\n",
      "neighbour prevented by\n",
      "neighbour prevented by\n",
      "neighbour decrease caspase-3\n",
      "neighbour decrease caspase-3\n",
      "Conjunction before fc upregulate\n",
      "fc add children for upregulate\n",
      "Conjuncts [prevented, upregulate, decrease]\n",
      "neighbour prevented by\n",
      "neighbour prevented by\n",
      "neighbour upregulate the\n",
      "neighbour upregulate the\n",
      "Conjunction before fc decrease\n",
      "fc add children for decrease\n",
      "Conjunctions ( conj )\n",
      "prevented [prevented, by, treatment, of, cells, with, miR-21, overexpression, before, upregulate, Bcl-2, expression, decrease, caspase-3, activity]\n",
      "achildren {miR-21}\n",
      "achildren False\n",
      "Conjuncts [prevented, treatment]\n",
      "neighbour prevented by\n",
      "neighbour prevented by\n",
      "Conjunction before fc treatment\n",
      "fc add children for treatment\n",
      "fc add children for treatment\n",
      "fc add children for treatment\n",
      "fc add children for treatment\n",
      "Conjuncts [treatment, cells]\n",
      "neighbour treatment of\n",
      "neighbour treatment of\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "Conjuncts [treatment, overexpression]\n",
      "neighbour treatment of\n",
      "neighbour treatment of\n",
      "Conjunction before fc overexpression\n",
      "fc add children for overexpression\n",
      "fc add children for overexpression\n",
      "Conjunctions ( nmod )\n",
      "treatment [prevented, by, treatment, of, cells, with, miR-21, overexpression, before]\n",
      "SDP sent: However, such effect was partly prevented by treatment of cells with miR-21 overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.\n",
      "SDP     : [(miR-21, 'NOUN', 'compound', False), (overexpression, 'NOUN', 'nmod', False), (treatment, 'NOUN', 'nmod', False), (prevented, 'VERB', 'ROOT', True), (upregulate, 'VERB', 'conj', False), (expression, 'NOUN', 'dobj', False), (Bcl-2, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule [to, downregulate, the, Bax, expression]\n",
      "deepcheck conj verb\n",
      "deepcheck connection rule noun found decrease caspase-3 [decrease, caspase-3, activity]\n",
      "deepcheck conj verb\n",
      "0 [However, ,, such, effect, was, partly, prevented, by, treatment, of, cells, with, miR-21, overexpression, before, ,, which, appeared, to, downregulate, the, Bax, expression, ,, upregulate, the, Bcl-2, expression, and, decrease, caspase-3, activity] 32 0 However, such effect was partly prevented by treatment of cells with miR-21 overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.\n",
      "DeepCheck Conj Subtree\n",
      "[by, treatment, of, cells, with, miR-21, overexpression, before, ,, which, appeared, to, downregulate, the, Bax, expression]\n",
      "DeepCheck Conj Subtree\n",
      "[of, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[with, miR-21, overexpression]\n",
      "deepcheck connection rule [to, downregulate, the, Bax, expression]\n",
      "deepcheck conj verb\n",
      "deepcheck connection rule noun found decrease caspase-3 [decrease, caspase-3, activity]\n",
      "deepcheck conj verb\n",
      "0 [However, ,, such, effect, was, partly, prevented, by, treatment, of, cells, with, miR-21, overexpression, before, ,, which, appeared, to, downregulate, the, Bax, expression, ,, upregulate, the, Bcl-2, expression, and, decrease, caspase-3, activity] 32 0 However, such effect was partly prevented by treatment of cells with miR-21 overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.\n",
      "Recognized overlapping entities: {Interval(165, 170, ('GENE_OR_GENE_PRODUCT', 'Bcl-2'))}\n",
      "passive? False\n",
      "Counter({'POS': 2, 'NEG': 1})\n",
      "Compartment Text However , such effect was partly prevented by treatment of cells with miR-21 overexpression before , which appeared to downregulate the Bax expression , upregulate the Bcl-2 expression and decrease caspase-3 activity\n",
      "[However, ,, such, effect, was, partly, prevented, by, treatment, of, cells, with, miR-21, overexpression, before, ,, which, appeared, to, downregulate, the, Bax, expression, ,, upregulate, the, Bcl-2, expression, and, decrease, caspase-3, activity]\n",
      "TextBetween: However , such effect was partly prevented by treatment of cells with miR-21 overexpression before , which appeared to downregulate the Bax expression , upregulate the Bcl-2 expression and decrease caspase-3 activity\n",
      "infer direction from token set [However, ,, such, effect, was, partly, prevented, by, treatment, of, cells, with, miR-21, overexpression, before, ,, which, appeared, to, downregulate, the, Bax, expression, ,, upregulate, the, Bcl-2, expression, and, decrease, caspase-3, activity]\n",
      "stemTypes2Count: Counter({'NEG': 3, 'POS': 2})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954e18>, {'mir': Counter({'POS': 1}), 'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'POS': 2, 'NEU': 2, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e26a8>, {'mir': Counter({'POS': 1}), 'gene': Counter({'POS': 1, 'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(overexpression, 'POS'), (downregulate, 'NEG'), (expression, 'NEU'), (upregulate, 'POS'), (expression, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between alternating', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between alternating', 'reg_detect_major': 'counts'}})\n",
      " hit sentence However, such effect was partly prevented by treatment of cells with miR-21 overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.\n",
      "full sentence However, such effect was partly prevented by treatment of cells with <e1>miR-21</e1> overexpression before, which appeared to downregulate the Bax expression, upregulate the <e2>Bcl-2</e2> expression and decrease caspase-3 activity.\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between alternating', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'However, such effect was partly prevented by treatment of cells with <e1>miR-21</e1> overexpression before, which appeared to downregulate the Bax expression, upregulate the <e2>Bcl-2</e2> expression and decrease caspase-3 activity.', 'full_sentence': 'However, such effect was partly prevented by treatment of cells with <e1>miR-21</e1> overexpression before, which appeared to downregulate the Bax expression, upregulate the <e2>Bcl-2</e2> expression and decrease caspase-3 activity.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (69, 75), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (165, 170), 'entity_text': 'Bcl-2'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d70.s12.p2\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case miR-21\n",
      "add special case caspase-3\n",
      "miR-21 12 caspase-3 30\n",
      "(u\"However, such effect was partly prevented by treatment of cells with miR-21 overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.\", 12, 30)\n",
      "geneword caspase-3 mirword miR-21\n",
      "Conjuncts [upregulate, decrease, prevented]\n",
      "neighbour upregulate the\n",
      "neighbour upregulate the\n",
      "neighbour decrease caspase-3\n",
      "neighbour decrease caspase-3\n",
      "Conjunction before fc prevented\n",
      "fc add children for prevented\n",
      "Conjuncts [prevented, decrease, upregulate]\n",
      "neighbour prevented by\n",
      "neighbour prevented by\n",
      "neighbour decrease caspase-3\n",
      "neighbour decrease caspase-3\n",
      "Conjunction before fc upregulate\n",
      "fc add children for upregulate\n",
      "Conjuncts [prevented, upregulate, decrease]\n",
      "neighbour prevented by\n",
      "neighbour prevented by\n",
      "neighbour upregulate the\n",
      "neighbour upregulate the\n",
      "Conjunction before fc decrease\n",
      "fc add children for decrease\n",
      "Conjunctions ( conj )\n",
      "prevented [prevented, by, treatment, of, cells, with, miR-21, overexpression, before, upregulate, Bcl-2, expression, decrease, caspase-3, activity]\n",
      "achildren {miR-21}\n",
      "achildren False\n",
      "Conjuncts [prevented, treatment]\n",
      "neighbour prevented by\n",
      "neighbour prevented by\n",
      "Conjunction before fc treatment\n",
      "fc add children for treatment\n",
      "fc add children for treatment\n",
      "fc add children for treatment\n",
      "fc add children for treatment\n",
      "Conjuncts [treatment, cells]\n",
      "neighbour treatment of\n",
      "neighbour treatment of\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "Conjuncts [treatment, overexpression]\n",
      "neighbour treatment of\n",
      "neighbour treatment of\n",
      "Conjunction before fc overexpression\n",
      "fc add children for overexpression\n",
      "fc add children for overexpression\n",
      "Conjunctions ( nmod )\n",
      "treatment [prevented, by, treatment, of, cells, with, miR-21, overexpression, before]\n",
      "SDP sent: However, such effect was partly prevented by treatment of cells with miR-21 overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.\n",
      "SDP     : [(miR-21, 'NOUN', 'compound', False), (overexpression, 'NOUN', 'nmod', False), (treatment, 'NOUN', 'nmod', False), (prevented, 'VERB', 'ROOT', True), (upregulate, 'VERB', 'conj', True), (decrease, 'VERB', 'conj', False), (activity, 'NOUN', 'dobj', False), (caspase-3, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule [to, downregulate, the, Bax, expression]\n",
      "deepcheck conj verb\n",
      "deepcheck connection rule noun found decrease caspase-3 [decrease, caspase-3, activity]\n",
      "deepcheck conj verb\n",
      "0 [However, ,, such, effect, was, partly, prevented, by, treatment, of, cells, with, miR-21, overexpression, before, ,, which, appeared, to, downregulate, the, Bax, expression, ,, upregulate, the, Bcl-2, expression, and, decrease, caspase-3, activity] 32 0 However, such effect was partly prevented by treatment of cells with miR-21 overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.\n",
      "DeepCheck Conj Subtree\n",
      "[by, treatment, of, cells, with, miR-21, overexpression, before, ,, which, appeared, to, downregulate, the, Bax, expression]\n",
      "DeepCheck Conj Subtree\n",
      "[of, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[with, miR-21, overexpression]\n",
      "deepcheck connection rule [to, downregulate, the, Bax, expression]\n",
      "deepcheck conj verb\n",
      "deepcheck connection rule noun found decrease caspase-3 [decrease, caspase-3, activity]\n",
      "deepcheck conj verb\n",
      "0 [However, ,, such, effect, was, partly, prevented, by, treatment, of, cells, with, miR-21, overexpression, before, ,, which, appeared, to, downregulate, the, Bax, expression, ,, upregulate, the, Bcl-2, expression, and, decrease, caspase-3, activity] 32 0 However, such effect was partly prevented by treatment of cells with miR-21 overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.\n",
      "Recognized overlapping entities: {Interval(195, 204, ('GENE_OR_GENE_PRODUCT', 'caspase-3'))}\n",
      "passive? False\n",
      "Counter({'POS': 2, 'NEG': 2})\n",
      "Compartment Text However , such effect was partly prevented by treatment of cells with miR-21 overexpression before , which appeared to downregulate the Bax expression , upregulate the Bcl-2 expression and decrease caspase-3 activity\n",
      "[However, ,, such, effect, was, partly, prevented, by, treatment, of, cells, with, miR-21, overexpression, before, ,, which, appeared, to, downregulate, the, Bax, expression, ,, upregulate, the, Bcl-2, expression, and, decrease, caspase-3, activity]\n",
      "TextBetween: However , such effect was partly prevented by treatment of cells with miR-21 overexpression before , which appeared to downregulate the Bax expression , upregulate the Bcl-2 expression and decrease caspase-3 activity\n",
      "infer direction from token set [However, ,, such, effect, was, partly, prevented, by, treatment, of, cells, with, miR-21, overexpression, before, ,, which, appeared, to, downregulate, the, Bax, expression, ,, upregulate, the, Bcl-2, expression, and, decrease, caspase-3, activity]\n",
      "stemTypes2Count: Counter({'NEG': 3, 'POS': 2})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e27b8>, {'mir': Counter({'POS': 1}), 'gene': Counter({'NEG': 1})})\n",
      "defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e27b8>, {'mir': Counter({'POS': 1}), 'gene': Counter({'NEG': 1})})\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}})\n",
      " hit sentence However, such effect was partly prevented by treatment of cells with miR-21 overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.\n",
      "full sentence However, such effect was partly prevented by treatment of cells with <e1>miR-21</e1> overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease <e2>caspase-3</e2> activity.\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'However, such effect was partly prevented by treatment of cells with <e1>miR-21</e1> overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease <e2>caspase-3</e2> activity.', 'full_sentence': 'However, such effect was partly prevented by treatment of cells with <e1>miR-21</e1> overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease <e2>caspase-3</e2> activity.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (69, 75), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (195, 204), 'entity_text': 'caspase-3'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d70.s14.p2\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miR-21\n",
      "add special case caspase-3\n",
      "miR-21 8 caspase-3 27\n",
      "(u\"Taken together, these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\", 8, 27)\n",
      "geneword caspase-3 mirword miR-21\n",
      "Conjuncts [activity, ratio]\n",
      "neighbour activity ,\n",
      "neighbour activity which\n",
      "Conjunction before fc ratio\n",
      "fc add children for ratio\n",
      "Conjuncts [ratio, activity]\n",
      "neighbour ratio and\n",
      "neighbour ratio and\n",
      "Conjunction before fc activity\n",
      "fc add children for activity\n",
      "Conjunctions ( conj )\n",
      "ratio [Bax/Bcl-2, ratio, caspase-3, activity]\n",
      "Conjuncts [apoptosis, cells]\n",
      "neighbour apoptosis in\n",
      "neighbour apoptosis in\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conjuncts [inhibit, part]\n",
      "neighbour inhibit TMZ-induced\n",
      "neighbour inhibit TMZ-induced\n",
      "Conjunction before fc part\n",
      "fc add children for part\n",
      "Conjuncts [possibility, overexpression]\n",
      "neighbour possibility of\n",
      "neighbour possibility of\n",
      "Conjunction before fc overexpression\n",
      "fc add children for overexpression\n",
      "fc add children for overexpression\n",
      "fc add children for overexpression\n",
      "Conjuncts [overexpression, resistance]\n",
      "neighbour overexpression in\n",
      "neighbour overexpression in\n",
      "Conjunction before fc resistance\n",
      "fc add children for resistance\n",
      "fc add children for resistance\n",
      "fc add children for resistance\n",
      "Conjuncts [resistance, therapy]\n",
      "neighbour resistance to\n",
      "neighbour resistance to\n",
      "Conjunction before fc therapy\n",
      "fc add children for therapy\n",
      "fc add children for therapy\n",
      "fc add children for therapy\n",
      "Conjuncts [therapy, TMZ]\n",
      "neighbour therapy of\n",
      "neighbour therapy of\n",
      "Conjunction before fc TMZ\n",
      "fc add children for TMZ\n",
      "Conunct merge check: part cells\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: cells overexpression\n",
      "cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the\n",
      "Conjunctions ( nmod )\n",
      "cells [apoptosis, in, U87MG, cells]\n",
      "part [inhibit, in, part]\n",
      "overexpression [possibility, of, miR-21, overexpression, in, clinical, resistance, to, chemotherapeutic, therapy, of, TMZ]\n",
      "SDP sent: Taken together, these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "SDP     : [(miR-21, 'NOUN', 'nsubj', False), (inhibit, 'VERB', 'ccomp', False), (decreasing, 'VERB', 'advcl', False), (ratio, 'NOUN', 'dobj', True), (activity, 'NOUN', 'conj', False), (caspase-3, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule [by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity, ,, which, highlighted, the, possibility, of, miR-21, overexpression, in, the, clinical, resistance, to, chemotherapeutic, therapy, of, TMZ]\n",
      "DeepCheck Conj Subtree\n",
      "[caspase-3, activity]\n",
      "0 [these, results, suggested, that, over-express, miR-21, could, inhibit, TMZ-induced, apoptosis, in, U87MG, cells, ,, at, least, in, part, ,, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity] 26 16 these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "1 [Taken, together] 2 0 Taken together, these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "2 [] 0\n",
      "empty compartment []\n",
      "3 [which, highlighted, the, possibility, of, miR-21, overexpression, in, the, clinical, resistance, to, chemotherapeutic, therapy, of, TMZ] 16 189 which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "DeepCheck Conj Subtree\n",
      "[in, U87MG, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[at, least, in, part]\n",
      "deepcheck connection rule [by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity, ,, which, highlighted, the, possibility, of, miR-21, overexpression, in, the, clinical, resistance, to, chemotherapeutic, therapy, of, TMZ]\n",
      "DeepCheck Conj Subtree\n",
      "[caspase-3, activity]\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-21, overexpression, in, the, clinical, resistance, to, chemotherapeutic, therapy, of, TMZ]\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, clinical, resistance, to, chemotherapeutic, therapy, of, TMZ]\n",
      "DeepCheck Conj Subtree\n",
      "[to, chemotherapeutic, therapy, of, TMZ]\n",
      "DeepCheck Conj Subtree\n",
      "[of, TMZ]\n",
      "0 [these, results, suggested, that, over-express, miR-21, could, inhibit, TMZ-induced, apoptosis, in, U87MG, cells, ,, at, least, in, part, ,, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity] 26 16 these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "1 [Taken, together] 2 0 Taken together, these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "2 [] 0\n",
      "empty compartment []\n",
      "3 [which, highlighted, the, possibility, of, miR-21, overexpression, in, the, clinical, resistance, to, chemotherapeutic, therapy, of, TMZ] 16 189 which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "Recognized overlapping entities: {Interval(169, 178, ('GENE_OR_GENE_PRODUCT', 'caspase-3'))}\n",
      "passive? False\n",
      "Counter({'NEG': 2})\n",
      "Compartment Text these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells , at least in part , by decreasing Bax/Bcl-2 ratio and caspase-3 activity\n",
      "[these, results, suggested, that, over-express, miR-21, could, inhibit, TMZ-induced, apoptosis, in, U87MG, cells, ,, at, least, in, part, ,, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity]\n",
      "TextBetween: these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells , at least in part , by decreasing Bax/Bcl-2 ratio and caspase-3 activity\n",
      "infer direction from token set [these, results, suggested, that, over-express, miR-21, could, inhibit, TMZ-induced, apoptosis, in, U87MG, cells, ,, at, least, in, part, ,, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity]\n",
      "stemTypes2Count: Counter({'NEG': 2, 'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954c80>, {'mir': Counter({'POS': 1, 'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70620>, {'mir': Counter({'POS': 1, 'NEG': 1}), 'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(over-express, 'POS'), (inhibit, 'NEG'), (decreasing, 'NEG'), (activity, 'NEU')]\n",
      "Between Tokens [(inhibit, 'NEG'), (decreasing, 'NEG')]\n",
      "Between Tokens Switch\n",
      "counts between final False\n",
      "['that', 'over-express', 'miR-21', 'could', 'inhibit', 'TMZ-induced', 'apoptosis', 'in', 'U87MG', 'cells', ',', 'at', 'least', 'in', 'part', ',', 'by', 'decreasing', 'Bax/Bcl-2', 'ratio', 'and', 'caspase-3', 'activity']\n",
      "Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "[(over-express, 'POS'), (inhibit, 'NEG'), (decreasing, 'NEG'), (activity, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Taken together, these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "full sentence Taken together, these results suggested that over-express <e1>miR-21</e1> could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and <e2>caspase-3</e2> activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Taken together, these results suggested that over-express <e1>miR-21</e1> could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and <e2>caspase-3</e2> activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.', 'full_sentence': 'Taken together, these results suggested that over-express <e1>miR-21</e1> could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and <e2>caspase-3</e2> activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (58, 64), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (169, 178), 'entity_text': 'caspase-3'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d80.s5.p1\n",
      "IS:  ('GENE_MIR', 'NEU')\n",
      "SB:  ('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'We identified putative miR sites in the <e2>CDK6</e2> including <e1>microRNA 124a</e1>, a brain enriched microRNA.', 'full_sentence': 'We identified putative miR sites in the <e2>CDK6</e2> including <e1>microRNA 124a</e1>, a brain enriched microRNA.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (40, 44), 'entity_text': 'CDK6'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (55, 68), 'entity_text': 'microRNA 124a'}}\n",
      "add special case CDK6\n",
      "add special case microRNA 124a\n",
      "CDK6 7 124a 10\n",
      "(u\"We identified putative miR sites in the CDK6 including microRNA 124a, a brain enriched microRNA.\", 7, 10)\n",
      "geneword CDK6 mirword 124a\n",
      "Conjunctions ( conj )\n",
      "microRNA [a, microRNA]\n",
      "Conjuncts [sites, CDK6]\n",
      "neighbour sites in\n",
      "neighbour sites in\n",
      "Conjunction before fc CDK6\n",
      "fc add children for CDK6\n",
      "fc add children for CDK6\n",
      "Conjuncts [CDK6, 124a]\n",
      "neighbour CDK6 including\n",
      "neighbour CDK6 including\n",
      "Conjunction before fc 124a\n",
      "fc add children for 124a\n",
      "fc add children for 124a\n",
      "fc add children for 124a\n",
      "Conjunctions ( nmod )\n",
      "CDK6 [sites, in, CDK6, including, microRNA, 124a, a, brain, enriched, microRNA]\n",
      "achildren {124a}\n",
      "achildren False\n",
      "SDP sent: We identified putative miR sites in the CDK6 including microRNA 124a, a brain enriched microRNA.\n",
      "SDP     : [(CDK6, 'NOUN', 'nmod', False), (identified, 'VERB', 'ROOT', False), (sites, 'NOUN', 'dobj', False), (CDK6, 'NOUN', 'nmod', False)]\n",
      "0 [We, identified, putative, miR, sites, in, the, CDK6, including, microRNA, 124a, ,, a, brain, enriched, microRNA] 16 0 We identified putative miR sites in the CDK6 including microRNA 124a, a brain enriched microRNA.\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, CDK6, including, microRNA, 124a, ,, a, brain, enriched, microRNA]\n",
      "DeepCheck Conj Subtree\n",
      "[including, microRNA, 124a, ,, a, brain, enriched, microRNA]\n",
      "deepcheck conj verb\n",
      "deepcheck conj verb\n",
      "0 [We, identified, putative, miR, sites, in, the, CDK6, including, microRNA, 124a, ,, a, brain, enriched, microRNA] 16 0 We identified putative miR sites in the CDK6 including microRNA 124a, a brain enriched microRNA.\n",
      "Recognized overlapping entities: {Interval(40, 44, ('GENE_OR_GENE_PRODUCT', 'CDK6'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text We identified putative miR sites in the CDK6 including microRNA 124a , a brain enriched microRNA\n",
      "[We, identified, putative, miR, sites, in, the, CDK6, including, microRNA, 124a, ,, a, brain, enriched, microRNA]\n",
      "TextBetween: We identified putative miR sites in the CDK6 including microRNA 124a , a brain enriched microRNA\n",
      "infer direction from token set [We, identified, putative, miR, sites, in, the, CDK6, including, microRNA, 124a, ,, a, brain, enriched, microRNA]\n",
      "stemTypes2Count: Counter({'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70488>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954d90>, {'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(including, 'NEU')]\n",
      "Between Tokens [(including, 'NEU')]\n",
      "counts between final False\n",
      "['in', 'the', 'CDK6', 'including', 'microRNA', '124a', ',', 'a']\n",
      "Counter({'NEU': 1})\n",
      "[(including, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence We identified putative miR sites in the CDK6 including microRNA 124a, a brain enriched microRNA.\n",
      "full sentence We identified putative miR sites in the <e2>CDK6</e2> including <e1>microRNA 124a</e1>, a brain enriched microRNA.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'We identified putative miR sites in the <e2>CDK6</e2> including <e1>microRNA 124a</e1>, a brain enriched microRNA.', 'full_sentence': 'We identified putative miR sites in the <e2>CDK6</e2> including <e1>microRNA 124a</e1>, a brain enriched microRNA.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (40, 44), 'entity_text': 'CDK6'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (55, 68), 'entity_text': 'microRNA 124a'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d80.s9.p1\n",
      "IS:  ('MIR_GENE', 'NEU')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Regulation of <e2>cyclin dependent kinase 6</e2> by <e1>microRNA 124</e1> in medulloblastoma.', 'full_sentence': 'Regulation of <e2>cyclin dependent kinase 6</e2> by <e1>microRNA 124</e1> in medulloblastoma.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (14, 39), 'entity_text': 'cyclin dependent kinase 6'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (43, 55), 'entity_text': 'microRNA 124'}}\n",
      "add special case cyclin dependent kinase 6\n",
      "add special case microRNA 124\n",
      "6 5 124 8\n",
      "(u\"Regulation of cyclin dependent kinase 6 by microRNA 124 in medulloblastoma.\", 5, 8)\n",
      "geneword 6 mirword 124\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [Regulation, kinase]\n",
      "neighbour Regulation of\n",
      "neighbour Regulation of\n",
      "Conjunction before fc kinase\n",
      "fc add children for kinase\n",
      "fc add children for kinase\n",
      "fc add children for kinase\n",
      "fc add children for kinase\n",
      "Conjuncts [Regulation, microRNA]\n",
      "neighbour Regulation of\n",
      "neighbour Regulation of\n",
      "Conjunction before fc microRNA\n",
      "fc add children for microRNA\n",
      "fc add children for microRNA\n",
      "fc add children for microRNA\n",
      "Conjuncts [microRNA, medulloblastoma]\n",
      "neighbour microRNA 124\n",
      "neighbour microRNA 124\n",
      "Conjunction before fc medulloblastoma\n",
      "fc add children for medulloblastoma\n",
      "Conjunctions ( nmod )\n",
      "kinase [Regulation, of, cyclin, dependent, kinase, 6, by, microRNA, 124, in, medulloblastoma]\n",
      "achildren {124}\n",
      "achildren False\n",
      "SDP sent: Regulation of cyclin dependent kinase 6 by microRNA 124 in medulloblastoma.\n",
      "SDP     : [(6, 'NUM', 'nummod', False), (kinase, 'NOUN', 'nmod', False), (Regulation, 'NOUN', 'ROOT', False), (microRNA, 'NOUN', 'nmod', False), (124, 'NUM', 'nummod', False)]\n",
      "0 [Regulation, of, cyclin, dependent, kinase, 6, by, microRNA, 124, in, medulloblastoma] 11 0 Regulation of cyclin dependent kinase 6 by microRNA 124 in medulloblastoma.\n",
      "DeepCheck Conj Subtree\n",
      "[of, cyclin, dependent, kinase, 6]\n",
      "DeepCheck Conj Subtree\n",
      "[by, microRNA, 124, in, medulloblastoma]\n",
      "DeepCheck Conj Subtree\n",
      "[in, medulloblastoma]\n",
      "0 [Regulation, of, cyclin, dependent, kinase, 6, by, microRNA, 124, in, medulloblastoma] 11 0 Regulation of cyclin dependent kinase 6 by microRNA 124 in medulloblastoma.\n",
      "Recognized overlapping entities: {Interval(14, 39, ('GENE_OR_GENE_PRODUCT', 'cyclin dependent kinase 6'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text Regulation of cyclin dependent kinase 6 by microRNA 124 in medulloblastoma\n",
      "[Regulation, of, cyclin, dependent, kinase, 6, by, microRNA, 124, in, medulloblastoma]\n",
      "TextBetween: Regulation of cyclin dependent kinase 6 by microRNA 124 in medulloblastoma\n",
      "infer direction from token set [Regulation, of, cyclin, dependent, kinase, 6, by, microRNA, 124, in, medulloblastoma]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954e18>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954b70>, {'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(dependent, 'NEU')]\n",
      "counts between 3 Counter() NEU\n",
      "Between Tokens []\n",
      "counts between final False\n",
      "['dependent', 'kinase', '6', 'by', 'microRNA', '124', 'in', 'medulloblastoma']\n",
      "Counter()\n",
      "[]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Regulation of cyclin dependent kinase 6 by microRNA 124 in medulloblastoma.\n",
      "full sentence Regulation of <e2>cyclin dependent kinase 6</e2> by <e1>microRNA 124</e1> in medulloblastoma.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Regulation of <e2>cyclin dependent kinase 6</e2> by <e1>microRNA 124</e1> in medulloblastoma.', 'full_sentence': 'Regulation of <e2>cyclin dependent kinase 6</e2> by <e1>microRNA 124</e1> in medulloblastoma.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (14, 39), 'entity_text': 'cyclin dependent kinase 6'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (43, 55), 'entity_text': 'microRNA 124'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d87.s9.p3\n",
      "IS:  ('GENE_MIR', 'NEU')\n",
      "SB:  ('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e2>BAG2</e2> levels in cells are under the physiological control of the microRNA <e1>miR-128a</e1>, which can tune paired helical filament Tau levels in neurons.', 'full_sentence': '<e2>BAG2</e2> levels in cells are under the physiological control of the microRNA <e1>miR-128a</e1>, which can tune paired helical filament Tau levels in neurons.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (0, 4), 'entity_text': 'BAG2'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (73, 81), 'entity_text': 'miR-128a'}}\n",
      "add special case BAG2\n",
      "add special case miR-128a\n",
      "BAG2 0 miR-128a 12\n",
      "(u\"BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons.\", 0, 12)\n",
      "geneword BAG2 mirword miR-128a\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [levels, cells]\n",
      "neighbour levels in\n",
      "neighbour levels in\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "Conjuncts [control, miR-128a]\n",
      "neighbour control of\n",
      "neighbour control of\n",
      "Conjunction before fc miR-128a\n",
      "fc add children for miR-128a\n",
      "fc add children for miR-128a\n",
      "Conjuncts [levels, neurons]\n",
      "neighbour levels in\n",
      "neighbour levels in\n",
      "Conjunction before fc neurons\n",
      "fc add children for neurons\n",
      "Conunct merge check: cells miR-128a\n",
      "cells are under the physiological\n",
      "Conunct merge check: miR-128a neurons\n",
      "miR-128a, which can tune paired helical filament Tau\n",
      "Conjunctions ( nmod )\n",
      "cells [levels, in, cells]\n",
      "miR-128a [control, of, microRNA, miR-128a]\n",
      "neurons [levels, in, neurons]\n",
      "SDP sent: BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons.\n",
      "SDP     : [(BAG2, 'NOUN', 'compound', False), (levels, 'NOUN', 'nsubj', False), (control, 'NOUN', 'ROOT', False), (miR-128a, 'NOUN', 'nmod', False)]\n",
      "0 [BAG2, levels, in, cells, are, under, the, physiological, control, of, the, microRNA, miR-128a, ,, which, can, tune, paired, helical, filament, Tau, levels, in, neurons] 24 0 BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons.\n",
      "DeepCheck Conj Subtree\n",
      "[in, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, microRNA, miR-128a, ,, which, can, tune, paired, helical, filament, Tau, levels, in, neurons]\n",
      "DeepCheck Conj Subtree\n",
      "[in, neurons]\n",
      "0 [BAG2, levels, in, cells, are, under, the, physiological, control, of, the, microRNA, miR-128a, ,, which, can, tune, paired, helical, filament, Tau, levels, in, neurons] 24 0 BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons.\n",
      "Recognized overlapping entities: {Interval(0, 4, ('GENE_OR_GENE_PRODUCT', 'BAG2'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text BAG2 levels in cells are under the physiological control of the microRNA miR-128a , which can tune paired helical filament Tau levels in neurons\n",
      "[BAG2, levels, in, cells, are, under, the, physiological, control, of, the, microRNA, miR-128a, ,, which, can, tune, paired, helical, filament, Tau, levels, in, neurons]\n",
      "TextBetween: BAG2 levels in cells are under the physiological control of the microRNA miR-128a , which can tune paired helical filament Tau levels in neurons\n",
      "infer direction from token set [BAG2, levels, in, cells, are, under, the, physiological, control, of, the, microRNA, miR-128a, ,, which, can, tune, paired, helical, filament, Tau, levels, in, neurons]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e20d0>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954e18>, {'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(levels, 'NEU'), (control, 'NEU')]\n",
      "Between Tokens [(levels, 'NEU'), (control, 'NEU')]\n",
      "counts between final False\n",
      "['BAG2', 'levels', 'in', 'cells', 'are', 'under', 'the', 'physiological', 'control', 'of', 'the', 'microRNA', 'miR-128a', ',', 'which']\n",
      "Counter({'NEU': 2})\n",
      "[(levels, 'NEU'), (control, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons.\n",
      "full sentence <e2>BAG2</e2> levels in cells are under the physiological control of the microRNA <e1>miR-128a</e1>, which can tune paired helical filament Tau levels in neurons.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e2>BAG2</e2> levels in cells are under the physiological control of the microRNA <e1>miR-128a</e1>, which can tune paired helical filament Tau levels in neurons.', 'full_sentence': '<e2>BAG2</e2> levels in cells are under the physiological control of the microRNA <e1>miR-128a</e1>, which can tune paired helical filament Tau levels in neurons.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (0, 4), 'entity_text': 'BAG2'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (73, 81), 'entity_text': 'miR-128a'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d88.s8.p2\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case miR-15a\n",
      "add special case arginyl-tRNA synthetase\n",
      "miR-15a 16 synthetase 26\n",
      "(u\"An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\", 16, 26)\n",
      "geneword synthetase mirword miR-15a\n",
      "Conjuncts [levels, leukemia]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "Conjuncts [miR-16-1, miR-15a]\n",
      "neighbour miR-16-1 expression\n",
      "neighbour miR-16-1 expression\n",
      "Conjunction before fc miR-15a\n",
      "fc add children for miR-15a\n",
      "Conjuncts [miR-15a, miR-16-1]\n",
      "neighbour miR-15a and\n",
      "neighbour miR-15a and\n",
      "Conjunction before fc miR-16-1\n",
      "Conjuncts [leukemia, levels]\n",
      "neighbour leukemia (\n",
      "neighbour leukemia B-CLL\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conunct merge check: leukemia B-CLL\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: B-CLL RARS\n",
      ") between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase\n",
      "Conunct merge check: RARS enzyme\n",
      "),\n",
      "Conjunctions ( conj )\n",
      "leukemia [in, B, cell, chronic, lymphocytic, leukemia, B-CLL, between, miR-15a, miR-16-1, expression, expression, levels, of, arginyl-tRNA, synthetase, RARS, enzyme]\n",
      "B-CLL [(, )]\n",
      "RARS [(, )]\n",
      "enzyme [an, complex]\n",
      "achildren {miR-15a}\n",
      "achildren False\n",
      "Conjuncts [shown, leukemia]\n",
      "neighbour shown in\n",
      "neighbour shown in\n",
      "Conjunction before fc leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "Conjuncts [leukemia, expression]\n",
      "neighbour leukemia (\n",
      "neighbour leukemia B-CLL\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "Conjuncts [levels, synthetase]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc synthetase\n",
      "fc add children for synthetase\n",
      "fc add children for synthetase\n",
      "fc add children for synthetase\n",
      "fc add children for synthetase\n",
      "Conjuncts [associates, p43]\n",
      "neighbour associates with\n",
      "neighbour associates with\n",
      "Conjunction before fc p43\n",
      "fc add children for p43\n",
      "fc add children for p43\n",
      "Conjuncts [associates, complex]\n",
      "neighbour associates with\n",
      "neighbour associates with\n",
      "Conjunction before fc complex\n",
      "fc add children for complex\n",
      "fc add children for complex\n",
      "fc add children for complex\n",
      "Conunct merge check: leukemia B-CLL\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: B-CLL synthetase\n",
      ") between miR-15a and miR-16-1 expression and the expression\n",
      "Conunct merge check: synthetase RARS\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: RARS enzyme\n",
      "),\n",
      "Conunct merge check: enzyme p43\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( nmod )\n",
      "leukemia [shown, in, B, cell, chronic, lymphocytic, leukemia, B-CLL, between, miR-15a, expression]\n",
      "B-CLL [(, )]\n",
      "synthetase [levels, of, arginyl-tRNA, synthetase, RARS, enzyme]\n",
      "RARS [(, )]\n",
      "enzyme [an, associates, in, aminoacyl-tRNA, synthetase, complex]\n",
      "p43 [associates, with, cofactor, p43]\n",
      "SDP sent: An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\n",
      "SDP     : [(miR-15a, 'NOUN', 'compound', False), (expression, 'NOUN', 'nmod', False), (leukemia, 'NOUN', 'nmod', True), (levels, 'NOUN', 'conj', False), (synthetase, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-16-1]\n",
      "DeepCheck Conj Subtree\n",
      "[the, expression, levels, of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex]\n",
      "0 [An, inverse, correlation, has, been, shown, in, B, cell, chronic, lymphocytic, leukemia, (, B-CLL, ), between, miR-15a, and, miR-16-1, expression, and, the, expression, levels, of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex] 44 0 An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\n",
      "DeepCheck Conj Subtree\n",
      "[in, B, cell, chronic, lymphocytic, leukemia, (, B-CLL, ), between, miR-15a, and, miR-16-1, expression, and, the, expression, levels, of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-16-1]\n",
      "DeepCheck Conj Subtree\n",
      "[between, miR-15a, and, miR-16-1, expression]\n",
      "DeepCheck Conj Subtree\n",
      "[the, expression, levels, of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex]\n",
      "DeepCheck Conj Subtree\n",
      "[of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex]\n",
      "DeepCheck Conj Subtree\n",
      "[with, the, cofactor, p43]\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, aminoacyl-tRNA, synthetase, complex]\n",
      "0 [An, inverse, correlation, has, been, shown, in, B, cell, chronic, lymphocytic, leukemia, (, B-CLL, ), between, miR-15a, and, miR-16-1, expression, and, the, expression, levels, of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex] 44 0 An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\n",
      "Recognized overlapping entities: {Interval(154, 177, ('GENE_OR_GENE_PRODUCT', 'arginyl-tRNA synthetase'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text An inverse correlation has been shown in B cell chronic lymphocytic leukemia ( B-CLL ) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase ( RARS ) , an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex\n",
      "(False, {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene neg corr', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\n",
      "full sentence An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between <e1>miR-15a</e1> and miR-16-1 expression and the expression levels of <e2>arginyl-tRNA synthetase</e2> (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene neg corr', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between <e1>miR-15a</e1> and miR-16-1 expression and the expression levels of <e2>arginyl-tRNA synthetase</e2> (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.', 'full_sentence': 'An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between <e1>miR-15a</e1> and miR-16-1 expression and the expression levels of <e2>arginyl-tRNA synthetase</e2> (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (93, 100), 'entity_text': 'miR-15a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (154, 177), 'entity_text': 'arginyl-tRNA synthetase'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d88.s8.p3\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case miR-15a\n",
      "add special case RARS\n",
      "miR-15a 16 RARS 28\n",
      "(u\"An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\", 16, 28)\n",
      "geneword RARS mirword miR-15a\n",
      "Conjuncts [levels, leukemia]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "Conjuncts [miR-16-1, miR-15a]\n",
      "neighbour miR-16-1 expression\n",
      "neighbour miR-16-1 expression\n",
      "Conjunction before fc miR-15a\n",
      "fc add children for miR-15a\n",
      "Conjuncts [miR-15a, miR-16-1]\n",
      "neighbour miR-15a and\n",
      "neighbour miR-15a and\n",
      "Conjunction before fc miR-16-1\n",
      "Conjuncts [leukemia, levels]\n",
      "neighbour leukemia (\n",
      "neighbour leukemia B-CLL\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conunct merge check: leukemia B-CLL\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: B-CLL RARS\n",
      ") between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase\n",
      "Conunct merge check: RARS enzyme\n",
      "),\n",
      "Conjunctions ( conj )\n",
      "leukemia [in, B, cell, chronic, lymphocytic, leukemia, B-CLL, between, miR-15a, miR-16-1, expression, expression, levels, of, arginyl-tRNA, synthetase, RARS, enzyme]\n",
      "B-CLL [(, )]\n",
      "RARS [(, )]\n",
      "enzyme [an, complex]\n",
      "achildren {miR-15a}\n",
      "achildren False\n",
      "Conjuncts [shown, leukemia]\n",
      "neighbour shown in\n",
      "neighbour shown in\n",
      "Conjunction before fc leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "Conjuncts [leukemia, expression]\n",
      "neighbour leukemia (\n",
      "neighbour leukemia B-CLL\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "Conjuncts [levels, synthetase]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc synthetase\n",
      "fc add children for synthetase\n",
      "fc add children for synthetase\n",
      "fc add children for synthetase\n",
      "fc add children for synthetase\n",
      "Conjuncts [associates, p43]\n",
      "neighbour associates with\n",
      "neighbour associates with\n",
      "Conjunction before fc p43\n",
      "fc add children for p43\n",
      "fc add children for p43\n",
      "Conjuncts [associates, complex]\n",
      "neighbour associates with\n",
      "neighbour associates with\n",
      "Conjunction before fc complex\n",
      "fc add children for complex\n",
      "fc add children for complex\n",
      "fc add children for complex\n",
      "Conunct merge check: leukemia B-CLL\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: B-CLL synthetase\n",
      ") between miR-15a and miR-16-1 expression and the expression\n",
      "Conunct merge check: synthetase RARS\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: RARS enzyme\n",
      "),\n",
      "Conunct merge check: enzyme p43\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( nmod )\n",
      "leukemia [shown, in, B, cell, chronic, lymphocytic, leukemia, B-CLL, between, miR-15a, expression]\n",
      "B-CLL [(, )]\n",
      "synthetase [levels, of, arginyl-tRNA, synthetase, RARS, enzyme]\n",
      "RARS [(, )]\n",
      "enzyme [an, associates, in, aminoacyl-tRNA, synthetase, complex]\n",
      "p43 [associates, with, cofactor, p43]\n",
      "SDP sent: An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\n",
      "SDP     : [(miR-15a, 'NOUN', 'compound', False), (expression, 'NOUN', 'nmod', False), (leukemia, 'NOUN', 'nmod', True), (levels, 'NOUN', 'conj', False), (synthetase, 'NOUN', 'nmod', False), (RARS, 'NOUN', 'appos', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-16-1]\n",
      "DeepCheck Conj Subtree\n",
      "[the, expression, levels, of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex]\n",
      "0 [An, inverse, correlation, has, been, shown, in, B, cell, chronic, lymphocytic, leukemia, (, B-CLL, ), between, miR-15a, and, miR-16-1, expression, and, the, expression, levels, of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex] 44 0 An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\n",
      "DeepCheck Conj Subtree\n",
      "[in, B, cell, chronic, lymphocytic, leukemia, (, B-CLL, ), between, miR-15a, and, miR-16-1, expression, and, the, expression, levels, of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-16-1]\n",
      "DeepCheck Conj Subtree\n",
      "[between, miR-15a, and, miR-16-1, expression]\n",
      "DeepCheck Conj Subtree\n",
      "[the, expression, levels, of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex]\n",
      "DeepCheck Conj Subtree\n",
      "[of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex]\n",
      "DeepCheck Conj Subtree\n",
      "[with, the, cofactor, p43]\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, aminoacyl-tRNA, synthetase, complex]\n",
      "0 [An, inverse, correlation, has, been, shown, in, B, cell, chronic, lymphocytic, leukemia, (, B-CLL, ), between, miR-15a, and, miR-16-1, expression, and, the, expression, levels, of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex] 44 0 An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\n",
      "Recognized overlapping entities: {Interval(179, 183, ('CANCER', 'RARS'))}\n",
      "Gene reject: {'CANCER'} due to {'CANCER'}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text An inverse correlation has been shown in B cell chronic lymphocytic leukemia ( B-CLL ) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase ( RARS ) , an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex\n",
      "(False, {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': False, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene neg corr', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\n",
      "full sentence An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between <e1>miR-15a</e1> and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (<e2>RARS</e2>), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': False, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene neg corr', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between <e1>miR-15a</e1> and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (<e2>RARS</e2>), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.', 'full_sentence': 'An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between <e1>miR-15a</e1> and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (<e2>RARS</e2>), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (93, 100), 'entity_text': 'miR-15a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (179, 183), 'entity_text': 'RARS'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d88.s8.p6\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case miR-16-1\n",
      "add special case arginyl-tRNA synthetase\n",
      "miR-16-1 18 synthetase 26\n",
      "(u\"An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\", 18, 26)\n",
      "geneword synthetase mirword miR-16-1\n",
      "Conjuncts [levels, leukemia]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "Conjuncts [miR-16-1, miR-15a]\n",
      "neighbour miR-16-1 expression\n",
      "neighbour miR-16-1 expression\n",
      "Conjunction before fc miR-15a\n",
      "fc add children for miR-15a\n",
      "Conjuncts [miR-15a, miR-16-1]\n",
      "neighbour miR-15a and\n",
      "neighbour miR-15a and\n",
      "Conjunction before fc miR-16-1\n",
      "Conjuncts [leukemia, levels]\n",
      "neighbour leukemia (\n",
      "neighbour leukemia B-CLL\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conunct merge check: leukemia B-CLL\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: B-CLL RARS\n",
      ") between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase\n",
      "Conunct merge check: RARS enzyme\n",
      "),\n",
      "Conjunctions ( conj )\n",
      "leukemia [in, B, cell, chronic, lymphocytic, leukemia, B-CLL, between, miR-15a, miR-16-1, expression, expression, levels, of, arginyl-tRNA, synthetase, RARS, enzyme]\n",
      "B-CLL [(, )]\n",
      "RARS [(, )]\n",
      "enzyme [an, complex]\n",
      "achildren {miR-16-1}\n",
      "achildren False\n",
      "Conjuncts [shown, leukemia]\n",
      "neighbour shown in\n",
      "neighbour shown in\n",
      "Conjunction before fc leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "Conjuncts [leukemia, expression]\n",
      "neighbour leukemia (\n",
      "neighbour leukemia B-CLL\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "Conjuncts [levels, synthetase]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc synthetase\n",
      "fc add children for synthetase\n",
      "fc add children for synthetase\n",
      "fc add children for synthetase\n",
      "fc add children for synthetase\n",
      "Conjuncts [associates, p43]\n",
      "neighbour associates with\n",
      "neighbour associates with\n",
      "Conjunction before fc p43\n",
      "fc add children for p43\n",
      "fc add children for p43\n",
      "Conjuncts [associates, complex]\n",
      "neighbour associates with\n",
      "neighbour associates with\n",
      "Conjunction before fc complex\n",
      "fc add children for complex\n",
      "fc add children for complex\n",
      "fc add children for complex\n",
      "Conunct merge check: leukemia B-CLL\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: B-CLL synthetase\n",
      ") between miR-15a and miR-16-1 expression and the expression\n",
      "Conunct merge check: synthetase RARS\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: RARS enzyme\n",
      "),\n",
      "Conunct merge check: enzyme p43\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( nmod )\n",
      "leukemia [shown, in, B, cell, chronic, lymphocytic, leukemia, B-CLL, between, miR-15a, expression]\n",
      "B-CLL [(, )]\n",
      "synthetase [levels, of, arginyl-tRNA, synthetase, RARS, enzyme]\n",
      "RARS [(, )]\n",
      "enzyme [an, associates, in, aminoacyl-tRNA, synthetase, complex]\n",
      "p43 [associates, with, cofactor, p43]\n",
      "SDP sent: An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\n",
      "SDP     : [(miR-16-1, 'NOUN', 'conj', False), (miR-15a, 'NOUN', 'compound', True), (expression, 'NOUN', 'nmod', False), (leukemia, 'NOUN', 'nmod', True), (levels, 'NOUN', 'conj', False), (synthetase, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-16-1]\n",
      "DeepCheck Conj Subtree\n",
      "[the, expression, levels, of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex]\n",
      "0 [An, inverse, correlation, has, been, shown, in, B, cell, chronic, lymphocytic, leukemia, (, B-CLL, ), between, miR-15a, and, miR-16-1, expression, and, the, expression, levels, of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex] 44 0 An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\n",
      "DeepCheck Conj Subtree\n",
      "[in, B, cell, chronic, lymphocytic, leukemia, (, B-CLL, ), between, miR-15a, and, miR-16-1, expression, and, the, expression, levels, of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-16-1]\n",
      "DeepCheck Conj Subtree\n",
      "[between, miR-15a, and, miR-16-1, expression]\n",
      "DeepCheck Conj Subtree\n",
      "[the, expression, levels, of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex]\n",
      "DeepCheck Conj Subtree\n",
      "[of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex]\n",
      "DeepCheck Conj Subtree\n",
      "[with, the, cofactor, p43]\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, aminoacyl-tRNA, synthetase, complex]\n",
      "0 [An, inverse, correlation, has, been, shown, in, B, cell, chronic, lymphocytic, leukemia, (, B-CLL, ), between, miR-15a, and, miR-16-1, expression, and, the, expression, levels, of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex] 44 0 An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\n",
      "Recognized overlapping entities: {Interval(154, 177, ('GENE_OR_GENE_PRODUCT', 'arginyl-tRNA synthetase'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text An inverse correlation has been shown in B cell chronic lymphocytic leukemia ( B-CLL ) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase ( RARS ) , an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex\n",
      "(False, {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene neg corr', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\n",
      "full sentence An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and <e1>miR-16-1</e1> expression and the expression levels of <e2>arginyl-tRNA synthetase</e2> (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene neg corr', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and <e1>miR-16-1</e1> expression and the expression levels of <e2>arginyl-tRNA synthetase</e2> (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.', 'full_sentence': 'An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and <e1>miR-16-1</e1> expression and the expression levels of <e2>arginyl-tRNA synthetase</e2> (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (105, 113), 'entity_text': 'miR-16-1'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (154, 177), 'entity_text': 'arginyl-tRNA synthetase'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d88.s8.p7\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case miR-16-1\n",
      "add special case RARS\n",
      "miR-16-1 18 RARS 28\n",
      "(u\"An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\", 18, 28)\n",
      "geneword RARS mirword miR-16-1\n",
      "Conjuncts [levels, leukemia]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "Conjuncts [miR-16-1, miR-15a]\n",
      "neighbour miR-16-1 expression\n",
      "neighbour miR-16-1 expression\n",
      "Conjunction before fc miR-15a\n",
      "fc add children for miR-15a\n",
      "Conjuncts [miR-15a, miR-16-1]\n",
      "neighbour miR-15a and\n",
      "neighbour miR-15a and\n",
      "Conjunction before fc miR-16-1\n",
      "Conjuncts [leukemia, levels]\n",
      "neighbour leukemia (\n",
      "neighbour leukemia B-CLL\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conunct merge check: leukemia B-CLL\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: B-CLL RARS\n",
      ") between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase\n",
      "Conunct merge check: RARS enzyme\n",
      "),\n",
      "Conjunctions ( conj )\n",
      "leukemia [in, B, cell, chronic, lymphocytic, leukemia, B-CLL, between, miR-15a, miR-16-1, expression, expression, levels, of, arginyl-tRNA, synthetase, RARS, enzyme]\n",
      "B-CLL [(, )]\n",
      "RARS [(, )]\n",
      "enzyme [an, complex]\n",
      "achildren {miR-16-1}\n",
      "achildren False\n",
      "Conjuncts [shown, leukemia]\n",
      "neighbour shown in\n",
      "neighbour shown in\n",
      "Conjunction before fc leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "fc add children for leukemia\n",
      "Conjuncts [leukemia, expression]\n",
      "neighbour leukemia (\n",
      "neighbour leukemia B-CLL\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "Conjuncts [levels, synthetase]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc synthetase\n",
      "fc add children for synthetase\n",
      "fc add children for synthetase\n",
      "fc add children for synthetase\n",
      "fc add children for synthetase\n",
      "Conjuncts [associates, p43]\n",
      "neighbour associates with\n",
      "neighbour associates with\n",
      "Conjunction before fc p43\n",
      "fc add children for p43\n",
      "fc add children for p43\n",
      "Conjuncts [associates, complex]\n",
      "neighbour associates with\n",
      "neighbour associates with\n",
      "Conjunction before fc complex\n",
      "fc add children for complex\n",
      "fc add children for complex\n",
      "fc add children for complex\n",
      "Conunct merge check: leukemia B-CLL\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: B-CLL synthetase\n",
      ") between miR-15a and miR-16-1 expression and the expression\n",
      "Conunct merge check: synthetase RARS\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: RARS enzyme\n",
      "),\n",
      "Conunct merge check: enzyme p43\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( nmod )\n",
      "leukemia [shown, in, B, cell, chronic, lymphocytic, leukemia, B-CLL, between, miR-15a, expression]\n",
      "B-CLL [(, )]\n",
      "synthetase [levels, of, arginyl-tRNA, synthetase, RARS, enzyme]\n",
      "RARS [(, )]\n",
      "enzyme [an, associates, in, aminoacyl-tRNA, synthetase, complex]\n",
      "p43 [associates, with, cofactor, p43]\n",
      "SDP sent: An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\n",
      "SDP     : [(miR-16-1, 'NOUN', 'conj', False), (miR-15a, 'NOUN', 'compound', True), (expression, 'NOUN', 'nmod', False), (leukemia, 'NOUN', 'nmod', True), (levels, 'NOUN', 'conj', False), (synthetase, 'NOUN', 'nmod', False), (RARS, 'NOUN', 'appos', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-16-1]\n",
      "DeepCheck Conj Subtree\n",
      "[the, expression, levels, of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex]\n",
      "0 [An, inverse, correlation, has, been, shown, in, B, cell, chronic, lymphocytic, leukemia, (, B-CLL, ), between, miR-15a, and, miR-16-1, expression, and, the, expression, levels, of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex] 44 0 An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\n",
      "DeepCheck Conj Subtree\n",
      "[in, B, cell, chronic, lymphocytic, leukemia, (, B-CLL, ), between, miR-15a, and, miR-16-1, expression, and, the, expression, levels, of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-16-1]\n",
      "DeepCheck Conj Subtree\n",
      "[between, miR-15a, and, miR-16-1, expression]\n",
      "DeepCheck Conj Subtree\n",
      "[the, expression, levels, of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex]\n",
      "DeepCheck Conj Subtree\n",
      "[of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex]\n",
      "DeepCheck Conj Subtree\n",
      "[with, the, cofactor, p43]\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, aminoacyl-tRNA, synthetase, complex]\n",
      "0 [An, inverse, correlation, has, been, shown, in, B, cell, chronic, lymphocytic, leukemia, (, B-CLL, ), between, miR-15a, and, miR-16-1, expression, and, the, expression, levels, of, arginyl-tRNA, synthetase, (, RARS, ), ,, an, enzyme, which, associates, with, the, cofactor, p43, in, the, aminoacyl-tRNA, synthetase, complex] 44 0 An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\n",
      "Recognized overlapping entities: {Interval(179, 183, ('CANCER', 'RARS'))}\n",
      "Gene reject: {'CANCER'} due to {'CANCER'}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text An inverse correlation has been shown in B cell chronic lymphocytic leukemia ( B-CLL ) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase ( RARS ) , an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex\n",
      "(False, {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': False, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene neg corr', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and miR-16-1 expression and the expression levels of arginyl-tRNA synthetase (RARS), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\n",
      "full sentence An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and <e1>miR-16-1</e1> expression and the expression levels of arginyl-tRNA synthetase (<e2>RARS</e2>), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': False, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'compartment mir gene neg corr', 'reg_detect_major': 'compartment'}}, 'tagged_sent': 'An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and <e1>miR-16-1</e1> expression and the expression levels of arginyl-tRNA synthetase (<e2>RARS</e2>), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.', 'full_sentence': 'An inverse correlation has been shown in B cell chronic lymphocytic leukemia (B-CLL) between miR-15a and <e1>miR-16-1</e1> expression and the expression levels of arginyl-tRNA synthetase (<e2>RARS</e2>), an enzyme which associates with the cofactor p43 in the aminoacyl-tRNA synthetase complex.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (105, 113), 'entity_text': 'miR-16-1'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (179, 183), 'entity_text': 'RARS'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d96.s2.p1\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e1>miR-29b</e1> and miR-125a regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'full_sentence': '<e1>miR-29b</e1> and miR-125a regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (0, 7), 'entity_text': 'miR-29b'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (30, 40), 'entity_text': 'podoplanin'}}\n",
      "add special case miR-29b\n",
      "add special case podoplanin\n",
      "miR-29b 0 podoplanin 4\n",
      "(u\"miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\", 0, 4)\n",
      "geneword podoplanin mirword miR-29b\n",
      "Conjuncts [miR-125a, miR-29b]\n",
      "neighbour miR-125a regulate\n",
      "neighbour miR-125a regulate\n",
      "Conjunction before fc miR-29b\n",
      "Conjuncts [miR-29b, miR-125a]\n",
      "neighbour miR-29b and\n",
      "neighbour miR-29b and\n",
      "Conjunction before fc miR-125a\n",
      "Conjuncts [suppress, regulate]\n",
      "neighbour suppress invasion\n",
      "neighbour suppress invasion\n",
      "Conjunction before fc regulate\n",
      "fc add children for regulate\n",
      "Conjuncts [regulate, suppress]\n",
      "neighbour regulate podoplanin\n",
      "neighbour regulate podoplanin\n",
      "Conjunction before fc suppress\n",
      "fc add children for suppress\n",
      "Conunct merge check: miR-29b regulate\n",
      "miR-125a\n",
      "Conjunctions ( conj )\n",
      "miR-29b [miR-29b, miR-125a]\n",
      "regulate [regulate, podoplanin, suppress, invasion, in, glioblastoma]\n",
      "Conjuncts [invasion, glioblastoma]\n",
      "neighbour invasion in\n",
      "neighbour invasion in\n",
      "Conjunction before fc glioblastoma\n",
      "fc add children for glioblastoma\n",
      "Conjunctions ( nmod )\n",
      "glioblastoma [invasion, in, glioblastoma]\n",
      "SDP sent: miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\n",
      "SDP     : [(miR-29b, 'NOUN', 'nsubj', False), (regulate, 'VERB', 'ROOT', False), (podoplanin, 'NOUN', 'dobj', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-125a]\n",
      "deepcheck connection rule noun found suppress invasion [suppress, invasion, in, glioblastoma]\n",
      "deepcheck conj verb\n",
      "0 [miR-29b, and, miR-125a, regulate, podoplanin, and, suppress, invasion, in, glioblastoma] 10 0 miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\n",
      "DeepCheck Conj Subtree\n",
      "[miR-125a]\n",
      "deepcheck connection rule noun found suppress invasion [suppress, invasion, in, glioblastoma]\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[in, glioblastoma]\n",
      "0 [miR-29b, and, miR-125a, regulate, podoplanin, and, suppress, invasion, in, glioblastoma] 10 0 miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\n",
      "Recognized overlapping entities: {Interval(30, 40, ('GENE_OR_GENE_PRODUCT', 'podoplanin'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma\n",
      "[miR-29b, and, miR-125a, regulate, podoplanin, and, suppress, invasion, in, glioblastoma]\n",
      "TextBetween: miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma\n",
      "infer direction from token set [miR-29b, and, miR-125a, regulate, podoplanin, and, suppress, invasion, in, glioblastoma]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62ae8>, {'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 1, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70ae8>, {'gene': Counter({'NEU': 1, 'NEG': 1})})\n",
      "counts between 1 stemTypes2Count [(regulate, 'NEU'), (suppress, 'NEG')]\n",
      "Between Tokens [(regulate, 'NEU')]\n",
      "counts between final False\n",
      "['miR-29b', 'and', 'miR-125a', 'regulate', 'podoplanin', 'and', 'suppress']\n",
      "Counter({'NEU': 1, 'NEG': 1})\n",
      "[(regulate, 'NEU'), (suppress, 'NEG')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\n",
      "full sentence <e1>miR-29b</e1> and miR-125a regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e1>miR-29b</e1> and miR-125a regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'full_sentence': '<e1>miR-29b</e1> and miR-125a regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (0, 7), 'entity_text': 'miR-29b'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (30, 40), 'entity_text': 'podoplanin'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d96.s2.p3\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'miR-29b and <e1>miR-125a</e1> regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'full_sentence': 'miR-29b and <e1>miR-125a</e1> regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (12, 20), 'entity_text': 'miR-125a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (30, 40), 'entity_text': 'podoplanin'}}\n",
      "add special case miR-125a\n",
      "add special case podoplanin\n",
      "miR-125a 2 podoplanin 4\n",
      "(u\"miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\", 2, 4)\n",
      "geneword podoplanin mirword miR-125a\n",
      "Conjuncts [miR-125a, miR-29b]\n",
      "neighbour miR-125a regulate\n",
      "neighbour miR-125a regulate\n",
      "Conjunction before fc miR-29b\n",
      "Conjuncts [miR-29b, miR-125a]\n",
      "neighbour miR-29b and\n",
      "neighbour miR-29b and\n",
      "Conjunction before fc miR-125a\n",
      "Conjuncts [suppress, regulate]\n",
      "neighbour suppress invasion\n",
      "neighbour suppress invasion\n",
      "Conjunction before fc regulate\n",
      "fc add children for regulate\n",
      "Conjuncts [regulate, suppress]\n",
      "neighbour regulate podoplanin\n",
      "neighbour regulate podoplanin\n",
      "Conjunction before fc suppress\n",
      "fc add children for suppress\n",
      "Conunct merge check: miR-29b regulate\n",
      "miR-125a\n",
      "Conjunctions ( conj )\n",
      "miR-29b [miR-29b, miR-125a]\n",
      "regulate [regulate, podoplanin, suppress, invasion, in, glioblastoma]\n",
      "Conjuncts [invasion, glioblastoma]\n",
      "neighbour invasion in\n",
      "neighbour invasion in\n",
      "Conjunction before fc glioblastoma\n",
      "fc add children for glioblastoma\n",
      "Conjunctions ( nmod )\n",
      "glioblastoma [invasion, in, glioblastoma]\n",
      "SDP sent: miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\n",
      "SDP     : [(miR-125a, 'NOUN', 'conj', False), (miR-29b, 'NOUN', 'nsubj', True), (regulate, 'VERB', 'ROOT', False), (podoplanin, 'NOUN', 'dobj', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-125a]\n",
      "deepcheck connection rule noun found suppress invasion [suppress, invasion, in, glioblastoma]\n",
      "deepcheck conj verb\n",
      "0 [miR-29b, and, miR-125a, regulate, podoplanin, and, suppress, invasion, in, glioblastoma] 10 0 miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\n",
      "DeepCheck Conj Subtree\n",
      "[miR-125a]\n",
      "deepcheck connection rule noun found suppress invasion [suppress, invasion, in, glioblastoma]\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[in, glioblastoma]\n",
      "0 [miR-29b, and, miR-125a, regulate, podoplanin, and, suppress, invasion, in, glioblastoma] 10 0 miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\n",
      "Recognized overlapping entities: {Interval(30, 40, ('GENE_OR_GENE_PRODUCT', 'podoplanin'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma\n",
      "[miR-29b, and, miR-125a, regulate, podoplanin, and, suppress, invasion, in, glioblastoma]\n",
      "TextBetween: miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma\n",
      "infer direction from token set [miR-29b, and, miR-125a, regulate, podoplanin, and, suppress, invasion, in, glioblastoma]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954e18>, {'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 1, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e2400>, {'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count [(regulate, 'NEU'), (suppress, 'NEG')]\n",
      "Between Tokens [(regulate, 'NEU')]\n",
      "counts between final False\n",
      "['miR-29b', 'and', 'miR-125a', 'regulate', 'podoplanin', 'and', 'suppress']\n",
      "Counter({'NEU': 1, 'NEG': 1})\n",
      "[(regulate, 'NEU'), (suppress, 'NEG')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\n",
      "full sentence miR-29b and <e1>miR-125a</e1> regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'miR-29b and <e1>miR-125a</e1> regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'full_sentence': 'miR-29b and <e1>miR-125a</e1> regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (12, 20), 'entity_text': 'miR-125a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (30, 40), 'entity_text': 'podoplanin'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d98.s7.p0\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case microRNA-433\n",
      "add special case FGF20\n",
      "microRNA-433 17 FGF20 22\n",
      "(u\"We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\", 17, 22)\n",
      "geneword FGF20 mirword microRNA-433\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [show, assays]\n",
      "neighbour show in\n",
      "neighbour show in\n",
      "Conjunction before fc assays\n",
      "fc add children for assays\n",
      "fc add children for assays\n",
      "fc add children for assays\n",
      "Conjuncts [allele, rs12720208]\n",
      "neighbour allele for\n",
      "neighbour allele for\n",
      "Conjunction before fc rs12720208\n",
      "fc add children for rs12720208\n",
      "Conjuncts [site, microRNA-433]\n",
      "neighbour site for\n",
      "neighbour site for\n",
      "Conjunction before fc microRNA-433\n",
      "fc add children for microRNA-433\n",
      "Conjuncts [disrupts, translation]\n",
      "neighbour disrupts a\n",
      "neighbour disrupts a\n",
      "Conjunction before fc translation\n",
      "fc add children for translation\n",
      "fc add children for translation\n",
      "Conjuncts [translation, FGF20]\n",
      "neighbour translation of\n",
      "neighbour translation of\n",
      "Conjunction before fc FGF20\n",
      "fc add children for FGF20\n",
      "Conunct merge check: assays rs12720208\n",
      "assays that the risk\n",
      "Conunct merge check: rs12720208 translation\n",
      "rs12720208\n",
      "Conunct merge check: translation microRNA-433\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( nmod )\n",
      "assays [show, in, several, functional, assays]\n",
      "rs12720208 [allele, for, rs12720208]\n",
      "microRNA-433 [site, for, microRNA-433]\n",
      "translation [disrupts, increasing, translation, of, FGF20]\n",
      "SDP sent: We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "SDP     : [(microRNA-433, 'NOUN', 'nmod', False), (site, 'NOUN', 'dobj', False), (disrupts, 'VERB', 'ccomp', False), (translation, 'NOUN', 'nmod', False), (FGF20, 'NOUN', 'nmod', False)]\n",
      "SDP DOBJ RULE\n",
      "0 [We, show, in, several, functional, assays] 6 0 We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "1 [the, risk, allele, for, rs12720208, disrupts, a, binding, site, for, microRNA-433, ,, increasing, translation, of, FGF20, in, vitro, and, in, vivo] 21 42 the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "DeepCheck Conj Subtree\n",
      "[in, several, functional, assays]\n",
      "DeepCheck Conj Subtree\n",
      "[for, rs12720208]\n",
      "DeepCheck Conj Subtree\n",
      "[for, microRNA-433]\n",
      "DeepCheck Conj Subtree\n",
      "[increasing, translation, of, FGF20, in, vitro, and, in, vivo]\n",
      "deepcheck conj verb\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[of, FGF20]\n",
      "0 [We, show, in, several, functional, assays] 6 0 We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "1 [the, risk, allele, for, rs12720208, disrupts, a, binding, site, for, microRNA-433, ,, increasing, translation, of, FGF20, in, vitro, and, in, vivo] 21 42 the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "Recognized overlapping entities: {Interval(141, 146, ('GENE_OR_GENE_PRODUCT', 'FGF20'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text the risk allele for rs12720208 disrupts a binding site for microRNA-433 , increasing translation of FGF20 in vitro and in vivo\n",
      "[the, risk, allele, for, rs12720208, disrupts, a, binding, site, for, microRNA-433, ,, increasing, translation, of, FGF20, in, vitro, and, in, vivo]\n",
      "TextBetween: the risk allele for rs12720208 disrupts a binding site for microRNA-433 , increasing translation of FGF20 in vitro and in vivo\n",
      "infer direction from token set [the, risk, allele, for, rs12720208, disrupts, a, binding, site, for, microRNA-433, ,, increasing, translation, of, FGF20, in, vitro, and, in, vivo]\n",
      "stemTypes2Count: Counter({'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf90954ae8>, {'mir': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'POS': 1, 'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70ae8>, {'mir': Counter({'POS': 1}), 'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(increasing, 'POS'), (translation, 'NEU')]\n",
      "Between Tokens [(increasing, 'POS'), (translation, 'NEU')]\n",
      "counts between final False\n",
      "['site', 'for', 'microRNA-433', ',', 'increasing', 'translation', 'of', 'FGF20', 'in', 'vitro']\n",
      "Counter({'POS': 1, 'NEU': 1})\n",
      "[(increasing, 'POS'), (translation, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "full sentence We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for <e1>microRNA-433</e1>, increasing translation of <e2>FGF20</e2> in vitro and in vivo.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for <e1>microRNA-433</e1>, increasing translation of <e2>FGF20</e2> in vitro and in vivo.', 'full_sentence': 'We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for <e1>microRNA-433</e1>, increasing translation of <e2>FGF20</e2> in vitro and in vivo.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (101, 113), 'entity_text': 'microRNA-433'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (141, 146), 'entity_text': 'FGF20'}}\n",
      "Total:      232\n",
      "Correct:    213 0.9181034482758621\n",
      "Incorrect:  19 0.08189655172413793\n",
      "classes Counter()\n",
      "precision 0.9136690647482014\n",
      "recall 0.9477611940298507\n",
      "specificity 0.8775510204081632\n",
      "f1 0.9304029304029303\n",
      "relationFalseCounter Counter({'conj': 8, 'conj_nmod': 3, 'context': 3, 'compartment': 2, 'compartment_nmod': 2, 'entity': 2})\n",
      "Correct classified:      127 0.5474137931034483 0.9136690647482014\n",
      "Incorrect classified:    12 0.05172413793103448 0.08633093525179857\n",
      "Random classified:  54 0.23275862068965517 0.38848920863309355\n",
      "Counter({'counts between': 10, 'between mir gene': 1, 'counts between alternating': 1})\n",
      "Correct interaction&classified:      117 0.5043103448275862 0.841726618705036\n",
      "Classified by:  Counter({'counts between': 64, 'counts opp': 10, 'counts between equal': 10, 'between gene mir': 10, 'compartment mir gene by': 8, 'return': 7, 'between mir gene': 6, 'compartment mir gene': 6, 'compartment gene mir': 5, 'compartment gene mir by': 4, 'compartment mir gene neg corr': 4, 'between mir gene reg corr': 3, 'counts between alternating': 2})\n",
      "[[[133   2]\n",
      "  [  0   4]]\n",
      "\n",
      " [[138   0]\n",
      "  [  0   1]]\n",
      "\n",
      " [[ 80   5]\n",
      "  [  5  49]]\n",
      "\n",
      " [[ 57   3]\n",
      "  [  6  73]]\n",
      "\n",
      " [[136   2]\n",
      "  [  1   0]]]\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      " GENE_MIR_NEU       0.67      1.00      0.80         4\n",
      "  GENE_MIR_UP       1.00      1.00      1.00         1\n",
      "MIR_GENE_DOWN       0.91      0.91      0.91        54\n",
      " MIR_GENE_NEU       0.96      0.92      0.94        79\n",
      "  MIR_GENE_UP       0.00      0.00      0.00         1\n",
      "\n",
      "     accuracy                           0.91       139\n",
      "    macro avg       0.71      0.77      0.73       139\n",
      " weighted avg       0.92      0.91      0.92       139\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%aimport MirGeneRelCheck\n",
    "\n",
    "relClassifier = SentenceRelationClassifier(relationsFile)\n",
    "\n",
    "test_results = runCheck(-1, \"TEST\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pickle\n",
    "\n",
    "outDirectory = mx_base + \"/scai_eval/\"\n",
    "\n",
    "with open(os.path.join(outDirectory, \"scai_test_f1.pickle\"), \"wb\") as fout:\n",
    "    pickle.dump(test_results, fout)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading relations scai_test_rels.tsv\n",
      "Interactions 232\n",
      "Non NA Interactions 141\n",
      "Will test 1 subsets.\n",
      "Testing checks ('compartment', 'between', 'counts', 'return')\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Setting relfile /mnt/extproj/projekte/textmining/miRExplore/python/allrels.csv\n",
      "Setting relfile /mnt/extproj/projekte/textmining/miRExplore/python/allrels.csv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "miRNA-corp.d1.s1.p0\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miRNA-128\n",
      "add special case ARP5\n",
      "miRNA-128 3 ARP5 18\n",
      "(u\"Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\", 3, 18)\n",
      "geneword ARP5 mirword miRNA-128\n",
      "Conjuncts [lines, CRL-1690]\n",
      "neighbour lines (\n",
      "neighbour lines a\n",
      "Conjunction before fc CRL-1690\n",
      "fc add children for CRL-1690\n",
      "Conjuncts [CRL-1690, lines]\n",
      "neighbour CRL-1690 and\n",
      "neighbour CRL-1690 and\n",
      "Conjunction before fc lines\n",
      "fc add children for lines\n",
      "fc add children for lines\n",
      "fc add children for lines\n",
      "Conjuncts [decreased, restored]\n",
      "neighbour decreased the\n",
      "neighbour decreased the\n",
      "Conjunction before fc restored\n",
      "fc add children for restored\n",
      "fc add children for restored\n",
      "Conjuncts [Bmi-1, E2F-3a, ARP5]\n",
      "neighbour Bmi-1 and\n",
      "neighbour Bmi-1 and\n",
      "neighbour E2F-3a expression\n",
      "neighbour E2F-3a expression\n",
      "Conjunction before fc ARP5\n",
      "fc add children for ARP5\n",
      "fc add children for ARP5\n",
      "Conjuncts [ARP5, E2F-3a, Bmi-1]\n",
      "neighbour ARP5 (\n",
      "neighbour ARP5 ANGPTL6\n",
      "neighbour E2F-3a expression\n",
      "neighbour E2F-3a expression\n",
      "Conjunction before fc Bmi-1\n",
      "Conjuncts [ARP5, Bmi-1, E2F-3a]\n",
      "neighbour ARP5 (\n",
      "neighbour ARP5 ANGPTL6\n",
      "neighbour Bmi-1 and\n",
      "neighbour Bmi-1 and\n",
      "Conjunction before fc E2F-3a\n",
      "Conjuncts [restored, decreased]\n",
      "neighbour restored '\n",
      "neighbour restored homeostatic\n",
      "Conjunction before fc decreased\n",
      "fc add children for decreased\n",
      "fc add children for decreased\n",
      "Conjuncts [lines, CRL-1690]\n",
      "neighbour lines .\n",
      "neighbour lines None\n",
      "Conjunction before fc CRL-1690\n",
      "fc add children for CRL-1690\n",
      "Conjuncts [CRL-1690, lines]\n",
      "neighbour CRL-1690 and\n",
      "neighbour CRL-1690 and\n",
      "Conjunction before fc lines\n",
      "fc add children for lines\n",
      "fc add children for lines\n",
      "Conunct merge check: CRL-1690 restored\n",
      "lines (\n",
      "Conunct merge check: restored ANGPTL6\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( conj )\n",
      "CRL-1690 [to, CRL-1690, CRL-2610, GBM, cell, lines]\n",
      "restored [a, restored, homeostatic, ARP5, ANGPTL6, Bmi-1, E2F-3a, expression, b, decreased, proliferation, of, CRL-1690, CRL-2610, cell, lines]\n",
      "ANGPTL6 [(, )]\n",
      "Conjuncts [Addition, miRNA-128]\n",
      "neighbour Addition of\n",
      "neighbour Addition of\n",
      "Conjunction before fc miRNA-128\n",
      "fc add children for miRNA-128\n",
      "fc add children for miRNA-128\n",
      "Conjuncts [Addition, CRL-1690]\n",
      "neighbour Addition of\n",
      "neighbour Addition of\n",
      "Conjunction before fc CRL-1690\n",
      "fc add children for CRL-1690\n",
      "Conjuncts [proliferation, CRL-1690]\n",
      "neighbour proliferation of\n",
      "neighbour proliferation of\n",
      "Conjunction before fc CRL-1690\n",
      "fc add children for CRL-1690\n",
      "Conunct merge check: miRNA-128 ANGPTL6\n",
      "CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5\n",
      "Conunct merge check: ANGPTL6 CRL-1690\n",
      "), Bmi-1 and E2F-3a expression, and (b) significantly decreased the\n",
      "Conjunctions ( nmod )\n",
      "miRNA-128 [Addition, of, exogenous, miRNA-128, to, CRL-1690]\n",
      "ANGPTL6 [(, )]\n",
      "CRL-1690 [proliferation, of, CRL-1690]\n",
      "SDP sent: Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\n",
      "SDP     : [(miRNA-128, 'NOUN', 'nmod', False), (Addition, 'NOUN', 'nsubj', False), (restored, 'VERB', 'ROOT', False), (expression, 'NOUN', 'dobj', False), (ARP5, 'NOUN', 'compound', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[CRL-2610, GBM, cell, lines]\n",
      "DeepCheck Conj Subtree\n",
      "[Bmi-1]\n",
      "DeepCheck Conj Subtree\n",
      "[E2F-3a]\n",
      "DeepCheck Conj Subtree\n",
      "[CRL-2610, cell, lines]\n",
      "0 [Addition, of, exogenous, miRNA-128, to, CRL-1690, and, CRL-2610, GBM, cell, lines, (, a, ), restored, ', homeostatic, ', ARP5, (, ANGPTL6, ), ,, Bmi-1, and, E2F-3a, expression] 27 0 Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\n",
      "1 [b, ), significantly, decreased, the, proliferation, of, CRL-1690, and, CRL-2610, cell, lines] 12 149 b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\n",
      "DeepCheck Conj Subtree\n",
      "[of, exogenous, miRNA-128]\n",
      "DeepCheck Conj Subtree\n",
      "[to, CRL-1690, and, CRL-2610, GBM, cell, lines]\n",
      "DeepCheck Conj Subtree\n",
      "[CRL-2610, GBM, cell, lines]\n",
      "DeepCheck Conj Subtree\n",
      "[Bmi-1]\n",
      "DeepCheck Conj Subtree\n",
      "[E2F-3a]\n",
      "DeepCheck Conj Subtree\n",
      "[of, CRL-1690, and, CRL-2610, cell, lines]\n",
      "DeepCheck Conj Subtree\n",
      "[CRL-2610, cell, lines]\n",
      "0 [Addition, of, exogenous, miRNA-128, to, CRL-1690, and, CRL-2610, GBM, cell, lines, (, a, ), restored, ', homeostatic, ', ARP5, (, ANGPTL6, ), ,, Bmi-1, and, E2F-3a, expression] 27 0 Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\n",
      "1 [b, ), significantly, decreased, the, proliferation, of, CRL-1690, and, CRL-2610, cell, lines] 12 149 b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\n",
      "Recognized overlapping entities: {Interval(99, 103, ('GENE_OR_GENE_PRODUCT', 'ARP5'))}\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "Compartment Text Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines ( a ) restored ' homeostatic ' ARP5 ( ANGPTL6 ) , Bmi-1 and E2F-3a expression\n",
      "[Addition, of, exogenous, miRNA-128, to, CRL-1690, and, CRL-2610, GBM, cell, lines, (, a, ), restored, ', homeostatic, ', ARP5, (, ANGPTL6, ), ,, Bmi-1, and, E2F-3a, expression]\n",
      "TextBetween: Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines ( a ) restored ' homeostatic ' ARP5 ( ANGPTL6 ) , Bmi-1 and E2F-3a expression\n",
      "infer direction from token set [Addition, of, exogenous, miRNA-128, to, CRL-1690, and, CRL-2610, GBM, cell, lines, (, a, ), restored, ', homeostatic, ', ARP5, (, ANGPTL6, ), ,, Bmi-1, and, E2F-3a, expression]\n",
      "stemTypes2Count: Counter({'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9a5e28c8>, {})\n",
      "counts between 1 stemTypes2Count Counter()\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70730>, {})\n",
      "counts between 1 stemTypes2Count []\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}})\n",
      " hit sentence Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\n",
      "full sentence Addition of exogenous <e1>miRNA-128</e1> to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' <e2>ARP5</e2> (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}}, 'tagged_sent': \"Addition of exogenous <e1>miRNA-128</e1> to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' <e2>ARP5</e2> (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\", 'full_sentence': \"Addition of exogenous <e1>miRNA-128</e1> to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' <e2>ARP5</e2> (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (22, 31), 'entity_text': 'miRNA-128'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (99, 103), 'entity_text': 'ARP5'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d1.s1.p1\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miRNA-128\n",
      "add special case ANGPTL6\n",
      "miRNA-128 3 ANGPTL6 20\n",
      "(u\"Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\", 3, 20)\n",
      "geneword ANGPTL6 mirword miRNA-128\n",
      "Conjuncts [lines, CRL-1690]\n",
      "neighbour lines (\n",
      "neighbour lines a\n",
      "Conjunction before fc CRL-1690\n",
      "fc add children for CRL-1690\n",
      "Conjuncts [CRL-1690, lines]\n",
      "neighbour CRL-1690 and\n",
      "neighbour CRL-1690 and\n",
      "Conjunction before fc lines\n",
      "fc add children for lines\n",
      "fc add children for lines\n",
      "fc add children for lines\n",
      "Conjuncts [decreased, restored]\n",
      "neighbour decreased the\n",
      "neighbour decreased the\n",
      "Conjunction before fc restored\n",
      "fc add children for restored\n",
      "fc add children for restored\n",
      "Conjuncts [Bmi-1, E2F-3a, ARP5]\n",
      "neighbour Bmi-1 and\n",
      "neighbour Bmi-1 and\n",
      "neighbour E2F-3a expression\n",
      "neighbour E2F-3a expression\n",
      "Conjunction before fc ARP5\n",
      "fc add children for ARP5\n",
      "fc add children for ARP5\n",
      "Conjuncts [ARP5, E2F-3a, Bmi-1]\n",
      "neighbour ARP5 (\n",
      "neighbour ARP5 ANGPTL6\n",
      "neighbour E2F-3a expression\n",
      "neighbour E2F-3a expression\n",
      "Conjunction before fc Bmi-1\n",
      "Conjuncts [ARP5, Bmi-1, E2F-3a]\n",
      "neighbour ARP5 (\n",
      "neighbour ARP5 ANGPTL6\n",
      "neighbour Bmi-1 and\n",
      "neighbour Bmi-1 and\n",
      "Conjunction before fc E2F-3a\n",
      "Conjuncts [restored, decreased]\n",
      "neighbour restored '\n",
      "neighbour restored homeostatic\n",
      "Conjunction before fc decreased\n",
      "fc add children for decreased\n",
      "fc add children for decreased\n",
      "Conjuncts [lines, CRL-1690]\n",
      "neighbour lines .\n",
      "neighbour lines None\n",
      "Conjunction before fc CRL-1690\n",
      "fc add children for CRL-1690\n",
      "Conjuncts [CRL-1690, lines]\n",
      "neighbour CRL-1690 and\n",
      "neighbour CRL-1690 and\n",
      "Conjunction before fc lines\n",
      "fc add children for lines\n",
      "fc add children for lines\n",
      "Conunct merge check: CRL-1690 restored\n",
      "lines (\n",
      "Conunct merge check: restored ANGPTL6\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( conj )\n",
      "CRL-1690 [to, CRL-1690, CRL-2610, GBM, cell, lines]\n",
      "restored [a, restored, homeostatic, ARP5, ANGPTL6, Bmi-1, E2F-3a, expression, b, decreased, proliferation, of, CRL-1690, CRL-2610, cell, lines]\n",
      "ANGPTL6 [(, )]\n",
      "Conjuncts [Addition, miRNA-128]\n",
      "neighbour Addition of\n",
      "neighbour Addition of\n",
      "Conjunction before fc miRNA-128\n",
      "fc add children for miRNA-128\n",
      "fc add children for miRNA-128\n",
      "Conjuncts [Addition, CRL-1690]\n",
      "neighbour Addition of\n",
      "neighbour Addition of\n",
      "Conjunction before fc CRL-1690\n",
      "fc add children for CRL-1690\n",
      "Conjuncts [proliferation, CRL-1690]\n",
      "neighbour proliferation of\n",
      "neighbour proliferation of\n",
      "Conjunction before fc CRL-1690\n",
      "fc add children for CRL-1690\n",
      "Conunct merge check: miRNA-128 ANGPTL6\n",
      "CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5\n",
      "Conunct merge check: ANGPTL6 CRL-1690\n",
      "), Bmi-1 and E2F-3a expression, and (b) significantly decreased the\n",
      "Conjunctions ( nmod )\n",
      "miRNA-128 [Addition, of, exogenous, miRNA-128, to, CRL-1690]\n",
      "ANGPTL6 [(, )]\n",
      "CRL-1690 [proliferation, of, CRL-1690]\n",
      "SDP sent: Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\n",
      "SDP     : [(miRNA-128, 'NOUN', 'nmod', False), (Addition, 'NOUN', 'nsubj', False), (restored, 'VERB', 'ROOT', False), (expression, 'NOUN', 'dobj', False), (ARP5, 'NOUN', 'compound', False), (ANGPTL6, 'NOUN', 'appos', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[CRL-2610, GBM, cell, lines]\n",
      "DeepCheck Conj Subtree\n",
      "[Bmi-1]\n",
      "DeepCheck Conj Subtree\n",
      "[E2F-3a]\n",
      "DeepCheck Conj Subtree\n",
      "[CRL-2610, cell, lines]\n",
      "0 [Addition, of, exogenous, miRNA-128, to, CRL-1690, and, CRL-2610, GBM, cell, lines, (, a, ), restored, ', homeostatic, ', ARP5, (, ANGPTL6, ), ,, Bmi-1, and, E2F-3a, expression] 27 0 Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\n",
      "1 [b, ), significantly, decreased, the, proliferation, of, CRL-1690, and, CRL-2610, cell, lines] 12 149 b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\n",
      "DeepCheck Conj Subtree\n",
      "[of, exogenous, miRNA-128]\n",
      "DeepCheck Conj Subtree\n",
      "[to, CRL-1690, and, CRL-2610, GBM, cell, lines]\n",
      "DeepCheck Conj Subtree\n",
      "[CRL-2610, GBM, cell, lines]\n",
      "DeepCheck Conj Subtree\n",
      "[Bmi-1]\n",
      "DeepCheck Conj Subtree\n",
      "[E2F-3a]\n",
      "DeepCheck Conj Subtree\n",
      "[of, CRL-1690, and, CRL-2610, cell, lines]\n",
      "DeepCheck Conj Subtree\n",
      "[CRL-2610, cell, lines]\n",
      "0 [Addition, of, exogenous, miRNA-128, to, CRL-1690, and, CRL-2610, GBM, cell, lines, (, a, ), restored, ', homeostatic, ', ARP5, (, ANGPTL6, ), ,, Bmi-1, and, E2F-3a, expression] 27 0 Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\n",
      "1 [b, ), significantly, decreased, the, proliferation, of, CRL-1690, and, CRL-2610, cell, lines] 12 149 b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\n",
      "Recognized overlapping entities: {Interval(105, 112, ('GENE_OR_GENE_PRODUCT', 'ANGPTL6'))}\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "Compartment Text Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines ( a ) restored ' homeostatic ' ARP5 ( ANGPTL6 ) , Bmi-1 and E2F-3a expression\n",
      "[Addition, of, exogenous, miRNA-128, to, CRL-1690, and, CRL-2610, GBM, cell, lines, (, a, ), restored, ', homeostatic, ', ARP5, (, ANGPTL6, ), ,, Bmi-1, and, E2F-3a, expression]\n",
      "TextBetween: Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines ( a ) restored ' homeostatic ' ARP5 ( ANGPTL6 ) , Bmi-1 and E2F-3a expression\n",
      "infer direction from token set [Addition, of, exogenous, miRNA-128, to, CRL-1690, and, CRL-2610, GBM, cell, lines, (, a, ), restored, ', homeostatic, ', ARP5, (, ANGPTL6, ), ,, Bmi-1, and, E2F-3a, expression]\n",
      "stemTypes2Count: Counter({'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62d90>, {})\n",
      "counts between 1 stemTypes2Count Counter()\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62ea0>, {})\n",
      "counts between 1 stemTypes2Count []\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}})\n",
      " hit sentence Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\n",
      "full sentence Addition of exogenous <e1>miRNA-128</e1> to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (<e2>ANGPTL6</e2>), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}}, 'tagged_sent': \"Addition of exogenous <e1>miRNA-128</e1> to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (<e2>ANGPTL6</e2>), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\", 'full_sentence': \"Addition of exogenous <e1>miRNA-128</e1> to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (<e2>ANGPTL6</e2>), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (22, 31), 'entity_text': 'miRNA-128'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (105, 112), 'entity_text': 'ANGPTL6'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d1.s1.p2\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miRNA-128\n",
      "add special case Bmi-1\n",
      "miRNA-128 3 Bmi-1 23\n",
      "(u\"Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\", 3, 23)\n",
      "geneword Bmi-1 mirword miRNA-128\n",
      "Conjuncts [lines, CRL-1690]\n",
      "neighbour lines (\n",
      "neighbour lines a\n",
      "Conjunction before fc CRL-1690\n",
      "fc add children for CRL-1690\n",
      "Conjuncts [CRL-1690, lines]\n",
      "neighbour CRL-1690 and\n",
      "neighbour CRL-1690 and\n",
      "Conjunction before fc lines\n",
      "fc add children for lines\n",
      "fc add children for lines\n",
      "fc add children for lines\n",
      "Conjuncts [decreased, restored]\n",
      "neighbour decreased the\n",
      "neighbour decreased the\n",
      "Conjunction before fc restored\n",
      "fc add children for restored\n",
      "fc add children for restored\n",
      "Conjuncts [Bmi-1, E2F-3a, ARP5]\n",
      "neighbour Bmi-1 and\n",
      "neighbour Bmi-1 and\n",
      "neighbour E2F-3a expression\n",
      "neighbour E2F-3a expression\n",
      "Conjunction before fc ARP5\n",
      "fc add children for ARP5\n",
      "fc add children for ARP5\n",
      "Conjuncts [ARP5, E2F-3a, Bmi-1]\n",
      "neighbour ARP5 (\n",
      "neighbour ARP5 ANGPTL6\n",
      "neighbour E2F-3a expression\n",
      "neighbour E2F-3a expression\n",
      "Conjunction before fc Bmi-1\n",
      "Conjuncts [ARP5, Bmi-1, E2F-3a]\n",
      "neighbour ARP5 (\n",
      "neighbour ARP5 ANGPTL6\n",
      "neighbour Bmi-1 and\n",
      "neighbour Bmi-1 and\n",
      "Conjunction before fc E2F-3a\n",
      "Conjuncts [restored, decreased]\n",
      "neighbour restored '\n",
      "neighbour restored homeostatic\n",
      "Conjunction before fc decreased\n",
      "fc add children for decreased\n",
      "fc add children for decreased\n",
      "Conjuncts [lines, CRL-1690]\n",
      "neighbour lines .\n",
      "neighbour lines None\n",
      "Conjunction before fc CRL-1690\n",
      "fc add children for CRL-1690\n",
      "Conjuncts [CRL-1690, lines]\n",
      "neighbour CRL-1690 and\n",
      "neighbour CRL-1690 and\n",
      "Conjunction before fc lines\n",
      "fc add children for lines\n",
      "fc add children for lines\n",
      "Conunct merge check: CRL-1690 restored\n",
      "lines (\n",
      "Conunct merge check: restored ANGPTL6\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( conj )\n",
      "CRL-1690 [to, CRL-1690, CRL-2610, GBM, cell, lines]\n",
      "restored [a, restored, homeostatic, ARP5, ANGPTL6, Bmi-1, E2F-3a, expression, b, decreased, proliferation, of, CRL-1690, CRL-2610, cell, lines]\n",
      "ANGPTL6 [(, )]\n",
      "Conjuncts [Addition, miRNA-128]\n",
      "neighbour Addition of\n",
      "neighbour Addition of\n",
      "Conjunction before fc miRNA-128\n",
      "fc add children for miRNA-128\n",
      "fc add children for miRNA-128\n",
      "Conjuncts [Addition, CRL-1690]\n",
      "neighbour Addition of\n",
      "neighbour Addition of\n",
      "Conjunction before fc CRL-1690\n",
      "fc add children for CRL-1690\n",
      "Conjuncts [proliferation, CRL-1690]\n",
      "neighbour proliferation of\n",
      "neighbour proliferation of\n",
      "Conjunction before fc CRL-1690\n",
      "fc add children for CRL-1690\n",
      "Conunct merge check: miRNA-128 ANGPTL6\n",
      "CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5\n",
      "Conunct merge check: ANGPTL6 CRL-1690\n",
      "), Bmi-1 and E2F-3a expression, and (b) significantly decreased the\n",
      "Conjunctions ( nmod )\n",
      "miRNA-128 [Addition, of, exogenous, miRNA-128, to, CRL-1690]\n",
      "ANGPTL6 [(, )]\n",
      "CRL-1690 [proliferation, of, CRL-1690]\n",
      "SDP sent: Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\n",
      "SDP     : [(miRNA-128, 'NOUN', 'nmod', False), (Addition, 'NOUN', 'nsubj', False), (restored, 'VERB', 'ROOT', False), (expression, 'NOUN', 'dobj', False), (ARP5, 'NOUN', 'compound', True), (Bmi-1, 'NOUN', 'conj', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[CRL-2610, GBM, cell, lines]\n",
      "DeepCheck Conj Subtree\n",
      "[Bmi-1]\n",
      "DeepCheck Conj Subtree\n",
      "[E2F-3a]\n",
      "DeepCheck Conj Subtree\n",
      "[CRL-2610, cell, lines]\n",
      "0 [Addition, of, exogenous, miRNA-128, to, CRL-1690, and, CRL-2610, GBM, cell, lines, (, a, ), restored, ', homeostatic, ', ARP5, (, ANGPTL6, ), ,, Bmi-1, and, E2F-3a, expression] 27 0 Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\n",
      "1 [b, ), significantly, decreased, the, proliferation, of, CRL-1690, and, CRL-2610, cell, lines] 12 149 b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\n",
      "DeepCheck Conj Subtree\n",
      "[of, exogenous, miRNA-128]\n",
      "DeepCheck Conj Subtree\n",
      "[to, CRL-1690, and, CRL-2610, GBM, cell, lines]\n",
      "DeepCheck Conj Subtree\n",
      "[CRL-2610, GBM, cell, lines]\n",
      "DeepCheck Conj Subtree\n",
      "[Bmi-1]\n",
      "DeepCheck Conj Subtree\n",
      "[E2F-3a]\n",
      "DeepCheck Conj Subtree\n",
      "[of, CRL-1690, and, CRL-2610, cell, lines]\n",
      "DeepCheck Conj Subtree\n",
      "[CRL-2610, cell, lines]\n",
      "0 [Addition, of, exogenous, miRNA-128, to, CRL-1690, and, CRL-2610, GBM, cell, lines, (, a, ), restored, ', homeostatic, ', ARP5, (, ANGPTL6, ), ,, Bmi-1, and, E2F-3a, expression] 27 0 Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\n",
      "1 [b, ), significantly, decreased, the, proliferation, of, CRL-1690, and, CRL-2610, cell, lines] 12 149 b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\n",
      "Recognized overlapping entities: {Interval(115, 120, ('GENE_OR_GENE_PRODUCT', 'Bmi-1'))}\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "Compartment Text Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines ( a ) restored ' homeostatic ' ARP5 ( ANGPTL6 ) , Bmi-1 and E2F-3a expression\n",
      "[Addition, of, exogenous, miRNA-128, to, CRL-1690, and, CRL-2610, GBM, cell, lines, (, a, ), restored, ', homeostatic, ', ARP5, (, ANGPTL6, ), ,, Bmi-1, and, E2F-3a, expression]\n",
      "TextBetween: Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines ( a ) restored ' homeostatic ' ARP5 ( ANGPTL6 ) , Bmi-1 and E2F-3a expression\n",
      "infer direction from token set [Addition, of, exogenous, miRNA-128, to, CRL-1690, and, CRL-2610, GBM, cell, lines, (, a, ), restored, ', homeostatic, ', ARP5, (, ANGPTL6, ), ,, Bmi-1, and, E2F-3a, expression]\n",
      "stemTypes2Count: Counter({'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed706a8>, {})\n",
      "counts between 1 stemTypes2Count Counter()\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c9d8>, {})\n",
      "counts between 1 stemTypes2Count []\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}})\n",
      " hit sentence Addition of exogenous miRNA-128 to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), Bmi-1 and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\n",
      "full sentence Addition of exogenous <e1>miRNA-128</e1> to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), <e2>Bmi-1</e2> and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}}, 'tagged_sent': \"Addition of exogenous <e1>miRNA-128</e1> to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), <e2>Bmi-1</e2> and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\", 'full_sentence': \"Addition of exogenous <e1>miRNA-128</e1> to CRL-1690 and CRL-2610 GBM cell lines (a) restored 'homeostatic' ARP5 (ANGPTL6), <e2>Bmi-1</e2> and E2F-3a expression, and (b) significantly decreased the proliferation of CRL-1690 and CRL-2610 cell lines.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (22, 31), 'entity_text': 'miRNA-128'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (115, 120), 'entity_text': 'Bmi-1'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d6.s1.p0\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case miR-21\n",
      "add special case caspase 9\n",
      "miR-21 4 9 7\n",
      "(u\"Moreover, reduction of miR-21 activated caspase 9 and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.\", 4, 7)\n",
      "geneword 9 mirword miR-21\n",
      "Conjuncts [3, caspase]\n",
      "neighbour 3 ,\n",
      "neighbour 3 which\n",
      "Conjunction before fc caspase\n",
      "fc add children for caspase\n",
      "fc add children for caspase\n",
      "fc add children for caspase\n",
      "fc add children for caspase\n",
      "Conjuncts [caspase, 3]\n",
      "neighbour caspase 9\n",
      "neighbour caspase 9\n",
      "Conjunction before fc 3\n",
      "Conunct merge check: reduction caspase\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( conj )\n",
      "reduction [reduction, .]\n",
      "caspase [of, miR-21, activated, caspase, 9, 3]\n",
      "achildren {miR-21}\n",
      "achildren False\n",
      "Conjuncts [reduction, caspase]\n",
      "neighbour reduction of\n",
      "neighbour reduction of\n",
      "Conjunction before fc caspase\n",
      "fc add children for caspase\n",
      "fc add children for caspase\n",
      "fc add children for caspase\n",
      "fc add children for caspase\n",
      "Conjuncts [genes, TIMP3]\n",
      "neighbour genes ,\n",
      "neighbour genes such\n",
      "Conjunction before fc TIMP3\n",
      "fc add children for TIMP3\n",
      "Conunct merge check: reduction TIMP3\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( nmod )\n",
      "reduction [reduction, of, miR-21, activated, caspase, 9, .]\n",
      "TIMP3 [genes, such, TIMP3]\n",
      "achildren {miR-21}\n",
      "achildren False\n",
      "SDP sent: Moreover, reduction of miR-21 activated caspase 9 and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.\n",
      "SDP     : [(miR-21, 'NOUN', 'compound', False), (caspase, 'NOUN', 'nmod', False), (9, 'NUM', 'nummod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[3]\n",
      "deepcheck connection rule [by, modulating, multiple, potential, target, genes, ,, such, as, TIMP3]\n",
      "0 [Moreover, ,, reduction, of, miR-21, activated, caspase, 9, and, 3, ,, which, may, be, mediated, by, modulating, multiple, potential, target, genes, ,, such, as, TIMP3] 25 0 Moreover, reduction of miR-21 activated caspase 9 and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-21, activated, caspase, 9, and, 3, ,, which, may, be, mediated, by, modulating, multiple, potential, target, genes, ,, such, as, TIMP3]\n",
      "DeepCheck Conj Subtree\n",
      "[3]\n",
      "deepcheck connection rule [by, modulating, multiple, potential, target, genes, ,, such, as, TIMP3]\n",
      "DeepCheck Conj Subtree\n",
      "[such, as, TIMP3]\n",
      "0 [Moreover, ,, reduction, of, miR-21, activated, caspase, 9, and, 3, ,, which, may, be, mediated, by, modulating, multiple, potential, target, genes, ,, such, as, TIMP3] 25 0 Moreover, reduction of miR-21 activated caspase 9 and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.\n",
      "Context Fail for Noun Rule\n",
      "Recognized overlapping entities: {Interval(40, 49, ('GENE_OR_GENE_PRODUCT', 'caspase 9'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text Moreover , reduction of miR-21 activated caspase 9 and 3 , which may be mediated by modulating multiple potential target genes , such as TIMP3\n",
      "[Moreover, ,, reduction, of, miR-21, activated, caspase, 9, and, 3, ,, which, may, be, mediated, by, modulating, multiple, potential, target, genes, ,, such, as, TIMP3]\n",
      "TextBetween: Moreover , reduction of miR-21 activated caspase 9 and 3 , which may be mediated by modulating multiple potential target genes , such as TIMP3\n",
      "infer direction from token set [Moreover, ,, reduction, of, miR-21, activated, caspase, 9, and, 3, ,, which, may, be, mediated, by, modulating, multiple, potential, target, genes, ,, such, as, TIMP3]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c048>, {'mir': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEG': 1, 'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c6a8>, {'mir': Counter({'NEG': 1, 'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(reduction, 'NEG'), (activated, 'NEU')]\n",
      "Between Tokens [(activated, 'NEU')]\n",
      "counts between final False\n",
      "['reduction', 'of', 'miR-21', 'activated', 'caspase', '9', 'and', '3']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "[(reduction, 'NEG'), (activated, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': False, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Moreover, reduction of miR-21 activated caspase 9 and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.\n",
      "full sentence Moreover, reduction of <e1>miR-21</e1> activated <e2>caspase 9</e2> and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Moreover, reduction of <e1>miR-21</e1> activated <e2>caspase 9</e2> and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.', 'full_sentence': 'Moreover, reduction of <e1>miR-21</e1> activated <e2>caspase 9</e2> and 3, which may be mediated by modulating multiple potential target genes, such as TIMP3.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (23, 29), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (40, 49), 'entity_text': 'caspase 9'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d9.s1.p0\n",
      "IS:  ('MIR_GENE', 'UP')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased <e2>FGF20</e2> expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 'full_sentence': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased <e2>FGF20</e2> expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (33, 40), 'entity_text': 'miR-433'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (81, 86), 'entity_text': 'FGF20'}}\n",
      "add special case miR-433\n",
      "add special case FGF20\n",
      "miR-433 6 FGF20 16\n",
      "(u\"The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 6, 16)\n",
      "geneword FGF20 mirword miR-433\n",
      "Conjuncts [upregulation, expression]\n",
      "neighbour upregulation of\n",
      "neighbour upregulation of\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conjuncts [expression, upregulation]\n",
      "neighbour expression and\n",
      "neighbour expression and\n",
      "Conjunction before fc upregulation\n",
      "fc add children for upregulation\n",
      "Conjunctions ( conj )\n",
      "expression [in, increased, FGF20, expression, upregulation, of, alpha-synuclein]\n",
      "Conjuncts [reduction, affinity]\n",
      "neighbour reduction of\n",
      "neighbour reduction of\n",
      "Conjunction before fc affinity\n",
      "fc add children for affinity\n",
      "fc add children for affinity\n",
      "fc add children for affinity\n",
      "Conjuncts [affinity, miR-433]\n",
      "neighbour affinity of\n",
      "neighbour affinity of\n",
      "Conjunction before fc miR-433\n",
      "fc add children for miR-433\n",
      "Conjuncts [affinity, UTR]\n",
      "neighbour affinity of\n",
      "neighbour affinity of\n",
      "Conjunction before fc UTR\n",
      "fc add children for UTR\n",
      "fc add children for UTR\n",
      "Conjuncts [result, expression]\n",
      "neighbour result in\n",
      "neighbour result in\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conjuncts [upregulation, alpha-synuclein]\n",
      "neighbour upregulation of\n",
      "neighbour upregulation of\n",
      "Conjunction before fc alpha-synuclein\n",
      "fc add children for alpha-synuclein\n",
      "Conjuncts [promote, turn]\n",
      "neighbour promote dopaminergic\n",
      "neighbour promote dopaminergic\n",
      "Conjunction before fc turn\n",
      "fc add children for turn\n",
      "Conunct merge check: affinity expression\n",
      "UTR would\n",
      "Conunct merge check: expression alpha-synuclein\n",
      "expression and\n",
      "Double Conjunction Found\n",
      "Conunct merge check: alpha-synuclein turn\n",
      "alpha-synuclein, which could\n",
      "Conjunctions ( nmod )\n",
      "affinity [reduction, of, affinity, of, miR-433, to, 3, UTR]\n",
      "expression [result, in, increased, FGF20, expression]\n",
      "alpha-synuclein [upregulation, of, alpha-synuclein]\n",
      "turn [in, turn, promote]\n",
      "(alpha-synuclein, 'conj') [result, in, increased, FGF20, expression, upregulation, of, alpha-synuclein]\n",
      "SDP sent: The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "SDP     : [(miR-433, 'NOUN', 'nmod', False), (affinity, 'NOUN', 'nmod', False), (reduction, 'NOUN', 'nsubj', False), (result, 'VERB', 'ROOT', False), (expression, 'NOUN', 'nmod', False), (FGF20, 'NOUN', 'compound', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "0 [The, reduction, of, the, affinity, of, miR-433, to, the, 3, ', UTR, would, result, in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration] 31 0 The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, affinity, of, miR-433, to, the, 3, ', UTR]\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-433]\n",
      "DeepCheck Conj Subtree\n",
      "[to, the, 3, ', UTR]\n",
      "DeepCheck Conj Subtree\n",
      "[in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "DeepCheck Conj Subtree\n",
      "[upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "DeepCheck Conj Subtree\n",
      "[of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "DeepCheck Conj Subtree\n",
      "[in, turn]\n",
      "0 [The, reduction, of, the, affinity, of, miR-433, to, the, 3, ', UTR, would, result, in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration] 31 0 The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "Recognized overlapping entities: {Interval(81, 86, ('GENE_OR_GENE_PRODUCT', 'FGF20'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text The reduction of the affinity of miR-433 to the 3 ' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein , which could in turn promote dopaminergic neurons degeneration\n",
      "[The, reduction, of, the, affinity, of, miR-433, to, the, 3, ', UTR, would, result, in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "TextBetween: The reduction of the affinity of miR-433 to the 3 ' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein , which could in turn promote dopaminergic neurons degeneration\n",
      "infer direction from token set [The, reduction, of, the, affinity, of, miR-433, to, the, 3, ', UTR, would, result, in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "stemTypes2Count: Counter({'POS': 3, 'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c7b8>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'POS': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70598>, {'gene': Counter({'POS': 1, 'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(result, 'NEU'), (increased, 'POS'), (expression, 'NEU')]\n",
      "Between Tokens [(result, 'NEU'), (increased, 'POS')]\n",
      "counts between final False\n",
      "['affinity', 'of', 'miR-433', 'to', 'the', '3', \"'\", 'UTR', 'would', 'result', 'in', 'increased', 'FGF20', 'expression', 'and']\n",
      "Counter({'NEU': 2, 'POS': 1})\n",
      "[(result, 'NEU'), (increased, 'POS'), (expression, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "full sentence The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased <e2>FGF20</e2> expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased <e2>FGF20</e2> expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 'full_sentence': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased <e2>FGF20</e2> expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (33, 40), 'entity_text': 'miR-433'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (81, 86), 'entity_text': 'FGF20'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d9.s1.p1\n",
      "IS:  ('MIR_GENE', 'UP')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased FGF20 expression and upregulation of <e2>alpha-synuclein</e2>, which could in turn promote dopaminergic neurons degeneration.\", 'full_sentence': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased FGF20 expression and upregulation of <e2>alpha-synuclein</e2>, which could in turn promote dopaminergic neurons degeneration.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (33, 40), 'entity_text': 'miR-433'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (118, 133), 'entity_text': 'alpha-synuclein'}}\n",
      "add special case miR-433\n",
      "add special case alpha-synuclein\n",
      "miR-433 6 alpha-synuclein 21\n",
      "(u\"The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\", 6, 21)\n",
      "geneword alpha-synuclein mirword miR-433\n",
      "Conjuncts [upregulation, expression]\n",
      "neighbour upregulation of\n",
      "neighbour upregulation of\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conjuncts [expression, upregulation]\n",
      "neighbour expression and\n",
      "neighbour expression and\n",
      "Conjunction before fc upregulation\n",
      "fc add children for upregulation\n",
      "Conjunctions ( conj )\n",
      "expression [in, increased, FGF20, expression, upregulation, of, alpha-synuclein]\n",
      "Conjuncts [reduction, affinity]\n",
      "neighbour reduction of\n",
      "neighbour reduction of\n",
      "Conjunction before fc affinity\n",
      "fc add children for affinity\n",
      "fc add children for affinity\n",
      "fc add children for affinity\n",
      "Conjuncts [affinity, miR-433]\n",
      "neighbour affinity of\n",
      "neighbour affinity of\n",
      "Conjunction before fc miR-433\n",
      "fc add children for miR-433\n",
      "Conjuncts [affinity, UTR]\n",
      "neighbour affinity of\n",
      "neighbour affinity of\n",
      "Conjunction before fc UTR\n",
      "fc add children for UTR\n",
      "fc add children for UTR\n",
      "Conjuncts [result, expression]\n",
      "neighbour result in\n",
      "neighbour result in\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conjuncts [upregulation, alpha-synuclein]\n",
      "neighbour upregulation of\n",
      "neighbour upregulation of\n",
      "Conjunction before fc alpha-synuclein\n",
      "fc add children for alpha-synuclein\n",
      "Conjuncts [promote, turn]\n",
      "neighbour promote dopaminergic\n",
      "neighbour promote dopaminergic\n",
      "Conjunction before fc turn\n",
      "fc add children for turn\n",
      "Conunct merge check: affinity expression\n",
      "UTR would\n",
      "Conunct merge check: expression alpha-synuclein\n",
      "expression and\n",
      "Double Conjunction Found\n",
      "Conunct merge check: alpha-synuclein turn\n",
      "alpha-synuclein, which could\n",
      "Conjunctions ( nmod )\n",
      "affinity [reduction, of, affinity, of, miR-433, to, 3, UTR]\n",
      "expression [result, in, increased, FGF20, expression]\n",
      "alpha-synuclein [upregulation, of, alpha-synuclein]\n",
      "turn [in, turn, promote]\n",
      "(alpha-synuclein, 'conj') [result, in, increased, FGF20, expression, upregulation, of, alpha-synuclein]\n",
      "SDP sent: The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "SDP     : [(miR-433, 'NOUN', 'nmod', False), (affinity, 'NOUN', 'nmod', False), (reduction, 'NOUN', 'nsubj', False), (result, 'VERB', 'ROOT', False), (expression, 'NOUN', 'nmod', True), (upregulation, 'NOUN', 'conj', False), (alpha-synuclein, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "0 [The, reduction, of, the, affinity, of, miR-433, to, the, 3, ', UTR, would, result, in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration] 31 0 The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, affinity, of, miR-433, to, the, 3, ', UTR]\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-433]\n",
      "DeepCheck Conj Subtree\n",
      "[to, the, 3, ', UTR]\n",
      "DeepCheck Conj Subtree\n",
      "[in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "DeepCheck Conj Subtree\n",
      "[upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "DeepCheck Conj Subtree\n",
      "[of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "DeepCheck Conj Subtree\n",
      "[in, turn]\n",
      "0 [The, reduction, of, the, affinity, of, miR-433, to, the, 3, ', UTR, would, result, in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration] 31 0 The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "Recognized overlapping entities: {Interval(118, 133, ('GENE_OR_GENE_PRODUCT', 'alpha-synuclein'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1, 'POS': 1})\n",
      "Compartment Text The reduction of the affinity of miR-433 to the 3 ' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein , which could in turn promote dopaminergic neurons degeneration\n",
      "[The, reduction, of, the, affinity, of, miR-433, to, the, 3, ', UTR, would, result, in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "TextBetween: The reduction of the affinity of miR-433 to the 3 ' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein , which could in turn promote dopaminergic neurons degeneration\n",
      "infer direction from token set [The, reduction, of, the, affinity, of, miR-433, to, the, 3, ', UTR, would, result, in, increased, FGF20, expression, and, upregulation, of, alpha-synuclein, ,, which, could, in, turn, promote, dopaminergic, neurons, degeneration]\n",
      "stemTypes2Count: Counter({'POS': 3, 'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70ae8>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'POS': 2})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62e18>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count [(result, 'NEU'), (increased, 'POS'), (expression, 'NEU'), (upregulation, 'POS')]\n",
      "Between Tokens [(result, 'NEU'), (increased, 'POS'), (expression, 'NEU'), (upregulation, 'POS')]\n",
      "counts between final False\n",
      "['affinity', 'of', 'miR-433', 'to', 'the', '3', \"'\", 'UTR', 'would', 'result', 'in', 'increased', 'FGF20', 'expression', 'and', 'upregulation', 'of', 'alpha-synuclein', ',', 'which']\n",
      "Counter({'NEU': 2, 'POS': 2})\n",
      "[(result, 'NEU'), (increased, 'POS'), (expression, 'NEU'), (upregulation, 'POS')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence The reduction of the affinity of miR-433 to the 3' UTR would result in increased FGF20 expression and upregulation of alpha-synuclein, which could in turn promote dopaminergic neurons degeneration.\n",
      "full sentence The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased FGF20 expression and upregulation of <e2>alpha-synuclein</e2>, which could in turn promote dopaminergic neurons degeneration.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased FGF20 expression and upregulation of <e2>alpha-synuclein</e2>, which could in turn promote dopaminergic neurons degeneration.\", 'full_sentence': \"The reduction of the affinity of <e1>miR-433</e1> to the 3' UTR would result in increased FGF20 expression and upregulation of <e2>alpha-synuclein</e2>, which could in turn promote dopaminergic neurons degeneration.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (33, 40), 'entity_text': 'miR-433'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (118, 133), 'entity_text': 'alpha-synuclein'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d11.s1.p0\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case apl-1\n",
      "add special case let-7\n",
      "apl-1 6 let-7 14\n",
      "(u\"Moreover, the levels of the apl-1 transcription are modulated by the activity of let-7 family microRNAs.\", 6, 14)\n",
      "geneword apl-1 mirword let-7\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [levels, transcription]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc transcription\n",
      "fc add children for transcription\n",
      "fc add children for transcription\n",
      "Conjuncts [modulated, activity]\n",
      "neighbour modulated by\n",
      "neighbour modulated by\n",
      "Conjunction before fc activity\n",
      "fc add children for activity\n",
      "fc add children for activity\n",
      "Conjuncts [activity, microRNAs]\n",
      "neighbour activity of\n",
      "neighbour activity of\n",
      "Conjunction before fc microRNAs\n",
      "fc add children for microRNAs\n",
      "fc add children for microRNAs\n",
      "fc add children for microRNAs\n",
      "Conunct merge check: transcription activity\n",
      "transcription are\n",
      "Conjunctions ( nmod )\n",
      "transcription [levels, of, apl-1, transcription]\n",
      "activity [modulated, by, activity, of, let-7, family, microRNAs]\n",
      "SDP sent: Moreover, the levels of the apl-1 transcription are modulated by the activity of let-7 family microRNAs.\n",
      "SDP     : [(apl-1, 'NOUN', 'compound', False), (transcription, 'NOUN', 'nmod', False), (levels, 'NOUN', 'nsubjpass', False), (modulated, 'VERB', 'ROOT', False), (activity, 'NOUN', 'nmod', False), (microRNAs, 'NOUN', 'nmod', False), (let-7, 'NOUN', 'compound', False)]\n",
      "0 [Moreover, ,, the, levels, of, the, apl-1, transcription, are, modulated, by, the, activity, of, let-7, family, microRNAs] 17 0 Moreover, the levels of the apl-1 transcription are modulated by the activity of let-7 family microRNAs.\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, apl-1, transcription]\n",
      "DeepCheck Conj Subtree\n",
      "[by, the, activity, of, let-7, family, microRNAs]\n",
      "DeepCheck Conj Subtree\n",
      "[of, let-7, family, microRNAs]\n",
      "0 [Moreover, ,, the, levels, of, the, apl-1, transcription, are, modulated, by, the, activity, of, let-7, family, microRNAs] 17 0 Moreover, the levels of the apl-1 transcription are modulated by the activity of let-7 family microRNAs.\n",
      "Context Fail <regex.Match object; span=(2, 21), match='-7 family microRNAs', fuzzy_counts=(0, 3, 0)> regex.Regex('(family microRNAs){e<=3}', flags=regex.V0)\n",
      "Recognized overlapping entities: {Interval(28, 33, ('GENE_OR_GENE_PRODUCT', 'apl-1'))}\n",
      "passive? True\n",
      "Counter()\n",
      "Compartment Text Moreover , the levels of the apl-1 transcription are modulated by the activity of let-7 family microRNAs\n",
      "[Moreover, ,, the, levels, of, the, apl-1, transcription, are, modulated, by, the, activity, of, let-7, family, microRNAs]\n",
      "TextBetween: Moreover , the levels of the apl-1 transcription are modulated by the activity of let-7 family microRNAs\n",
      "infer direction from token set [Moreover, ,, the, levels, of, the, apl-1, transcription, are, modulated, by, the, activity, of, let-7, family, microRNAs]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62a60>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 3})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62ea0>, {'gene': Counter({'NEU': 1}), 'mir': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(transcription, 'NEU'), (modulated, 'NEU'), (activity, 'NEU')]\n",
      "Between Tokens [(transcription, 'NEU'), (modulated, 'NEU'), (activity, 'NEU')]\n",
      "counts between final True\n",
      "['of', 'the', 'apl-1', 'transcription', 'are', 'modulated', 'by', 'the', 'activity', 'of', 'let-7', 'family', 'microRNAs']\n",
      "Counter({'NEU': 3})\n",
      "[(transcription, 'NEU'), (modulated, 'NEU'), (activity, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Moreover, the levels of the apl-1 transcription are modulated by the activity of let-7 family microRNAs.\n",
      "full sentence Moreover, the levels of the <e2>apl-1</e2> transcription are modulated by the activity of <e1>let-7</e1> family microRNAs.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Moreover, the levels of the <e2>apl-1</e2> transcription are modulated by the activity of <e1>let-7</e1> family microRNAs.', 'full_sentence': 'Moreover, the levels of the <e2>apl-1</e2> transcription are modulated by the activity of <e1>let-7</e1> family microRNAs.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (28, 33), 'entity_text': 'apl-1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (81, 86), 'entity_text': 'let-7'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d11.s1.p0\n",
      "IS:  ('MIR_GENE', 'NEU')\n",
      "SB:  ('NA', 'NA', 'FALSE', 'FALSE')\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Moreover, the levels of the <e2>apl-1</e2> transcription are modulated by the activity of <e1>let-7</e1> family microRNAs.', 'full_sentence': 'Moreover, the levels of the <e2>apl-1</e2> transcription are modulated by the activity of <e1>let-7</e1> family microRNAs.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (28, 33), 'entity_text': 'apl-1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (81, 86), 'entity_text': 'let-7'}}\n",
      "add special case apl-1\n",
      "add special case let-7\n",
      "apl-1 6 let-7 14\n",
      "(u\"Moreover, the levels of the apl-1 transcription are modulated by the activity of let-7 family microRNAs.\", 6, 14)\n",
      "geneword apl-1 mirword let-7\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [levels, transcription]\n",
      "neighbour levels of\n",
      "neighbour levels of\n",
      "Conjunction before fc transcription\n",
      "fc add children for transcription\n",
      "fc add children for transcription\n",
      "Conjuncts [modulated, activity]\n",
      "neighbour modulated by\n",
      "neighbour modulated by\n",
      "Conjunction before fc activity\n",
      "fc add children for activity\n",
      "fc add children for activity\n",
      "Conjuncts [activity, microRNAs]\n",
      "neighbour activity of\n",
      "neighbour activity of\n",
      "Conjunction before fc microRNAs\n",
      "fc add children for microRNAs\n",
      "fc add children for microRNAs\n",
      "fc add children for microRNAs\n",
      "Conunct merge check: transcription activity\n",
      "transcription are\n",
      "Conjunctions ( nmod )\n",
      "transcription [levels, of, apl-1, transcription]\n",
      "activity [modulated, by, activity, of, let-7, family, microRNAs]\n",
      "SDP sent: Moreover, the levels of the apl-1 transcription are modulated by the activity of let-7 family microRNAs.\n",
      "SDP     : [(apl-1, 'NOUN', 'compound', False), (transcription, 'NOUN', 'nmod', False), (levels, 'NOUN', 'nsubjpass', False), (modulated, 'VERB', 'ROOT', False), (activity, 'NOUN', 'nmod', False), (microRNAs, 'NOUN', 'nmod', False), (let-7, 'NOUN', 'compound', False)]\n",
      "0 [Moreover, ,, the, levels, of, the, apl-1, transcription, are, modulated, by, the, activity, of, let-7, family, microRNAs] 17 0 Moreover, the levels of the apl-1 transcription are modulated by the activity of let-7 family microRNAs.\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, apl-1, transcription]\n",
      "DeepCheck Conj Subtree\n",
      "[by, the, activity, of, let-7, family, microRNAs]\n",
      "DeepCheck Conj Subtree\n",
      "[of, let-7, family, microRNAs]\n",
      "0 [Moreover, ,, the, levels, of, the, apl-1, transcription, are, modulated, by, the, activity, of, let-7, family, microRNAs] 17 0 Moreover, the levels of the apl-1 transcription are modulated by the activity of let-7 family microRNAs.\n",
      "Context Fail <regex.Match object; span=(2, 21), match='-7 family microRNAs', fuzzy_counts=(0, 3, 0)> regex.Regex('(family microRNAs){e<=3}', flags=regex.V0)\n",
      "Recognized overlapping entities: {Interval(28, 33, ('GENE_OR_GENE_PRODUCT', 'apl-1'))}\n",
      "passive? True\n",
      "Counter()\n",
      "Compartment Text Moreover , the levels of the apl-1 transcription are modulated by the activity of let-7 family microRNAs\n",
      "[Moreover, ,, the, levels, of, the, apl-1, transcription, are, modulated, by, the, activity, of, let-7, family, microRNAs]\n",
      "TextBetween: Moreover , the levels of the apl-1 transcription are modulated by the activity of let-7 family microRNAs\n",
      "infer direction from token set [Moreover, ,, the, levels, of, the, apl-1, transcription, are, modulated, by, the, activity, of, let-7, family, microRNAs]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62f28>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 3})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62a60>, {'gene': Counter({'NEU': 1}), 'mir': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(transcription, 'NEU'), (modulated, 'NEU'), (activity, 'NEU')]\n",
      "Between Tokens [(transcription, 'NEU'), (modulated, 'NEU'), (activity, 'NEU')]\n",
      "counts between final True\n",
      "['of', 'the', 'apl-1', 'transcription', 'are', 'modulated', 'by', 'the', 'activity', 'of', 'let-7', 'family', 'microRNAs']\n",
      "Counter({'NEU': 3})\n",
      "[(transcription, 'NEU'), (modulated, 'NEU'), (activity, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Moreover, the levels of the apl-1 transcription are modulated by the activity of let-7 family microRNAs.\n",
      "full sentence Moreover, the levels of the <e2>apl-1</e2> transcription are modulated by the activity of <e1>let-7</e1> family microRNAs.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Moreover, the levels of the <e2>apl-1</e2> transcription are modulated by the activity of <e1>let-7</e1> family microRNAs.', 'full_sentence': 'Moreover, the levels of the <e2>apl-1</e2> transcription are modulated by the activity of <e1>let-7</e1> family microRNAs.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (28, 33), 'entity_text': 'apl-1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (81, 86), 'entity_text': 'let-7'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d18.s1.p0\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case microRNA-30\n",
      "add special case Oas1\n",
      "microRNA-30 12 Oas1 18\n",
      "(u\"Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\", 12, 18)\n",
      "geneword Oas1 mirword microRNA-30\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [Modification, construct]\n",
      "neighbour Modification of\n",
      "neighbour Modification of\n",
      "Conjunction before fc construct\n",
      "fc add children for construct\n",
      "fc add children for construct\n",
      "Conjuncts [features, precursor]\n",
      "neighbour features of\n",
      "neighbour features of\n",
      "Conjunction before fc precursor\n",
      "fc add children for precursor\n",
      "fc add children for precursor\n",
      "fc add children for precursor\n",
      "fc add children for precursor\n",
      "Conjuncts [induction, cultures]\n",
      "neighbour induction in\n",
      "neighbour induction in\n",
      "Conjunction before fc cultures\n",
      "fc add children for cultures\n",
      "fc add children for cultures\n",
      "fc add children for cultures\n",
      "Conjuncts [modification, strand]\n",
      "neighbour modification of\n",
      "neighbour modification of\n",
      "Conjunction before fc strand\n",
      "fc add children for strand\n",
      "fc add children for strand\n",
      "Conunct merge check: construct precursor\n",
      "construct by implementing\n",
      "Conunct merge check: precursor cultures\n",
      "precursor avoided Oas1\n",
      "Conunct merge check: cultures strand\n",
      "cultures, whereby\n",
      "Conjunctions ( nmod )\n",
      "construct [Modification, of, shRNA, construct]\n",
      "precursor [features, of, occurring, microRNA-30, miR-30, precursor]\n",
      "cultures [induction, in, transduced, primary, cultures]\n",
      "strand [modification, of, passenger, strand]\n",
      "SDP sent: Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "SDP     : [(microRNA-30, 'ADJ', 'compound', False), (precursor, 'NOUN', 'nmod', False), (features, 'NOUN', 'dobj', False), (implementing, 'VERB', 'advcl', False), (Modification, 'NOUN', 'nsubj', False), (avoided, 'VERB', 'ROOT', False), (induction, 'NOUN', 'dobj', False), (Oas1, 'NOUN', 'compound', False)]\n",
      "SDP DOBJ RULE\n",
      "deepcheck connection rule [by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor]\n",
      "deepcheck connection rule noun found seems modification [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression]\n",
      "0 [Modification, of, the, shRNA, construct, by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor, avoided, Oas1, induction, in, transduced, primary, cultures] 24 0 Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "1 [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression] 17 174 whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, shRNA, construct]\n",
      "deepcheck connection rule [by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor]\n",
      "DeepCheck Conj Subtree\n",
      "[in, transduced, primary, cultures]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, passenger, strand]\n",
      "deepcheck connection rule noun found seems modification [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression]\n",
      "0 [Modification, of, the, shRNA, construct, by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor, avoided, Oas1, induction, in, transduced, primary, cultures] 24 0 Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "1 [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression] 17 174 whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "Recognized overlapping entities: {Interval(127, 131, ('GENE_OR_GENE_PRODUCT', 'Oas1'))}\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "Compartment Text Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 ( miR-30 ) precursor avoided Oas1 induction in transduced primary cultures\n",
      "[Modification, of, the, shRNA, construct, by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor, avoided, Oas1, induction, in, transduced, primary, cultures]\n",
      "TextBetween: Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 ( miR-30 ) precursor avoided Oas1 induction in transduced primary cultures\n",
      "infer direction from token set [Modification, of, the, shRNA, construct, by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor, avoided, Oas1, induction, in, transduced, primary, cultures]\n",
      "stemTypes2Count: Counter({'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62a60>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'POS': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70b70>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count [(induction, 'POS')]\n",
      "Between Tokens []\n",
      "counts between final False\n",
      "['naturally', 'occurring', 'microRNA-30', '(', 'miR-30', ')', 'precursor', 'avoided', 'Oas1', 'induction', 'in']\n",
      "Counter({'POS': 1})\n",
      "[(induction, 'POS')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "full sentence Modification of the shRNA construct by implementing features of the naturally occurring <e1>microRNA-30</e1> (miR-30) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Modification of the shRNA construct by implementing features of the naturally occurring <e1>microRNA-30</e1> (miR-30) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.', 'full_sentence': 'Modification of the shRNA construct by implementing features of the naturally occurring <e1>microRNA-30</e1> (miR-30) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (88, 99), 'entity_text': 'microRNA-30'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (127, 131), 'entity_text': 'Oas1'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d18.s1.p1\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miR-30\n",
      "add special case Oas1\n",
      "miR-30 14 Oas1 18\n",
      "(u\"Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\", 14, 18)\n",
      "geneword Oas1 mirword miR-30\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [Modification, construct]\n",
      "neighbour Modification of\n",
      "neighbour Modification of\n",
      "Conjunction before fc construct\n",
      "fc add children for construct\n",
      "fc add children for construct\n",
      "Conjuncts [features, precursor]\n",
      "neighbour features of\n",
      "neighbour features of\n",
      "Conjunction before fc precursor\n",
      "fc add children for precursor\n",
      "fc add children for precursor\n",
      "fc add children for precursor\n",
      "fc add children for precursor\n",
      "Conjuncts [induction, cultures]\n",
      "neighbour induction in\n",
      "neighbour induction in\n",
      "Conjunction before fc cultures\n",
      "fc add children for cultures\n",
      "fc add children for cultures\n",
      "fc add children for cultures\n",
      "Conjuncts [modification, strand]\n",
      "neighbour modification of\n",
      "neighbour modification of\n",
      "Conjunction before fc strand\n",
      "fc add children for strand\n",
      "fc add children for strand\n",
      "Conunct merge check: construct precursor\n",
      "construct by implementing\n",
      "Conunct merge check: precursor cultures\n",
      "precursor avoided Oas1\n",
      "Conunct merge check: cultures strand\n",
      "cultures, whereby\n",
      "Conjunctions ( nmod )\n",
      "construct [Modification, of, shRNA, construct]\n",
      "precursor [features, of, occurring, microRNA-30, miR-30, precursor]\n",
      "cultures [induction, in, transduced, primary, cultures]\n",
      "strand [modification, of, passenger, strand]\n",
      "SDP sent: Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "SDP     : [(miR-30, 'NOUN', 'compound', False), (precursor, 'NOUN', 'nmod', False), (features, 'NOUN', 'dobj', False), (implementing, 'VERB', 'advcl', False), (Modification, 'NOUN', 'nsubj', False), (avoided, 'VERB', 'ROOT', False), (induction, 'NOUN', 'dobj', False), (Oas1, 'NOUN', 'compound', False)]\n",
      "SDP DOBJ RULE\n",
      "deepcheck connection rule [by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor]\n",
      "deepcheck connection rule noun found seems modification [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression]\n",
      "0 [Modification, of, the, shRNA, construct, by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor, avoided, Oas1, induction, in, transduced, primary, cultures] 24 0 Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "1 [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression] 17 174 whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, shRNA, construct]\n",
      "deepcheck connection rule [by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor]\n",
      "DeepCheck Conj Subtree\n",
      "[in, transduced, primary, cultures]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, passenger, strand]\n",
      "deepcheck connection rule noun found seems modification [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression]\n",
      "0 [Modification, of, the, shRNA, construct, by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor, avoided, Oas1, induction, in, transduced, primary, cultures] 24 0 Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "1 [whereby, modification, of, the, passenger, strand, seems, to, be, a, crucial, feature, to, circumvent, interferon-stimulated, gene, expression] 17 174 whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "Recognized overlapping entities: {Interval(127, 131, ('GENE_OR_GENE_PRODUCT', 'Oas1'))}\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "Compartment Text Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 ( miR-30 ) precursor avoided Oas1 induction in transduced primary cultures\n",
      "[Modification, of, the, shRNA, construct, by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor, avoided, Oas1, induction, in, transduced, primary, cultures]\n",
      "TextBetween: Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 ( miR-30 ) precursor avoided Oas1 induction in transduced primary cultures\n",
      "infer direction from token set [Modification, of, the, shRNA, construct, by, implementing, features, of, the, naturally, occurring, microRNA-30, (, miR-30, ), precursor, avoided, Oas1, induction, in, transduced, primary, cultures]\n",
      "stemTypes2Count: Counter({'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62a60>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'POS': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70b70>, {'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count [(induction, 'POS')]\n",
      "Between Tokens []\n",
      "counts between final False\n",
      "['microRNA-30', '(', 'miR-30', ')', 'precursor', 'avoided', 'Oas1', 'induction', 'in']\n",
      "Counter({'POS': 1})\n",
      "[(induction, 'POS')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (miR-30) precursor avoided Oas1 induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "full sentence Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (<e1>miR-30</e1>) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (<e1>miR-30</e1>) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.', 'full_sentence': 'Modification of the shRNA construct by implementing features of the naturally occurring microRNA-30 (<e1>miR-30</e1>) precursor avoided <e2>Oas1</e2> induction in transduced primary cultures, whereby modification of the passenger strand seems to be a crucial feature to circumvent interferon-stimulated gene expression.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (101, 107), 'entity_text': 'miR-30'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (127, 131), 'entity_text': 'Oas1'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d48.s1.p0\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miR-221/222\n",
      "add special case p27Kip1\n",
      "miR-221/222 2 p27Kip1 8\n",
      "(u\"Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\", 2, 8)\n",
      "geneword p27Kip1 mirword miR-221/222\n",
      "Conjuncts [reduced, p27Kip1]\n",
      "neighbour reduced tumor\n",
      "neighbour reduced tumor\n",
      "Conjunction before fc p27Kip1\n",
      "fc add children for p27Kip1\n",
      "fc add children for p27Kip1\n",
      "Conjuncts [p27Kip1, reduced]\n",
      "neighbour p27Kip1 in\n",
      "neighbour p27Kip1 in\n",
      "Conjunction before fc reduced\n",
      "fc add children for reduced\n",
      "fc add children for reduced\n",
      "Conjunctions ( conj )\n",
      "p27Kip1 [increased, p27Kip1, in, U251, glioma, subcutaneous, mice, reduced, tumor, growth, through, up, regulation, of, p27Kip1]\n",
      "Conjuncts [knocked-down, 2'-OME-oligonucleotides]\n",
      "neighbour knocked-down through\n",
      "neighbour knocked-down through\n",
      "Conjunction before fc 2'-OME-oligonucleotides\n",
      "fc add children for 2'-OME-oligonucleotides\n",
      "fc add children for 2'-OME-oligonucleotides\n",
      "Conjuncts [p27Kip1, mice]\n",
      "neighbour p27Kip1 in\n",
      "neighbour p27Kip1 in\n",
      "Conjunction before fc mice\n",
      "fc add children for mice\n",
      "fc add children for mice\n",
      "fc add children for mice\n",
      "fc add children for mice\n",
      "Conjuncts [reduced, regulation]\n",
      "neighbour reduced tumor\n",
      "neighbour reduced tumor\n",
      "Conjunction before fc regulation\n",
      "fc add children for regulation\n",
      "fc add children for regulation\n",
      "Conjuncts [regulation, p27Kip1]\n",
      "neighbour regulation of\n",
      "neighbour regulation of\n",
      "Conjunction before fc p27Kip1\n",
      "fc add children for p27Kip1\n",
      "Conunct merge check: 2'-OME-oligonucleotides mice\n",
      "2'-OME-oligonucleotides increased\n",
      "Conunct merge check: mice regulation\n",
      "mice and strongly\n",
      "Double Conjunction Found\n",
      "Conjunctions ( nmod )\n",
      "2'-OME-oligonucleotides [knocked-down, through, antisense, 2'-OME-oligonucleotides]\n",
      "mice [p27Kip1, in, U251, glioma, subcutaneous, mice]\n",
      "regulation [reduced, through, up, regulation, of, p27Kip1]\n",
      "(regulation, 'conj') [p27Kip1, in, U251, glioma, subcutaneous, mice, reduced, through, up, regulation, of, p27Kip1]\n",
      "SDP sent: Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "SDP     : [(miR-221/222, 'NOUN', 'amod', False), (knocked-down, 'VERB', 'nsubj', False), (increased, 'VERB', 'amod', False), (p27Kip1, 'NOUN', 'ROOT', False)]\n",
      "0 [Consistently, ,, miR-221/222, knocked-down, through, antisense, 2'-OME-oligonucleotides, increased, p27Kip1, in, U251, glioma, subcutaneous, mice] 14 0 Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "1 [strongly, reduced, tumor, growth, in, vivo, through, up, regulation, of, p27Kip1] 11 136 strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "DeepCheck Conj Subtree\n",
      "[through, antisense, 2'-OME-oligonucleotides]\n",
      "DeepCheck Conj Subtree\n",
      "[in, U251, glioma, subcutaneous, mice]\n",
      "DeepCheck Conj Subtree\n",
      "[through, up, regulation, of, p27Kip1]\n",
      "DeepCheck Conj Subtree\n",
      "[of, p27Kip1]\n",
      "0 [Consistently, ,, miR-221/222, knocked-down, through, antisense, 2'-OME-oligonucleotides, increased, p27Kip1, in, U251, glioma, subcutaneous, mice] 14 0 Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "1 [strongly, reduced, tumor, growth, in, vivo, through, up, regulation, of, p27Kip1] 11 136 strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "Recognized overlapping entities: {Interval(91, 98, ('GENE_OR_GENE_PRODUCT', 'p27Kip1'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1, 'POS': 1})\n",
      "Compartment Text Consistently , miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice\n",
      "[Consistently, ,, miR-221/222, knocked-down, through, antisense, 2'-OME-oligonucleotides, increased, p27Kip1, in, U251, glioma, subcutaneous, mice]\n",
      "TextBetween: Consistently , miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice\n",
      "infer direction from token set [Consistently, ,, miR-221/222, knocked-down, through, antisense, 2'-OME-oligonucleotides, increased, p27Kip1, in, U251, glioma, subcutaneous, mice]\n",
      "stemTypes2Count: Counter({'NEG': 1, 'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70950>, {'mir': Counter({'NEG': 1}), 'gene': Counter({'POS': 1})})\n",
      "defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70950>, {'mir': Counter({'NEG': 1}), 'gene': Counter({'POS': 1})})\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "full sentence Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased <e2>p27Kip1</e2> in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased <e2>p27Kip1</e2> in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\", 'full_sentence': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased <e2>p27Kip1</e2> in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (14, 25), 'entity_text': 'miR-221/222'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (91, 98), 'entity_text': 'p27Kip1'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d48.s1.p3\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case miR-221/222\n",
      "add special case p27Kip1\n",
      "miR-221/222 2 p27Kip1 25\n",
      "(u\"Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\", 2, 25)\n",
      "geneword p27Kip1 mirword miR-221/222\n",
      "Conjuncts [reduced, p27Kip1]\n",
      "neighbour reduced tumor\n",
      "neighbour reduced tumor\n",
      "Conjunction before fc p27Kip1\n",
      "fc add children for p27Kip1\n",
      "fc add children for p27Kip1\n",
      "Conjuncts [p27Kip1, reduced]\n",
      "neighbour p27Kip1 in\n",
      "neighbour p27Kip1 in\n",
      "Conjunction before fc reduced\n",
      "fc add children for reduced\n",
      "fc add children for reduced\n",
      "Conjunctions ( conj )\n",
      "p27Kip1 [increased, p27Kip1, in, U251, glioma, subcutaneous, mice, reduced, tumor, growth, through, up, regulation, of, p27Kip1]\n",
      "Conjuncts [knocked-down, 2'-OME-oligonucleotides]\n",
      "neighbour knocked-down through\n",
      "neighbour knocked-down through\n",
      "Conjunction before fc 2'-OME-oligonucleotides\n",
      "fc add children for 2'-OME-oligonucleotides\n",
      "fc add children for 2'-OME-oligonucleotides\n",
      "Conjuncts [p27Kip1, mice]\n",
      "neighbour p27Kip1 in\n",
      "neighbour p27Kip1 in\n",
      "Conjunction before fc mice\n",
      "fc add children for mice\n",
      "fc add children for mice\n",
      "fc add children for mice\n",
      "fc add children for mice\n",
      "Conjuncts [reduced, regulation]\n",
      "neighbour reduced tumor\n",
      "neighbour reduced tumor\n",
      "Conjunction before fc regulation\n",
      "fc add children for regulation\n",
      "fc add children for regulation\n",
      "Conjuncts [regulation, p27Kip1]\n",
      "neighbour regulation of\n",
      "neighbour regulation of\n",
      "Conjunction before fc p27Kip1\n",
      "fc add children for p27Kip1\n",
      "Conunct merge check: 2'-OME-oligonucleotides mice\n",
      "2'-OME-oligonucleotides increased\n",
      "Conunct merge check: mice regulation\n",
      "mice and strongly\n",
      "Double Conjunction Found\n",
      "Conjunctions ( nmod )\n",
      "2'-OME-oligonucleotides [knocked-down, through, antisense, 2'-OME-oligonucleotides]\n",
      "mice [p27Kip1, in, U251, glioma, subcutaneous, mice]\n",
      "regulation [reduced, through, up, regulation, of, p27Kip1]\n",
      "(regulation, 'conj') [p27Kip1, in, U251, glioma, subcutaneous, mice, reduced, through, up, regulation, of, p27Kip1]\n",
      "SDP sent: Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "SDP     : [(miR-221/222, 'NOUN', 'amod', False), (knocked-down, 'VERB', 'nsubj', False), (increased, 'VERB', 'amod', False), (p27Kip1, 'NOUN', 'ROOT', True), (reduced, 'VERB', 'conj', False), (regulation, 'NOUN', 'nmod', False), (p27Kip1, 'NOUN', 'nmod', False)]\n",
      "0 [Consistently, ,, miR-221/222, knocked-down, through, antisense, 2'-OME-oligonucleotides, increased, p27Kip1, in, U251, glioma, subcutaneous, mice] 14 0 Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "1 [strongly, reduced, tumor, growth, in, vivo, through, up, regulation, of, p27Kip1] 11 136 strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "DeepCheck Conj Subtree\n",
      "[through, antisense, 2'-OME-oligonucleotides]\n",
      "DeepCheck Conj Subtree\n",
      "[in, U251, glioma, subcutaneous, mice]\n",
      "DeepCheck Conj Subtree\n",
      "[through, up, regulation, of, p27Kip1]\n",
      "DeepCheck Conj Subtree\n",
      "[of, p27Kip1]\n",
      "0 [Consistently, ,, miR-221/222, knocked-down, through, antisense, 2'-OME-oligonucleotides, increased, p27Kip1, in, U251, glioma, subcutaneous, mice] 14 0 Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "1 [strongly, reduced, tumor, growth, in, vivo, through, up, regulation, of, p27Kip1] 11 136 strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "Recognized overlapping entities: {Interval(199, 206, ('GENE_OR_GENE_PRODUCT', 'p27Kip1'))}\n",
      "passive? False\n",
      "Counter({'NEG': 2, 'POS': 1})\n",
      "Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "TextBetween: Consistently , miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1 .\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between mir gene', 'reg_detect_major': 'between'}})\n",
      " hit sentence Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "full sentence Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between mir gene', 'reg_detect_major': 'between'}}, 'tagged_sent': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\", 'full_sentence': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (14, 25), 'entity_text': 'miR-221/222'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (199, 206), 'entity_text': 'p27Kip1'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d48.s1.p3\n",
      "IS:  ('MIR_GENE', 'NEU')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between mir gene', 'reg_detect_major': 'between'}}, 'tagged_sent': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\", 'full_sentence': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (14, 25), 'entity_text': 'miR-221/222'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (199, 206), 'entity_text': 'p27Kip1'}}\n",
      "add special case miR-221/222\n",
      "add special case p27Kip1\n",
      "miR-221/222 2 p27Kip1 25\n",
      "(u\"Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\", 2, 25)\n",
      "geneword p27Kip1 mirword miR-221/222\n",
      "Conjuncts [reduced, p27Kip1]\n",
      "neighbour reduced tumor\n",
      "neighbour reduced tumor\n",
      "Conjunction before fc p27Kip1\n",
      "fc add children for p27Kip1\n",
      "fc add children for p27Kip1\n",
      "Conjuncts [p27Kip1, reduced]\n",
      "neighbour p27Kip1 in\n",
      "neighbour p27Kip1 in\n",
      "Conjunction before fc reduced\n",
      "fc add children for reduced\n",
      "fc add children for reduced\n",
      "Conjunctions ( conj )\n",
      "p27Kip1 [increased, p27Kip1, in, U251, glioma, subcutaneous, mice, reduced, tumor, growth, through, up, regulation, of, p27Kip1]\n",
      "Conjuncts [knocked-down, 2'-OME-oligonucleotides]\n",
      "neighbour knocked-down through\n",
      "neighbour knocked-down through\n",
      "Conjunction before fc 2'-OME-oligonucleotides\n",
      "fc add children for 2'-OME-oligonucleotides\n",
      "fc add children for 2'-OME-oligonucleotides\n",
      "Conjuncts [p27Kip1, mice]\n",
      "neighbour p27Kip1 in\n",
      "neighbour p27Kip1 in\n",
      "Conjunction before fc mice\n",
      "fc add children for mice\n",
      "fc add children for mice\n",
      "fc add children for mice\n",
      "fc add children for mice\n",
      "Conjuncts [reduced, regulation]\n",
      "neighbour reduced tumor\n",
      "neighbour reduced tumor\n",
      "Conjunction before fc regulation\n",
      "fc add children for regulation\n",
      "fc add children for regulation\n",
      "Conjuncts [regulation, p27Kip1]\n",
      "neighbour regulation of\n",
      "neighbour regulation of\n",
      "Conjunction before fc p27Kip1\n",
      "fc add children for p27Kip1\n",
      "Conunct merge check: 2'-OME-oligonucleotides mice\n",
      "2'-OME-oligonucleotides increased\n",
      "Conunct merge check: mice regulation\n",
      "mice and strongly\n",
      "Double Conjunction Found\n",
      "Conjunctions ( nmod )\n",
      "2'-OME-oligonucleotides [knocked-down, through, antisense, 2'-OME-oligonucleotides]\n",
      "mice [p27Kip1, in, U251, glioma, subcutaneous, mice]\n",
      "regulation [reduced, through, up, regulation, of, p27Kip1]\n",
      "(regulation, 'conj') [p27Kip1, in, U251, glioma, subcutaneous, mice, reduced, through, up, regulation, of, p27Kip1]\n",
      "SDP sent: Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "SDP     : [(miR-221/222, 'NOUN', 'amod', False), (knocked-down, 'VERB', 'nsubj', False), (increased, 'VERB', 'amod', False), (p27Kip1, 'NOUN', 'ROOT', True), (reduced, 'VERB', 'conj', False), (regulation, 'NOUN', 'nmod', False), (p27Kip1, 'NOUN', 'nmod', False)]\n",
      "0 [Consistently, ,, miR-221/222, knocked-down, through, antisense, 2'-OME-oligonucleotides, increased, p27Kip1, in, U251, glioma, subcutaneous, mice] 14 0 Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "1 [strongly, reduced, tumor, growth, in, vivo, through, up, regulation, of, p27Kip1] 11 136 strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "DeepCheck Conj Subtree\n",
      "[through, antisense, 2'-OME-oligonucleotides]\n",
      "DeepCheck Conj Subtree\n",
      "[in, U251, glioma, subcutaneous, mice]\n",
      "DeepCheck Conj Subtree\n",
      "[through, up, regulation, of, p27Kip1]\n",
      "DeepCheck Conj Subtree\n",
      "[of, p27Kip1]\n",
      "0 [Consistently, ,, miR-221/222, knocked-down, through, antisense, 2'-OME-oligonucleotides, increased, p27Kip1, in, U251, glioma, subcutaneous, mice] 14 0 Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "1 [strongly, reduced, tumor, growth, in, vivo, through, up, regulation, of, p27Kip1] 11 136 strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "Recognized overlapping entities: {Interval(199, 206, ('GENE_OR_GENE_PRODUCT', 'p27Kip1'))}\n",
      "passive? False\n",
      "Counter({'NEG': 2, 'POS': 1})\n",
      "Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "TextBetween: Consistently , miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1 .\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between mir gene', 'reg_detect_major': 'between'}})\n",
      " hit sentence Consistently, miR-221/222 knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of p27Kip1.\n",
      "full sentence Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between mir gene', 'reg_detect_major': 'between'}}, 'tagged_sent': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\", 'full_sentence': \"Consistently, <e1>miR-221/222</e1> knocked-down through antisense 2'-OME-oligonucleotides increased p27Kip1 in U251 glioma subcutaneous mice and strongly reduced tumor growth in vivo through up regulation of <e2>p27Kip1</e2>.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (14, 25), 'entity_text': 'miR-221/222'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (199, 206), 'entity_text': 'p27Kip1'}}\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "The data strongly suggested that a regulatory loop between miR-21 and STAT3 might provide an insight into the mechanism of modulating EGFR/STAT3 signaling. \n",
      "('miR-21', 'Specific_miRNAs', (59, 65)) ('STAT3', 'Genes/Proteins', (70, 75))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d53.s6.p0\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miR-29b\n",
      "add special case PGRN\n",
      "miR-29b 0 PGRN 10\n",
      "(u\"miR-29b downregulates the expression of luciferase through hPGRN or mouse PGRN (mPGRN) 3'UTRs, and the regulation was abolished by mutations in the miR-29b binding site.\", 0, 10)\n",
      "geneword PGRN mirword miR-29b\n",
      "Conjuncts [abolished, downregulates]\n",
      "neighbour abolished by\n",
      "neighbour abolished by\n",
      "Conjunction before fc downregulates\n",
      "fc add children for downregulates\n",
      "Conjuncts [mouse, hPGRN]\n",
      "neighbour mouse PGRN\n",
      "neighbour mouse PGRN\n",
      "Conjunction before fc hPGRN\n",
      "Conjuncts [hPGRN, mouse]\n",
      "neighbour hPGRN or\n",
      "neighbour hPGRN or\n",
      "Conjunction before fc mouse\n",
      "Conjuncts [downregulates, abolished]\n",
      "neighbour downregulates the\n",
      "neighbour downregulates the\n",
      "Conjunction before fc abolished\n",
      "fc add children for abolished\n",
      "Conjunctions ( conj )\n",
      "downregulates [downregulates, expression, of, luciferase, through, hPGRN, mouse, PGRN, mPGRN, 3'UTRs, abolished, by, mutations, in, miR-29b, binding, site]\n",
      "Conjuncts [expression, luciferase]\n",
      "neighbour expression of\n",
      "neighbour expression of\n",
      "Conjunction before fc luciferase\n",
      "fc add children for luciferase\n",
      "Conjuncts [expression, 3'UTRs]\n",
      "neighbour expression of\n",
      "neighbour expression of\n",
      "Conjunction before fc 3'UTRs\n",
      "fc add children for 3'UTRs\n",
      "fc add children for 3'UTRs\n",
      "fc add children for 3'UTRs\n",
      "fc add children for 3'UTRs\n",
      "Conjuncts [abolished, mutations]\n",
      "neighbour abolished by\n",
      "neighbour abolished by\n",
      "Conjunction before fc mutations\n",
      "fc add children for mutations\n",
      "fc add children for mutations\n",
      "Conjuncts [mutations, site]\n",
      "neighbour mutations in\n",
      "neighbour mutations in\n",
      "Conjunction before fc site\n",
      "fc add children for site\n",
      "fc add children for site\n",
      "fc add children for site\n",
      "Conunct merge check: luciferase mutations\n",
      "3'UTRs, and the regulation was\n",
      "Double Conjunction Found\n",
      "Conjunctions ( nmod )\n",
      "luciferase [expression, of, luciferase, through, hPGRN, PGRN, mPGRN, 3'UTRs]\n",
      "mutations [abolished, by, mutations, in, miR-29b, binding, site]\n",
      "(mutations, 'conj') [expression, of, luciferase, through, hPGRN, PGRN, mPGRN, 3'UTRs, abolished, by, mutations, in, miR-29b, binding, site]\n",
      "SDP sent: miR-29b downregulates the expression of luciferase through hPGRN or mouse PGRN (mPGRN) 3'UTRs, and the regulation was abolished by mutations in the miR-29b binding site.\n",
      "SDP     : [(miR-29b, 'NOUN', 'nsubj', False), (downregulates, 'VERB', 'ROOT', False), (expression, 'NOUN', 'dobj', False), (3'UTRs, 'ADJ', 'nmod', False), (PGRN, 'NOUN', 'compound', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[mouse]\n",
      "deepcheck connection rule noun found abolished regulation [the, regulation, was, abolished, by, mutations, in, the, miR-29b, binding, site]\n",
      "0 [miR-29b, downregulates, the, expression, of, luciferase, through, hPGRN, or, mouse, PGRN, (, mPGRN, ), 3'UTRs] 15 0 miR-29b downregulates the expression of luciferase through hPGRN or mouse PGRN (mPGRN) 3'UTRs, and the regulation was abolished by mutations in the miR-29b binding site.\n",
      "1 [the, regulation, was, abolished, by, mutations, in, the, miR-29b, binding, site] 11 99 the regulation was abolished by mutations in the miR-29b binding site.\n",
      "DeepCheck Conj Subtree\n",
      "[of, luciferase]\n",
      "DeepCheck Conj Subtree\n",
      "[mouse]\n",
      "DeepCheck Conj Subtree\n",
      "[through, hPGRN, or, mouse, PGRN, (, mPGRN, ), 3'UTRs]\n",
      "deepcheck connection rule noun found abolished regulation [the, regulation, was, abolished, by, mutations, in, the, miR-29b, binding, site]\n",
      "DeepCheck Conj Subtree\n",
      "[by, mutations, in, the, miR-29b, binding, site]\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, miR-29b, binding, site]\n",
      "0 [miR-29b, downregulates, the, expression, of, luciferase, through, hPGRN, or, mouse, PGRN, (, mPGRN, ), 3'UTRs] 15 0 miR-29b downregulates the expression of luciferase through hPGRN or mouse PGRN (mPGRN) 3'UTRs, and the regulation was abolished by mutations in the miR-29b binding site.\n",
      "1 [the, regulation, was, abolished, by, mutations, in, the, miR-29b, binding, site] 11 99 the regulation was abolished by mutations in the miR-29b binding site.\n",
      "Recognized overlapping entities: {Interval(68, 78, ('ORGANISM', 'mouse PGRN'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text miR-29b downregulates the expression of luciferase through hPGRN or mouse PGRN ( mPGRN ) 3'UTRs\n",
      "[miR-29b, downregulates, the, expression, of, luciferase, through, hPGRN, or, mouse, PGRN, (, mPGRN, ), 3'UTRs]\n",
      "TextBetween: miR-29b downregulates the expression of luciferase through hPGRN or mouse PGRN ( mPGRN ) 3'UTRs\n",
      "infer direction from token set [miR-29b, downregulates, the, expression, of, luciferase, through, hPGRN, or, mouse, PGRN, (, mPGRN, ), 3'UTRs]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62ea0>, {'mir': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEG': 1, 'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62d08>, {'mir': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count [(downregulates, 'NEG'), (expression, 'NEU')]\n",
      "Between Tokens [(downregulates, 'NEG'), (expression, 'NEU')]\n",
      "counts between final False\n",
      "['miR-29b', 'downregulates', 'the', 'expression', 'of', 'luciferase', 'through', 'hPGRN', 'or', 'mouse', 'PGRN', '(', 'mPGRN']\n",
      "Counter({'NEG': 1, 'NEU': 1})\n",
      "[(downregulates, 'NEG'), (expression, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence miR-29b downregulates the expression of luciferase through hPGRN or mouse PGRN (mPGRN) 3'UTRs, and the regulation was abolished by mutations in the miR-29b binding site.\n",
      "full sentence <e1>miR-29b</e1> downregulates the expression of luciferase through hPGRN or mouse <e2>PGRN</e2> (mPGRN) 3'UTRs, and the regulation was abolished by mutations in the miR-29b binding site.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"<e1>miR-29b</e1> downregulates the expression of luciferase through hPGRN or mouse <e2>PGRN</e2> (mPGRN) 3'UTRs, and the regulation was abolished by mutations in the miR-29b binding site.\", 'full_sentence': \"<e1>miR-29b</e1> downregulates the expression of luciferase through hPGRN or mouse <e2>PGRN</e2> (mPGRN) 3'UTRs, and the regulation was abolished by mutations in the miR-29b binding site.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (0, 7), 'entity_text': 'miR-29b'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (74, 78), 'entity_text': 'PGRN'}}\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We identified an evolutionarily conserved binding site for microRNA-29b (miR-29b) in the 3' untranslated region (3'UTR) of the human PGRN (hPGRN) mRNA. \n",
      "('microRNA-29b', 'Specific_miRNAs', (59, 71)) ('GRN', 'Genes/Proteins', (133, 137))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}\n",
      "We identified an evolutionarily conserved binding site for microRNA-29b (miR-29b) in the 3' untranslated region (3'UTR) of the human PGRN (hPGRN) mRNA. \n",
      "('miR-29b', 'Specific_miRNAs', (73, 80)) ('GRN', 'Genes/Proteins', (133, 137))\n",
      "('MIR_GENE', 'NEU') return passive? False\n",
      "('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d54.s9.p0\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case miR-21\n",
      "add special case FASLG\n",
      "miR-21 6 FASLG 17\n",
      "(u\"Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\", 6, 17)\n",
      "geneword FASLG mirword miR-21\n",
      "Conjuncts [mediates, protects]\n",
      "neighbour mediates the\n",
      "neighbour mediates the\n",
      "Conjunction before fc protects\n",
      "fc add children for protects\n",
      "Conjuncts [ligand, FASLG]\n",
      "neighbour ligand whose\n",
      "neighbour ligand whose\n",
      "Conjunction before fc FASLG\n",
      "fc add children for FASLG\n",
      "fc add children for FASLG\n",
      "Conjuncts [FASLG, ligand]\n",
      "neighbour FASLG ,\n",
      "neighbour FASLG a\n",
      "Conjunction before fc ligand\n",
      "fc add children for ligand\n",
      "fc add children for ligand\n",
      "fc add children for ligand\n",
      "Conjuncts [protects, mediates]\n",
      "neighbour protects against\n",
      "neighbour protects against\n",
      "Conjunction before fc mediates\n",
      "fc add children for mediates\n",
      "Conunct merge check: protects FASLG\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( conj )\n",
      "protects [protects, against, ischemic, neuronal, death, mediates, neuroprotective, effect]\n",
      "FASLG [of, FASLG, a, tumor, necrosis, factor-, family, member, important, cell, death-inducing, ligand]\n",
      "Conjuncts [overexpression, miR-21]\n",
      "neighbour overexpression of\n",
      "neighbour overexpression of\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conjuncts [protects, death]\n",
      "neighbour protects against\n",
      "neighbour protects against\n",
      "Conjunction before fc death\n",
      "fc add children for death\n",
      "fc add children for death\n",
      "fc add children for death\n",
      "Conjuncts [downregulation, FASLG]\n",
      "neighbour downregulation of\n",
      "neighbour downregulation of\n",
      "Conjunction before fc FASLG\n",
      "fc add children for FASLG\n",
      "fc add children for FASLG\n",
      "Conjuncts [targeted, miR-21]\n",
      "neighbour targeted by\n",
      "neighbour targeted by\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conunct merge check: miR-21 death\n",
      "miR-21\n",
      "Conunct merge check: death FASLG\n",
      "death, and that\n",
      "Double Conjunction Found\n",
      "Conunct merge check: FASLG miR-21\n",
      "member and an important cell death-inducing ligand whose gene is\n",
      "Conjunctions ( nmod )\n",
      "miR-21 [overexpression, of, miR-21]\n",
      "death [protects, against, ischemic, neuronal, death]\n",
      "FASLG [downregulation, of, FASLG, a, tumor, necrosis, factor-, family, member]\n",
      "miR-21 [targeted, by, miR-21]\n",
      "(FASLG, 'conj') [protects, against, ischemic, neuronal, death, downregulation, of, FASLG, a, tumor, necrosis, factor-, family, member]\n",
      "SDP sent: Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "SDP     : [(miR-21, 'NOUN', 'nmod', False), (overexpression, 'NOUN', 'nsubj', False), (protects, 'VERB', 'ccomp', True), (mediates, 'VERB', 'conj', False), (downregulation, 'NOUN', 'nsubj', False), (FASLG, 'NOUN', 'nmod', False)]\n",
      "deepcheck connection rule noun found protects overexpression [that, overexpression, of, miR-21, protects, against, ischemic, neuronal, death, ,, and, that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "DeepCheck Conj Subtree\n",
      "[an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "deepcheck connection rule noun found mediates downregulation [that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "0 [Our, data, indicate] 3 0 Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "1 [overexpression, of, miR-21, protects, against, ischemic, neuronal, death] 8 23 overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "2 [downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect] 28 99 downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-21]\n",
      "deepcheck connection rule noun found protects overexpression [that, overexpression, of, miR-21, protects, against, ischemic, neuronal, death, ,, and, that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "DeepCheck Conj Subtree\n",
      "[against, ischemic, neuronal, death]\n",
      "DeepCheck Conj Subtree\n",
      "[of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[by, miR-21]\n",
      "deepcheck connection rule noun found mediates downregulation [that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "0 [Our, data, indicate] 3 0 Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "1 [overexpression, of, miR-21, protects, against, ischemic, neuronal, death] 8 23 overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "2 [downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect] 28 99 downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "Context Fail <regex.Match object; span=(7, 13), match=' tumor', fuzzy_counts=(0, 1, 0)> regex.Regex('(tumor){e<=1}', flags=regex.V0)\n",
      "Recognized overlapping entities: {Interval(117, 122, ('GENE_OR_GENE_PRODUCT', 'FASLG'))}\n",
      "passive? False\n",
      "Counter({'POS': 1, 'NEG': 1})\n",
      "Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "TextBetween: overexpression of miR-21 protects against ischemic neuronal death , and that downregulation of FASLG , a\n",
      "infer direction from token set [overexpression, of, miR-21, protects, against, ischemic, neuronal, death, ,, and, that, downregulation, of, FASLG, ,, a]\n",
      "stemTypes2Count: Counter({'POS': 1, 'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c7b8>, {'mir': Counter({'POS': 1}), 'gene': Counter({'NEG': 1})})\n",
      "defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c7b8>, {'mir': Counter({'POS': 1}), 'gene': Counter({'NEG': 1})})\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "full sentence Our data indicate that overexpression of <e1>miR-21</e1> protects against ischemic neuronal death, and that downregulation of <e2>FASLG</e2>, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Our data indicate that overexpression of <e1>miR-21</e1> protects against ischemic neuronal death, and that downregulation of <e2>FASLG</e2>, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.', 'full_sentence': 'Our data indicate that overexpression of <e1>miR-21</e1> protects against ischemic neuronal death, and that downregulation of <e2>FASLG</e2>, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (41, 47), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (117, 122), 'entity_text': 'FASLG'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d54.s9.p0\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('NA', 'NA', 'FALSE', 'FALSE')\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Our data indicate that overexpression of <e1>miR-21</e1> protects against ischemic neuronal death, and that downregulation of <e2>FASLG</e2>, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.', 'full_sentence': 'Our data indicate that overexpression of <e1>miR-21</e1> protects against ischemic neuronal death, and that downregulation of <e2>FASLG</e2>, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (41, 47), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (117, 122), 'entity_text': 'FASLG'}}\n",
      "add special case miR-21\n",
      "add special case FASLG\n",
      "miR-21 6 FASLG 17\n",
      "(u\"Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\", 6, 17)\n",
      "geneword FASLG mirword miR-21\n",
      "Conjuncts [mediates, protects]\n",
      "neighbour mediates the\n",
      "neighbour mediates the\n",
      "Conjunction before fc protects\n",
      "fc add children for protects\n",
      "Conjuncts [ligand, FASLG]\n",
      "neighbour ligand whose\n",
      "neighbour ligand whose\n",
      "Conjunction before fc FASLG\n",
      "fc add children for FASLG\n",
      "fc add children for FASLG\n",
      "Conjuncts [FASLG, ligand]\n",
      "neighbour FASLG ,\n",
      "neighbour FASLG a\n",
      "Conjunction before fc ligand\n",
      "fc add children for ligand\n",
      "fc add children for ligand\n",
      "fc add children for ligand\n",
      "Conjuncts [protects, mediates]\n",
      "neighbour protects against\n",
      "neighbour protects against\n",
      "Conjunction before fc mediates\n",
      "fc add children for mediates\n",
      "Conunct merge check: protects FASLG\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( conj )\n",
      "protects [protects, against, ischemic, neuronal, death, mediates, neuroprotective, effect]\n",
      "FASLG [of, FASLG, a, tumor, necrosis, factor-, family, member, important, cell, death-inducing, ligand]\n",
      "Conjuncts [overexpression, miR-21]\n",
      "neighbour overexpression of\n",
      "neighbour overexpression of\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conjuncts [protects, death]\n",
      "neighbour protects against\n",
      "neighbour protects against\n",
      "Conjunction before fc death\n",
      "fc add children for death\n",
      "fc add children for death\n",
      "fc add children for death\n",
      "Conjuncts [downregulation, FASLG]\n",
      "neighbour downregulation of\n",
      "neighbour downregulation of\n",
      "Conjunction before fc FASLG\n",
      "fc add children for FASLG\n",
      "fc add children for FASLG\n",
      "Conjuncts [targeted, miR-21]\n",
      "neighbour targeted by\n",
      "neighbour targeted by\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conunct merge check: miR-21 death\n",
      "miR-21\n",
      "Conunct merge check: death FASLG\n",
      "death, and that\n",
      "Double Conjunction Found\n",
      "Conunct merge check: FASLG miR-21\n",
      "member and an important cell death-inducing ligand whose gene is\n",
      "Conjunctions ( nmod )\n",
      "miR-21 [overexpression, of, miR-21]\n",
      "death [protects, against, ischemic, neuronal, death]\n",
      "FASLG [downregulation, of, FASLG, a, tumor, necrosis, factor-, family, member]\n",
      "miR-21 [targeted, by, miR-21]\n",
      "(FASLG, 'conj') [protects, against, ischemic, neuronal, death, downregulation, of, FASLG, a, tumor, necrosis, factor-, family, member]\n",
      "SDP sent: Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "SDP     : [(miR-21, 'NOUN', 'nmod', False), (overexpression, 'NOUN', 'nsubj', False), (protects, 'VERB', 'ccomp', True), (mediates, 'VERB', 'conj', False), (downregulation, 'NOUN', 'nsubj', False), (FASLG, 'NOUN', 'nmod', False)]\n",
      "deepcheck connection rule noun found protects overexpression [that, overexpression, of, miR-21, protects, against, ischemic, neuronal, death, ,, and, that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "DeepCheck Conj Subtree\n",
      "[an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "deepcheck connection rule noun found mediates downregulation [that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "0 [Our, data, indicate] 3 0 Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "1 [overexpression, of, miR-21, protects, against, ischemic, neuronal, death] 8 23 overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "2 [downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect] 28 99 downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-21]\n",
      "deepcheck connection rule noun found protects overexpression [that, overexpression, of, miR-21, protects, against, ischemic, neuronal, death, ,, and, that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "DeepCheck Conj Subtree\n",
      "[against, ischemic, neuronal, death]\n",
      "DeepCheck Conj Subtree\n",
      "[of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[by, miR-21]\n",
      "deepcheck connection rule noun found mediates downregulation [that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "0 [Our, data, indicate] 3 0 Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "1 [overexpression, of, miR-21, protects, against, ischemic, neuronal, death] 8 23 overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "2 [downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect] 28 99 downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "Context Fail <regex.Match object; span=(7, 13), match=' tumor', fuzzy_counts=(0, 1, 0)> regex.Regex('(tumor){e<=1}', flags=regex.V0)\n",
      "Recognized overlapping entities: {Interval(117, 122, ('GENE_OR_GENE_PRODUCT', 'FASLG'))}\n",
      "passive? False\n",
      "Counter({'POS': 1, 'NEG': 1})\n",
      "Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "TextBetween: overexpression of miR-21 protects against ischemic neuronal death , and that downregulation of FASLG , a\n",
      "infer direction from token set [overexpression, of, miR-21, protects, against, ischemic, neuronal, death, ,, and, that, downregulation, of, FASLG, ,, a]\n",
      "stemTypes2Count: Counter({'POS': 1, 'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70598>, {'mir': Counter({'POS': 1}), 'gene': Counter({'NEG': 1})})\n",
      "defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70598>, {'mir': Counter({'POS': 1}), 'gene': Counter({'NEG': 1})})\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "full sentence Our data indicate that overexpression of <e1>miR-21</e1> protects against ischemic neuronal death, and that downregulation of <e2>FASLG</e2>, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Our data indicate that overexpression of <e1>miR-21</e1> protects against ischemic neuronal death, and that downregulation of <e2>FASLG</e2>, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.', 'full_sentence': 'Our data indicate that overexpression of <e1>miR-21</e1> protects against ischemic neuronal death, and that downregulation of <e2>FASLG</e2>, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (41, 47), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (117, 122), 'entity_text': 'FASLG'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d54.s9.p1\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case miR-21\n",
      "add special case tumor necrosis factor- \n",
      "miR-21 6 factor- 22\n",
      "(u\"Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\", 6, 22)\n",
      "geneword factor- mirword miR-21\n",
      "Conjuncts [mediates, protects]\n",
      "neighbour mediates the\n",
      "neighbour mediates the\n",
      "Conjunction before fc protects\n",
      "fc add children for protects\n",
      "Conjuncts [ligand, FASLG]\n",
      "neighbour ligand whose\n",
      "neighbour ligand whose\n",
      "Conjunction before fc FASLG\n",
      "fc add children for FASLG\n",
      "fc add children for FASLG\n",
      "Conjuncts [FASLG, ligand]\n",
      "neighbour FASLG ,\n",
      "neighbour FASLG a\n",
      "Conjunction before fc ligand\n",
      "fc add children for ligand\n",
      "fc add children for ligand\n",
      "fc add children for ligand\n",
      "Conjuncts [protects, mediates]\n",
      "neighbour protects against\n",
      "neighbour protects against\n",
      "Conjunction before fc mediates\n",
      "fc add children for mediates\n",
      "Conunct merge check: protects FASLG\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( conj )\n",
      "protects [protects, against, ischemic, neuronal, death, mediates, neuroprotective, effect]\n",
      "FASLG [of, FASLG, a, tumor, necrosis, factor-, family, member, important, cell, death-inducing, ligand]\n",
      "Conjuncts [overexpression, miR-21]\n",
      "neighbour overexpression of\n",
      "neighbour overexpression of\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conjuncts [protects, death]\n",
      "neighbour protects against\n",
      "neighbour protects against\n",
      "Conjunction before fc death\n",
      "fc add children for death\n",
      "fc add children for death\n",
      "fc add children for death\n",
      "Conjuncts [downregulation, FASLG]\n",
      "neighbour downregulation of\n",
      "neighbour downregulation of\n",
      "Conjunction before fc FASLG\n",
      "fc add children for FASLG\n",
      "fc add children for FASLG\n",
      "Conjuncts [targeted, miR-21]\n",
      "neighbour targeted by\n",
      "neighbour targeted by\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conunct merge check: miR-21 death\n",
      "miR-21\n",
      "Conunct merge check: death FASLG\n",
      "death, and that\n",
      "Double Conjunction Found\n",
      "Conunct merge check: FASLG miR-21\n",
      "member and an important cell death-inducing ligand whose gene is\n",
      "Conjunctions ( nmod )\n",
      "miR-21 [overexpression, of, miR-21]\n",
      "death [protects, against, ischemic, neuronal, death]\n",
      "FASLG [downregulation, of, FASLG, a, tumor, necrosis, factor-, family, member]\n",
      "miR-21 [targeted, by, miR-21]\n",
      "(FASLG, 'conj') [protects, against, ischemic, neuronal, death, downregulation, of, FASLG, a, tumor, necrosis, factor-, family, member]\n",
      "SDP sent: Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "SDP     : [(miR-21, 'NOUN', 'nmod', False), (overexpression, 'NOUN', 'nsubj', False), (protects, 'VERB', 'ccomp', True), (mediates, 'VERB', 'conj', False), (downregulation, 'NOUN', 'nsubj', False), (FASLG, 'NOUN', 'nmod', False), (member, 'NOUN', 'appos', False), (factor-, 'NOUN', 'amod', False)]\n",
      "deepcheck connection rule noun found protects overexpression [that, overexpression, of, miR-21, protects, against, ischemic, neuronal, death, ,, and, that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "DeepCheck Conj Subtree\n",
      "[an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "deepcheck connection rule noun found mediates downregulation [that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "0 [Our, data, indicate] 3 0 Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "1 [overexpression, of, miR-21, protects, against, ischemic, neuronal, death] 8 23 overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "2 [downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect] 28 99 downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-21]\n",
      "deepcheck connection rule noun found protects overexpression [that, overexpression, of, miR-21, protects, against, ischemic, neuronal, death, ,, and, that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "DeepCheck Conj Subtree\n",
      "[against, ischemic, neuronal, death]\n",
      "DeepCheck Conj Subtree\n",
      "[of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[by, miR-21]\n",
      "deepcheck connection rule noun found mediates downregulation [that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "0 [Our, data, indicate] 3 0 Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "1 [overexpression, of, miR-21, protects, against, ischemic, neuronal, death] 8 23 overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "2 [downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect] 28 99 downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "Context Fail <regex.Match object; span=(7, 21), match=' family member', fuzzy_counts=(0, 1, 0)> regex.Regex('(family member){e<=1}', flags=regex.V0)\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter({'POS': 1, 'NEG': 1})\n",
      "Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "TextBetween: overexpression of miR-21 protects against ischemic neuronal death , and that downregulation of FASLG , a tumor necrosis factor- family member\n",
      "infer direction from token set [overexpression, of, miR-21, protects, against, ischemic, neuronal, death, ,, and, that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member]\n",
      "stemTypes2Count: Counter({'POS': 1, 'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70b70>, {'mir': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'POS': 1, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8ce18>, {'mir': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count [(overexpression, 'POS'), (downregulation, 'NEG')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "full sentence Our data indicate that overexpression of <e1>miR-21</e1> protects against ischemic neuronal death, and that downregulation of FASLG, a <e2>tumor necrosis factor- </e2>family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Our data indicate that overexpression of <e1>miR-21</e1> protects against ischemic neuronal death, and that downregulation of FASLG, a <e2>tumor necrosis factor- </e2>family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.', 'full_sentence': 'Our data indicate that overexpression of <e1>miR-21</e1> protects against ischemic neuronal death, and that downregulation of FASLG, a <e2>tumor necrosis factor- </e2>family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (41, 47), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (126, 150), 'entity_text': 'tumor necrosis factor- '}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d54.s9.p1\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('NA', 'NA', 'FALSE', 'FALSE')\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Our data indicate that overexpression of <e1>miR-21</e1> protects against ischemic neuronal death, and that downregulation of FASLG, a <e2>tumor necrosis factor- </e2>family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.', 'full_sentence': 'Our data indicate that overexpression of <e1>miR-21</e1> protects against ischemic neuronal death, and that downregulation of FASLG, a <e2>tumor necrosis factor- </e2>family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (41, 47), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (126, 150), 'entity_text': 'tumor necrosis factor- '}}\n",
      "add special case miR-21\n",
      "add special case tumor necrosis factor- \n",
      "miR-21 6 factor- 22\n",
      "(u\"Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\", 6, 22)\n",
      "geneword factor- mirword miR-21\n",
      "Conjuncts [mediates, protects]\n",
      "neighbour mediates the\n",
      "neighbour mediates the\n",
      "Conjunction before fc protects\n",
      "fc add children for protects\n",
      "Conjuncts [ligand, FASLG]\n",
      "neighbour ligand whose\n",
      "neighbour ligand whose\n",
      "Conjunction before fc FASLG\n",
      "fc add children for FASLG\n",
      "fc add children for FASLG\n",
      "Conjuncts [FASLG, ligand]\n",
      "neighbour FASLG ,\n",
      "neighbour FASLG a\n",
      "Conjunction before fc ligand\n",
      "fc add children for ligand\n",
      "fc add children for ligand\n",
      "fc add children for ligand\n",
      "Conjuncts [protects, mediates]\n",
      "neighbour protects against\n",
      "neighbour protects against\n",
      "Conjunction before fc mediates\n",
      "fc add children for mediates\n",
      "Conunct merge check: protects FASLG\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( conj )\n",
      "protects [protects, against, ischemic, neuronal, death, mediates, neuroprotective, effect]\n",
      "FASLG [of, FASLG, a, tumor, necrosis, factor-, family, member, important, cell, death-inducing, ligand]\n",
      "Conjuncts [overexpression, miR-21]\n",
      "neighbour overexpression of\n",
      "neighbour overexpression of\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conjuncts [protects, death]\n",
      "neighbour protects against\n",
      "neighbour protects against\n",
      "Conjunction before fc death\n",
      "fc add children for death\n",
      "fc add children for death\n",
      "fc add children for death\n",
      "Conjuncts [downregulation, FASLG]\n",
      "neighbour downregulation of\n",
      "neighbour downregulation of\n",
      "Conjunction before fc FASLG\n",
      "fc add children for FASLG\n",
      "fc add children for FASLG\n",
      "Conjuncts [targeted, miR-21]\n",
      "neighbour targeted by\n",
      "neighbour targeted by\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conunct merge check: miR-21 death\n",
      "miR-21\n",
      "Conunct merge check: death FASLG\n",
      "death, and that\n",
      "Double Conjunction Found\n",
      "Conunct merge check: FASLG miR-21\n",
      "member and an important cell death-inducing ligand whose gene is\n",
      "Conjunctions ( nmod )\n",
      "miR-21 [overexpression, of, miR-21]\n",
      "death [protects, against, ischemic, neuronal, death]\n",
      "FASLG [downregulation, of, FASLG, a, tumor, necrosis, factor-, family, member]\n",
      "miR-21 [targeted, by, miR-21]\n",
      "(FASLG, 'conj') [protects, against, ischemic, neuronal, death, downregulation, of, FASLG, a, tumor, necrosis, factor-, family, member]\n",
      "SDP sent: Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "SDP     : [(miR-21, 'NOUN', 'nmod', False), (overexpression, 'NOUN', 'nsubj', False), (protects, 'VERB', 'ccomp', True), (mediates, 'VERB', 'conj', False), (downregulation, 'NOUN', 'nsubj', False), (FASLG, 'NOUN', 'nmod', False), (member, 'NOUN', 'appos', False), (factor-, 'NOUN', 'amod', False)]\n",
      "deepcheck connection rule noun found protects overexpression [that, overexpression, of, miR-21, protects, against, ischemic, neuronal, death, ,, and, that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "DeepCheck Conj Subtree\n",
      "[an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "deepcheck connection rule noun found mediates downregulation [that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "0 [Our, data, indicate] 3 0 Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "1 [overexpression, of, miR-21, protects, against, ischemic, neuronal, death] 8 23 overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "2 [downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect] 28 99 downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-21]\n",
      "deepcheck connection rule noun found protects overexpression [that, overexpression, of, miR-21, protects, against, ischemic, neuronal, death, ,, and, that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "DeepCheck Conj Subtree\n",
      "[against, ischemic, neuronal, death]\n",
      "DeepCheck Conj Subtree\n",
      "[of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[by, miR-21]\n",
      "deepcheck connection rule noun found mediates downregulation [that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "0 [Our, data, indicate] 3 0 Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "1 [overexpression, of, miR-21, protects, against, ischemic, neuronal, death] 8 23 overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "2 [downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect] 28 99 downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "Context Fail <regex.Match object; span=(7, 21), match=' family member', fuzzy_counts=(0, 1, 0)> regex.Regex('(family member){e<=1}', flags=regex.V0)\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter({'POS': 1, 'NEG': 1})\n",
      "Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "TextBetween: overexpression of miR-21 protects against ischemic neuronal death , and that downregulation of FASLG , a tumor necrosis factor- family member\n",
      "infer direction from token set [overexpression, of, miR-21, protects, against, ischemic, neuronal, death, ,, and, that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member]\n",
      "stemTypes2Count: Counter({'POS': 1, 'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c510>, {'mir': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'POS': 1, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70ae8>, {'mir': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count [(overexpression, 'POS'), (downregulation, 'NEG')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "full sentence Our data indicate that overexpression of <e1>miR-21</e1> protects against ischemic neuronal death, and that downregulation of FASLG, a <e2>tumor necrosis factor- </e2>family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Our data indicate that overexpression of <e1>miR-21</e1> protects against ischemic neuronal death, and that downregulation of FASLG, a <e2>tumor necrosis factor- </e2>family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.', 'full_sentence': 'Our data indicate that overexpression of <e1>miR-21</e1> protects against ischemic neuronal death, and that downregulation of FASLG, a <e2>tumor necrosis factor- </e2>family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (41, 47), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (126, 150), 'entity_text': 'tumor necrosis factor- '}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d54.s9.p2\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case FASLG\n",
      "add special case miR-21\n",
      "FASLG 17 miR-21 36\n",
      "(u\"Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\", 17, 36)\n",
      "geneword FASLG mirword miR-21\n",
      "Conjuncts [mediates, protects]\n",
      "neighbour mediates the\n",
      "neighbour mediates the\n",
      "Conjunction before fc protects\n",
      "fc add children for protects\n",
      "Conjuncts [ligand, FASLG]\n",
      "neighbour ligand whose\n",
      "neighbour ligand whose\n",
      "Conjunction before fc FASLG\n",
      "fc add children for FASLG\n",
      "fc add children for FASLG\n",
      "Conjuncts [FASLG, ligand]\n",
      "neighbour FASLG ,\n",
      "neighbour FASLG a\n",
      "Conjunction before fc ligand\n",
      "fc add children for ligand\n",
      "fc add children for ligand\n",
      "fc add children for ligand\n",
      "Conjuncts [protects, mediates]\n",
      "neighbour protects against\n",
      "neighbour protects against\n",
      "Conjunction before fc mediates\n",
      "fc add children for mediates\n",
      "Conunct merge check: protects FASLG\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( conj )\n",
      "protects [protects, against, ischemic, neuronal, death, mediates, neuroprotective, effect]\n",
      "FASLG [of, FASLG, a, tumor, necrosis, factor-, family, member, important, cell, death-inducing, ligand]\n",
      "Conjuncts [overexpression, miR-21]\n",
      "neighbour overexpression of\n",
      "neighbour overexpression of\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conjuncts [protects, death]\n",
      "neighbour protects against\n",
      "neighbour protects against\n",
      "Conjunction before fc death\n",
      "fc add children for death\n",
      "fc add children for death\n",
      "fc add children for death\n",
      "Conjuncts [downregulation, FASLG]\n",
      "neighbour downregulation of\n",
      "neighbour downregulation of\n",
      "Conjunction before fc FASLG\n",
      "fc add children for FASLG\n",
      "fc add children for FASLG\n",
      "Conjuncts [targeted, miR-21]\n",
      "neighbour targeted by\n",
      "neighbour targeted by\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conunct merge check: miR-21 death\n",
      "miR-21\n",
      "Conunct merge check: death FASLG\n",
      "death, and that\n",
      "Double Conjunction Found\n",
      "Conunct merge check: FASLG miR-21\n",
      "member and an important cell death-inducing ligand whose gene is\n",
      "Conjunctions ( nmod )\n",
      "miR-21 [overexpression, of, miR-21]\n",
      "death [protects, against, ischemic, neuronal, death]\n",
      "FASLG [downregulation, of, FASLG, a, tumor, necrosis, factor-, family, member]\n",
      "miR-21 [targeted, by, miR-21]\n",
      "(FASLG, 'conj') [protects, against, ischemic, neuronal, death, downregulation, of, FASLG, a, tumor, necrosis, factor-, family, member]\n",
      "SDP sent: Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "SDP     : [(FASLG, 'NOUN', 'nmod', False), (indicate, 'VERB', 'ROOT', False), (protects, 'VERB', 'ccomp', True), (mediates, 'VERB', 'conj', False), (downregulation, 'NOUN', 'nsubj', False), (FASLG, 'NOUN', 'nmod', False)]\n",
      "deepcheck connection rule noun found protects overexpression [that, overexpression, of, miR-21, protects, against, ischemic, neuronal, death, ,, and, that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "DeepCheck Conj Subtree\n",
      "[an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "deepcheck connection rule noun found mediates downregulation [that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "0 [Our, data, indicate] 3 0 Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "1 [overexpression, of, miR-21, protects, against, ischemic, neuronal, death] 8 23 overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "2 [downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect] 28 99 downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-21]\n",
      "deepcheck connection rule noun found protects overexpression [that, overexpression, of, miR-21, protects, against, ischemic, neuronal, death, ,, and, that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "DeepCheck Conj Subtree\n",
      "[against, ischemic, neuronal, death]\n",
      "DeepCheck Conj Subtree\n",
      "[of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[by, miR-21]\n",
      "deepcheck connection rule noun found mediates downregulation [that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "0 [Our, data, indicate] 3 0 Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "1 [overexpression, of, miR-21, protects, against, ischemic, neuronal, death] 8 23 overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "2 [downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect] 28 99 downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "Context Fail <regex.Match object; span=(7, 13), match=' tumor', fuzzy_counts=(0, 1, 0)> regex.Regex('(tumor){e<=1}', flags=regex.V0)\n",
      "Recognized overlapping entities: {Interval(117, 122, ('GENE_OR_GENE_PRODUCT', 'FASLG'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text downregulation of FASLG , a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21 , probably mediates the neuroprotective effect\n",
      "[downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "TextBetween: downregulation of FASLG , a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21 , probably mediates the neuroprotective effect\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between gene mir', 'reg_detect_major': 'between'}})\n",
      " hit sentence Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "full sentence Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of <e2>FASLG</e2>, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by <e1>miR-21</e1>, probably mediates the neuroprotective effect.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between gene mir', 'reg_detect_major': 'between'}}, 'tagged_sent': 'Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of <e2>FASLG</e2>, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by <e1>miR-21</e1>, probably mediates the neuroprotective effect.', 'full_sentence': 'Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of <e2>FASLG</e2>, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by <e1>miR-21</e1>, probably mediates the neuroprotective effect.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (117, 122), 'entity_text': 'FASLG'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (234, 240), 'entity_text': 'miR-21'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d54.s9.p3\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case tumor necrosis factor- \n",
      "add special case miR-21\n",
      "factor- 22 miR-21 36\n",
      "(u\"Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\", 22, 36)\n",
      "geneword factor- mirword miR-21\n",
      "Conjuncts [mediates, protects]\n",
      "neighbour mediates the\n",
      "neighbour mediates the\n",
      "Conjunction before fc protects\n",
      "fc add children for protects\n",
      "Conjuncts [ligand, FASLG]\n",
      "neighbour ligand whose\n",
      "neighbour ligand whose\n",
      "Conjunction before fc FASLG\n",
      "fc add children for FASLG\n",
      "fc add children for FASLG\n",
      "Conjuncts [FASLG, ligand]\n",
      "neighbour FASLG ,\n",
      "neighbour FASLG a\n",
      "Conjunction before fc ligand\n",
      "fc add children for ligand\n",
      "fc add children for ligand\n",
      "fc add children for ligand\n",
      "Conjuncts [protects, mediates]\n",
      "neighbour protects against\n",
      "neighbour protects against\n",
      "Conjunction before fc mediates\n",
      "fc add children for mediates\n",
      "Conunct merge check: protects FASLG\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( conj )\n",
      "protects [protects, against, ischemic, neuronal, death, mediates, neuroprotective, effect]\n",
      "FASLG [of, FASLG, a, tumor, necrosis, factor-, family, member, important, cell, death-inducing, ligand]\n",
      "Conjuncts [overexpression, miR-21]\n",
      "neighbour overexpression of\n",
      "neighbour overexpression of\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conjuncts [protects, death]\n",
      "neighbour protects against\n",
      "neighbour protects against\n",
      "Conjunction before fc death\n",
      "fc add children for death\n",
      "fc add children for death\n",
      "fc add children for death\n",
      "Conjuncts [downregulation, FASLG]\n",
      "neighbour downregulation of\n",
      "neighbour downregulation of\n",
      "Conjunction before fc FASLG\n",
      "fc add children for FASLG\n",
      "fc add children for FASLG\n",
      "Conjuncts [targeted, miR-21]\n",
      "neighbour targeted by\n",
      "neighbour targeted by\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conunct merge check: miR-21 death\n",
      "miR-21\n",
      "Conunct merge check: death FASLG\n",
      "death, and that\n",
      "Double Conjunction Found\n",
      "Conunct merge check: FASLG miR-21\n",
      "member and an important cell death-inducing ligand whose gene is\n",
      "Conjunctions ( nmod )\n",
      "miR-21 [overexpression, of, miR-21]\n",
      "death [protects, against, ischemic, neuronal, death]\n",
      "FASLG [downregulation, of, FASLG, a, tumor, necrosis, factor-, family, member]\n",
      "miR-21 [targeted, by, miR-21]\n",
      "(FASLG, 'conj') [protects, against, ischemic, neuronal, death, downregulation, of, FASLG, a, tumor, necrosis, factor-, family, member]\n",
      "SDP sent: Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "SDP     : [(factor-, 'NOUN', 'amod', False), (member, 'NOUN', 'appos', False), (FASLG, 'NOUN', 'nmod', True), (ligand, 'NOUN', 'conj', False), (targeted, 'VERB', 'acl:relcl', False), (miR-21, 'NOUN', 'nmod', False)]\n",
      "deepcheck connection rule noun found protects overexpression [that, overexpression, of, miR-21, protects, against, ischemic, neuronal, death, ,, and, that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "DeepCheck Conj Subtree\n",
      "[an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "deepcheck connection rule noun found mediates downregulation [that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "0 [Our, data, indicate] 3 0 Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "1 [overexpression, of, miR-21, protects, against, ischemic, neuronal, death] 8 23 overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "2 [downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect] 28 99 downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-21]\n",
      "deepcheck connection rule noun found protects overexpression [that, overexpression, of, miR-21, protects, against, ischemic, neuronal, death, ,, and, that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "DeepCheck Conj Subtree\n",
      "[against, ischemic, neuronal, death]\n",
      "DeepCheck Conj Subtree\n",
      "[of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[by, miR-21]\n",
      "deepcheck connection rule noun found mediates downregulation [that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "0 [Our, data, indicate] 3 0 Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "1 [overexpression, of, miR-21, protects, against, ischemic, neuronal, death] 8 23 overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "2 [downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect] 28 99 downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "Context Fail <regex.Match object; span=(7, 21), match=' family member', fuzzy_counts=(0, 1, 0)> regex.Regex('(family member){e<=1}', flags=regex.V0)\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text downregulation of FASLG , a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21 , probably mediates the neuroprotective effect\n",
      "[downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "TextBetween: downregulation of FASLG , a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21 , probably mediates the neuroprotective effect\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between gene mir', 'reg_detect_major': 'between'}})\n",
      " hit sentence Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "full sentence Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a <e2>tumor necrosis factor- </e2>family member and an important cell death-inducing ligand whose gene is targeted by <e1>miR-21</e1>, probably mediates the neuroprotective effect.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between gene mir', 'reg_detect_major': 'between'}}, 'tagged_sent': 'Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a <e2>tumor necrosis factor- </e2>family member and an important cell death-inducing ligand whose gene is targeted by <e1>miR-21</e1>, probably mediates the neuroprotective effect.', 'full_sentence': 'Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a <e2>tumor necrosis factor- </e2>family member and an important cell death-inducing ligand whose gene is targeted by <e1>miR-21</e1>, probably mediates the neuroprotective effect.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (126, 150), 'entity_text': 'tumor necrosis factor- '}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (234, 240), 'entity_text': 'miR-21'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d54.s9.p3\n",
      "IS:  ('MIR_GENE', 'NEU')\n",
      "SB:  ('NA', 'NA', 'FALSE', 'FALSE')\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between gene mir', 'reg_detect_major': 'between'}}, 'tagged_sent': 'Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a <e2>tumor necrosis factor- </e2>family member and an important cell death-inducing ligand whose gene is targeted by <e1>miR-21</e1>, probably mediates the neuroprotective effect.', 'full_sentence': 'Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a <e2>tumor necrosis factor- </e2>family member and an important cell death-inducing ligand whose gene is targeted by <e1>miR-21</e1>, probably mediates the neuroprotective effect.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (126, 150), 'entity_text': 'tumor necrosis factor- '}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (234, 240), 'entity_text': 'miR-21'}}\n",
      "add special case tumor necrosis factor- \n",
      "add special case miR-21\n",
      "factor- 22 miR-21 36\n",
      "(u\"Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\", 22, 36)\n",
      "geneword factor- mirword miR-21\n",
      "Conjuncts [mediates, protects]\n",
      "neighbour mediates the\n",
      "neighbour mediates the\n",
      "Conjunction before fc protects\n",
      "fc add children for protects\n",
      "Conjuncts [ligand, FASLG]\n",
      "neighbour ligand whose\n",
      "neighbour ligand whose\n",
      "Conjunction before fc FASLG\n",
      "fc add children for FASLG\n",
      "fc add children for FASLG\n",
      "Conjuncts [FASLG, ligand]\n",
      "neighbour FASLG ,\n",
      "neighbour FASLG a\n",
      "Conjunction before fc ligand\n",
      "fc add children for ligand\n",
      "fc add children for ligand\n",
      "fc add children for ligand\n",
      "Conjuncts [protects, mediates]\n",
      "neighbour protects against\n",
      "neighbour protects against\n",
      "Conjunction before fc mediates\n",
      "fc add children for mediates\n",
      "Conunct merge check: protects FASLG\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( conj )\n",
      "protects [protects, against, ischemic, neuronal, death, mediates, neuroprotective, effect]\n",
      "FASLG [of, FASLG, a, tumor, necrosis, factor-, family, member, important, cell, death-inducing, ligand]\n",
      "Conjuncts [overexpression, miR-21]\n",
      "neighbour overexpression of\n",
      "neighbour overexpression of\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conjuncts [protects, death]\n",
      "neighbour protects against\n",
      "neighbour protects against\n",
      "Conjunction before fc death\n",
      "fc add children for death\n",
      "fc add children for death\n",
      "fc add children for death\n",
      "Conjuncts [downregulation, FASLG]\n",
      "neighbour downregulation of\n",
      "neighbour downregulation of\n",
      "Conjunction before fc FASLG\n",
      "fc add children for FASLG\n",
      "fc add children for FASLG\n",
      "Conjuncts [targeted, miR-21]\n",
      "neighbour targeted by\n",
      "neighbour targeted by\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conunct merge check: miR-21 death\n",
      "miR-21\n",
      "Conunct merge check: death FASLG\n",
      "death, and that\n",
      "Double Conjunction Found\n",
      "Conunct merge check: FASLG miR-21\n",
      "member and an important cell death-inducing ligand whose gene is\n",
      "Conjunctions ( nmod )\n",
      "miR-21 [overexpression, of, miR-21]\n",
      "death [protects, against, ischemic, neuronal, death]\n",
      "FASLG [downregulation, of, FASLG, a, tumor, necrosis, factor-, family, member]\n",
      "miR-21 [targeted, by, miR-21]\n",
      "(FASLG, 'conj') [protects, against, ischemic, neuronal, death, downregulation, of, FASLG, a, tumor, necrosis, factor-, family, member]\n",
      "SDP sent: Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "SDP     : [(factor-, 'NOUN', 'amod', False), (member, 'NOUN', 'appos', False), (FASLG, 'NOUN', 'nmod', True), (ligand, 'NOUN', 'conj', False), (targeted, 'VERB', 'acl:relcl', False), (miR-21, 'NOUN', 'nmod', False)]\n",
      "deepcheck connection rule noun found protects overexpression [that, overexpression, of, miR-21, protects, against, ischemic, neuronal, death, ,, and, that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "DeepCheck Conj Subtree\n",
      "[an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "deepcheck connection rule noun found mediates downregulation [that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "0 [Our, data, indicate] 3 0 Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "1 [overexpression, of, miR-21, protects, against, ischemic, neuronal, death] 8 23 overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "2 [downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect] 28 99 downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-21]\n",
      "deepcheck connection rule noun found protects overexpression [that, overexpression, of, miR-21, protects, against, ischemic, neuronal, death, ,, and, that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "DeepCheck Conj Subtree\n",
      "[against, ischemic, neuronal, death]\n",
      "DeepCheck Conj Subtree\n",
      "[of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[by, miR-21]\n",
      "deepcheck connection rule noun found mediates downregulation [that, downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "0 [Our, data, indicate] 3 0 Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "1 [overexpression, of, miR-21, protects, against, ischemic, neuronal, death] 8 23 overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "2 [downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect] 28 99 downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "Context Fail <regex.Match object; span=(7, 21), match=' family member', fuzzy_counts=(0, 1, 0)> regex.Regex('(family member){e<=1}', flags=regex.V0)\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text downregulation of FASLG , a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21 , probably mediates the neuroprotective effect\n",
      "[downregulation, of, FASLG, ,, a, tumor, necrosis, factor-, family, member, and, an, important, cell, death-inducing, ligand, whose, gene, is, targeted, by, miR-21, ,, probably, mediates, the, neuroprotective, effect]\n",
      "TextBetween: downregulation of FASLG , a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21 , probably mediates the neuroprotective effect\n",
      "(False, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between gene mir', 'reg_detect_major': 'between'}})\n",
      " hit sentence Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a tumor necrosis factor- family member and an important cell death-inducing ligand whose gene is targeted by miR-21, probably mediates the neuroprotective effect.\n",
      "full sentence Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a <e2>tumor necrosis factor- </e2>family member and an important cell death-inducing ligand whose gene is targeted by <e1>miR-21</e1>, probably mediates the neuroprotective effect.\n",
      "{'accept_relation': False, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': False, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between gene mir', 'reg_detect_major': 'between'}}, 'tagged_sent': 'Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a <e2>tumor necrosis factor- </e2>family member and an important cell death-inducing ligand whose gene is targeted by <e1>miR-21</e1>, probably mediates the neuroprotective effect.', 'full_sentence': 'Our data indicate that overexpression of miR-21 protects against ischemic neuronal death, and that downregulation of FASLG, a <e2>tumor necrosis factor- </e2>family member and an important cell death-inducing ligand whose gene is targeted by <e1>miR-21</e1>, probably mediates the neuroprotective effect.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (126, 150), 'entity_text': 'tumor necrosis factor- '}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (234, 240), 'entity_text': 'miR-21'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d54.s10.p2\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miR-21\n",
      "add special case Faslg\n",
      "miR-21 16 Faslg 33\n",
      "(u\"Moreover, overexpression of miR-21 in neurons significantly reduced FASLG levels, and introduction of an miR-21 mimic into 293-HEK cells substantially reduced luciferase activity in a reporter system containing the 3'-UTR of Faslg.\", 16, 33)\n",
      "geneword Faslg mirword miR-21\n",
      "Conjuncts [reduced, reduced]\n",
      "neighbour reduced luciferase\n",
      "neighbour reduced luciferase\n",
      "Conjunction before fc reduced\n",
      "fc add children for reduced\n",
      "Conjuncts [reduced, reduced]\n",
      "neighbour reduced FASLG\n",
      "neighbour reduced FASLG\n",
      "Conjunction before fc reduced\n",
      "fc add children for reduced\n",
      "fc add children for reduced\n",
      "Conjunctions ( conj )\n",
      "reduced [reduced, FASLG, levels, reduced, luciferase, activity, in, reporter, system, containing, 3'-UTR, of, Faslg]\n",
      "Conjuncts [overexpression, miR-21]\n",
      "neighbour overexpression of\n",
      "neighbour overexpression of\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "Conjuncts [overexpression, neurons]\n",
      "neighbour overexpression of\n",
      "neighbour overexpression of\n",
      "Conjunction before fc neurons\n",
      "fc add children for neurons\n",
      "Conjuncts [introduction, miR-21]\n",
      "neighbour introduction of\n",
      "neighbour introduction of\n",
      "Conjunction before fc miR-21\n",
      "fc add children for miR-21\n",
      "fc add children for miR-21\n",
      "Conjuncts [mimic, cells]\n",
      "neighbour mimic into\n",
      "neighbour mimic into\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conjuncts [reduced, system]\n",
      "neighbour reduced luciferase\n",
      "neighbour reduced luciferase\n",
      "Conjunction before fc system\n",
      "fc add children for system\n",
      "fc add children for system\n",
      "fc add children for system\n",
      "Conjuncts [3'-UTR, Faslg]\n",
      "neighbour 3'-UTR of\n",
      "neighbour 3'-UTR of\n",
      "Conjunction before fc Faslg\n",
      "fc add children for Faslg\n",
      "Conunct merge check: miR-21 miR-21\n",
      "neurons significantly reduced FASLG levels, and\n",
      "Conunct merge check: miR-21 system\n",
      "cells substantially\n",
      "Conjunctions ( nmod )\n",
      "miR-21 [overexpression, of, miR-21, in, neurons]\n",
      "miR-21 [introduction, of, miR-21, mimic, into, 293-HEK, cells]\n",
      "system [reduced, in, reporter, system, containing, 3'-UTR, of, Faslg]\n",
      "SDP sent: Moreover, overexpression of miR-21 in neurons significantly reduced FASLG levels, and introduction of an miR-21 mimic into 293-HEK cells substantially reduced luciferase activity in a reporter system containing the 3'-UTR of Faslg.\n",
      "SDP     : [(miR-21, 'NOUN', 'nmod', False), (introduction, 'NOUN', 'nsubj', False), (reduced, 'VERB', 'conj', False), (system, 'NOUN', 'nmod', False), (containing, 'VERB', 'acl', False), (3'-UTR, 'NOUN', 'dobj', False), (Faslg, 'NOUN', 'nmod', False)]\n",
      "deepcheck connection rule noun found reduced introduction [introduction, of, an, miR-21, mimic, into, 293-HEK, cells, substantially, reduced, luciferase, activity, in, a, reporter, system, containing, the, 3'-UTR, of, Faslg]\n",
      "0 [Moreover, ,, overexpression, of, miR-21, in, neurons, significantly, reduced, FASLG, levels] 11 0 Moreover, overexpression of miR-21 in neurons significantly reduced FASLG levels, and introduction of an miR-21 mimic into 293-HEK cells substantially reduced luciferase activity in a reporter system containing the 3'-UTR of Faslg.\n",
      "1 [introduction, of, an, miR-21, mimic, into, 293-HEK, cells, substantially, reduced, luciferase, activity, in, a, reporter, system, containing, the, 3'-UTR, of, Faslg] 21 86 introduction of an miR-21 mimic into 293-HEK cells substantially reduced luciferase activity in a reporter system containing the 3'-UTR of Faslg.\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-21]\n",
      "DeepCheck Conj Subtree\n",
      "[in, neurons]\n",
      "DeepCheck Conj Subtree\n",
      "[of, an, miR-21, mimic, into, 293-HEK, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[into, 293-HEK, cells]\n",
      "deepcheck connection rule noun found reduced introduction [introduction, of, an, miR-21, mimic, into, 293-HEK, cells, substantially, reduced, luciferase, activity, in, a, reporter, system, containing, the, 3'-UTR, of, Faslg]\n",
      "DeepCheck Conj Subtree\n",
      "[in, a, reporter, system, containing, the, 3'-UTR, of, Faslg]\n",
      "DeepCheck Conj Subtree\n",
      "[of, Faslg]\n",
      "0 [Moreover, ,, overexpression, of, miR-21, in, neurons, significantly, reduced, FASLG, levels] 11 0 Moreover, overexpression of miR-21 in neurons significantly reduced FASLG levels, and introduction of an miR-21 mimic into 293-HEK cells substantially reduced luciferase activity in a reporter system containing the 3'-UTR of Faslg.\n",
      "1 [introduction, of, an, miR-21, mimic, into, 293-HEK, cells, substantially, reduced, luciferase, activity, in, a, reporter, system, containing, the, 3'-UTR, of, Faslg] 21 86 introduction of an miR-21 mimic into 293-HEK cells substantially reduced luciferase activity in a reporter system containing the 3'-UTR of Faslg.\n",
      "Recognized overlapping entities: {Interval(225, 230, ('SIMPLE_CHEMICAL', 'Faslg'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text introduction of an miR-21 mimic into 293-HEK cells substantially reduced luciferase activity in a reporter system containing the 3'-UTR of Faslg\n",
      "[introduction, of, an, miR-21, mimic, into, 293-HEK, cells, substantially, reduced, luciferase, activity, in, a, reporter, system, containing, the, 3'-UTR, of, Faslg]\n",
      "TextBetween: introduction of an miR-21 mimic into 293-HEK cells substantially reduced luciferase activity in a reporter system containing the 3'-UTR of Faslg\n",
      "infer direction from token set [introduction, of, an, miR-21, mimic, into, 293-HEK, cells, substantially, reduced, luciferase, activity, in, a, reporter, system, containing, the, 3'-UTR, of, Faslg]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70950>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62ea0>, {})\n",
      "counts between 1 stemTypes2Count [(reduced, 'NEG'), (activity, 'NEU'), (containing, 'NEU')]\n",
      "Between Tokens [(reduced, 'NEG'), (activity, 'NEU'), (containing, 'NEU')]\n",
      "counts between final False\n",
      "['of', 'an', 'miR-21', 'mimic', 'into', '293-HEK', 'cells', 'substantially', 'reduced', 'luciferase', 'activity', 'in', 'a', 'reporter', 'system', 'containing', 'the', \"3'-UTR\", 'of', 'Faslg']\n",
      "Counter({'NEU': 2, 'NEG': 1})\n",
      "[(reduced, 'NEG'), (activity, 'NEU'), (containing, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Moreover, overexpression of miR-21 in neurons significantly reduced FASLG levels, and introduction of an miR-21 mimic into 293-HEK cells substantially reduced luciferase activity in a reporter system containing the 3'-UTR of Faslg.\n",
      "full sentence Moreover, overexpression of miR-21 in neurons significantly reduced FASLG levels, and introduction of an <e1>miR-21</e1> mimic into 293-HEK cells substantially reduced luciferase activity in a reporter system containing the 3'-UTR of <e2>Faslg</e2>.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"Moreover, overexpression of miR-21 in neurons significantly reduced FASLG levels, and introduction of an <e1>miR-21</e1> mimic into 293-HEK cells substantially reduced luciferase activity in a reporter system containing the 3'-UTR of <e2>Faslg</e2>.\", 'full_sentence': \"Moreover, overexpression of miR-21 in neurons significantly reduced FASLG levels, and introduction of an <e1>miR-21</e1> mimic into 293-HEK cells substantially reduced luciferase activity in a reporter system containing the 3'-UTR of <e2>Faslg</e2>.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (105, 111), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (225, 230), 'entity_text': 'Faslg'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d56.s2.p9\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case CD47\n",
      "add special case microRNA-155\n",
      "CD47 20 microRNA-155 26\n",
      "(u\"Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites.\", 20, 26)\n",
      "geneword CD47 mirword microRNA-155\n",
      "Conjuncts [microRNA-155, microRNA-326, microRNA-34a]\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-326 )\n",
      "neighbour microRNA-326 targeted\n",
      "Conjunction before fc microRNA-34a\n",
      "Conjuncts [microRNA-34a, microRNA-326, microRNA-155]\n",
      "neighbour microRNA-34a ,\n",
      "neighbour microRNA-34a microRNA-155\n",
      "neighbour microRNA-326 )\n",
      "neighbour microRNA-326 targeted\n",
      "Conjunction before fc microRNA-155\n",
      "Conjuncts [microRNA-34a, microRNA-155, microRNA-326]\n",
      "neighbour microRNA-34a ,\n",
      "neighbour microRNA-34a microRNA-155\n",
      "neighbour microRNA-155 and\n",
      "neighbour microRNA-155 and\n",
      "Conjunction before fc microRNA-326\n",
      "Conjunctions ( conj )\n",
      "microRNA-34a [microRNA-34a, microRNA-155, microRNA-326]\n",
      "Conjuncts [upregulated, lesions]\n",
      "neighbour upregulated in\n",
      "neighbour upregulated in\n",
      "Conjunction before fc lesions\n",
      "fc add children for lesions\n",
      "fc add children for lesions\n",
      "fc add children for lesions\n",
      "fc add children for lesions\n",
      "fc add children for lesions\n",
      "Conjuncts [region, CD47]\n",
      "neighbour region of\n",
      "neighbour region of\n",
      "Conjunction before fc CD47\n",
      "fc add children for CD47\n",
      "Conjuncts [targeted, assays]\n",
      "neighbour targeted the\n",
      "neighbour targeted the\n",
      "Conjunction before fc assays\n",
      "fc add children for assays\n",
      "fc add children for assays\n",
      "Conjuncts [targeted, microRNA-155]\n",
      "neighbour targeted the\n",
      "neighbour targeted the\n",
      "Conjunction before fc microRNA-155\n",
      "fc add children for microRNA-155\n",
      "Conjuncts [targeted, sites]\n",
      "neighbour targeted the\n",
      "neighbour targeted the\n",
      "Conjunction before fc sites\n",
      "fc add children for sites\n",
      "fc add children for sites\n",
      "fc add children for sites\n",
      "Conunct merge check: lesions assays\n",
      "microRNA-34a, microRNA-155 and microRNA-326)\n",
      "Conunct merge check: assays CD47\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( nmod )\n",
      "lesions [upregulated, in, active, multiple, sclerosis, lesions, microRNA-34a]\n",
      "CD47 [region, of, CD47]\n",
      "assays [targeted, in, reporter, assays, with, microRNA-155, at, two, distinct, sites]\n",
      "SDP sent: Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites.\n",
      "SDP     : [(CD47, 'NOUN', 'nmod', False), (region, 'NOUN', 'dobj', False), (targeted, 'VERB', 'ROOT', False), (microRNA-155, 'NOUN', 'nmod', False)]\n",
      "SDP DOBJ RULE\n",
      "DeepCheck Conj Subtree\n",
      "[microRNA-155]\n",
      "DeepCheck Conj Subtree\n",
      "[microRNA-326]\n",
      "0 [Three, microRNA, upregulated, in, active, multiple, sclerosis, lesions, (, microRNA-34a, ,, microRNA-155, and, microRNA-326, ), targeted, the, 3'-untranslated, region, of, CD47, in, reporter, assays, ,, with, microRNA-155, even, at, two, distinct, sites] 32 0 Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites.\n",
      "DeepCheck Conj Subtree\n",
      "[in, active, multiple, sclerosis, lesions, (, microRNA-34a, ,, microRNA-155, and, microRNA-326, )]\n",
      "DeepCheck Conj Subtree\n",
      "[microRNA-155]\n",
      "DeepCheck Conj Subtree\n",
      "[microRNA-326]\n",
      "DeepCheck Conj Subtree\n",
      "[of, CD47]\n",
      "DeepCheck Conj Subtree\n",
      "[in, reporter, assays]\n",
      "DeepCheck Conj Subtree\n",
      "[with, microRNA-155]\n",
      "DeepCheck Conj Subtree\n",
      "[even, at, two, distinct, sites]\n",
      "0 [Three, microRNA, upregulated, in, active, multiple, sclerosis, lesions, (, microRNA-34a, ,, microRNA-155, and, microRNA-326, ), targeted, the, 3'-untranslated, region, of, CD47, in, reporter, assays, ,, with, microRNA-155, even, at, two, distinct, sites] 32 0 Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites.\n",
      "Recognized overlapping entities: {Interval(149, 153, ('GENE_OR_GENE_PRODUCT', 'CD47'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text Three microRNA upregulated in active multiple sclerosis lesions ( microRNA-34a , microRNA-155 and microRNA-326 ) targeted the 3'-untranslated region of CD47 in reporter assays , with microRNA-155 even at two distinct sites\n",
      "[Three, microRNA, upregulated, in, active, multiple, sclerosis, lesions, (, microRNA-34a, ,, microRNA-155, and, microRNA-326, ), targeted, the, 3'-untranslated, region, of, CD47, in, reporter, assays, ,, with, microRNA-155, even, at, two, distinct, sites]\n",
      "TextBetween: Three microRNA upregulated in active multiple sclerosis lesions ( microRNA-34a , microRNA-155 and microRNA-326 ) targeted the 3'-untranslated region of CD47 in reporter assays , with microRNA-155 even at two distinct sites\n",
      "infer direction from token set [Three, microRNA, upregulated, in, active, multiple, sclerosis, lesions, (, microRNA-34a, ,, microRNA-155, and, microRNA-326, ), targeted, the, 3'-untranslated, region, of, CD47, in, reporter, assays, ,, with, microRNA-155, even, at, two, distinct, sites]\n",
      "stemTypes2Count: Counter({'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbecba9e048>, {})\n",
      "counts between 1 stemTypes2Count Counter()\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c730>, {})\n",
      "counts between 1 stemTypes2Count []\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}})\n",
      " hit sentence Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of CD47 in reporter assays, with microRNA-155 even at two distinct sites.\n",
      "full sentence Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of <e2>CD47</e2> in reporter assays, with <e1>microRNA-155</e1> even at two distinct sites.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}}, 'tagged_sent': \"Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of <e2>CD47</e2> in reporter assays, with <e1>microRNA-155</e1> even at two distinct sites.\", 'full_sentence': \"Three microRNA upregulated in active multiple sclerosis lesions (microRNA-34a, microRNA-155 and microRNA-326) targeted the 3'-untranslated region of <e2>CD47</e2> in reporter assays, with <e1>microRNA-155</e1> even at two distinct sites.\", 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (149, 153), 'entity_text': 'CD47'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (179, 191), 'entity_text': 'microRNA-155'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d57.s2.p4\n",
      "IS: False\n",
      "SB: True\n",
      "Repeat Analysis\n",
      "add special case miR-23a\n",
      "add special case XIAP\n",
      "miR-23a 11 XIAP 18\n",
      "(u\"miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\", 11, 18)\n",
      "geneword XIAP mirword miR-23a\n",
      "Conjuncts [led, bound]\n",
      "neighbour led to\n",
      "neighbour led to\n",
      "Conjunction before fc bound\n",
      "fc add children for bound\n",
      "Conjuncts [miR-23a, UTR]\n",
      "neighbour miR-23a inhibition\n",
      "neighbour miR-23a inhibition\n",
      "Conjunction before fc UTR\n",
      "fc add children for UTR\n",
      "fc add children for UTR\n",
      "Conjuncts [UTR, miR-23a]\n",
      "neighbour UTR of\n",
      "neighbour UTR of\n",
      "Conjunction before fc miR-23a\n",
      "Conjuncts [bound, led]\n",
      "neighbour bound the\n",
      "neighbour bound the\n",
      "Conjunction before fc led\n",
      "fc add children for led\n",
      "Conjunctions ( conj )\n",
      "bound [bound, 3, UTR, of, XIAP, miR-23a, led, to, increase, in, XIAP, mRNA]\n",
      "achildren {miR-23a}\n",
      "achildren False\n",
      "Conjuncts [UTR, XIAP]\n",
      "neighbour UTR of\n",
      "neighbour UTR of\n",
      "Conjunction before fc XIAP\n",
      "fc add children for XIAP\n",
      "Conjuncts [led, increase]\n",
      "neighbour led to\n",
      "neighbour led to\n",
      "Conjunction before fc increase\n",
      "fc add children for increase\n",
      "fc add children for increase\n",
      "Conjuncts [increase, mRNA]\n",
      "neighbour increase in\n",
      "neighbour increase in\n",
      "Conjunction before fc mRNA\n",
      "fc add children for mRNA\n",
      "fc add children for mRNA\n",
      "Conjuncts [target, miR-23a]\n",
      "neighbour target for\n",
      "neighbour target for\n",
      "Conjunction before fc miR-23a\n",
      "fc add children for miR-23a\n",
      "Conunct merge check: XIAP increase\n",
      "XIAP, and miR-23a inhibition\n",
      "Double Conjunction Found\n",
      "Conunct merge check: increase miR-23a\n",
      "mRNA in vitro, demonstrating that XIAP is a previously uncharacterized\n",
      "Conjunctions ( nmod )\n",
      "XIAP [UTR, of, XIAP]\n",
      "increase [led, to, increase, in, XIAP, mRNA]\n",
      "miR-23a [target, for, miR-23a]\n",
      "(increase, 'conj') [UTR, of, XIAP, led, to, increase, in, XIAP, mRNA]\n",
      "SDP sent: miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "SDP     : [(miR-23a, 'NOUN', 'conj', False), (UTR, 'NOUN', 'dobj', True), (bound, 'VERB', 'ROOT', True), (led, 'VERB', 'conj', False), (increase, 'NOUN', 'nmod', False), (mRNA, 'NOUN', 'nmod', False), (XIAP, 'NOUN', 'compound', False)]\n",
      "SDP DOBJ RULE\n",
      "DeepCheck Conj Subtree\n",
      "[miR-23a]\n",
      "deepcheck connection rule noun found led inhibition [inhibition, led, to, an, increase, in, XIAP, mRNA, in, vitro]\n",
      "0 [miR-23a, directly, bound, the, 3, ', UTR, of, XIAP, ,, and, miR-23a] 12 0 miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "1 [inhibition, led, to, an, increase, in, XIAP, mRNA, in, vitro] 10 55 inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "2 [demonstrating, that, XIAP, is, a, previously, uncharacterized, target, for, miR-23a] 10 108 demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "DeepCheck Conj Subtree\n",
      "[of, XIAP]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-23a]\n",
      "deepcheck connection rule noun found led inhibition [inhibition, led, to, an, increase, in, XIAP, mRNA, in, vitro]\n",
      "DeepCheck Conj Subtree\n",
      "[to, an, increase, in, XIAP, mRNA]\n",
      "DeepCheck Conj Subtree\n",
      "[in, XIAP, mRNA]\n",
      "DeepCheck Conj Subtree\n",
      "[for, miR-23a]\n",
      "0 [miR-23a, directly, bound, the, 3, ', UTR, of, XIAP, ,, and, miR-23a] 12 0 miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "1 [inhibition, led, to, an, increase, in, XIAP, mRNA, in, vitro] 10 55 inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "2 [demonstrating, that, XIAP, is, a, previously, uncharacterized, target, for, miR-23a] 10 108 demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "Recognized overlapping entities: {Interval(88, 92, ('GENE_OR_GENE_PRODUCT', 'XIAP'))}\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "TextBetween: , and miR-23a inhibition led to an increase in XIAP mRNA in\n",
      "infer direction from token set [,, and, miR-23a, inhibition, led, to, an, increase, in, XIAP, mRNA, in]\n",
      "stemTypes2Count: Counter({'NEG': 1, 'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c6a8>, {'mir': Counter({'NEG': 1}), 'gene': Counter({'POS': 1})})\n",
      "defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c6a8>, {'mir': Counter({'NEG': 1}), 'gene': Counter({'POS': 1})})\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}})\n",
      " hit sentence miR-23a directly bound the 3' UTR of XIAP, and miR-23a inhibition led to an increase in XIAP mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "full sentence miR-23a directly bound the 3' UTR of XIAP, and <e1>miR-23a</e1> inhibition led to an increase in <e2>XIAP</e2> mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\n",
      "{'accept_relation': False, 'check_results': {'conj': False, 'conj_nmod': True, 'sdp': True, 'compartment': False, 'compartment_nmod': False, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"miR-23a directly bound the 3' UTR of XIAP, and <e1>miR-23a</e1> inhibition led to an increase in <e2>XIAP</e2> mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\", 'full_sentence': \"miR-23a directly bound the 3' UTR of XIAP, and <e1>miR-23a</e1> inhibition led to an increase in <e2>XIAP</e2> mRNA in vitro, demonstrating that XIAP is a previously uncharacterized target for miR-23a.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (47, 54), 'entity_text': 'miR-23a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (88, 92), 'entity_text': 'XIAP'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d66.s2.p0\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'We found that the A allele of rs2507800 suppressed <e2>angiopoietin-1</e2> translation by facilitating <e1>miR-211</e1> binding, but not the T allele.', 'full_sentence': 'We found that the A allele of rs2507800 suppressed <e2>angiopoietin-1</e2> translation by facilitating <e1>miR-211</e1> binding, but not the T allele.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (51, 65), 'entity_text': 'angiopoietin-1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (94, 101), 'entity_text': 'miR-211'}}\n",
      "add special case angiopoietin-1\n",
      "add special case miR-211\n",
      "angiopoietin-1 9 miR-211 13\n",
      "(u\"We found that the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding, but not the T allele.\", 9, 13)\n",
      "geneword angiopoietin-1 mirword miR-211\n",
      "Conjuncts [allele, binding]\n",
      "neighbour allele .\n",
      "neighbour allele None\n",
      "Conjunction before fc binding\n",
      "fc add children for binding\n",
      "Conjuncts [binding, allele]\n",
      "neighbour binding ,\n",
      "neighbour binding but\n",
      "Conjunction before fc allele\n",
      "fc add children for allele\n",
      "Conjunctions ( conj )\n",
      "binding [miR-211, binding, T, allele]\n",
      "Conjuncts [allele, rs2507800]\n",
      "neighbour allele of\n",
      "neighbour allele of\n",
      "Conjunction before fc rs2507800\n",
      "fc add children for rs2507800\n",
      "Conjunctions ( nmod )\n",
      "rs2507800 [allele, of, rs2507800]\n",
      "SDP sent: We found that the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding, but not the T allele.\n",
      "SDP     : [(angiopoietin-1, 'NOUN', 'compound', False), (translation, 'NOUN', 'dobj', False), (suppressed, 'VERB', 'ccomp', False), (facilitating, 'VERB', 'advcl', False), (binding, 'NOUN', 'dobj', False), (miR-211, 'NOUN', 'compound', False)]\n",
      "SDP DOBJ RULE\n",
      "deepcheck connection rule [by, facilitating, miR-211, binding, ,, but, not, the, T, allele]\n",
      "DeepCheck Conj Subtree\n",
      "[the, T, allele]\n",
      "0 [We, found] 2 0 We found that the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding, but not the T allele.\n",
      "1 [the, A, allele, of, rs2507800, suppressed, angiopoietin-1, translation, by, facilitating, miR-211, binding, ,, but, not, the, T, allele] 18 14 the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding, but not the T allele.\n",
      "DeepCheck Conj Subtree\n",
      "[of, rs2507800]\n",
      "deepcheck connection rule [by, facilitating, miR-211, binding, ,, but, not, the, T, allele]\n",
      "DeepCheck Conj Subtree\n",
      "[the, T, allele]\n",
      "0 [We, found] 2 0 We found that the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding, but not the T allele.\n",
      "1 [the, A, allele, of, rs2507800, suppressed, angiopoietin-1, translation, by, facilitating, miR-211, binding, ,, but, not, the, T, allele] 18 14 the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding, but not the T allele.\n",
      "Recognized overlapping entities: {Interval(51, 65, ('GENE_OR_GENE_PRODUCT', 'angiopoietin-1'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding , but not the T allele\n",
      "[the, A, allele, of, rs2507800, suppressed, angiopoietin-1, translation, by, facilitating, miR-211, binding, ,, but, not, the, T, allele]\n",
      "TextBetween: the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding , but not the T allele\n",
      "infer direction from token set [the, A, allele, of, rs2507800, suppressed, angiopoietin-1, translation, by, facilitating, miR-211, binding, ,, but, not, the, T, allele]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fc0b806d048>, {'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c510>, {'gene': Counter({'NEG': 1, 'NEU': 1}), 'mir': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(suppressed, 'NEG'), (translation, 'NEU'), (binding, 'NEU')]\n",
      "counts between 3 Counter() NEG\n",
      "Between Tokens []\n",
      "counts between final False\n",
      "['rs2507800', 'suppressed', 'angiopoietin-1', 'translation', 'by', 'facilitating', 'miR-211', 'binding', ',']\n",
      "Counter()\n",
      "[]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence We found that the A allele of rs2507800 suppressed angiopoietin-1 translation by facilitating miR-211 binding, but not the T allele.\n",
      "full sentence We found that the A allele of rs2507800 suppressed <e2>angiopoietin-1</e2> translation by facilitating <e1>miR-211</e1> binding, but not the T allele.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'We found that the A allele of rs2507800 suppressed <e2>angiopoietin-1</e2> translation by facilitating <e1>miR-211</e1> binding, but not the T allele.', 'full_sentence': 'We found that the A allele of rs2507800 suppressed <e2>angiopoietin-1</e2> translation by facilitating <e1>miR-211</e1> binding, but not the T allele.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (51, 65), 'entity_text': 'angiopoietin-1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (94, 101), 'entity_text': 'miR-211'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d66.s4.p0\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miR-211\n",
      "add special case angiopoietin-1\n",
      "miR-211-binding 9 angiopoietin-1 12\n",
      "(u\"variant rs2507800 was identified to be located in the miR-211-binding site of angiopoietin-1.\", 9, 12)\n",
      "geneword angiopoietin-1 mirword miR-211-binding\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [located, site]\n",
      "neighbour located in\n",
      "neighbour located in\n",
      "Conjunction before fc site\n",
      "fc add children for site\n",
      "fc add children for site\n",
      "fc add children for site\n",
      "Conjuncts [site, angiopoietin-1]\n",
      "neighbour site of\n",
      "neighbour site of\n",
      "Conjunction before fc angiopoietin-1\n",
      "fc add children for angiopoietin-1\n",
      "Conjunctions ( nmod )\n",
      "site [located, in, miR-211-binding, site, of, angiopoietin-1]\n",
      "achildren {miR-211-binding, site}\n",
      "achildren False\n",
      "SDP sent: variant rs2507800 was identified to be located in the miR-211-binding site of angiopoietin-1.\n",
      "SDP     : [(miR-211-binding, 'VERB', 'amod', False), (site, 'NOUN', 'nmod', False), (angiopoietin-1, 'NOUN', 'nmod', False)]\n",
      "0 [variant, rs2507800, was, identified, to, be, located, in, the, miR-211-binding, site, of, angiopoietin-1] 13 0 variant rs2507800 was identified to be located in the miR-211-binding site of angiopoietin-1.\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, miR-211-binding, site, of, angiopoietin-1]\n",
      "DeepCheck Conj Subtree\n",
      "[of, angiopoietin-1]\n",
      "0 [variant, rs2507800, was, identified, to, be, located, in, the, miR-211-binding, site, of, angiopoietin-1] 13 0 variant rs2507800 was identified to be located in the miR-211-binding site of angiopoietin-1.\n",
      "Context Fail for Noun Rule\n",
      "Recognized overlapping entities: {Interval(78, 92, ('GENE_OR_GENE_PRODUCT', 'angiopoietin-1'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text variant rs2507800 was identified to be located in the miR-211-binding site of angiopoietin-1\n",
      "[variant, rs2507800, was, identified, to, be, located, in, the, miR-211-binding, site, of, angiopoietin-1]\n",
      "TextBetween: variant rs2507800 was identified to be located in the miR-211-binding site of angiopoietin-1\n",
      "infer direction from token set [variant, rs2507800, was, identified, to, be, located, in, the, miR-211-binding, site, of, angiopoietin-1]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8ce18>, {})\n",
      "counts between 1 stemTypes2Count Counter()\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c6a8>, {})\n",
      "counts between 1 stemTypes2Count []\n",
      "(False, {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}})\n",
      " hit sentence variant rs2507800 was identified to be located in the miR-211-binding site of angiopoietin-1.\n",
      "full sentence variant rs2507800 was identified to be located in the <e1>miR-211</e1>-binding site of <e2>angiopoietin-1</e2>.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'return', 'reg_detect_major': 'return'}}, 'tagged_sent': 'variant rs2507800 was identified to be located in the <e1>miR-211</e1>-binding site of <e2>angiopoietin-1</e2>.', 'full_sentence': \"To test our hypothesis, a predicted microRNA-binding site was found in the 3'-UTR of angiopoietin-1 using bioinformatics; variant rs2507800 was identified to be located in the <e1>miR-211</e1>-binding site of <e2>angiopoietin-1</e2>.\", 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (54, 61), 'entity_text': 'miR-211'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (78, 92), 'entity_text': 'angiopoietin-1'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d66.s8.p0\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case angiopoietin-1\n",
      "add special case microRNA 211\n",
      "angiopoietin-1 7 211 20\n",
      "(u\"A functional variant in the 3'-UTR of angiopoietin-1 might reduce stroke risk by interfering with the binding efficiency of microRNA 211.\", 7, 20)\n",
      "geneword angiopoietin-1 mirword 211\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [variant, 3'-UTR]\n",
      "neighbour variant in\n",
      "neighbour variant in\n",
      "Conjunction before fc 3'-UTR\n",
      "fc add children for 3'-UTR\n",
      "fc add children for 3'-UTR\n",
      "Conjuncts [3'-UTR, angiopoietin-1]\n",
      "neighbour 3'-UTR of\n",
      "neighbour 3'-UTR of\n",
      "Conjunction before fc angiopoietin-1\n",
      "fc add children for angiopoietin-1\n",
      "Conjuncts [interfering, efficiency]\n",
      "neighbour interfering with\n",
      "neighbour interfering with\n",
      "Conjunction before fc efficiency\n",
      "fc add children for efficiency\n",
      "fc add children for efficiency\n",
      "fc add children for efficiency\n",
      "Conjuncts [efficiency, microRNA]\n",
      "neighbour efficiency of\n",
      "neighbour efficiency of\n",
      "Conjunction before fc microRNA\n",
      "fc add children for microRNA\n",
      "fc add children for microRNA\n",
      "Conunct merge check: 3'-UTR efficiency\n",
      "angiopoietin-1 might reduce stroke risk by\n",
      "Conjunctions ( nmod )\n",
      "3'-UTR [variant, in, 3'-UTR, of, angiopoietin-1]\n",
      "efficiency [interfering, with, binding, efficiency, of, microRNA, 211]\n",
      "SDP sent: A functional variant in the 3'-UTR of angiopoietin-1 might reduce stroke risk by interfering with the binding efficiency of microRNA 211.\n",
      "SDP     : [(angiopoietin-1, 'NOUN', 'nmod', False), (3'-UTR, 'NOUN', 'nmod', False), (variant, 'NOUN', 'nsubj', False), (reduce, 'VERB', 'ROOT', False), (interfering, 'VERB', 'advcl', False), (efficiency, 'NOUN', 'nmod', False), (microRNA, 'NOUN', 'nmod', False), (211, 'NUM', 'nummod', False)]\n",
      "deepcheck connection rule [by, interfering, with, the, binding, efficiency, of, microRNA, 211]\n",
      "0 [A, functional, variant, in, the, 3'-UTR, of, angiopoietin-1, might, reduce, stroke, risk, by, interfering, with, the, binding, efficiency, of, microRNA, 211] 21 0 A functional variant in the 3'-UTR of angiopoietin-1 might reduce stroke risk by interfering with the binding efficiency of microRNA 211.\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, 3'-UTR, of, angiopoietin-1]\n",
      "DeepCheck Conj Subtree\n",
      "[of, angiopoietin-1]\n",
      "deepcheck connection rule [by, interfering, with, the, binding, efficiency, of, microRNA, 211]\n",
      "DeepCheck Conj Subtree\n",
      "[with, the, binding, efficiency, of, microRNA, 211]\n",
      "DeepCheck Conj Subtree\n",
      "[of, microRNA, 211]\n",
      "0 [A, functional, variant, in, the, 3'-UTR, of, angiopoietin-1, might, reduce, stroke, risk, by, interfering, with, the, binding, efficiency, of, microRNA, 211] 21 0 A functional variant in the 3'-UTR of angiopoietin-1 might reduce stroke risk by interfering with the binding efficiency of microRNA 211.\n",
      "Recognized overlapping entities: {Interval(38, 52, ('GENE_OR_GENE_PRODUCT', 'angiopoietin-1'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text A functional variant in the 3'-UTR of angiopoietin-1 might reduce stroke risk by interfering with the binding efficiency of microRNA 211\n",
      "[A, functional, variant, in, the, 3'-UTR, of, angiopoietin-1, might, reduce, stroke, risk, by, interfering, with, the, binding, efficiency, of, microRNA, 211]\n",
      "TextBetween: A functional variant in the 3'-UTR of angiopoietin-1 might reduce stroke risk by interfering with the binding efficiency of microRNA 211\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between gene mir', 'reg_detect_major': 'between'}})\n",
      " hit sentence A functional variant in the 3'-UTR of angiopoietin-1 might reduce stroke risk by interfering with the binding efficiency of microRNA 211.\n",
      "full sentence A functional variant in the 3'-UTR of <e2>angiopoietin-1</e2> might reduce stroke risk by interfering with the binding efficiency of <e1>microRNA 211</e1>.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'between gene mir', 'reg_detect_major': 'between'}}, 'tagged_sent': \"A functional variant in the 3'-UTR of <e2>angiopoietin-1</e2> might reduce stroke risk by interfering with the binding efficiency of <e1>microRNA 211</e1>.\", 'full_sentence': \"A functional variant in the 3'-UTR of <e2>angiopoietin-1</e2> might reduce stroke risk by interfering with the binding efficiency of <e1>microRNA 211</e1>.\", 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (38, 52), 'entity_text': 'angiopoietin-1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (124, 136), 'entity_text': 'microRNA 211'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d66.s9.p1\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('MIR_GENE', 'NEU', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of <e2>angiopoietin-1</e2> might reduce the risk of stroke by interfering with <e1>miR-211</e1> binding.\", 'full_sentence': \"Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of <e2>angiopoietin-1</e2> might reduce the risk of stroke by interfering with <e1>miR-211</e1> binding.\", 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (79, 93), 'entity_text': 'angiopoietin-1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (146, 153), 'entity_text': 'miR-211'}}\n",
      "add special case angiopoietin-1\n",
      "add special case miR-211\n",
      "angiopoietin-1 15 miR-211 25\n",
      "(u\"Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding.\", 15, 25)\n",
      "geneword angiopoietin-1 mirword miR-211\n",
      "Conjunctions ( conj )\n",
      "rs2507800 [(, )]\n",
      "Conjuncts [genotype, 3'-UTR]\n",
      "neighbour genotype (\n",
      "neighbour genotype rs2507800\n",
      "Conjunction before fc 3'-UTR\n",
      "fc add children for 3'-UTR\n",
      "fc add children for 3'-UTR\n",
      "Conjuncts [3'-UTR, angiopoietin-1]\n",
      "neighbour 3'-UTR of\n",
      "neighbour 3'-UTR of\n",
      "Conjunction before fc angiopoietin-1\n",
      "fc add children for angiopoietin-1\n",
      "Conjuncts [risk, stroke]\n",
      "neighbour risk of\n",
      "neighbour risk of\n",
      "Conjunction before fc stroke\n",
      "fc add children for stroke\n",
      "Conjuncts [interfering, binding]\n",
      "neighbour interfering with\n",
      "neighbour interfering with\n",
      "Conjunction before fc binding\n",
      "fc add children for binding\n",
      "fc add children for binding\n",
      "Conunct merge check: 3'-UTR rs2507800\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: rs2507800 stroke\n",
      ") in the 3'-UTR of angiopoietin-1 might reduce the\n",
      "Conunct merge check: stroke binding\n",
      "stroke by\n",
      "Conjunctions ( nmod )\n",
      "rs2507800 [(, )]\n",
      "3'-UTR [genotype, in, 3'-UTR, of, angiopoietin-1]\n",
      "stroke [risk, of, stroke]\n",
      "binding [interfering, with, miR-211, binding]\n",
      "SDP sent: Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding.\n",
      "SDP     : [(angiopoietin-1, 'NOUN', 'nmod', False), (3'-UTR, 'NOUN', 'nmod', False), (genotype, 'NOUN', 'nsubj', False), (reduce, 'VERB', 'ccomp', False), (interfering, 'VERB', 'advcl', False), (binding, 'NOUN', 'nmod', False), (miR-211, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule [by, interfering, with, miR-211, binding]\n",
      "0 [Our, results, provide, evidence] 4 0 Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding.\n",
      "1 [the, TT, genotype, (, rs2507800, ), in, the, 3'-UTR, of, angiopoietin-1, might, reduce, the, risk, of, stroke, by, interfering, with, miR-211, binding] 22 34 the TT genotype (rs2507800) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding.\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, 3'-UTR, of, angiopoietin-1]\n",
      "DeepCheck Conj Subtree\n",
      "[of, angiopoietin-1]\n",
      "DeepCheck Conj Subtree\n",
      "[of, stroke]\n",
      "deepcheck connection rule [by, interfering, with, miR-211, binding]\n",
      "DeepCheck Conj Subtree\n",
      "[with, miR-211, binding]\n",
      "0 [Our, results, provide, evidence] 4 0 Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding.\n",
      "1 [the, TT, genotype, (, rs2507800, ), in, the, 3'-UTR, of, angiopoietin-1, might, reduce, the, risk, of, stroke, by, interfering, with, miR-211, binding] 22 34 the TT genotype (rs2507800) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding.\n",
      "Recognized overlapping entities: {Interval(79, 93, ('GENE_OR_GENE_PRODUCT', 'angiopoietin-1'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text the TT genotype ( rs2507800 ) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding\n",
      "[the, TT, genotype, (, rs2507800, ), in, the, 3'-UTR, of, angiopoietin-1, might, reduce, the, risk, of, stroke, by, interfering, with, miR-211, binding]\n",
      "TextBetween: the TT genotype ( rs2507800 ) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding\n",
      "infer direction from token set [the, TT, genotype, (, rs2507800, ), in, the, 3'-UTR, of, angiopoietin-1, might, reduce, the, risk, of, stroke, by, interfering, with, miR-211, binding]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c158>, {'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c9d8>, {'gene': Counter({'NEG': 1}), 'mir': Counter({'NEU': 2})})\n",
      "counts between 1 stemTypes2Count [(reduce, 'NEG'), (interfering, 'NEU'), (binding, 'NEU')]\n",
      "counts between 3 Counter({'NEG': 1}) NEG\n",
      "Between Tokens [(reduce, 'NEG')]\n",
      "counts between final False\n",
      "[\"3'-UTR\", 'of', 'angiopoietin-1', 'might', 'reduce', 'the', 'risk', 'of', 'stroke', 'by', 'interfering', 'with', 'miR-211', 'binding']\n",
      "Counter({'NEG': 1})\n",
      "[(reduce, 'NEG')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of angiopoietin-1 might reduce the risk of stroke by interfering with miR-211 binding.\n",
      "full sentence Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of <e2>angiopoietin-1</e2> might reduce the risk of stroke by interfering with <e1>miR-211</e1> binding.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': \"Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of <e2>angiopoietin-1</e2> might reduce the risk of stroke by interfering with <e1>miR-211</e1> binding.\", 'full_sentence': \"Our results provide evidence that the TT genotype (rs2507800) in the 3'-UTR of <e2>angiopoietin-1</e2> might reduce the risk of stroke by interfering with <e1>miR-211</e1> binding.\", 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (79, 93), 'entity_text': 'angiopoietin-1'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (146, 153), 'entity_text': 'miR-211'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d69.s5.p3\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case GRN\n",
      "add special case miR-107\n",
      "GRN 24 miR-107 29\n",
      "(u\"miR-107 has been implicated in Alzheimer's disease pathogenesis, and sequence elements in the open reading frame-rather than the 3' untranslated region-of GRN mRNA are recognized by miR-107 and are highly conserved among vertebrate species.\", 24, 29)\n",
      "geneword GRN mirword miR-107\n",
      "Conjuncts [elements, recognized, conserved, implicated]\n",
      "neighbour elements in\n",
      "neighbour elements in\n",
      "neighbour recognized by\n",
      "neighbour recognized by\n",
      "neighbour conserved among\n",
      "neighbour conserved among\n",
      "Conjunction before fc implicated\n",
      "fc add children for implicated\n",
      "Conjuncts [implicated, recognized, conserved, elements]\n",
      "neighbour implicated in\n",
      "neighbour implicated in\n",
      "neighbour recognized by\n",
      "neighbour recognized by\n",
      "neighbour conserved among\n",
      "neighbour conserved among\n",
      "Conjunction before fc elements\n",
      "fc add children for elements\n",
      "fc add children for elements\n",
      "Conjuncts [implicated, elements, conserved, recognized]\n",
      "neighbour implicated in\n",
      "neighbour implicated in\n",
      "neighbour elements in\n",
      "neighbour elements in\n",
      "neighbour conserved among\n",
      "neighbour conserved among\n",
      "Conjunction before fc recognized\n",
      "fc add children for recognized\n",
      "Conjuncts [implicated, elements, recognized, conserved]\n",
      "neighbour implicated in\n",
      "neighbour implicated in\n",
      "neighbour elements in\n",
      "neighbour elements in\n",
      "neighbour recognized by\n",
      "neighbour recognized by\n",
      "Conjunction before fc conserved\n",
      "fc add children for conserved\n",
      "Conjunctions ( conj )\n",
      "implicated [implicated, in, 's, disease, pathogenesis, sequence, elements, in, open, reading, frame-rather, recognized, by, miR-107, conserved, among, vertebrate, species]\n",
      "Conjuncts [implicated, pathogenesis]\n",
      "neighbour implicated in\n",
      "neighbour implicated in\n",
      "Conjunction before fc pathogenesis\n",
      "fc add children for pathogenesis\n",
      "fc add children for pathogenesis\n",
      "fc add children for pathogenesis\n",
      "Conjuncts [elements, frame-rather]\n",
      "neighbour elements in\n",
      "neighbour elements in\n",
      "Conjunction before fc frame-rather\n",
      "fc add children for frame-rather\n",
      "fc add children for frame-rather\n",
      "fc add children for frame-rather\n",
      "Conjuncts [recognized, miR-107]\n",
      "neighbour recognized by\n",
      "neighbour recognized by\n",
      "Conjunction before fc miR-107\n",
      "fc add children for miR-107\n",
      "Conjuncts [conserved, species]\n",
      "neighbour conserved among\n",
      "neighbour conserved among\n",
      "Conjunction before fc species\n",
      "fc add children for species\n",
      "fc add children for species\n",
      "Conunct merge check: pathogenesis frame-rather\n",
      "pathogenesis, and sequence\n",
      "Double Conjunction Found\n",
      "Conunct merge check: frame-rather miR-107\n",
      "frame-rather than the 3' untranslated region-of GRN mRNA are\n",
      "Conunct merge check: miR-107 species\n",
      "miR-107 and are highly\n",
      "Double Conjunction Found\n",
      "Conjunctions ( nmod )\n",
      "pathogenesis [implicated, in, 's, disease, pathogenesis]\n",
      "frame-rather [elements, in, open, reading, frame-rather]\n",
      "miR-107 [recognized, by, miR-107]\n",
      "species [conserved, among, vertebrate, species]\n",
      "(frame-rather, 'conj') [implicated, in, 's, disease, pathogenesis, elements, in, open, reading, frame-rather]\n",
      "(species, 'conj') [recognized, by, miR-107, conserved, among, vertebrate, species]\n",
      "SDP sent: miR-107 has been implicated in Alzheimer's disease pathogenesis, and sequence elements in the open reading frame-rather than the 3' untranslated region-of GRN mRNA are recognized by miR-107 and are highly conserved among vertebrate species.\n",
      "SDP     : [(GRN, 'NOUN', 'compound', False), (mRNA, 'NOUN', 'nsubjpass', False), (recognized, 'VERB', 'conj', False), (miR-107, 'NOUN', 'nmod', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[sequence, elements, in, the, open, reading, frame-rather]\n",
      "0 [miR-107, has, been, implicated, in, Alzheimer, 's, disease, pathogenesis, ,, and, sequence, elements, in, the, open, reading, frame-rather] 18 0 miR-107 has been implicated in Alzheimer's disease pathogenesis, and sequence elements in the open reading frame-rather than the 3' untranslated region-of GRN mRNA are recognized by miR-107 and are highly conserved among vertebrate species.\n",
      "1 [the, 3, ', untranslated, region-of, GRN, mRNA, are, recognized, by, miR-107] 11 125 the 3' untranslated region-of GRN mRNA are recognized by miR-107 and are highly conserved among vertebrate species.\n",
      "2 [are, highly, conserved, among, vertebrate, species] 6 194 are highly conserved among vertebrate species.\n",
      "DeepCheck Conj Subtree\n",
      "[in, Alzheimer, 's, disease, pathogenesis]\n",
      "DeepCheck Conj Subtree\n",
      "[sequence, elements, in, the, open, reading, frame-rather]\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, open, reading, frame-rather]\n",
      "DeepCheck Conj Subtree\n",
      "[by, miR-107]\n",
      "DeepCheck Conj Subtree\n",
      "[among, vertebrate, species]\n",
      "0 [miR-107, has, been, implicated, in, Alzheimer, 's, disease, pathogenesis, ,, and, sequence, elements, in, the, open, reading, frame-rather] 18 0 miR-107 has been implicated in Alzheimer's disease pathogenesis, and sequence elements in the open reading frame-rather than the 3' untranslated region-of GRN mRNA are recognized by miR-107 and are highly conserved among vertebrate species.\n",
      "1 [the, 3, ', untranslated, region-of, GRN, mRNA, are, recognized, by, miR-107] 11 125 the 3' untranslated region-of GRN mRNA are recognized by miR-107 and are highly conserved among vertebrate species.\n",
      "2 [are, highly, conserved, among, vertebrate, species] 6 194 are highly conserved among vertebrate species.\n",
      "Recognized overlapping entities: set()\n",
      "passive? True\n",
      "Counter()\n",
      "Compartment Text the 3 ' untranslated region-of GRN mRNA are recognized by miR-107\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'compartment gene mir by', 'reg_detect_major': 'compartment'}})\n",
      " hit sentence miR-107 has been implicated in Alzheimer's disease pathogenesis, and sequence elements in the open reading frame-rather than the 3' untranslated region-of GRN mRNA are recognized by miR-107 and are highly conserved among vertebrate species.\n",
      "full sentence miR-107 has been implicated in Alzheimer's disease pathogenesis, and sequence elements in the open reading frame-rather than the 3' untranslated region-of <e2>GRN</e2> mRNA are recognized by <e1>miR-107</e1> and are highly conserved among vertebrate species.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': True, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': True, 'reg_detect': 'compartment gene mir by', 'reg_detect_major': 'compartment'}}, 'tagged_sent': \"miR-107 has been implicated in Alzheimer's disease pathogenesis, and sequence elements in the open reading frame-rather than the 3' untranslated region-of <e2>GRN</e2> mRNA are recognized by <e1>miR-107</e1> and are highly conserved among vertebrate species.\", 'full_sentence': \"miR-107 has been implicated in Alzheimer's disease pathogenesis, and sequence elements in the open reading frame-rather than the 3' untranslated region-of <e2>GRN</e2> mRNA are recognized by <e1>miR-107</e1> and are highly conserved among vertebrate species.\", 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (155, 158), 'entity_text': 'GRN'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (182, 189), 'entity_text': 'miR-107'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d69.s11.p1\n",
      "IS:  ('MIR_GENE', 'NEU')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Glucose supplementation in cultured cells that leads to increased <e1>miR-107</e1> levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of <e2>GRN</e2> protein.', 'full_sentence': 'Glucose supplementation in cultured cells that leads to increased <e1>miR-107</e1> levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of <e2>GRN</e2> protein.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (66, 73), 'entity_text': 'miR-107'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (193, 196), 'entity_text': 'GRN'}}\n",
      "add special case miR-107\n",
      "add special case GRN\n",
      "miR-107 9 GRN 27\n",
      "(u\"Glucose supplementation in cultured cells that leads to increased miR-107 levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of GRN protein.\", 9, 27)\n",
      "geneword GRN mirword miR-107\n",
      "Conjuncts [secretion, expression]\n",
      "neighbour secretion of\n",
      "neighbour secretion of\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conjuncts [expression, secretion]\n",
      "neighbour expression ,\n",
      "neighbour expression including\n",
      "Conjunction before fc secretion\n",
      "fc add children for secretion\n",
      "fc add children for secretion\n",
      "Conjunctions ( conj )\n",
      "expression [in, decreased, GRN, expression, including, changes, in, cell, compartmentation, decreased, secretion, of, GRN, protein]\n",
      "Conjuncts [supplementation, cells]\n",
      "neighbour supplementation in\n",
      "neighbour supplementation in\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conjuncts [leads, levels]\n",
      "neighbour leads to\n",
      "neighbour leads to\n",
      "Conjunction before fc levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "fc add children for levels\n",
      "Conjuncts [results, expression]\n",
      "neighbour results in\n",
      "neighbour results in\n",
      "Conjunction before fc expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "fc add children for expression\n",
      "Conjuncts [expression, changes]\n",
      "neighbour expression ,\n",
      "neighbour expression including\n",
      "Conjunction before fc changes\n",
      "fc add children for changes\n",
      "fc add children for changes\n",
      "Conjuncts [changes, compartmentation]\n",
      "neighbour changes in\n",
      "neighbour changes in\n",
      "Conjunction before fc compartmentation\n",
      "fc add children for compartmentation\n",
      "fc add children for compartmentation\n",
      "Conjuncts [secretion, protein]\n",
      "neighbour secretion of\n",
      "neighbour secretion of\n",
      "Conjunction before fc protein\n",
      "fc add children for protein\n",
      "fc add children for protein\n",
      "Conunct merge check: cells levels\n",
      "cells that\n",
      "Conunct merge check: levels expression\n",
      "levels also\n",
      "Conunct merge check: expression protein\n",
      "compartmentation and decreased\n",
      "Double Conjunction Found\n",
      "Conjunctions ( nmod )\n",
      "cells [supplementation, in, cultured, cells]\n",
      "levels [leads, to, increased, miR-107, levels]\n",
      "expression [results, in, decreased, GRN, expression, including, changes, in, cell, compartmentation]\n",
      "protein [secretion, of, GRN, protein]\n",
      "(protein, 'conj') [results, in, decreased, GRN, expression, including, changes, in, cell, compartmentation, secretion, of, GRN, protein]\n",
      "SDP sent: Glucose supplementation in cultured cells that leads to increased miR-107 levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of GRN protein.\n",
      "SDP     : [(miR-107, 'NOUN', 'compound', False), (levels, 'NOUN', 'nmod', False), (leads, 'VERB', 'acl:relcl', False), (cells, 'NOUN', 'nmod', False), (supplementation, 'NOUN', 'nsubj', False), (results, 'VERB', 'ROOT', False), (expression, 'NOUN', 'nmod', True), (secretion, 'NOUN', 'conj', False), (protein, 'NOUN', 'nmod', False), (GRN, 'NOUN', 'compound', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[decreased, secretion, of, GRN, protein]\n",
      "deepcheck conj verb\n",
      "deepcheck conj verb\n",
      "0 [Glucose, supplementation, in, cultured, cells, that, leads, to, increased, miR-107, levels, also, results, in, decreased, GRN, expression, ,, including, changes, in, cell, compartmentation, and, decreased, secretion, of, GRN, protein] 29 0 Glucose supplementation in cultured cells that leads to increased miR-107 levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of GRN protein.\n",
      "DeepCheck Conj Subtree\n",
      "[in, cultured, cells, that, leads, to, increased, miR-107, levels]\n",
      "DeepCheck Conj Subtree\n",
      "[to, increased, miR-107, levels]\n",
      "DeepCheck Conj Subtree\n",
      "[in, decreased, GRN, expression, ,, including, changes, in, cell, compartmentation, and, decreased, secretion, of, GRN, protein]\n",
      "DeepCheck Conj Subtree\n",
      "[including, changes, in, cell, compartmentation]\n",
      "deepcheck conj verb\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[in, cell, compartmentation]\n",
      "DeepCheck Conj Subtree\n",
      "[decreased, secretion, of, GRN, protein]\n",
      "deepcheck conj verb\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[of, GRN, protein]\n",
      "0 [Glucose, supplementation, in, cultured, cells, that, leads, to, increased, miR-107, levels, also, results, in, decreased, GRN, expression, ,, including, changes, in, cell, compartmentation, and, decreased, secretion, of, GRN, protein] 29 0 Glucose supplementation in cultured cells that leads to increased miR-107 levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of GRN protein.\n",
      "Recognized overlapping entities: set()\n",
      "passive? False\n",
      "Counter({'POS': 1})\n",
      "Compartment Text Glucose supplementation in cultured cells that leads to increased miR-107 levels also results in decreased GRN expression , including changes in cell compartmentation and decreased secretion of GRN protein\n",
      "[Glucose, supplementation, in, cultured, cells, that, leads, to, increased, miR-107, levels, also, results, in, decreased, GRN, expression, ,, including, changes, in, cell, compartmentation, and, decreased, secretion, of, GRN, protein]\n",
      "TextBetween: Glucose supplementation in cultured cells that leads to increased miR-107 levels also results in decreased GRN expression , including changes in cell compartmentation and decreased secretion of GRN protein\n",
      "infer direction from token set [Glucose, supplementation, in, cultured, cells, that, leads, to, increased, miR-107, levels, also, results, in, decreased, GRN, expression, ,, including, changes, in, cell, compartmentation, and, decreased, secretion, of, GRN, protein]\n",
      "stemTypes2Count: Counter({'POS': 2, 'NEG': 2})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8ce18>, {'mir': Counter({'POS': 1}), 'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 4, 'POS': 2, 'NEG': 2})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c620>, {'mir': Counter({'POS': 1, 'NEU': 1}), 'gene': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count [(increased, 'POS'), (levels, 'NEU'), (results, 'NEU'), (decreased, 'NEG'), (expression, 'NEU'), (including, 'NEU'), (decreased, 'NEG'), (secretion, 'POS')]\n",
      "Between Tokens [(levels, 'NEU'), (results, 'NEU'), (decreased, 'NEG'), (expression, 'NEU'), (including, 'NEU'), (decreased, 'NEG'), (secretion, 'POS')]\n",
      "counts between final False\n",
      "['to', 'increased', 'miR-107', 'levels', 'also', 'results', 'in', 'decreased', 'GRN', 'expression', ',', 'including', 'changes', 'in', 'cell', 'compartmentation', 'and', 'decreased', 'secretion', 'of', 'GRN', 'protein']\n",
      "Counter({'NEU': 4, 'POS': 2, 'NEG': 2})\n",
      "[(increased, 'POS'), (levels, 'NEU'), (results, 'NEU'), (decreased, 'NEG'), (expression, 'NEU'), (including, 'NEU'), (decreased, 'NEG'), (secretion, 'POS')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Glucose supplementation in cultured cells that leads to increased miR-107 levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of GRN protein.\n",
      "full sentence Glucose supplementation in cultured cells that leads to increased <e1>miR-107</e1> levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of <e2>GRN</e2> protein.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Glucose supplementation in cultured cells that leads to increased <e1>miR-107</e1> levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of <e2>GRN</e2> protein.', 'full_sentence': 'Glucose supplementation in cultured cells that leads to increased <e1>miR-107</e1> levels also results in decreased GRN expression, including changes in cell compartmentation and decreased secretion of <e2>GRN</e2> protein.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (66, 73), 'entity_text': 'miR-107'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (193, 196), 'entity_text': 'GRN'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d70.s1.p0\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case MiR-21\n",
      "add special case caspase-3\n",
      "MiR-21 0 caspase-3 17\n",
      "(u\"MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\", 0, 17)\n",
      "geneword caspase-3 mirword MiR-21\n",
      "Conjuncts [induced, protected]\n",
      "neighbour induced apoptosis\n",
      "neighbour induced apoptosis\n",
      "Conjunction before fc protected\n",
      "fc add children for protected\n",
      "fc add children for protected\n",
      "Conjuncts [protected, induced]\n",
      "neighbour protected human\n",
      "neighbour protected human\n",
      "Conjunction before fc induced\n",
      "fc add children for induced\n",
      "Conjuncts [activity, ratio]\n",
      "neighbour activity .\n",
      "neighbour activity None\n",
      "Conjunction before fc ratio\n",
      "fc add children for ratio\n",
      "Conjuncts [ratio, activity]\n",
      "neighbour ratio and\n",
      "neighbour ratio and\n",
      "Conjunction before fc activity\n",
      "fc add children for activity\n",
      "Conunct merge check: protected ratio\n",
      "apoptosis by decreasing\n",
      "Conjunctions ( conj )\n",
      "protected [protected, human, glioblastoma, U87MG, cells, from, chemotherapeutic, drug, temozolomide, induced, apoptosis]\n",
      "ratio [Bax/Bcl-2, ratio, caspase-3, activity]\n",
      "Conjuncts [protected, temozolomide]\n",
      "neighbour protected human\n",
      "neighbour protected human\n",
      "Conjunction before fc temozolomide\n",
      "fc add children for temozolomide\n",
      "fc add children for temozolomide\n",
      "fc add children for temozolomide\n",
      "Conjunctions ( nmod )\n",
      "temozolomide [protected, from, chemotherapeutic, drug, temozolomide]\n",
      "SDP sent: MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "SDP     : [(MiR-21, 'NOUN', 'nsubj', False), (protected, 'VERB', 'ROOT', False), (decreasing, 'VERB', 'advcl', False), (ratio, 'NOUN', 'dobj', True), (activity, 'NOUN', 'conj', False), (caspase-3, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule noun found induced apoptosis [induced, apoptosis]\n",
      "deepcheck conj verb\n",
      "deepcheck connection rule [by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity]\n",
      "DeepCheck Conj Subtree\n",
      "[caspase-3, activity]\n",
      "0 [MiR-21, protected, human, glioblastoma, U87MG, cells, from, chemotherapeutic, drug, temozolomide, induced, apoptosis, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity] 19 0 MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "DeepCheck Conj Subtree\n",
      "[from, chemotherapeutic, drug, temozolomide]\n",
      "deepcheck connection rule noun found induced apoptosis [induced, apoptosis]\n",
      "deepcheck conj verb\n",
      "deepcheck connection rule [by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity]\n",
      "DeepCheck Conj Subtree\n",
      "[caspase-3, activity]\n",
      "0 [MiR-21, protected, human, glioblastoma, U87MG, cells, from, chemotherapeutic, drug, temozolomide, induced, apoptosis, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity] 19 0 MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "Recognized overlapping entities: {Interval(140, 149, ('GENE_OR_GENE_PRODUCT', 'caspase-3'))}\n",
      "passive? False\n",
      "Counter({'NEG': 1})\n",
      "Compartment Text MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity\n",
      "[MiR-21, protected, human, glioblastoma, U87MG, cells, from, chemotherapeutic, drug, temozolomide, induced, apoptosis, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity]\n",
      "TextBetween: MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity\n",
      "infer direction from token set [MiR-21, protected, human, glioblastoma, U87MG, cells, from, chemotherapeutic, drug, temozolomide, induced, apoptosis, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity]\n",
      "stemTypes2Count: Counter({'POS': 1, 'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbecba9e1e0>, {})\n",
      "counts between 1 stemTypes2Count Counter({'POS': 1, 'NEG': 1, 'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c510>, {'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(induced, 'POS'), (decreasing, 'NEG'), (activity, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}})\n",
      " hit sentence MiR-21 protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and caspase-3 activity.\n",
      "full sentence <e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and <e2>caspase-3</e2> activity.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between equal', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and <e2>caspase-3</e2> activity.', 'full_sentence': '<e1>MiR-21</e1> protected human glioblastoma U87MG cells from chemotherapeutic drug temozolomide induced apoptosis by decreasing Bax/Bcl-2 ratio and <e2>caspase-3</e2> activity.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (0, 6), 'entity_text': 'MiR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (140, 149), 'entity_text': 'caspase-3'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d70.s12.p0\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miR-21\n",
      "add special case Bax\n",
      "miR-21 12 Bax 21\n",
      "(u\"However, such effect was partly prevented by treatment of cells with miR-21 overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.\", 12, 21)\n",
      "geneword Bax mirword miR-21\n",
      "Conjuncts [upregulate, decrease, prevented]\n",
      "neighbour upregulate the\n",
      "neighbour upregulate the\n",
      "neighbour decrease caspase-3\n",
      "neighbour decrease caspase-3\n",
      "Conjunction before fc prevented\n",
      "fc add children for prevented\n",
      "Conjuncts [prevented, decrease, upregulate]\n",
      "neighbour prevented by\n",
      "neighbour prevented by\n",
      "neighbour decrease caspase-3\n",
      "neighbour decrease caspase-3\n",
      "Conjunction before fc upregulate\n",
      "fc add children for upregulate\n",
      "Conjuncts [prevented, upregulate, decrease]\n",
      "neighbour prevented by\n",
      "neighbour prevented by\n",
      "neighbour upregulate the\n",
      "neighbour upregulate the\n",
      "Conjunction before fc decrease\n",
      "fc add children for decrease\n",
      "Conjunctions ( conj )\n",
      "prevented [prevented, by, treatment, of, cells, with, miR-21, overexpression, before, upregulate, Bcl-2, expression, decrease, caspase-3, activity]\n",
      "Conjuncts [prevented, treatment]\n",
      "neighbour prevented by\n",
      "neighbour prevented by\n",
      "Conjunction before fc treatment\n",
      "fc add children for treatment\n",
      "fc add children for treatment\n",
      "fc add children for treatment\n",
      "fc add children for treatment\n",
      "Conjuncts [treatment, cells]\n",
      "neighbour treatment of\n",
      "neighbour treatment of\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "Conjuncts [treatment, overexpression]\n",
      "neighbour treatment of\n",
      "neighbour treatment of\n",
      "Conjunction before fc overexpression\n",
      "fc add children for overexpression\n",
      "fc add children for overexpression\n",
      "Conjunctions ( nmod )\n",
      "treatment [prevented, by, treatment, of, cells, with, miR-21, overexpression, before]\n",
      "SDP sent: However, such effect was partly prevented by treatment of cells with miR-21 overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.\n",
      "SDP     : [(miR-21, 'NOUN', 'compound', False), (overexpression, 'NOUN', 'nmod', False), (treatment, 'NOUN', 'nmod', False), (appeared, 'VERB', 'acl:relcl', False), (downregulate, 'VERB', 'xcomp', False), (expression, 'NOUN', 'dobj', False), (Bax, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule [to, downregulate, the, Bax, expression]\n",
      "deepcheck conj verb\n",
      "deepcheck connection rule noun found decrease caspase-3 [decrease, caspase-3, activity]\n",
      "deepcheck conj verb\n",
      "0 [However, ,, such, effect, was, partly, prevented, by, treatment, of, cells, with, miR-21, overexpression, before, ,, which, appeared, to, downregulate, the, Bax, expression, ,, upregulate, the, Bcl-2, expression, and, decrease, caspase-3, activity] 32 0 However, such effect was partly prevented by treatment of cells with miR-21 overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.\n",
      "DeepCheck Conj Subtree\n",
      "[by, treatment, of, cells, with, miR-21, overexpression, before, ,, which, appeared, to, downregulate, the, Bax, expression]\n",
      "DeepCheck Conj Subtree\n",
      "[of, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[with, miR-21, overexpression]\n",
      "deepcheck connection rule [to, downregulate, the, Bax, expression]\n",
      "deepcheck conj verb\n",
      "deepcheck connection rule noun found decrease caspase-3 [decrease, caspase-3, activity]\n",
      "deepcheck conj verb\n",
      "0 [However, ,, such, effect, was, partly, prevented, by, treatment, of, cells, with, miR-21, overexpression, before, ,, which, appeared, to, downregulate, the, Bax, expression, ,, upregulate, the, Bcl-2, expression, and, decrease, caspase-3, activity] 32 0 However, such effect was partly prevented by treatment of cells with miR-21 overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.\n",
      "Recognized overlapping entities: {Interval(134, 137, ('GENE_OR_GENE_PRODUCT', 'Bax'))}\n",
      "passive? False\n",
      "Counter({'POS': 1, 'NEG': 1})\n",
      "Compartment Text However , such effect was partly prevented by treatment of cells with miR-21 overexpression before , which appeared to downregulate the Bax expression , upregulate the Bcl-2 expression and decrease caspase-3 activity\n",
      "[However, ,, such, effect, was, partly, prevented, by, treatment, of, cells, with, miR-21, overexpression, before, ,, which, appeared, to, downregulate, the, Bax, expression, ,, upregulate, the, Bcl-2, expression, and, decrease, caspase-3, activity]\n",
      "TextBetween: However , such effect was partly prevented by treatment of cells with miR-21 overexpression before , which appeared to downregulate the Bax expression , upregulate the Bcl-2 expression and decrease caspase-3 activity\n",
      "infer direction from token set [However, ,, such, effect, was, partly, prevented, by, treatment, of, cells, with, miR-21, overexpression, before, ,, which, appeared, to, downregulate, the, Bax, expression, ,, upregulate, the, Bcl-2, expression, and, decrease, caspase-3, activity]\n",
      "stemTypes2Count: Counter({'NEG': 3, 'POS': 2})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62a60>, {'mir': Counter({'POS': 1}), 'gene': Counter({'NEG': 1})})\n",
      "defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62a60>, {'mir': Counter({'POS': 1}), 'gene': Counter({'NEG': 1})})\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}})\n",
      " hit sentence However, such effect was partly prevented by treatment of cells with miR-21 overexpression before, which appeared to downregulate the Bax expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.\n",
      "full sentence However, such effect was partly prevented by treatment of cells with <e1>miR-21</e1> overexpression before, which appeared to downregulate the <e2>Bax</e2> expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts opp', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'However, such effect was partly prevented by treatment of cells with <e1>miR-21</e1> overexpression before, which appeared to downregulate the <e2>Bax</e2> expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.', 'full_sentence': 'However, such effect was partly prevented by treatment of cells with <e1>miR-21</e1> overexpression before, which appeared to downregulate the <e2>Bax</e2> expression, upregulate the Bcl-2 expression and decrease caspase-3 activity.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (69, 75), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (134, 137), 'entity_text': 'Bax'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d70.s14.p2\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miR-21\n",
      "add special case caspase-3\n",
      "miR-21 8 caspase-3 27\n",
      "(u\"Taken together, these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\", 8, 27)\n",
      "geneword caspase-3 mirword miR-21\n",
      "Conjuncts [activity, ratio]\n",
      "neighbour activity ,\n",
      "neighbour activity which\n",
      "Conjunction before fc ratio\n",
      "fc add children for ratio\n",
      "Conjuncts [ratio, activity]\n",
      "neighbour ratio and\n",
      "neighbour ratio and\n",
      "Conjunction before fc activity\n",
      "fc add children for activity\n",
      "Conjunctions ( conj )\n",
      "ratio [Bax/Bcl-2, ratio, caspase-3, activity]\n",
      "Conjuncts [apoptosis, cells]\n",
      "neighbour apoptosis in\n",
      "neighbour apoptosis in\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "fc add children for cells\n",
      "Conjuncts [inhibit, part]\n",
      "neighbour inhibit TMZ-induced\n",
      "neighbour inhibit TMZ-induced\n",
      "Conjunction before fc part\n",
      "fc add children for part\n",
      "Conjuncts [possibility, overexpression]\n",
      "neighbour possibility of\n",
      "neighbour possibility of\n",
      "Conjunction before fc overexpression\n",
      "fc add children for overexpression\n",
      "fc add children for overexpression\n",
      "fc add children for overexpression\n",
      "Conjuncts [overexpression, resistance]\n",
      "neighbour overexpression in\n",
      "neighbour overexpression in\n",
      "Conjunction before fc resistance\n",
      "fc add children for resistance\n",
      "fc add children for resistance\n",
      "fc add children for resistance\n",
      "Conjuncts [resistance, therapy]\n",
      "neighbour resistance to\n",
      "neighbour resistance to\n",
      "Conjunction before fc therapy\n",
      "fc add children for therapy\n",
      "fc add children for therapy\n",
      "fc add children for therapy\n",
      "Conjuncts [therapy, TMZ]\n",
      "neighbour therapy of\n",
      "neighbour therapy of\n",
      "Conjunction before fc TMZ\n",
      "fc add children for TMZ\n",
      "Conunct merge check: part cells\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conunct merge check: cells overexpression\n",
      "cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the\n",
      "Conjunctions ( nmod )\n",
      "cells [apoptosis, in, U87MG, cells]\n",
      "part [inhibit, in, part]\n",
      "overexpression [possibility, of, miR-21, overexpression, in, clinical, resistance, to, chemotherapeutic, therapy, of, TMZ]\n",
      "SDP sent: Taken together, these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "SDP     : [(miR-21, 'NOUN', 'nsubj', False), (inhibit, 'VERB', 'ccomp', False), (decreasing, 'VERB', 'advcl', False), (ratio, 'NOUN', 'dobj', True), (activity, 'NOUN', 'conj', False), (caspase-3, 'NOUN', 'compound', False)]\n",
      "deepcheck connection rule [by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity, ,, which, highlighted, the, possibility, of, miR-21, overexpression, in, the, clinical, resistance, to, chemotherapeutic, therapy, of, TMZ]\n",
      "DeepCheck Conj Subtree\n",
      "[caspase-3, activity]\n",
      "0 [these, results, suggested, that, over-express, miR-21, could, inhibit, TMZ-induced, apoptosis, in, U87MG, cells, ,, at, least, in, part, ,, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity] 26 16 these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "1 [Taken, together] 2 0 Taken together, these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "2 [] 0\n",
      "empty compartment []\n",
      "3 [which, highlighted, the, possibility, of, miR-21, overexpression, in, the, clinical, resistance, to, chemotherapeutic, therapy, of, TMZ] 16 189 which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "DeepCheck Conj Subtree\n",
      "[in, U87MG, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[at, least, in, part]\n",
      "deepcheck connection rule [by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity, ,, which, highlighted, the, possibility, of, miR-21, overexpression, in, the, clinical, resistance, to, chemotherapeutic, therapy, of, TMZ]\n",
      "DeepCheck Conj Subtree\n",
      "[caspase-3, activity]\n",
      "DeepCheck Conj Subtree\n",
      "[of, miR-21, overexpression, in, the, clinical, resistance, to, chemotherapeutic, therapy, of, TMZ]\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, clinical, resistance, to, chemotherapeutic, therapy, of, TMZ]\n",
      "DeepCheck Conj Subtree\n",
      "[to, chemotherapeutic, therapy, of, TMZ]\n",
      "DeepCheck Conj Subtree\n",
      "[of, TMZ]\n",
      "0 [these, results, suggested, that, over-express, miR-21, could, inhibit, TMZ-induced, apoptosis, in, U87MG, cells, ,, at, least, in, part, ,, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity] 26 16 these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "1 [Taken, together] 2 0 Taken together, these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "2 [] 0\n",
      "empty compartment []\n",
      "3 [which, highlighted, the, possibility, of, miR-21, overexpression, in, the, clinical, resistance, to, chemotherapeutic, therapy, of, TMZ] 16 189 which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "Recognized overlapping entities: {Interval(169, 178, ('GENE_OR_GENE_PRODUCT', 'caspase-3'))}\n",
      "passive? False\n",
      "Counter({'NEG': 2})\n",
      "Compartment Text these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells , at least in part , by decreasing Bax/Bcl-2 ratio and caspase-3 activity\n",
      "[these, results, suggested, that, over-express, miR-21, could, inhibit, TMZ-induced, apoptosis, in, U87MG, cells, ,, at, least, in, part, ,, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity]\n",
      "TextBetween: these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells , at least in part , by decreasing Bax/Bcl-2 ratio and caspase-3 activity\n",
      "infer direction from token set [these, results, suggested, that, over-express, miR-21, could, inhibit, TMZ-induced, apoptosis, in, U87MG, cells, ,, at, least, in, part, ,, by, decreasing, Bax/Bcl-2, ratio, and, caspase-3, activity]\n",
      "stemTypes2Count: Counter({'NEG': 2, 'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbecba9e048>, {'mir': Counter({'POS': 1, 'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62ea0>, {'mir': Counter({'POS': 1, 'NEG': 1}), 'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(over-express, 'POS'), (inhibit, 'NEG'), (decreasing, 'NEG'), (activity, 'NEU')]\n",
      "Between Tokens [(inhibit, 'NEG'), (decreasing, 'NEG')]\n",
      "Between Tokens Switch\n",
      "counts between final False\n",
      "['that', 'over-express', 'miR-21', 'could', 'inhibit', 'TMZ-induced', 'apoptosis', 'in', 'U87MG', 'cells', ',', 'at', 'least', 'in', 'part', ',', 'by', 'decreasing', 'Bax/Bcl-2', 'ratio', 'and', 'caspase-3', 'activity']\n",
      "Counter({'NEG': 2, 'POS': 1, 'NEU': 1})\n",
      "[(over-express, 'POS'), (inhibit, 'NEG'), (decreasing, 'NEG'), (activity, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Taken together, these results suggested that over-express miR-21 could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and caspase-3 activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "full sentence Taken together, these results suggested that over-express <e1>miR-21</e1> could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and <e2>caspase-3</e2> activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Taken together, these results suggested that over-express <e1>miR-21</e1> could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and <e2>caspase-3</e2> activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.', 'full_sentence': 'Taken together, these results suggested that over-express <e1>miR-21</e1> could inhibit TMZ-induced apoptosis in U87MG cells, at least in part, by decreasing Bax/Bcl-2 ratio and <e2>caspase-3</e2> activity, which highlighted the possibility of miR-21 overexpression in the clinical resistance to chemotherapeutic therapy of TMZ.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (58, 64), 'entity_text': 'miR-21'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (169, 178), 'entity_text': 'caspase-3'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d80.s5.p1\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case CDK6\n",
      "add special case microRNA 124a\n",
      "CDK6 7 124a 10\n",
      "(u\"We identified putative miR sites in the CDK6 including microRNA 124a, a brain enriched microRNA.\", 7, 10)\n",
      "geneword CDK6 mirword 124a\n",
      "Conjunctions ( conj )\n",
      "microRNA [a, microRNA]\n",
      "Conjuncts [sites, CDK6]\n",
      "neighbour sites in\n",
      "neighbour sites in\n",
      "Conjunction before fc CDK6\n",
      "fc add children for CDK6\n",
      "fc add children for CDK6\n",
      "Conjuncts [CDK6, 124a]\n",
      "neighbour CDK6 including\n",
      "neighbour CDK6 including\n",
      "Conjunction before fc 124a\n",
      "fc add children for 124a\n",
      "fc add children for 124a\n",
      "fc add children for 124a\n",
      "Conjunctions ( nmod )\n",
      "CDK6 [sites, in, CDK6, including, microRNA, 124a, a, brain, enriched, microRNA]\n",
      "achildren {124a}\n",
      "achildren False\n",
      "SDP sent: We identified putative miR sites in the CDK6 including microRNA 124a, a brain enriched microRNA.\n",
      "SDP     : [(CDK6, 'NOUN', 'nmod', False), (identified, 'VERB', 'ROOT', False), (sites, 'NOUN', 'dobj', False), (CDK6, 'NOUN', 'nmod', False)]\n",
      "0 [We, identified, putative, miR, sites, in, the, CDK6, including, microRNA, 124a, ,, a, brain, enriched, microRNA] 16 0 We identified putative miR sites in the CDK6 including microRNA 124a, a brain enriched microRNA.\n",
      "DeepCheck Conj Subtree\n",
      "[in, the, CDK6, including, microRNA, 124a, ,, a, brain, enriched, microRNA]\n",
      "DeepCheck Conj Subtree\n",
      "[including, microRNA, 124a, ,, a, brain, enriched, microRNA]\n",
      "deepcheck conj verb\n",
      "deepcheck conj verb\n",
      "0 [We, identified, putative, miR, sites, in, the, CDK6, including, microRNA, 124a, ,, a, brain, enriched, microRNA] 16 0 We identified putative miR sites in the CDK6 including microRNA 124a, a brain enriched microRNA.\n",
      "Recognized overlapping entities: {Interval(40, 44, ('GENE_OR_GENE_PRODUCT', 'CDK6'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text We identified putative miR sites in the CDK6 including microRNA 124a , a brain enriched microRNA\n",
      "[We, identified, putative, miR, sites, in, the, CDK6, including, microRNA, 124a, ,, a, brain, enriched, microRNA]\n",
      "TextBetween: We identified putative miR sites in the CDK6 including microRNA 124a , a brain enriched microRNA\n",
      "infer direction from token set [We, identified, putative, miR, sites, in, the, CDK6, including, microRNA, 124a, ,, a, brain, enriched, microRNA]\n",
      "stemTypes2Count: Counter({'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbecba9e2f0>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c598>, {'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(including, 'NEU')]\n",
      "Between Tokens [(including, 'NEU')]\n",
      "counts between final False\n",
      "['in', 'the', 'CDK6', 'including', 'microRNA', '124a', ',', 'a']\n",
      "Counter({'NEU': 1})\n",
      "[(including, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence We identified putative miR sites in the CDK6 including microRNA 124a, a brain enriched microRNA.\n",
      "full sentence We identified putative miR sites in the <e2>CDK6</e2> including <e1>microRNA 124a</e1>, a brain enriched microRNA.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'We identified putative miR sites in the <e2>CDK6</e2> including <e1>microRNA 124a</e1>, a brain enriched microRNA.', 'full_sentence': 'We identified putative miR sites in the <e2>CDK6</e2> including <e1>microRNA 124a</e1>, a brain enriched microRNA.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (40, 44), 'entity_text': 'CDK6'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (55, 68), 'entity_text': 'microRNA 124a'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d80.s9.p1\n",
      "IS:  ('MIR_GENE', 'NEU')\n",
      "SB:  ('MIR_GENE', 'DOWN', 'TRUE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Regulation of <e2>cyclin dependent kinase 6</e2> by <e1>microRNA 124</e1> in medulloblastoma.', 'full_sentence': 'Regulation of <e2>cyclin dependent kinase 6</e2> by <e1>microRNA 124</e1> in medulloblastoma.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (14, 39), 'entity_text': 'cyclin dependent kinase 6'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (43, 55), 'entity_text': 'microRNA 124'}}\n",
      "add special case cyclin dependent kinase 6\n",
      "add special case microRNA 124\n",
      "6 5 124 8\n",
      "(u\"Regulation of cyclin dependent kinase 6 by microRNA 124 in medulloblastoma.\", 5, 8)\n",
      "geneword 6 mirword 124\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [Regulation, kinase]\n",
      "neighbour Regulation of\n",
      "neighbour Regulation of\n",
      "Conjunction before fc kinase\n",
      "fc add children for kinase\n",
      "fc add children for kinase\n",
      "fc add children for kinase\n",
      "fc add children for kinase\n",
      "Conjuncts [Regulation, microRNA]\n",
      "neighbour Regulation of\n",
      "neighbour Regulation of\n",
      "Conjunction before fc microRNA\n",
      "fc add children for microRNA\n",
      "fc add children for microRNA\n",
      "fc add children for microRNA\n",
      "Conjuncts [microRNA, medulloblastoma]\n",
      "neighbour microRNA 124\n",
      "neighbour microRNA 124\n",
      "Conjunction before fc medulloblastoma\n",
      "fc add children for medulloblastoma\n",
      "Conjunctions ( nmod )\n",
      "kinase [Regulation, of, cyclin, dependent, kinase, 6, by, microRNA, 124, in, medulloblastoma]\n",
      "achildren {124}\n",
      "achildren False\n",
      "SDP sent: Regulation of cyclin dependent kinase 6 by microRNA 124 in medulloblastoma.\n",
      "SDP     : [(6, 'NUM', 'nummod', False), (kinase, 'NOUN', 'nmod', False), (Regulation, 'NOUN', 'ROOT', False), (microRNA, 'NOUN', 'nmod', False), (124, 'NUM', 'nummod', False)]\n",
      "0 [Regulation, of, cyclin, dependent, kinase, 6, by, microRNA, 124, in, medulloblastoma] 11 0 Regulation of cyclin dependent kinase 6 by microRNA 124 in medulloblastoma.\n",
      "DeepCheck Conj Subtree\n",
      "[of, cyclin, dependent, kinase, 6]\n",
      "DeepCheck Conj Subtree\n",
      "[by, microRNA, 124, in, medulloblastoma]\n",
      "DeepCheck Conj Subtree\n",
      "[in, medulloblastoma]\n",
      "0 [Regulation, of, cyclin, dependent, kinase, 6, by, microRNA, 124, in, medulloblastoma] 11 0 Regulation of cyclin dependent kinase 6 by microRNA 124 in medulloblastoma.\n",
      "Recognized overlapping entities: {Interval(14, 39, ('GENE_OR_GENE_PRODUCT', 'cyclin dependent kinase 6'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text Regulation of cyclin dependent kinase 6 by microRNA 124 in medulloblastoma\n",
      "[Regulation, of, cyclin, dependent, kinase, 6, by, microRNA, 124, in, medulloblastoma]\n",
      "TextBetween: Regulation of cyclin dependent kinase 6 by microRNA 124 in medulloblastoma\n",
      "infer direction from token set [Regulation, of, cyclin, dependent, kinase, 6, by, microRNA, 124, in, medulloblastoma]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8cae8>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8cb70>, {'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(dependent, 'NEU')]\n",
      "counts between 3 Counter() NEU\n",
      "Between Tokens []\n",
      "counts between final False\n",
      "['dependent', 'kinase', '6', 'by', 'microRNA', '124', 'in', 'medulloblastoma']\n",
      "Counter()\n",
      "[]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(False, {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence Regulation of cyclin dependent kinase 6 by microRNA 124 in medulloblastoma.\n",
      "full sentence Regulation of <e2>cyclin dependent kinase 6</e2> by <e1>microRNA 124</e1> in medulloblastoma.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': False, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'Regulation of <e2>cyclin dependent kinase 6</e2> by <e1>microRNA 124</e1> in medulloblastoma.', 'full_sentence': 'Regulation of <e2>cyclin dependent kinase 6</e2> by <e1>microRNA 124</e1> in medulloblastoma.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (14, 39), 'entity_text': 'cyclin dependent kinase 6'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (43, 55), 'entity_text': 'microRNA 124'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d87.s9.p2\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case miR-128a\n",
      "add special case Tau\n",
      "miR-128a 12 Tau 20\n",
      "(u\"BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons.\", 12, 20)\n",
      "geneword Tau mirword miR-128a\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [levels, cells]\n",
      "neighbour levels in\n",
      "neighbour levels in\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "Conjuncts [control, miR-128a]\n",
      "neighbour control of\n",
      "neighbour control of\n",
      "Conjunction before fc miR-128a\n",
      "fc add children for miR-128a\n",
      "fc add children for miR-128a\n",
      "Conjuncts [levels, neurons]\n",
      "neighbour levels in\n",
      "neighbour levels in\n",
      "Conjunction before fc neurons\n",
      "fc add children for neurons\n",
      "Conunct merge check: cells miR-128a\n",
      "cells are under the physiological\n",
      "Conunct merge check: miR-128a neurons\n",
      "miR-128a, which can tune paired helical filament Tau\n",
      "Conjunctions ( nmod )\n",
      "cells [levels, in, cells]\n",
      "miR-128a [control, of, microRNA, miR-128a]\n",
      "neurons [levels, in, neurons]\n",
      "SDP sent: BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons.\n",
      "SDP     : [(miR-128a, 'NOUN', 'nmod', False), (control, 'NOUN', 'ROOT', False), (miR-128a, 'NOUN', 'nmod', False)]\n",
      "0 [BAG2, levels, in, cells, are, under, the, physiological, control, of, the, microRNA, miR-128a, ,, which, can, tune, paired, helical, filament, Tau, levels, in, neurons] 24 0 BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons.\n",
      "DeepCheck Conj Subtree\n",
      "[in, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, microRNA, miR-128a, ,, which, can, tune, paired, helical, filament, Tau, levels, in, neurons]\n",
      "DeepCheck Conj Subtree\n",
      "[in, neurons]\n",
      "0 [BAG2, levels, in, cells, are, under, the, physiological, control, of, the, microRNA, miR-128a, ,, which, can, tune, paired, helical, filament, Tau, levels, in, neurons] 24 0 BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons.\n",
      "Recognized overlapping entities: {Interval(122, 125, ('GENE_OR_GENE_PRODUCT', 'Tau'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text BAG2 levels in cells are under the physiological control of the microRNA miR-128a , which can tune paired helical filament Tau levels in neurons\n",
      "[BAG2, levels, in, cells, are, under, the, physiological, control, of, the, microRNA, miR-128a, ,, which, can, tune, paired, helical, filament, Tau, levels, in, neurons]\n",
      "TextBetween: BAG2 levels in cells are under the physiological control of the microRNA miR-128a , which can tune paired helical filament Tau levels in neurons\n",
      "infer direction from token set [BAG2, levels, in, cells, are, under, the, physiological, control, of, the, microRNA, miR-128a, ,, which, can, tune, paired, helical, filament, Tau, levels, in, neurons]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c730>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c598>, {'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(levels, 'NEU')]\n",
      "Between Tokens []\n",
      "counts between final False\n",
      "['the', 'microRNA', 'miR-128a', ',', 'which', 'can', 'tune', 'paired', 'helical', 'filament', 'Tau', 'levels', 'in']\n",
      "Counter({'NEU': 1})\n",
      "[(levels, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons.\n",
      "full sentence BAG2 levels in cells are under the physiological control of the microRNA <e1>miR-128a</e1>, which can tune paired helical filament <e2>Tau</e2> levels in neurons.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'BAG2 levels in cells are under the physiological control of the microRNA <e1>miR-128a</e1>, which can tune paired helical filament <e2>Tau</e2> levels in neurons.', 'full_sentence': 'BAG2 levels in cells are under the physiological control of the microRNA <e1>miR-128a</e1>, which can tune paired helical filament <e2>Tau</e2> levels in neurons.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (73, 81), 'entity_text': 'miR-128a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (122, 125), 'entity_text': 'Tau'}}\n",
      "\n",
      "\n",
      "miRNA-corp.d87.s9.p3\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case BAG2\n",
      "add special case miR-128a\n",
      "BAG2 0 miR-128a 12\n",
      "(u\"BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons.\", 0, 12)\n",
      "geneword BAG2 mirword miR-128a\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [levels, cells]\n",
      "neighbour levels in\n",
      "neighbour levels in\n",
      "Conjunction before fc cells\n",
      "fc add children for cells\n",
      "Conjuncts [control, miR-128a]\n",
      "neighbour control of\n",
      "neighbour control of\n",
      "Conjunction before fc miR-128a\n",
      "fc add children for miR-128a\n",
      "fc add children for miR-128a\n",
      "Conjuncts [levels, neurons]\n",
      "neighbour levels in\n",
      "neighbour levels in\n",
      "Conjunction before fc neurons\n",
      "fc add children for neurons\n",
      "Conunct merge check: cells miR-128a\n",
      "cells are under the physiological\n",
      "Conunct merge check: miR-128a neurons\n",
      "miR-128a, which can tune paired helical filament Tau\n",
      "Conjunctions ( nmod )\n",
      "cells [levels, in, cells]\n",
      "miR-128a [control, of, microRNA, miR-128a]\n",
      "neurons [levels, in, neurons]\n",
      "SDP sent: BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons.\n",
      "SDP     : [(BAG2, 'NOUN', 'compound', False), (levels, 'NOUN', 'nsubj', False), (control, 'NOUN', 'ROOT', False), (miR-128a, 'NOUN', 'nmod', False)]\n",
      "0 [BAG2, levels, in, cells, are, under, the, physiological, control, of, the, microRNA, miR-128a, ,, which, can, tune, paired, helical, filament, Tau, levels, in, neurons] 24 0 BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons.\n",
      "DeepCheck Conj Subtree\n",
      "[in, cells]\n",
      "DeepCheck Conj Subtree\n",
      "[of, the, microRNA, miR-128a, ,, which, can, tune, paired, helical, filament, Tau, levels, in, neurons]\n",
      "DeepCheck Conj Subtree\n",
      "[in, neurons]\n",
      "0 [BAG2, levels, in, cells, are, under, the, physiological, control, of, the, microRNA, miR-128a, ,, which, can, tune, paired, helical, filament, Tau, levels, in, neurons] 24 0 BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons.\n",
      "Recognized overlapping entities: {Interval(0, 4, ('GENE_OR_GENE_PRODUCT', 'BAG2'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text BAG2 levels in cells are under the physiological control of the microRNA miR-128a , which can tune paired helical filament Tau levels in neurons\n",
      "[BAG2, levels, in, cells, are, under, the, physiological, control, of, the, microRNA, miR-128a, ,, which, can, tune, paired, helical, filament, Tau, levels, in, neurons]\n",
      "TextBetween: BAG2 levels in cells are under the physiological control of the microRNA miR-128a , which can tune paired helical filament Tau levels in neurons\n",
      "infer direction from token set [BAG2, levels, in, cells, are, under, the, physiological, control, of, the, microRNA, miR-128a, ,, which, can, tune, paired, helical, filament, Tau, levels, in, neurons]\n",
      "stemTypes2Count: Counter()\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fc0b806d158>, {})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 2})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c158>, {'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(levels, 'NEU'), (control, 'NEU')]\n",
      "Between Tokens [(levels, 'NEU'), (control, 'NEU')]\n",
      "counts between final False\n",
      "['BAG2', 'levels', 'in', 'cells', 'are', 'under', 'the', 'physiological', 'control', 'of', 'the', 'microRNA', 'miR-128a', ',', 'which']\n",
      "Counter({'NEU': 2})\n",
      "[(levels, 'NEU'), (control, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence BAG2 levels in cells are under the physiological control of the microRNA miR-128a, which can tune paired helical filament Tau levels in neurons.\n",
      "full sentence <e2>BAG2</e2> levels in cells are under the physiological control of the microRNA <e1>miR-128a</e1>, which can tune paired helical filament Tau levels in neurons.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'NEU', 'interaction_dir': 'GENE_MIR', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e2>BAG2</e2> levels in cells are under the physiological control of the microRNA <e1>miR-128a</e1>, which can tune paired helical filament Tau levels in neurons.', 'full_sentence': '<e2>BAG2</e2> levels in cells are under the physiological control of the microRNA <e1>miR-128a</e1>, which can tune paired helical filament Tau levels in neurons.', 'entity1': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (0, 4), 'entity_text': 'BAG2'}, 'entity2': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (73, 81), 'entity_text': 'miR-128a'}}\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d96.s2.p1\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e1>miR-29b</e1> and miR-125a regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'full_sentence': '<e1>miR-29b</e1> and miR-125a regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (0, 7), 'entity_text': 'miR-29b'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (30, 40), 'entity_text': 'podoplanin'}}\n",
      "add special case miR-29b\n",
      "add special case podoplanin\n",
      "miR-29b 0 podoplanin 4\n",
      "(u\"miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\", 0, 4)\n",
      "geneword podoplanin mirword miR-29b\n",
      "Conjuncts [miR-125a, miR-29b]\n",
      "neighbour miR-125a regulate\n",
      "neighbour miR-125a regulate\n",
      "Conjunction before fc miR-29b\n",
      "Conjuncts [miR-29b, miR-125a]\n",
      "neighbour miR-29b and\n",
      "neighbour miR-29b and\n",
      "Conjunction before fc miR-125a\n",
      "Conjuncts [suppress, regulate]\n",
      "neighbour suppress invasion\n",
      "neighbour suppress invasion\n",
      "Conjunction before fc regulate\n",
      "fc add children for regulate\n",
      "Conjuncts [regulate, suppress]\n",
      "neighbour regulate podoplanin\n",
      "neighbour regulate podoplanin\n",
      "Conjunction before fc suppress\n",
      "fc add children for suppress\n",
      "Conunct merge check: miR-29b regulate\n",
      "miR-125a\n",
      "Conjunctions ( conj )\n",
      "miR-29b [miR-29b, miR-125a]\n",
      "regulate [regulate, podoplanin, suppress, invasion, in, glioblastoma]\n",
      "Conjuncts [invasion, glioblastoma]\n",
      "neighbour invasion in\n",
      "neighbour invasion in\n",
      "Conjunction before fc glioblastoma\n",
      "fc add children for glioblastoma\n",
      "Conjunctions ( nmod )\n",
      "glioblastoma [invasion, in, glioblastoma]\n",
      "SDP sent: miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\n",
      "SDP     : [(miR-29b, 'NOUN', 'nsubj', False), (regulate, 'VERB', 'ROOT', False), (podoplanin, 'NOUN', 'dobj', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-125a]\n",
      "deepcheck connection rule noun found suppress invasion [suppress, invasion, in, glioblastoma]\n",
      "deepcheck conj verb\n",
      "0 [miR-29b, and, miR-125a, regulate, podoplanin, and, suppress, invasion, in, glioblastoma] 10 0 miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\n",
      "DeepCheck Conj Subtree\n",
      "[miR-125a]\n",
      "deepcheck connection rule noun found suppress invasion [suppress, invasion, in, glioblastoma]\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[in, glioblastoma]\n",
      "0 [miR-29b, and, miR-125a, regulate, podoplanin, and, suppress, invasion, in, glioblastoma] 10 0 miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\n",
      "Recognized overlapping entities: {Interval(30, 40, ('GENE_OR_GENE_PRODUCT', 'podoplanin'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma\n",
      "[miR-29b, and, miR-125a, regulate, podoplanin, and, suppress, invasion, in, glioblastoma]\n",
      "TextBetween: miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma\n",
      "infer direction from token set [miR-29b, and, miR-125a, regulate, podoplanin, and, suppress, invasion, in, glioblastoma]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8c9d8>, {'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 1, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe9fe8cf28>, {'gene': Counter({'NEU': 1, 'NEG': 1})})\n",
      "counts between 1 stemTypes2Count [(regulate, 'NEU'), (suppress, 'NEG')]\n",
      "Between Tokens [(regulate, 'NEU')]\n",
      "counts between final False\n",
      "['miR-29b', 'and', 'miR-125a', 'regulate', 'podoplanin', 'and', 'suppress']\n",
      "Counter({'NEU': 1, 'NEG': 1})\n",
      "[(regulate, 'NEU'), (suppress, 'NEG')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\n",
      "full sentence <e1>miR-29b</e1> and miR-125a regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': '<e1>miR-29b</e1> and miR-125a regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'full_sentence': '<e1>miR-29b</e1> and miR-125a regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (0, 7), 'entity_text': 'miR-29b'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (30, 40), 'entity_text': 'podoplanin'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d96.s2.p3\n",
      "IS:  ('MIR_GENE', 'DOWN')\n",
      "SB:  ('MIR_GENE', 'NEU', 'FALSE', 'FALSE')\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'miR-29b and <e1>miR-125a</e1> regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'full_sentence': 'miR-29b and <e1>miR-125a</e1> regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (12, 20), 'entity_text': 'miR-125a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (30, 40), 'entity_text': 'podoplanin'}}\n",
      "add special case miR-125a\n",
      "add special case podoplanin\n",
      "miR-125a 2 podoplanin 4\n",
      "(u\"miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\", 2, 4)\n",
      "geneword podoplanin mirword miR-125a\n",
      "Conjuncts [miR-125a, miR-29b]\n",
      "neighbour miR-125a regulate\n",
      "neighbour miR-125a regulate\n",
      "Conjunction before fc miR-29b\n",
      "Conjuncts [miR-29b, miR-125a]\n",
      "neighbour miR-29b and\n",
      "neighbour miR-29b and\n",
      "Conjunction before fc miR-125a\n",
      "Conjuncts [suppress, regulate]\n",
      "neighbour suppress invasion\n",
      "neighbour suppress invasion\n",
      "Conjunction before fc regulate\n",
      "fc add children for regulate\n",
      "Conjuncts [regulate, suppress]\n",
      "neighbour regulate podoplanin\n",
      "neighbour regulate podoplanin\n",
      "Conjunction before fc suppress\n",
      "fc add children for suppress\n",
      "Conunct merge check: miR-29b regulate\n",
      "miR-125a\n",
      "Conjunctions ( conj )\n",
      "miR-29b [miR-29b, miR-125a]\n",
      "regulate [regulate, podoplanin, suppress, invasion, in, glioblastoma]\n",
      "Conjuncts [invasion, glioblastoma]\n",
      "neighbour invasion in\n",
      "neighbour invasion in\n",
      "Conjunction before fc glioblastoma\n",
      "fc add children for glioblastoma\n",
      "Conjunctions ( nmod )\n",
      "glioblastoma [invasion, in, glioblastoma]\n",
      "SDP sent: miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\n",
      "SDP     : [(miR-125a, 'NOUN', 'conj', False), (miR-29b, 'NOUN', 'nsubj', True), (regulate, 'VERB', 'ROOT', False), (podoplanin, 'NOUN', 'dobj', False)]\n",
      "DeepCheck Conj Subtree\n",
      "[miR-125a]\n",
      "deepcheck connection rule noun found suppress invasion [suppress, invasion, in, glioblastoma]\n",
      "deepcheck conj verb\n",
      "0 [miR-29b, and, miR-125a, regulate, podoplanin, and, suppress, invasion, in, glioblastoma] 10 0 miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\n",
      "DeepCheck Conj Subtree\n",
      "[miR-125a]\n",
      "deepcheck connection rule noun found suppress invasion [suppress, invasion, in, glioblastoma]\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[in, glioblastoma]\n",
      "0 [miR-29b, and, miR-125a, regulate, podoplanin, and, suppress, invasion, in, glioblastoma] 10 0 miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\n",
      "Recognized overlapping entities: {Interval(30, 40, ('GENE_OR_GENE_PRODUCT', 'podoplanin'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma\n",
      "[miR-29b, and, miR-125a, regulate, podoplanin, and, suppress, invasion, in, glioblastoma]\n",
      "TextBetween: miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma\n",
      "infer direction from token set [miR-29b, and, miR-125a, regulate, podoplanin, and, suppress, invasion, in, glioblastoma]\n",
      "stemTypes2Count: Counter({'NEG': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70510>, {'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'NEU': 1, 'NEG': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62e18>, {'gene': Counter({'NEG': 1})})\n",
      "counts between 1 stemTypes2Count [(regulate, 'NEU'), (suppress, 'NEG')]\n",
      "Between Tokens [(regulate, 'NEU')]\n",
      "counts between final False\n",
      "['miR-29b', 'and', 'miR-125a', 'regulate', 'podoplanin', 'and', 'suppress']\n",
      "Counter({'NEU': 1, 'NEG': 1})\n",
      "[(regulate, 'NEU'), (suppress, 'NEG')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence miR-29b and miR-125a regulate podoplanin and suppress invasion in glioblastoma.\n",
      "full sentence miR-29b and <e1>miR-125a</e1> regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'DOWN', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'miR-29b and <e1>miR-125a</e1> regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'full_sentence': 'miR-29b and <e1>miR-125a</e1> regulate <e2>podoplanin</e2> and suppress invasion in glioblastoma.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (12, 20), 'entity_text': 'miR-125a'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (30, 40), 'entity_text': 'podoplanin'}}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "miRNA-corp.d98.s7.p0\n",
      "IS: True\n",
      "SB: False\n",
      "Repeat Analysis\n",
      "add special case microRNA-433\n",
      "add special case FGF20\n",
      "microRNA-433 17 FGF20 22\n",
      "(u\"We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\", 17, 22)\n",
      "geneword FGF20 mirword microRNA-433\n",
      "Conjunctions ( conj )\n",
      "Conjuncts [show, assays]\n",
      "neighbour show in\n",
      "neighbour show in\n",
      "Conjunction before fc assays\n",
      "fc add children for assays\n",
      "fc add children for assays\n",
      "fc add children for assays\n",
      "Conjuncts [allele, rs12720208]\n",
      "neighbour allele for\n",
      "neighbour allele for\n",
      "Conjunction before fc rs12720208\n",
      "fc add children for rs12720208\n",
      "Conjuncts [site, microRNA-433]\n",
      "neighbour site for\n",
      "neighbour site for\n",
      "Conjunction before fc microRNA-433\n",
      "fc add children for microRNA-433\n",
      "Conjuncts [disrupts, translation]\n",
      "neighbour disrupts a\n",
      "neighbour disrupts a\n",
      "Conjunction before fc translation\n",
      "fc add children for translation\n",
      "fc add children for translation\n",
      "Conjuncts [translation, FGF20]\n",
      "neighbour translation of\n",
      "neighbour translation of\n",
      "Conjunction before fc FGF20\n",
      "fc add children for FGF20\n",
      "Conunct merge check: assays rs12720208\n",
      "assays that the risk\n",
      "Conunct merge check: rs12720208 translation\n",
      "rs12720208\n",
      "Conunct merge check: translation microRNA-433\n",
      "\n",
      "Double Conunction: Empty text bewteen\n",
      "Conjunctions ( nmod )\n",
      "assays [show, in, several, functional, assays]\n",
      "rs12720208 [allele, for, rs12720208]\n",
      "microRNA-433 [site, for, microRNA-433]\n",
      "translation [disrupts, increasing, translation, of, FGF20]\n",
      "SDP sent: We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "SDP     : [(microRNA-433, 'NOUN', 'nmod', False), (site, 'NOUN', 'dobj', False), (disrupts, 'VERB', 'ccomp', False), (translation, 'NOUN', 'nmod', False), (FGF20, 'NOUN', 'nmod', False)]\n",
      "SDP DOBJ RULE\n",
      "0 [We, show, in, several, functional, assays] 6 0 We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "1 [the, risk, allele, for, rs12720208, disrupts, a, binding, site, for, microRNA-433, ,, increasing, translation, of, FGF20, in, vitro, and, in, vivo] 21 42 the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "DeepCheck Conj Subtree\n",
      "[in, several, functional, assays]\n",
      "DeepCheck Conj Subtree\n",
      "[for, rs12720208]\n",
      "DeepCheck Conj Subtree\n",
      "[for, microRNA-433]\n",
      "DeepCheck Conj Subtree\n",
      "[increasing, translation, of, FGF20, in, vitro, and, in, vivo]\n",
      "deepcheck conj verb\n",
      "deepcheck conj verb\n",
      "DeepCheck Conj Subtree\n",
      "[of, FGF20]\n",
      "0 [We, show, in, several, functional, assays] 6 0 We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "1 [the, risk, allele, for, rs12720208, disrupts, a, binding, site, for, microRNA-433, ,, increasing, translation, of, FGF20, in, vitro, and, in, vivo] 21 42 the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "Recognized overlapping entities: {Interval(141, 146, ('GENE_OR_GENE_PRODUCT', 'FGF20'))}\n",
      "passive? False\n",
      "Counter()\n",
      "Compartment Text the risk allele for rs12720208 disrupts a binding site for microRNA-433 , increasing translation of FGF20 in vitro and in vivo\n",
      "[the, risk, allele, for, rs12720208, disrupts, a, binding, site, for, microRNA-433, ,, increasing, translation, of, FGF20, in, vitro, and, in, vivo]\n",
      "TextBetween: the risk allele for rs12720208 disrupts a binding site for microRNA-433 , increasing translation of FGF20 in vitro and in vivo\n",
      "infer direction from token set [the, risk, allele, for, rs12720208, disrupts, a, binding, site, for, microRNA-433, ,, increasing, translation, of, FGF20, in, vitro, and, in, vivo]\n",
      "stemTypes2Count: Counter({'POS': 1})\n",
      "mirgene2stem: defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbe99a62d90>, {'mir': Counter({'POS': 1})})\n",
      "counts between 1 stemTypes2Count Counter({'POS': 1, 'NEU': 1})\n",
      "counts between 1 stemTypes2Count defaultdict(<function SentenceRelationClassifier.countStems.<locals>.<lambda> at 0x7fbf8ed70950>, {'mir': Counter({'POS': 1}), 'gene': Counter({'NEU': 1})})\n",
      "counts between 1 stemTypes2Count [(increasing, 'POS'), (translation, 'NEU')]\n",
      "Between Tokens [(increasing, 'POS'), (translation, 'NEU')]\n",
      "counts between final False\n",
      "['site', 'for', 'microRNA-433', ',', 'increasing', 'translation', 'of', 'FGF20', 'in', 'vitro']\n",
      "Counter({'POS': 1, 'NEU': 1})\n",
      "[(increasing, 'POS'), (translation, 'NEU')]\n",
      "Return After Infer\n",
      "{'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}\n",
      "(True, {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}})\n",
      " hit sentence We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for microRNA-433, increasing translation of FGF20 in vitro and in vivo.\n",
      "full sentence We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for <e1>microRNA-433</e1>, increasing translation of <e2>FGF20</e2> in vitro and in vivo.\n",
      "{'accept_relation': True, 'check_results': {'conj': True, 'conj_nmod': True, 'sdp': True, 'compartment': True, 'compartment_nmod': True, 'context': True, 'entity': True, 'passive': False, 'negation': False, 'classification': {'regulation_dir': 'UP', 'interaction_dir': 'MIR_GENE', 'passive': False, 'reg_detect': 'counts between', 'reg_detect_major': 'counts'}}, 'tagged_sent': 'We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for <e1>microRNA-433</e1>, increasing translation of <e2>FGF20</e2> in vitro and in vivo.', 'full_sentence': 'We show in several functional assays that the risk allele for rs12720208 disrupts a binding site for <e1>microRNA-433</e1>, increasing translation of <e2>FGF20</e2> in vitro and in vivo.', 'entity1': {'entity_type': 'mirna', 'entity_type_token': 'e1', 'entity_location': (101, 113), 'entity_text': 'microRNA-433'}, 'entity2': {'entity_type': 'gene', 'entity_type_token': 'e2', 'entity_location': (141, 146), 'entity_text': 'FGF20'}}\n",
      "Total:      232\n",
      "Correct:    205 0.8836206896551724\n",
      "Incorrect:  27 0.11637931034482758\n",
      "classes Counter()\n",
      "precision 0.9349593495934959\n",
      "recall 0.8582089552238806\n",
      "specificity 0.9183673469387755\n",
      "f1 0.8949416342412452\n",
      "relationFalseCounter Counter({'context': 5, 'compartment': 4, 'compartment_nmod': 4, 'conj_nmod': 3, 'conj': 2})\n",
      "Correct classified:      110 0.47413793103448276 0.8943089430894309\n",
      "Incorrect classified:    13 0.05603448275862069 0.10569105691056911\n",
      "Random classified:  42 0.1810344827586207 0.34146341463414637\n",
      "Counter({'counts between': 9, 'between mir gene': 1, 'counts opp': 1, 'counts between equal': 1, 'between gene mir': 1})\n",
      "Correct interaction&classified:      107 0.46120689655172414 0.8699186991869918\n",
      "Classified by:  Counter({'counts between': 62, 'between gene mir': 10, 'counts opp': 8, 'counts between equal': 8, 'compartment mir gene by': 8, 'between mir gene': 6, 'compartment mir gene': 6, 'compartment gene mir': 5, 'between mir gene reg corr': 3, 'return': 3, 'compartment gene mir by': 3, 'counts between alternating': 1})\n",
      "[[[119   0]\n",
      "  [  0   4]]\n",
      "\n",
      " [[122   0]\n",
      "  [  0   1]]\n",
      "\n",
      " [[ 75   6]\n",
      "  [  5  37]]\n",
      "\n",
      " [[ 46   5]\n",
      "  [  4  68]]\n",
      "\n",
      " [[121   2]\n",
      "  [  0   0]]\n",
      "\n",
      " [[119   0]\n",
      "  [  4   0]]]\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      " GENE_MIR_NEU       1.00      1.00      1.00         4\n",
      "  GENE_MIR_UP       1.00      1.00      1.00         1\n",
      "MIR_GENE_DOWN       0.86      0.88      0.87        42\n",
      " MIR_GENE_NEU       0.93      0.94      0.94        72\n",
      "  MIR_GENE_UP       0.00      0.00      0.00         0\n",
      "        NA_NA       0.00      0.00      0.00         4\n",
      "\n",
      "     accuracy                           0.89       123\n",
      "    macro avg       0.63      0.64      0.63       123\n",
      " weighted avg       0.88      0.89      0.89       123\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/j/joppich/.local/lib/python3.6/site-packages/sklearn/metrics/_classification.py:1248: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/home/j/joppich/.local/lib/python3.6/site-packages/sklearn/metrics/_classification.py:1248: UndefinedMetricWarning: Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/home/j/joppich/.local/lib/python3.6/site-packages/sklearn/metrics/_classification.py:1248: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/home/j/joppich/.local/lib/python3.6/site-packages/sklearn/metrics/_classification.py:1248: UndefinedMetricWarning: Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/home/j/joppich/.local/lib/python3.6/site-packages/sklearn/metrics/_classification.py:1248: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/home/j/joppich/.local/lib/python3.6/site-packages/sklearn/metrics/_classification.py:1248: UndefinedMetricWarning: Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/home/j/joppich/.local/lib/python3.6/site-packages/sklearn/metrics/_classification.py:1248: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/home/j/joppich/.local/lib/python3.6/site-packages/sklearn/metrics/_classification.py:1248: UndefinedMetricWarning: Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/home/j/joppich/.local/lib/python3.6/site-packages/sklearn/metrics/_classification.py:1248: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/home/j/joppich/.local/lib/python3.6/site-packages/sklearn/metrics/_classification.py:1248: UndefinedMetricWarning: Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/home/j/joppich/.local/lib/python3.6/site-packages/sklearn/metrics/_classification.py:1248: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/home/j/joppich/.local/lib/python3.6/site-packages/sklearn/metrics/_classification.py:1248: UndefinedMetricWarning: Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    }
   ],
   "source": [
    "%aimport MirGeneRelCheck\n",
    "\n",
    "relClassifier = SentenceRelationClassifier(relationsFile)\n",
    "\n",
    "test_results_original = runCheck(-1, \"TEST\", original=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
