{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['/mnt/raidbio/extproj/projekte/textmining/miRExplore/python/synonymes/../', '/usr/lib64/python311.zip', '/usr/lib64/python3.11', '/usr/lib64/python3.11/lib-dynload', '', '/home/j/joppich/.local/lib/python3.11/site-packages', '/usr/lib64/python3.11/site-packages', '/usr/lib64/python3.11/_import_failed', '/usr/lib/python3.11/site-packages', './../', './../', './../', './../', './../', './../']\n"
     ]
    }
   ],
   "source": [
    "from gtfparse import read_gtf\n",
    "import polars as pl\n",
    "from intervaltree import Interval, IntervalTree\n",
    "import pandas as pd\n",
    "pd.set_option('display.max_rows', 60)\n",
    "from collections import defaultdict\n",
    "\n",
    "import os, sys\n",
    "\n",
    "sys.path.append(\"./../\")\n",
    "print(sys.path)\n",
    "\n",
    "from synonymes.mirnaID import miRNA, miRNASynonymeTYPE, miRNAPART\n",
    "from Bio import SeqIO\n",
    "from io import StringIO\n",
    "import base64, gzip\n",
    "import json\n",
    "\n",
    "from Bio.Seq import Seq \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_fasta_seq = {}\n",
    "\n",
    "with gzip.open(\"/mnt/extproj/projekte/textmining/genome_annotations/GRCh37.p13.genome.fa.gz\", \"rt\") as fin:\n",
    "    for seq_record in SeqIO.parse(fin, \"fasta\"):\n",
    "        all_fasta_seq[seq_record.id] = str(seq_record.seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['chr1',\n",
       " 'chr2',\n",
       " 'chr3',\n",
       " 'chr4',\n",
       " 'chr5',\n",
       " 'chr6',\n",
       " 'chr7',\n",
       " 'chr8',\n",
       " 'chr9',\n",
       " 'chr10']"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[x for x in all_fasta_seq][:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:root:Extracted GTF attributes: ['gene_id', 'transcript_id', 'gene_type', 'gene_status', 'gene_name', 'transcript_type', 'transcript_status', 'transcript_name', 'level', 'havana_gene', 'tag', 'havana_transcript', 'exon_number', 'exon_id', 'ont', 'protein_id', 'ccdsid']\n"
     ]
    }
   ],
   "source": [
    "df = read_gtf(\"/mnt/extproj/projekte/textmining/genome_annotations/gencode.v19.annotation.gtf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><style>\n",
       ".dataframe > thead > tr,\n",
       ".dataframe > tbody > tr {\n",
       "  text-align: right;\n",
       "  white-space: pre-wrap;\n",
       "}\n",
       "</style>\n",
       "<small>shape: (5, 25)</small><table border=\"1\" class=\"dataframe\"><thead><tr><th>seqname</th><th>source</th><th>feature</th><th>start</th><th>end</th><th>score</th><th>strand</th><th>frame</th><th>gene_id</th><th>transcript_id</th><th>gene_type</th><th>gene_status</th><th>gene_name</th><th>transcript_type</th><th>transcript_status</th><th>transcript_name</th><th>level</th><th>havana_gene</th><th>tag</th><th>havana_transcript</th><th>exon_number</th><th>exon_id</th><th>ont</th><th>protein_id</th><th>ccdsid</th></tr><tr><td>cat</td><td>cat</td><td>cat</td><td>i64</td><td>i64</td><td>f32</td><td>cat</td><td>i64</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td><td>str</td></tr></thead><tbody><tr><td>&quot;chr1&quot;</td><td>&quot;HAVANA&quot;</td><td>&quot;gene&quot;</td><td>11869</td><td>14412</td><td>null</td><td>&quot;+&quot;</td><td>0</td><td>&quot;ENSG00000223972.4&quot;</td><td>&quot;ENSG00000223972.4&quot;</td><td>&quot;pseudogene&quot;</td><td>&quot;KNOWN&quot;</td><td>&quot;DDX11L1&quot;</td><td>&quot;pseudogene&quot;</td><td>&quot;KNOWN&quot;</td><td>&quot;DDX11L1&quot;</td><td>&quot;2&quot;</td><td>&quot;OTTHUMG00000000961.2&quot;</td><td>&quot;&quot;</td><td>&quot;&quot;</td><td>&quot;&quot;</td><td>&quot;&quot;</td><td>&quot;&quot;</td><td>&quot;&quot;</td><td>&quot;&quot;</td></tr><tr><td>&quot;chr1&quot;</td><td>&quot;HAVANA&quot;</td><td>&quot;transcript&quot;</td><td>11869</td><td>14409</td><td>null</td><td>&quot;+&quot;</td><td>0</td><td>&quot;ENSG00000223972.4&quot;</td><td>&quot;ENST00000456328.2&quot;</td><td>&quot;pseudogene&quot;</td><td>&quot;KNOWN&quot;</td><td>&quot;DDX11L1&quot;</td><td>&quot;processed_transcript&quot;</td><td>&quot;KNOWN&quot;</td><td>&quot;DDX11L1-002&quot;</td><td>&quot;2&quot;</td><td>&quot;OTTHUMG00000000961.2&quot;</td><td>&quot;basic&quot;</td><td>&quot;OTTHUMT00000362751.1&quot;</td><td>&quot;&quot;</td><td>&quot;&quot;</td><td>&quot;&quot;</td><td>&quot;&quot;</td><td>&quot;&quot;</td></tr><tr><td>&quot;chr1&quot;</td><td>&quot;HAVANA&quot;</td><td>&quot;exon&quot;</td><td>11869</td><td>12227</td><td>null</td><td>&quot;+&quot;</td><td>0</td><td>&quot;ENSG00000223972.4&quot;</td><td>&quot;ENST00000456328.2&quot;</td><td>&quot;pseudogene&quot;</td><td>&quot;KNOWN&quot;</td><td>&quot;DDX11L1&quot;</td><td>&quot;processed_transcript&quot;</td><td>&quot;KNOWN&quot;</td><td>&quot;DDX11L1-002&quot;</td><td>&quot;2&quot;</td><td>&quot;OTTHUMG00000000961.2&quot;</td><td>&quot;basic&quot;</td><td>&quot;OTTHUMT00000362751.1&quot;</td><td>&quot;1&quot;</td><td>&quot;ENSE00002234944.1&quot;</td><td>&quot;&quot;</td><td>&quot;&quot;</td><td>&quot;&quot;</td></tr><tr><td>&quot;chr1&quot;</td><td>&quot;HAVANA&quot;</td><td>&quot;exon&quot;</td><td>12613</td><td>12721</td><td>null</td><td>&quot;+&quot;</td><td>0</td><td>&quot;ENSG00000223972.4&quot;</td><td>&quot;ENST00000456328.2&quot;</td><td>&quot;pseudogene&quot;</td><td>&quot;KNOWN&quot;</td><td>&quot;DDX11L1&quot;</td><td>&quot;processed_transcript&quot;</td><td>&quot;KNOWN&quot;</td><td>&quot;DDX11L1-002&quot;</td><td>&quot;2&quot;</td><td>&quot;OTTHUMG00000000961.2&quot;</td><td>&quot;basic&quot;</td><td>&quot;OTTHUMT00000362751.1&quot;</td><td>&quot;2&quot;</td><td>&quot;ENSE00003582793.1&quot;</td><td>&quot;&quot;</td><td>&quot;&quot;</td><td>&quot;&quot;</td></tr><tr><td>&quot;chr1&quot;</td><td>&quot;HAVANA&quot;</td><td>&quot;exon&quot;</td><td>13221</td><td>14409</td><td>null</td><td>&quot;+&quot;</td><td>0</td><td>&quot;ENSG00000223972.4&quot;</td><td>&quot;ENST00000456328.2&quot;</td><td>&quot;pseudogene&quot;</td><td>&quot;KNOWN&quot;</td><td>&quot;DDX11L1&quot;</td><td>&quot;processed_transcript&quot;</td><td>&quot;KNOWN&quot;</td><td>&quot;DDX11L1-002&quot;</td><td>&quot;2&quot;</td><td>&quot;OTTHUMG00000000961.2&quot;</td><td>&quot;basic&quot;</td><td>&quot;OTTHUMT00000362751.1&quot;</td><td>&quot;3&quot;</td><td>&quot;ENSE00002312635.1&quot;</td><td>&quot;&quot;</td><td>&quot;&quot;</td><td>&quot;&quot;</td></tr></tbody></table></div>"
      ],
      "text/plain": [
       "shape: (5, 25)\n",
       "┌─────────┬────────┬────────────┬───────┬───┬───────────────────┬─────┬────────────┬────────┐\n",
       "│ seqname ┆ source ┆ feature    ┆ start ┆ … ┆ exon_id           ┆ ont ┆ protein_id ┆ ccdsid │\n",
       "│ ---     ┆ ---    ┆ ---        ┆ ---   ┆   ┆ ---               ┆ --- ┆ ---        ┆ ---    │\n",
       "│ cat     ┆ cat    ┆ cat        ┆ i64   ┆   ┆ str               ┆ str ┆ str        ┆ str    │\n",
       "╞═════════╪════════╪════════════╪═══════╪═══╪═══════════════════╪═════╪════════════╪════════╡\n",
       "│ chr1    ┆ HAVANA ┆ gene       ┆ 11869 ┆ … ┆                   ┆     ┆            ┆        │\n",
       "│ chr1    ┆ HAVANA ┆ transcript ┆ 11869 ┆ … ┆                   ┆     ┆            ┆        │\n",
       "│ chr1    ┆ HAVANA ┆ exon       ┆ 11869 ┆ … ┆ ENSE00002234944.1 ┆     ┆            ┆        │\n",
       "│ chr1    ┆ HAVANA ┆ exon       ┆ 12613 ┆ … ┆ ENSE00003582793.1 ┆     ┆            ┆        │\n",
       "│ chr1    ┆ HAVANA ┆ exon       ┆ 13221 ┆ … ┆ ENSE00002312635.1 ┆     ┆            ┆        │\n",
       "└─────────┴────────┴────────────┴───────┴───┴───────────────────┴─────┴────────────┴────────┘"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><style>\n",
       ".dataframe > thead > tr,\n",
       ".dataframe > tbody > tr {\n",
       "  text-align: right;\n",
       "  white-space: pre-wrap;\n",
       "}\n",
       "</style>\n",
       "<small>shape: (112, 7)</small><table border=\"1\" class=\"dataframe\"><thead><tr><th>seqname</th><th>start</th><th>end</th><th>gene_id</th><th>gene_type</th><th>gene_name</th><th>feature</th></tr><tr><td>cat</td><td>i64</td><td>i64</td><td>str</td><td>str</td><td>str</td><td>cat</td></tr></thead><tbody><tr><td>&quot;chr5&quot;</td><td>102364591</td><td>102364814</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102285241</td><td>102285321</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102200948</td><td>102203013</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;transcript&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102201183</td><td>102201214</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102364591</td><td>102364896</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102343157</td><td>102364971</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;transcript&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102285241</td><td>102285277</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102201527</td><td>102201984</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102090487</td><td>102091052</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102360838</td><td>102361038</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102364591</td><td>102365417</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102325475</td><td>102342563</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;transcript&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102201441</td><td>102201988</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102338737</td><td>102338853</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102201523</td><td>102209508</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;transcript&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102208997</td><td>102209508</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102343161</td><td>102343361</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102201527</td><td>102201784</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102325976</td><td>102326105</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102089685</td><td>102366809</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;gene&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102143761</td><td>102143935</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102343246</td><td>102363926</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;transcript&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102091047</td><td>102201974</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;transcript&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102201430</td><td>102366809</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;transcript&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102201820</td><td>102201988</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td><td>&hellip;</td></tr><tr><td>&quot;chr5&quot;</td><td>102343273</td><td>102364896</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;transcript&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102364591</td><td>102366809</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102089685</td><td>102089794</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102364591</td><td>102364948</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102202977</td><td>102203046</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102343157</td><td>102343361</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102345455</td><td>102345570</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102361083</td><td>102361083</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102201527</td><td>102201988</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102295579</td><td>102295763</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102325475</td><td>102326105</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102173925</td><td>102174293</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102143767</td><td>102201896</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;transcript&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102202291</td><td>102202420</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102363886</td><td>102363942</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102249638</td><td>102249725</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102090371</td><td>102201616</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;transcript&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102282541</td><td>102282580</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102200241</td><td>102200271</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102201820</td><td>102364869</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;transcript&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102295701</td><td>102295763</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102295701</td><td>102310541</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;transcript&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102360835</td><td>102361038</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102364591</td><td>102365215</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr><tr><td>&quot;chr5&quot;</td><td>102201761</td><td>102201988</td><td>&quot;ENSG00000145730.16&quot;</td><td>&quot;protein_coding&quot;</td><td>&quot;PAM&quot;</td><td>&quot;exon&quot;</td></tr></tbody></table></div>"
      ],
      "text/plain": [
       "shape: (112, 7)\n",
       "┌─────────┬───────────┬───────────┬────────────────────┬────────────────┬───────────┬────────────┐\n",
       "│ seqname ┆ start     ┆ end       ┆ gene_id            ┆ gene_type      ┆ gene_name ┆ feature    │\n",
       "│ ---     ┆ ---       ┆ ---       ┆ ---                ┆ ---            ┆ ---       ┆ ---        │\n",
       "│ cat     ┆ i64       ┆ i64       ┆ str                ┆ str            ┆ str       ┆ cat        │\n",
       "╞═════════╪═══════════╪═══════════╪════════════════════╪════════════════╪═══════════╪════════════╡\n",
       "│ chr5    ┆ 102364591 ┆ 102364814 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102285241 ┆ 102285321 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102200948 ┆ 102203013 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ transcript │\n",
       "│ chr5    ┆ 102201183 ┆ 102201214 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102364591 ┆ 102364896 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102343157 ┆ 102364971 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ transcript │\n",
       "│ chr5    ┆ 102285241 ┆ 102285277 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102201527 ┆ 102201984 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102090487 ┆ 102091052 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102360838 ┆ 102361038 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102364591 ┆ 102365417 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102325475 ┆ 102342563 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ transcript │\n",
       "│ chr5    ┆ 102201441 ┆ 102201988 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102338737 ┆ 102338853 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102201523 ┆ 102209508 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ transcript │\n",
       "│ chr5    ┆ 102208997 ┆ 102209508 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102343161 ┆ 102343361 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102201527 ┆ 102201784 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102325976 ┆ 102326105 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102089685 ┆ 102366809 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ gene       │\n",
       "│ chr5    ┆ 102143761 ┆ 102143935 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102343246 ┆ 102363926 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ transcript │\n",
       "│ chr5    ┆ 102091047 ┆ 102201974 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ transcript │\n",
       "│ chr5    ┆ 102201430 ┆ 102366809 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ transcript │\n",
       "│ chr5    ┆ 102201820 ┆ 102201988 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ …       ┆ …         ┆ …         ┆ …                  ┆ …              ┆ …         ┆ …          │\n",
       "│ chr5    ┆ 102343273 ┆ 102364896 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ transcript │\n",
       "│ chr5    ┆ 102364591 ┆ 102366809 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102089685 ┆ 102089794 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102364591 ┆ 102364948 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102202977 ┆ 102203046 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102343157 ┆ 102343361 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102345455 ┆ 102345570 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102361083 ┆ 102361083 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102201527 ┆ 102201988 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102295579 ┆ 102295763 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102325475 ┆ 102326105 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102173925 ┆ 102174293 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102143767 ┆ 102201896 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ transcript │\n",
       "│ chr5    ┆ 102202291 ┆ 102202420 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102363886 ┆ 102363942 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102249638 ┆ 102249725 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102090371 ┆ 102201616 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ transcript │\n",
       "│ chr5    ┆ 102282541 ┆ 102282580 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102200241 ┆ 102200271 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102201820 ┆ 102364869 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ transcript │\n",
       "│ chr5    ┆ 102295701 ┆ 102295763 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102295701 ┆ 102310541 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ transcript │\n",
       "│ chr5    ┆ 102360835 ┆ 102361038 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102364591 ┆ 102365215 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "│ chr5    ┆ 102201761 ┆ 102201988 ┆ ENSG00000145730.16 ┆ protein_coding ┆ PAM       ┆ exon       │\n",
       "└─────────┴───────────┴───────────┴────────────────────┴────────────────┴───────────┴────────────┘"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pl.Config.set_tbl_rows(50)\n",
    "df.filter(pl.col(\"gene_name\") == \"PAM\").filter(pl.col(\"feature\").is_in([\"gene\", \"transcript\", \"exon\"])).select([\"seqname\", \"start\", \"end\", \"gene_id\", \"gene_type\", \"gene_name\",\"feature\"]).unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>gene_mir</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>start2</th>\n",
       "      <th>end2</th>\n",
       "      <th>score8</th>\n",
       "      <th>score9</th>\n",
       "      <th>score10</th>\n",
       "      <th>score11</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>900599</td>\n",
       "      <td>900606</td>\n",
       "      <td>KLHL17:miR-299-3p</td>\n",
       "      <td>86</td>\n",
       "      <td>+</td>\n",
       "      <td>900599</td>\n",
       "      <td>900606</td>\n",
       "      <td>255,0,0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>900605</td>\n",
       "      <td>900612</td>\n",
       "      <td>KLHL17:miR-124-3p.1</td>\n",
       "      <td>88</td>\n",
       "      <td>+</td>\n",
       "      <td>900605</td>\n",
       "      <td>900612</td>\n",
       "      <td>30,144,255</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>900605</td>\n",
       "      <td>900613</td>\n",
       "      <td>KLHL17:miR-124-3p.2/506-3p</td>\n",
       "      <td>96</td>\n",
       "      <td>+</td>\n",
       "      <td>900605</td>\n",
       "      <td>900613</td>\n",
       "      <td>128,0,128</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>900933</td>\n",
       "      <td>900941</td>\n",
       "      <td>KLHL17:miR-19-3p</td>\n",
       "      <td>91</td>\n",
       "      <td>+</td>\n",
       "      <td>900933</td>\n",
       "      <td>900941</td>\n",
       "      <td>128,0,128</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>901054</td>\n",
       "      <td>901061</td>\n",
       "      <td>KLHL17:miR-137</td>\n",
       "      <td>84</td>\n",
       "      <td>+</td>\n",
       "      <td>901054</td>\n",
       "      <td>901061</td>\n",
       "      <td>255,0,0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr   start     end                    gene_mir  score strand  start2  \\\n",
       "0  chr1  900599  900606           KLHL17:miR-299-3p     86      +  900599   \n",
       "1  chr1  900605  900612         KLHL17:miR-124-3p.1     88      +  900605   \n",
       "2  chr1  900605  900613  KLHL17:miR-124-3p.2/506-3p     96      +  900605   \n",
       "3  chr1  900933  900941            KLHL17:miR-19-3p     91      +  900933   \n",
       "4  chr1  901054  901061              KLHL17:miR-137     84      +  901054   \n",
       "\n",
       "     end2      score8  score9 score10 score11  \n",
       "0  900606     255,0,0       1       7       0  \n",
       "1  900612  30,144,255       1       7       0  \n",
       "2  900613   128,0,128       1       8       0  \n",
       "3  900941   128,0,128       1       8       0  \n",
       "4  901061     255,0,0       1       7       0  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mirBSDF = pd.read_csv(\"../../../genome_annotations/Predicted_Target_Locations.default_predictions.hg19.bed\", sep=\"\\t\",header=None)\n",
    "mirBSDF.columns = (\"chr\", \"start\", \"end\", \"gene_mir\", \"score\", \"strand\", \"start2\", \"end2\", \"score8\", \"score9\", \"score10\", \"score11\")\n",
    "mirBSDF.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "            miR family  Seed+m8  Species ID       MiRBase ID  \\\n",
      "33  let-7-3p/miR-98-3p  UAUACAA       10090  mmu-let-7a-1-3p   \n",
      "34  let-7-3p/miR-98-3p  UAUACAA       10090    mmu-let-7b-3p   \n",
      "35  let-7-3p/miR-98-3p  UAUACAA       10090  mmu-let-7c-2-3p   \n",
      "36  let-7-3p/miR-98-3p  UAUACAA       10090  mmu-let-7f-1-3p   \n",
      "37  let-7-3p/miR-98-3p  UAUACAA       10090    mmu-miR-98-3p   \n",
      "\n",
      "           Mature sequence  Family Conservation? MiRBase Accession  \n",
      "33  CUAUACAAUCUACUGUCUUUCC                    -1      MIMAT0004620  \n",
      "34  CUAUACAACCUACUGCCUUCCC                    -1      MIMAT0004621  \n",
      "35  CUAUACAAUCUACUGUCUUUCC                    -1      MIMAT0005439  \n",
      "36  CUAUACAAUCUAUUGCCUUCCC                    -1      MIMAT0004623  \n",
      "37  CUAUACAACUUACUACUUUCCU                    -1      MIMAT0017023  \n"
     ]
    }
   ],
   "source": [
    "mirFamilies = pd.read_csv(\"../../../genome_annotations/miR_Family_Info.txt\", sep=\"\\t\")\n",
    "mirFamilies = mirFamilies[mirFamilies[\"Species ID\"].isin([9606, 10090])]\n",
    "print(mirFamilies.head())\n",
    "\n",
    "mirFamily2miRNA = defaultdict(set)\n",
    "\n",
    "for ri, row in mirFamilies.iterrows():\n",
    "    mirFamily2miRNA[ row[\"miR family\"] ].add( row[\"MiRBase ID\"] )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene2mir2locations = defaultdict(lambda : defaultdict(set))\n",
    "\n",
    "for ri, row in mirBSDF.iterrows():\n",
    "    loc = (row[\"chr\"], row[\"start\"], row[\"end\"], row[\"strand\"])\n",
    "        \n",
    "    gene = row[\"gene_mir\"].split(\":\")[0]\n",
    "    mirFamily = row[\"gene_mir\"].split(\":\")[1]\n",
    "    \n",
    "    mirnas = mirFamily2miRNA[mirFamily]\n",
    "        \n",
    "\n",
    "    for mirna in mirnas:\n",
    "        \n",
    "        entryName = \"-\".join(mirna.split(\"-\")[1:3]).replace(\"mir\", \"miR\")\n",
    "        \n",
    "        gene2mir2locations[gene][entryName].add(loc)\n",
    "        \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(set,\n",
       "            {'miR-873': {('chr21', 16333589, 16333597, '-')},\n",
       "             'miR-30e': {('chr21', 16333597, 16333604, '-')},\n",
       "             'miR-30d': {('chr21', 16333597, 16333604, '-')},\n",
       "             'miR-30b': {('chr21', 16333597, 16333604, '-')},\n",
       "             'miR-30a': {('chr21', 16333597, 16333604, '-')},\n",
       "             'miR-30c': {('chr21', 16333597, 16333604, '-')},\n",
       "             'miR-582': {('chr21', 16333649, 16333656, '-')},\n",
       "             'miR-489': {('chr21', 16333659, 16333666, '-'),\n",
       "              ('chr21', 16336852, 16336859, '-')},\n",
       "             'miR-330': {('chr21', 16333916, 16333924, '-'),\n",
       "              ('chr21', 16333918, 16333925, '-')},\n",
       "             'miR-6835': {('chr21', 16333918, 16333925, '-')},\n",
       "             'miR-140': {('chr21', 16333944, 16333951, '-'),\n",
       "              ('chr21', 16336136, 16336144, '-')},\n",
       "             'miR-200c': {('chr21', 16333970, 16333977, '-')},\n",
       "             'miR-200b': {('chr21', 16333970, 16333977, '-')},\n",
       "             'miR-429': {('chr21', 16333970, 16333977, '-')},\n",
       "             'miR-208b': {('chr21', 16334008, 16334015, '-')},\n",
       "             'miR-208a': {('chr21', 16334008, 16334015, '-')},\n",
       "             'miR-499a': {('chr21', 16334008, 16334015, '-')},\n",
       "             'miR-103a': {('chr21', 16334061, 16334068, '-')},\n",
       "             'miR-107': {('chr21', 16334061, 16334068, '-')},\n",
       "             'miR-33': {('chr21', 16334218, 16334225, '-')},\n",
       "             'miR-33b': {('chr21', 16334218, 16334225, '-')},\n",
       "             'miR-33a': {('chr21', 16334218, 16334225, '-')},\n",
       "             'miR-411': {('chr21', 16334303, 16334310, '-')},\n",
       "             'miR-379': {('chr21', 16334303, 16334310, '-')},\n",
       "             'miR-194': {('chr21', 16334304, 16334311, '-')},\n",
       "             'miR-193a': {('chr21', 16334991, 16334998, '-')},\n",
       "             'miR-193b': {('chr21', 16334991, 16334998, '-')},\n",
       "             'miR-4465': {('chr21', 16335187, 16335195, '-')},\n",
       "             'miR-26b': {('chr21', 16335187, 16335195, '-')},\n",
       "             'miR-26a': {('chr21', 16335187, 16335195, '-')},\n",
       "             'miR-1297': {('chr21', 16335187, 16335195, '-')},\n",
       "             'miR-490': {('chr21', 16335260, 16335268, '-')},\n",
       "             'miR-455': {('chr21', 16335470, 16335478, '-')},\n",
       "             'miR-5129': {('chr21', 16335470, 16335478, '-')},\n",
       "             'miR-6395': {('chr21', 16335645, 16335652, '-')},\n",
       "             'miR-328': {('chr21', 16335645, 16335652, '-')},\n",
       "             'miR-3118': {('chr21', 16335851, 16335859, '-')},\n",
       "             'miR-134': {('chr21', 16335851, 16335859, '-')},\n",
       "             'miR-27b': {('chr21', 16336786, 16336793, '-')},\n",
       "             'miR-27a': {('chr21', 16336786, 16336793, '-')},\n",
       "             'miR-199b': {('chr21', 16336837, 16336845, '-')},\n",
       "             'miR-199a': {('chr21', 16336837, 16336845, '-')},\n",
       "             'miR-3129': {('chr21', 16336837, 16336845, '-')},\n",
       "             'miR-670': {('chr21', 16336844, 16336852, '-')},\n",
       "             'miR-802': {('chr21', 16336899, 16336906, '-')}})"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene2mir2locations[\"NRIP1\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [],
   "source": [
    "ctable = {\"gene\": \"blueviolet\", \"mirna\": \"purple\", \"*\": \"black\" }\n",
    "\n",
    "def rowToExonGTF(row, exonID, padding, startOffset = 0, genomeName=None):\n",
    "    \n",
    "    startOffset = int(startOffset)-padding\n",
    "    \n",
    "    geneID = row[3]\n",
    "    geneName = row[5]\n",
    "    geneType = row[4]\n",
    "    \n",
    "    if genomeName is None:\n",
    "        genomeName = geneName\n",
    "    \n",
    "    geneType = \"gene\"\n",
    "    \n",
    "    entryStart = int(row[1])\n",
    "    entryEnd = int(row[2])\n",
    "        \n",
    "    if row[7] == \"gene\":\n",
    "        gtfEntry = (genomeName, \"MXPLORE\", row[7], entryStart-startOffset, entryEnd-startOffset, \".\", row[6], \".\", 'gene_id \"{geneName}\"; gene_type \"{geneType}\"; gene_name \"{geneName}\";'.format(geneName=geneName, geneType=geneType) )\n",
    "    elif row[7] == \"transcript\":\n",
    "        gtfEntry = (genomeName, \"MXPLORE\", row[7], entryStart-startOffset, entryEnd-startOffset, \".\", row[6], \".\", 'gene_id \"{geneName}\"; gene_type \"{geneType}\"; gene_name \"{geneName}\"; transcript_type \"{geneType}\"; transcript_name \"{geneName}\"; transcript_id \"{geneName}\";'.format(geneName=geneName, geneType=geneType) )\n",
    "    elif row[7] == \"mirna\":\n",
    "        geneType = \"mirna\"\n",
    "        gtfEntry = (genomeName, \"MXPLORE\", row[7], entryStart-startOffset, entryEnd-startOffset, \".\", row[6], \".\", 'gene_id \"{geneName}\"; gene_type \"{geneType}\"; gene_name \"{geneName}\"; transcript_type \"{geneType}\"; transcript_name \"{geneName}\"; transcript_id \"{geneName}\"; exon_id \"exon_{exonID}\"; exon_number {exonID};'.format(geneName=geneName, geneType=geneType, exonID=exonID) )\n",
    "    elif row[7] in [\"exon\", \"CDS\"]:\n",
    "        gtfEntry = (genomeName, \"MXPLORE\", row[7], entryStart-startOffset, entryEnd-startOffset, \".\", row[6], \".\", 'gene_id \"{geneName}\"; gene_type \"{geneType}\"; gene_name \"{geneName}\"; transcript_type \"{geneType}\"; transcript_name \"{geneName}\"; transcript_id \"{geneName}\"; exon_id \"exon_{exonID}\"; exon_number {exonID};'.format(geneName=geneName, geneType=geneType, exonID=exonID) )\n",
    "    \n",
    "    gtfEntry = list(gtfEntry)\n",
    "    gtfEntry.append(\" \")\n",
    "    \n",
    "    return \"\\t\".join([str(x) for x in gtfEntry])\n",
    "\n",
    "def create_annotation_for_gene(selected_gene, padding=100):\n",
    "\n",
    "    tree = IntervalTree()\n",
    "    geneDF = df.filter(pl.col(\"gene_name\") == selected_gene).filter(pl.col(\"feature\").is_in([\"gene\", \"exon\"])).select([\"seqname\", \"start\", \"end\", \"gene_id\", \"gene_type\", \"gene_name\", \"strand\", \"feature\"]).unique()\n",
    "\n",
    "    geneRow = None\n",
    "    for row in geneDF.iter_rows():\n",
    "        \n",
    "        row = list(row)\n",
    "        row[2] = row[2] + 1\n",
    "        row = tuple(row)\n",
    "        \n",
    "        if row[-1] == \"exon\":\n",
    "            tree.addi(row[1], row[2], row)\n",
    "            \n",
    "        elif row[-1] == \"gene\":\n",
    "            geneRow = row\n",
    "            \n",
    "    def add_set_elem(inset, elem):\n",
    "        inset.add(elem)\n",
    "        return inset\n",
    "\n",
    "    tree.merge_overlaps(data_reducer=lambda x,y: add_set_elem(x,y), data_initializer=set())\n",
    "\n",
    "    gene_io = StringIO(\"\")\n",
    "    #print(rowToExonGTF(geneRow, None, startOffset = geneRow[1]), file=gene_io)\n",
    "    #print(geneRow)\n",
    "\n",
    "    txRow = list(geneRow)\n",
    "    txRow[7] = \"transcript\"\n",
    "    print(rowToExonGTF(txRow, None, padding=padding, startOffset = geneRow[1]), file=gene_io)\n",
    "\n",
    "    treeIntervals = [x for x in tree]\n",
    "\n",
    "    if geneRow[6] == \"+\":\n",
    "        treeIntervals = sorted(treeIntervals, key=lambda x: x.begin)\n",
    "    else:\n",
    "        treeIntervals = sorted(treeIntervals, key=lambda x: x.end, reverse=True)\n",
    "\n",
    "    for i, intv in enumerate(treeIntervals):\n",
    "        dataElem = list(list(intv.data)[0]) \n",
    "        \n",
    "        dataElem[1] = intv.begin\n",
    "        dataElem[2] = intv.end\n",
    "        \n",
    "        print(rowToExonGTF(dataElem, i, padding=padding, startOffset = geneRow[1]), file=gene_io)\n",
    "\n",
    "    gene_io.seek(0)\n",
    "    geneStr = gene_io.read()\n",
    "\n",
    "\n",
    "    geneData = base64.b64encode(gzip.compress(bytes(geneStr, \"utf-8\")))\n",
    "\n",
    "    all_tracks = [\n",
    "        {\"name\": \"{} structure\".format(selected_gene), \"format\": \"gtf\", \"url\": \"data:application/gzip;base64,{}\".format(geneData.decode()),\"displayMode\":\"EXPANDED\",\"type\":\"annotation\",\"autoHeight\":\"true\", \"colorBy\": \"gene_type\", \"colorTable\": ctable},\n",
    "    ]\n",
    "\n",
    "\n",
    "    gene_sequence = all_fasta_seq[geneRow[0]][ (geneRow[1]-padding):(geneRow[2]+padding) ]\n",
    "    \n",
    "    if geneRow[6] == \"-\":\n",
    "        # complement sequence\n",
    "        rawSeq = Seq(gene_sequence)\n",
    "        newSeq = rawSeq.complement()\n",
    "        gene_sequence = str(newSeq)\n",
    "    \n",
    "    genomeSeq = \">{}\\n{}\".format(selected_gene, gene_sequence)\n",
    "    genomeData = base64.b64encode(gzip.compress(bytes(genomeSeq, \"utf-8\")))\n",
    "\n",
    "\n",
    "    idxName = selected_gene\n",
    "    idxLength = len(gene_sequence)\n",
    "    idxOffset = len(selected_gene) + 2\n",
    "    idxLineBases = len(gene_sequence)\n",
    "    idxLineWidth = idxLineBases\n",
    "\n",
    "    idxStr = \"{}\\t{}\\t{}\\t{}\\t{}\\n\".format(idxName, idxLength, idxOffset, idxLineBases, idxLineWidth)\n",
    "    idxData = base64.b64encode(gzip.compress(bytes(idxStr, \"utf-8\")))\n",
    "\n",
    "    outputjson = {\n",
    "        \"geneIdEnsembl\":selected_gene,\n",
    "        \"geneSymbolHgnc\":selected_gene,\n",
    "        \"genome\":\n",
    "            {\n",
    "                \"id\":selected_gene,\n",
    "                \"name\":selected_gene,\n",
    "                \"fastaURL\": \"data:application/gzip;base64,{}\".format(genomeData.decode()),\n",
    "                \"indexURL\":\"data:application/gzip;base64,{}\".format(idxData.decode())\n",
    "            },\n",
    "        \"tracks\": all_tracks,\n",
    "        \"startOffset\": geneRow[1]\n",
    "    }\n",
    "\n",
    "    return outputjson\n",
    "\n",
    "def create_tracks_mirna(selected_gene, selected_mirna, startOffset, padding):   \n",
    "    \n",
    "    mirnaBindingSites = gene2mir2locations.get(selected_gene, {}).get(selected_mirna, []) #{('chr2', 136872166, 136872174, '-')}\n",
    "\n",
    "    mirna_io = StringIO(\"\")\n",
    "\n",
    "    for i, bs in enumerate(mirnaBindingSites):\n",
    "        mirna_data = (bs[0], bs[1], bs[2], selected_mirna, 'mirna', selected_mirna, bs[3], 'mirna')\n",
    "        print(rowToExonGTF(mirna_data, i, padding=padding, startOffset = startOffset, genomeName=selected_gene), file=mirna_io)\n",
    "        \n",
    "    mirna_io.seek(0)\n",
    "    mirStr = mirna_io.read()\n",
    "    mirnaData = base64.b64encode(gzip.compress(bytes(mirStr, \"utf-8\")))\n",
    "    \n",
    "    mirTrack = {\"name\": \"{} binding sites\".format(selected_mirna), \"format\": \"gtf\", \"url\": \"data:application/gzip;base64,{}\".format(mirnaData.decode()),\"displayMode\":\"EXPANDED\",\"type\":\"annotation\",\"autoHeight\":\"true\", \"colorBy\": \"gene_type\", \"colorTable\": ctable}\n",
    "\n",
    "    return mirTrack"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "55765"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "allGenes = set(df[\"gene_name\"].unique())\n",
    "len(allGenes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "padding=100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gene index 0\n",
      "gene index 1000\n",
      "gene index 2000\n",
      "gene index 3000\n",
      "gene index 4000\n",
      "gene index 5000\n",
      "gene index 6000\n",
      "gene index 7000\n",
      "gene index 8000\n",
      "gene index 9000\n",
      "gene index 10000\n",
      "gene index 11000\n",
      "gene index 12000\n",
      "gene index 13000\n",
      "gene index 14000\n",
      "gene index 15000\n",
      "gene index 16000\n",
      "gene index 17000\n",
      "gene index 18000\n",
      "gene index 19000\n",
      "gene index 20000\n",
      "gene index 21000\n",
      "gene index 22000\n",
      "gene index 23000\n",
      "gene index 24000\n",
      "gene index 25000\n",
      "gene index 26000\n",
      "gene index 27000\n",
      "gene index 28000\n",
      "gene index 29000\n",
      "gene index 30000\n",
      "gene index 31000\n",
      "gene index 32000\n",
      "gene index 33000\n",
      "gene index 34000\n",
      "gene index 35000\n",
      "gene index 36000\n",
      "gene index 37000\n",
      "gene index 38000\n",
      "gene index 39000\n",
      "gene index 40000\n",
      "gene index 41000\n",
      "gene index 42000\n",
      "gene index 43000\n",
      "gene index 44000\n",
      "gene index 45000\n",
      "gene index 46000\n",
      "gene index 47000\n",
      "gene index 48000\n",
      "gene index 49000\n",
      "gene index 50000\n",
      "gene index 51000\n",
      "gene index 52000\n",
      "gene index 53000\n",
      "gene index 54000\n",
      "gene index 55000\n"
     ]
    }
   ],
   "source": [
    "allGeneAnnots = {}\n",
    "\n",
    "for i, gene in enumerate(allGenes):\n",
    "    \n",
    "    if i % 1000 == 0:\n",
    "        print(\"gene index\", i)\n",
    "    geneAnnot = create_annotation_for_gene(gene, padding=padding)\n",
    "    allGeneAnnots[gene] = geneAnnot\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pickle\n",
    "with open(\"hg19_gene_annotation.pickle\", \"wb\") as fout:\n",
    "    pickle.dump(allGeneAnnots, fout)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>gene_mir</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>start2</th>\n",
       "      <th>end2</th>\n",
       "      <th>score8</th>\n",
       "      <th>score9</th>\n",
       "      <th>score10</th>\n",
       "      <th>score11</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>900599</td>\n",
       "      <td>900606</td>\n",
       "      <td>KLHL17:miR-299-3p</td>\n",
       "      <td>86</td>\n",
       "      <td>+</td>\n",
       "      <td>900599</td>\n",
       "      <td>900606</td>\n",
       "      <td>255,0,0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>900605</td>\n",
       "      <td>900612</td>\n",
       "      <td>KLHL17:miR-124-3p.1</td>\n",
       "      <td>88</td>\n",
       "      <td>+</td>\n",
       "      <td>900605</td>\n",
       "      <td>900612</td>\n",
       "      <td>30,144,255</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>900605</td>\n",
       "      <td>900613</td>\n",
       "      <td>KLHL17:miR-124-3p.2/506-3p</td>\n",
       "      <td>96</td>\n",
       "      <td>+</td>\n",
       "      <td>900605</td>\n",
       "      <td>900613</td>\n",
       "      <td>128,0,128</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>900933</td>\n",
       "      <td>900941</td>\n",
       "      <td>KLHL17:miR-19-3p</td>\n",
       "      <td>91</td>\n",
       "      <td>+</td>\n",
       "      <td>900933</td>\n",
       "      <td>900941</td>\n",
       "      <td>128,0,128</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>901054</td>\n",
       "      <td>901061</td>\n",
       "      <td>KLHL17:miR-137</td>\n",
       "      <td>84</td>\n",
       "      <td>+</td>\n",
       "      <td>901054</td>\n",
       "      <td>901061</td>\n",
       "      <td>255,0,0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr   start     end                    gene_mir  score strand  start2  \\\n",
       "0  chr1  900599  900606           KLHL17:miR-299-3p     86      +  900599   \n",
       "1  chr1  900605  900612         KLHL17:miR-124-3p.1     88      +  900605   \n",
       "2  chr1  900605  900613  KLHL17:miR-124-3p.2/506-3p     96      +  900605   \n",
       "3  chr1  900933  900941            KLHL17:miR-19-3p     91      +  900933   \n",
       "4  chr1  901054  901061              KLHL17:miR-137     84      +  901054   \n",
       "\n",
       "     end2      score8  score9 score10 score11  \n",
       "0  900606     255,0,0       1       7       0  \n",
       "1  900612  30,144,255       1       7       0  \n",
       "2  900613   128,0,128       1       8       0  \n",
       "3  900941   128,0,128       1       8       0  \n",
       "4  901061     255,0,0       1       7       0  "
      ]
     },
     "execution_count": 143,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mirBSDF.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'name': 'miR-299a binding sites', 'format': 'gtf', 'url': 'data:application/gzip;base64,H4sIAF0LYmYC//P28fAxNOf0jQjw8Q9y5czNLMpL5DQxNzYCEZacepzaQJyempcan5mioJSbGaRrZGmZqGStABYrqSxIBYkCNcGE8hJzU1EUlhQl5hUnF2UWlKApR5LApwnN3tSK/DywGJhhABPJK81NSi1SMLDmVOACACRCsGzUAAAA', 'displayMode': 'EXPANDED', 'type': 'annotation', 'autoHeight': 'true', 'colorBy': 'gene_type', 'colorTable': {'gene': 'blueviolet', 'mirna': 'purple', '*': 'black'}}\n",
      "{'name': 'miR-299b binding sites', 'format': 'gtf', 'url': 'data:application/gzip;base64,H4sIAF0LYmYC//P28fAxNOf0jQjw8Q9y5czNLMpL5DQxNzYCEZacepzaQJyempcan5mioJSbGaRrZGmZpGStABYrqSxIBYkCNcGE8hJzU1EUlhQl5hUnF2UWlKApR5LApwnN3tSK/DywGJhhABPJK81NSi1SMLDmVOACAB98rurUAAAA', 'displayMode': 'EXPANDED', 'type': 'annotation', 'autoHeight': 'true', 'colorBy': 'gene_type', 'colorTable': {'gene': 'blueviolet', 'mirna': 'purple', '*': 'black'}}\n",
      "{'name': 'miR-299 binding sites', 'format': 'gtf', 'url': 'data:application/gzip;base64,H4sIAF0LYmYC//P28fAxNOf0jQjw8Q9y5czNLMpL5DQxNzYCEZacepzaQJyempcan5mioJSbGaRrZGmpZK0AFiqpLEgFCQL1wITyEnNTkdWVFCXmFScXZRaUoKlGksCjB9XS1Ir8PLAQmGEAE8krzU1KLVIwsOZU4AIAte6Y/tAAAAA=', 'displayMode': 'EXPANDED', 'type': 'annotation', 'autoHeight': 'true', 'colorBy': 'gene_type', 'colorTable': {'gene': 'blueviolet', 'mirna': 'purple', '*': 'black'}}\n",
      "{'name': 'miR-124 binding sites', 'format': 'gtf', 'url': 'data:application/gzip;base64,H4sIAF0LYmYC//P28fAxNOf0jQjw8Q9y5czNLMpL5DQxN7YAEiamnHqc2kCcnpqXGp+ZoqCUmxmka2hkomStABYqqSxIBQkC9cCE8hJzU5HVlRQl5hUnF2UWlKCpRpLAowfV0tSK/DywEJhhABPJK81NSi1SMLDmVODyxucfs0HtH0M0/xiC/AMAqIGyBKABAAA=', 'displayMode': 'EXPANDED', 'type': 'annotation', 'autoHeight': 'true', 'colorBy': 'gene_type', 'colorTable': {'gene': 'blueviolet', 'mirna': 'purple', '*': 'black'}}\n",
      "{'name': 'miR-506 binding sites', 'format': 'gtf', 'url': 'data:application/gzip;base64,H4sIAF0LYmYC//P28fAxNOf0jQjw8Q9y5czNLMpL5DQxN7YAEiZmnHqc2kCcnpqXGp+ZoqCUmxmka2pgpmStABYqqSxIBQkC9cCE8hJzU5HVlRQl5hUnF2UWlKCpRpLAowfV0tSK/DywEJhhABPJK81NSi1SMLDmVOACAFJWjpjQAAAA', 'displayMode': 'EXPANDED', 'type': 'annotation', 'autoHeight': 'true', 'colorBy': 'gene_type', 'colorTable': {'gene': 'blueviolet', 'mirna': 'purple', '*': 'black'}}\n",
      "{'name': 'miR-19b binding sites', 'format': 'gtf', 'url': 'data:application/gzip;base64,H4sIAF0LYmYC//P28fAxNOf0jQjw8Q9y5czNLMpL5DQ1MDMDEuYmnHqc2kCcnpqXGp+ZoqCUmxmka2iZpGStABYqqSxIBQkC9cCE8hJzU5HVlRQl5hUnF2UWlKCpRpLAowfV0tSK/DywEJhhABPJK81NSi1SMLDmVOACAI6mc6zQAAAA', 'displayMode': 'EXPANDED', 'type': 'annotation', 'autoHeight': 'true', 'colorBy': 'gene_type', 'colorTable': {'gene': 'blueviolet', 'mirna': 'purple', '*': 'black'}}\n",
      "{'name': 'miR-19a binding sites', 'format': 'gtf', 'url': 'data:application/gzip;base64,H4sIAF0LYmYC//P28fAxNOf0jQjw8Q9y5czNLMpL5DQ1MDMDEuYmnHqc2kCcnpqXGp+ZoqCUmxmka2iZqGStABYqqSxIBQkC9cCE8hJzU5HVlRQl5hUnF2UWlKCpRpLAowfV0tSK/DywEJhhABPJK81NSi1SMLDmVOACAH8ziBDQAAAA', 'displayMode': 'EXPANDED', 'type': 'annotation', 'autoHeight': 'true', 'colorBy': 'gene_type', 'colorTable': {'gene': 'blueviolet', 'mirna': 'purple', '*': 'black'}}\n",
      "{'name': 'miR-137 binding sites', 'format': 'gtf', 'url': 'data:application/gzip;base64,H4sIAF0LYmYC//P28fAxNOf0jQjw8Q9y5czNLMpL5DQ1tDAHEpYmnHqc2kCcnpqXGp+ZoqCUmxmka2hsrmStABYqqSxIBQkC9cCE8hJzU5HVlRQl5hUnF2UWlKCpRpLAowfV0tSK/DywEJhhABPJK81NSi1SMLDmVOACAB2UyabQAAAA', 'displayMode': 'EXPANDED', 'type': 'annotation', 'autoHeight': 'true', 'colorBy': 'gene_type', 'colorTable': {'gene': 'blueviolet', 'mirna': 'purple', '*': 'black'}}\n"
     ]
    }
   ],
   "source": [
    "allMirnaAnnots = dict()\n",
    "\n",
    "for ri, row in mirBSDF.iterrows():\n",
    "    \n",
    "    gene_mir = row[\"gene_mir\"]\n",
    "    gene, mir = gene_mir.split(\":\",1)\n",
    "    \n",
    "    for mir in gene2mir2locations[gene]:\n",
    "        \n",
    "        startOffset = allGeneAnnots.get(gene, {}).get(\"startOffset\", -1)\n",
    "        \n",
    "        if startOffset < 0:\n",
    "            continue\n",
    "        \n",
    "        mirAnnot = create_tracks_mirna(gene, mir, startOffset, padding)\n",
    "        \n",
    "        if not gene in allMirnaAnnots:\n",
    "            allMirnaAnnots[gene] = dict()\n",
    "        \n",
    "        allMirnaAnnots[gene][mir] = mirAnnot\n",
    "        \n",
    "        if ri == 0:\n",
    "            print(mirAnnot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pickle\n",
    "with open(\"hg19_mirna_annotation.pickle\", \"wb\") as fout:\n",
    "    pickle.dump(allMirnaAnnots, fout)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'name': 'miR-124 binding sites',\n",
       " 'format': 'gtf',\n",
       " 'url': 'data:application/gzip;base64,H4sIAIULYmYC/3N29jHi9I0I8PEPcuXMzSzKS+Q0NDc1ARJmRpx6nNpAnJ6alxqfmaKglJsZpGtoZKJkrQAWKqksSAUJAvXAhPISc1OR1ZUUJeYVJxdlFpSgqUaSwKMH1dLUivw8sBCYYQATySvNTUotUjCw5lTgAgCCJKf1zgAAAA==',\n",
       " 'displayMode': 'EXPANDED',\n",
       " 'type': 'annotation',\n",
       " 'autoHeight': 'true',\n",
       " 'colorBy': 'gene_type',\n",
       " 'colorTable': {'gene': 'blueviolet', 'mirna': 'purple', '*': 'black'}}"
      ]
     },
     "execution_count": 146,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "allMirnaAnnots[\"CCL2\"][\"miR-124\"]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
